{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "81c02b39",
   "metadata": {},
   "outputs": [],
   "source": [
    "# load required packages\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import networkx\n",
    "import networkx as nx\n",
    "import pandas as pd\n",
    "import re\n",
    "import random\n",
    "from ndex2.cx2 import CX2Network\n",
    "\n",
    "from IPython.display import display\n",
    "\n",
    "import getpass\n",
    "import ndex2\n",
    "\n",
    "import json\n",
    "import cdapsutil\n",
    "\n",
    "#from gprofiler import GProfiler\n",
    "#gp = GProfiler(\"MyToolName/0.1\")\n",
    "\n",
    "from scipy.stats import hypergeom\n",
    "from scipy.stats import norm\n",
    "\n",
    "# latex rendering of text in graphs\n",
    "import matplotlib as mpl\n",
    "mpl.rc('text', usetex = False)\n",
    "mpl.rc('font', family = 'serif')\n",
    "\n",
    "from matplotlib import rcParams\n",
    "rcParams['font.family'] = 'sans-serif'\n",
    "rcParams['font.sans-serif'] = ['Arial']\n",
    "\n",
    "sns.set(font_scale=1.4)\n",
    "\n",
    "sns.set_style('white')\n",
    "\n",
    "sns.set_style(\"ticks\", {\"xtick.major.size\": 15, \"ytick.major.size\": 15})\n",
    "plt.rcParams['svg.fonttype'] = 'none'\n",
    "\n",
    "from datetime import datetime\n",
    "import sys\n",
    "%matplotlib inline\n",
    "import networkx"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "71b99a6a",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tscc/nfs/home/hiyoothere/miniconda3/envs/r4/lib/python3.12/site-packages/netcoloc/validation.py:20: UserWarning: Import of ddot failed. NetColoc will not work properly: + cannot import name 'Ontology' from 'ddot' (/tscc/nfs/home/hiyoothere/miniconda3/envs/r4/lib/python3.12/site-packages/ddot/__init__.py)\n",
      "  warnings.warn('Import of ddot failed. NetColoc will not work properly: + ' + str(ie))\n"
     ]
    }
   ],
   "source": [
    "# verify DDOT was installed\n",
    "import ddot\n",
    "\n",
    "from netcoloc import netprop_zscore, netprop, network_colocalization, validation\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e34d974c-ce06-4442-a47f-f5388a53724e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "4234e004",
   "metadata": {},
   "source": [
    "## 2. Select two gene sets of interest. Load gene sets from text files into python."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "ca67b020",
   "metadata": {},
   "outputs": [],
   "source": [
    "d1_name='MM DNM gens'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "906f878d",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Get alias gene \n",
    "f_alias = open(\"/tscc/projects/ps-gleesonlab7/gleeson3/resources/HGNC/Gene_SYNONYM.txt\", 'r')\n",
    "dic_alias = {} #alias:general\n",
    "\n",
    "for line in f_alias:\n",
    "    if \"Approved\" not in line:\n",
    "        s = line.strip().split(\"\\t\")\n",
    "        #print (len(s))\n",
    "        approved = s[0]\n",
    "        if len(s) ==2:\n",
    "            if ',' in s[1]:\n",
    "                l_alias = s[1].split(',')\n",
    "                for i in l_alias:\n",
    "                    dic_alias[i.strip()] = approved\n",
    "            else:\n",
    "                dic_alias[s[1]] = approved\n",
    "        elif len(s) == 3:\n",
    "            if ',' in s[1]:\n",
    "                l_alias = s[1].split(',')\n",
    "                for i in l_alias:\n",
    "                    \n",
    "                    dic_alias[i.strip()] = approved\n",
    "            else:\n",
    "                dic_alias[s[1]] = approved\n",
    "            if ',' in s[2]:\n",
    "                l_previous = s[2].split(',')\n",
    "\n",
    "                for i in l_previous:\n",
    "                    dic_alias[i.strip()] = approved\n",
    "            else:\n",
    "                dic_alias[s[2]] = approved\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "396756c5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of MM DNM gens genes: 187\n",
      "First 5 genes: ['GRIA3', 'SLC30A6', 'RXFP2', 'FAM83G', 'ZSWIM6']\n"
     ]
    }
   ],
   "source": [
    "# ------ customize this section based your input genesets -------\n",
    "\n",
    "D1_df = pd.read_csv('/tscc/projects/ps-gleesonlab8/User/hiyoothere/NTD/5.Analysis/DNM/Burden/NTD_DamMC2_woINVALID.den_input',sep='\\t')\n",
    "D1_genes = D1_df['SYMBOL'].tolist()\n",
    "#Remove duplicates\n",
    "D1_genes = list(set(D1_genes))\n",
    "\n",
    "        \n",
    "print('Number of '+d1_name+' genes:',len(D1_genes))\n",
    "print('First 5 genes:', D1_genes[0:5])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0c878ed7",
   "metadata": {},
   "source": [
    "# Output the overlap between the two gene sets\n",
    "print('Number of '+d1_name+' and '+d2_name+' genes overlapping:', len(np.intersect1d(D1_genes,D2_genes)))\n",
    "\n",
    "# Create a dictionary for each geneset\n",
    "seed_dict = {d1_name:D1_genes,d2_name:D2_genes}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c7991d4e",
   "metadata": {},
   "source": [
    "## 3. Select gene interaction network to use for the analysis."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "f89c0864",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of nodes: 18820\n",
      "\n",
      "Number of edges: 2693109\n"
     ]
    }
   ],
   "source": [
    "interactome_uuid='4de852d9-9908-11e9-bcaf-0ac135e8bacf' # for PCNet\n",
    "ndex_server='public.ndexbio.org'\n",
    "ndex_user=None\n",
    "ndex_password=None\n",
    "G_int = ndex2.create_nice_cx_from_server(\n",
    "            ndex_server, \n",
    "            username=ndex_user, \n",
    "            password=ndex_password, \n",
    "            uuid=interactome_uuid\n",
    "        ).to_networkx()\n",
    "nodes = list(G_int.nodes)\n",
    "\n",
    "# remove self edges from network\n",
    "G_int.remove_edges_from(nx.selfloop_edges(G_int))\n",
    "\n",
    "# print out the numbers of nodes and edges in the interatome for diagnostic purposes:\n",
    "print('Number of nodes:', len(G_int.nodes))\n",
    "print('\\nNumber of edges:', len(G_int.edges))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "e3c0f8b1",
   "metadata": {},
   "outputs": [],
   "source": [
    "int_nodes = list(G_int.nodes)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bf57167e",
   "metadata": {},
   "source": [
    "# Identify network colocalized gene network"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d2d13f0e",
   "metadata": {},
   "source": [
    "## 4. Precalculate matrices needed for propagation. This step should take a few minutes (more for larger/denser networks)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "40749cc5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "calculating w_prime\n",
      "\n",
      "calculating w_double_prime\n"
     ]
    }
   ],
   "source": [
    "# pre-calculate matrices used for network propagation. this step takes a few minutes, more for denser interactomes\n",
    "print('\\ncalculating w_prime')\n",
    "w_prime = netprop.get_normalized_adjacency_matrix(G_int, conserve_heat=True)\n",
    "\n",
    "print('\\ncalculating w_double_prime')\n",
    "w_double_prime = netprop.get_individual_heats_matrix(w_prime, .5)\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7a3ed731",
   "metadata": {},
   "source": [
    "## 5. Subset seed genes to those found in the selected network."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "f51078a3",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LAMA3\n",
      "H3-4\n",
      "SLCO1B3-SLCO1B7\n",
      "MGA\n",
      "MAP2\n",
      "POP1\n"
     ]
    }
   ],
   "source": [
    "for MM in D1_genes:\n",
    "    if MM not in int_nodes:\n",
    "        print (MM)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "91994a17",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of D1 genes: 187\n",
      "Not present H3-4\n",
      "Not present SLCO1B3-SLCO1B7\n",
      "Number of D1 genes in interactome: 185\n"
     ]
    }
   ],
   "source": [
    "# subset seed genes to those found in interactome\n",
    "print(\"Number of D1 genes:\", len(D1_genes))\n",
    "\n",
    "\n",
    "D1_new = []\n",
    "for MM in D1_genes:\n",
    "    if MM in int_nodes:\n",
    "        D1_new.append(MM)\n",
    "    if MM not in int_nodes:\n",
    "        if MM in dic_alias:\n",
    "            #print (MM, dic_alias[MM])\n",
    "            D1_new.append(MM)\n",
    "        else:\n",
    "            print (\"Not present\", MM)\n",
    "        \n",
    "D1_genes = D1_new\n",
    "print(\"Number of D1 genes in interactome:\", len(D1_genes))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3e77b2a5",
   "metadata": {},
   "source": [
    "##  6. Compute network proximity scores from seed gene set.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "88bc29a9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Calculating D1 z-scores: \n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "0f74344f83f746309724e921fbc9ae95",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "  0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>z</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CLTC</th>\n",
       "      <td>21.341222</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HNRNPL</th>\n",
       "      <td>19.767824</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VDAC2</th>\n",
       "      <td>19.187881</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CNPY1</th>\n",
       "      <td>19.097940</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FAM83G</th>\n",
       "      <td>17.947948</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                z\n",
       "CLTC    21.341222\n",
       "HNRNPL  19.767824\n",
       "VDAC2   19.187881\n",
       "CNPY1   19.097940\n",
       "FAM83G  17.947948"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# D1 network propagation\n",
    "print('\\nCalculating D1 z-scores: ')\n",
    "z_D1, Fnew_D1, Fnew_rand_D1 = netprop_zscore.calculate_heat_zscores(w_double_prime, int_nodes, \n",
    "                                                                    dict(G_int.degree), \n",
    "                                                                    D1_genes, num_reps=1000,\n",
    "                                                                    minimum_bin_size=100)\n",
    "\n",
    "z_D1 = pd.DataFrame({'z':z_D1})\n",
    "\n",
    "z_D1.sort_values('z',ascending=False).head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a34d941d",
   "metadata": {},
   "source": [
    "## 7. Build proximal subnetwork by taking z> thresh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "9b33a657",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Nodes in proximal subgraph: 439\n",
      "Edges in proximal subgraph: 2447\n",
      "          d1_seeds      d1_name       z_d1\n",
      "CLTC             1  MM DNM gens  21.341222\n",
      "HNRNPL           1  MM DNM gens  19.767824\n",
      "VDAC2            1  MM DNM gens  19.187881\n",
      "CNPY1            1  MM DNM gens  19.097940\n",
      "FAM83G           1  MM DNM gens  17.947948\n",
      "GTF3C3           1  MM DNM gens  17.852848\n",
      "MFAP1            1  MM DNM gens  17.846594\n",
      "DPYSL2           1  MM DNM gens  17.695655\n",
      "LEMD3            1  MM DNM gens  16.780574\n",
      "TOP2B            1  MM DNM gens  16.752226\n",
      "C6orf132         1  MM DNM gens  16.370847\n",
      "CRHR2            1  MM DNM gens  16.271097\n",
      "SMURF2           1  MM DNM gens  16.069396\n",
      "ATAD3B           1  MM DNM gens  15.869375\n",
      "SPEN             1  MM DNM gens  15.622895\n"
     ]
    }
   ],
   "source": [
    "\n",
    "# ----------- select thresholds for NetColoc -----------------\n",
    "zthresh=3 # default = 3\n",
    "\n",
    "# select the genes in the network intersection, make a subgraph\n",
    "\n",
    "G_prox = nx.subgraph(G_int,z_D1[z_D1['z']>zthresh].index.tolist()) \n",
    "print(\"Nodes in proximal subgraph:\", len(G_prox.nodes()))\n",
    "print(\"Edges in proximal subgraph:\", len(G_prox.edges()))\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "# compile dataframe of metadata for overlapping nodes\n",
    "node_df = pd.DataFrame(index=list(G_prox.nodes))\n",
    "node_df = node_df.assign(d1_seeds=0, d1_name=d1_name)\n",
    "node_df.head()\n",
    "node_df.loc[list(np.intersect1d(node_df['d1_seeds'].keys() ,D1_genes)), 'd1_seeds']=1\n",
    "node_df['z_d1']=z_D1.loc[list(G_prox.nodes)]['z']\n",
    "#node_df['z_d2']=z_D2.loc[list(G_overlap.nodes)]['z']\n",
    "#node_df['z_both']=node_df['z_d1']*node_df['z_d2']\n",
    "#node_df['sum_seeds']=node_df['d1_seeds']+node_df['d2_seeds']\n",
    "\n",
    "node_df = node_df.sort_values('z_d1',ascending=False)\n",
    "print (node_df.head(15))\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e51de6c6",
   "metadata": {},
   "source": [
    "# ** Hypergeometric test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "8645edba",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of Mouse NTD genes: 374\n",
      "First 5 genes: ['EFNA5', 'EPHA7', 'FREM1', 'FREM2', 'GJA1']\n",
      "Number of DNM NTD genes: 185\n",
      "First 5 genes: ['GRIA3', 'SLC30A6', 'RXFP2', 'FAM83G', 'ZSWIM6'] \n",
      "\n",
      "\n",
      "Number of overalps of DNM NTD genes and Mouse NTD genes: 11 \n",
      "\n",
      "THOSE genes:: ['CELSR1' 'EP300' 'FREM1' 'GRHL2' 'GRHL3' 'MIB2' 'PAX3' 'SCRIB' 'SMURF2'\n",
      " 'TBXT' 'TIAM1']\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Number of Prop_genes: 439\n",
      "First 5 genes: Index(['CLTC', 'HNRNPL', 'VDAC2', 'CNPY1', 'FAM83G'], dtype='object')\n",
      "Number of overalps of Prop genes genes and Mouse NTD genes: 17 \n",
      "\n",
      "THOSE genes:: ['BMP7' 'CELSR1' 'DYNC2H1' 'EP300' 'FREM1' 'FREM2' 'GRHL2' 'GRHL3' 'MIB2'\n",
      " 'PAX3' 'SCRIB' 'SLC25A19' 'SMURF2' 'SS18' 'TBXT' 'TIAM1' 'TULP3']\n",
      "\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#1. Mouse gene and the DNM gene\n",
    "MO_df = pd.read_csv('/tscc/projects/ps-gleesonlab8/User/hiyoothere/NTD/12.Network_propagation/MouseNTDgene/raw_221017.f.HumanConv.txt',sep='\\t', encoding = \"cp1252\")\n",
    "MO_genes = MO_df['gene'].tolist()\n",
    "print('Number of Mouse NTD genes:',len(MO_genes))\n",
    "print('First 5 genes:', MO_genes[0:5])\n",
    "\n",
    "print('Number of DNM NTD genes:',len(D1_genes))\n",
    "print('First 5 genes:', D1_genes[0:5], '\\n\\n')\n",
    "\n",
    "\n",
    "overlap_DNM = np.intersect1d(D1_genes,MO_genes)\n",
    "print('Number of overalps of DNM NTD genes and Mouse NTD genes:',len(overlap_DNM), '\\n')\n",
    "print('THOSE genes::', overlap_DNM)\n",
    "cnt_overlap_DNM = len(overlap_DNM)\n",
    "print (\"\\n\\n\\n\")\n",
    "\n",
    "#2. Mouse gene and propagated genes\n",
    "\n",
    "Prop_genes = node_df['d1_seeds'].keys()\n",
    "print('Number of Prop_genes:',len(Prop_genes))\n",
    "print('First 5 genes:', Prop_genes[0:5])\n",
    "\n",
    "\n",
    "\n",
    "overlap_prop = np.intersect1d(MO_genes,Prop_genes)\n",
    "print('Number of overalps of Prop genes genes and Mouse NTD genes:',len(overlap_prop), '\\n')\n",
    "print('THOSE genes::', overlap_prop)\n",
    "cnt_overlap_prop = len(overlap_prop)\n",
    "print (\"\\n\\n\\n\")\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "d415cca7",
   "metadata": {},
   "outputs": [],
   "source": [
    "from scipy.stats import hypergeom\n",
    "import matplotlib.pyplot as plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "659bcf49",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([1.34076912e-004, 1.22243180e-003, 5.54482682e-003, 1.66831247e-002,\n",
       "       3.74576026e-002, 6.69420512e-002, 9.91922661e-002, 1.25344513e-001,\n",
       "       1.37889401e-001, 1.34149284e-001, 1.16860355e-001, 9.20717947e-002,\n",
       "       6.61553653e-002, 4.36519228e-002, 2.66080628e-002, 1.50594962e-002,\n",
       "       7.94920474e-003, 3.92868290e-003, 1.82423114e-003, 7.98286880e-004,\n",
       "       3.30128623e-004, 1.29340228e-004, 4.81160856e-005, 1.70312077e-005,\n",
       "       5.74665119e-006, 1.85159957e-006, 5.70599659e-007, 1.68424387e-007,\n",
       "       4.76823320e-008, 1.29639451e-008, 3.38886739e-009, 8.52680339e-010,\n",
       "       2.06717905e-010, 4.83334880e-011, 1.09091026e-011, 2.37886745e-012,\n",
       "       5.01579822e-013, 1.02335723e-013, 2.02182114e-014, 3.87062255e-015,\n",
       "       7.18489806e-016, 1.29397842e-016, 2.26230890e-017, 3.84180072e-018,\n",
       "       6.34021624e-019, 1.01736611e-019, 1.58804269e-020, 2.41244417e-021,\n",
       "       3.56822606e-022, 5.14078087e-023, 7.21707897e-024, 9.87679965e-025,\n",
       "       1.31811545e-025, 1.71603710e-026, 2.18013535e-027, 2.70374442e-028,\n",
       "       3.27423341e-029, 3.87299333e-030, 4.47613746e-031, 5.05593469e-032,\n",
       "       5.58288023e-033, 6.02819444e-034, 6.36645366e-035, 6.57801447e-036,\n",
       "       6.65090736e-037, 6.58194555e-038, 6.37690846e-039, 6.04979489e-040,\n",
       "       5.62127390e-041, 5.11656693e-042, 4.56305628e-043, 3.98792557e-044,\n",
       "       3.41610012e-045, 2.86868136e-046, 2.36197734e-047, 1.90713799e-048,\n",
       "       1.51032571e-049, 1.17329786e-050, 8.94252544e-052, 6.68789551e-053,\n",
       "       4.90859805e-054, 3.53609302e-055, 2.50060973e-056, 1.73612234e-057,\n",
       "       1.18353595e-058, 7.92323065e-060, 5.20947993e-061, 3.36440077e-062,\n",
       "       2.13447476e-063, 1.33042862e-064, 8.14808195e-066, 4.90373711e-067,\n",
       "       2.90035028e-068, 1.68603859e-069, 9.63428540e-071, 5.41185894e-072,\n",
       "       2.98873445e-073, 1.62285594e-074, 8.66485382e-076, 4.54953480e-077,\n",
       "       2.34926550e-078, 1.19313752e-079, 5.96039729e-081, 2.92899303e-082,\n",
       "       1.41596050e-083, 6.73447895e-085, 3.15142150e-086, 1.45106672e-087,\n",
       "       6.57468018e-089, 2.93153749e-090, 1.28640078e-091, 5.55575246e-093,\n",
       "       2.36167554e-094, 9.88172216e-096, 4.07008739e-097, 1.65027559e-098,\n",
       "       6.58738690e-100, 2.58878203e-101, 1.00166952e-102, 3.81610677e-104,\n",
       "       1.43154003e-105, 5.28801444e-107, 1.92356379e-108, 6.89070221e-110,\n",
       "       2.43097696e-111, 8.44647851e-113, 2.89044340e-114, 9.74233718e-116,\n",
       "       3.23435119e-117, 1.05767308e-118, 3.40698660e-120, 1.08108051e-121,\n",
       "       3.37930784e-123, 1.04062097e-124, 3.15693050e-126, 9.43533167e-128,\n",
       "       2.77830965e-129, 8.06021503e-131, 2.30391185e-132, 6.48856485e-134,\n",
       "       1.80054878e-135, 4.92316118e-137, 1.32640208e-138, 3.52133191e-140,\n",
       "       9.21186583e-142, 2.37467932e-143, 6.03234984e-145, 1.51007780e-146,\n",
       "       3.72520401e-148, 9.05618003e-150, 2.16965265e-151, 5.12260787e-153,\n",
       "       1.19193603e-154, 2.73325747e-156, 6.17701094e-158, 1.37578491e-159,\n",
       "       3.01995739e-161, 6.53329997e-163, 1.39299306e-164, 2.92720130e-166,\n",
       "       6.06241806e-168, 1.23745912e-169, 2.48948332e-171, 4.93608147e-173,\n",
       "       9.64608712e-175, 1.85787502e-176, 3.52678212e-178, 6.59838448e-180,\n",
       "       1.21672706e-181, 2.21128469e-183, 3.96087516e-185, 6.99248258e-187,\n",
       "       1.21664476e-188, 2.08634650e-190, 3.52612023e-192, 5.87345010e-194,\n",
       "       9.64210997e-196, 1.56002059e-197, 2.48750831e-199, 3.90904708e-201,\n",
       "       6.05404003e-203, 9.24024022e-205, 1.38988883e-206, 2.06030079e-208,\n",
       "       3.00974405e-210, 4.33282522e-212, 6.14680488e-214, 8.59326097e-216,\n",
       "       1.18383245e-217, 1.60708620e-219, 2.14979730e-221, 2.83372470e-223,\n",
       "       3.68054111e-225, 4.71032981e-227, 5.93974799e-229, 7.37994096e-231,\n",
       "       9.03434568e-233, 1.08965729e-234, 1.29485961e-236, 1.51594966e-238,\n",
       "       1.74850019e-240, 1.98680174e-242, 2.22402736e-244, 2.45251084e-246,\n",
       "       2.66412800e-248, 2.85075810e-250, 3.00479067e-252, 3.11963473e-254,\n",
       "       3.19018370e-256, 3.21319037e-258, 3.18751387e-260, 3.11421262e-262,\n",
       "       2.99647263e-264, 2.83937799e-266, 2.64954649e-268, 2.43466686e-270,\n",
       "       2.20298256e-272, 1.96277071e-274, 1.72186130e-276, 1.48723486e-278,\n",
       "       1.26472488e-280, 1.05883821e-282, 8.72693434e-285, 7.08065605e-287,\n",
       "       5.65517030e-289, 4.44588860e-291, 3.44026619e-293, 2.62014640e-295,\n",
       "       1.96398488e-297, 1.44879982e-299, 1.05175467e-301, 7.51335716e-304,\n",
       "       5.28134330e-306, 3.65277005e-308, 2.48567398e-310, 1.66412425e-312,\n",
       "       1.09603003e-314, 7.10115810e-317, 4.52564132e-319, 2.83593681e-321,\n",
       "       1.97626258e-323, 0.00000000e+000, 0.00000000e+000, 0.00000000e+000,\n",
       "       0.00000000e+000, 0.00000000e+000, 0.00000000e+000, 0.00000000e+000,\n",
       "       0.00000000e+000, 0.00000000e+000, 0.00000000e+000, 0.00000000e+000,\n",
       "       0.00000000e+000, 0.00000000e+000, 0.00000000e+000, 0.00000000e+000,\n",
       "       0.00000000e+000, 0.00000000e+000, 0.00000000e+000, 0.00000000e+000,\n",
       "       0.00000000e+000, 0.00000000e+000, 0.00000000e+000, 0.00000000e+000,\n",
       "       0.00000000e+000, 0.00000000e+000, 0.00000000e+000, 0.00000000e+000,\n",
       "       0.00000000e+000, 0.00000000e+000, 0.00000000e+000, 0.00000000e+000,\n",
       "       0.00000000e+000, 0.00000000e+000, 0.00000000e+000, 0.00000000e+000,\n",
       "       0.00000000e+000, 0.00000000e+000, 0.00000000e+000, 0.00000000e+000,\n",
       "       0.00000000e+000, 0.00000000e+000, 0.00000000e+000, 0.00000000e+000,\n",
       "       0.00000000e+000, 0.00000000e+000, 0.00000000e+000, 0.00000000e+000,\n",
       "       0.00000000e+000, 0.00000000e+000, 0.00000000e+000, 0.00000000e+000,\n",
       "       0.00000000e+000, 0.00000000e+000, 0.00000000e+000, 0.00000000e+000,\n",
       "       0.00000000e+000, 0.00000000e+000, 0.00000000e+000, 0.00000000e+000,\n",
       "       0.00000000e+000, 0.00000000e+000, 0.00000000e+000, 0.00000000e+000,\n",
       "       0.00000000e+000, 0.00000000e+000, 0.00000000e+000, 0.00000000e+000,\n",
       "       0.00000000e+000, 0.00000000e+000, 0.00000000e+000, 0.00000000e+000,\n",
       "       0.00000000e+000, 0.00000000e+000, 0.00000000e+000, 0.00000000e+000,\n",
       "       0.00000000e+000, 0.00000000e+000, 0.00000000e+000, 0.00000000e+000,\n",
       "       0.00000000e+000, 0.00000000e+000, 0.00000000e+000, 0.00000000e+000,\n",
       "       0.00000000e+000, 0.00000000e+000, 0.00000000e+000, 0.00000000e+000,\n",
       "       0.00000000e+000, 0.00000000e+000, 0.00000000e+000, 0.00000000e+000,\n",
       "       0.00000000e+000, 0.00000000e+000, 0.00000000e+000, 0.00000000e+000,\n",
       "       0.00000000e+000, 0.00000000e+000, 0.00000000e+000, 0.00000000e+000,\n",
       "       0.00000000e+000, 0.00000000e+000, 0.00000000e+000, 0.00000000e+000,\n",
       "       0.00000000e+000, 0.00000000e+000, 0.00000000e+000, 0.00000000e+000,\n",
       "       0.00000000e+000, 0.00000000e+000, 0.00000000e+000, 0.00000000e+000,\n",
       "       0.00000000e+000, 0.00000000e+000, 0.00000000e+000, 0.00000000e+000,\n",
       "       0.00000000e+000, 0.00000000e+000, 0.00000000e+000, 0.00000000e+000,\n",
       "       0.00000000e+000, 0.00000000e+000, 0.00000000e+000, 0.00000000e+000,\n",
       "       0.00000000e+000, 0.00000000e+000, 0.00000000e+000, 0.00000000e+000,\n",
       "       0.00000000e+000, 0.00000000e+000, 0.00000000e+000, 0.00000000e+000,\n",
       "       0.00000000e+000, 0.00000000e+000, 0.00000000e+000])"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "[M, n, N] = [18820, len(MO_genes), len(G_prox.nodes())]\n",
    "#https://www.science.org/doi/10.1126/science.abj6987\n",
    "#M = KNOWN NTD GENE ( MOUSE GENE)\n",
    "#N = CNT_PROP_GENE\n",
    "rv = hypergeom(M, n, N)\n",
    "#rv = hypergeom(20, 7, 12)\n",
    "x = np.arange(0, n+1)\n",
    "pmf = rv.pmf(x)\n",
    "pmf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "e5268c6d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAl0AAAHDCAYAAAAeBq+FAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjguMywgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/H5lhTAAAACXBIWXMAAA9hAAAPYQGoP6dpAAB3LUlEQVR4nO3deVhU1f8H8PewMyCCmAubhjooi7mAC0qmmKlRbuGSialoKtmmLaap2ULaYmVZKgJqq7hhZq6glYiAlIKouYuAKCAi+wD394e/uV9GZmAYhmF7v56nJzzn3HM/M3eWz5x77rkSQRAEEBEREVG9MmjoAIiIiIhaAiZdRERERHrApIuIiIhID5h0EREREekBky4iIiIiPWDSRURERKQHTLqIiIiI9IBJFxEREZEeGDV0APSAp6cnSktL8cgjjzR0KERERKShO3fuwMTEBAkJCTW2ZdLVSJSUlKC8vLyhwyAiIqJaKCsrg6Y392HS1Ui0a9cOAHDkyJEGjoSIiIg05evrq3FbzukiIiIi0gMmXURERER60GROL8rlcoSFhSEyMhKpqamQSqXw9PTE/Pnz4erqqnE/OTk5iIqKQlJSEpKSkvDff/9BLpfj5ZdfxoIFCzTuRxAETJ8+HSdPngQA7Nu3D126dKn14yIiIqKWoUkkXXK5HIGBgYiNjYWtrS2GDh2KO3fu4NChQzh69CjWr1+PQYMGadRXYmIilixZUueYfv31V5w8eRISiUTjCXRERETUcjWJpCskJASxsbHw8PBAeHg4LC0tAQB79+7FwoULsWjRIhw+fBgWFhY19mVra4spU6bA3d0d7u7uiIyMRGhoaK3iuXXrFj799FP4+PjgypUrSEtL0+pxERERUcvR6Od0lZWVITw8HACwfPlyMeECAD8/PwwZMgQ5OTnYsWOHRv317t0bK1aswHPPPYfu3bvD0NCw1jEtW7YMFRUVWLFiRa23JSIiopap0SddiYmJyM3NhYODAzw8PKrUjx49GoD+llrYvXs3jh07hldffRUODg562ScRERE1fY0+6Tp//jwAwM3NTWW9YhK9ol19ysrKQnBwMNzd3TFt2rR63x8RERE1H41+Tld6ejoAoEOHDirrFeW5ubkoKCjQaF6XtlauXIn8/Hx89NFHWp2WrG4BtYyMDHTs2LEu4REREVEj1uiTroKCAgCAubm5ynqpVKrUtr6SroMHD+LAgQOYM2cOunfvXi/7aEjlFQJSrmQjJ68YbazM4OpsC0MDSUOHRURE1Gw0+qRLsRyDRNJwCUBubi7ef/99ODk5ISgoSOt+qpt3VpvbCOhazJl0bNidhOx7xWKZbWszzBnrAe+edg0WFxERUXPS6Od0KUauCgsLVdYXFRVVaatrwcHByMrKwsqVK2FmZlYv+2goMWfSEbw5XinhAoDse8UI3hyPmDPpDRQZERFR89LoR7rs7B6MtNy6dUtlvaLc2tq63pKuI0eOwNTUFOvWrcO6deuU6u7cuQMAePvtt2Fubo6pU6di5MiR9RKHrpVXCNiwO6naNhsjk9HfvSNPNRIREdVRo0+6evToAQBISUlRWX/27FkAgIuLS73GUVJSgri4OLX1SUkPkpeGPE1YWylXsquMcD0sK7cIKVey4dG1rZ6iIiIiap4afdLVp08fWFtbIzU1FcnJyXB3d1eq379/P4D6TXYSEhLU1g0bNgxpaWlN8t6LOXnVJ1y1bUdERETqNfo5XUZGRggICADwvyUbFPbt24fo6GjY2NhgwoQJYnlmZiZGjhyJkSNHIjMzU+8xNxVtrDSbn6ZpOyIiIlKv0Y90AcDs2bMRGxuLuLg4jBgxAl5eXsjKykJCQgKMjY2xevVqpdsDyeVyXL16Vfz7YRMnThT/zsjIAABERETgr7/+Esu3bdtWXw+n0XB1toVta7NqTzG2tTaHq7OtHqMiIiJqnppE0mViYoJNmzYhNDQUe/bsQVRUFKRSKXx9fREUFKR2tXp1Tp8+XaUsMzOzxY2KGRpIMGesB4I3x6ttM3uMOyfRExER6YBEUCyERQ1KMSdNX/eQrEzVOl1trc0xe4w71+kiIiKqRm2+v5vESBfVL++edujv3hEHT1zDup1n8GQ/JzzexwEeXXjFIhERka4w6SIAwMnkDPx48MFNww/F3cChuBtclZ6IiEiHGv3Vi1T/FKvS38svVSrnqvRERES6w6SrhdN0VfryCk79IyIiqgsmXS1cbValJyIiIu0x6WrhuCo9ERGRfjDpauG4Kj0REZF+MOlq4RSr0leHq9ITERHVHZOuFk6xKn11uCo9ERFR3THpInj3tMPi6V5obWmiVN7W2hyLp3txnS4iIiId4OKoBOBB4tXW2hwLv/oTABD03GN4sn8njnARERHpCEe6SGRQKcHq6mjNhIuIiEiHmHQRERER6QGTLiIiIiI9YNJFREREpAdMuoiIiIj0gEkXERERkR4w6SIiIiLSAyZdRERERHrApIuIiIhID5h0EREREekBbwPUwpVXCEi5ko2cvGIUlZQ1dDhERETNFpOuFizmTDo27E5C9r3ihg6FiIio2ePpxRYq5kw6gjfHq024zly8o+eIiIiImjcmXS1QeYWADbuTqm2z6+hllFcIeoqIiIio+WPS1QKlXMmu8ZRibn4JUq5k6ykiIiKi5o9JVwuUk6fZHC5N2xEREVHNmHS1QG2szHTajoiIiGrGpKsFcnW2hW3r6hMqa0tTuDrb6ikiIiKi5o9JVwtkaCDBnLEe1bYZ90QXGBpI9BQRERFR88ekq4Xy7mmHxdO91I549ez2iJ4jIiIiat6azOKocrkcYWFhiIyMRGpqKqRSKTw9PTF//ny4urpq3E9OTg6ioqKQlJSEpKQk/Pfff5DL5Xj55ZexYMECtfs+efIkjh49isTERKSlpaGwsBAdOnTA4MGDERgYCHt7e109VL3x7mmH/u4dcejkdXy7/TQm+nbDtiMXGzosIiKiZqlJJF1yuRyBgYGIjY2Fra0thg4dijt37uDQoUM4evQo1q9fj0GDBmnUV2JiIpYsWVKr/cfHx2PWrFkAAHt7e3h6egIAzpw5g59++gl79uxBSEgIevfuXbsH1ggYGkjQ1dEaAODQvlXDBkNERNSMNYmkKyQkBLGxsfDw8EB4eDgsLS0BAHv37sXChQuxaNEiHD58GBYWFjX2ZWtriylTpsDd3R3u7u6IjIxEaGhotdtIJBKMHj0aM2bMQM+ePcXykpISrFixAjt37sTChQtx4MABGBsb1+3BEhERUbPU6Od0lZWVITw8HACwfPlyMeECAD8/PwwZMgQ5OTnYsWOHRv317t0bK1aswHPPPYfu3bvD0NCwxm0GDhyINWvWKCVcAGBqaooVK1agVatWSEtLwz///KP5AyMiIqIWpdEnXYmJicjNzYWDgwM8PKpecTd69GgAwJEjR/QdGoAHiVfnzp0BALdv326QGIiIiKjxa/RJ1/nz5wEAbm5uKusVk+gV7fStvLwcaWlpAIC2bds2SAxERETU+DX6pCs9PR0A0KFDB5X1ivLc3FwUFBToLS6FyMhI5OTkwNbWFn369NH7/omIiKhpaPQT6RWJlLm5ucp6qVSq1FaTyfS6cvPmTaxatQoA8Prrr8PExKTa9r6+vmrrMjIy0LFjR53GR0RERI1Hox/pEgQBwIMrCBuT+/fvY/78+cjNzcWoUaPg7+/f0CERERFRI9boR7oUI1eFhYUq64uKiqq0rW/FxcWYN28eLly4gAEDBmD16tUabVfdZP/qRsGIiIio6Wv0I112dnYAgFu3bqmsV5RbW1vrJekqLS3FggULEB8fj969e2PdunU1nlYkIiIiavRJV48ePQAAKSkpKuvPnj0LAHBxcan3WMrLy7Fo0SL8+eef6NGjBzZs2KDXOWT6lHj+NpIuZaG8QmjoUIiIiJqFRn96sU+fPrC2tkZqaiqSk5Ph7u6uVL9//34A9X96ThAELFmyBAcOHECXLl0QGhoKKyuret2nvl1MvSv+vfWPcwAA29ZmmDPWA9497RoqLCIiomah0Y90GRkZISAgAACwcuVK5Ofni3X79u1DdHQ0bGxsMGHCBLE8MzMTI0eOxMiRI5GZmamTOD7++GPs2rULjo6OCAsLQ5s2bXTSb2Py219Xq5Rl3ytG8OZ4xJxJb4CIiIiImo9GP9IFALNnz0ZsbCzi4uIwYsQIeHl5ISsrCwkJCTA2Nsbq1auVbg8kl8tx9epV8e+HTZw4Ufw7IyMDABAREYG//vpLLN+2bZv49+HDh7FlyxYAD+aYrVmzRmWcw4cPx/Dhw+vwSBtGhQanEDdGJqO/e0cYGjSuq0iJiIiaiiaRdJmYmGDTpk0IDQ3Fnj17EBUVBalUCl9fXwQFBaldrV6d06dPVynLzMxUOyqWl5cn/n3y5Em1/drb2zfJpOtK2r0a22TlFiHlSjY8unLVfSIiIm00iaQLeJB4zZ07F3Pnzq2xrYODAy5cuKC2vro6VcaPH4/x48fXapumJK+gVKN2OXnF9RwJERFR89Xo53RR/bOy0GzJizZWZvUcCRERUfPFpIvgbN+6xjZtrc3h6myrh2iIiIiaJyZdBAMNJsfPHuPOSfRERER1wKSLRM/4PFqlrK21ORZP9+I6XURERHXUZCbSU/3r5mgD4H9rdQU99xie7N+JI1xEREQ6wJEuUqurozUTLiIiIh1h0kVERESkB0y6iIiIiPSASRcRERGRHjDpIiIiItIDJl1EREREesCki4iIiEgPmHQRERER6QGTLiIiIiI9YNJFREREpAdMuoiIiIj0gEkXERERkR4w6SIiIiLSAyZdRERERHrApIuIiIhID5h0EREREekBky4iIiIiPWDSRURERKQHTLqIiIiI9EDjpCsgIAAfffSRyrr8/HwUFRXpLCgiIiKi5kbjpCsuLg4pKSkq6zw9PTF79mydBUVERETU3Ojs9KIgCLrqioiIiKjZ4ZwuIiIiIj1g0kVERESkB0y6iIiIiPSASRcRERGRHhjVpvH9+/cRHx9f6zoFLy+v2uyOiIiIqNmoVdJ18eJFBAQEVCmXSCRq6yq3UbfkhCbkcjnCwsIQGRmJ1NRUSKVSeHp6Yv78+XB1ddW4n5ycHERFRSEpKQlJSUn477//IJfL8fLLL2PBggV6iYGIiIhanlolXXVZFqIu28rlcgQGBiI2Nha2trYYOnQo7ty5g0OHDuHo0aNYv349Bg0apFFfiYmJWLJkSYPGQERERC2PxknX+fPn6zOOaoWEhCA2NhYeHh4IDw+HpaUlAGDv3r1YuHAhFi1ahMOHD8PCwqLGvmxtbTFlyhS4u7vD3d0dkZGRCA0N1WsMRERE1PI0+on0ZWVlCA8PBwAsX75cTHYAwM/PD0OGDEFOTg527NihUX+9e/fGihUr8Nxzz6F79+4wNDTUewxERETU8jT6pCsxMRG5ublwcHCAh4dHlfrRo0cDAI4cOdKsYyAiIqKmrVZzuhqC4rSmm5ubynrFBPb6PP3ZGGKoD+UVAlKuZOPctRwAvJUTERFRfdI46QoLC6vzzmbMmFHrbdLT0wEAHTp0UFmvKM/NzUVBQUG9zKlqDDHoWsyZdGzYnYTse8Vi2cbdSQ0YERERUfOmcdK1atUqSCSSOu1Mm6SroKAAAGBubq6yXiqVKrWtj4RHVzH4+vqq3UdGRgY6duxYhyg1F3MmHcGbq66pll9Uppf9ExERtUS1Pr3YqVMnPPLII/URi0qKU151Tfiaegy6Ul4hYANHtIiIiPSu1ut03bx5E507d8b48eMxbNgwGBnV77QwxahRYWGhyvqioqIqbRtrDNVNtK9uFEyXUq5kK51SJCIiIv3QOGM6dOgQdu7cicjISBw9ehTHjh2DjY0N/Pz8MH78eHTv3r1eArSzswMA3Lp1S2W9otza2rrekq7GEIOu5OQx4SIiImoIGi8Z4ejoiFdffRVHjhzBpk2bMGrUKBQUFGDLli0YN24cxo8fjx9//BG5ubk6DbBHjx4AoPYWQmfPngUAuLi46HS/jS0GXWljZdbQIRAREbVItV6nSyKRYNCgQfjiiy/w999/Y9myZXB1dUVKSgo+/PBDPP7443jttddw7NgxnSxB0KdPH1hbWyM1NRXJyclV6vfv3w+gfk/PNYYYdMXV2Ra2rZl4ERER6VudFkdt1aoVnn/+eezYsQO//fYbAgICYGlpif3792Pu3Ll4+eWX6xygkZGReCPtlStXIj8/X6zbt28foqOjYWNjgwkTJojlmZmZGDlyJEaOHInMzMwGiaGxMjSQYM7Yqgu8EhERUf3S2Sz4bt26YfHixXjiiSewdOlSpKenK00wr4vZs2cjNjYWcXFxGDFiBLy8vJCVlYWEhAQYGxtj9erVSrfmkcvluHr1qvj3wyZOnCj+nZGRAQCIiIjAX3/9JZZv27atTjE0Zt497bB4uleVdboszY2RX1T1+SIiIqK600nSlZ6ejl27dmH37t24efMmBEGAlZUVBg4cqIvuYWJigk2bNiE0NBR79uxBVFQUpFIpfH19ERQUpHaleHVOnz5dpSwzM7PaUTFdx9DQvHvaob97Rxw6eR3fbn/wfMwe6441P//TwJERERE1T1onXSUlJThw4AB27tyJuLg4VFRUwMDAAN7e3hg/fjyefPJJmJiY6CxQExMTzJ07F3Pnzq2xrYODAy5cuKC2vro6XcXQFBgaSNDV0Vr8d3NYh4yIiKixqnXSlZiYiJ07d2L//v0oKCiAIAhwcnLCuHHjMG7cOLW3yiEiIiJqyTROutavX4+dO3fixo0bEAQB5ubmGDt2LMaPHw8vL6/6jJGIiIioydM46VqzZg0kEgm6dOmCcePGYdSoUeI9BzVdm8va2lqbGImIiIiavFqfXrx8+TI+++wzfPbZZ7XaTiKRqF1clIiIiKi5q/W9F7Wli4VSiYiIiJoqjZOu6m7WTM1T4vnbKCoug6uzLQwNeGUjERFRXWicdNnb29dnHNQIbf3jHADAtrUZ5oz1gHdPuwaOiIiIqOmq022AqHm5mHpXZXn2vWIEb45HzJl0PUdERETUfNR6Iv3hw4cRGRmJa9euAQAeffRRjBkzpknc7JmqF52QWm39xshk9HfvyFONREREWqhV0vXhhx/ixx9/BPC/ifGXLl3CoUOHMHXqVCxdulT3EZLe5BeVVVuflVuElCvZ8OjaVk8RERERNR8aJ11//vknfvjhBwAPbm7dr18/CIKA+Ph4XLx4ET/++COGDBkCHx+feguWGl5OXnHNjYiIiKgKjZOuiIgIAMDEiROxYsUKGBg8mA5WUVGB5cuXIyIiAhEREUy6mrk2VmYNHQIREVGTpPFE+qSkJJiZmeGdd94REy4AMDAwwDvvvANTU1MkJSXVS5CkH5bm1efgba3N4epsq6doiIiImheNk67s7Gw4OjqKt/6pzMLCAk5OTsjOztZpcKRfQz0dq62fPcadk+iJiIi0pPHpRblcDisrK7X1rVq1glwu10lQ1DC6OdoAuFqlvK21OWaPcec6XURERHVQ6yUjqGUJeu4xPNm/E0e4iIiI6qhWSVdpaSnS01UvkFlaWgoAyMjIUHufRTs7jpQ0NV0drZlwERER6UCtkq7k5OQaF0EdNmyYynKJRIKUlJTa7I6IiIio2ahV0qVuBKu+tyUiIiJq6jROuo4cOVKfcRARERE1axonXfb29vUZBxEREVGzpvE6XURERESkPSZdRERERHrApIuIiIhID5h0EREREekBky4iIiIiPWDSRURERKQHTLqIiIiI9IBJFxEREZEeMOkiIiIi0oNa3XtRnfz8fKSmpqKgoKDaeyx6eXnpYndERERETU6dkq5//vkHn3/+OU6dOlVjW4lEgpSUlLrsjoiIiKjJ0jrpio+Px6xZsyCXyyEIAszMzNCmTRtdxqZELpcjLCwMkZGRSE1NhVQqhaenJ+bPnw9XV9da93fixAmEhIQgKSkJpaWlcHZ2xqRJkzBx4kRIJBKV21y5cgUbN27EyZMncfv2bRgZGcHJyQnDhw/HzJkzYWlpWdeHSURERM2U1knXl19+idLSUgwaNAiLFy9G165ddRmXErlcjsDAQMTGxsLW1hZDhw7FnTt3cOjQIRw9ehTr16/HoEGDNO4vIiIC7733HgwMDDBgwABYWFjg+PHjWLZsGf79918EBwdX2SYhIQGzZs1CcXExOnfujGHDhqGoqAiJiYn49ttvsW/fPvzyyy+wtrbW4SMnIiKi5kLrpCslJQUWFhb49ttvYWZmpsuYqggJCUFsbCw8PDwQHh4ujijt3bsXCxcuxKJFi3D48GFYWFjU2Fdqairef/99GBkZISwsTJxnlpmZieeffx47d+6Ej48PRo8erbTdihUrUFxcjPnz5+OVV14RR8Nyc3Mxc+ZMnD17FiEhIVi0aJGOHz0RERE1B1pfvWhiYoLOnTvXe8JVVlaG8PBwAMDy5cuVTuH5+flhyJAhyMnJwY4dOzTqb/PmzZDL5Zg4caLSxP727duLCdPGjRuVtrl79y4uXrwIY2NjzJs3T+n0o7W1NWbOnAkAOHPmjFaPkYiIiJo/rZMud3d3ZGZm6jIWlRITE5GbmwsHBwd4eHhUqVeMSB05ckSj/qKiogAAo0aNqlLn6+sLU1NTpKSkICMjQyw3NjbWqG8bGxuN2hEREVHLo3XSFRgYiOzsbI1HmLR1/vx5AICbm5vKesUkekW76uTn5yMtLU1pu8pMTEzEuWmV+7O0tESvXr0gl8vx3XffKS2LkZubi9DQUADAhAkTNHlIRERE1AJpPadr4MCBWLp0KT744AOcO3cO/v7+6NSpk85PN6anpwMAOnTooLJeUZ6bm4uCgoJq53UpEi4rKyu17Tp06ICzZ8+KbRU+/PBDBAYGYt26ddi3bx9cXFxQXFyMU6dOwdzcHKtWrcLjjz9e68dHRERELUOd1uny8/NDTEwMfvzxR/z444/VttV2na6CggIAgLm5ucp6qVSq1La6pKumvir3p2ir0K1bN/z666945ZVXcPr0aVy7dk2sGzhwoEZXb/r6+qqty8jIQMeOHWvsg4iIiJomrU8v3rp1CxMmTEBUVBQEQajxv4qKCq32oziVp27tLF33pW5F/djYWDz77LPIz89HSEgI4uPj8eeff2LlypWIjY3F888/j7/++qvOMRIREVHzpPVI1xdffIGbN2/C3t4es2bNgru7O9q0aaOT5KgyxchVYWGhyvqioqIqbbXtCwCKi4ur9HXv3j28+uqrKC0tRUhICOzs7AA8OE05adIktGrVCq+//jpWrFiBgwcPwtDQUGXf1U32r24UjIiIiJo+rZOumJgYGBsbY/PmzXBwcNBlTEoUCc6tW7dU1ivKra2ta0y67O3tAQB5eXkoLCxUOjX5cH+K/QLA0aNHkZubi4EDByqVK4wYMQLGxsa4efMmUlNT0blz55ofGBEREbUoWp9eLCgogLOzc70mXADQo0cPAFA7H+zs2bMAABcXlxr7atWqlZh4qeqvtLQUFy9eBAB0795dLFckYq1atVLZr5GRkZjA3bt3r8Y4iIiIqOXROunq3LmzeCquPvXp0wfW1tZITU1FcnJylfr9+/cD0Pz03LBhw5S2qyw6OholJSXo0aOH0ojWI488AuBBgldWVlZluytXrojJliKpIyIiIqpM66RrwoQJuHHjRr2vwm5kZISAgAAAwMqVK5Gfny/W7du3D9HR0bCxsVFaIyszMxMjR47EyJEjqyzgGhAQAGNjY/zyyy9ISEgQy2/fvo1PP/0UwIM1yCp7/PHHYWZmhrS0NHz22WdKiVdOTg6WLl0KAOjXrx/atm2ro0dOREREzYnWc7peeOEF/PvvvwgKCsJ7772HESNG6DIuJbNnz0ZsbCzi4uIwYsQIeHl5ISsrCwkJCTA2Nsbq1auVbg8kl8tx9epV8e/KnJycsHTpUqxYsQIBAQEYOHAgpFIpYmJikJ+fjzFjxsDPz09pm7Zt22Lp0qVYtmwZwsLCcODAAbi6uqK4uBinT5/G/fv30bZtW6xcubLengMiIiJq2rROuhSjT3fv3sWrr74KKysrODk5qV0DSyKRYPPmzVrty8TEBJs2bUJoaCj27NmDqKgoSKVS+Pr6IigoSO1q9epMnjwZTk5O2LhxI06fPg25XA5nZ2dMnDgRkydPVrmNv78/ZDIZNm/ejMTERBw7dgyGhoZwcHCAv78/AgMDYWtrq9XjIyIiouZP66QrLi5O6d/37t1DUlKS2vZ1XUrCxMQEc+fOxdy5c2ts6+DggAsXLlTbxtvbG97e3rWK4bHHHsMXX3xRq22IiIiIgDokXcHBwbqMg4iIiKhZ0zrpGjdunC7jICIiImrWtL56kYiIiIg0V6cbXldWUVGB9PR08abTdnZ2MDBgTkdEREQE6CDpOnv2LL7//nv8/fffSoulmpmZwcfHBy+99FKtry4kIiIiam7qNBS1fft2TJo0CYcPH0ZRUREEQRD/KyoqwsGDBzFp0iTs2LFDV/ESERERNUlaj3SdO3cOy5cvR3l5OTw9PTFz5kzIZDK0a9cOt2/fxsWLFxEaGor4+HgsX74crq6u4n0UiYiIiFoarUe6Nm3ahPLycsyYMQM//PADhg0bBgcHB5iYmMDBwQFDhw7F1q1bMXPmTJSVlSE0NFSXcRMRERE1KVonXfHx8bCyssIbb7xRbbvXXnsNrVq1qrKYKhEREVFLonXSlZ2djU6dOsHY2LjadiYmJujcuTNycnK03RURERFRk6d10mVhYYGsrCyN2mZnZ0MqlWq7KyIiIqImT+uky9XVFbdu3cKRI0eqbRcVFYX09HS4urpquysiIiKiJk/rpGvChAkQBAGLFi1CeHi40hpdAFBSUoLNmzfjzTffhEQiwXPPPVfnYImIiIiaKq2XjPDz88PBgwdx8OBBrFq1Cl999RUcHR3xyCOP4M6dO0hNTUVxcTEEQcBTTz2Fp59+WpdxExERETUpdVqRfs2aNfjuu+8QHh6O/Px8/Pfff/jvv//EektLS7z44ouYN29enQMlIiIiasrqlHQZGhri5ZdfxqxZs5CQkICrV6+K9150dnZG3759YW5urqtYiYiIiJosndzw2tzcHD4+PvDx8dFFd0RERETNjk6SLmp6yisEpFzJxrlrXD+NiIhIH+qcdF27dg3h4eGIjY1FZmYmSkpKkJKSItZv374dt27dwowZM2BhYVHX3ZEOxJxJx4bdSci+p3zF6cXUuw0UERERUfNXp6Rr7969WLJkCUpLSyEIAgBAIpEotbl37x6+/fZbdOnSBaNGjarL7kgHYs6kI3hzvMq63/66WqUs8fxtFBWXwdXZFoYGEhVbERERkSa0Xqfr3LlzeOeddyCXyzF16lRs3boVbm5uVdqNHDkSgiDUuIgq1b/yCgEbdifVaputf5zDu98dx6wPDyLmTHo9RUZERNT8aZ10hYSEoLy8HIsXL8bSpUvh5eUFU1PTKu3s7e3Rtm1bpaUkqGGkXMmuckpRU9n3ihG8OZ6JFxERkZa0Trri4+NhaWmJadOm1di2ffv2uH37tra7Ih3JydMu4apsY2QyyisEHURDRETUsmiddOXk5MDR0VGznRgYoLCwUNtdkY60sTKrcx9ZuUVIuZKtg2iIiIhaFq2TrlatWuHOnTsatb1x4wZsbGy03RXpiKuzLWxb1z3x0sWIGRERUUujddLVo0cPZGVlKS0PoUp0dDTu3buHxx57TNtdkY4YGkgwZ6xHnfvRxYgZERFRS6N10jVmzBgIgoD33nsPWVlZKttcunQJy5cvh0Qiwfjx47UOknTHu6cdFk/3Ujni5Teoc43bt7U2h6uzbT1ERkRE1LxpvU7Xs88+i927d+PEiRPw8/PD0KFDcevWLQDA1q1bkZiYiMOHD0Mul8PX1xdPPPGErmKmOvLuaYf+7h1x6OR1fLv9tFgu69QGOH6t2m1nj3Hnel1ERERa0Drpkkgk+Oabb7BkyRLs378fu3btEus+/vhjcbHUkSNH4pNPPql7pKRThgYSdHW01rh9W2tzzB7jDu+edvUXFBERUTNWpxXpLSws8OWXX+LMmTPYv38/Lly4gHv37kEqlUImk2HUqFHo27evrmKlBhL03GN4sn8njnARERHVgU5ueN2zZ0/07NlTF11RI9TV0ZoJFxERUR1pPZGeiIiIiDSnk5EufZDL5QgLC0NkZCRSU1MhlUrh6emJ+fPnw9XVtdb9nThxAiEhIUhKSkJpaSmcnZ0xadIkTJw4scpNuyu7f/8+QkNDcfjwYdy8eRPAgxX3+/bti1deeQXt27fX+jESERFR86V10hUQEKBxW0NDQ1haWsLBwQGenp544oknYGhoqPH2crkcgYGBiI2Nha2tLYYOHYo7d+7g0KFDOHr0KNavX49BgwZp3F9ERATee+89GBgYYMCAAbCwsMDx48exbNky/PvvvwgODla53aVLlzBjxgzcvn0bdnZ28PHxQXl5Oa5fv47t27dj3LhxTLqIiIhIJa2Trri4OAAQR4UUVytW9nCdRCJBeHg4HB0d8fnnn8PDQ7OFOkNCQhAbGwsPDw+Eh4fD0tISALB3714sXLgQixYtwuHDh2FhYVFjX6mpqXj//fdhZGSEsLAweHl5AQAyMzPx/PPPY+fOnfDx8cHo0aOVtsvLy8PMmTORlZWFpUuX4oUXXlAaEbtx44YYFxEREdHDtJ7TFRwcjKCgIHEUa9y4cXjnnXfwwQcf4J133sG4ceNgaWkJIyMjBAUF4b333sOUKVNgY2ODGzduIDAwEJmZmTXup6ysDOHh4QCA5cuXKyU2fn5+GDJkCHJycrBjxw6N4t68eTPkcjkmTpwoJlzAg1OEixYtAgBs3LixynbffPMNMjMz8eKLL2LatGlVTkE6OTmhTZs2GsVARERELY/WSZe3tzd++ukn9O3bF4cOHUJwcDBefPFF+Pv748UXX0RwcDAOHjyIXr164aeffoKvry+WLVuGgwcPwsvLC3l5eQgLC6txP4mJicjNzYWDg4PKkTHFiNSRI0c0ijsqKgoAMGrUqCp1vr6+MDU1RUpKCjIyMsTykpIS7Nq1CwYGBpgxY4ZG+yEiIiKqTOuk6+uvv0Z+fj7WrFkDa2trlW1sbGywZs0a3L9/H19//TUAwNLSEh9//DEA4K+//qpxP+fPnwcAuLm5qaxXTKJXtKtOfn4+0tLSlLarzMTEBF27dq3SX3JyMvLy8tCtWze0a9cOf/75J1atWoVly5Zhw4YNuHLlSo37JiIiopZN6zldf/31F7p161bjKbW2bdtCJpPh77//FsscHR3h5OSE9PT0GvejaNOhQweV9Yry3NxcFBQUVDuvS5FwWVlZqW3XoUMHnD17VmwLPJhADwD29vaYO3cuoqOjlbZZs2YNXnrpJbz22mvVPhZfX1+1dRkZGejYsWO12xMREVHTpfVIV25uLkpLSzVqW1paitzcXKUyKysrVFRU1LhtQUEBAMDc3FxlvVQqrdJW274q91e5r3v37gEA/vzzT/z555947bXXcOzYMfGKR2NjY3z33XeIiIio8fEQERFRy6T1SFf79u1x5coVXL58GV26dFHb7uLFi7h8+TIcHByUytPT09Welqys8pWPdaVJX6quwiwvLwfwYFL/zJkzMW/ePLFu6tSpkMvlCA4Oxrp16+Dv76+27+rmnVU3CkZERERNn9YjXSNGjEBFRQXmzZuH5ORklW2SkpIQFBQEAHjqqafE8szMTGRnZ6Nz58417kdxGrCwsFBlfVFRUZW22vYFAMXFxVX6qvz3xIkTq2yjKEtPT0dqamq1MRAREVHLpPVI17x583D06FFcvnwZ/v7+6NatG7p37w4LCwsUFBTg/PnzuHjxIgRBQNeuXZVGh7Zt2wYAGi1oamdnBwC4deuWynpFubW1dY1Jl729PYAHa24VFhYqnZp8uD/Ffitv9/DfClKpFLa2tsjOzsadO3fg6OhYbRxERETU8middFlaWuKHH37A+++/jwMHDuC///7Df//9p9RGIpFg1KhRWLZsmVJCNGvWLLz44ovVzq1S6NGjBwAgJSVFZf3Zs2cBAC4uLjX21apVK9jb2yMtLQ0pKSnw9PRUqi8tLcXFixcBAN27dxfLK1/pmJeXh7Zt2yptV1FRgby8PABQmcgRERER1eneizY2Nvjyyy+RmpqKv//+G1evXhVHkB599FEMHjxY5ahPbRKTPn36wNraGqmpqUhOToa7u7tS/f79+wFoPidq2LBh2Lp1K/bv318l6YqOjkZJSQl69OihNNLVsWNHuLm54ezZs4iNjYWfn5/SdqdOnYJcLoe5uTmcnZ01fmxERETUcmg9p2vLli3YunUrSktL4ejoiClTpuDdd9/Fhx9+iHfffRdTpkzRyWk2IyMj8T6PK1euRH5+vli3b98+REdHw8bGBhMmTBDLMzMzMXLkSIwcObLKqvcBAQEwNjbGL7/8goSEBLH89u3b+PTTTwEAgYGBVeKYM2cOAOCLL77A9evXlbb76KOPAADPPfccTExM6vqQiYiIqBnSeqQrODgYTk5OmDZtmi7jUWn27NmIjY1FXFwcRowYAS8vL2RlZSEhIQHGxsZYvXq10u2B5HI5rl69Kv5dmZOTE5YuXYoVK1YgICAAAwcOhFQqRUxMDPLz8zFmzJgqI1kAMHLkSEyZMgU///wzxo4di169esHIyAj//PMP7t+/j169emHhwoX1+0QQERFRk6V10tWmTRtYWVnpMha1TExMsGnTJoSGhmLPnj2IioqCVCqFr68vgoKC1K5Wr87kyZPh5OSEjRs34vTp05DL5XB2dsbEiRMxefJktdutWLECffv2xY8//ojTp0+jrKwMnTt3hp+fH6ZPnw5TU9O6PlQiIiJqprROunr16oWTJ09CLpfD2NhYlzGpZGJigrlz52Lu3Lk1tnVwcMCFCxeqbePt7Q1vb+9ax/HMM8/gmWeeqfV2RERE1LJpPadr1qxZKCoqwjfffKPLeIiIiIiaJa1Hujp06IDXX38dX375Jc6fP4/x48eja9eu1S4DUfmKQCIiIqKWROukq/ISDYp7ElZHIpGoXWuLiIiIqLnTOulSdY9CXbYnIiIiak60TrrOnz+vyziIiIiImjWtJ9ITERERkeaYdBERERHpQZ3uvahw5swZxMbG4tatWyguLsbHH38s1t2+fRtlZWW8cpGIiIhatDolXZmZmXjzzTcRHx8P4MFkeYlEopR0ffXVV9i5cyd+/vln9OrVq07BEhERETVVWp9ezM/PR0BAAOLi4tCuXTuMHTsWHTp0qNJu3LhxEAQBR44cqVOgRERERE2Z1klXaGgorl+/jiFDhmDfvn0IDg5WeQqxT58+MDMzE0fDiIiIiFoirZOuQ4cOwcjICB9//DEsLCzU78DAAI6OjsjMzNR2V0RERERNntZJV2pqKjp16gRbW9sa21pYWCArK0vbXRERERE1eVonXQYGBqioqNCobU5ODqRSqba7IiIiImrytE66HBwccPPmTeTn51fbLjU1FTdu3ICzs7O2uyIiIiJq8rROuoYMGQK5XI61a9eqbSMIAj7++GNIJBIMGzZM210RERERNXlar9M1Y8YMbNu2DVu2bEF2djYmTZqEsrIyAA+Wk0hMTMSGDRuQkJCA9u3bY8qUKToLmoiIiKip0TrpatOmDdatW4f58+dj7969+P3338U6Ly8vAA9GuqytrfHNN9/A0tKy7tESERERNVF1uvdi3759sWfPHkyePBk2NjYQBEH8r1WrVnjuueewa9cueHh46CpeIiIioiapzvdebN++PVasWIEVK1bg9u3byMvLg1QqRYcOHWBgwPtpExEREQE6uuG1Qrt27dCuXTtddklERETULOgs6aqoqEB6ejoKCgpgYWEBOzs7jnQRERER/b86J11JSUnYsGED/v77bxQXF4vlZmZmGDx4MGbPno2ePXvWdTdERERETVqdhqLCw8MxefJkHD58GEVFRUoT6YuKinDo0CFMmTIFoaGhuoqXiIiIqEnSeqTrr7/+wieffAIAGDRoEAICAtCtWze0bdsWWVlZuHjxIrZs2YLjx4/j008/Rbdu3eDj46OzwImIiIiaEq2TrpCQEEgkEsyfPx8LFixQqrOzs4OdnR2GDBmCb775Bt988w02bdrEpKuJupSai7Tb+WhjZQZXZ1sYGkgaOiQiIqImR+ukKzk5GVZWVggKCqq23bx587B161YkJSVpuytqYN9uPy3+bdvaDHPGesC7p10DRkRERNT0aD2nSxAEODk51XiFoqGhIZycnLTdDTUy2feKEbw5HjFn0hs6FCIioiZF66SrS5cuyMjI0KhtRkYGunTpou2uqBHaGJmM8gqhocMgIiJqMrROuqZMmYKsrCxERERU2y4iIgJZWVm84XUzk5VbhJQr2Q0dBhERUZOh9Zyu8ePH47///sP777+P5ORkBAQEwNnZGRKJBIIg4OrVq9iyZQu2b9+OGTNmYNy4cbqMmxqBnLzimhsRERERgDokXb6+vuLf27Ztw7Zt22BoaAgbGxvcvXsX5eXlAB7M6Tp48CAOHjxYpQ+JRILDhw9rtD+5XI6wsDBERkYiNTUVUqkUnp6emD9/PlxdXWsd/4kTJxASEoKkpCSUlpbC2dkZkyZNwsSJEyGR1Hx1niAImD59Ok6ePAkA2LdvX4s7hdrGyqyhQyAiImoytE660tLSqpSVlZXhzp07VcpUtQWgUXIDPEi4AgMDERsbC1tbWwwdOhR37tzBoUOHcPToUaxfvx6DBg3SOPaIiAi89957MDAwwIABA2BhYYHjx49j2bJl+PfffxEcHFxjH7/++itOnjwpjuy1NG2tzeHqbNvQYRARETUZWiddW7Zs0WUc1QoJCUFsbCw8PDwQHh4OS0tLAMDevXuxcOFCLFq0CIcPH4aFhUWNfaWmpuL999+HkZERwsLC4OXlBQDIzMzE888/j507d8LHxwejR49W28etW7fw6aefwsfHB1euXFGbVDZns8e4c70uIiKiWtA66erXr58u41CrrKwM4eHhAIDly5eLCRcA+Pn5Yc+ePTh27Bh27NiBgICAGvvbvHkz5HI5pk6dKiZcANC+fXssWrQIr732GjZu3Fht0rVs2TJUVFRgxYoVGu2zOWlrbY7ZY9y5ThcREVEt1enei/qQmJiI3NxcODg4wMPDo0q9Ijk6cuSIRv1FRUUBAEaNGlWlztfXF6ampkhJSVG7HMbu3btx7NgxvPrqq3BwcND0YTRpE327AQCCnnsMIUueZMJFRESkBa2TrrFjx+Knn35Cfn6+LuOp4vz58wAANzc3lfWKSfSKdtXJz88XTwWqmnxvYmKCrl27qu0vKysLwcHBcHd3x7Rp0zR7AM2AQ/tWAICujtY8pUhERKQlrZOu8+fP44MPPsDgwYOxePFiJCQk6DIuUXr6g5XPO3TooLJeUZ6bm4uCgoJq+1IkXFZWVmrnfyn6UzVPa+XKlcjPz8dHH30EQ0NDzR4AEREREeowp+v9999HREQEkpOTsWvXLuzevRuPPvooJk6ciDFjxsDGxkYnASoSKXNzc5X1UqlUqW11k+lr6qtyfw8ncAcPHsSBAwcwZ84cdO/eXbPgH1J5mY2HZWRkoGPHjlr1S0RERI2f1iNdkyZNwvbt27Fnzx688MILsLKywpUrV7Bq1So8/vjjeP311xETE1PnABXLMWi6vERd+1K1/ENubi7ef/99ODk51XiDbyIiIiJVtB7pUpDJZFi6dCneeustHDx4EBEREYiLi8Mff/yB/fv3w97eHv7+/hg3bhzatWtX6/4VI1eFhYUq64uKiqq01bYvACguLq7SV3BwMLKyshAeHg4zM+0XBK1usn91o2C6Vl4hIOVKNs5dy9HbPomIiFq6OiddCiYmJvDz84Ofnx9SU1Oxfft27Nq1Czdv3sSXX36JtWvX4vHHH8ekSZMwZMgQjfu1s3twpdytW7dU1ivKra2ta0y67O3tAQB5eXkoLCxUOjX5cH+K/QIPkiVTU1OsW7cO69atU2qvWAz27bffhrm5OaZOnYqRI0dq8tAaRMyZdGzYnYTse7yFDxERkT7pLOmqzNHRERMmTIBcLseWLVtQVlaGsrIyREVFITo6Gp07d8abb76JYcOG1dhXjx49AAApKSkq68+ePQsAcHFxqbGvVq1awd7eHmlpaUhJSYGnp6dSfWlpKS5evAgAVeZtlZSUIC4uTm3fSUlJAPQ7YlVbMWfSEbw5Xm39xdS7eoyGiIioZdFp0lVaWooDBw4gIiIC8fEPvtwFQcBjjz0Gf39/ZGdn49dff8XVq1cRFBSETz/9FH5+ftX22adPH1hbWyM1NRXJyclwd3dXqt+/fz8AzZOdYcOGYevWrdi/f3+VpCs6OholJSXo0aOH0khXdVdmDhs2DGlpaY3+3ovlFQI27E6qtk10wk09RUNERNTy6GRxVMXyET4+PnjrrbcQFxcHCwsLTJ48Gbt378avv/6K5557Di+99BIOHz6Mt956C4IgYP369TX2bWRkJK76rliyQWHfvn2Ijo6GjY0NJkyYIJZnZmZi5MiRGDlyJDIzM5X6CwgIgLGxMX755RelZOr27dv49NNPAQCBgYF1ej4ao5Qr2TWeUswvkuspGiIiopZH65Gu/Px87N27FxEREUhJSRGv+nN3d8ekSZPg5+encmkGAwMDzJw5Ezt37sS1a9c02tfs2bMRGxuLuLg4jBgxAl5eXsjKykJCQgKMjY2xevVqpdsDyeVyXL16Vfy7MicnJyxdulS8hc/AgQMhlUoRExOD/Px8jBkzpsbRt6YoJ49zuIiIiBqS1kmXj48PiouLIQgCpFIp/Pz8MHnyZJUrvatiY2ODy5cva9TWxMQEmzZtQmhoKPbs2YOoqChIpVL4+voiKChI7Wr16kyePBlOTk7YuHEjTp8+DblcDmdnZ0ycOBGTJ0+uVV9NRRsr7a+6JCIiorrTOukqKiqCq6urOKpV05WDD1uyZAny8vI0bm9iYoK5c+di7ty5NbZ1cHDAhQsXqm3j7e0Nb29vjfevjuJejo2dq7MtbFubVXuK0dLcmKcYiYiI6onWSVdERITKG1BrSttV3Uk7hgYSzBnrUe3Vi0M9HfDbX1f1GBUREVHLofVE+rokXNQwvHvaYfF0L9i2Vn2qsZujbm7dRERERFXVyzpd1Hh597RDf/eOOHTyOr7dfrqhwyEiImox6pR0VVRUIDIyEtHR0bh+/ToKCwtRUVGhsq1EIsHhw4frsjvSEUMDCbo6Wjd0GERERC1KnZaMCAwMxOnTp1XeJPphurhhNREREVFTpXXS9e233+Lff/+FmZkZJkyYgD59+sDW1hYGBjpZb5WIiIioWdE66Tpw4AAkEgm+++47DBw4UJcxERERETU7Wg9L3blzB/b29ky4iIiIiDSgddJlY2OD1q1b6zIWIiIiomZL66Rr8ODBuHTpktINqImIiIhINa2TrgULFsDU1BQfffQRysvLdRkTERERUbOj0UT6+HjVt45ZsGABVq1ahaSkJPj7+6Nz586QSqVq+/Hy8tIuSiIiIqImTqOka9q0adWus3Xp0iV88skn1fYhkUiQkpJSu+iIiIiImgmNki47O7v6joOIiIioWdMo6YqKiqrvOIiIiIiaNS4fT0RERKQHTLqIiIiI9IBJFxEREZEeMOkiIiIi0gMmXVQjQRAAAInnbyPpUhbKK4QGjoiIiKjp0ejqRWrZNu5OAgBs/eMcAMC2tRnmjPWAd08uJUJERKQpjnRRjfKLypT+nX2vGMGb4xFzJr2BIiIiImp6mHSR1jZGJvNUIxERkYaYdJHWsnKLkHIlu6HDICIiahKYdFGd5OQVN3QIRERETQKTLqqTNlZmDR0CERFRk8Cki7TW1tocrs62DR0GERFRk8Cki7Q2e4w7DA0kDR0GERFRk8B1uqhGluZGSstGtLU2x+wx7lyni4iIqBaYdFGNZo/1wJqf/wEABD33GJ7s34kjXERERLXE04tUI4nkfwlWV0drJlxERERaYNJFREREpAdN5vSiXC5HWFgYIiMjkZqaCqlUCk9PT8yfPx+urq617u/EiRMICQlBUlISSktL4ezsjEmTJmHixIlKIzuKfZ88eRJHjx5FYmIi0tLSUFhYiA4dOmDw4MEIDAyEvb29rh4qERERNUNNYqRLLpcjMDAQn3/+Oe7evYuhQ4fC2dkZhw4dwsSJE3H8+PFa9RcREYEZM2bgxIkTcHd3h4+PD65du4Zly5bh3XffrdI+Pj4es2bNwtatW5GbmwtPT088/vjjKC4uxk8//YRnn30W//zzj64eLhERETVDTWKkKyQkBLGxsfDw8EB4eDgsLS0BAHv37sXChQuxaNEiHD58GBYWFjX2lZqaivfffx9GRkYICwuDl5cXACAzMxPPP/88du7cCR8fH4wePVrcRiKRYPTo0ZgxYwZ69uwplpeUlGDFihXYuXMnFi5ciAMHDsDY2FjHj56IiIiag0Y/0lVWVobw8HAAwPLly8WECwD8/PwwZMgQ5OTkYMeOHRr1t3nzZsjlckycOFFMuACgffv2WLRoEQBg48aNStsMHDgQa9asUUq4AMDU1BQrVqxAq1atkJaWxtEuIiIiUqvRJ12JiYnIzc2Fg4MDPDw8qtQrRqSOHDmiUX9RUVEAgFGjRlWp8/X1hampKVJSUpCRkaFRf6ampujcuTMA4Pbt2xptQ0RERC1Po0+6zp8/DwBwc3NTWa+YRK9oV538/HykpaUpbVeZiYkJunbtqnF/AFBeXi722bZtW422ISIiopan0Sdd6enpAIAOHTqorFeU5+bmoqCgoNq+FMmRlZWV2vlfiv4UbWsSGRmJnJwc2Nraok+fPhptQ0RERC1Po59Ir0ikzM3NVdZLpVKlttVNpq+pr8r91ZTAAcDNmzexatUqAMDrr78OExOTatv7+vqqrcvIyEDHjh1r3CcRERE1TY1+pEsQBACosnZWffWlaFOT+/fvY/78+cjNzcWoUaPg7+9f5/iIiIio+Wr0I12KkavCwkKV9UVFRVXaatsXABQXF9fYV3FxMebNm4cLFy5gwIABWL16dbX7Vahusn91o2BERETU9DX6kS47OzsAwK1bt1TWK8qtra1rTLoUq8bn5eWpTbwU/Sn2+7DS0lIsWLAA8fHx6N27N9atW1fjaUUiIiKiRp909ejRAwCQkpKisv7s2bMAABcXlxr7atWqlZh4qeqvtLQUFy9eBAB07969Sn15eTkWLVqEP//8Ez169MCGDRs0WpC1MSmvEJB0KQuJ57m8BRERkT41+qSrT58+sLa2RmpqKpKTk6vU79+/H4Dmp+eGDRumtF1l0dHRKCkpQY8ePaqMdAmCgCVLluDAgQPo0qULQkNDYWVlVduH06BizqRj1ocH8e53x7H1j3MNHQ4REVGL0uiTLiMjIwQEBAAAVq5cifz8fLFu3759iI6Oho2NDSZMmCCWZ2ZmYuTIkRg5ciQyMzOV+gsICICxsTF++eUXJCQkiOW3b9/Gp59+CgAIDAysEsfHH3+MXbt2wdHREWFhYWjTpo1OH2d9izmTjuDN8ci+V9zQoRAREbVIjX4iPQDMnj0bsbGxiIuLw4gRI+Dl5YWsrCwkJCTA2NgYq1evVro9kFwux9WrV8W/K3NycsLSpUuxYsUKBAQEYODAgZBKpYiJiUF+fj7GjBkDPz8/pW0OHz6MLVu2AHgw12vNmjUq4xw+fDiGDx+uy4euE+UVAjbsTqqxnaZXbhIREVHtNYmky8TEBJs2bUJoaCj27NmDqKgoSKVS+Pr6IigoSO1q9epMnjwZTk5O2LhxI06fPg25XA5nZ2dMnDgRkydPrtI+Ly9P/PvkyZNq+7W3t2+USVfKlWyNRrjSbufX2Cbx/G0UFZfB1dkWhgZ1X8aDiIiopZAIHN5oFBRz0jS9h2RtHEu8ic9+PFVju9HenbEv5lqV8md8HsVvf11VKrNtbYY5Yz3g3VP1VZ5EREQtQW2+vxv9nC6quzZWZhq1szQ3Vln+cMIFANn3ihG8OR4xZ9LrFBsREVFLwaSrBXB1toVt65oTL/t2ljW2edjGyGSUV3CwlIiIqCZMuloAQwMJ5oz1qLGdNrdaysotQsqVbG3CIiIialGYdLUQ3j3tsHi6l0YjXrWVk8dlKIiIiGrSJK5eJN3w7mmH/u4dcejkdXy7/bTO+tV0zhgREVFLxpGuFsbQQIKujtY666+ttTlcnW111h8REVFzxaSL6mT2GHeu10VERKQBJl1Uo2d8Hq1S1tbaHIune3GdLiIiIg1xThfVqJujDYD/rdUV9NxjeLJ/J45wERER1QJHuqjWujpaM+EiIiKqJSZdRERERHrApIuIiIhID5h0EREREekBky4iIiIiPWDSRURERKQHTLqo1hLP30bSpSyUVwgNHQoREVGTwXW6qNa2/nEOAGDb2gxzxnpwgVQiIiINcKSLanQx9a7K8ux7xQjeHI+YM+l6joiIiKjpYdJFNYpOSK22fmNkMk81EhER1YBJF9Uov6is2vqs3CKkXMnWUzRERERNE5Mu0omcvOKGDoGIiKhR40T6FqK8QkDKlWzk5BWjqKT6kStttLEy03mfREREzQmTrhYg5kw6NuxOQvY97UajLM2Nqj3F2NbaHK7OttqGR0RE1CLw9GIzF3MmHcGb47VOuABgqKdjtfWzx7jD0ECidf9EREQtAUe6mrHyCgEbdifVuZ9ujjYArlYpb2ttjtlj3LlOFxERkQaYdDVjKVey6zTCVZ0n+znh8T4O8OjStl76JyIiam6YdDVj9XlF4aG4GzgUd4Or0hMREWmIc7qaMX1cUchV6YmIiDTDpKsZc3W2hW3ruideglDzavNclZ6IiKh6TLqaMUMDCeaM9ahzP2m382tsw1XpiYiIqsekq5nz7mmHxdO96jTilV8k16gdV6UnIiJSj0lXC9DfvSNen9wHT/ZzAgA8N6xrrba3NDfWqF16VkGtYyMiImopmszVi3K5HGFhYYiMjERqaiqkUik8PT0xf/58uLq61rq/EydOICQkBElJSSgtLYWzszMmTZqEiRMnQiJRvdCnrmPQB1Wr0e8/ca1Wfdi3s9So3U8HzqNTh1a8kpGIiEiFJjHSJZfLERgYiM8//xx3797F0KFD4ezsjEOHDmHixIk4fvx4rfqLiIjAjBkzcOLECbi7u8PHxwfXrl3DsmXL8O677+olBn1Qtxp9dbf0UUVdEqoKJ9QTERGp1iRGukJCQhAbGwsPDw+Eh4fD0vLByMvevXuxcOFCLFq0CIcPH4aFhUWNfaWmpuL999+HkZERwsLC4OXlBQDIzMzE888/j507d8LHxwejR4+utxj0QVer0ddWVm4Rvt9xGpZSY2TdLUIFBEgkEjxibQ5LqQnyC0uRdbcIggRVytS1bWzlTSF2xsjYGSNjZ4zK5QVFchhIJHDv2hYeXdo2yO3rGn3SVVZWhvDwcADA8uXLxWQHAPz8/LBnzx4cO3YMO3bsQEBAQI39bd68GXK5HFOnThUTLgBo3749Fi1ahNdeew0bN25USrp0HYM+1Odq9DXZH3u9QfZLRERUk18P/4dWUmO87N9L79NhGv3pxcTEROTm5sLBwQEeHlWXP1AkR0eOHNGov6ioKADAqFGjqtT5+vrC1NQUKSkpyMjIqLcY9CE2OaPmRkRERC3Q/UJ5gyzs3eiTrvPnzwMA3NzcVNYrJrAr2lUnPz8faWlpSttVZmJigq5du1bpT5cx6EN5hYCjiakNHQYREVGjtjEySa/zkBv96cX09AdZaIcOHVTWK8pzc3NRUFBQ7ZwqRcJlZWWltl2HDh1w9uxZsa0uY/D19VUbW0ZGBjp27Ki2vjZSrmQjr0CztbWIiIhaqqzcYqRcyYZH17Z62V+jT7oKCh6s/WRubq6yXiqVKrWtLumqqa/K/Sna6joGfdB0kVKnDpZw7WwLADAzNUKnDlYYO6QLikv+d3Xjw+XZ94oRfy6zXuImIiLSN30u7N3oky7Fff9qs2xBXfpSdZ9BXcVQ3Zyv6kbBakvTG13PHfdYleze2b61yraK8vIKAQEr/uBIGhERNQuafmfqQqOf06UYNSosLFRZX1RUVKWttn0BQHFxcZW+dBmDPmhyo+u21uZwdbatdd+GBhLMG/+YtqERERE1Gm2tzbT6LtRWo0+67OweXM5569YtlfWKcmtr6xoTHnt7ewBAXl6e2gRK0Z9iv7qOQR80udH17DHuWq9RMriXPcY/0UWrbYmIiBqL2WM89LpeV6NPunr06AEASElJUVl/9uxZAICLi0uNfbVq1UpMvFT1V1paiosXLwIAunfvXi8x6Iu6G123tTbH4uledV6bZMYz7ng7wBPmpo3+DDUREZGSVlITnXwX1laj/8bs06cPrK2tkZqaiuTkZLi7uyvV79+/H4Dmc6KGDRuGrVu3Yv/+/fD09FSqi46ORklJCXr06KE00qXrGPTFu6cd+rt3RMqVbOTkFaON1YNhVF1l9YMfs8dADzskX8rCv5fuIPtuEWytzRrdKsRNeQVlxtg4y5tC7IyRsTNGrkhfa0ZGRggICMDXX3+NlStXIjQ0VFwRft++fYiOjoaNjQ0mTJggbpOZmYnp06cDeLACffv27cW6gIAA/PLLL/jll18wcuRIMfG6ffs2Pv30UwBAYGBgnWNoLAwNJPV6KayhgQSPyR7BY7JH6m0fREREzUGjT7oAYPbs2YiNjUVcXBxGjBgBLy8vZGVlISEhAcbGxli9erXSrXnkcjmuXr0q/l2Zk5MTli5dihUrViAgIAADBw6EVCpFTEwM8vPzMWbMGPj5+dU5BiIiIqLKmkTSZWJigk2bNiE0NBR79uxBVFQUpFIpfH19ERQUpHaleHUmT54MJycnbNy4EadPn4ZcLoezszMmTpyIyZMn6yUGIiIialkkgqqFqUjvFPPBGtP9G4mIiKh6tfn+bvRXLxIRERE1B0y6iIiIiPSASRcRERGRHjDpIiIiItIDJl1EREREesCki4iIiEgPmsQ6XS3B7du3UV5e3uhuJURERETqZWRkwNDQUKO2HOlqJExNTWFkVD85cEZGBjIyMuqlb2p4PL7NH49x88dj3HQZGRnB1NRUo7ZcHLUF4MKrzRuPb/PHY9z88Ri3DBzpIiIiItIDJl1EREREesCki4iIiEgPmHQRERER6QGTLiIiIiI9YNJFREREpAdcMoKIiIhIDzjSRURERKQHTLqIiIiI9IBJFxEREZEeMOkiIiIi0gMmXURERER6YNTQAVD9kMvlCAsLQ2RkJFJTUyGVSuHp6Yn58+fD1dW1ocOj/3f27FnExMQgKSkJycnJSEtLA/DgprcODg5qt0tNTcXXX3+NEydO4N69e+jQoQNGjhyJefPmQSqVqt3ujz/+wJYtW3DhwgUAQPfu3TF9+nQ89dRTun1gBODB+/DkyZM4evQoEhMTkZaWhsLCQnTo0AGDBw9GYGAg7O3tVW7LY9x0/Prrrzhx4gQuXLiA7OxsFBQUoHXr1vDw8MDkyZMxdOhQldvxGLc8XDKiGZLL5QgMDERsbCxsbW3h5eWFO3fu4NSpUzA2Nsb69esxaNCghg6TAMyfPx9HjhypUl5d0pWSkoIXXngBBQUFcHNzg5OTE86cOYO0tDS4uLjgp59+gqWlZZXtvvzyS3z33XcwMTERj//x48dRWlqKBQsW4OWXX9btgyPExMRgxowZAAB7e3v06NEDAHDmzBncvn0blpaWCAkJQe/evZW24zFuWkaOHInU1FTIZDK0b98eZmZmSE1NRXJyMgBg5syZePvtt5W24TFuoQRqdtatWyfIZDJhwoQJwv3798Xy3377TZDJZMKAAQOE/Pz8BoyQFNavXy+sWbNGOHTokHDr1i3B29tbkMlkQmpqqsr2ZWVlwsiRIwWZTCasX79eLC8pKRFeeuklQSaTCcuWLauyXUJCgiCTyQRPT0/h0qVLYvmlS5cET09PQSaTCf/++6/uH2ALFxMTI7z22mvC6dOnlcqLi4uFd955R5DJZMLQoUOF0tJSsY7HuOlJTExU+ZkaHx8v9OrVq8pzz2PccjHpambkcrnQr18/QSaTCWfOnKlSP3v2bEEmkwmbN29ugOioJjUlXQcPHhRkMpng5+cnVFRUKNVlZmYKrq6ugpubm5CTk6NUN3fuXEEmkwkhISFV+ty4caMgk8mEoKAg3T0QqlFxcbHQt29fQSaTCSdPnhTLeYybl8WLFwsymUz47rvvxDIe45aLE+mbmcTEROTm5sLBwQEeHh5V6kePHg0AKk9pUeMXFRUFAHjqqacgkUiU6tq1a4e+fftCLpfj2LFjYnlpaSmOHz8OABg1alSVPhWvib/++gulpaX1FTo9xNTUFJ07dwYA3L59WyznMW5ejIweTJ02MTERy3iMWy4mXc3M+fPnAQBubm4q6xWT6BXtqGlRHDd3d3eV9YrjrphgCwBXrlxBSUkJbGxsYGdnV2UbOzs7WFtbo7i4GNeuXdN90KRSeXm5eOFE27ZtxXIe4+bj3Llz+OOPP2BoaAgfHx+xnMe45eLVi81Meno6AKBDhw4q6xXlubm5KCgogIWFhd5io7qr6fi2b98eAMQvc022UdTl5uYiLS0NMplMV+FSNSIjI5GTkwNbW1v06dNHLOcxbrp27NiB+Ph4yOVypKWl4d9//4WRkRFWrFiBbt26ie14jFsuJl3NTEFBAQDA3NxcZX3ly5CZdDU9NR1fxfFUtNNkG+B/r4vK21H9uXnzJlatWgUAeP3115VOPfEYN12JiYnYtWuX+G9zc3MsWbIEEyZMUGrHY9xy8fRiMyP8/wogD88ToOZF3fEVVKwAo8lrQtV2VD/u37+P+fPnIzc3F6NGjYK/v7/KdjzGTc9HH32ECxcu4J9//sHu3bsxevRoLF26FPPmzUNJSUmV9jzGLQ+TrmZG8QupsLBQZX1RUVGVttR0KH7J1nR8Kx/bml4TAFBcXFxlO9K94uJizJs3DxcuXMCAAQOwevXqKm14jJs+qVSKHj164OOPP4a/vz+OHj2K0NBQpXqAx7glYtLVzCgmWN66dUtlvaLc2tqab8wmSNPjW3mirWLFc3XbqNuOdEuxeGV8fDx69+6NdevWKZ1WVOAxbl7GjBkDQPmKcR7jlotJVzOjWPE6JSVFZf3Zs2cBAC4uLnqLiXRHcXwVx/FhiuPevXt3sezRRx+Fqakp7t69i4yMjCrbZGRk4O7duzAzM8Ojjz5aD1FTeXk5Fi1ahD///BM9evTAhg0b1P7o4TFuXtq0aQMAyMnJEct4jFsuJl3NTJ8+fWBtba10C4rK9u/fDwDw9fXVd2ikA4p7uB04cKBKneJWT0ZGRnj88cfFclNTU3h7ewP43/Gv7I8//gAADB48WOXIC9WNIAhYsmQJDhw4gC5duiA0NBRWVlZq2/MYNy+xsbEAgE6dOollPMYtF5OuZsbIyAgBAQEAgJUrVyI/P1+s27dvH6Kjo2FjY1PlahpqGoYNG4bOnTvjwoUL2Lhxo1gul8uxbNkyyOVyTJgwQfx1rTBr1iwAwHfffYfLly+L5ZcvX8b3338PAAgMDNTDI2h5Pv74Y+zatQuOjo4ICwurcmwexmPctCQlJWHXrl0qFySNjo7Gl19+CQBKF0zwGLdcvOF1M1RaWopZs2YhLi5OvOF1VlYWEhISYGxsjHXr1in9gqKGc/ToUaxbt078d0pKCuRyOXr06CH+Wh0yZAiCgoLENsnJyZg2bRoKCwvh7u4OJycnnD59Wlyb56effkKrVq2q7OuLL77A+vXrlX4xx8TEoKSkBPPnz8err75az4+25Tl8+LB47Pr37692rs3w4cMxfPhw8d88xk2H4hi3atUK7u7usLW1xf3793H16lXcuHEDgOobXvMYt0xMupqp0tJShIaGYs+ePUhNTYVUKkXfvn0RFBSkdrV60r+dO3di8eLF1bYZN24cPvnkE6Wy69evY+3atThx4gTu3buHDh064KmnnsL8+fOrvUBi37592LJli7jStYuLC6ZPn67ytiJUd5ocXwB4+eWXsWDBAqUyHuOmITMzE7/++ivi4+Nx48YN5OTkwMDAAO3atUOvXr0wadIkeHp6qtyWx7jlYdJFREREpAec00VERESkB0y6iIiIiPSASRcRERGRHjDpIiIiItIDJl1EREREesCki4iIiEgPmHQRERER6QGTLiIiIiI9YNLVDOXl5eGDDz7AsGHD4ObmBhcXF0ybNq2hwyKiWhg2bBhcXFxw8uTJhg6FWggXFxe4uLjg5s2bDR1Ks2XU0AE0B1OnTsWpU6cQGxsLa2trAMDixYuxc+dO7Nq1C66urnqNJygoCHFxcTAzM4OLiwtMTU0hk8k02tbFxUX829/fHx9++KHatgUFBRg8eDAKCwsBPHgeli1bprb9iRMnsGfPHiQkJCArKwuCIKBt27bo27cvnn32WQwaNEjttsOGDUNaWprK26VUdvLkSfGG30eOHIGDg4NSfVFREX799VccOnQIly5dQn5+PiwtLdGmTRs8+uij8PT0xJAhQ9ClSxeV+6+Jvb09oqKiamynqt+FCxdizpw5KtslJSXhueeeAwDx1h/Tpk1DXFycxvtSUNxW6ObNm/D19VWqk0gkkEqlsLS0RKdOneDm5oYnn3wSffv2rfV+iKh2wsPDcf/+fYwbN67KZxcpW7t2LQBg+vTpsLKyauBoNMekq45KS0uRlJSELl26iAkXAJw6dQpSqVQpidGH//77T0y4fv/99zq9cf/44w8sWbIE5ubmausVCVd17t+/j0WLFuHo0aMAAKlUCkdHRxgYGODGjRvYvXs3du/ejSFDhuDzzz9XeZNXXbhx4wZmzZol3oTW2toa3bp1g0Qiwc2bN3HlyhUcOXIEly9fxkcffaSyj86dO6NNmzZq9/HII49oHd/GjRsxadIktG7dWqP2MpkMZWVlVcr/++8/5Ofnw9bWFp06dapS37lz5ypl7u7u4g22S0pKkJubi7i4OMTFxSEsLAxubm4IDg7W++u5JXN0dISJiYna9x81P1u2bEFaWhr69evHpKsG33zzDYAHPyKZdLUgKSkpKCkpURoJuHPnDq5fvw5vb28YGhrqNZ5Lly4BALp161anN22XLl1w+fJlHDx4EGPGjFHZZseOHUptVSkqKsILL7yA8+fPw8bGBm+//Taefvpp8Qu+tLQU+/btw6pVq3Ds2DG88MIL+OWXX3T+RSMIAl555RXcuHEDnTt3xooVKzBw4EClNhcuXMD+/ftR3e1IX3rpJYwfP16nsQGAoaEh8vLysGHDBrz55psabfPee++pLFeMgD3++ONVbpStzldffVXl9XLv3j0cPHgQ3333Hc6ePQt/f39s3rwZvXv31qhPqpvNmzc3dAhEpGOc01VHiYmJAIA+ffqIZQkJCVXK9KWkpAQA6py0jBs3DgCwc+dOlfXXrl1DYmIiunbtip49e6rtJzg4GOfPn4eVlRV+/PFHjBs3Tky4AMDExARjx47FDz/8ACsrK5w/f17jRKE2kpKScO7cOQDAp59+WiXhAh6cWn311Vfx2muv6Xz/NVEktj/88AMyMzP1vn9VWrduDX9/f+zevRs9e/ZESUkJFixYgKKiooYOjYioSWLSVUeKpKvySNepU6cA1D3punv3Lr744gv4+fmhV69e6N27N5555hmsXbsW9+/fV2q7c+dOuLi44J133gEAxMXFiZMitZmM279/f9jb2+PkyZMq5zIpRrmqG/VJT08X2y1atKjKPKnKunTpgoULFwIAtm/fjvT09FrFWxPFKUUAjfIU2ZAhQ+Dl5YXi4mJx2LyxsLKywhdffAEjIyPcuXMHv/76a637uH79OhYuXAhvb294eHjgqaeewtdff42SkhK88847cHFxEedoPCw/Px/ff/89JkyYgL59+4rbf/zxx7hz547KbSpPQk9PT8e7774LHx8fuLu7Y9iwYfjkk0+Qn5+vNt6KigpERkZi5syZGDBgANzd3eHj44M33ngDKSkparfZuXMnpk2bhv79+8PNzQ0DBgzAM888gyVLliA2NrZWz5m6ifRr164V3+vl5eUIDw/HM888g549e8LLywsvvfQSkpOTa7Uv4MF8SBcXFwwbNgzAg88Uf39/9O7dG3369EFAQAD+/PNPldsqPn+mTZuG8vJybNmyBePHj0efPn3g4uKCvLw8se3ly5exePFiDBs2DO7u7vDy8sILL7yAiIgIlJeXq+y/8gTvhIQEzJkzB/3790fPnj0xZswY/PDDD2q3TUhIwKpVq/Dcc89h8ODBcHd3h7e3N+bMmYPo6Ohqn5M7d+5g+fLlGDJkCDw8PDBs2DB89NFHuHfvntJxqOs+Fc+f4rM2ICBA6fNb3T5ef/11PP7443B3d0e/fv0wY8YMHDhwoNrHdPDgQUyZMgW9e/eGp6cnpk6disOHD1e7TXUeft0cOXIE06ZNg6enJ3r16gV/f3/8/vvv1faRnZ2Nzz//HM888wx69+6Nxx57DH5+fvjqq6+qfNcpnncFX19fpedK8TkyZMgQuLi4ICkpqcr+nn32Wbi4uGDw4MFV6m7evAkXFxf06dOnyvQNuVyOn376CVOmTIGXlxc8PDzg6+uL9957D9evX1f52Cq/TkpKSph01Zbig1Dx36FDhwAAw4cPF8u2bt0KAJg5c6ZYVturBy9duoQxY8Zg/fr1uHLlCpycnODg4IBLly7hm2++wdixY5Gamiq2t7W1RZ8+fcT5OpaWlujTp4/4X23nSUkkEowbNw6CIGD37t1KdeXl5di9ezeMjIzUnnoEgH379qGsrAxWVlbiyFl1FOfmy8rKsG/fvlrFWxNLS0vxb0Wi3Ngoks4dO3bgypUrDRyNMkdHRwwdOhTAgw/V2jh9+jTGjRuHvXv34v79++I8um+//RbTp0+HXC5Xu+3ly5fx7LPPYs2aNTh37hysra3RuXNnpKenY/PmzRg7dizOnz+vdvsLFy5gzJgx+O233/DII4+gXbt2SEtLQ1hYGGbNmqVyTlx+fj5mzZqFt956C8ePH4eRkRG6deuGwsJC/P777/D398eePXuqbPfuu+9i8eLFiIuLg6mpKbp3747WrVvj5s2b2L59O7Zv316r560mZWVlmDNnDoKDg1FSUoJHH30UxcXFOHr0KKZOnYozZ85o3XdwcDAWL16M9PR0ODs7w8jICCdPnsTs2bMRFhamdjtBELBgwQJ89NFHyM3NhbOzs9IcxX379mHMmDHYuXMn7t69C5lMhlatWiE+Ph5Lly7FSy+9JI7Wq3L48GEEBAQgPj4eDg4OaNOmDc6fP48PPvgAr776qsrEKygoCKGhoUhNTYW1tTVkMhkEQcCxY8cwd+5cfPnllyr3df36dYwfPx6//PIL7ty5g0cffRQWFhb44Ycf4O/vr5RI1nWfis9vxVkAmUym9Pn98DzMzz77DFOnTsW+fftQWFiIrl27wsTEBDExMXjllVewfPlylXF99dVXWLBgARITE2Fubo5OnTrh0qVLCAoK0snp7G+++Qbz588Xv7OMjIxw5swZvPHGG/jhhx9UbnPq1CmMHj0aGzZswNWrV9GuXTvY2dnh6tWrWLduHZ577jncunVLbN+xY0elAQ13d3el56pjx44AHgwcAA8u4KosJycH//33H4AHSbViSo6Cor2npyeMjP43Ays/Px8vvvgi3n//fSQmJqJ169bo1q0bsrKysG3bNowZMwbHjh1T+9yUlJTghRdeAASqlYULFwrTp08Xpk+fLowdO1aQyWTCE088IZZNmjRJkMlkwsCBA8Wy6dOnC8HBwRrvo6SkRHjqqacEmUwm+Pv7C2lpaWLdtWvXhGeeeUaQyWTCuHHjhPLycqVtd+zYIchkMuGFF17Q6vHJZDJBJpMJZ86cEVJTUwUXFxfB19dXqKioENscPXpUkMlkwksvvSQIgiC8/fbbgkwmE95//32lvubOnavUThOzZ88WZDKZMG/ePKXyoUOHCjKZTPj666+r3T42NlZ8DKmpqWL5/fv3hb59+woymUzo37+/EBoaKly7dk3juBT737Fjh8bb1KbfP/74QxAEQZg3b54gk8mEBQsWKLU7c+aM+Lhq8sILLwgymUx4++23q22Xmpqq8rlSJzw8XJDJZIKHh0eNbRWKiorExzhr1iwhOztbrEtOThZ8fHwENzc3lcc2Pz9fGD58uCCTyYTXXntNuHXrllh3//598XU3YsQIobS0VGlbxT7d3NyEN998U8jLyxPr/v77b6Fnz56CTCYTtm/fXiXmV199VZDJZMLYsWOFM2fOiOXl5eVCeHi40L17d8Hd3V24fPmyWHfu3DlBJpMJffr0EWJjY5X6q6ioEOLi4oTff/9d4+et8mN4uL+vv/5afGxDhw4VkpKSxLrs7Gxh4sSJgkwmE6ZOnVqr/SneO66urkKPHj2En3/+WXzfy+Vy4csvvxRkMpnQvXt34fTp00rbKj53evToIfTr10+IiYkR60pKSoTy8nLh0qVLgoeHhyCTyYQlS5YIBQUFYpu///5bfH9++OGHVWJTvE7d3NyEd999VygsLBTrDh06JB7PkJCQKttu27ZNuHHjRpXy48ePCwMHDhRkMlmVxyMIguDv7y9+zt68eVMsv3btmuDn5ye+blW9z7Tdp7pjXtmPP/4ofo49/Jr6+++/xf4f/qw6fvy4+Dxu2rRJ/O6Qy+XCV199JT4eTT8PFBSvGzc3N6Fnz57Cnj17xDq5XC4sW7ZMkMlkQq9evYT79+8rbZuRkSH069dPkMlkwgcffCDcu3dPrLt9+7YQGBio9vusplgVr8mZM2cqle/bt0+QyWTC448/LshkMmHr1q1K9W+88Yb4HFX2zjvvCDKZTBgwYIAQHx8vlt+/f194/fXXxfd/5e9rQfjf+7VHjx7C0KFDBY501dJnn32G8PBwhIeH49lnnwXw4FeNomzWrFkAgClTpohl4eHhKoeH1dm3bx+uXr0KExMTfP3117CzsxPrOnXqhK+++gqGhoY4e/ZsrZYnqC0HBwf069cPqamp4jw14H+nFidMmFDt9opfJ05OThrvU3G1XeVfNrpgaWmJjz76CCYmJrh79y4++eQTjBgxAv369cOLL76ItWvXVjtiorB48WKlkc6H/1N3ikxTb7zxBgwNDXHgwIE6jVTUB3t7ewAPfrFVd2qusn379iEtLQ22trb48ssvla78dHNzwyeffKJ2pOunn37CjRs30KdPH3z22Wdo3769WKc4nq6urrh27RoOHjyosg9HR0d89NFHSiO9gwYNEl+7D5/qOXPmDP744w9YW1vj+++/h4eHh1hnYGCA6dOnY+rUqSgtLUV4eLhYp7iQZMCAAeIvbAWJRAIvLy+MHj26uqeq1uRyOVavXg13d3exrE2bNli6dCmAB6efHj41o4mysjJMmDABkydPhkQiAQAYGRnh1VdfxaBBg1BRUYH169er3La8vBzLly9XmjNpYmICAwMDbNq0CSUlJZDJZPjggw8glUrFNoMGDcLbb78NAPj555+RnZ2tsn8HBwd88MEHSnNWhw8fjnnz5gEANm3ahNLSUqVt/P394ejoWKUvb29vvP766wCqzl09efIkTp8+DRMTE3zzzTfiax/432dwRUWFyhi13acmioqK8PXXXwN4MDf14dfUoEGDsGLFCgDAhg0blOoUx+ypp57CzJkzYWDw4OvfyMgIr7zyCvr161freCqTy+V46aWX8Mwzz4hlRkZGWLx4Mdq0aYPCwsIqp8q///575Obm4umnn8bSpUuVrkJ85JFHsGbNGrRv3x5xcXH4999/axWP4n146tQppdeE4jS/4jXz8Gl/RYyV38c3b94Uz/gsX74cnp6eYp2lpSVWr14NBwcH5OfnK30uVFZeXo41a9bw9GJdHD9+HACUzgvHxMQAgMqJ2ppSDFE+/fTT6NChQ5X6Rx99VDx/Xt1wpi4ovpwUHxC5ubmIiopCmzZt8MQTT1S7bUFBAQAofbjWRNFW0y/12njqqacQGRkJf39/cXmPe/fu4cSJE/jmm28wZswYzJ07Fzk5OWr76Ny5s9JQ9sP/KYa2tdW1a1fxlO3nn39ep750rfJxVBzbmijmAI0aNUrpFK+Ct7e30hdaZfv37wfw4AtM1VXAhoaG4jpj6uZLTZo0CcbGxlXKFacnHp6H8ccffwAAhg4dqpTkVTZixAgAUPoCURz3f//9F9euXVO5na65uLgoffgrKJb/EARBaS5jbUyfPr3a8uPHj6s8NWtpaSk+Pw9TvBZmzJghJnOVjR07Fra2tpDL5eLn6MOmTZsmJguVTZ06FUZGRsjOzsbZs2er1CumZSxYsADTpk3DlClTMGXKFGzZsgUAqszTU8Q6aNAgpR+9Cs7OzjWuXVfbfWri5MmTuHv3Luzt7eHj46OyzdChQ2FsbIyrV6+KF+UUFhYiPj4eAMR1DB+m7pjXxvPPP1+lzMzMTFyr8uHXo2L+2aRJk1T2Z2lpKa7fWNs5kfb29nB0dERRURFOnz4tlsfGxsLMzAzjx49Hu3btEB8fLybQly9fxp07d9C6dWv06NFD3Oavv/5CRUUF7O3t8dRTT1XZl5GRkfj8qftO7tq1Kx577DEuGaGt0tJSJCQkwNnZWSkxiomJgVQqxWOPPaZ131evXgWAahc0lclkOHToUL3P/XnqqaewcuVK7N+/H++99x727NkDuVyOZ599VuWXWWUWFhYAoNFaXgqKtqq+oHXB2dkZH374IT744ANcuXIFZ8+eRUJCAqKjo3H79m1ER0dj5syZ2L59u9L5fIX6WjKisldeeQV79+5FbGws/v77b5WTPRtC5URLcWxrongtd+/eXW2byhOIK1MsAhseHo6IiAiV2ypGRNSNjKpakwx4MIcGqPraVOwzLi4OU6ZMUbmtYs5R5X326tULnp6eSEhIwKhRo9C3b194eXnhscceg6enZ728ntU9NolEAltbW2RkZNTqvadgZGSERx99VGVd165dATwYccnIyKgymtO5c2eV75v79++LFz2o+1wzNjaGs7MzsrOz1X6uqdu2VatWaN++PdLS0nD58mWlZU0+++wzhISEVLsUTG5urtK/Fa/b6i66cXFxUbs4sTb71IRiNP7+/ftqX5+VZWZmon379rhx44Y4301xDB+mrlxTNjY2SmtVVqZ4v1X+DMnMzBR/4H722WcqXzcAxIuqtDn70b9/f6SmpiI2NhZeXl7IzMzEtWvX4O3tDRMTE/Tv3x+//fYbzp07Bzc3NzGx8/LyUkruK38nq/rBoKgDHvyQKy8vr/JDUXEhGZMuDaWkpOCDDz4Q/11SUoKioiLk5OSIL35BEHDt2jVIpVKlXxNz587FkCFDNN6X4oXZtm1btW0Ui3BqOuKgLTMzM4wePRrbtm3DgQMHxBEvTRKP9u3bIyUlpVa/thUjDw+P8CneAOquUFKoXK/uTQw8+GLq0qULunTpgmeffRalpaUIDg7GTz/9hHPnzuHAgQN4+umnNY5blzp27IipU6ciLCwMX3zxRbUr9euTIjEyNzfXOIlQfOlXl6SpqissLBRPOyoSoeoUFxerLFe3dIri9fTwl+K9e/cAPHisNd2BoPI+DQwMsH79emzYsAG7d+9GfHy8OLJgamoKPz8/vPnmm7CxsanxsWiquhFkxeOr7hSYOjY2NmrXF6z8maTqs0ddTJXbVreAcE2fa4ovb3WxpaWlKW37+++/Y+PGjTAwMEBQUBCefPJJODg4QCqVwsDAACdOnMCLL75YZdRO29dtXfapCcXrMy8vT6MLghTLuyjOHBgZGalNjKr7vtGEJq/Hyu+3yhciaDKVQt17vDoDBgzA9u3bERsbiwULFoiT5AcMGCD+/7fffkNsbKxS0qWoV6jNd7IgCCgsLKxy8Zri+WHSpaH79++rfJHn5uZWKS8sLFQqUzc/QR3FmzkrK0ttG8WvRk1HHOpi/Pjx2LZtG9auXYu0tDTxfo416du3L6Kjo5GYmIjS0lKl9blUKSkpwT///AOg6nIbinP91V0xBCj/eqzNFZsmJiZYunQpDhw4gOzsbPz7778NlnQBD0bUIiIicPbsWfzxxx8q54fomyKJqM0oruKDprofB6rqzM3NYWhoiPLyckRGRlY7UqZLinjffffdWp9usbS0xBtvvIE33ngD169fx6lTp3D8+HEcOnQIO3bswLVr1/DDDz+oPD3WmNy9e1flL3VA+TOpNp89ldveuXNH7anbmj7XsrOz4ezsrLJOEVvlbRU/EmfMmKHy9mHqRpu0fd3WZZ+aUDy24cOH49tvv9V4O8WPpLKyMuTm5qpMvKr7vqkPlZO0+Pj4ellVXjEv6/Tp0ygqKhKnBCiSKsU0oJMnT2LmzJniyOXD8zJr852suJWaOo373d+I9O/fHxcuXBD/8/b2BvDgdKKibOLEiQCAX375RaltbU9HKT5ULl68qLaNoq66ta90pXfv3nB2dhZ/+dc0gV5h1KhRMDIywr1796osO6HK7t27kZeXByMjI4waNUqpTnG6o6bJ7opRkfbt29c6ITU0NBRXZa9uGQN9sLGxQWBgIADgyy+/rHGEr76lpqaKt3F6+H6N1dHkuKkayVKMRAL/W/dOHxQ/Juq6rEinTp0wfvx4fP7559i2bRskEglOnTql0ahdQysrK1M7L01xeb25uXmt5i+2atVKHAlQ97lWVlYmnsZR97mmbtv79++L85cqb6u4cbOquW8AxB95D1O8bqs7XurqtN2nJhSnsE6fPl2rUUwnJycxiX54iQQFdeX1pUOHDmKiVZfnpDrt2rWDs7Mz5HI5EhIScPLkSVhaWooXn9jb28PBwQEJCQlITk5Gbm4ubG1tq5zGrvydrO6UseK12alTp2rvRMOkSwvl5eU4ffo0OnfurDTcnZiYCFNTU7i5udWpf8WpyN9//x23b9+uUn/9+nVxraTanLasi9mzZ2PgwIHw9vbWeATIwcFBTDg//fRTtbcKAh5MYPzss88APEjqHp5crXic//zzj/jB/LDS0lL89ttvSu0V7t27V+Wqpofl5uaKbxx182X0afr06XjkkUdw/fp1bNu2rcHiuH//Pt544w2UlZXhkUcegb+/v8bbKib7/vHHHyovjjhx4oTa03iKxHvz5s1azU3SxsiRIwE8WItMsZZPXXXv3l0cdVX1fm6M1K3ZpJgE7u3tXe3pe1UU78nw8HCVX1yRkZHIysqCiYmJ2guRfvjhB5Xb/vTTTygrK0ObNm2UPn8Vp5dVPe/Z2dlqfwwqXrfHjx9XuVDz1atX1f4Y0HaflbdVdyrN29sbVlZWuHPnTq0+E6RSqZgEqlsvS3Fs9cXQ0FC86GL9+vW1/mGpeK6qW9cN+N+oVUREBNLS0uDl5aWUFA0YMAAFBQXYuHEjAKi8itPHxwcGBga4efOmyoVky8rKxOevpgvMmHRp4dy5cygoKFA6BXbv3j1cvnwZPXv2rPE0Wk1GjRqFRx99FCUlJXjllVeQkZEh1qWmpoqLALq5uYkLVta38ePHIzw8HGFhYWrnBKiyePFiyGQy5OXlYerUqdi1a5dS8lNaWorIyEi88MILyMvLg0wmU7m8xqhRo9ClSxdUVFQgKCioypU/t2/fxmuvvYa0tDSYm5uLS3coJCYmYsSIEfj+++9V/or/559/8NJLL6GwsBAWFhY6v7xfG1KpVLysWZORQl27d+8eIiIiMGbMGJw5cwampqZYu3ZtrW4x9fTTT8Pe3h7Z2dl4/fXXcffuXbHu7NmzePvtt9VekBEQEAAnJydcv34dM2fOrDJaJggCkpOT8fHHH+tseQ1PT0+MHDkScrkcs2bNQlRUVJUv+Zs3byIkJERpcn9kZCS++uqrKj8s5HI5QkJCxBFcfZ0mrQsjIyNs374d27ZtEx97WVkZvvnmG/z9998wMDDAnDlzat3vzJkzYWpqivPnz2PFihVKt5M6ceIEVq1aBeDBcjvq5m7duHEDy5cvV0pKjhw5gu+++07cR+XPXy8vLwAPvtQrH5vU1FTMmTNHbXLTv39/PPbYYygtLcWCBQuUEq8bN27glVdeUXuaWNt9Av9bXkfdlXqWlpbibco+/PBDhIeHV+lPcWZB8XwqKI7Z/v37ER4eLo6UKY5tbe9aogtBQUGwtrbGqVOnEBQUpLTgN/BggCMhIQHvvvtulduj1fRcKShOJSqWlXl4vlZN9cCDEbGxY8cCgLg4qkJ+fj7eeecd3LhxA5aWljVOS+CcLi2ous3PP//8A0EQaryMWBMmJiZYu3YtZs6ciX/++Qe+vr7o1q0bKioqcOnSJVRUVMDBwQFfffVVo58fIpVK8eOPP+KNN97AX3/9hXfeeQcrV66Ek5MTJBIJbty4Ic6NGDx4ML744guV58ONjY2xbt06zJ49G5cvX8a4ceNgb2+Ptm3bIj8/H1evXkVFRQWkUik+//zzKiNVEokEGRkZWLNmDdasWQNra2vxUvBbt26JV9FYWFhgzZo1aNeuncrHs379erVX0imEhITobK7dxIkTER4ervWl/5p69dVXxS+rkpIS5ObmIj09XfzSdXNzQ3BwcK1voWRmZoYvvvgCM2fOxJ9//onHH38c3bp1Q1FREa5cuYJevXrBy8sLe/furfJatrS0REhICObOnYt//vkHY8aMQceOHdGuXTuUlJQgNTVVfO3U5pRnTT755BOUlpYiKioK8+bNg7W1NRwdHVFRUYHMzExxXsfLL78sbnP37l2sW7cO69atg7W1Nezt7SEIAm7evCnOQ3zzzTfVzmVqTNq3b48nn3wS7733Hr7++mt06NABqamp4lykN954A7169ap1v126dMEnn3yCt956C7/88gv27NkDZ2dn3Lt3T/yyHTx4sHhnBlUWLVqE1atXY+/evXB2dkZOTo44Ujp06FDMnDlTqX1gYCD27duH9PR0PPvss+jcuTMMDAxw6dIlWFpa4q233lK6QEpBIpHg008/xdSpU5GcnIzhw4crfQY7ODhg8uTJ2Lp1a5XXrbb7BB4smxEVFYXQ0FAcOnQI7du3h4GBAXx8fMSkaerUqcjNzcXatWsRHByML774As7OzjA2NhafD0EQqozYDB48GC+99BLWr1+P4OBgbNiwAXZ2duKxfffdd/Hxxx/XcBR1y87ODhs3bkRQUBCio6MRHR0NJycncV2vGzduiEnl/PnzlbYdO3YsVq1ahZUrV+LHH39EmzZtxDupVJ7S069fP0gkEvGzTF3Spah/eD6XwpIlS8S5mlOmTEGnTp3QqlUrXL58GUVFReJnnaolRipj0qUFVTe5VlVWF926dcOePXsQFhaGI0eO4Nq1a5BIJOjatSuefPJJvPjii/Uy8bA+WFlZISQkBMePH8eePXtw6tQpXL9+HYIgoG3btvD19cWzzz6rdt0Zhc6dO2P37t2IiIjA4cOHcfHiRWRmZsLMzAwuLi7w9vbGtGnTVM41efzxx7F9+3YcP34ccXFxSEtLw40bN1BSUiLeMmnQoEGYNGlStVdXXbt2rcZ1mHQ5/8rY2BivvfYa3njjDZ31qYriXn0SiQTm5uZo1aoVvLy84ObmhieffLJOPyZ69eqFnTt3Yu3atYiJicHFixfRsWNHzJ07F/PmzcNbb70FQPUyIZ06dcLu3buxfft2HDhwABcuXEBKSop4O6BevXph+PDhOvmxo2Bubo5169YhOjoaO3fuxOnTp3H+/HlYWFigXbt2GDBgAIYNG6Z0Cvupp55CRUUFTp48iUuXLuHq1auQy+Vo27YtBg8ejKlTp6qd49MYKUaof/75Z3G0pl+/fggMDKzTlIbRo0dDJpNh06ZNiI2NxYULF2BmZgZPT0+MHTsW48ePr3Y+zPDhw+Hu7o7169fjzJkzKCwshEwmg7+/P6ZOnVpl2/bt22Pbtm1Ys2YNjh8/juvXr6Nt27YYO3YsgoKCqr1CtVOnTti1axfWrl2L6OhoXL58GY888gief/55LFiwQFxs9OHXbV32qbin6K+//opLly7h5s2bEAShynSLoKAg+Pr64ocffkBcXByuXr0KAwMDtGvXDj4+PhgyZAiGDx9epf833ngDPXr0wObNm3H+/HlcvXoV3bt3x4wZMzB8+HC9J10A0LNnT+zbtw8///wzoqKicPnyZWRkZMDW1hbdu3dH3759MXz48CrPwYsvvgjgwSjz9evXlV6nlbVp0wYymQwXLlyAjY1NlR+OjzzyCLp27YpLly6hffv2apdLsbS0xObNm7Ft2zb89ttvuHjxItLT09GuXTt4e3tj9uzZ4uLe1ZEI1S0kQkRUz55++mlcunQJ69at0+mIFdXOyZMnERAQAHt7+3q904U2FF+UR44cES92aWhz5szBsWPHsGTJErULjhI9rHGfmyKiZu2ff/7BpUuXYGRkpLSgJVFjlpaWJq6ar5jDRaQJJl1EVK/+/fdfREREVLl6MS4uTpwUPHr0aKX7MhI1tBs3biAsLKzKbcHOnz+PuXPnQi6Xo2/fvkq3iyGqCed0EVG9ysrKwtKlS7Fs2TJ07NgRtra2uHXrlnhJvUwmw7vvvtvAURIpKygowCeffIJVq1ahXbt2aNeundLE/Y4dO+KTTz5p4CipqWHSRUT1yt3dHTNnzkRsbCxu3bqFlJQUmJmZoWfPnhgxYgSmTp1aq5uiE+mDvb095s+fj5iYGKSlpeH8+fMwNjZG9+7d8cQTT+DFF1/U6W2dqGXgRHoiIiIiPeCcLiIiIiI9YNJFREREpAdMuoiIiIj0gEkXERERkR4w6SIiIiLSAyZdRERERHrApIuIiIhID5h0EREREekBky4iIiIiPfg/3z6MvMYZQowAAAAASUVORK5CYII=",
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "fig = plt.figure()\n",
    "ax = fig.add_subplot(111)\n",
    "ax.plot(x, pmf, 'bo')\n",
    "ax.vlines(x, 0, pmf, lw=1)\n",
    "ax.set_xlabel('# of MOUSE NTD genes in propagated network')\n",
    "ax.set_ylabel('hypergeom PMF')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "389fb975",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.0015777682995173348"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prb = hypergeom.cdf(len(overlap_prop), M, n, N) \n",
    "prb\n",
    "(1-prb)/2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "959bc1c2-e14c-48ad-a36a-0c62028c219e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Generating CX\n"
     ]
    }
   ],
   "source": [
    "G_prox_cx = ndex2.create_nice_cx_from_networkx(G_prox)\n",
    "network_uuid_hier = G_prox_cx.upload_to('ndexbio.org', 'jiny', 'xhflxhfl12')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "34de9942-b86a-4c3f-b3df-44148e6ba806",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Generating CX\n"
     ]
    },
    {
     "ename": "HTTPError",
     "evalue": "401 Client Error: 401 for url: http://public.ndexbio.org/v2/network",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mHTTPError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[68], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m uuid \u001b[38;5;241m=\u001b[39m \u001b[43mG_prox_cx\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mupload_to\u001b[49m\u001b[43m(\u001b[49m\u001b[43mserver\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mpublic.ndexbio.org\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[1;32m      2\u001b[0m \u001b[43m                  \u001b[49m\u001b[43musername\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mhijiny01@gmail.com\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[1;32m      3\u001b[0m \u001b[43m                  \u001b[49m\u001b[43mpassword\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mxhflxhfl12\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/miniconda3/envs/r4/lib/python3.12/site-packages/ndex2/nice_cx_network.py:1444\u001b[0m, in \u001b[0;36mNiceCXNetwork.upload_to\u001b[0;34m(self, server, username, password, user_agent, client)\u001b[0m\n\u001b[1;32m   1442\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m   1443\u001b[0m     ndex \u001b[38;5;241m=\u001b[39m Ndex2(server, username, password, user_agent\u001b[38;5;241m=\u001b[39muser_agent)\n\u001b[0;32m-> 1444\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mndex\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msave_new_network\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mto_cx\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/miniconda3/envs/r4/lib/python3.12/site-packages/ndex2/client.py:469\u001b[0m, in \u001b[0;36mNdex2.save_new_network\u001b[0;34m(self, cx, visibility)\u001b[0m\n\u001b[1;32m    466\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    467\u001b[0m         stream \u001b[38;5;241m=\u001b[39m io\u001b[38;5;241m.\u001b[39mBytesIO(json\u001b[38;5;241m.\u001b[39mdumps(cx, \u001b[38;5;28mcls\u001b[39m\u001b[38;5;241m=\u001b[39mDecimalEncoder))\n\u001b[0;32m--> 469\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msave_cx_stream_as_new_network\u001b[49m\u001b[43m(\u001b[49m\u001b[43mstream\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mvisibility\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mvisibility\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/miniconda3/envs/r4/lib/python3.12/site-packages/ndex2/client.py:497\u001b[0m, in \u001b[0;36mNdex2.save_cx_stream_as_new_network\u001b[0;34m(self, cx_stream, visibility)\u001b[0m\n\u001b[1;32m    491\u001b[0m     route \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124m/network\u001b[39m\u001b[38;5;124m'\u001b[39m\n\u001b[1;32m    493\u001b[0m fields \u001b[38;5;241m=\u001b[39m {\n\u001b[1;32m    494\u001b[0m     \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mCXNetworkStream\u001b[39m\u001b[38;5;124m'\u001b[39m: (\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mfilename\u001b[39m\u001b[38;5;124m'\u001b[39m, cx_stream, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mapplication/octet-stream\u001b[39m\u001b[38;5;124m'\u001b[39m)\n\u001b[1;32m    495\u001b[0m }\n\u001b[0;32m--> 497\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mpost_multipart\u001b[49m\u001b[43m(\u001b[49m\u001b[43mroute\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfields\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mquery_string\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mquery_string\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/miniconda3/envs/r4/lib/python3.12/site-packages/ndex2/client.py:430\u001b[0m, in \u001b[0;36mNdex2.post_multipart\u001b[0;34m(self, route, fields, query_string, alt_version_endpoint, returnjsonundertry, returnfullresponse)\u001b[0m\n\u001b[1;32m    424\u001b[0m headers \u001b[38;5;241m=\u001b[39m {\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mContent-Type\u001b[39m\u001b[38;5;124m'\u001b[39m: multipart_data\u001b[38;5;241m.\u001b[39mcontent_type,\n\u001b[1;32m    425\u001b[0m            Ndex2\u001b[38;5;241m.\u001b[39mUSER_AGENT_KEY: \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_get_user_agent(),\n\u001b[1;32m    426\u001b[0m            \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mConnection\u001b[39m\u001b[38;5;124m'\u001b[39m: \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mclose\u001b[39m\u001b[38;5;124m'\u001b[39m\n\u001b[1;32m    427\u001b[0m            }\n\u001b[1;32m    429\u001b[0m response \u001b[38;5;241m=\u001b[39m requests\u001b[38;5;241m.\u001b[39mpost(url, data\u001b[38;5;241m=\u001b[39mmultipart_data, headers\u001b[38;5;241m=\u001b[39mheaders, auth\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_get_auth_tuple())\n\u001b[0;32m--> 430\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_return_response\u001b[49m\u001b[43m(\u001b[49m\u001b[43mresponse\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    431\u001b[0m \u001b[43m                             \u001b[49m\u001b[43mreturnjsonundertry\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mreturnjsonundertry\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    432\u001b[0m \u001b[43m                             \u001b[49m\u001b[43mreturnfullresponse\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mreturnfullresponse\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/miniconda3/envs/r4/lib/python3.12/site-packages/ndex2/client.py:219\u001b[0m, in \u001b[0;36mNdex2._return_response\u001b[0;34m(self, response, raiseforstatus, returnfullresponse, returnjsonundertry)\u001b[0m\n\u001b[1;32m    217\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mdebug_response(response)\n\u001b[1;32m    218\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m raiseforstatus:\n\u001b[0;32m--> 219\u001b[0m     \u001b[43mresponse\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mraise_for_status\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    220\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m returnfullresponse \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mTrue\u001b[39;00m:\n\u001b[1;32m    221\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m response\n",
      "File \u001b[0;32m~/miniconda3/envs/r4/lib/python3.12/site-packages/requests/models.py:1021\u001b[0m, in \u001b[0;36mResponse.raise_for_status\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m   1016\u001b[0m     http_error_msg \u001b[38;5;241m=\u001b[39m (\n\u001b[1;32m   1017\u001b[0m         \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;132;01m{\u001b[39;00m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mstatus_code\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m Server Error: \u001b[39m\u001b[38;5;132;01m{\u001b[39;00mreason\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m for url: \u001b[39m\u001b[38;5;132;01m{\u001b[39;00m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39murl\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m   1018\u001b[0m     )\n\u001b[1;32m   1020\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m http_error_msg:\n\u001b[0;32m-> 1021\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m HTTPError(http_error_msg, response\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m)\n",
      "\u001b[0;31mHTTPError\u001b[0m: 401 Client Error: 401 for url: http://public.ndexbio.org/v2/network"
     ]
    }
   ],
   "source": [
    "uuid = G_prox_cx.upload_to(server='public.ndexbio.org',\n",
    "                  username='hijiny01@gmail.com',\n",
    "                  password='xhflxhfl12')\n",
    "\n",
    "G_prox_cx = ndex2.create_nice_cx_from_networkx(G_prox)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "77f01f30-ba72-4406-a547-3a6b14e506de",
   "metadata": {},
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "'NiceCXNetwork' object has no attribute 'get'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[50], line 4\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[38;5;28mtype\u001b[39m(G_prox_cx)\n\u001b[1;32m      3\u001b[0m __all__ \u001b[38;5;241m=\u001b[39m [\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mcytoscape_data\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mcytoscape_graph\u001b[39m\u001b[38;5;124m\"\u001b[39m]\n\u001b[0;32m----> 4\u001b[0m \u001b[43mnetworkx\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcytoscape_graph\u001b[49m\u001b[43m(\u001b[49m\u001b[43mG_prox_cx\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mname\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mMM.cyjs\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mident\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mid\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/miniconda3/envs/r4/lib/python3.12/site-packages/networkx/utils/backends.py:412\u001b[0m, in \u001b[0;36m_dispatch.__call__\u001b[0;34m(self, backend, *args, **kwargs)\u001b[0m\n\u001b[1;32m    409\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21m__call__\u001b[39m(\u001b[38;5;28mself\u001b[39m, \u001b[38;5;241m/\u001b[39m, \u001b[38;5;241m*\u001b[39margs, backend\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs):\n\u001b[1;32m    410\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m backends:\n\u001b[1;32m    411\u001b[0m         \u001b[38;5;66;03m# Fast path if no backends are installed\u001b[39;00m\n\u001b[0;32m--> 412\u001b[0m         \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43morig_func\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    414\u001b[0m     \u001b[38;5;66;03m# Use `backend_name` in this function instead of `backend`\u001b[39;00m\n\u001b[1;32m    415\u001b[0m     backend_name \u001b[38;5;241m=\u001b[39m backend\n",
      "File \u001b[0;32m~/miniconda3/envs/r4/lib/python3.12/site-packages/networkx/readwrite/json_graph/cytoscape.py:142\u001b[0m, in \u001b[0;36mcytoscape_graph\u001b[0;34m(data, name, ident)\u001b[0m\n\u001b[1;32m    139\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m name \u001b[38;5;241m==\u001b[39m ident:\n\u001b[1;32m    140\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m nx\u001b[38;5;241m.\u001b[39mNetworkXError(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mname and ident must be different.\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[0;32m--> 142\u001b[0m multigraph \u001b[38;5;241m=\u001b[39m \u001b[43mdata\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mmultigraph\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m    143\u001b[0m directed \u001b[38;5;241m=\u001b[39m data\u001b[38;5;241m.\u001b[39mget(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mdirected\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m    144\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m multigraph:\n",
      "\u001b[0;31mAttributeError\u001b[0m: 'NiceCXNetwork' object has no attribute 'get'"
     ]
    }
   ],
   "source": [
    "import networkx as nx\n",
    "type(G_prox_cx)\n",
    "__all__ = [\"cytoscape_data\", \"cytoscape_graph\"]\n",
    "networkx.cytoscape_graph(G_prox_cx, name='MM.cyjs', ident='id')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "8ba56538-afd0-4459-9d6f-d251a2cfc6bd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdin",
     "output_type": "stream",
     "text": [
      "NDEx server : http://public.ndexbio.org/ \n",
      "NDEx user name: hijiny01@gmail.com \n",
      "NDEx password: xhflxhfl12 ········\n"
     ]
    },
    {
     "ename": "InvalidURL",
     "evalue": "Invalid URL 'http:///rest/admin/status': No host supplied",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mInvalidURL\u001b[0m                                Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[43], line 9\u001b[0m\n\u001b[1;32m      7\u001b[0m USERNAME \u001b[38;5;241m=\u001b[39m \u001b[38;5;28minput\u001b[39m(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mNDEx user name: hijiny01@gmail.com\u001b[39m\u001b[38;5;124m'\u001b[39m)\n\u001b[1;32m      8\u001b[0m PASSWORD \u001b[38;5;241m=\u001b[39m getpass\u001b[38;5;241m.\u001b[39mgetpass(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mNDEx password: xhflxhfl12\u001b[39m\u001b[38;5;124m'\u001b[39m)\n\u001b[0;32m----> 9\u001b[0m network_uuid_hier \u001b[38;5;241m=\u001b[39m \u001b[43mG_prox_cx\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mupload_to\u001b[49m\u001b[43m(\u001b[49m\u001b[43mSERVER\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mUSERNAME\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mPASSWORD\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/miniconda3/envs/r4/lib/python3.12/site-packages/ndex2/nice_cx_network.py:1443\u001b[0m, in \u001b[0;36mNiceCXNetwork.upload_to\u001b[0;34m(self, server, username, password, user_agent, client)\u001b[0m\n\u001b[1;32m   1441\u001b[0m     ndex \u001b[38;5;241m=\u001b[39m client\n\u001b[1;32m   1442\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m-> 1443\u001b[0m     ndex \u001b[38;5;241m=\u001b[39m \u001b[43mNdex2\u001b[49m\u001b[43m(\u001b[49m\u001b[43mserver\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43musername\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mpassword\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43muser_agent\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43muser_agent\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1444\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m ndex\u001b[38;5;241m.\u001b[39msave_new_network(\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mto_cx())\n",
      "File \u001b[0;32m~/miniconda3/envs/r4/lib/python3.12/site-packages/ndex2/client.py:120\u001b[0m, in \u001b[0;36mNdex2.__init__\u001b[0;34m(self, host, username, password, update_status, debug, user_agent, timeout, skip_version_check)\u001b[0m\n\u001b[1;32m    117\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m    118\u001b[0m     version_url \u001b[38;5;241m=\u001b[39m urljoin(host, status_url)\n\u001b[0;32m--> 120\u001b[0m     response \u001b[38;5;241m=\u001b[39m \u001b[43mrequests\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[43mversion_url\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    121\u001b[0m \u001b[43m                            \u001b[49m\u001b[43mheaders\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m{\u001b[49m\u001b[43mNdex2\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mUSER_AGENT_KEY\u001b[49m\u001b[43m:\u001b[49m\n\u001b[1;32m    122\u001b[0m \u001b[43m                                     \u001b[49m\u001b[43muserAgent\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m+\u001b[39;49m\n\u001b[1;32m    123\u001b[0m \u001b[43m                                     \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43muser_agent\u001b[49m\u001b[43m}\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    124\u001b[0m     response\u001b[38;5;241m.\u001b[39mraise_for_status()\n\u001b[1;32m    125\u001b[0m     data \u001b[38;5;241m=\u001b[39m response\u001b[38;5;241m.\u001b[39mjson()\n",
      "File \u001b[0;32m~/miniconda3/envs/r4/lib/python3.12/site-packages/requests/api.py:73\u001b[0m, in \u001b[0;36mget\u001b[0;34m(url, params, **kwargs)\u001b[0m\n\u001b[1;32m     62\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mget\u001b[39m(url, params\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs):\n\u001b[1;32m     63\u001b[0m \u001b[38;5;250m    \u001b[39m\u001b[38;5;124mr\u001b[39m\u001b[38;5;124;03m\"\"\"Sends a GET request.\u001b[39;00m\n\u001b[1;32m     64\u001b[0m \n\u001b[1;32m     65\u001b[0m \u001b[38;5;124;03m    :param url: URL for the new :class:`Request` object.\u001b[39;00m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m     70\u001b[0m \u001b[38;5;124;03m    :rtype: requests.Response\u001b[39;00m\n\u001b[1;32m     71\u001b[0m \u001b[38;5;124;03m    \"\"\"\u001b[39;00m\n\u001b[0;32m---> 73\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mrequest\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mget\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mparams\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mparams\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/miniconda3/envs/r4/lib/python3.12/site-packages/requests/api.py:59\u001b[0m, in \u001b[0;36mrequest\u001b[0;34m(method, url, **kwargs)\u001b[0m\n\u001b[1;32m     55\u001b[0m \u001b[38;5;66;03m# By using the 'with' statement we are sure the session is closed, thus we\u001b[39;00m\n\u001b[1;32m     56\u001b[0m \u001b[38;5;66;03m# avoid leaving sockets open which can trigger a ResourceWarning in some\u001b[39;00m\n\u001b[1;32m     57\u001b[0m \u001b[38;5;66;03m# cases, and look like a memory leak in others.\u001b[39;00m\n\u001b[1;32m     58\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m sessions\u001b[38;5;241m.\u001b[39mSession() \u001b[38;5;28;01mas\u001b[39;00m session:\n\u001b[0;32m---> 59\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43msession\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[43m(\u001b[49m\u001b[43mmethod\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mmethod\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43murl\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/miniconda3/envs/r4/lib/python3.12/site-packages/requests/sessions.py:575\u001b[0m, in \u001b[0;36mSession.request\u001b[0;34m(self, method, url, params, data, headers, cookies, files, auth, timeout, allow_redirects, proxies, hooks, stream, verify, cert, json)\u001b[0m\n\u001b[1;32m    562\u001b[0m \u001b[38;5;66;03m# Create the Request.\u001b[39;00m\n\u001b[1;32m    563\u001b[0m req \u001b[38;5;241m=\u001b[39m Request(\n\u001b[1;32m    564\u001b[0m     method\u001b[38;5;241m=\u001b[39mmethod\u001b[38;5;241m.\u001b[39mupper(),\n\u001b[1;32m    565\u001b[0m     url\u001b[38;5;241m=\u001b[39murl,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    573\u001b[0m     hooks\u001b[38;5;241m=\u001b[39mhooks,\n\u001b[1;32m    574\u001b[0m )\n\u001b[0;32m--> 575\u001b[0m prep \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mprepare_request\u001b[49m\u001b[43m(\u001b[49m\u001b[43mreq\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    577\u001b[0m proxies \u001b[38;5;241m=\u001b[39m proxies \u001b[38;5;129;01mor\u001b[39;00m {}\n\u001b[1;32m    579\u001b[0m settings \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mmerge_environment_settings(\n\u001b[1;32m    580\u001b[0m     prep\u001b[38;5;241m.\u001b[39murl, proxies, stream, verify, cert\n\u001b[1;32m    581\u001b[0m )\n",
      "File \u001b[0;32m~/miniconda3/envs/r4/lib/python3.12/site-packages/requests/sessions.py:486\u001b[0m, in \u001b[0;36mSession.prepare_request\u001b[0;34m(self, request)\u001b[0m\n\u001b[1;32m    483\u001b[0m     auth \u001b[38;5;241m=\u001b[39m get_netrc_auth(request\u001b[38;5;241m.\u001b[39murl)\n\u001b[1;32m    485\u001b[0m p \u001b[38;5;241m=\u001b[39m PreparedRequest()\n\u001b[0;32m--> 486\u001b[0m \u001b[43mp\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mprepare\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    487\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmethod\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmethod\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mupper\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    488\u001b[0m \u001b[43m    \u001b[49m\u001b[43murl\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    489\u001b[0m \u001b[43m    \u001b[49m\u001b[43mfiles\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mfiles\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    490\u001b[0m \u001b[43m    \u001b[49m\u001b[43mdata\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mdata\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    491\u001b[0m \u001b[43m    \u001b[49m\u001b[43mjson\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mjson\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    492\u001b[0m \u001b[43m    \u001b[49m\u001b[43mheaders\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mmerge_setting\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    493\u001b[0m \u001b[43m        \u001b[49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mheaders\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mheaders\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mdict_class\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mCaseInsensitiveDict\u001b[49m\n\u001b[1;32m    494\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    495\u001b[0m \u001b[43m    \u001b[49m\u001b[43mparams\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mmerge_setting\u001b[49m\u001b[43m(\u001b[49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mparams\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mparams\u001b[49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    496\u001b[0m \u001b[43m    \u001b[49m\u001b[43mauth\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mmerge_setting\u001b[49m\u001b[43m(\u001b[49m\u001b[43mauth\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mauth\u001b[49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    497\u001b[0m \u001b[43m    \u001b[49m\u001b[43mcookies\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mmerged_cookies\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    498\u001b[0m \u001b[43m    \u001b[49m\u001b[43mhooks\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mmerge_hooks\u001b[49m\u001b[43m(\u001b[49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mhooks\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mhooks\u001b[49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    499\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    500\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m p\n",
      "File \u001b[0;32m~/miniconda3/envs/r4/lib/python3.12/site-packages/requests/models.py:368\u001b[0m, in \u001b[0;36mPreparedRequest.prepare\u001b[0;34m(self, method, url, headers, files, data, params, auth, cookies, hooks, json)\u001b[0m\n\u001b[1;32m    365\u001b[0m \u001b[38;5;250m\u001b[39m\u001b[38;5;124;03m\"\"\"Prepares the entire request with the given parameters.\"\"\"\u001b[39;00m\n\u001b[1;32m    367\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mprepare_method(method)\n\u001b[0;32m--> 368\u001b[0m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mprepare_url\u001b[49m\u001b[43m(\u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mparams\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    369\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mprepare_headers(headers)\n\u001b[1;32m    370\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mprepare_cookies(cookies)\n",
      "File \u001b[0;32m~/miniconda3/envs/r4/lib/python3.12/site-packages/requests/models.py:445\u001b[0m, in \u001b[0;36mPreparedRequest.prepare_url\u001b[0;34m(self, url, params)\u001b[0m\n\u001b[1;32m    439\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m MissingSchema(\n\u001b[1;32m    440\u001b[0m         \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mInvalid URL \u001b[39m\u001b[38;5;132;01m{\u001b[39;00murl\u001b[38;5;132;01m!r}\u001b[39;00m\u001b[38;5;124m: No scheme supplied. \u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    441\u001b[0m         \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mPerhaps you meant https://\u001b[39m\u001b[38;5;132;01m{\u001b[39;00murl\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m?\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    442\u001b[0m     )\n\u001b[1;32m    444\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m host:\n\u001b[0;32m--> 445\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m InvalidURL(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mInvalid URL \u001b[39m\u001b[38;5;132;01m{\u001b[39;00murl\u001b[38;5;132;01m!r}\u001b[39;00m\u001b[38;5;124m: No host supplied\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m    447\u001b[0m \u001b[38;5;66;03m# In general, we want to try IDNA encoding the hostname if the string contains\u001b[39;00m\n\u001b[1;32m    448\u001b[0m \u001b[38;5;66;03m# non-ASCII characters. This allows users to automatically get the correct IDNA\u001b[39;00m\n\u001b[1;32m    449\u001b[0m \u001b[38;5;66;03m# behaviour. For strings containing only ASCII characters, we need to also verify\u001b[39;00m\n\u001b[1;32m    450\u001b[0m \u001b[38;5;66;03m# it doesn't start with a wildcard (*), before allowing the unencoded hostname.\u001b[39;00m\n\u001b[1;32m    451\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m unicode_is_ascii(host):\n",
      "\u001b[0;31mInvalidURL\u001b[0m: Invalid URL 'http:///rest/admin/status': No host supplied"
     ]
    }
   ],
   "source": [
    "#G_prox_cx.upload_to('ndexbio.org', 'hijiny01@gmail.com', 'xhflxhfl12')\n",
    "\n",
    "#Upload to NDEx. Enter your ndex username and password. \n",
    "# If you are new to ndex, make a new account on the website (ndexbio.org)\n",
    "G_prox_cx.set_name(d1_name+'_systems_map') \n",
    "SERVER = input('NDEx server : http://public.ndexbio.org/')\n",
    "USERNAME = input('NDEx user name: hijiny01@gmail.com')\n",
    "PASSWORD = getpass.getpass('NDEx password: xhflxhfl12')\n",
    "network_uuid_hier = G_prox_cx.upload_to(SERVER, USERNAME, PASSWORD)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0a6e3b51",
   "metadata": {},
   "source": [
    "### you used N as the total number of genes (~19,000), M as the number of known NTD genes, and s (sample size) as the number of genes from network propagation and the k (number of success in the sample) as the number of genes overlapping with the NTD genes in the network propagation result. Did I understand the method clearly?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "7b4c6ca5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         d1_seeds      d1_name       z_d1     gene\n",
      "CLTC            1  MM DNM gens  21.341222     CLTC\n",
      "HNRNPL          1  MM DNM gens  19.767824   HNRNPL\n",
      "VDAC2           1  MM DNM gens  19.187881    VDAC2\n",
      "CNPY1           1  MM DNM gens  19.097940    CNPY1\n",
      "FAM83G          1  MM DNM gens  17.947948   FAM83G\n",
      "...           ...          ...        ...      ...\n",
      "TTN             0  MM DNM gens   3.014179      TTN\n",
      "MRPL9           0  MM DNM gens   3.010165    MRPL9\n",
      "SLC4A10         0  MM DNM gens   3.007231  SLC4A10\n",
      "WIPF3           0  MM DNM gens   3.006245    WIPF3\n",
      "TMIGD1          0  MM DNM gens   3.005495   TMIGD1\n",
      "\n",
      "[439 rows x 4 columns]\n",
      "CLTC          CLTC\n",
      "HNRNPL      HNRNPL\n",
      "VDAC2        VDAC2\n",
      "CNPY1        CNPY1\n",
      "FAM83G      FAM83G\n",
      "            ...   \n",
      "TTN            TTN\n",
      "MRPL9        MRPL9\n",
      "SLC4A10    SLC4A10\n",
      "WIPF3        WIPF3\n",
      "TMIGD1      TMIGD1\n",
      "Name: gene, Length: 439, dtype: object\n"
     ]
    }
   ],
   "source": [
    "node_df[\"gene\"] =node_df.index\n",
    "print (node_df)\n",
    "print (node_df[\"gene\"])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2c6de719",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "5f142b35",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "d1_seeds\n",
      "d1_name\n",
      "z_d1\n",
      "gene\n",
      "['BMP7' 'CELSR1' 'DYNC2H1' 'EP300' 'FREM1' 'FREM2' 'GRHL2' 'GRHL3' 'MIB2'\n",
      " 'PAX3' 'SCRIB' 'SLC25A19' 'SMURF2' 'SS18' 'TBXT' 'TIAM1' 'TULP3']\n",
      "CLTC          CLTC\n",
      "HNRNPL      HNRNPL\n",
      "VDAC2        VDAC2\n",
      "CNPY1        CNPY1\n",
      "FAM83G      FAM83G\n",
      "            ...   \n",
      "TTN            TTN\n",
      "MRPL9        MRPL9\n",
      "SLC4A10    SLC4A10\n",
      "WIPF3        WIPF3\n",
      "TMIGD1      TMIGD1\n",
      "Name: gene, Length: 439, dtype: object\n",
      "         d1_seeds      d1_name       z_d1     gene Mouse_Gene\n",
      "CLTC            1  MM DNM gens  21.341222     CLTC          0\n",
      "HNRNPL          1  MM DNM gens  19.767824   HNRNPL          0\n",
      "VDAC2           1  MM DNM gens  19.187881    VDAC2          0\n",
      "CNPY1           1  MM DNM gens  19.097940    CNPY1          0\n",
      "FAM83G          1  MM DNM gens  17.947948   FAM83G          0\n",
      "...           ...          ...        ...      ...        ...\n",
      "TTN             0  MM DNM gens   3.014179      TTN          0\n",
      "MRPL9           0  MM DNM gens   3.010165    MRPL9          0\n",
      "SLC4A10         0  MM DNM gens   3.007231  SLC4A10          0\n",
      "WIPF3           0  MM DNM gens   3.006245    WIPF3          0\n",
      "TMIGD1          0  MM DNM gens   3.005495   TMIGD1          0\n",
      "\n",
      "[439 rows x 5 columns]\n"
     ]
    }
   ],
   "source": [
    "#print (node_df.head(15))\n",
    "for col in node_df.columns:\n",
    "    print(col)\n",
    "#node_df.rename(columns = {\"Gene\", \"seed\", \"geneSet\", \"z_d1\", \"Mouse_NTDgene\"})\n",
    "print (overlap_prop)\n",
    "print (node_df[\"gene\"])\n",
    "node_df[\"Mouse_Gene\"] = np.where(node_df[\"gene\"].isin(overlap_prop), \"MouseNTD\", \"0\")\n",
    "print (node_df)\n",
    "node_df.to_csv(\"/projects/ps-gleesonlab8/User/hiyoothere/NTD/5.Analysis/Network/output/NTD_DamMC2_coloc.txt\", sep = '\\t')\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "7c8a13da",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "83c7f765",
   "metadata": {},
   "outputs": [
    {
     "ename": "SyntaxError",
     "evalue": "invalid syntax (1744613418.py, line 5)",
     "output_type": "error",
     "traceback": [
      "\u001b[0;36m  Input \u001b[0;32mIn [2]\u001b[0;36m\u001b[0m\n\u001b[0;31m    oe = oe[!duplicated(oe$SYMBOL), ]\u001b[0m\n\u001b[0m            ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m invalid syntax\n"
     ]
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b30524bc",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "1eafc0a4",
   "metadata": {},
   "source": [
    "# Compute systems map from proximal subgraph"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "99f7da96",
   "metadata": {},
   "source": [
    "## 9. Convert network colocalization subnetwork to form used in community detection module"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "7a7e7e7a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>d1_seeds</th>\n",
       "      <th>d1_name</th>\n",
       "      <th>z_d1</th>\n",
       "      <th>sum_seeds</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CLTC</th>\n",
       "      <td>1</td>\n",
       "      <td>MM DNM gens</td>\n",
       "      <td>21.341222</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HNRNPL</th>\n",
       "      <td>1</td>\n",
       "      <td>MM DNM gens</td>\n",
       "      <td>19.767824</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VDAC2</th>\n",
       "      <td>1</td>\n",
       "      <td>MM DNM gens</td>\n",
       "      <td>19.187881</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CNPY1</th>\n",
       "      <td>1</td>\n",
       "      <td>MM DNM gens</td>\n",
       "      <td>19.097940</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FAM83G</th>\n",
       "      <td>1</td>\n",
       "      <td>MM DNM gens</td>\n",
       "      <td>17.947948</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GTF3C3</th>\n",
       "      <td>1</td>\n",
       "      <td>MM DNM gens</td>\n",
       "      <td>17.852848</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MFAP1</th>\n",
       "      <td>1</td>\n",
       "      <td>MM DNM gens</td>\n",
       "      <td>17.846594</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DPYSL2</th>\n",
       "      <td>1</td>\n",
       "      <td>MM DNM gens</td>\n",
       "      <td>17.695655</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LEMD3</th>\n",
       "      <td>1</td>\n",
       "      <td>MM DNM gens</td>\n",
       "      <td>16.780574</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TOP2B</th>\n",
       "      <td>1</td>\n",
       "      <td>MM DNM gens</td>\n",
       "      <td>16.752226</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C6orf132</th>\n",
       "      <td>1</td>\n",
       "      <td>MM DNM gens</td>\n",
       "      <td>16.370847</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRHR2</th>\n",
       "      <td>1</td>\n",
       "      <td>MM DNM gens</td>\n",
       "      <td>16.271097</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SMURF2</th>\n",
       "      <td>1</td>\n",
       "      <td>MM DNM gens</td>\n",
       "      <td>16.069396</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATAD3B</th>\n",
       "      <td>1</td>\n",
       "      <td>MM DNM gens</td>\n",
       "      <td>15.869375</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPEN</th>\n",
       "      <td>1</td>\n",
       "      <td>MM DNM gens</td>\n",
       "      <td>15.622895</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          d1_seeds      d1_name       z_d1  sum_seeds\n",
       "CLTC             1  MM DNM gens  21.341222          1\n",
       "HNRNPL           1  MM DNM gens  19.767824          1\n",
       "VDAC2            1  MM DNM gens  19.187881          1\n",
       "CNPY1            1  MM DNM gens  19.097940          1\n",
       "FAM83G           1  MM DNM gens  17.947948          1\n",
       "GTF3C3           1  MM DNM gens  17.852848          1\n",
       "MFAP1            1  MM DNM gens  17.846594          1\n",
       "DPYSL2           1  MM DNM gens  17.695655          1\n",
       "LEMD3            1  MM DNM gens  16.780574          1\n",
       "TOP2B            1  MM DNM gens  16.752226          1\n",
       "C6orf132         1  MM DNM gens  16.370847          1\n",
       "CRHR2            1  MM DNM gens  16.271097          1\n",
       "SMURF2           1  MM DNM gens  16.069396          1\n",
       "ATAD3B           1  MM DNM gens  15.869375          1\n",
       "SPEN             1  MM DNM gens  15.622895          1"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# compile dataframe of metadata for overlapping nodes\n",
    "node_df = pd.DataFrame(index=list(G_prox.nodes))\n",
    "node_df = node_df.assign(d1_seeds=0, d1_name=d1_name,)\n",
    "node_df.loc[list(np.intersect1d(D1_genes,node_df.index.tolist())), 'd1_seeds']=1\n",
    "node_df['z_d1']=z_D1.loc[list(G_prox.nodes)]['z']\n",
    "node_df['sum_seeds']=node_df['d1_seeds']\n",
    "\n",
    "node_df = node_df.sort_values('z_d1',ascending=False)\n",
    "node_df.head(15)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0aea77fb",
   "metadata": {},
   "source": [
    "## 10. Run community detection on NetColoc subnetwork (recommend HiDef)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "3b9850ec-07af-427e-98c2-b80c0873bbd9",
   "metadata": {},
   "outputs": [],
   "source": [
    "G_prox_cx = ndex2.create_nice_cx_from_networkx(G_prox)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "8dc3145c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Nodes in overlap subgraph: 439\n",
      "Edges in overlap subgraph: 2447\n"
     ]
    }
   ],
   "source": [
    "print(\"Nodes in overlap subgraph:\", len(G_prox.nodes()))\n",
    "print(\"Edges in overlap subgraph:\", len(G_prox.edges()))\n",
    "# Create cx format of overlap subgraph\n",
    "G_prox_cx = ndex2.create_nice_cx_from_networkx(G_prox)\n",
    "G_prox_cx.set_name(d1_name+'_NetColoc_subgraph') \n",
    "for node_id, node in G_prox_cx.get_nodes():\n",
    "    data = node_df.loc[node['n']]\n",
    "    for row, value in data.items():\n",
    "        if row == 'd1_seeds' or row == 'd2_seeds' or row=='sum_seeds':\n",
    "            data_type = 'double'\n",
    "        elif row=='d1_name' or row=='d2_name':\n",
    "            data_type='string'\n",
    "        else:\n",
    "            data_type = 'double'\n",
    "        G_prox_cx.set_node_attribute(node_id, row, value, type=data_type)\n",
    "\n",
    "cd = cdapsutil.CommunityDetection()\n",
    "\n",
    "# Run HiDeF on CDAPS REST service\n",
    "G_hier = cd.run_community_detection(G_prox_cx, algorithm='hidefv1.1beta',arguments={'--maxres':'20'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "b45ec87b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Hierarchy name: hidefv1.1beta_(none)_MM DNM gens_NetColoc_subgraph\n",
      "# nodes: 117\n",
      "# edges: 131\n"
     ]
    }
   ],
   "source": [
    "# Print information about hierarchy\n",
    "print('Hierarchy name: ' + str(G_hier.get_name()))\n",
    "print('# nodes: ' + str(len(G_hier.get_nodes())))\n",
    "print('# edges: ' + str(len(G_hier.get_edges())))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a445397d",
   "metadata": {},
   "source": [
    "## 11. Convert the NetColoc hierarchy to networkx format, and write out features of the hierarchy to a pandas dataframe, for easier access in Python."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "263c1aff",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of nodes: 117\n",
      "\n",
      "Number of edges: 131\n"
     ]
    }
   ],
   "source": [
    "G_hier = G_hier.to_networkx(mode='default')\n",
    "G_hier\n",
    "\n",
    "nodes = G_hier.nodes()\n",
    "\n",
    "# print the number of nodes and edges in the hierarchy for diagnostic purposes\n",
    "print('Number of nodes:', len(G_hier.nodes()))\n",
    "\n",
    "print('\\nNumber of edges:', len(G_hier.edges()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "470f59b4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CD_MemberList</th>\n",
       "      <th>CD_MemberList_Size</th>\n",
       "      <th>CD_Labeled</th>\n",
       "      <th>CD_MemberList_LogSize</th>\n",
       "      <th>CD_CommunityName</th>\n",
       "      <th>CD_AnnotatedMembers</th>\n",
       "      <th>CD_AnnotatedMembers_Size</th>\n",
       "      <th>CD_AnnotatedMembers_Overlap</th>\n",
       "      <th>CD_AnnotatedMembers_Pvalue</th>\n",
       "      <th>HiDeF_persistence</th>\n",
       "      <th>represents</th>\n",
       "      <th>name</th>\n",
       "      <th>system_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TMEM143 USP17L1 PLCL2 BORCS6 ZNF791 CTAG1B FNI...</td>\n",
       "      <td>434</td>\n",
       "      <td>False</td>\n",
       "      <td>8.762</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>120</td>\n",
       "      <td>C439</td>\n",
       "      <td>C439</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ZNF180 NUFIP1 ZNF181 MFAP1 UBIAD1</td>\n",
       "      <td>5</td>\n",
       "      <td>False</td>\n",
       "      <td>2.322</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5</td>\n",
       "      <td>C516</td>\n",
       "      <td>C516</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SQOR CXXC5 EHD4 SMAD3 STYXL1</td>\n",
       "      <td>5</td>\n",
       "      <td>False</td>\n",
       "      <td>2.322</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5</td>\n",
       "      <td>C517</td>\n",
       "      <td>C517</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FNIP2 MED13L METRNL ZSWIM6</td>\n",
       "      <td>4</td>\n",
       "      <td>False</td>\n",
       "      <td>2.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>43</td>\n",
       "      <td>C518</td>\n",
       "      <td>C518</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>VWA8 MMUT PCK2 MCM9</td>\n",
       "      <td>4</td>\n",
       "      <td>False</td>\n",
       "      <td>2.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>27</td>\n",
       "      <td>C524</td>\n",
       "      <td>C524</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       CD_MemberList  CD_MemberList_Size  \\\n",
       "0  TMEM143 USP17L1 PLCL2 BORCS6 ZNF791 CTAG1B FNI...                 434   \n",
       "1                  ZNF180 NUFIP1 ZNF181 MFAP1 UBIAD1                   5   \n",
       "2                       SQOR CXXC5 EHD4 SMAD3 STYXL1                   5   \n",
       "3                         FNIP2 MED13L METRNL ZSWIM6                   4   \n",
       "4                                VWA8 MMUT PCK2 MCM9                   4   \n",
       "\n",
       "  CD_Labeled CD_MemberList_LogSize CD_CommunityName CD_AnnotatedMembers  \\\n",
       "0      False                 8.762                                        \n",
       "1      False                 2.322                                        \n",
       "2      False                 2.322                                        \n",
       "3      False                   2.0                                        \n",
       "4      False                   2.0                                        \n",
       "\n",
       "  CD_AnnotatedMembers_Size CD_AnnotatedMembers_Overlap  \\\n",
       "0                        0                         0.0   \n",
       "1                        0                         0.0   \n",
       "2                        0                         0.0   \n",
       "3                        0                         0.0   \n",
       "4                        0                         0.0   \n",
       "\n",
       "  CD_AnnotatedMembers_Pvalue  HiDeF_persistence represents  name  system_ID  \n",
       "0                        0.0                120       C439  C439          0  \n",
       "1                        0.0                  5       C516  C516          1  \n",
       "2                        0.0                  5       C517  C517          2  \n",
       "3                        0.0                 43       C518  C518          3  \n",
       "4                        0.0                 27       C524  C524          4  "
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# add node attributes to dataframe for easier access\n",
    "hier_df = pd.DataFrame.from_dict(dict(G_hier.nodes(data=True)), orient='index')\n",
    "hier_df['system_ID']=hier_df.index.tolist()\n",
    "# some columns are not the right type\n",
    "hier_df['CD_MemberList_Size']=[int(x) for x in hier_df['CD_MemberList_Size'].tolist()]\n",
    "hier_df['HiDeF_persistence']=[int(x) for x in hier_df['HiDeF_persistence'].tolist()]\n",
    "hier_df.head()\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1f0ab2cc",
   "metadata": {},
   "source": [
    "## 12. Remove systems with no seed genes (OPTIONAL)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "2bc9bbf3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of nodes with seed genes: 117\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CD_MemberList</th>\n",
       "      <th>CD_MemberList_Size</th>\n",
       "      <th>CD_Labeled</th>\n",
       "      <th>CD_MemberList_LogSize</th>\n",
       "      <th>CD_CommunityName</th>\n",
       "      <th>CD_AnnotatedMembers</th>\n",
       "      <th>CD_AnnotatedMembers_Size</th>\n",
       "      <th>CD_AnnotatedMembers_Overlap</th>\n",
       "      <th>CD_AnnotatedMembers_Pvalue</th>\n",
       "      <th>HiDeF_persistence</th>\n",
       "      <th>represents</th>\n",
       "      <th>name</th>\n",
       "      <th>system_ID</th>\n",
       "      <th>num_d1_seeds</th>\n",
       "      <th>frac_d1_seeds</th>\n",
       "      <th>frac_no_seeds</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C439</th>\n",
       "      <td>TMEM143 USP17L1 PLCL2 BORCS6 ZNF791 CTAG1B FNI...</td>\n",
       "      <td>434</td>\n",
       "      <td>False</td>\n",
       "      <td>8.762</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>120</td>\n",
       "      <td>C439</td>\n",
       "      <td>C439</td>\n",
       "      <td>0</td>\n",
       "      <td>178</td>\n",
       "      <td>0.410138</td>\n",
       "      <td>0.589862</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C516</th>\n",
       "      <td>ZNF180 NUFIP1 ZNF181 MFAP1 UBIAD1</td>\n",
       "      <td>5</td>\n",
       "      <td>False</td>\n",
       "      <td>2.322</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5</td>\n",
       "      <td>C516</td>\n",
       "      <td>C516</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>0.600000</td>\n",
       "      <td>0.400000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C517</th>\n",
       "      <td>SQOR CXXC5 EHD4 SMAD3 STYXL1</td>\n",
       "      <td>5</td>\n",
       "      <td>False</td>\n",
       "      <td>2.322</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5</td>\n",
       "      <td>C517</td>\n",
       "      <td>C517</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0.400000</td>\n",
       "      <td>0.600000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C518</th>\n",
       "      <td>FNIP2 MED13L METRNL ZSWIM6</td>\n",
       "      <td>4</td>\n",
       "      <td>False</td>\n",
       "      <td>2.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>43</td>\n",
       "      <td>C518</td>\n",
       "      <td>C518</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>0.500000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C524</th>\n",
       "      <td>VWA8 MMUT PCK2 MCM9</td>\n",
       "      <td>4</td>\n",
       "      <td>False</td>\n",
       "      <td>2.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>27</td>\n",
       "      <td>C524</td>\n",
       "      <td>C524</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>0.750000</td>\n",
       "      <td>0.250000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C529</th>\n",
       "      <td>TBCB SCAP LIN37 MAP4</td>\n",
       "      <td>4</td>\n",
       "      <td>False</td>\n",
       "      <td>2.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>15</td>\n",
       "      <td>C529</td>\n",
       "      <td>C529</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>0.250000</td>\n",
       "      <td>0.750000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C533</th>\n",
       "      <td>DCP1A NSMAF NBEAL1 WDFY4</td>\n",
       "      <td>4</td>\n",
       "      <td>False</td>\n",
       "      <td>2.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>12</td>\n",
       "      <td>C533</td>\n",
       "      <td>C533</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>0.500000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C535</th>\n",
       "      <td>P2RX5 C7orf26 PPP4R1 BRAT1</td>\n",
       "      <td>4</td>\n",
       "      <td>False</td>\n",
       "      <td>2.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11</td>\n",
       "      <td>C535</td>\n",
       "      <td>C535</td>\n",
       "      <td>7</td>\n",
       "      <td>2</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>0.500000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C539</th>\n",
       "      <td>PPP1R2 CLIP1 CLASP2 DCTN6</td>\n",
       "      <td>4</td>\n",
       "      <td>False</td>\n",
       "      <td>2.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9</td>\n",
       "      <td>C539</td>\n",
       "      <td>C539</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>0.250000</td>\n",
       "      <td>0.750000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C542</th>\n",
       "      <td>PPP1R35 DCP1A KIF22 DCTN2</td>\n",
       "      <td>4</td>\n",
       "      <td>False</td>\n",
       "      <td>2.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>8</td>\n",
       "      <td>C542</td>\n",
       "      <td>C542</td>\n",
       "      <td>9</td>\n",
       "      <td>2</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>0.500000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C547</th>\n",
       "      <td>DAG1 PLEC ILK AGR2</td>\n",
       "      <td>4</td>\n",
       "      <td>False</td>\n",
       "      <td>2.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>12</td>\n",
       "      <td>C547</td>\n",
       "      <td>C547</td>\n",
       "      <td>10</td>\n",
       "      <td>1</td>\n",
       "      <td>0.250000</td>\n",
       "      <td>0.750000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C440</th>\n",
       "      <td>FICD COL20A1 UCN3 FREM1 ADGRV1 FREM2 ADGRL2 SR...</td>\n",
       "      <td>149</td>\n",
       "      <td>False</td>\n",
       "      <td>7.219</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5</td>\n",
       "      <td>C440</td>\n",
       "      <td>C440</td>\n",
       "      <td>11</td>\n",
       "      <td>72</td>\n",
       "      <td>0.483221</td>\n",
       "      <td>0.516779</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C441</th>\n",
       "      <td>USP17L1 POTEE METAP1 RAB4A CHORDC1 CSNK1D FBXO...</td>\n",
       "      <td>95</td>\n",
       "      <td>False</td>\n",
       "      <td>6.57</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5</td>\n",
       "      <td>C441</td>\n",
       "      <td>C441</td>\n",
       "      <td>12</td>\n",
       "      <td>35</td>\n",
       "      <td>0.368421</td>\n",
       "      <td>0.631579</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C442</th>\n",
       "      <td>CAPN3 MED13L ZNF782 ZNF181 ZNF280A DCP1A FNDC9...</td>\n",
       "      <td>58</td>\n",
       "      <td>False</td>\n",
       "      <td>5.858</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>27</td>\n",
       "      <td>C442</td>\n",
       "      <td>C442</td>\n",
       "      <td>13</td>\n",
       "      <td>31</td>\n",
       "      <td>0.534483</td>\n",
       "      <td>0.465517</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C445</th>\n",
       "      <td>FICD CYB561 VDAC2 MYO10 ZCRB1 SCLY NDUFB9 ARFI...</td>\n",
       "      <td>33</td>\n",
       "      <td>False</td>\n",
       "      <td>5.044</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5</td>\n",
       "      <td>C445</td>\n",
       "      <td>C445</td>\n",
       "      <td>14</td>\n",
       "      <td>12</td>\n",
       "      <td>0.363636</td>\n",
       "      <td>0.636364</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C446</th>\n",
       "      <td>SLC25A19 VDAC2 TDRD15 ZCRB1 USP17L1 SCLY SCML2...</td>\n",
       "      <td>31</td>\n",
       "      <td>False</td>\n",
       "      <td>4.954</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7</td>\n",
       "      <td>C446</td>\n",
       "      <td>C446</td>\n",
       "      <td>15</td>\n",
       "      <td>11</td>\n",
       "      <td>0.354839</td>\n",
       "      <td>0.645161</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C447</th>\n",
       "      <td>VDAC2 ZCRB1 NDUFB9 MRGBP TIMMDC1 OPALIN HEPN1 ...</td>\n",
       "      <td>29</td>\n",
       "      <td>False</td>\n",
       "      <td>4.858</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6</td>\n",
       "      <td>C447</td>\n",
       "      <td>C447</td>\n",
       "      <td>16</td>\n",
       "      <td>7</td>\n",
       "      <td>0.241379</td>\n",
       "      <td>0.758621</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C448</th>\n",
       "      <td>COL20A1 TM6SF1 PLG COL15A1 SLC30A6 TMEM189 OPA...</td>\n",
       "      <td>28</td>\n",
       "      <td>False</td>\n",
       "      <td>4.807</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7</td>\n",
       "      <td>C448</td>\n",
       "      <td>C448</td>\n",
       "      <td>17</td>\n",
       "      <td>15</td>\n",
       "      <td>0.535714</td>\n",
       "      <td>0.464286</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C453</th>\n",
       "      <td>BMP7 TDRD15 USP17L1 KRTAP9-7 SCML2 LAD1 KRBA2 ...</td>\n",
       "      <td>19</td>\n",
       "      <td>False</td>\n",
       "      <td>4.248</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5</td>\n",
       "      <td>C453</td>\n",
       "      <td>C453</td>\n",
       "      <td>18</td>\n",
       "      <td>6</td>\n",
       "      <td>0.315789</td>\n",
       "      <td>0.684211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C457</th>\n",
       "      <td>DNAH17 DNAH6 DNAH1 DNAH10 ZFR2 DNAH2 DNAH5 DNH...</td>\n",
       "      <td>12</td>\n",
       "      <td>False</td>\n",
       "      <td>3.585</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>100</td>\n",
       "      <td>C457</td>\n",
       "      <td>C457</td>\n",
       "      <td>19</td>\n",
       "      <td>3</td>\n",
       "      <td>0.250000</td>\n",
       "      <td>0.750000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C464</th>\n",
       "      <td>ZNF383 UCN3 AC012309.1 ZNF782 ZNF791 ZNF528 CN...</td>\n",
       "      <td>9</td>\n",
       "      <td>False</td>\n",
       "      <td>3.17</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>18</td>\n",
       "      <td>C464</td>\n",
       "      <td>C464</td>\n",
       "      <td>20</td>\n",
       "      <td>6</td>\n",
       "      <td>0.666667</td>\n",
       "      <td>0.333333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C468</th>\n",
       "      <td>TTC24 FREM1 ADGRV1 FREM2 FRAS1 CLEC2L TTC26 FR...</td>\n",
       "      <td>9</td>\n",
       "      <td>False</td>\n",
       "      <td>3.17</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5</td>\n",
       "      <td>C468</td>\n",
       "      <td>C468</td>\n",
       "      <td>21</td>\n",
       "      <td>3</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>0.666667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C471</th>\n",
       "      <td>RUFY3 SNRK TRIM3 RUFY4 ADAMTS12 ZFYVE28 ATXN7 ...</td>\n",
       "      <td>8</td>\n",
       "      <td>False</td>\n",
       "      <td>3.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>23</td>\n",
       "      <td>C471</td>\n",
       "      <td>C471</td>\n",
       "      <td>22</td>\n",
       "      <td>3</td>\n",
       "      <td>0.375000</td>\n",
       "      <td>0.625000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C473</th>\n",
       "      <td>TMEM143 TNK2 SPATA1 SS18 SH3BP1 MINK1 PRRC2A B...</td>\n",
       "      <td>8</td>\n",
       "      <td>False</td>\n",
       "      <td>3.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6</td>\n",
       "      <td>C473</td>\n",
       "      <td>C473</td>\n",
       "      <td>23</td>\n",
       "      <td>4</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>0.500000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C480</th>\n",
       "      <td>MYO16 DYNC2H1 TTC24 DNAH7 DNAH2 TTC26 DNAH14</td>\n",
       "      <td>7</td>\n",
       "      <td>False</td>\n",
       "      <td>2.807</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6</td>\n",
       "      <td>C480</td>\n",
       "      <td>C480</td>\n",
       "      <td>24</td>\n",
       "      <td>2</td>\n",
       "      <td>0.285714</td>\n",
       "      <td>0.714286</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C481</th>\n",
       "      <td>STK24 DEF8 TUBE1 RADIL GLYR1 LIN7A CENPBD1</td>\n",
       "      <td>7</td>\n",
       "      <td>False</td>\n",
       "      <td>2.807</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6</td>\n",
       "      <td>C481</td>\n",
       "      <td>C481</td>\n",
       "      <td>25</td>\n",
       "      <td>2</td>\n",
       "      <td>0.285714</td>\n",
       "      <td>0.714286</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C484</th>\n",
       "      <td>PPP1R14C BMP7 PPP1R14A PPP1R14D SAP25 IYD</td>\n",
       "      <td>6</td>\n",
       "      <td>False</td>\n",
       "      <td>2.585</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>42</td>\n",
       "      <td>C484</td>\n",
       "      <td>C484</td>\n",
       "      <td>26</td>\n",
       "      <td>1</td>\n",
       "      <td>0.166667</td>\n",
       "      <td>0.833333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C488</th>\n",
       "      <td>CTAG1B DTNBP1 BORCS8 TSKS CTAG1A BORCS6</td>\n",
       "      <td>6</td>\n",
       "      <td>False</td>\n",
       "      <td>2.585</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>32</td>\n",
       "      <td>C488</td>\n",
       "      <td>C488</td>\n",
       "      <td>27</td>\n",
       "      <td>2</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>0.666667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C489</th>\n",
       "      <td>MYO6 LIN7A MTUS1 ADD2 ALDH16A1 ROCK2</td>\n",
       "      <td>6</td>\n",
       "      <td>False</td>\n",
       "      <td>2.585</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>8</td>\n",
       "      <td>C489</td>\n",
       "      <td>C489</td>\n",
       "      <td>28</td>\n",
       "      <td>3</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>0.500000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C490</th>\n",
       "      <td>UBL7 EHD4 RFXANK BSG STYXL1 SZRD1</td>\n",
       "      <td>6</td>\n",
       "      <td>False</td>\n",
       "      <td>2.585</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7</td>\n",
       "      <td>C490</td>\n",
       "      <td>C490</td>\n",
       "      <td>29</td>\n",
       "      <td>2</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>0.666667</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                          CD_MemberList  CD_MemberList_Size  \\\n",
       "name                                                                          \n",
       "C439  TMEM143 USP17L1 PLCL2 BORCS6 ZNF791 CTAG1B FNI...                 434   \n",
       "C516                  ZNF180 NUFIP1 ZNF181 MFAP1 UBIAD1                   5   \n",
       "C517                       SQOR CXXC5 EHD4 SMAD3 STYXL1                   5   \n",
       "C518                         FNIP2 MED13L METRNL ZSWIM6                   4   \n",
       "C524                                VWA8 MMUT PCK2 MCM9                   4   \n",
       "C529                               TBCB SCAP LIN37 MAP4                   4   \n",
       "C533                           DCP1A NSMAF NBEAL1 WDFY4                   4   \n",
       "C535                         P2RX5 C7orf26 PPP4R1 BRAT1                   4   \n",
       "C539                          PPP1R2 CLIP1 CLASP2 DCTN6                   4   \n",
       "C542                          PPP1R35 DCP1A KIF22 DCTN2                   4   \n",
       "C547                                 DAG1 PLEC ILK AGR2                   4   \n",
       "C440  FICD COL20A1 UCN3 FREM1 ADGRV1 FREM2 ADGRL2 SR...                 149   \n",
       "C441  USP17L1 POTEE METAP1 RAB4A CHORDC1 CSNK1D FBXO...                  95   \n",
       "C442  CAPN3 MED13L ZNF782 ZNF181 ZNF280A DCP1A FNDC9...                  58   \n",
       "C445  FICD CYB561 VDAC2 MYO10 ZCRB1 SCLY NDUFB9 ARFI...                  33   \n",
       "C446  SLC25A19 VDAC2 TDRD15 ZCRB1 USP17L1 SCLY SCML2...                  31   \n",
       "C447  VDAC2 ZCRB1 NDUFB9 MRGBP TIMMDC1 OPALIN HEPN1 ...                  29   \n",
       "C448  COL20A1 TM6SF1 PLG COL15A1 SLC30A6 TMEM189 OPA...                  28   \n",
       "C453  BMP7 TDRD15 USP17L1 KRTAP9-7 SCML2 LAD1 KRBA2 ...                  19   \n",
       "C457  DNAH17 DNAH6 DNAH1 DNAH10 ZFR2 DNAH2 DNAH5 DNH...                  12   \n",
       "C464  ZNF383 UCN3 AC012309.1 ZNF782 ZNF791 ZNF528 CN...                   9   \n",
       "C468  TTC24 FREM1 ADGRV1 FREM2 FRAS1 CLEC2L TTC26 FR...                   9   \n",
       "C471  RUFY3 SNRK TRIM3 RUFY4 ADAMTS12 ZFYVE28 ATXN7 ...                   8   \n",
       "C473  TMEM143 TNK2 SPATA1 SS18 SH3BP1 MINK1 PRRC2A B...                   8   \n",
       "C480       MYO16 DYNC2H1 TTC24 DNAH7 DNAH2 TTC26 DNAH14                   7   \n",
       "C481         STK24 DEF8 TUBE1 RADIL GLYR1 LIN7A CENPBD1                   7   \n",
       "C484          PPP1R14C BMP7 PPP1R14A PPP1R14D SAP25 IYD                   6   \n",
       "C488            CTAG1B DTNBP1 BORCS8 TSKS CTAG1A BORCS6                   6   \n",
       "C489               MYO6 LIN7A MTUS1 ADD2 ALDH16A1 ROCK2                   6   \n",
       "C490                  UBL7 EHD4 RFXANK BSG STYXL1 SZRD1                   6   \n",
       "\n",
       "     CD_Labeled CD_MemberList_LogSize CD_CommunityName CD_AnnotatedMembers  \\\n",
       "name                                                                         \n",
       "C439      False                 8.762                                        \n",
       "C516      False                 2.322                                        \n",
       "C517      False                 2.322                                        \n",
       "C518      False                   2.0                                        \n",
       "C524      False                   2.0                                        \n",
       "C529      False                   2.0                                        \n",
       "C533      False                   2.0                                        \n",
       "C535      False                   2.0                                        \n",
       "C539      False                   2.0                                        \n",
       "C542      False                   2.0                                        \n",
       "C547      False                   2.0                                        \n",
       "C440      False                 7.219                                        \n",
       "C441      False                  6.57                                        \n",
       "C442      False                 5.858                                        \n",
       "C445      False                 5.044                                        \n",
       "C446      False                 4.954                                        \n",
       "C447      False                 4.858                                        \n",
       "C448      False                 4.807                                        \n",
       "C453      False                 4.248                                        \n",
       "C457      False                 3.585                                        \n",
       "C464      False                  3.17                                        \n",
       "C468      False                  3.17                                        \n",
       "C471      False                   3.0                                        \n",
       "C473      False                   3.0                                        \n",
       "C480      False                 2.807                                        \n",
       "C481      False                 2.807                                        \n",
       "C484      False                 2.585                                        \n",
       "C488      False                 2.585                                        \n",
       "C489      False                 2.585                                        \n",
       "C490      False                 2.585                                        \n",
       "\n",
       "     CD_AnnotatedMembers_Size CD_AnnotatedMembers_Overlap  \\\n",
       "name                                                        \n",
       "C439                        0                         0.0   \n",
       "C516                        0                         0.0   \n",
       "C517                        0                         0.0   \n",
       "C518                        0                         0.0   \n",
       "C524                        0                         0.0   \n",
       "C529                        0                         0.0   \n",
       "C533                        0                         0.0   \n",
       "C535                        0                         0.0   \n",
       "C539                        0                         0.0   \n",
       "C542                        0                         0.0   \n",
       "C547                        0                         0.0   \n",
       "C440                        0                         0.0   \n",
       "C441                        0                         0.0   \n",
       "C442                        0                         0.0   \n",
       "C445                        0                         0.0   \n",
       "C446                        0                         0.0   \n",
       "C447                        0                         0.0   \n",
       "C448                        0                         0.0   \n",
       "C453                        0                         0.0   \n",
       "C457                        0                         0.0   \n",
       "C464                        0                         0.0   \n",
       "C468                        0                         0.0   \n",
       "C471                        0                         0.0   \n",
       "C473                        0                         0.0   \n",
       "C480                        0                         0.0   \n",
       "C481                        0                         0.0   \n",
       "C484                        0                         0.0   \n",
       "C488                        0                         0.0   \n",
       "C489                        0                         0.0   \n",
       "C490                        0                         0.0   \n",
       "\n",
       "     CD_AnnotatedMembers_Pvalue  HiDeF_persistence represents  name  \\\n",
       "name                                                                  \n",
       "C439                        0.0                120       C439  C439   \n",
       "C516                        0.0                  5       C516  C516   \n",
       "C517                        0.0                  5       C517  C517   \n",
       "C518                        0.0                 43       C518  C518   \n",
       "C524                        0.0                 27       C524  C524   \n",
       "C529                        0.0                 15       C529  C529   \n",
       "C533                        0.0                 12       C533  C533   \n",
       "C535                        0.0                 11       C535  C535   \n",
       "C539                        0.0                  9       C539  C539   \n",
       "C542                        0.0                  8       C542  C542   \n",
       "C547                        0.0                 12       C547  C547   \n",
       "C440                        0.0                  5       C440  C440   \n",
       "C441                        0.0                  5       C441  C441   \n",
       "C442                        0.0                 27       C442  C442   \n",
       "C445                        0.0                  5       C445  C445   \n",
       "C446                        0.0                  7       C446  C446   \n",
       "C447                        0.0                  6       C447  C447   \n",
       "C448                        0.0                  7       C448  C448   \n",
       "C453                        0.0                  5       C453  C453   \n",
       "C457                        0.0                100       C457  C457   \n",
       "C464                        0.0                 18       C464  C464   \n",
       "C468                        0.0                  5       C468  C468   \n",
       "C471                        0.0                 23       C471  C471   \n",
       "C473                        0.0                  6       C473  C473   \n",
       "C480                        0.0                  6       C480  C480   \n",
       "C481                        0.0                  6       C481  C481   \n",
       "C484                        0.0                 42       C484  C484   \n",
       "C488                        0.0                 32       C488  C488   \n",
       "C489                        0.0                  8       C489  C489   \n",
       "C490                        0.0                  7       C490  C490   \n",
       "\n",
       "      system_ID  num_d1_seeds  frac_d1_seeds  frac_no_seeds  \n",
       "name                                                         \n",
       "C439          0           178       0.410138       0.589862  \n",
       "C516          1             3       0.600000       0.400000  \n",
       "C517          2             2       0.400000       0.600000  \n",
       "C518          3             2       0.500000       0.500000  \n",
       "C524          4             3       0.750000       0.250000  \n",
       "C529          5             1       0.250000       0.750000  \n",
       "C533          6             2       0.500000       0.500000  \n",
       "C535          7             2       0.500000       0.500000  \n",
       "C539          8             1       0.250000       0.750000  \n",
       "C542          9             2       0.500000       0.500000  \n",
       "C547         10             1       0.250000       0.750000  \n",
       "C440         11            72       0.483221       0.516779  \n",
       "C441         12            35       0.368421       0.631579  \n",
       "C442         13            31       0.534483       0.465517  \n",
       "C445         14            12       0.363636       0.636364  \n",
       "C446         15            11       0.354839       0.645161  \n",
       "C447         16             7       0.241379       0.758621  \n",
       "C448         17            15       0.535714       0.464286  \n",
       "C453         18             6       0.315789       0.684211  \n",
       "C457         19             3       0.250000       0.750000  \n",
       "C464         20             6       0.666667       0.333333  \n",
       "C468         21             3       0.333333       0.666667  \n",
       "C471         22             3       0.375000       0.625000  \n",
       "C473         23             4       0.500000       0.500000  \n",
       "C480         24             2       0.285714       0.714286  \n",
       "C481         25             2       0.285714       0.714286  \n",
       "C484         26             1       0.166667       0.833333  \n",
       "C488         27             2       0.333333       0.666667  \n",
       "C489         28             3       0.500000       0.500000  \n",
       "C490         29             2       0.333333       0.666667  "
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hier_df.index=hier_df['name']\n",
    "hier_df.head()\n",
    "\n",
    "num_d1_seeds = []\n",
    "frac_d1_seeds=[]\n",
    "\n",
    "systems_keep = []\n",
    "for c in hier_df.index.tolist():\n",
    "    system_genes = hier_df['CD_MemberList'].loc[c].split(' ')\n",
    "    d1_temp = list(np.intersect1d(system_genes,D1_genes))\n",
    "\n",
    "    num_d1_temp = len(d1_temp)\n",
    "    if (num_d1_temp)>0: # keep the system if it has at least 1 seed gene\n",
    "        systems_keep.append(c)\n",
    "        num_d1_seeds.append(num_d1_temp)\n",
    "        \n",
    "        frac_d1_seeds.append((num_d1_temp)/float(len(system_genes)))\n",
    "\n",
    "        \n",
    "frac_no_seeds = np.subtract(1.0,np.array([frac_d1_seeds]).sum(axis=0))\n",
    "\n",
    "hier_df = hier_df.loc[systems_keep]\n",
    "hier_df['num_d1_seeds']=num_d1_seeds\n",
    "hier_df['frac_d1_seeds']=frac_d1_seeds\n",
    "hier_df['frac_no_seeds']=frac_no_seeds\n",
    "print(\"Number of nodes with seed genes:\", len(hier_df))\n",
    "\n",
    "hier_df.head(30)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "6ca54a44",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "name\n",
      "C541      2.0\n",
      "C522      2.0\n",
      "C531      2.0\n",
      "C532      2.0\n",
      "C540      2.0\n",
      "        ...  \n",
      "C443      5.7\n",
      "C442    5.858\n",
      "C441     6.57\n",
      "C440    7.219\n",
      "C439    8.762\n",
      "Name: CD_MemberList_LogSize, Length: 117, dtype: object\n"
     ]
    }
   ],
   "source": [
    "print (hier_df[\"CD_MemberList_LogSize\"].sort_values())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "id": "f0f4d081",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of nodes with seed genes: 117\n",
      "Number of edges remaining: 131\n"
     ]
    }
   ],
   "source": [
    "# prune G_hier--> only keep systems with at least one seed gene\n",
    "\n",
    "nkeep=[]\n",
    "for n in list(G_hier.nodes()):\n",
    "    if G_hier.nodes(data=True)[n]['name'] in systems_keep:\n",
    "        nkeep.append(n)\n",
    "        \n",
    "\n",
    "G_hier = nx.subgraph(G_hier, nkeep)\n",
    "print(\"Number of nodes with seed genes:\", len(G_hier.nodes()))\n",
    "print(\"Number of edges remaining:\", len(G_hier.edges()))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ba23aa8c",
   "metadata": {},
   "source": [
    "## 13. Sneak peek of hierarchy visualization (OPTIONAL)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "id": "04c63201",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "76706c026d7145958badaaf20adbf90d",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "CytoscapeWidget(cytoscape_layout={'name': 'cose'}, cytoscape_style=[{'selector': 'node', 'css': {'content': 'd…"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "network_colocalization.view_G_hier(G_hier)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a01af1a8",
   "metadata": {},
   "source": [
    "## 14. Annotate systems with gprofiler."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "id": "5e8dd9a7",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C439\n",
      "434\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>RHO GTPase cycle</td>\n",
       "      <td>10835</td>\n",
       "      <td>30</td>\n",
       "      <td>RHO GTPase cycle</td>\n",
       "      <td>REAC:R-HSA-9012999</td>\n",
       "      <td>0.000008</td>\n",
       "      <td>[REAC:R-HSA-194315]</td>\n",
       "      <td>0.129310</td>\n",
       "      <td>query_1</td>\n",
       "      <td>232</td>\n",
       "      <td>0.066815</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>449</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Signaling by Rho GTPases</td>\n",
       "      <td>10835</td>\n",
       "      <td>40</td>\n",
       "      <td>Signaling by Rho GTPases</td>\n",
       "      <td>REAC:R-HSA-194315</td>\n",
       "      <td>0.000008</td>\n",
       "      <td>[REAC:R-HSA-9716542]</td>\n",
       "      <td>0.172414</td>\n",
       "      <td>query_1</td>\n",
       "      <td>232</td>\n",
       "      <td>0.056980</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>702</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Signaling by Rho GTPases, Miro GTPases and RHO...</td>\n",
       "      <td>10835</td>\n",
       "      <td>40</td>\n",
       "      <td>Signaling by Rho GTPases, Miro GTPases and RHO...</td>\n",
       "      <td>REAC:R-HSA-9716542</td>\n",
       "      <td>0.000008</td>\n",
       "      <td>[REAC:R-HSA-162582]</td>\n",
       "      <td>0.172414</td>\n",
       "      <td>query_1</td>\n",
       "      <td>232</td>\n",
       "      <td>0.055788</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>717</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                         description  effective_domain_size  \\\n",
       "2                                   RHO GTPase cycle                  10835   \n",
       "0                           Signaling by Rho GTPases                  10835   \n",
       "1  Signaling by Rho GTPases, Miro GTPases and RHO...                  10835   \n",
       "\n",
       "   intersection_size                                               name  \\\n",
       "2                 30                                   RHO GTPase cycle   \n",
       "0                 40                           Signaling by Rho GTPases   \n",
       "1                 40  Signaling by Rho GTPases, Miro GTPases and RHO...   \n",
       "\n",
       "               native   p_value               parents  precision    query  \\\n",
       "2  REAC:R-HSA-9012999  0.000008   [REAC:R-HSA-194315]   0.129310  query_1   \n",
       "0   REAC:R-HSA-194315  0.000008  [REAC:R-HSA-9716542]   0.172414  query_1   \n",
       "1  REAC:R-HSA-9716542  0.000008   [REAC:R-HSA-162582]   0.172414  query_1   \n",
       "\n",
       "   query_size    recall  significant source  term_size  \n",
       "2         232  0.066815         True   REAC        449  \n",
       "0         232  0.056980         True   REAC        702  \n",
       "1         232  0.055788         True   REAC        717  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C516\n",
      "5\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C517\n",
      "5\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C518\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C524\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C529\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C533\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C535\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C539\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C542\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C547\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C440\n",
      "149\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Nervous system development</td>\n",
       "      <td>10835</td>\n",
       "      <td>22</td>\n",
       "      <td>Nervous system development</td>\n",
       "      <td>REAC:R-HSA-9675108</td>\n",
       "      <td>2.541969e-07</td>\n",
       "      <td>[REAC:R-HSA-1266738]</td>\n",
       "      <td>0.250000</td>\n",
       "      <td>query_1</td>\n",
       "      <td>88</td>\n",
       "      <td>0.038328</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>574</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Axon guidance</td>\n",
       "      <td>10835</td>\n",
       "      <td>21</td>\n",
       "      <td>Axon guidance</td>\n",
       "      <td>REAC:R-HSA-422475</td>\n",
       "      <td>3.504396e-07</td>\n",
       "      <td>[REAC:R-HSA-9675108]</td>\n",
       "      <td>0.238636</td>\n",
       "      <td>query_1</td>\n",
       "      <td>88</td>\n",
       "      <td>0.038251</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>549</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  description  effective_domain_size  intersection_size  \\\n",
       "0  Nervous system development                  10835                 22   \n",
       "1               Axon guidance                  10835                 21   \n",
       "\n",
       "                         name              native       p_value  \\\n",
       "0  Nervous system development  REAC:R-HSA-9675108  2.541969e-07   \n",
       "1               Axon guidance   REAC:R-HSA-422475  3.504396e-07   \n",
       "\n",
       "                parents  precision    query  query_size    recall  \\\n",
       "0  [REAC:R-HSA-1266738]   0.250000  query_1          88  0.038328   \n",
       "1  [REAC:R-HSA-9675108]   0.238636  query_1          88  0.038251   \n",
       "\n",
       "   significant source  term_size  \n",
       "0         True   REAC        574  \n",
       "1         True   REAC        549  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C441\n",
      "95\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C442\n",
      "58\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Chromatin modifying enzymes</td>\n",
       "      <td>10835</td>\n",
       "      <td>11</td>\n",
       "      <td>Chromatin modifying enzymes</td>\n",
       "      <td>REAC:R-HSA-3247509</td>\n",
       "      <td>3.294937e-09</td>\n",
       "      <td>[REAC:R-HSA-4839726]</td>\n",
       "      <td>0.392857</td>\n",
       "      <td>query_1</td>\n",
       "      <td>28</td>\n",
       "      <td>0.040293</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>273</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Chromatin organization</td>\n",
       "      <td>10835</td>\n",
       "      <td>11</td>\n",
       "      <td>Chromatin organization</td>\n",
       "      <td>REAC:R-HSA-4839726</td>\n",
       "      <td>3.294937e-09</td>\n",
       "      <td>[REAC:0000000]</td>\n",
       "      <td>0.392857</td>\n",
       "      <td>query_1</td>\n",
       "      <td>28</td>\n",
       "      <td>0.040293</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>273</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   description  effective_domain_size  intersection_size  \\\n",
       "0  Chromatin modifying enzymes                  10835                 11   \n",
       "1       Chromatin organization                  10835                 11   \n",
       "\n",
       "                          name              native       p_value  \\\n",
       "0  Chromatin modifying enzymes  REAC:R-HSA-3247509  3.294937e-09   \n",
       "1       Chromatin organization  REAC:R-HSA-4839726  3.294937e-09   \n",
       "\n",
       "                parents  precision    query  query_size    recall  \\\n",
       "0  [REAC:R-HSA-4839726]   0.392857  query_1          28  0.040293   \n",
       "1        [REAC:0000000]   0.392857  query_1          28  0.040293   \n",
       "\n",
       "   significant source  term_size  \n",
       "0         True   REAC        273  \n",
       "1         True   REAC        273  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C445\n",
      "33\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C446\n",
      "31\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C447\n",
      "29\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C448\n",
      "28\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C453\n",
      "19\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C457\n",
      "12\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C464\n",
      "9\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C468\n",
      "9\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C471\n",
      "8\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C473\n",
      "8\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C480\n",
      "7\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C481\n",
      "7\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C484\n",
      "6\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C488\n",
      "6\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C489\n",
      "6\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C490\n",
      "6\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C491\n",
      "6\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C495\n",
      "6\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C496\n",
      "5\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C501\n",
      "5\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C511\n",
      "5\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C512\n",
      "5\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C449\n",
      "27\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Clathrin-mediated endocytosis</td>\n",
       "      <td>10835</td>\n",
       "      <td>10</td>\n",
       "      <td>Clathrin-mediated endocytosis</td>\n",
       "      <td>REAC:R-HSA-8856828</td>\n",
       "      <td>1.349967e-11</td>\n",
       "      <td>[REAC:R-HSA-199991]</td>\n",
       "      <td>0.434783</td>\n",
       "      <td>query_1</td>\n",
       "      <td>23</td>\n",
       "      <td>0.069444</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>144</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Parasite infection</td>\n",
       "      <td>10835</td>\n",
       "      <td>6</td>\n",
       "      <td>Parasite infection</td>\n",
       "      <td>REAC:R-HSA-9664407</td>\n",
       "      <td>7.158181e-07</td>\n",
       "      <td>[REAC:R-HSA-9658195]</td>\n",
       "      <td>0.260870</td>\n",
       "      <td>query_1</td>\n",
       "      <td>23</td>\n",
       "      <td>0.053097</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Leishmania phagocytosis</td>\n",
       "      <td>10835</td>\n",
       "      <td>6</td>\n",
       "      <td>Leishmania phagocytosis</td>\n",
       "      <td>REAC:R-HSA-9664417</td>\n",
       "      <td>7.158181e-07</td>\n",
       "      <td>[REAC:R-HSA-9664407]</td>\n",
       "      <td>0.260870</td>\n",
       "      <td>query_1</td>\n",
       "      <td>23</td>\n",
       "      <td>0.053097</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>FCGR3A-mediated phagocytosis</td>\n",
       "      <td>10835</td>\n",
       "      <td>6</td>\n",
       "      <td>FCGR3A-mediated phagocytosis</td>\n",
       "      <td>REAC:R-HSA-9664422</td>\n",
       "      <td>7.158181e-07</td>\n",
       "      <td>[REAC:R-HSA-9664417]</td>\n",
       "      <td>0.260870</td>\n",
       "      <td>query_1</td>\n",
       "      <td>23</td>\n",
       "      <td>0.053097</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Regulation of actin dynamics for phagocytic cu...</td>\n",
       "      <td>10835</td>\n",
       "      <td>6</td>\n",
       "      <td>Regulation of actin dynamics for phagocytic cu...</td>\n",
       "      <td>REAC:R-HSA-2029482</td>\n",
       "      <td>7.837873e-07</td>\n",
       "      <td>[REAC:R-HSA-2029480]</td>\n",
       "      <td>0.260870</td>\n",
       "      <td>query_1</td>\n",
       "      <td>23</td>\n",
       "      <td>0.051724</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>116</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                          description  effective_domain_size  \\\n",
       "0                       Clathrin-mediated endocytosis                  10835   \n",
       "12                                 Parasite infection                  10835   \n",
       "13                            Leishmania phagocytosis                  10835   \n",
       "14                       FCGR3A-mediated phagocytosis                  10835   \n",
       "15  Regulation of actin dynamics for phagocytic cu...                  10835   \n",
       "\n",
       "    intersection_size                                               name  \\\n",
       "0                  10                      Clathrin-mediated endocytosis   \n",
       "12                  6                                 Parasite infection   \n",
       "13                  6                            Leishmania phagocytosis   \n",
       "14                  6                       FCGR3A-mediated phagocytosis   \n",
       "15                  6  Regulation of actin dynamics for phagocytic cu...   \n",
       "\n",
       "                native       p_value               parents  precision  \\\n",
       "0   REAC:R-HSA-8856828  1.349967e-11   [REAC:R-HSA-199991]   0.434783   \n",
       "12  REAC:R-HSA-9664407  7.158181e-07  [REAC:R-HSA-9658195]   0.260870   \n",
       "13  REAC:R-HSA-9664417  7.158181e-07  [REAC:R-HSA-9664407]   0.260870   \n",
       "14  REAC:R-HSA-9664422  7.158181e-07  [REAC:R-HSA-9664417]   0.260870   \n",
       "15  REAC:R-HSA-2029482  7.837873e-07  [REAC:R-HSA-2029480]   0.260870   \n",
       "\n",
       "      query  query_size    recall  significant source  term_size  \n",
       "0   query_1          23  0.069444         True   REAC        144  \n",
       "12  query_1          23  0.053097         True   REAC        113  \n",
       "13  query_1          23  0.053097         True   REAC        113  \n",
       "14  query_1          23  0.053097         True   REAC        113  \n",
       "15  query_1          23  0.051724         True   REAC        116  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C544\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C451\n",
      "21\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C549\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C487\n",
      "6\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C522\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C459\n",
      "11\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C460\n",
      "11\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C492\n",
      "6\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C462\n",
      "10\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C466\n",
      "9\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C531\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C532\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C472\n",
      "8\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Ion homeostasis</td>\n",
       "      <td>10835</td>\n",
       "      <td>4</td>\n",
       "      <td>Ion homeostasis</td>\n",
       "      <td>REAC:R-HSA-5578775</td>\n",
       "      <td>0.000002</td>\n",
       "      <td>[REAC:R-HSA-5576891]</td>\n",
       "      <td>0.5</td>\n",
       "      <td>query_1</td>\n",
       "      <td>8</td>\n",
       "      <td>0.074074</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>54</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       description  effective_domain_size  intersection_size             name  \\\n",
       "0  Ion homeostasis                  10835                  4  Ion homeostasis   \n",
       "\n",
       "               native   p_value               parents  precision    query  \\\n",
       "0  REAC:R-HSA-5578775  0.000002  [REAC:R-HSA-5576891]        0.5  query_1   \n",
       "\n",
       "   query_size    recall  significant source  term_size  \n",
       "0           8  0.074074         True   REAC         54  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C540\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C478\n",
      "7\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C546\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C458\n",
      "12\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C527\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C500\n",
      "5\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C469\n",
      "8\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C538\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C443\n",
      "52\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C477\n",
      "7\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C479\n",
      "7\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C523\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C552\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C452\n",
      "19\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Chromatin modifying enzymes</td>\n",
       "      <td>10835</td>\n",
       "      <td>5</td>\n",
       "      <td>Chromatin modifying enzymes</td>\n",
       "      <td>REAC:R-HSA-3247509</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>[REAC:R-HSA-4839726]</td>\n",
       "      <td>0.833333</td>\n",
       "      <td>query_1</td>\n",
       "      <td>6</td>\n",
       "      <td>0.018315</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>273</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Chromatin organization</td>\n",
       "      <td>10835</td>\n",
       "      <td>5</td>\n",
       "      <td>Chromatin organization</td>\n",
       "      <td>REAC:R-HSA-4839726</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>[REAC:0000000]</td>\n",
       "      <td>0.833333</td>\n",
       "      <td>query_1</td>\n",
       "      <td>6</td>\n",
       "      <td>0.018315</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>273</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   description  effective_domain_size  intersection_size  \\\n",
       "0  Chromatin modifying enzymes                  10835                  5   \n",
       "1       Chromatin organization                  10835                  5   \n",
       "\n",
       "                          name              native   p_value  \\\n",
       "0  Chromatin modifying enzymes  REAC:R-HSA-3247509  0.000001   \n",
       "1       Chromatin organization  REAC:R-HSA-4839726  0.000001   \n",
       "\n",
       "                parents  precision    query  query_size    recall  \\\n",
       "0  [REAC:R-HSA-4839726]   0.833333  query_1           6  0.018315   \n",
       "1        [REAC:0000000]   0.833333  query_1           6  0.018315   \n",
       "\n",
       "   significant source  term_size  \n",
       "0         True   REAC        273  \n",
       "1         True   REAC        273  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C444\n",
      "34\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C519\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C493\n",
      "6\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C497\n",
      "5\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Antigen processing: Ubiquitination &amp; Proteasom...</td>\n",
       "      <td>10835</td>\n",
       "      <td>5</td>\n",
       "      <td>Antigen processing: Ubiquitination &amp; Proteasom...</td>\n",
       "      <td>REAC:R-HSA-983168</td>\n",
       "      <td>9.170868e-07</td>\n",
       "      <td>[REAC:R-HSA-983169]</td>\n",
       "      <td>1.0</td>\n",
       "      <td>query_1</td>\n",
       "      <td>5</td>\n",
       "      <td>0.016234</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>308</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Class I MHC mediated antigen processing &amp; pres...</td>\n",
       "      <td>10835</td>\n",
       "      <td>5</td>\n",
       "      <td>Class I MHC mediated antigen processing &amp; pres...</td>\n",
       "      <td>REAC:R-HSA-983169</td>\n",
       "      <td>1.318986e-06</td>\n",
       "      <td>[REAC:R-HSA-1280218]</td>\n",
       "      <td>1.0</td>\n",
       "      <td>query_1</td>\n",
       "      <td>5</td>\n",
       "      <td>0.013158</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>380</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                         description  effective_domain_size  \\\n",
       "0  Antigen processing: Ubiquitination & Proteasom...                  10835   \n",
       "1  Class I MHC mediated antigen processing & pres...                  10835   \n",
       "\n",
       "   intersection_size                                               name  \\\n",
       "0                  5  Antigen processing: Ubiquitination & Proteasom...   \n",
       "1                  5  Class I MHC mediated antigen processing & pres...   \n",
       "\n",
       "              native       p_value               parents  precision    query  \\\n",
       "0  REAC:R-HSA-983168  9.170868e-07   [REAC:R-HSA-983169]        1.0  query_1   \n",
       "1  REAC:R-HSA-983169  1.318986e-06  [REAC:R-HSA-1280218]        1.0  query_1   \n",
       "\n",
       "   query_size    recall  significant source  term_size  \n",
       "0           5  0.016234         True   REAC        308  \n",
       "1           5  0.013158         True   REAC        380  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C498\n",
      "5\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Macroautophagy</td>\n",
       "      <td>10835</td>\n",
       "      <td>4</td>\n",
       "      <td>Macroautophagy</td>\n",
       "      <td>REAC:R-HSA-1632852</td>\n",
       "      <td>1.889756e-07</td>\n",
       "      <td>[REAC:R-HSA-9612973]</td>\n",
       "      <td>1.0</td>\n",
       "      <td>query_1</td>\n",
       "      <td>4</td>\n",
       "      <td>0.029851</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>134</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Autophagy</td>\n",
       "      <td>10835</td>\n",
       "      <td>4</td>\n",
       "      <td>Autophagy</td>\n",
       "      <td>REAC:R-HSA-9612973</td>\n",
       "      <td>1.889756e-07</td>\n",
       "      <td>[REAC:0000000]</td>\n",
       "      <td>1.0</td>\n",
       "      <td>query_1</td>\n",
       "      <td>4</td>\n",
       "      <td>0.026846</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>149</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      description  effective_domain_size  intersection_size            name  \\\n",
       "1  Macroautophagy                  10835                  4  Macroautophagy   \n",
       "0       Autophagy                  10835                  4       Autophagy   \n",
       "\n",
       "               native       p_value               parents  precision    query  \\\n",
       "1  REAC:R-HSA-1632852  1.889756e-07  [REAC:R-HSA-9612973]        1.0  query_1   \n",
       "0  REAC:R-HSA-9612973  1.889756e-07        [REAC:0000000]        1.0  query_1   \n",
       "\n",
       "   query_size    recall  significant source  term_size  \n",
       "1           4  0.029851         True   REAC        134  \n",
       "0           4  0.026846         True   REAC        149  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C502\n",
      "5\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C536\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C537\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C514\n",
      "5\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C467\n",
      "9\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C454\n",
      "17\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C545\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C450\n",
      "22\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C482\n",
      "7\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C455\n",
      "15\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C476\n",
      "7\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C456\n",
      "15\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C461\n",
      "11\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Clathrin-mediated endocytosis</td>\n",
       "      <td>10835</td>\n",
       "      <td>7</td>\n",
       "      <td>Clathrin-mediated endocytosis</td>\n",
       "      <td>REAC:R-HSA-8856828</td>\n",
       "      <td>1.178411e-09</td>\n",
       "      <td>[REAC:R-HSA-199991]</td>\n",
       "      <td>0.636364</td>\n",
       "      <td>query_1</td>\n",
       "      <td>11</td>\n",
       "      <td>0.048611</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>144</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     description  effective_domain_size  intersection_size  \\\n",
       "0  Clathrin-mediated endocytosis                  10835                  7   \n",
       "\n",
       "                            name              native       p_value  \\\n",
       "0  Clathrin-mediated endocytosis  REAC:R-HSA-8856828  1.178411e-09   \n",
       "\n",
       "               parents  precision    query  query_size    recall  significant  \\\n",
       "0  [REAC:R-HSA-199991]   0.636364  query_1          11  0.048611         True   \n",
       "\n",
       "  source  term_size  \n",
       "0   REAC        144  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C463\n",
      "9\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FCGR3A-mediated phagocytosis</td>\n",
       "      <td>10835</td>\n",
       "      <td>4</td>\n",
       "      <td>FCGR3A-mediated phagocytosis</td>\n",
       "      <td>REAC:R-HSA-9664422</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>[REAC:R-HSA-9664417]</td>\n",
       "      <td>0.571429</td>\n",
       "      <td>query_1</td>\n",
       "      <td>7</td>\n",
       "      <td>0.035398</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Parasite infection</td>\n",
       "      <td>10835</td>\n",
       "      <td>4</td>\n",
       "      <td>Parasite infection</td>\n",
       "      <td>REAC:R-HSA-9664407</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>[REAC:R-HSA-9658195]</td>\n",
       "      <td>0.571429</td>\n",
       "      <td>query_1</td>\n",
       "      <td>7</td>\n",
       "      <td>0.035398</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Leishmania phagocytosis</td>\n",
       "      <td>10835</td>\n",
       "      <td>4</td>\n",
       "      <td>Leishmania phagocytosis</td>\n",
       "      <td>REAC:R-HSA-9664417</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>[REAC:R-HSA-9664407]</td>\n",
       "      <td>0.571429</td>\n",
       "      <td>query_1</td>\n",
       "      <td>7</td>\n",
       "      <td>0.035398</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Regulation of actin dynamics for phagocytic cu...</td>\n",
       "      <td>10835</td>\n",
       "      <td>4</td>\n",
       "      <td>Regulation of actin dynamics for phagocytic cu...</td>\n",
       "      <td>REAC:R-HSA-2029482</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>[REAC:R-HSA-2029480]</td>\n",
       "      <td>0.571429</td>\n",
       "      <td>query_1</td>\n",
       "      <td>7</td>\n",
       "      <td>0.034483</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Fcgamma receptor (FCGR) dependent phagocytosis</td>\n",
       "      <td>10835</td>\n",
       "      <td>4</td>\n",
       "      <td>Fcgamma receptor (FCGR) dependent phagocytosis</td>\n",
       "      <td>REAC:R-HSA-2029480</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>[REAC:R-HSA-168249]</td>\n",
       "      <td>0.571429</td>\n",
       "      <td>query_1</td>\n",
       "      <td>7</td>\n",
       "      <td>0.028571</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>140</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                         description  effective_domain_size  \\\n",
       "3                       FCGR3A-mediated phagocytosis                  10835   \n",
       "5                                 Parasite infection                  10835   \n",
       "7                            Leishmania phagocytosis                  10835   \n",
       "2  Regulation of actin dynamics for phagocytic cu...                  10835   \n",
       "9     Fcgamma receptor (FCGR) dependent phagocytosis                  10835   \n",
       "\n",
       "   intersection_size                                               name  \\\n",
       "3                  4                       FCGR3A-mediated phagocytosis   \n",
       "5                  4                                 Parasite infection   \n",
       "7                  4                            Leishmania phagocytosis   \n",
       "2                  4  Regulation of actin dynamics for phagocytic cu...   \n",
       "9                  4     Fcgamma receptor (FCGR) dependent phagocytosis   \n",
       "\n",
       "               native   p_value               parents  precision    query  \\\n",
       "3  REAC:R-HSA-9664422  0.000003  [REAC:R-HSA-9664417]   0.571429  query_1   \n",
       "5  REAC:R-HSA-9664407  0.000003  [REAC:R-HSA-9658195]   0.571429  query_1   \n",
       "7  REAC:R-HSA-9664417  0.000003  [REAC:R-HSA-9664407]   0.571429  query_1   \n",
       "2  REAC:R-HSA-2029482  0.000003  [REAC:R-HSA-2029480]   0.571429  query_1   \n",
       "9  REAC:R-HSA-2029480  0.000005   [REAC:R-HSA-168249]   0.571429  query_1   \n",
       "\n",
       "   query_size    recall  significant source  term_size  \n",
       "3           7  0.035398         True   REAC        113  \n",
       "5           7  0.035398         True   REAC        113  \n",
       "7           7  0.035398         True   REAC        113  \n",
       "2           7  0.034483         True   REAC        116  \n",
       "9           7  0.028571         True   REAC        140  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C483\n",
      "6\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C553\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C474\n",
      "7\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C465\n",
      "9\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C515\n",
      "5\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C485\n",
      "6\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C555\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C507\n",
      "5\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C534\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C470\n",
      "8\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C494\n",
      "6\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C508\n",
      "5\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C499\n",
      "5\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C509\n",
      "5\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C504\n",
      "5\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C520\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Clathrin-mediated endocytosis</td>\n",
       "      <td>10835</td>\n",
       "      <td>4</td>\n",
       "      <td>Clathrin-mediated endocytosis</td>\n",
       "      <td>REAC:R-HSA-8856828</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>[REAC:R-HSA-199991]</td>\n",
       "      <td>1.0</td>\n",
       "      <td>query_1</td>\n",
       "      <td>4</td>\n",
       "      <td>0.027778</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>144</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     description  effective_domain_size  intersection_size  \\\n",
       "0  Clathrin-mediated endocytosis                  10835                  4   \n",
       "\n",
       "                            name              native   p_value  \\\n",
       "0  Clathrin-mediated endocytosis  REAC:R-HSA-8856828  0.000001   \n",
       "\n",
       "               parents  precision    query  query_size    recall  significant  \\\n",
       "0  [REAC:R-HSA-199991]        1.0  query_1           4  0.027778         True   \n",
       "\n",
       "  source  term_size  \n",
       "0   REAC        144  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C475\n",
      "7\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C530\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C486\n",
      "6\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C505\n",
      "5\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C513\n",
      "5\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Ion homeostasis</td>\n",
       "      <td>10835</td>\n",
       "      <td>3</td>\n",
       "      <td>Ion homeostasis</td>\n",
       "      <td>REAC:R-HSA-5578775</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>[REAC:R-HSA-5576891]</td>\n",
       "      <td>0.6</td>\n",
       "      <td>query_1</td>\n",
       "      <td>5</td>\n",
       "      <td>0.055556</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Transport of small molecules</td>\n",
       "      <td>10835</td>\n",
       "      <td>5</td>\n",
       "      <td>Transport of small molecules</td>\n",
       "      <td>REAC:R-HSA-382551</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>[REAC:0000000]</td>\n",
       "      <td>1.0</td>\n",
       "      <td>query_1</td>\n",
       "      <td>5</td>\n",
       "      <td>0.006964</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>718</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    description  effective_domain_size  intersection_size  \\\n",
       "0               Ion homeostasis                  10835                  3   \n",
       "1  Transport of small molecules                  10835                  5   \n",
       "\n",
       "                           name              native   p_value  \\\n",
       "0               Ion homeostasis  REAC:R-HSA-5578775  0.000009   \n",
       "1  Transport of small molecules   REAC:R-HSA-382551  0.000009   \n",
       "\n",
       "                parents  precision    query  query_size    recall  \\\n",
       "0  [REAC:R-HSA-5576891]        0.6  query_1           5  0.055556   \n",
       "1        [REAC:0000000]        1.0  query_1           5  0.006964   \n",
       "\n",
       "   significant source  term_size  \n",
       "0         True   REAC         54  \n",
       "1         True   REAC        718  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C526\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C506\n",
      "5\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C543\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C525\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C503\n",
      "5\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C554\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C551\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C548\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C510\n",
      "5\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C521\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C528\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C550\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C541\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# gprofiler annotation of clusters\n",
    "\n",
    "system_name_list = []\n",
    "for p in hier_df.index.tolist():\n",
    "    focal_genes=hier_df['CD_MemberList'].loc[p].split(' ')\n",
    "    print(p)\n",
    "    print(len(focal_genes))\n",
    "    if len(focal_genes)>2:\n",
    "        gp_temp = pd.DataFrame(gp.profile(focal_genes,significance_threshold_method='fdr',\n",
    "                                               sources=['REAC']))\n",
    "        if len(gp_temp)>0: # make sure data is not empty\n",
    "            \n",
    "            # make sure terms are specific, and overlap with at least 3 genes\n",
    "            gp_temp = gp_temp[(gp_temp['term_size']<1000)&(gp_temp['term_size']>50)]\n",
    "            gp_temp = gp_temp[gp_temp['intersection_size']>=3]\n",
    "            \n",
    "            gp_temp = gp_temp[gp_temp['p_value']<1E-5] # set a stringent pvalue threshold\n",
    "            \n",
    "            gp_temp = gp_temp.sort_values('recall',ascending=False)\n",
    "            \n",
    "            if len(gp_temp)>1:\n",
    "                system_name_list.append(gp_temp.head(1)['name'].tolist()[0])\n",
    "            else:\n",
    "                system_name_list.append(p)\n",
    "        else:\n",
    "            system_name_list.append(p)\n",
    "            \n",
    "\n",
    "        display(gp_temp.head())\n",
    "        \n",
    "    else:\n",
    "        system_name_list.append(p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "67220bc9",
   "metadata": {},
   "outputs": [],
   "source": [
    "hier_df.to_csv(\"/projects/ps-gleesonlab8/User/hiyoothere/NTD/5.Analysis/Network/output/9.subnet_coloc_heir.txt\", sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "id": "3fa72b83",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'head' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Input \u001b[0;32mIn [88]\u001b[0m, in \u001b[0;36m<cell line: 1>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0m \u001b[43mhead\u001b[49m(hier_df)\n",
      "\u001b[0;31mNameError\u001b[0m: name 'head' is not defined"
     ]
    }
   ],
   "source": [
    "head(hier_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f85629a1",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "51f4a2f6",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "eba841ca",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cea82321",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "53ef5323",
   "metadata": {},
   "source": [
    "## Compute network colocalized systems map"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b0393820",
   "metadata": {},
   "source": [
    "## 9. Convert network colocalization subnetwork to form used in community detection module"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "id": "efa69d63",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'G_overlap' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Input \u001b[0;32mIn [91]\u001b[0m, in \u001b[0;36m<cell line: 2>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;66;03m# compile dataframe of metadata for overlapping nodes\u001b[39;00m\n\u001b[0;32m----> 2\u001b[0m node_df \u001b[38;5;241m=\u001b[39m pd\u001b[38;5;241m.\u001b[39mDataFrame(index\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mlist\u001b[39m(\u001b[43mG_overlap\u001b[49m\u001b[38;5;241m.\u001b[39mnodes))\n\u001b[1;32m      3\u001b[0m node_df \u001b[38;5;241m=\u001b[39m node_df\u001b[38;5;241m.\u001b[39massign(d1_seeds\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m0\u001b[39m, d2_seeds\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m0\u001b[39m, d1_name\u001b[38;5;241m=\u001b[39md1_name, d2_name\u001b[38;5;241m=\u001b[39md2_name)\n\u001b[1;32m      4\u001b[0m node_df\u001b[38;5;241m.\u001b[39mloc[\u001b[38;5;28mlist\u001b[39m(np\u001b[38;5;241m.\u001b[39mintersect1d(seed_dict[d1_name],node_df\u001b[38;5;241m.\u001b[39mindex\u001b[38;5;241m.\u001b[39mtolist())), \u001b[38;5;124m'\u001b[39m\u001b[38;5;124md1_seeds\u001b[39m\u001b[38;5;124m'\u001b[39m]\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m1\u001b[39m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'G_overlap' is not defined"
     ]
    }
   ],
   "source": [
    "# compile dataframe of metadata for overlapping nodes\n",
    "node_df = pd.DataFrame(index=list(G_overlap.nodes))\n",
    "node_df = node_df.assign(d1_seeds=0, d2_seeds=0, d1_name=d1_name, d2_name=d2_name)\n",
    "node_df.loc[list(np.intersect1d(seed_dict[d1_name],node_df.index.tolist())), 'd1_seeds']=1\n",
    "node_df.loc[list(np.intersect1d(seed_dict[d2_name],node_df.index.tolist())), 'd2_seeds']=1\n",
    "node_df['z_d1']=z_D1.loc[list(G_overlap.nodes)]['z']\n",
    "#node_df['z_d2']=z_D2.loc[list(G_overlap.nodes)]['z']\n",
    "#node_df['z_both']=node_df['z_d1']*node_df['z_d2']\n",
    "#node_df['sum_seeds']=node_df['d1_seeds']+node_df['d2_seeds']\n",
    "\n",
    "node_df = node_df.sort_values('z_d1',ascending=False)\n",
    "print (node_df.head(15))\n",
    "node_df.to_csv(\"/projects/ps-gleesonlab8/User/hiyoothere/NTD/5.Analysis/Network/output/output/9.subnet_coloc.txt\", sep = '\\t')\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f06f19cf",
   "metadata": {},
   "source": [
    "## 10. Run community detection on NetColoc subnetwork (recommend HiDef)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "id": "d5369e39",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'G_overlap' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Input \u001b[0;32mIn [92]\u001b[0m, in \u001b[0;36m<cell line: 1>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0m \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mNodes in overlap subgraph:\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;28mlen\u001b[39m(\u001b[43mG_overlap\u001b[49m\u001b[38;5;241m.\u001b[39mnodes()))\n\u001b[1;32m      2\u001b[0m \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mEdges in overlap subgraph:\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;28mlen\u001b[39m(G_overlap\u001b[38;5;241m.\u001b[39medges()))\n\u001b[1;32m      3\u001b[0m \u001b[38;5;66;03m# Create cx format of overlap subgraph\u001b[39;00m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'G_overlap' is not defined"
     ]
    }
   ],
   "source": [
    "print(\"Nodes in overlap subgraph:\", len(G_overlap.nodes()))\n",
    "print(\"Edges in overlap subgraph:\", len(G_overlap.edges()))\n",
    "# Create cx format of overlap subgraph\n",
    "G_overlap_cx = ndex2.create_nice_cx_from_networkx(G_overlap)\n",
    "G_overlap_cx.set_name(d1_name+'_'+d2_name+'_NetColoc_subgraph') \n",
    "for node_id, node in G_overlap_cx.get_nodes():\n",
    "    data = node_df.loc[node['n']]\n",
    "    for row, value in data.items():\n",
    "        if row == 'd1_seeds' or row == 'd2_seeds' or row=='sum_seeds':\n",
    "            data_type = 'double'\n",
    "        elif row=='d1_name' or row=='d2_name':\n",
    "            data_type='string'\n",
    "        else:\n",
    "            data_type = 'double'\n",
    "        G_overlap_cx.set_node_attribute(node_id, row, value, type=data_type)\n",
    "\n",
    "cd = cdapsutil.CommunityDetection()\n",
    "print (G_overlap_cx)\n",
    "# Run HiDeF on CDAPS REST service\n",
    "G_hier = cd.run_community_detection(G_overlap_cx, algorithm='hidefv1.1beta',arguments={'--maxres':'20'})\n",
    "print (G_hier)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "id": "29d8c022",
   "metadata": {},
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "'MultiDiGraph' object has no attribute 'get_name'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "Input \u001b[0;32mIn [93]\u001b[0m, in \u001b[0;36m<cell line: 2>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;66;03m# Print information about hierarchy\u001b[39;00m\n\u001b[0;32m----> 2\u001b[0m \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mHierarchy name: \u001b[39m\u001b[38;5;124m'\u001b[39m \u001b[38;5;241m+\u001b[39m \u001b[38;5;28mstr\u001b[39m(\u001b[43mG_hier\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget_name\u001b[49m()))\n\u001b[1;32m      3\u001b[0m \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m# nodes: \u001b[39m\u001b[38;5;124m'\u001b[39m \u001b[38;5;241m+\u001b[39m \u001b[38;5;28mstr\u001b[39m(\u001b[38;5;28mlen\u001b[39m(G_hier\u001b[38;5;241m.\u001b[39mget_nodes())))\n\u001b[1;32m      4\u001b[0m \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m# edges: \u001b[39m\u001b[38;5;124m'\u001b[39m \u001b[38;5;241m+\u001b[39m \u001b[38;5;28mstr\u001b[39m(\u001b[38;5;28mlen\u001b[39m(G_hier\u001b[38;5;241m.\u001b[39mget_edges())))\n",
      "\u001b[0;31mAttributeError\u001b[0m: 'MultiDiGraph' object has no attribute 'get_name'"
     ]
    }
   ],
   "source": [
    "# Print information about hierarchy\n",
    "print('Hierarchy name: ' + str(G_hier.get_name()))\n",
    "print('# nodes: ' + str(len(G_hier.get_nodes())))\n",
    "print('# edges: ' + str(len(G_hier.get_edges())))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "id": "81f3531e",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DISCLAIMIER: cdapsutil is experimental and may contain errors and interfaces may change\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'algorithms': {'hidef': {'name': 'hidef',\n",
       "   'displayName': 'HiDeF',\n",
       "   'description': \"Runs <a href='https://github.com/fanzheng10/HiDeF'>HiDeF</a> community detection algorithm.<br/><br/>Paper:<br/>Zheng, F., Zhang, S., Churas, C., Pratt, D., Bahar, I., Ideker, T. HiDeF: identifying persistent structures in<br/>multiscale ‘omics data. Genome Biol 22, 21 (2021).<br/><a href='https://doi.org/10.1186/s13059-020-02228-4'>10.1186/s13059-020-02228-4</a><br/><br/><b>NOTE:</b> For reasonable runtime, the maximum size of input network should not exceed 20k nodes and 400k edges.<br/><br/><a href='https://github.com/idekerlab/cdhidef'>Click here for more information</a>\",\n",
       "   'version': '0.2.2',\n",
       "   'dockerImage': 'coleslawndex/cdhidef:0.2.2',\n",
       "   'inputDataFormat': 'EDGELISTV2',\n",
       "   'outputDataFormat': 'COMMUNITYDETECTRESULTV2',\n",
       "   'customParameters': [{'name': '--maxres',\n",
       "     'displayName': 'Maximum resolution parameter',\n",
       "     'description': 'Maximum resolution parameter. Increase to get more smaller communities',\n",
       "     'type': 'value',\n",
       "     'defaultValue': '50.0',\n",
       "     'validationType': 'number',\n",
       "     'validationHelp': 'Should be a number',\n",
       "     'validationRegex': None,\n",
       "     'minValue': None,\n",
       "     'maxValue': None},\n",
       "    {'name': '--k',\n",
       "     'displayName': 'Persistent threshold',\n",
       "     'description': 'Persistence threshold. Increase to delete unstable clusters, and get fewer communities',\n",
       "     'type': 'value',\n",
       "     'defaultValue': '5',\n",
       "     'validationType': 'digits',\n",
       "     'validationHelp': 'Should be a number',\n",
       "     'validationRegex': None,\n",
       "     'minValue': None,\n",
       "     'maxValue': None},\n",
       "    {'name': '--alg',\n",
       "     'displayName': 'Algorithm',\n",
       "     'description': 'Choose to use Louvain or newer Leiden algorithm',\n",
       "     'type': 'value',\n",
       "     'defaultValue': 'louvain',\n",
       "     'validationType': 'string',\n",
       "     'validationHelp': 'Must be set to louvain or leiden',\n",
       "     'validationRegex': 'louvain|leiden',\n",
       "     'minValue': None,\n",
       "     'maxValue': None},\n",
       "    {'name': '--n',\n",
       "     'displayName': 'Target community number',\n",
       "     'description': 'Target community number. Explore the maximum resolution parameter until the number of generated communities at this resolution is close enough to this value. Increase this number to get more smaller communities',\n",
       "     'type': 'value',\n",
       "     'defaultValue': None,\n",
       "     'validationType': 'digits',\n",
       "     'validationHelp': 'Should be either unset or a whole number',\n",
       "     'validationRegex': None,\n",
       "     'minValue': None,\n",
       "     'maxValue': None},\n",
       "    {'name': '--ct',\n",
       "     'displayName': 'Consensus threshold',\n",
       "     'description': 'Consensus threshold. Threshold of collapsing community graph and choose genes for each community',\n",
       "     'type': 'value',\n",
       "     'defaultValue': '75',\n",
       "     'validationType': 'digits',\n",
       "     'validationHelp': 'Should be a number',\n",
       "     'validationRegex': None,\n",
       "     'minValue': None,\n",
       "     'maxValue': None}]},\n",
       "  'hidefv1.1beta': {'name': 'hidefv1.1beta',\n",
       "   'displayName': 'HiDeF 1.1 beta',\n",
       "   'description': \"Runs <a href='https://github.com/fanzheng10/HiDeF'>HiDeF</a> community detection algorithm. This is a 1.1 beta version which uses Leiden as default algorithm. <br/><br/>Paper:<br/>Zheng, F., Zhang, S., Churas, C., Pratt, D., Bahar, I., Ideker, T. HiDeF: identifying persistent structures in<br/>multiscale ‘omics data. Genome Biol 22, 21 (2021).<br/><a href='https://doi.org/10.1186/s13059-020-02228-4'>10.1186/s13059-020-02228-4</a><br/><br/>NOTE: For reasonable runtime, the maximum size of input network should not exceed 20k nodes and 400k edges.<br/><br/><a href='https://github.com/idekerlab/cdhidef'>Click here for more information</a>\",\n",
       "   'version': '1.1.1',\n",
       "   'dockerImage': 'coleslawndex/cdhidef:1.1.1b1',\n",
       "   'inputDataFormat': 'EDGELISTV2',\n",
       "   'outputDataFormat': 'COMMUNITYDETECTRESULTV2',\n",
       "   'customParameters': [{'name': '--maxres',\n",
       "     'displayName': 'Maximum resolution parameter',\n",
       "     'description': 'Maximum resolution parameter. Increase to get more smaller communities',\n",
       "     'type': 'value',\n",
       "     'defaultValue': '25.0',\n",
       "     'validationType': 'number',\n",
       "     'validationHelp': 'Should be a number',\n",
       "     'validationRegex': None,\n",
       "     'minValue': None,\n",
       "     'maxValue': None},\n",
       "    {'name': '--p',\n",
       "     'displayName': 'Consensus threshold',\n",
       "     'description': 'Consensus threshold. Threshold of collapsing community graph and choose genes for each community',\n",
       "     'type': 'value',\n",
       "     'defaultValue': '75',\n",
       "     'validationType': 'digits',\n",
       "     'validationHelp': 'Should be a number',\n",
       "     'validationRegex': None,\n",
       "     'minValue': None,\n",
       "     'maxValue': None},\n",
       "    {'name': '--alg',\n",
       "     'displayName': 'Algorithm',\n",
       "     'description': 'Choose to use Louvain or newer Leiden algorithm',\n",
       "     'type': 'value',\n",
       "     'defaultValue': 'leiden',\n",
       "     'validationType': 'string',\n",
       "     'validationHelp': 'Must be set to louvain or leiden',\n",
       "     'validationRegex': 'louvain|leiden',\n",
       "     'minValue': None,\n",
       "     'maxValue': None},\n",
       "    {'name': '--n',\n",
       "     'displayName': 'Target community number',\n",
       "     'description': 'Target community number. Explore the maximum resolution parameter until the number of generated communities at this resolution is close enough to this value. Increase this number to get more smaller communities',\n",
       "     'type': 'value',\n",
       "     'defaultValue': None,\n",
       "     'validationType': 'digits',\n",
       "     'validationHelp': 'Should be either unset or a whole number',\n",
       "     'validationRegex': None,\n",
       "     'minValue': None,\n",
       "     'maxValue': None},\n",
       "    {'name': '--k',\n",
       "     'displayName': 'Persistent threshold',\n",
       "     'description': 'Persistence threshold. Increase to delete unstable clusters, and get fewer communities',\n",
       "     'type': 'value',\n",
       "     'defaultValue': '5',\n",
       "     'validationType': 'digits',\n",
       "     'validationHelp': 'Should be a number',\n",
       "     'validationRegex': None,\n",
       "     'minValue': None,\n",
       "     'maxValue': None}]},\n",
       "  'oslom': {'name': 'oslom',\n",
       "   'displayName': 'OSLOM',\n",
       "   'description': \"Runs <a href='http://www.oslom.org'>Order Statistics Local Optimization Method (OSLOM)</a> community detection algorithm.<br/><br/><a href='https://github.com/idekerlab/cdoslom'>Click here for more information</a>\",\n",
       "   'version': '0.3.0',\n",
       "   'dockerImage': 'coleslawndex/cdoslom:0.3.0',\n",
       "   'inputDataFormat': 'EDGELIST',\n",
       "   'outputDataFormat': 'COMMUNITYDETECTRESULT',\n",
       "   'customParameters': [{'name': '--seed',\n",
       "     'displayName': 'Random number seed',\n",
       "     'description': 'Random number generator seed. -1 means to not set it',\n",
       "     'type': 'value',\n",
       "     'defaultValue': '-1',\n",
       "     'validationType': 'number',\n",
       "     'validationHelp': 'Should be a number',\n",
       "     'validationRegex': None,\n",
       "     'minValue': None,\n",
       "     'maxValue': None},\n",
       "    {'name': '--singlet',\n",
       "     'displayName': 'Singlet',\n",
       "     'description': 'If set, do NOT merge singlet with existing clusters/modules',\n",
       "     'type': 'flag',\n",
       "     'defaultValue': None,\n",
       "     'validationType': None,\n",
       "     'validationHelp': None,\n",
       "     'validationRegex': None,\n",
       "     'minValue': None,\n",
       "     'maxValue': None},\n",
       "    {'name': '--cp',\n",
       "     'displayName': 'Coverage parameter',\n",
       "     'description': 'Larger value leads to bigger clusters/modules',\n",
       "     'type': 'value',\n",
       "     'defaultValue': '0.5',\n",
       "     'validationType': 'number',\n",
       "     'validationHelp': 'Should be a number',\n",
       "     'validationRegex': None,\n",
       "     'minValue': None,\n",
       "     'maxValue': None},\n",
       "    {'name': '--directed',\n",
       "     'displayName': 'Assume Directed Links',\n",
       "     'description': 'If set, infomap assumes directed links',\n",
       "     'type': 'flag',\n",
       "     'defaultValue': None,\n",
       "     'validationType': None,\n",
       "     'validationHelp': None,\n",
       "     'validationRegex': None,\n",
       "     'minValue': None,\n",
       "     'maxValue': None},\n",
       "    {'name': '--p_val',\n",
       "     'displayName': 'P-value',\n",
       "     'description': 'Increase for more clusters/modules',\n",
       "     'type': 'value',\n",
       "     'defaultValue': '0.1',\n",
       "     'validationType': 'number',\n",
       "     'validationHelp': 'Should be a number',\n",
       "     'validationRegex': None,\n",
       "     'minValue': None,\n",
       "     'maxValue': None}]},\n",
       "  'louvain': {'name': 'louvain',\n",
       "   'displayName': 'Louvain',\n",
       "   'description': \"Runs <a href='https://anaconda.org/conda-forge/louvain'>Louvain</a> community detection algorithm.<br/><br/>Louvain works by iteratively maximazing modularity in each community, building hierarchy from the bottom layer (single nodes) up to the root layer. NOTE: Louvain may not return a result with a fully connected network as input.<br/><br/><a href='https://github.com/idekerlab/cdlouvain'>Click here for more information</a>\",\n",
       "   'version': '0.2.0',\n",
       "   'dockerImage': 'coleslawndex/cdlouvain:0.2.0',\n",
       "   'inputDataFormat': 'EDGELIST',\n",
       "   'outputDataFormat': 'COMMUNITYDETECTRESULT',\n",
       "   'customParameters': [{'name': '--resolution_parameter',\n",
       "     'displayName': 'Cluster resolution',\n",
       "     'description': 'Guides number of communities the algorithm will detect. Higher the value the more communities detected.',\n",
       "     'type': 'value',\n",
       "     'defaultValue': '10',\n",
       "     'validationType': 'number',\n",
       "     'validationHelp': None,\n",
       "     'validationRegex': None,\n",
       "     'minValue': 0,\n",
       "     'maxValue': None},\n",
       "    {'name': '--directed',\n",
       "     'displayName': 'Treat input as directed graph',\n",
       "     'description': 'If set, input is treated as directed graph',\n",
       "     'type': 'flag',\n",
       "     'defaultValue': None,\n",
       "     'validationType': None,\n",
       "     'validationHelp': None,\n",
       "     'validationRegex': None,\n",
       "     'minValue': None,\n",
       "     'maxValue': None},\n",
       "    {'name': '--seed',\n",
       "     'displayName': 'Random number generator seed',\n",
       "     'description': 'Sets random number generator seed. Leave blank to not set.',\n",
       "     'type': 'value',\n",
       "     'defaultValue': '',\n",
       "     'validationType': 'digits',\n",
       "     'validationHelp': 'Should be a whole number',\n",
       "     'validationRegex': None,\n",
       "     'minValue': None,\n",
       "     'maxValue': None},\n",
       "    {'name': '--deep',\n",
       "     'displayName': 'Generate deep hierarchy',\n",
       "     'description': 'If set, generate deep hierarchy, otherwise algorithm will just create clusters',\n",
       "     'type': 'flag',\n",
       "     'defaultValue': None,\n",
       "     'validationType': None,\n",
       "     'validationHelp': None,\n",
       "     'validationRegex': None,\n",
       "     'minValue': None,\n",
       "     'maxValue': None},\n",
       "    {'name': '--overlap',\n",
       "     'displayName': 'Generate overlapping communities',\n",
       "     'description': 'If set, generate potentially overlapping communities',\n",
       "     'type': 'flag',\n",
       "     'defaultValue': None,\n",
       "     'validationType': None,\n",
       "     'validationHelp': None,\n",
       "     'validationRegex': None,\n",
       "     'minValue': None,\n",
       "     'maxValue': None}]},\n",
       "  'infomap': {'name': 'infomap',\n",
       "   'displayName': 'Infomap',\n",
       "   'description': \"Runs <a href='https://www.mapequation.org'>Infomap</a> community detection algorithm.<br/><br/>A method based on Flux. Detect communities from dynamics on the network.<br/><br/><a href='https://github.com/idekerlab/cdinfomap'>Click here for more information</a> <br/><b>NOTE:</b> July 2022, --enableoverlapping parameter removed cause it caused Infomap 0.20 to use excessive amounts of ram.\",\n",
       "   'version': '0.1.0',\n",
       "   'dockerImage': 'coleslawndex/cdinfomap:0.1.0',\n",
       "   'inputDataFormat': 'EDGELIST',\n",
       "   'outputDataFormat': 'COMMUNITYDETECTRESULT',\n",
       "   'customParameters': [{'name': '--markovtime',\n",
       "     'displayName': 'Markov time',\n",
       "     'description': 'Scale link flow with this value to change the cost of moving between modules/communities. Higher for less modules/communities',\n",
       "     'type': 'value',\n",
       "     'defaultValue': '0.75',\n",
       "     'validationType': 'number',\n",
       "     'validationHelp': 'Should be a number',\n",
       "     'validationRegex': None,\n",
       "     'minValue': 0,\n",
       "     'maxValue': None},\n",
       "    {'name': '--seed',\n",
       "     'displayName': 'Random number generator seed',\n",
       "     'description': 'Sets random number generator seed. Leave blank to not set.',\n",
       "     'type': 'value',\n",
       "     'defaultValue': '',\n",
       "     'validationType': 'digits',\n",
       "     'validationHelp': 'Should be a whole number',\n",
       "     'validationRegex': None,\n",
       "     'minValue': None,\n",
       "     'maxValue': None},\n",
       "    {'name': '--directed',\n",
       "     'displayName': 'Assume Directed Links',\n",
       "     'description': 'If set, infomap assumes directed links',\n",
       "     'type': 'flag',\n",
       "     'defaultValue': None,\n",
       "     'validationType': None,\n",
       "     'validationHelp': None,\n",
       "     'validationRegex': None,\n",
       "     'minValue': None,\n",
       "     'maxValue': None}]},\n",
       "  'clixo': {'name': 'clixo',\n",
       "   'displayName': 'CliXO',\n",
       "   'description': \"Runs <a href='https://github.com/fanzheng10/CliXO-1.0'>CliXO-1.0 Beta</a> community detection algorithm.<br/><br/>Original paper: Kramer M, Dutkowski J, Yu M, Bafna V, Ideker T. Inferring gene ontologies from pairwise similarity data. Bioinformatics, 30: i34-i42. 2014. doi: 10.1093/bioinformatics/btu282<br/><br/>A classical method that iteratively detects hierarchy based on cliques from changing edge weight threshold.<br/>CliXO only works on graphs with edge weights.<br/>NOTE: For reasonable runtime, the maximum size of input network should not exceed hundreds of nodes.<br/><br/><a href='https://github.com/idekerlab/cdclixo'>Click here for more information</a>\",\n",
       "   'version': '1.0.0b1',\n",
       "   'dockerImage': 'coleslawndex/cdclixo:1.0.0b1',\n",
       "   'inputDataFormat': 'EDGELIST',\n",
       "   'outputDataFormat': 'COMMUNITYDETECTRESULT',\n",
       "   'customParameters': [{'name': '--alpha',\n",
       "     'displayName': 'Alpha',\n",
       "     'description': 'Threshold between cluster layers',\n",
       "     'type': 'value',\n",
       "     'defaultValue': '0.1',\n",
       "     'validationType': 'number',\n",
       "     'validationHelp': None,\n",
       "     'validationRegex': None,\n",
       "     'minValue': 0,\n",
       "     'maxValue': None},\n",
       "    {'name': '--beta',\n",
       "     'displayName': 'Beta',\n",
       "     'description': 'Merge similarity for overlapping clusters',\n",
       "     'type': 'value',\n",
       "     'defaultValue': '0.5',\n",
       "     'validationType': 'number',\n",
       "     'validationHelp': None,\n",
       "     'validationRegex': None,\n",
       "     'minValue': 0,\n",
       "     'maxValue': 1}]},\n",
       "  'gprofilersingletermv2': {'name': 'gprofilersingletermv2',\n",
       "   'displayName': 'gProfiler',\n",
       "   'description': \"Uses <a href='https://pypi.org/project/gprofiler-official/'>Gprofiler</a> to find best term below pvalue cut offusing a list of genes as input. For more information <a href='https://github.com/idekerlab/cdgprofilergenestoterm'>click here</a>\",\n",
       "   'version': '0.3.0',\n",
       "   'dockerImage': 'coleslawndex/cdgprofilergenestoterm:0.3.0',\n",
       "   'inputDataFormat': 'GENELIST',\n",
       "   'outputDataFormat': 'MAPPEDTERMJSON',\n",
       "   'customParameters': [{'name': '--omit_intersections',\n",
       "     'displayName': 'Skip gene intersections (faster)',\n",
       "     'description': 'If set, skips calculation of gene intersections which makes operation much faster',\n",
       "     'type': 'flag',\n",
       "     'defaultValue': None,\n",
       "     'validationType': None,\n",
       "     'validationHelp': None,\n",
       "     'validationRegex': None,\n",
       "     'minValue': None,\n",
       "     'maxValue': None},\n",
       "    {'name': '--maxgenelistsize',\n",
       "     'displayName': 'Max gene list size',\n",
       "     'description': 'Maximum number of genes allowed in query',\n",
       "     'type': 'value',\n",
       "     'defaultValue': '5000',\n",
       "     'validationType': 'number',\n",
       "     'validationHelp': 'Must be a number',\n",
       "     'validationRegex': None,\n",
       "     'minValue': 0,\n",
       "     'maxValue': 20000},\n",
       "    {'name': '--minoverlap',\n",
       "     'displayName': 'Minimum overlap',\n",
       "     'description': 'Minimum Jaccard value to allow for results',\n",
       "     'type': 'value',\n",
       "     'defaultValue': '0.05',\n",
       "     'validationType': 'number',\n",
       "     'validationHelp': 'Must be a number',\n",
       "     'validationRegex': None,\n",
       "     'minValue': 0,\n",
       "     'maxValue': 1.1},\n",
       "    {'name': '--organism',\n",
       "     'displayName': 'Organism',\n",
       "     'description': 'Organism to use for query. scerevisiae for Yeast(Saccharomyces cerevisiae), hsapiens for Human (Homo sapiens)',\n",
       "     'type': 'value',\n",
       "     'defaultValue': 'hsapiens',\n",
       "     'validationType': 'string',\n",
       "     'validationHelp': 'Must be a valid name under id column on this page: https://biit.cs.ut.ee/gprofiler/page/organism-list',\n",
       "     'validationRegex': None,\n",
       "     'minValue': None,\n",
       "     'maxValue': None},\n",
       "    {'name': '--maxpval',\n",
       "     'displayName': 'Maximum Pvalue',\n",
       "     'description': 'Maximum pvalue to allow for results',\n",
       "     'type': 'value',\n",
       "     'defaultValue': '0.00001',\n",
       "     'validationType': 'number',\n",
       "     'validationHelp': 'Must be a number',\n",
       "     'validationRegex': None,\n",
       "     'minValue': None,\n",
       "     'maxValue': None}]},\n",
       "  'iquerysingleterm': {'name': 'iquerysingleterm',\n",
       "   'displayName': 'iQuery',\n",
       "   'description': \"Uses <a href='https://github.com/ndexbio/ndexsearch-rest'>Integrated Query</a> to find best term below pvalue cut off using a list of genes as input.\",\n",
       "   'version': '0.4.0',\n",
       "   'dockerImage': 'coleslawndex/cdiquerygenestoterm:0.4.0',\n",
       "   'inputDataFormat': 'GENELIST',\n",
       "   'outputDataFormat': 'MAPPEDTERMJSON',\n",
       "   'customParameters': [{'name': '--url',\n",
       "     'displayName': 'REST endpoint',\n",
       "     'description': 'Integrated Query REST endpoint URL',\n",
       "     'type': 'value',\n",
       "     'defaultValue': 'http://public.ndexbio.org',\n",
       "     'validationType': 'string',\n",
       "     'validationHelp': 'Must be a URL starting with http',\n",
       "     'validationRegex': None,\n",
       "     'minValue': None,\n",
       "     'maxValue': None}]},\n",
       "  'cdenrichrgenestoterm': {'name': 'cdenrichrgenestoterm',\n",
       "   'displayName': 'Enrichr',\n",
       "   'description': \"Uses <a href='http://amp.pharm.mssm.edu/Enrichr/'>Enrichr</a> to find best term below pvalue cut off using a list of genes as input. For more information <a href='https://github.com/idekerlab/cdenrichrgenestoterm'>click here</a>\",\n",
       "   'version': '0.4.0',\n",
       "   'dockerImage': 'coleslawndex/cdenrichrgenestoterm:0.4.0',\n",
       "   'inputDataFormat': 'GENELIST',\n",
       "   'outputDataFormat': 'MAPPEDTERMJSON',\n",
       "   'customParameters': [{'name': '--maxpval',\n",
       "     'displayName': 'Maximum Pvalue',\n",
       "     'description': 'Maximum pvalue to allow for results',\n",
       "     'type': 'value',\n",
       "     'defaultValue': '0.05',\n",
       "     'validationType': 'number',\n",
       "     'validationHelp': 'Must be a number',\n",
       "     'validationRegex': None,\n",
       "     'minValue': None,\n",
       "     'maxValue': None},\n",
       "    {'name': '--genesets',\n",
       "     'displayName': 'Gene sets',\n",
       "     'description': \"Gene sets to query. For options <a href='http://amp.pharm.mssm.edu/Enrichr/#stats'>click here</a>\",\n",
       "     'type': 'value',\n",
       "     'defaultValue': 'GO_Biological_Process_2018,GO_Cellular_Component_2018,GO_Molecular_Function_2018,KEGG_2019_Human,Reactome_2016,WikiPathways_2019_Human,Human_Phenotype_Ontology,Jensen_DISEASES',\n",
       "     'validationType': 'string',\n",
       "     'validationHelp': 'Comma delimited string',\n",
       "     'validationRegex': None,\n",
       "     'minValue': None,\n",
       "     'maxValue': None}]}}}"
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sr = cdapsutil.ServiceRunner()\n",
    "sr.get_algorithms() # this will print out available options for each algorithm"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0d5da38e",
   "metadata": {},
   "source": [
    "## 11. Convert the NetColoc hierarchy to networkx format, and write out features of the hierarchy to a pandas dataframe, for easier access in Python."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "id": "6f83e8f1",
   "metadata": {},
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "'MultiDiGraph' object has no attribute 'to_networkx'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "Input \u001b[0;32mIn [95]\u001b[0m, in \u001b[0;36m<cell line: 1>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0m G_hier \u001b[38;5;241m=\u001b[39m \u001b[43mG_hier\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mto_networkx\u001b[49m(mode\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mdefault\u001b[39m\u001b[38;5;124m'\u001b[39m)\n\u001b[1;32m      2\u001b[0m \u001b[38;5;66;03m#G_hier.to_csv(\"output/G_hier.txt\", sep = '\\t')\u001b[39;00m\n\u001b[1;32m      3\u001b[0m \u001b[38;5;28mprint\u001b[39m (G_hier)\n",
      "\u001b[0;31mAttributeError\u001b[0m: 'MultiDiGraph' object has no attribute 'to_networkx'"
     ]
    }
   ],
   "source": [
    "G_hier = G_hier.to_networkx(mode='default')\n",
    "#G_hier.to_csv(\"output/G_hier.txt\", sep = '\\t')\n",
    "print (G_hier)\n",
    "nodes = G_hier.nodes()\n",
    "\n",
    "# print the number of nodes and edges in the hierarchy for diagnostic purposes\n",
    "print('Number of nodes:', len(G_hier.nodes()))\n",
    "\n",
    "print('\\nNumber of edges:', len(G_hier.edges()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "id": "e4053027",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CD_MemberList</th>\n",
       "      <th>CD_MemberList_Size</th>\n",
       "      <th>CD_Labeled</th>\n",
       "      <th>CD_MemberList_LogSize</th>\n",
       "      <th>CD_CommunityName</th>\n",
       "      <th>CD_AnnotatedMembers</th>\n",
       "      <th>CD_AnnotatedMembers_Size</th>\n",
       "      <th>CD_AnnotatedMembers_Overlap</th>\n",
       "      <th>CD_AnnotatedMembers_Pvalue</th>\n",
       "      <th>HiDeF_persistence</th>\n",
       "      <th>represents</th>\n",
       "      <th>name</th>\n",
       "      <th>CD_MemberList_LogSize_viz</th>\n",
       "      <th>system_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TMEM143 USP17L1 PLCL2 BORCS6 ZNF791 CTAG1B FNI...</td>\n",
       "      <td>434</td>\n",
       "      <td>False</td>\n",
       "      <td>8.762</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>120</td>\n",
       "      <td>C439</td>\n",
       "      <td>C439</td>\n",
       "      <td>87.62</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ZNF180 NUFIP1 ZNF181 MFAP1 UBIAD1</td>\n",
       "      <td>5</td>\n",
       "      <td>False</td>\n",
       "      <td>2.322</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5</td>\n",
       "      <td>C516</td>\n",
       "      <td>C516</td>\n",
       "      <td>23.22</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SQOR CXXC5 EHD4 SMAD3 STYXL1</td>\n",
       "      <td>5</td>\n",
       "      <td>False</td>\n",
       "      <td>2.322</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5</td>\n",
       "      <td>C517</td>\n",
       "      <td>C517</td>\n",
       "      <td>23.22</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FNIP2 MED13L METRNL ZSWIM6</td>\n",
       "      <td>4</td>\n",
       "      <td>False</td>\n",
       "      <td>2.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>43</td>\n",
       "      <td>C518</td>\n",
       "      <td>C518</td>\n",
       "      <td>20.0</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>VWA8 MMUT PCK2 MCM9</td>\n",
       "      <td>4</td>\n",
       "      <td>False</td>\n",
       "      <td>2.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>27</td>\n",
       "      <td>C524</td>\n",
       "      <td>C524</td>\n",
       "      <td>20.0</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       CD_MemberList  CD_MemberList_Size  \\\n",
       "0  TMEM143 USP17L1 PLCL2 BORCS6 ZNF791 CTAG1B FNI...                 434   \n",
       "1                  ZNF180 NUFIP1 ZNF181 MFAP1 UBIAD1                   5   \n",
       "2                       SQOR CXXC5 EHD4 SMAD3 STYXL1                   5   \n",
       "3                         FNIP2 MED13L METRNL ZSWIM6                   4   \n",
       "4                                VWA8 MMUT PCK2 MCM9                   4   \n",
       "\n",
       "  CD_Labeled CD_MemberList_LogSize CD_CommunityName CD_AnnotatedMembers  \\\n",
       "0      False                 8.762                                        \n",
       "1      False                 2.322                                        \n",
       "2      False                 2.322                                        \n",
       "3      False                   2.0                                        \n",
       "4      False                   2.0                                        \n",
       "\n",
       "  CD_AnnotatedMembers_Size CD_AnnotatedMembers_Overlap  \\\n",
       "0                        0                         0.0   \n",
       "1                        0                         0.0   \n",
       "2                        0                         0.0   \n",
       "3                        0                         0.0   \n",
       "4                        0                         0.0   \n",
       "\n",
       "  CD_AnnotatedMembers_Pvalue  HiDeF_persistence represents  name  \\\n",
       "0                        0.0                120       C439  C439   \n",
       "1                        0.0                  5       C516  C516   \n",
       "2                        0.0                  5       C517  C517   \n",
       "3                        0.0                 43       C518  C518   \n",
       "4                        0.0                 27       C524  C524   \n",
       "\n",
       "  CD_MemberList_LogSize_viz  system_ID  \n",
       "0                     87.62          0  \n",
       "1                     23.22          1  \n",
       "2                     23.22          2  \n",
       "3                      20.0          3  \n",
       "4                      20.0          4  "
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# add node attributes to dataframe for easier access\n",
    "hier_df = pd.DataFrame.from_dict(dict(G_hier.nodes(data=True)), orient='index')\n",
    "hier_df['system_ID']=hier_df.index.tolist()\n",
    "# some columns are not the right type\n",
    "hier_df['CD_MemberList_Size']=[int(x) for x in hier_df['CD_MemberList_Size'].tolist()]\n",
    "hier_df['HiDeF_persistence']=[int(x) for x in hier_df['HiDeF_persistence'].tolist()]\n",
    "hier_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f08f350c",
   "metadata": {},
   "source": [
    "## 12. Remove systems with no seed genes (OPTIONAL)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "id": "c03fd7cd",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'seed_dict' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Input \u001b[0;32mIn [97]\u001b[0m, in \u001b[0;36m<cell line: 8>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      8\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m c \u001b[38;5;129;01min\u001b[39;00m hier_df\u001b[38;5;241m.\u001b[39mindex\u001b[38;5;241m.\u001b[39mtolist():\n\u001b[1;32m      9\u001b[0m     system_genes \u001b[38;5;241m=\u001b[39m hier_df[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mCD_MemberList\u001b[39m\u001b[38;5;124m'\u001b[39m]\u001b[38;5;241m.\u001b[39mloc[c]\u001b[38;5;241m.\u001b[39msplit(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m \u001b[39m\u001b[38;5;124m'\u001b[39m)\n\u001b[0;32m---> 10\u001b[0m     d1_temp \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mlist\u001b[39m(np\u001b[38;5;241m.\u001b[39mintersect1d(system_genes,\u001b[43mseed_dict\u001b[49m[d1_name]))\n\u001b[1;32m     11\u001b[0m     d2_temp \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mlist\u001b[39m(np\u001b[38;5;241m.\u001b[39mintersect1d(system_genes,seed_dict[d2_name]))\n\u001b[1;32m     12\u001b[0m     num_d1_temp \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mlen\u001b[39m(d1_temp)\n",
      "\u001b[0;31mNameError\u001b[0m: name 'seed_dict' is not defined"
     ]
    }
   ],
   "source": [
    "hier_df.index=hier_df['name']\n",
    "hier_df.head()\n",
    "\n",
    "num_d1_seeds, num_d2_seeds, num_both_seeds = [],[],[]\n",
    "frac_d1_seeds, frac_d2_seeds, frac_both_seeds=[],[],[]\n",
    "\n",
    "systems_keep = []\n",
    "for c in hier_df.index.tolist():\n",
    "    system_genes = hier_df['CD_MemberList'].loc[c].split(' ')\n",
    "    d1_temp = list(np.intersect1d(system_genes,seed_dict[d1_name]))\n",
    "    d2_temp = list(np.intersect1d(system_genes,seed_dict[d2_name]))\n",
    "    num_d1_temp = len(d1_temp)\n",
    "    num_d2_temp = len(d2_temp)\n",
    "    if (num_d1_temp+num_d2_temp)>0: # keep the system if it has at least 1 seed gene\n",
    "        systems_keep.append(c)\n",
    "        num_both_temp = len(np.intersect1d(d1_temp,d2_temp))\n",
    "        num_both_seeds.append(num_both_temp)\n",
    "        num_d1_seeds.append(num_d1_temp-num_both_temp)\n",
    "        num_d2_seeds.append(num_d2_temp-num_both_temp)\n",
    "        \n",
    "        frac_both_seeds.append(num_both_temp/float(len(system_genes)))\n",
    "        frac_d1_seeds.append((num_d1_temp-num_both_temp)/float(len(system_genes)))\n",
    "        frac_d2_seeds.append((num_d2_temp-num_both_temp)/float(len(system_genes)))\n",
    "\n",
    "        \n",
    "frac_no_seeds = np.subtract(1.0,np.array([frac_d1_seeds,frac_d2_seeds,frac_both_seeds]).sum(axis=0))\n",
    "\n",
    "hier_df = hier_df.loc[systems_keep]\n",
    "hier_df['num_d1_seeds']=num_d1_seeds\n",
    "hier_df['num_d2_seeds']=num_d2_seeds\n",
    "hier_df['num_both_seeds']=num_d2_seeds\n",
    "hier_df['frac_d1_seeds']=frac_d1_seeds\n",
    "hier_df['frac_d2_seeds']=frac_d2_seeds\n",
    "hier_df['frac_both_seeds']=frac_both_seeds\n",
    "hier_df['frac_no_seeds']=frac_no_seeds\n",
    "print(\"Number of nodes with seed genes:\", len(hier_df))\n",
    "\n",
    "hier_df.head()\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "eacafd30",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "id": "b9525685",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of nodes with seed genes: 0\n",
      "Number of edges remaining: 0\n"
     ]
    }
   ],
   "source": [
    "# prune G_hier--> only keep systems with at least one seed gene\n",
    "\n",
    "nkeep=[]\n",
    "for n in list(G_hier.nodes()):\n",
    "    if G_hier.nodes(data=True)[n]['name'] in systems_keep:\n",
    "        nkeep.append(n)\n",
    "        \n",
    "\n",
    "G_hier = nx.subgraph(G_hier, nkeep)\n",
    "print(\"Number of nodes with seed genes:\", len(G_hier.nodes()))\n",
    "print(\"Number of edges remaining:\", len(G_hier.edges()))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "545780ac",
   "metadata": {},
   "source": [
    "## 13. Sneak peek of hierarchy visualization (OPTIONAL)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "id": "b98bae88",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "81e4b9162cd5425390dd619a1f0ba3bc",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "CytoscapeWidget(cytoscape_layout={'name': 'cose'}, cytoscape_style=[{'selector': 'node', 'css': {'content': 'd…"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "network_colocalization.view_G_hier(G_hier)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f1391f8e",
   "metadata": {},
   "source": [
    "## 14. Annotate systems with gprofiler."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "id": "033f7e41",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C439\n",
      "434\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>RHO GTPase cycle</td>\n",
       "      <td>10835</td>\n",
       "      <td>30</td>\n",
       "      <td>RHO GTPase cycle</td>\n",
       "      <td>REAC:R-HSA-9012999</td>\n",
       "      <td>0.000008</td>\n",
       "      <td>[REAC:R-HSA-194315]</td>\n",
       "      <td>0.129310</td>\n",
       "      <td>query_1</td>\n",
       "      <td>232</td>\n",
       "      <td>0.066815</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>449</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Signaling by Rho GTPases</td>\n",
       "      <td>10835</td>\n",
       "      <td>40</td>\n",
       "      <td>Signaling by Rho GTPases</td>\n",
       "      <td>REAC:R-HSA-194315</td>\n",
       "      <td>0.000008</td>\n",
       "      <td>[REAC:R-HSA-9716542]</td>\n",
       "      <td>0.172414</td>\n",
       "      <td>query_1</td>\n",
       "      <td>232</td>\n",
       "      <td>0.056980</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>702</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Signaling by Rho GTPases, Miro GTPases and RHO...</td>\n",
       "      <td>10835</td>\n",
       "      <td>40</td>\n",
       "      <td>Signaling by Rho GTPases, Miro GTPases and RHO...</td>\n",
       "      <td>REAC:R-HSA-9716542</td>\n",
       "      <td>0.000008</td>\n",
       "      <td>[REAC:R-HSA-162582]</td>\n",
       "      <td>0.172414</td>\n",
       "      <td>query_1</td>\n",
       "      <td>232</td>\n",
       "      <td>0.055788</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>717</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                         description  effective_domain_size  \\\n",
       "2                                   RHO GTPase cycle                  10835   \n",
       "0                           Signaling by Rho GTPases                  10835   \n",
       "1  Signaling by Rho GTPases, Miro GTPases and RHO...                  10835   \n",
       "\n",
       "   intersection_size                                               name  \\\n",
       "2                 30                                   RHO GTPase cycle   \n",
       "0                 40                           Signaling by Rho GTPases   \n",
       "1                 40  Signaling by Rho GTPases, Miro GTPases and RHO...   \n",
       "\n",
       "               native   p_value               parents  precision    query  \\\n",
       "2  REAC:R-HSA-9012999  0.000008   [REAC:R-HSA-194315]   0.129310  query_1   \n",
       "0   REAC:R-HSA-194315  0.000008  [REAC:R-HSA-9716542]   0.172414  query_1   \n",
       "1  REAC:R-HSA-9716542  0.000008   [REAC:R-HSA-162582]   0.172414  query_1   \n",
       "\n",
       "   query_size    recall  significant source  term_size  \n",
       "2         232  0.066815         True   REAC        449  \n",
       "0         232  0.056980         True   REAC        702  \n",
       "1         232  0.055788         True   REAC        717  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C516\n",
      "5\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C517\n",
      "5\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C518\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C524\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C529\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C533\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C535\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C539\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C542\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C547\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C440\n",
      "149\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Nervous system development</td>\n",
       "      <td>10835</td>\n",
       "      <td>22</td>\n",
       "      <td>Nervous system development</td>\n",
       "      <td>REAC:R-HSA-9675108</td>\n",
       "      <td>2.541969e-07</td>\n",
       "      <td>[REAC:R-HSA-1266738]</td>\n",
       "      <td>0.250000</td>\n",
       "      <td>query_1</td>\n",
       "      <td>88</td>\n",
       "      <td>0.038328</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>574</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Axon guidance</td>\n",
       "      <td>10835</td>\n",
       "      <td>21</td>\n",
       "      <td>Axon guidance</td>\n",
       "      <td>REAC:R-HSA-422475</td>\n",
       "      <td>3.504396e-07</td>\n",
       "      <td>[REAC:R-HSA-9675108]</td>\n",
       "      <td>0.238636</td>\n",
       "      <td>query_1</td>\n",
       "      <td>88</td>\n",
       "      <td>0.038251</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>549</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  description  effective_domain_size  intersection_size  \\\n",
       "0  Nervous system development                  10835                 22   \n",
       "1               Axon guidance                  10835                 21   \n",
       "\n",
       "                         name              native       p_value  \\\n",
       "0  Nervous system development  REAC:R-HSA-9675108  2.541969e-07   \n",
       "1               Axon guidance   REAC:R-HSA-422475  3.504396e-07   \n",
       "\n",
       "                parents  precision    query  query_size    recall  \\\n",
       "0  [REAC:R-HSA-1266738]   0.250000  query_1          88  0.038328   \n",
       "1  [REAC:R-HSA-9675108]   0.238636  query_1          88  0.038251   \n",
       "\n",
       "   significant source  term_size  \n",
       "0         True   REAC        574  \n",
       "1         True   REAC        549  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C441\n",
      "95\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C442\n",
      "58\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Chromatin modifying enzymes</td>\n",
       "      <td>10835</td>\n",
       "      <td>11</td>\n",
       "      <td>Chromatin modifying enzymes</td>\n",
       "      <td>REAC:R-HSA-3247509</td>\n",
       "      <td>3.294937e-09</td>\n",
       "      <td>[REAC:R-HSA-4839726]</td>\n",
       "      <td>0.392857</td>\n",
       "      <td>query_1</td>\n",
       "      <td>28</td>\n",
       "      <td>0.040293</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>273</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Chromatin organization</td>\n",
       "      <td>10835</td>\n",
       "      <td>11</td>\n",
       "      <td>Chromatin organization</td>\n",
       "      <td>REAC:R-HSA-4839726</td>\n",
       "      <td>3.294937e-09</td>\n",
       "      <td>[REAC:0000000]</td>\n",
       "      <td>0.392857</td>\n",
       "      <td>query_1</td>\n",
       "      <td>28</td>\n",
       "      <td>0.040293</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>273</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   description  effective_domain_size  intersection_size  \\\n",
       "0  Chromatin modifying enzymes                  10835                 11   \n",
       "1       Chromatin organization                  10835                 11   \n",
       "\n",
       "                          name              native       p_value  \\\n",
       "0  Chromatin modifying enzymes  REAC:R-HSA-3247509  3.294937e-09   \n",
       "1       Chromatin organization  REAC:R-HSA-4839726  3.294937e-09   \n",
       "\n",
       "                parents  precision    query  query_size    recall  \\\n",
       "0  [REAC:R-HSA-4839726]   0.392857  query_1          28  0.040293   \n",
       "1        [REAC:0000000]   0.392857  query_1          28  0.040293   \n",
       "\n",
       "   significant source  term_size  \n",
       "0         True   REAC        273  \n",
       "1         True   REAC        273  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C445\n",
      "33\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C446\n",
      "31\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C447\n",
      "29\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C448\n",
      "28\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C453\n",
      "19\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C457\n",
      "12\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C464\n",
      "9\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C468\n",
      "9\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C471\n",
      "8\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C473\n",
      "8\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C480\n",
      "7\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C481\n",
      "7\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C484\n",
      "6\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C488\n",
      "6\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C489\n",
      "6\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C490\n",
      "6\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C491\n",
      "6\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C495\n",
      "6\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C496\n",
      "5\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C501\n",
      "5\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C511\n",
      "5\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C512\n",
      "5\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C449\n",
      "27\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Clathrin-mediated endocytosis</td>\n",
       "      <td>10835</td>\n",
       "      <td>10</td>\n",
       "      <td>Clathrin-mediated endocytosis</td>\n",
       "      <td>REAC:R-HSA-8856828</td>\n",
       "      <td>1.349967e-11</td>\n",
       "      <td>[REAC:R-HSA-199991]</td>\n",
       "      <td>0.434783</td>\n",
       "      <td>query_1</td>\n",
       "      <td>23</td>\n",
       "      <td>0.069444</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>144</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Parasite infection</td>\n",
       "      <td>10835</td>\n",
       "      <td>6</td>\n",
       "      <td>Parasite infection</td>\n",
       "      <td>REAC:R-HSA-9664407</td>\n",
       "      <td>7.158181e-07</td>\n",
       "      <td>[REAC:R-HSA-9658195]</td>\n",
       "      <td>0.260870</td>\n",
       "      <td>query_1</td>\n",
       "      <td>23</td>\n",
       "      <td>0.053097</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Leishmania phagocytosis</td>\n",
       "      <td>10835</td>\n",
       "      <td>6</td>\n",
       "      <td>Leishmania phagocytosis</td>\n",
       "      <td>REAC:R-HSA-9664417</td>\n",
       "      <td>7.158181e-07</td>\n",
       "      <td>[REAC:R-HSA-9664407]</td>\n",
       "      <td>0.260870</td>\n",
       "      <td>query_1</td>\n",
       "      <td>23</td>\n",
       "      <td>0.053097</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>FCGR3A-mediated phagocytosis</td>\n",
       "      <td>10835</td>\n",
       "      <td>6</td>\n",
       "      <td>FCGR3A-mediated phagocytosis</td>\n",
       "      <td>REAC:R-HSA-9664422</td>\n",
       "      <td>7.158181e-07</td>\n",
       "      <td>[REAC:R-HSA-9664417]</td>\n",
       "      <td>0.260870</td>\n",
       "      <td>query_1</td>\n",
       "      <td>23</td>\n",
       "      <td>0.053097</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Regulation of actin dynamics for phagocytic cu...</td>\n",
       "      <td>10835</td>\n",
       "      <td>6</td>\n",
       "      <td>Regulation of actin dynamics for phagocytic cu...</td>\n",
       "      <td>REAC:R-HSA-2029482</td>\n",
       "      <td>7.837873e-07</td>\n",
       "      <td>[REAC:R-HSA-2029480]</td>\n",
       "      <td>0.260870</td>\n",
       "      <td>query_1</td>\n",
       "      <td>23</td>\n",
       "      <td>0.051724</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>116</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                          description  effective_domain_size  \\\n",
       "0                       Clathrin-mediated endocytosis                  10835   \n",
       "12                                 Parasite infection                  10835   \n",
       "13                            Leishmania phagocytosis                  10835   \n",
       "14                       FCGR3A-mediated phagocytosis                  10835   \n",
       "15  Regulation of actin dynamics for phagocytic cu...                  10835   \n",
       "\n",
       "    intersection_size                                               name  \\\n",
       "0                  10                      Clathrin-mediated endocytosis   \n",
       "12                  6                                 Parasite infection   \n",
       "13                  6                            Leishmania phagocytosis   \n",
       "14                  6                       FCGR3A-mediated phagocytosis   \n",
       "15                  6  Regulation of actin dynamics for phagocytic cu...   \n",
       "\n",
       "                native       p_value               parents  precision  \\\n",
       "0   REAC:R-HSA-8856828  1.349967e-11   [REAC:R-HSA-199991]   0.434783   \n",
       "12  REAC:R-HSA-9664407  7.158181e-07  [REAC:R-HSA-9658195]   0.260870   \n",
       "13  REAC:R-HSA-9664417  7.158181e-07  [REAC:R-HSA-9664407]   0.260870   \n",
       "14  REAC:R-HSA-9664422  7.158181e-07  [REAC:R-HSA-9664417]   0.260870   \n",
       "15  REAC:R-HSA-2029482  7.837873e-07  [REAC:R-HSA-2029480]   0.260870   \n",
       "\n",
       "      query  query_size    recall  significant source  term_size  \n",
       "0   query_1          23  0.069444         True   REAC        144  \n",
       "12  query_1          23  0.053097         True   REAC        113  \n",
       "13  query_1          23  0.053097         True   REAC        113  \n",
       "14  query_1          23  0.053097         True   REAC        113  \n",
       "15  query_1          23  0.051724         True   REAC        116  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C544\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C451\n",
      "21\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C549\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C487\n",
      "6\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C522\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C459\n",
      "11\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C460\n",
      "11\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C492\n",
      "6\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C462\n",
      "10\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C466\n",
      "9\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C531\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C532\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C472\n",
      "8\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Ion homeostasis</td>\n",
       "      <td>10835</td>\n",
       "      <td>4</td>\n",
       "      <td>Ion homeostasis</td>\n",
       "      <td>REAC:R-HSA-5578775</td>\n",
       "      <td>0.000002</td>\n",
       "      <td>[REAC:R-HSA-5576891]</td>\n",
       "      <td>0.5</td>\n",
       "      <td>query_1</td>\n",
       "      <td>8</td>\n",
       "      <td>0.074074</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>54</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       description  effective_domain_size  intersection_size             name  \\\n",
       "0  Ion homeostasis                  10835                  4  Ion homeostasis   \n",
       "\n",
       "               native   p_value               parents  precision    query  \\\n",
       "0  REAC:R-HSA-5578775  0.000002  [REAC:R-HSA-5576891]        0.5  query_1   \n",
       "\n",
       "   query_size    recall  significant source  term_size  \n",
       "0           8  0.074074         True   REAC         54  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C540\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C478\n",
      "7\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C546\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C458\n",
      "12\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C527\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C500\n",
      "5\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C469\n",
      "8\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C538\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C443\n",
      "52\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C477\n",
      "7\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C479\n",
      "7\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C523\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C552\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C452\n",
      "19\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Chromatin modifying enzymes</td>\n",
       "      <td>10835</td>\n",
       "      <td>5</td>\n",
       "      <td>Chromatin modifying enzymes</td>\n",
       "      <td>REAC:R-HSA-3247509</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>[REAC:R-HSA-4839726]</td>\n",
       "      <td>0.833333</td>\n",
       "      <td>query_1</td>\n",
       "      <td>6</td>\n",
       "      <td>0.018315</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>273</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Chromatin organization</td>\n",
       "      <td>10835</td>\n",
       "      <td>5</td>\n",
       "      <td>Chromatin organization</td>\n",
       "      <td>REAC:R-HSA-4839726</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>[REAC:0000000]</td>\n",
       "      <td>0.833333</td>\n",
       "      <td>query_1</td>\n",
       "      <td>6</td>\n",
       "      <td>0.018315</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>273</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   description  effective_domain_size  intersection_size  \\\n",
       "0  Chromatin modifying enzymes                  10835                  5   \n",
       "1       Chromatin organization                  10835                  5   \n",
       "\n",
       "                          name              native   p_value  \\\n",
       "0  Chromatin modifying enzymes  REAC:R-HSA-3247509  0.000001   \n",
       "1       Chromatin organization  REAC:R-HSA-4839726  0.000001   \n",
       "\n",
       "                parents  precision    query  query_size    recall  \\\n",
       "0  [REAC:R-HSA-4839726]   0.833333  query_1           6  0.018315   \n",
       "1        [REAC:0000000]   0.833333  query_1           6  0.018315   \n",
       "\n",
       "   significant source  term_size  \n",
       "0         True   REAC        273  \n",
       "1         True   REAC        273  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C444\n",
      "34\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C519\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C493\n",
      "6\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C497\n",
      "5\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Antigen processing: Ubiquitination &amp; Proteasom...</td>\n",
       "      <td>10835</td>\n",
       "      <td>5</td>\n",
       "      <td>Antigen processing: Ubiquitination &amp; Proteasom...</td>\n",
       "      <td>REAC:R-HSA-983168</td>\n",
       "      <td>9.170868e-07</td>\n",
       "      <td>[REAC:R-HSA-983169]</td>\n",
       "      <td>1.0</td>\n",
       "      <td>query_1</td>\n",
       "      <td>5</td>\n",
       "      <td>0.016234</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>308</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Class I MHC mediated antigen processing &amp; pres...</td>\n",
       "      <td>10835</td>\n",
       "      <td>5</td>\n",
       "      <td>Class I MHC mediated antigen processing &amp; pres...</td>\n",
       "      <td>REAC:R-HSA-983169</td>\n",
       "      <td>1.318986e-06</td>\n",
       "      <td>[REAC:R-HSA-1280218]</td>\n",
       "      <td>1.0</td>\n",
       "      <td>query_1</td>\n",
       "      <td>5</td>\n",
       "      <td>0.013158</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>380</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                         description  effective_domain_size  \\\n",
       "0  Antigen processing: Ubiquitination & Proteasom...                  10835   \n",
       "1  Class I MHC mediated antigen processing & pres...                  10835   \n",
       "\n",
       "   intersection_size                                               name  \\\n",
       "0                  5  Antigen processing: Ubiquitination & Proteasom...   \n",
       "1                  5  Class I MHC mediated antigen processing & pres...   \n",
       "\n",
       "              native       p_value               parents  precision    query  \\\n",
       "0  REAC:R-HSA-983168  9.170868e-07   [REAC:R-HSA-983169]        1.0  query_1   \n",
       "1  REAC:R-HSA-983169  1.318986e-06  [REAC:R-HSA-1280218]        1.0  query_1   \n",
       "\n",
       "   query_size    recall  significant source  term_size  \n",
       "0           5  0.016234         True   REAC        308  \n",
       "1           5  0.013158         True   REAC        380  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C498\n",
      "5\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Macroautophagy</td>\n",
       "      <td>10835</td>\n",
       "      <td>4</td>\n",
       "      <td>Macroautophagy</td>\n",
       "      <td>REAC:R-HSA-1632852</td>\n",
       "      <td>1.889756e-07</td>\n",
       "      <td>[REAC:R-HSA-9612973]</td>\n",
       "      <td>1.0</td>\n",
       "      <td>query_1</td>\n",
       "      <td>4</td>\n",
       "      <td>0.029851</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>134</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Autophagy</td>\n",
       "      <td>10835</td>\n",
       "      <td>4</td>\n",
       "      <td>Autophagy</td>\n",
       "      <td>REAC:R-HSA-9612973</td>\n",
       "      <td>1.889756e-07</td>\n",
       "      <td>[REAC:0000000]</td>\n",
       "      <td>1.0</td>\n",
       "      <td>query_1</td>\n",
       "      <td>4</td>\n",
       "      <td>0.026846</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>149</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      description  effective_domain_size  intersection_size            name  \\\n",
       "1  Macroautophagy                  10835                  4  Macroautophagy   \n",
       "0       Autophagy                  10835                  4       Autophagy   \n",
       "\n",
       "               native       p_value               parents  precision    query  \\\n",
       "1  REAC:R-HSA-1632852  1.889756e-07  [REAC:R-HSA-9612973]        1.0  query_1   \n",
       "0  REAC:R-HSA-9612973  1.889756e-07        [REAC:0000000]        1.0  query_1   \n",
       "\n",
       "   query_size    recall  significant source  term_size  \n",
       "1           4  0.029851         True   REAC        134  \n",
       "0           4  0.026846         True   REAC        149  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C502\n",
      "5\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C536\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C537\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C514\n",
      "5\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C467\n",
      "9\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C454\n",
      "17\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C545\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C450\n",
      "22\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C482\n",
      "7\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C455\n",
      "15\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C476\n",
      "7\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C456\n",
      "15\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C461\n",
      "11\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Clathrin-mediated endocytosis</td>\n",
       "      <td>10835</td>\n",
       "      <td>7</td>\n",
       "      <td>Clathrin-mediated endocytosis</td>\n",
       "      <td>REAC:R-HSA-8856828</td>\n",
       "      <td>1.178411e-09</td>\n",
       "      <td>[REAC:R-HSA-199991]</td>\n",
       "      <td>0.636364</td>\n",
       "      <td>query_1</td>\n",
       "      <td>11</td>\n",
       "      <td>0.048611</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>144</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     description  effective_domain_size  intersection_size  \\\n",
       "0  Clathrin-mediated endocytosis                  10835                  7   \n",
       "\n",
       "                            name              native       p_value  \\\n",
       "0  Clathrin-mediated endocytosis  REAC:R-HSA-8856828  1.178411e-09   \n",
       "\n",
       "               parents  precision    query  query_size    recall  significant  \\\n",
       "0  [REAC:R-HSA-199991]   0.636364  query_1          11  0.048611         True   \n",
       "\n",
       "  source  term_size  \n",
       "0   REAC        144  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C463\n",
      "9\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FCGR3A-mediated phagocytosis</td>\n",
       "      <td>10835</td>\n",
       "      <td>4</td>\n",
       "      <td>FCGR3A-mediated phagocytosis</td>\n",
       "      <td>REAC:R-HSA-9664422</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>[REAC:R-HSA-9664417]</td>\n",
       "      <td>0.571429</td>\n",
       "      <td>query_1</td>\n",
       "      <td>7</td>\n",
       "      <td>0.035398</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Parasite infection</td>\n",
       "      <td>10835</td>\n",
       "      <td>4</td>\n",
       "      <td>Parasite infection</td>\n",
       "      <td>REAC:R-HSA-9664407</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>[REAC:R-HSA-9658195]</td>\n",
       "      <td>0.571429</td>\n",
       "      <td>query_1</td>\n",
       "      <td>7</td>\n",
       "      <td>0.035398</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Leishmania phagocytosis</td>\n",
       "      <td>10835</td>\n",
       "      <td>4</td>\n",
       "      <td>Leishmania phagocytosis</td>\n",
       "      <td>REAC:R-HSA-9664417</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>[REAC:R-HSA-9664407]</td>\n",
       "      <td>0.571429</td>\n",
       "      <td>query_1</td>\n",
       "      <td>7</td>\n",
       "      <td>0.035398</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Regulation of actin dynamics for phagocytic cu...</td>\n",
       "      <td>10835</td>\n",
       "      <td>4</td>\n",
       "      <td>Regulation of actin dynamics for phagocytic cu...</td>\n",
       "      <td>REAC:R-HSA-2029482</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>[REAC:R-HSA-2029480]</td>\n",
       "      <td>0.571429</td>\n",
       "      <td>query_1</td>\n",
       "      <td>7</td>\n",
       "      <td>0.034483</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Fcgamma receptor (FCGR) dependent phagocytosis</td>\n",
       "      <td>10835</td>\n",
       "      <td>4</td>\n",
       "      <td>Fcgamma receptor (FCGR) dependent phagocytosis</td>\n",
       "      <td>REAC:R-HSA-2029480</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>[REAC:R-HSA-168249]</td>\n",
       "      <td>0.571429</td>\n",
       "      <td>query_1</td>\n",
       "      <td>7</td>\n",
       "      <td>0.028571</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>140</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                         description  effective_domain_size  \\\n",
       "3                       FCGR3A-mediated phagocytosis                  10835   \n",
       "5                                 Parasite infection                  10835   \n",
       "7                            Leishmania phagocytosis                  10835   \n",
       "2  Regulation of actin dynamics for phagocytic cu...                  10835   \n",
       "9     Fcgamma receptor (FCGR) dependent phagocytosis                  10835   \n",
       "\n",
       "   intersection_size                                               name  \\\n",
       "3                  4                       FCGR3A-mediated phagocytosis   \n",
       "5                  4                                 Parasite infection   \n",
       "7                  4                            Leishmania phagocytosis   \n",
       "2                  4  Regulation of actin dynamics for phagocytic cu...   \n",
       "9                  4     Fcgamma receptor (FCGR) dependent phagocytosis   \n",
       "\n",
       "               native   p_value               parents  precision    query  \\\n",
       "3  REAC:R-HSA-9664422  0.000003  [REAC:R-HSA-9664417]   0.571429  query_1   \n",
       "5  REAC:R-HSA-9664407  0.000003  [REAC:R-HSA-9658195]   0.571429  query_1   \n",
       "7  REAC:R-HSA-9664417  0.000003  [REAC:R-HSA-9664407]   0.571429  query_1   \n",
       "2  REAC:R-HSA-2029482  0.000003  [REAC:R-HSA-2029480]   0.571429  query_1   \n",
       "9  REAC:R-HSA-2029480  0.000005   [REAC:R-HSA-168249]   0.571429  query_1   \n",
       "\n",
       "   query_size    recall  significant source  term_size  \n",
       "3           7  0.035398         True   REAC        113  \n",
       "5           7  0.035398         True   REAC        113  \n",
       "7           7  0.035398         True   REAC        113  \n",
       "2           7  0.034483         True   REAC        116  \n",
       "9           7  0.028571         True   REAC        140  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C483\n",
      "6\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C553\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C474\n",
      "7\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C465\n",
      "9\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C515\n",
      "5\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C485\n",
      "6\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C555\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C507\n",
      "5\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C534\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C470\n",
      "8\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C494\n",
      "6\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C508\n",
      "5\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C499\n",
      "5\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C509\n",
      "5\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C504\n",
      "5\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C520\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Clathrin-mediated endocytosis</td>\n",
       "      <td>10835</td>\n",
       "      <td>4</td>\n",
       "      <td>Clathrin-mediated endocytosis</td>\n",
       "      <td>REAC:R-HSA-8856828</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>[REAC:R-HSA-199991]</td>\n",
       "      <td>1.0</td>\n",
       "      <td>query_1</td>\n",
       "      <td>4</td>\n",
       "      <td>0.027778</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>144</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     description  effective_domain_size  intersection_size  \\\n",
       "0  Clathrin-mediated endocytosis                  10835                  4   \n",
       "\n",
       "                            name              native   p_value  \\\n",
       "0  Clathrin-mediated endocytosis  REAC:R-HSA-8856828  0.000001   \n",
       "\n",
       "               parents  precision    query  query_size    recall  significant  \\\n",
       "0  [REAC:R-HSA-199991]        1.0  query_1           4  0.027778         True   \n",
       "\n",
       "  source  term_size  \n",
       "0   REAC        144  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C475\n",
      "7\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C530\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C486\n",
      "6\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C505\n",
      "5\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C513\n",
      "5\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Ion homeostasis</td>\n",
       "      <td>10835</td>\n",
       "      <td>3</td>\n",
       "      <td>Ion homeostasis</td>\n",
       "      <td>REAC:R-HSA-5578775</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>[REAC:R-HSA-5576891]</td>\n",
       "      <td>0.6</td>\n",
       "      <td>query_1</td>\n",
       "      <td>5</td>\n",
       "      <td>0.055556</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Transport of small molecules</td>\n",
       "      <td>10835</td>\n",
       "      <td>5</td>\n",
       "      <td>Transport of small molecules</td>\n",
       "      <td>REAC:R-HSA-382551</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>[REAC:0000000]</td>\n",
       "      <td>1.0</td>\n",
       "      <td>query_1</td>\n",
       "      <td>5</td>\n",
       "      <td>0.006964</td>\n",
       "      <td>True</td>\n",
       "      <td>REAC</td>\n",
       "      <td>718</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    description  effective_domain_size  intersection_size  \\\n",
       "0               Ion homeostasis                  10835                  3   \n",
       "1  Transport of small molecules                  10835                  5   \n",
       "\n",
       "                           name              native   p_value  \\\n",
       "0               Ion homeostasis  REAC:R-HSA-5578775  0.000009   \n",
       "1  Transport of small molecules   REAC:R-HSA-382551  0.000009   \n",
       "\n",
       "                parents  precision    query  query_size    recall  \\\n",
       "0  [REAC:R-HSA-5576891]        0.6  query_1           5  0.055556   \n",
       "1        [REAC:0000000]        1.0  query_1           5  0.006964   \n",
       "\n",
       "   significant source  term_size  \n",
       "0         True   REAC         54  \n",
       "1         True   REAC        718  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C526\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C506\n",
      "5\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C543\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C525\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C503\n",
      "5\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C554\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C551\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C548\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C510\n",
      "5\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C521\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C528\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C550\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C541\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>name</th>\n",
       "      <th>native</th>\n",
       "      <th>p_value</th>\n",
       "      <th>parents</th>\n",
       "      <th>precision</th>\n",
       "      <th>query</th>\n",
       "      <th>query_size</th>\n",
       "      <th>recall</th>\n",
       "      <th>significant</th>\n",
       "      <th>source</th>\n",
       "      <th>term_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [description, effective_domain_size, intersection_size, name, native, p_value, parents, precision, query, query_size, recall, significant, source, term_size]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# gprofiler annotation of clusters\n",
    "\n",
    "system_name_list = []\n",
    "for p in hier_df.index.tolist():\n",
    "    focal_genes=hier_df['CD_MemberList'].loc[p].split(' ')\n",
    "    print(p)\n",
    "    print(len(focal_genes))\n",
    "    if len(focal_genes)>2:\n",
    "        gp_temp = pd.DataFrame(gp.profile(focal_genes,significance_threshold_method='fdr',\n",
    "                                               sources=['REAC']))\n",
    "        if len(gp_temp)>0: # make sure data is not empty\n",
    "            \n",
    "            # make sure terms are specific, and overlap with at least 3 genes\n",
    "            gp_temp = gp_temp[(gp_temp['term_size']<1000)&(gp_temp['term_size']>50)]\n",
    "            gp_temp = gp_temp[gp_temp['intersection_size']>=3]\n",
    "            \n",
    "            gp_temp = gp_temp[gp_temp['p_value']<1E-5] # set a stringent pvalue threshold\n",
    "            \n",
    "            gp_temp = gp_temp.sort_values('recall',ascending=False)\n",
    "            \n",
    "            if len(gp_temp)>1:\n",
    "                system_name_list.append(gp_temp.head(1)['name'].tolist()[0])\n",
    "            else:\n",
    "                system_name_list.append(p)\n",
    "        else:\n",
    "            system_name_list.append(p)\n",
    "            \n",
    "\n",
    "        display(gp_temp.head())\n",
    "        \n",
    "    else:\n",
    "        system_name_list.append(p)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ac6a07cc",
   "metadata": {},
   "source": [
    "# Validate identified genes and systems\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8cf619aa",
   "metadata": {},
   "source": [
    "## 15. Load and parse mouse variant database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "id": "9e936cad",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "querying 1-1000...done.\n",
      "querying 1001-2000...done.\n",
      "querying 2001-3000...done.\n",
      "querying 3001-4000...done.\n",
      "querying 4001-5000...done.\n",
      "querying 5001-6000...done.\n",
      "querying 6001-7000...done.\n",
      "querying 7001-8000...done.\n",
      "querying 8001-9000...done.\n",
      "querying 9001-10000...done.\n",
      "querying 10001-11000...done.\n",
      "querying 11001-12000...done.\n",
      "querying 12001-13000...done.\n",
      "querying 13001-14000...done.\n",
      "querying 14001-15000...done.\n",
      "querying 15001-16000...done.\n",
      "querying 16001-17000...done.\n",
      "querying 17001-18000...done.\n",
      "querying 18001-19000...done.\n",
      "querying 19001-20000...done.\n",
      "querying 20001-21000...done.\n",
      "querying 21001-22000...done.\n",
      "querying 22001-23000...done.\n",
      "querying 23001-24000...done.\n",
      "querying 24001-25000...done.\n",
      "querying 25001-25247...done.\n",
      "Finished.\n",
      "33 input query terms found dup hits:\n",
      "\t[('Adam1a', 2), ('Adam1b', 2), ('Betalinc1', 2), ('Chaserr', 2), ('Dnm3os', 2), ('Eldr', 2), ('Fer1l\n",
      "12515 input query terms found no hit:\n",
      "\t['+|Cav1', '+|Cm', '+|Del(10Prmt2-Pdxk)4Yey', '+|Del(11Mpo-Chad)18Brd', '+|Del(11Rr325)1Smun', '+|De\n",
      "Pass \"returnall=True\" to return complete lists of duplicate or missing query terms.\n",
      "25286\n",
      "12771\n",
      "12732\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MGI_Allele_Accession_ID</th>\n",
       "      <th>Allele symbol</th>\n",
       "      <th>involves</th>\n",
       "      <th>MP</th>\n",
       "      <th>PMID</th>\n",
       "      <th>MGI_marker_accession_ID</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>human_ortholog</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Rb1</th>\n",
       "      <td>Rb1&lt;tm1Tyj&gt;/Rb1&lt;tm1Tyj&gt;</td>\n",
       "      <td>Rb1&lt;tm1Tyj&gt;</td>\n",
       "      <td>involves: 129S2/SvPas</td>\n",
       "      <td>MP:0000600</td>\n",
       "      <td>12529408</td>\n",
       "      <td>MGI:97874</td>\n",
       "      <td>Rb1</td>\n",
       "      <td>RB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Rb1</th>\n",
       "      <td>Rb1&lt;tm1Tyj&gt;/Rb1&lt;tm1Tyj&gt;</td>\n",
       "      <td>Rb1&lt;tm1Tyj&gt;</td>\n",
       "      <td>involves: 129S2/SvPas</td>\n",
       "      <td>MP:0001716</td>\n",
       "      <td>16449662</td>\n",
       "      <td>MGI:97874</td>\n",
       "      <td>Rb1</td>\n",
       "      <td>RB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Rb1</th>\n",
       "      <td>Rb1&lt;tm1Tyj&gt;/Rb1&lt;tm1Tyj&gt;</td>\n",
       "      <td>Rb1&lt;tm1Tyj&gt;</td>\n",
       "      <td>involves: 129S2/SvPas</td>\n",
       "      <td>MP:0001698</td>\n",
       "      <td>16449662</td>\n",
       "      <td>MGI:97874</td>\n",
       "      <td>Rb1</td>\n",
       "      <td>RB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Rb1</th>\n",
       "      <td>Rb1&lt;tm1Tyj&gt;/Rb1&lt;tm1Tyj&gt;</td>\n",
       "      <td>Rb1&lt;tm1Tyj&gt;</td>\n",
       "      <td>involves: 129S2/SvPas</td>\n",
       "      <td>MP:0001092</td>\n",
       "      <td>16449662</td>\n",
       "      <td>MGI:97874</td>\n",
       "      <td>Rb1</td>\n",
       "      <td>RB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Rb1</th>\n",
       "      <td>Rb1&lt;tm1Tyj&gt;/Rb1&lt;tm1Tyj&gt;</td>\n",
       "      <td>Rb1&lt;tm1Tyj&gt;</td>\n",
       "      <td>involves: 129S2/SvPas</td>\n",
       "      <td>MP:0000961</td>\n",
       "      <td>16449662</td>\n",
       "      <td>MGI:97874</td>\n",
       "      <td>Rb1</td>\n",
       "      <td>RB1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           MGI_Allele_Accession_ID Allele symbol               involves  \\\n",
       "gene_name                                                                 \n",
       "Rb1        Rb1<tm1Tyj>/Rb1<tm1Tyj>   Rb1<tm1Tyj>  involves: 129S2/SvPas   \n",
       "Rb1        Rb1<tm1Tyj>/Rb1<tm1Tyj>   Rb1<tm1Tyj>  involves: 129S2/SvPas   \n",
       "Rb1        Rb1<tm1Tyj>/Rb1<tm1Tyj>   Rb1<tm1Tyj>  involves: 129S2/SvPas   \n",
       "Rb1        Rb1<tm1Tyj>/Rb1<tm1Tyj>   Rb1<tm1Tyj>  involves: 129S2/SvPas   \n",
       "Rb1        Rb1<tm1Tyj>/Rb1<tm1Tyj>   Rb1<tm1Tyj>  involves: 129S2/SvPas   \n",
       "\n",
       "                   MP      PMID MGI_marker_accession_ID gene_name  \\\n",
       "gene_name                                                           \n",
       "Rb1        MP:0000600  12529408               MGI:97874       Rb1   \n",
       "Rb1        MP:0001716  16449662               MGI:97874       Rb1   \n",
       "Rb1        MP:0001698  16449662               MGI:97874       Rb1   \n",
       "Rb1        MP:0001092  16449662               MGI:97874       Rb1   \n",
       "Rb1        MP:0000961  16449662               MGI:97874       Rb1   \n",
       "\n",
       "          human_ortholog  \n",
       "gene_name                 \n",
       "Rb1                  RB1  \n",
       "Rb1                  RB1  \n",
       "Rb1                  RB1  \n",
       "Rb1                  RB1  \n",
       "Rb1                  RB1  "
      ]
     },
     "execution_count": 101,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mgi_df = validation.load_MGI_mouseKO_data()\n",
    "mgi_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "id": "6c66f779",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15054\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:385: FutureWarning: The default dtype for empty Series will be 'object' instead of 'float64' in a future version. Specify a dtype explicitly to silence this warning.\n",
      "  pd.Series(id2namespace).to_csv(id2namespace_file, sep='\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MP</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>MP:0000001</th>\n",
       "      <td>mammalian phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MP:0000002</th>\n",
       "      <td>obsolete Morphology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MP:0000003</th>\n",
       "      <td>abnormal adipose tissue morphology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MP:0000011</th>\n",
       "      <td>abnormal adipose tissue morphology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MP:0000005</th>\n",
       "      <td>increased brown adipose tissue amount</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      description\n",
       "MP                                               \n",
       "MP:0000001                    mammalian phenotype\n",
       "MP:0000002                    obsolete Morphology\n",
       "MP:0000003     abnormal adipose tissue morphology\n",
       "MP:0000011     abnormal adipose tissue morphology\n",
       "MP:0000005  increased brown adipose tissue amount"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Parent</th>\n",
       "      <th>Child</th>\n",
       "      <th>Relation</th>\n",
       "      <th>Namespace</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MP:0005375</td>\n",
       "      <td>MP:0000003</td>\n",
       "      <td>is_a</td>\n",
       "      <td>MPheno.ontology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MP:0001778</td>\n",
       "      <td>MP:0000005</td>\n",
       "      <td>is_a</td>\n",
       "      <td>MPheno.ontology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MP:0001781</td>\n",
       "      <td>MP:0000008</td>\n",
       "      <td>is_a</td>\n",
       "      <td>MPheno.ontology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MP:0005334</td>\n",
       "      <td>MP:0000010</td>\n",
       "      <td>is_a</td>\n",
       "      <td>MPheno.ontology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MP:0000003</td>\n",
       "      <td>MP:0000013</td>\n",
       "      <td>is_a</td>\n",
       "      <td>MPheno.ontology</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Parent       Child Relation        Namespace\n",
       "0  MP:0005375  MP:0000003     is_a  MPheno.ontology\n",
       "1  MP:0001778  MP:0000005     is_a  MPheno.ontology\n",
       "2  MP:0001781  MP:0000008     is_a  MPheno.ontology\n",
       "3  MP:0005334  MP:0000010     is_a  MPheno.ontology\n",
       "4  MP:0000003  MP:0000013     is_a  MPheno.ontology"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "0 genes, 13919 terms, 0 gene-term relations, 17675 term-term relations\n",
       "node_attributes: ['description']\n",
       "edge_attributes: ['Relation', 'Namespace']"
      ]
     },
     "execution_count": 102,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "MPO = validation.load_MPO()\n",
    "MPO"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "92b3bf70",
   "metadata": {},
   "source": [
    "## 16. Identify phenotype(s) of interest."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "id": "c4e456dd",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_MP_description(term_id, ontology=MPO):\n",
    "    return ontology.node_attr.loc[term_id, \"description\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "id": "27dd2393",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of brain phenotypes 199\n",
      "Example brain phenotypes:\n",
      "MP:0000786 - abnormal embryonic neuroepithelial layer differentiation\n",
      "MP:0000929 - open neural tube\n",
      "MP:0000930 - wavy neural tube\n",
      "MP:0000936 - small embryonic telencephalon\n",
      "MP:0001672 - abnormal embryo development\n",
      "MP:0001688 - abnormal somite development\n",
      "MP:0001689 - incomplete somite formation\n",
      "MP:0001690 - failure of somite differentiation\n",
      "MP:0001691 - abnormal somite shape\n",
      "MP:0001697 - abnormal embryo size\n"
     ]
    }
   ],
   "source": [
    "# find terms related to brain\n",
    "# ---- modify this part as needed for your project -----\n",
    "MP_focal_brain_list = []\n",
    "for t in MPO.node_attr.index.tolist():\n",
    "    descr_temp = MPO.node_attr.loc[t]['description']\n",
    "    if descr_temp.find('neural tube')>-1:\n",
    "        MP_focal_brain_list.append(t)\n",
    "    #elif descr_temp.find('neural')>-1:\n",
    "    #    MP_focal_brain_list.append(t)\n",
    "    elif descr_temp.find('abnormal neural fold formation')>-1:\n",
    "        MP_focal_brain_list.append(t)\n",
    "    elif descr_temp.find('neural fold')>-1:\n",
    "        MP_focal_brain_list.append(t)\n",
    "    elif descr_temp.find('abnormal neural tube closure')>-1:\n",
    "        MP_focal_brain_list.append(t)\n",
    "    elif descr_temp.find('neural tube detects')>-1:\n",
    "        MP_focal_brain_list.append(t)\n",
    "    elif descr_temp.find('neuroepithelium')>-1:\n",
    "        MP_focal_brain_list.append(t)      \n",
    "    elif descr_temp.find('anencephaly')>-1:\n",
    "        MP_focal_brain_list.append(t) \n",
    "    elif descr_temp.find('neural tube')>-1:\n",
    "        MP_focal_brain_list.append(t) \n",
    "    elif descr_temp.find('somite')>-1:\n",
    "        MP_focal_brain_list.append(t) \n",
    "    elif descr_temp.find('embryo')>-1:\n",
    "        MP_focal_brain_list.append(t) \n",
    "    elif descr_temp.find('neural')>-1:\n",
    "        MP_focal_brain_list.append(t) \n",
    "    elif descr_temp.find('neural tube')>-1:\n",
    "        MP_focal_brain_list.append(t) \n",
    "print(\"Number of brain phenotypes\", len(MP_focal_brain_list))\n",
    "print(\"Example brain phenotypes:\")\n",
    "print(\"\\n\".join([mp+\" - \"+get_MP_description(mp) for mp in MP_focal_brain_list[0:10]]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "id": "c5bc6012",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of heart phenotypes: 324\n",
      "Example heart phenotypes:\n",
      "MP:0000266 - abnormal heart morphology\n",
      "MP:0000267 - abnormal heart development\n",
      "MP:0000269 - abnormal heart looping\n",
      "MP:0000270 - abnormal heart tube morphology\n",
      "MP:0000274 - enlarged heart\n",
      "MP:0000275 - heart hyperplasia\n",
      "MP:0000276 - heart right ventricle hypertrophy\n",
      "MP:0000277 - abnormal heart shape\n",
      "MP:0000278 - abnormal myocardial fiber morphology\n",
      "MP:0000285 - abnormal heart valve morphology\n"
     ]
    }
   ],
   "source": [
    "# find terms related to heart\n",
    "MP_focal_heart_list = []\n",
    "for t in MPO.node_attr.index.tolist():\n",
    "    descr_temp = MPO.node_attr.loc[t]['description']\n",
    "    if descr_temp.find('cardi')>-1:\n",
    "        MP_focal_heart_list.append(t)\n",
    "    elif descr_temp.find('heart')>-1:\n",
    "        MP_focal_heart_list.append(t)\n",
    "        \n",
    "print(\"Number of heart phenotypes:\", len(MP_focal_heart_list))\n",
    "print(\"Example heart phenotypes:\")\n",
    "print(\"\\n\".join([mp+\" - \"+get_MP_description(mp) for mp in MP_focal_heart_list[0:10]]))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "de820363",
   "metadata": {},
   "source": [
    "## 17. Compute the enrichment of selected phenotype(s) in NetColoc subnetwork as a whole."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "id": "b8d04e05",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal innate immunity\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 606\n",
      "number overlapping genes = 16\n",
      "0.5777338694120189\n",
      "(-0.3622586627287402, 0.6496861522111229)\n",
      "0.14371374474119136\n",
      "\n",
      "abnormal embryonic neuroepithelial layer differentiation\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 19\n",
      "number overlapping genes = 0\n",
      "0.9401226419991326\n",
      "(-2.701974643515019, 2.91733496611851)\n",
      "0.10768016130174551\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "open neural tube\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 242\n",
      "number overlapping genes = 15\n",
      "0.00010421778826489033\n",
      "(0.5207379334766425, 1.5835637630402917)\n",
      "1.0521508482584672\n",
      "\n",
      "wavy neural tube\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 38\n",
      "number overlapping genes = 3\n",
      "0.03195700894006393\n",
      "(0.11164811939269659, 2.477542850578552)\n",
      "1.2945954849856243\n",
      "\n",
      "small embryonic telencephalon\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 18\n",
      "number overlapping genes = 1\n",
      "0.37469493120659214\n",
      "(-1.1045676969306806, 2.933505033105395)\n",
      "0.9144686680873573\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal embryo development\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 1219\n",
      "number overlapping genes = 43\n",
      "0.0036021186519489753\n",
      "(0.15547281846896766, 0.7962558144414525)\n",
      "0.4758643164552101\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal somite development\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 221\n",
      "number overlapping genes = 7\n",
      "0.3929403429358297\n",
      "(-0.428118679930223, 1.0896933752984854)\n",
      "0.3307873476841312\n",
      "\n",
      "incomplete somite formation\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 47\n",
      "number overlapping genes = 2\n",
      "0.3806713460776284\n",
      "(-0.7846229284599917, 2.0545296734530853)\n",
      "0.6349533724965468\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "failure of somite differentiation\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 49\n",
      "number overlapping genes = 1\n",
      "0.901488999510235\n",
      "(-2.1077746754017124, 1.857358574800644)\n",
      "-0.12520805030053417\n",
      "\n",
      "abnormal somite shape\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 52\n",
      "number overlapping genes = 0\n",
      "0.5263171652800704\n",
      "(-3.687650844050985, 1.88580505461087)\n",
      "-0.9009228947200576\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal embryo size\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 882\n",
      "number overlapping genes = 29\n",
      "0.04793586944246572\n",
      "(0.0035118363599124236, 0.7694667913568587)\n",
      "0.3864893138583856\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "decreased embryo size\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 627\n",
      "number overlapping genes = 25\n",
      "0.004921965072379773\n",
      "(0.17907439475726705, 1.0028409635639528)\n",
      "0.5909576791606099\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal embryo turning\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 241\n",
      "number overlapping genes = 9\n",
      "0.1412089021659327\n",
      "(-0.16773886290637985, 1.1778273385705098)\n",
      "0.505044237832065\n",
      "\n",
      "incomplete embryo turning\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 67\n",
      "number overlapping genes = 2\n",
      "0.7114823370113719\n",
      "(-1.1441570056956216, 1.6764320943185034)\n",
      "0.2661375443114409\n",
      "\n",
      "abnormal extraembryonic endoderm formation\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 16\n",
      "number overlapping genes = 1\n",
      "0.31460277435193573\n",
      "(-0.9867431941643781, 3.066224562572851)\n",
      "1.0397406842042365\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal embryo implantation\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 65\n",
      "number overlapping genes = 1\n",
      "0.6817392844068424\n",
      "(-2.3912773596510695, 1.563751342077057)\n",
      "-0.4137630087870061\n",
      "\n",
      "failure of embryo implantation\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 29\n",
      "number overlapping genes = 0\n",
      "0.8249207885576681\n",
      "(-3.11294175044143, 2.4814991654090806)\n",
      "-0.31572129251617476\n",
      "\n",
      "impaired embryo implantation\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 16\n",
      "number overlapping genes = 0\n",
      "0.8456728580855998\n",
      "(-2.5366992037844955, 3.096086686402577)\n",
      "0.2796937413090408\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "embryonic growth arrest\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 329\n",
      "number overlapping genes = 11\n",
      "0.20881940300917934\n",
      "(-0.2183979225636179, 0.9991367380564391)\n",
      "0.3903694077464106\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "anencephaly\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 22\n",
      "number overlapping genes = 3\n",
      "0.0022195555031533517\n",
      "(0.6849873713295977, 3.1277647777233177)\n",
      "1.9063760745264577\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal embryonic tissue morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 1085\n",
      "number overlapping genes = 40\n",
      "0.002023553861734718\n",
      "(0.1903996597435212, 0.8527709386877567)\n",
      "0.5215852992156389\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal extraembryonic tissue morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 802\n",
      "number overlapping genes = 26\n",
      "0.07272695392416627\n",
      "(-0.03400569657327446, 0.771803786421984)\n",
      "0.36889904492435477\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal embryonic growth/weight/body size\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 1198\n",
      "number overlapping genes = 36\n",
      "0.0969470705097574\n",
      "(-0.05302047143566546, 0.6394765410436951)\n",
      "0.2932280348040148\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal neural tube morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 607\n",
      "number overlapping genes = 23\n",
      "0.014451619907356763\n",
      "(0.10609392169647447, 0.9620382945506545)\n",
      "0.5340661081235645\n",
      "\n",
      "disorganized extraembryonic tissue\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 24\n",
      "number overlapping genes = 2\n",
      "0.06778708860872322\n",
      "(-0.09884307167424788, 2.802496360327512)\n",
      "1.351826644326632\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "delayed neural tube closure\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 33\n",
      "number overlapping genes = 3\n",
      "0.017066868648387384\n",
      "(0.2583645394720726, 2.639672645361788)\n",
      "1.4490185924169303\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal neural crest cell migration\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 44\n",
      "number overlapping genes = 1\n",
      "0.98823416348311\n",
      "(-1.999846679896033, 1.969977611132633)\n",
      "-0.014934534381700004\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal embryonic hematopoiesis\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 58\n",
      "number overlapping genes = 0\n",
      "0.47707173443946593\n",
      "(-3.795805954512891, 1.7749069523660146)\n",
      "-1.0104495010734382\n",
      "\n",
      "kinked neural tube\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 56\n",
      "number overlapping genes = 2\n",
      "0.5310053118993276\n",
      "(-0.9624152470922211, 1.8666972298533693)\n",
      "0.45214099138057406\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal neural tube closure\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 387\n",
      "number overlapping genes = 17\n",
      "0.006715023953070608\n",
      "(0.18988180004586824, 1.1813974950712585)\n",
      "0.6856396475585633\n",
      "\n",
      "delayed somite formation\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 15\n",
      "number overlapping genes = 0\n",
      "0.810832524130851\n",
      "(-2.4749464053043386, 3.1635198003970917)\n",
      "0.34428669754637653\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal embryonic epiblast morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 78\n",
      "number overlapping genes = 4\n",
      "0.10589846735567747\n",
      "(-0.1768971918572233, 1.8444090335782533)\n",
      "0.833755920860515\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal embryonic-extraembryonic boundary morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 31\n",
      "number overlapping genes = 1\n",
      "0.7340331390454671\n",
      "(-1.648863264765388, 2.340416596465544)\n",
      "0.34577666585007805\n",
      "\n",
      "embryonic growth retardation\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 696\n",
      "number overlapping genes = 23\n",
      "0.07548485663986741\n",
      "(-0.0397461841044916, 0.8141657521860609)\n",
      "0.38720978404078465\n",
      "\n",
      "disorganized embryonic tissue\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 29\n",
      "number overlapping genes = 0\n",
      "0.8249207885576681\n",
      "(-3.11294175044143, 2.4814991654090806)\n",
      "-0.31572129251617476\n",
      "\n",
      "absent embryonic cilia\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 19\n",
      "number overlapping genes = 0\n",
      "0.9401226419991326\n",
      "(-2.701974643515019, 2.91733496611851)\n",
      "0.10768016130174551\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "failure of initiation of embryo turning\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 55\n",
      "number overlapping genes = 2\n",
      "0.5142539595295207\n",
      "(-0.9441429171554885, 1.8859182618089638)\n",
      "0.47088767232673767\n",
      "\n",
      "delayed embryo turning\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 20\n",
      "number overlapping genes = 3\n",
      "0.0013167021253983476\n",
      "(0.7866230818211066, 3.248798107487314)\n",
      "2.0177105946542104\n",
      "\n",
      "abnormal direction of embryo turning\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 13\n",
      "number overlapping genes = 1\n",
      "0.2254453944459337\n",
      "(-0.7791721489886974, 3.305267195070543)\n",
      "1.2630475230409228\n",
      "\n",
      "abnormal embryonic erythropoiesis\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 27\n",
      "number overlapping genes = 0\n",
      "0.8642498026572712\n",
      "(-3.043127200229141, 2.5548204318201906)\n",
      "-0.2441533842044752\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal embryonic erythrocyte morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 34\n",
      "number overlapping genes = 0\n",
      "0.7389226833773714\n",
      "(-3.268794667514915, 2.3186778677001962)\n",
      "-0.47505839990735943\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal embryonic neuroepithelium morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 110\n",
      "number overlapping genes = 3\n",
      "0.7681282592046035\n",
      "(-0.9781248016119497, 1.3244918076500716)\n",
      "0.17318350301906094\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal extraembryonic tissue physiology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 317\n",
      "number overlapping genes = 10\n",
      "0.31221749665578624\n",
      "(-0.308617171155967, 0.9656375968697771)\n",
      "0.32851021285690507\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal embryonic/fetal subventricular zone morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 50\n",
      "number overlapping genes = 2\n",
      "0.43047904838277107\n",
      "(-0.8474445891242932, 1.9879471291598736)\n",
      "0.5702512700177902\n",
      "\n",
      "abnormal crista ampullaris neuroepithelium morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 11\n",
      "number overlapping genes = 1\n",
      "0.1685955456781253\n",
      "(-0.612357220595233, 3.5033130673400317)\n",
      "1.4454779233723993\n",
      "\n",
      "sensorineural hearing loss\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 39\n",
      "number overlapping genes = 1\n",
      "0.9144538590352755\n",
      "(-1.8789189881590127, 2.0968229241353864)\n",
      "0.10895196798818674\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal neural fold formation\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 28\n",
      "number overlapping genes = 2\n",
      "0.10723723872720109\n",
      "(-0.2568263283399237, 2.6259357652775934)\n",
      "1.184554718468835\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal somite size\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 40\n",
      "number overlapping genes = 0\n",
      "0.6541440853169642\n",
      "(-3.4286058095999263, 2.1527971641905563)\n",
      "-0.637904322704685\n",
      "\n",
      "abnormal dorsal-ventral polarity of the somites\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 12\n",
      "number overlapping genes = 0\n",
      "0.6943062389475754\n",
      "(-2.2629717018239894, 3.3981587745901116)\n",
      "0.5675935363830611\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal neural plate morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 29\n",
      "number overlapping genes = 1\n",
      "0.6838624058265801\n",
      "(-1.5820528483568495, 2.411809823559534)\n",
      "0.41487848760134227\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "embryonic lethality before implantation\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 183\n",
      "number overlapping genes = 2\n",
      "0.283502912530474\n",
      "(-2.1611260177745546, 0.6324782610936397)\n",
      "-0.7643238783404573\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "embryonic lethality between implantation and somite formation\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 301\n",
      "number overlapping genes = 7\n",
      "0.98184962325707\n",
      "(-0.7469101387847025, 0.7644531369561846)\n",
      "0.008771499085741041\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "embryonic lethality between somite formation and embryo turning\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 109\n",
      "number overlapping genes = 4\n",
      "0.3461095147318306\n",
      "(-0.5208759151255469, 1.4852086158892381)\n",
      "0.48216635038184563\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "embryonic lethality during organogenesis\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 891\n",
      "number overlapping genes = 30\n",
      "0.03181939713535046\n",
      "(0.03592716033132698, 0.7902162620208051)\n",
      "0.413071711176066\n",
      "\n",
      "abnormal extraembryonic mesoderm development\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 12\n",
      "number overlapping genes = 1\n",
      "0.19662155695955585\n",
      "(-0.6992190407373404, 3.399445687297693)\n",
      "1.3501133232801763\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "sensorineural hearing impairment\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 40\n",
      "number overlapping genes = 1\n",
      "0.9348179265232167\n",
      "(-1.904296900545638, 2.070140857038808)\n",
      "0.08292197824658487\n",
      "\n",
      "absent somites\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 39\n",
      "number overlapping genes = 1\n",
      "0.9144538590352755\n",
      "(-1.8789189881590127, 2.0968229241353864)\n",
      "0.10895196798818674\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "embryonic lethality at implantation\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 33\n",
      "number overlapping genes = 3\n",
      "0.017066868648387384\n",
      "(0.2583645394720726, 2.639672645361788)\n",
      "1.4490185924169303\n",
      "\n",
      "abnormal rostral-caudal patterning of the somites\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 11\n",
      "number overlapping genes = 0\n",
      "0.650920527384572\n",
      "(-2.1810419024243837, 3.4903605756690164)\n",
      "0.6546593366223163\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal embryonic tissue physiology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 287\n",
      "number overlapping genes = 11\n",
      "0.08614436905142463\n",
      "(-0.07593310935453268, 1.1446166671793376)\n",
      "0.5343417789124025\n",
      "\n",
      "decreased embryo weight\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 22\n",
      "number overlapping genes = 1\n",
      "0.4927090029037302\n",
      "(-1.305407363719734, 2.7112909487754457)\n",
      "0.7029417925278558\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "impaired somite development\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 104\n",
      "number overlapping genes = 4\n",
      "0.29969330496167623\n",
      "(-0.47272369655270485, 1.5351836668898315)\n",
      "0.5312299851685633\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal preimplantation embryo development\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 100\n",
      "number overlapping genes = 3\n",
      "0.6439778582957942\n",
      "(-0.8810662307990249, 1.4247628013301359)\n",
      "0.2718482852655555\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "decreased neural crest cell number\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 15\n",
      "number overlapping genes = 0\n",
      "0.810832524130851\n",
      "(-2.4749464053043386, 3.1635198003970917)\n",
      "0.34428669754637653\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal neural crest cell apoptosis\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 13\n",
      "number overlapping genes = 0\n",
      "0.7353056736663703\n",
      "(-2.338715699078265, 3.3137085040739587)\n",
      "0.48749640249784676\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal neural crest cell morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 44\n",
      "number overlapping genes = 0\n",
      "0.6062784730063095\n",
      "(-3.5225693856547537, 2.0557042707353106)\n",
      "-0.7334325574597216\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal neural crest morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 46\n",
      "number overlapping genes = 0\n",
      "0.5844887704740971\n",
      "(-3.5664496338336957, 2.0104629031557164)\n",
      "-0.7779933653389897\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "embryonic lethality between implantation and placentation\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 464\n",
      "number overlapping genes = 10\n",
      "0.8264840627024309\n",
      "(-0.7048447087233569, 0.5630377987154986)\n",
      "-0.07090345500392914\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal extraembryonic ectoderm morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 22\n",
      "number overlapping genes = 3\n",
      "0.0022195555031533517\n",
      "(0.6849873713295977, 3.1277647777233177)\n",
      "1.9063760745264577\n",
      "\n",
      "embryonic lethality, complete penetrance\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 471\n",
      "number overlapping genes = 18\n",
      "0.028300540980726156\n",
      "(0.05721104890446366, 1.0191734352961208)\n",
      "0.5381922421002923\n",
      "\n",
      "embryonic lethality at implantation, complete penetrance\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 26\n",
      "number overlapping genes = 1\n",
      "0.6047917153710545\n",
      "(-1.472678679036409, 2.529419831139403)\n",
      "0.528370576051497\n",
      "\n",
      "embryonic lethality before implantation, complete penetrance\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 154\n",
      "number overlapping genes = 1\n",
      "0.19897273787521463\n",
      "(-3.2588539911742647, 0.6784795284628247)\n",
      "-1.2901872313557199\n",
      "\n",
      "embryonic lethality between implantation and placentation, complete penetrance\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 66\n",
      "number overlapping genes = 1\n",
      "0.6704271740674494\n",
      "(-2.4066026340834723, 1.5479590820697051)\n",
      "-0.4293217760068835\n",
      "\n",
      "embryonic lethality between implantation and somite formation, complete penetrance\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 276\n",
      "number overlapping genes = 7\n",
      "0.7975251289220582\n",
      "(-0.657463724721029, 0.8555052431227487)\n",
      "0.09902075920085984\n",
      "\n",
      "embryonic lethality between somite formation and embryo turning, complete penetrance\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 86\n",
      "number overlapping genes = 3\n",
      "0.46746945795467365\n",
      "(-0.7273239573242087, 1.5842916584996616)\n",
      "0.4284838505877264\n",
      "\n",
      "embryonic lethality during organogenesis, complete penetrance\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 714\n",
      "number overlapping genes = 25\n",
      "0.03143332837862604\n",
      "(0.04016452580390828, 0.8619750951619394)\n",
      "0.4510698104829238\n",
      "\n",
      "embryonic lethality, incomplete penetrance\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 69\n",
      "number overlapping genes = 4\n",
      "0.06249020682065847\n",
      "(-0.05027772143183551, 1.9781279302876156)\n",
      "0.96392510442789\n",
      "\n",
      "embryonic lethality before implantation, incomplete penetrance\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 24\n",
      "number overlapping genes = 1\n",
      "0.5496390725284372\n",
      "(-1.3925240107009893, 2.616246221876069)\n",
      "0.6118611055875398\n",
      "\n",
      "embryonic lethality between implantation and placentation, incomplete penetrance\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 16\n",
      "number overlapping genes = 1\n",
      "0.31460277435193573\n",
      "(-0.9867431941643781, 3.066224562572851)\n",
      "1.0397406842042365\n",
      "\n",
      "embryonic lethality between implantation and somite formation, incomplete penetrance\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 24\n",
      "number overlapping genes = 0\n",
      "0.9296577160457551\n",
      "(-2.9283559368148526, 2.6759420282055504)\n",
      "-0.12620695430465112\n",
      "\n",
      "embryonic lethality between somite formation and embryo turning, incomplete penetrance\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 23\n",
      "number overlapping genes = 1\n",
      "0.5213827610946205\n",
      "(-1.3499106973183825, 2.662645345334998)\n",
      "0.6563673240083077\n",
      "\n",
      "embryonic lethality during organogenesis, incomplete penetrance\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 279\n",
      "number overlapping genes = 10\n",
      "0.15548098505071298\n",
      "(-0.17576644716632006, 1.101258969438837)\n",
      "0.4627462611362585\n",
      "\n",
      "absent embryonic epiblast\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 13\n",
      "number overlapping genes = 1\n",
      "0.2254453944459337\n",
      "(-0.7791721489886974, 3.305267195070543)\n",
      "1.2630475230409228\n",
      "\n",
      "small embryonic epiblast\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 12\n",
      "number overlapping genes = 0\n",
      "0.6943062389475754\n",
      "(-2.2629717018239894, 3.3981587745901116)\n",
      "0.5675935363830611\n",
      "\n",
      "increased embryonic epiblast cell apoptosis\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 11\n",
      "number overlapping genes = 0\n",
      "0.650920527384572\n",
      "(-2.1810419024243837, 3.4903605756690164)\n",
      "0.6546593366223163\n",
      "\n",
      "decreased embryonic epiblast cell proliferation\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 11\n",
      "number overlapping genes = 0\n",
      "0.650920527384572\n",
      "(-2.1810419024243837, 3.4903605756690164)\n",
      "0.6546593366223163\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal extraembryonic coelom morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 16\n",
      "number overlapping genes = 0\n",
      "0.8456728580855998\n",
      "(-2.5366992037844955, 3.096086686402577)\n",
      "0.2796937413090408\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal neural fold morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 49\n",
      "number overlapping genes = 2\n",
      "0.4138096950803526\n",
      "(-0.8269370615469838, 2.0096554801522064)\n",
      "0.5913592093026114\n",
      "\n",
      "abnormal cephalic neural fold morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 12\n",
      "number overlapping genes = 0\n",
      "0.6943062389475754\n",
      "(-2.2629717018239894, 3.3981587745901116)\n",
      "0.5675935363830611\n",
      "\n",
      "decreased somite size\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 30\n",
      "number overlapping genes = 0\n",
      "0.806392931317837\n",
      "(-3.1461084091285785, 2.4467537675130147)\n",
      "-0.3496773208077819\n",
      "\n",
      "fused somites\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 12\n",
      "number overlapping genes = 0\n",
      "0.6943062389475754\n",
      "(-2.2629717018239894, 3.3981587745901116)\n",
      "0.5675935363830611\n",
      "\n",
      "decreased embryonic cilium length\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 13\n",
      "number overlapping genes = 0\n",
      "0.7353056736663703\n",
      "(-2.338715699078265, 3.3137085040739587)\n",
      "0.48749640249784676\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal bilaminar embryonic disk morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 85\n",
      "number overlapping genes = 4\n",
      "0.14872280663753024\n",
      "(-0.26544173677326244, 1.7514207834016244)\n",
      "0.742989523314181\n",
      "\n",
      "embryonic-extraembryonic boundary constriction\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 15\n",
      "number overlapping genes = 0\n",
      "0.810832524130851\n",
      "(-2.4749464053043386, 3.1635198003970917)\n",
      "0.34428669754637653\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "increased neural tube apoptosis\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 34\n",
      "number overlapping genes = 3\n",
      "0.019564230837919767\n",
      "(0.22725609084206044, 2.605092489178241)\n",
      "1.4161742900101508\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "increased embryonic neuroepithelium apoptosis\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 20\n",
      "number overlapping genes = 2\n",
      "0.03763959283760957\n",
      "(0.088732485513904, 3.016697781564024)\n",
      "1.552715133538964\n",
      "\n",
      "decreased embryonic neuroepithelium thickness\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 23\n",
      "number overlapping genes = 1\n",
      "0.5213827610946205\n",
      "(-1.3499106973183825, 2.662645345334998)\n",
      "0.6563673240083077\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal embryonic neuroepithelial cell proliferation\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 15\n",
      "number overlapping genes = 0\n",
      "0.810832524130851\n",
      "(-2.4749464053043386, 3.1635198003970917)\n",
      "0.34428669754637653\n",
      "\n",
      "decreased embryonic neuroepithelial cell proliferation\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 12\n",
      "number overlapping genes = 0\n",
      "0.6943062389475754\n",
      "(-2.2629717018239894, 3.3981587745901116)\n",
      "0.5675935363830611\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal neural crest cell physiology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 56\n",
      "number overlapping genes = 1\n",
      "0.7955780657546137\n",
      "(-2.241718184343156, 1.7182741473119658)\n",
      "-0.26172201851559507\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal cranial neural crest cell morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 11\n",
      "number overlapping genes = 0\n",
      "0.650920527384572\n",
      "(-2.1810419024243837, 3.4903605756690164)\n",
      "0.6546593366223163\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "embryo tissue necrosis\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 32\n",
      "number overlapping genes = 1\n",
      "0.7583472123710618\n",
      "(-1.6806718759494526, 2.30653660283605)\n",
      "0.31293236344329856\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "embryonic lethality prior to organogenesis\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 1049\n",
      "number overlapping genes = 25\n",
      "0.8639533068651056\n",
      "(-0.3729568558040807, 0.44441296015947923)\n",
      "0.035728052177699254\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "embryonic lethality prior to tooth bud stage\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 1210\n",
      "number overlapping genes = 28\n",
      "0.9847231571723669\n",
      "(-0.38368069455320136, 0.39125138757530803)\n",
      "0.0037853465110533335\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal embryonic tissue cell apoptosis\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 166\n",
      "number overlapping genes = 7\n",
      "0.10510065698387978\n",
      "(-0.13210442287690904, 1.3938668705396915)\n",
      "0.6308812238313912\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "increased embryonic tissue cell apoptosis\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 139\n",
      "number overlapping genes = 5\n",
      "0.31302060595233805\n",
      "(-0.43567310733343534, 1.3600153785383635)\n",
      "0.46217113560246403\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "embryo tumor\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 15\n",
      "number overlapping genes = 0\n",
      "0.810832524130851\n",
      "(-2.4749464053043386, 3.1635198003970917)\n",
      "0.34428669754637653\n",
      "\n",
      "embryo cyst\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 11\n",
      "number overlapping genes = 0\n",
      "0.650920527384572\n",
      "(-2.1810419024243837, 3.4903605756690164)\n",
      "0.6546593366223163\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal embryo morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 1597\n",
      "number overlapping genes = 49\n",
      "0.034665257536861394\n",
      "(0.02344910834777103, 0.6270537427937857)\n",
      "0.32525142557077835\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal preimplantation embryo morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 216\n",
      "number overlapping genes = 3\n",
      "0.37178107380254777\n",
      "(-1.664619026978641, 0.6224276028510396)\n",
      "-0.5210957120638007\n",
      "\n",
      "embryonic lethality prior to organogenesis, complete penetrance\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 14\n",
      "number overlapping genes = 0\n",
      "0.7740935236234859\n",
      "(-2.409141497019146, 3.2358094993592985)\n",
      "0.41333400117007635\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    }
   ],
   "source": [
    "# add a negative control phenotype: abnormal innate immunity: MP:0002419\n",
    "# negative controls are tough here because we're dealing with development, which impacts almost everything.\n",
    "MP_focal_list = ['MP:0002419']+MP_focal_brain_list\n",
    "root_KO_brain_df=validation.MPO_enrichment_root(hier_df,MPO,mgi_df,MP_focal_list,G_int,verbose=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "id": "08d48abd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>OR_p</th>\n",
       "      <th>log_OR</th>\n",
       "      <th>log_OR_CI_lower</th>\n",
       "      <th>log_OR_CI_upper</th>\n",
       "      <th>num_genes_in_term</th>\n",
       "      <th>MP_description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>MP:0002419</th>\n",
       "      <td>0.577734</td>\n",
       "      <td>0.143714</td>\n",
       "      <td>-0.362259</td>\n",
       "      <td>0.649686</td>\n",
       "      <td>606</td>\n",
       "      <td>abnormal innate immunity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MP:0000786</th>\n",
       "      <td>0.940123</td>\n",
       "      <td>0.107680</td>\n",
       "      <td>-2.701975</td>\n",
       "      <td>2.917335</td>\n",
       "      <td>19</td>\n",
       "      <td>abnormal embryonic neuroepithelial layer diffe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MP:0000929</th>\n",
       "      <td>0.000104</td>\n",
       "      <td>1.052151</td>\n",
       "      <td>0.520738</td>\n",
       "      <td>1.583564</td>\n",
       "      <td>242</td>\n",
       "      <td>open neural tube</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MP:0000930</th>\n",
       "      <td>0.031957</td>\n",
       "      <td>1.294595</td>\n",
       "      <td>0.111648</td>\n",
       "      <td>2.477543</td>\n",
       "      <td>38</td>\n",
       "      <td>wavy neural tube</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MP:0000936</th>\n",
       "      <td>0.374695</td>\n",
       "      <td>0.914469</td>\n",
       "      <td>-1.104568</td>\n",
       "      <td>2.933505</td>\n",
       "      <td>18</td>\n",
       "      <td>small embryonic telencephalon</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                OR_p    log_OR  log_OR_CI_lower  log_OR_CI_upper  \\\n",
       "MP:0002419  0.577734  0.143714        -0.362259         0.649686   \n",
       "MP:0000786  0.940123  0.107680        -2.701975         2.917335   \n",
       "MP:0000929  0.000104  1.052151         0.520738         1.583564   \n",
       "MP:0000930  0.031957  1.294595         0.111648         2.477543   \n",
       "MP:0000936  0.374695  0.914469        -1.104568         2.933505   \n",
       "\n",
       "            num_genes_in_term  \\\n",
       "MP:0002419                606   \n",
       "MP:0000786                 19   \n",
       "MP:0000929                242   \n",
       "MP:0000930                 38   \n",
       "MP:0000936                 18   \n",
       "\n",
       "                                               MP_description  \n",
       "MP:0002419                           abnormal innate immunity  \n",
       "MP:0000786  abnormal embryonic neuroepithelial layer diffe...  \n",
       "MP:0000929                                   open neural tube  \n",
       "MP:0000930                                   wavy neural tube  \n",
       "MP:0000936                      small embryonic telencephalon  "
      ]
     },
     "execution_count": 107,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "root_KO_brain_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "id": "f207dcf9",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal heart development\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 392\n",
      "number overlapping genes = 12\n",
      "0.3158710399628364\n",
      "(-0.28451621527333604, 0.8808970106732916)\n",
      "0.2981903976999778\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal heart looping\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 138\n",
      "number overlapping genes = 4\n",
      "0.6424001042744849\n",
      "(-0.762388490056235, 1.2357870791135637)\n",
      "0.23669929452866434\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal heart tube morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 84\n",
      "number overlapping genes = 1\n",
      "0.5028911916316607\n",
      "(-2.6487953268540547, 1.2992791748958032)\n",
      "-0.6747580759791258\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "enlarged heart\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 993\n",
      "number overlapping genes = 29\n",
      "0.186265158343777\n",
      "(-0.12449517924614267, 0.6401082245425973)\n",
      "0.25780652264822734\n",
      "\n",
      "heart right ventricle hypertrophy\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 42\n",
      "number overlapping genes = 3\n",
      "0.04846753425567264\n",
      "(0.007984167584619728, 2.364343642042255)\n",
      "1.1861639048134371\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal heart shape\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 46\n",
      "number overlapping genes = 2\n",
      "0.36423434117145004\n",
      "(-0.7627787315418448, 2.0777402305747703)\n",
      "0.6574807495164627\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal myocardial fiber morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 214\n",
      "number overlapping genes = 9\n",
      "0.06700031257273598\n",
      "(-0.0441438580730481, 1.3046594600583807)\n",
      "0.6302578009926663\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal heart valve morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 185\n",
      "number overlapping genes = 10\n",
      "0.006324849587189133\n",
      "(0.25335620586667496, 1.5423372940021114)\n",
      "0.8978467499343932\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal pericardium morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 232\n",
      "number overlapping genes = 10\n",
      "0.04438757283186128\n",
      "(0.016495744840830384, 1.2982465513744699)\n",
      "0.6573711481076501\n",
      "\n",
      "enlarged pericardium\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 44\n",
      "number overlapping genes = 3\n",
      "0.058338048485040955\n",
      "(-0.040094421746090125, 2.3121833590449654)\n",
      "1.1360444686494375\n",
      "\n",
      "distended pericardium\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 57\n",
      "number overlapping genes = 2\n",
      "0.5477288346531834\n",
      "(-0.9803616843914391, 1.8478362879968977)\n",
      "0.4337373018027293\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "absent myocardial trabeculae\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 21\n",
      "number overlapping genes = 1\n",
      "0.4636536249636122\n",
      "(-1.2588385341754846, 2.762411347409461)\n",
      "0.7517864066169881\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal cardiac stroke volume\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 47\n",
      "number overlapping genes = 1\n",
      "0.9349947986903643\n",
      "(-2.0659833487458386, 1.9009045913835794)\n",
      "-0.0825393786811297\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal response to cardiac infarction\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 88\n",
      "number overlapping genes = 2\n",
      "0.9833416707827917\n",
      "(-1.4201382936503057, 1.3901995634339868)\n",
      "-0.014969365108159494\n",
      "\n",
      "dextrocardia\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 40\n",
      "number overlapping genes = 2\n",
      "0.2685893003092279\n",
      "(-0.6206929409894073, 2.2295155176024295)\n",
      "0.8044112883065111\n",
      "\n",
      "mesocardia\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 15\n",
      "number overlapping genes = 2\n",
      "0.013587833555349371\n",
      "(0.3866565160008535, 3.370162902947212)\n",
      "1.8784097094740329\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal cardiovascular system physiology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 1951\n",
      "number overlapping genes = 53\n",
      "0.2026248236222431\n",
      "(-0.10191381189886939, 0.4805631755505553)\n",
      "0.18932468182584294\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "cardiac hypertrophy\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 206\n",
      "number overlapping genes = 9\n",
      "0.05153371400057777\n",
      "(-0.004461559722879538, 1.3454695085501909)\n",
      "0.6705039744136556\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal cardiac output\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 53\n",
      "number overlapping genes = 0\n",
      "0.517546952261781\n",
      "(-3.7065034870696394, 1.866452217770746)\n",
      "-0.9200256346494466\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal heart rate\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 405\n",
      "number overlapping genes = 10\n",
      "0.8250945658977016\n",
      "(-0.5633439521523371, 0.7065295215475089)\n",
      "0.0715927846975859\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "irregular heartbeat\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 73\n",
      "number overlapping genes = 2\n",
      "0.8048958280852898\n",
      "(-1.2310054079825297, 1.5860401889573947)\n",
      "0.17751739048743254\n",
      "\n",
      "pericardial edema\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 86\n",
      "number overlapping genes = 5\n",
      "0.03662284672974005\n",
      "(0.06023149833648078, 1.8766912304394805)\n",
      "0.9684613643879807\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "heart inflammation\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 44\n",
      "number overlapping genes = 1\n",
      "0.98823416348311\n",
      "(-1.999846679896033, 1.969977611132633)\n",
      "-0.014934534381700004\n",
      "\n",
      "myocarditis\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 15\n",
      "number overlapping genes = 0\n",
      "0.810832524130851\n",
      "(-2.4749464053043386, 3.1635198003970917)\n",
      "0.34428669754637653\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "small heart\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 335\n",
      "number overlapping genes = 11\n",
      "0.231724077293763\n",
      "(-0.2372384839518531, 0.979928765492869)\n",
      "0.37134514077050795\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal myocardial trabeculae morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 153\n",
      "number overlapping genes = 8\n",
      "0.018981439571397165\n",
      "(0.14143146862923828, 1.5779596107243714)\n",
      "0.8596955396768049\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "disorganized myocardium\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 47\n",
      "number overlapping genes = 3\n",
      "0.07518896954509728\n",
      "(-0.1081566408520882, 2.2386833523820693)\n",
      "1.0652633557649906\n",
      "\n",
      "heart left ventricle hypertrophy\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 67\n",
      "number overlapping genes = 6\n",
      "0.000841679426855348\n",
      "(0.5937443210273007, 2.281823551428297)\n",
      "1.4377839362277989\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "increased heart rate\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 112\n",
      "number overlapping genes = 2\n",
      "0.7136718432642984\n",
      "(-1.6641116782932435, 1.1392821367064037)\n",
      "-0.2624147707934199\n",
      "\n",
      "thin myocardium\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 97\n",
      "number overlapping genes = 3\n",
      "0.6061437040559445\n",
      "(-0.8500340088726621, 1.45689101101727)\n",
      "0.3034285010723039\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "heart hypoplasia\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 112\n",
      "number overlapping genes = 5\n",
      "0.1344914811985455\n",
      "(-0.2132032476247625, 1.5905238435463227)\n",
      "0.6886602979607801\n",
      "\n",
      "dilated heart left ventricle\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 115\n",
      "number overlapping genes = 6\n",
      "0.04292047567813384\n",
      "(0.027238361951443135, 1.6821358563916715)\n",
      "0.8546871091715573\n",
      "\n",
      "dilated heart right ventricle\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 51\n",
      "number overlapping genes = 2\n",
      "0.44719510253683314\n",
      "(-0.8675422654230129, 1.9666971649317564)\n",
      "0.5495774497543717\n",
      "\n",
      "dilated cardiomyopathy\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 87\n",
      "number overlapping genes = 2\n",
      "0.9964183887372254\n",
      "(-1.4085745848640348, 1.4021372208056353)\n",
      "-0.0032186820291997265\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "increased heart weight\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 398\n",
      "number overlapping genes = 14\n",
      "0.10646655358256603\n",
      "(-0.09559204693048229, 0.9883884620647778)\n",
      "0.44639820756714776\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "decreased heart weight\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 105\n",
      "number overlapping genes = 3\n",
      "0.7065385513687841\n",
      "(-0.9307592957081838, 1.3733853451716365)\n",
      "0.22131302473172632\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "dystrophic cardiac calcinosis\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 15\n",
      "number overlapping genes = 1\n",
      "0.2846142546247056\n",
      "(-0.9222042718710344, 3.139780247526907)\n",
      "1.1087879878279363\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal cardiovascular development\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 806\n",
      "number overlapping genes = 22\n",
      "0.4135867971399072\n",
      "(-0.25342522589290883, 0.6161810150771858)\n",
      "0.1813778945921385\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal cardiac muscle contractility\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 321\n",
      "number overlapping genes = 8\n",
      "0.8227064572919257\n",
      "(-0.6273241726172583, 0.7892709640291582)\n",
      "0.08097339570594997\n",
      "\n",
      "increased myocardial infarct size\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 30\n",
      "number overlapping genes = 1\n",
      "0.7092047653494131\n",
      "(-1.6160137605248914, 2.375479148808262)\n",
      "0.3797326941416852\n",
      "\n",
      "decreased myocardial infarct size\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 26\n",
      "number overlapping genes = 0\n",
      "0.885144444349316\n",
      "(-3.0063097621811607, 2.5935925851664585)\n",
      "-0.20635858850735112\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal epicardium morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 17\n",
      "number overlapping genes = 2\n",
      "0.021432662587959284\n",
      "(0.25671157529782196, 3.2136884337472997)\n",
      "1.7352000045225608\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal heart atrium morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 250\n",
      "number overlapping genes = 13\n",
      "0.0029106120617515613\n",
      "(0.2940216294318302, 1.4272326193690201)\n",
      "0.8606271244004251\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "dilated heart atrium\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 61\n",
      "number overlapping genes = 3\n",
      "0.18457456827647067\n",
      "(-0.37615936539527617, 1.9526521896978242)\n",
      "0.788246412151274\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "cardiac fibrosis\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 179\n",
      "number overlapping genes = 6\n",
      "0.3519151265866314\n",
      "(-0.43029744912069523, 1.2087686150147383)\n",
      "0.38923558294702154\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal cardiomyocyte apoptosis\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 61\n",
      "number overlapping genes = 4\n",
      "0.034947344957591914\n",
      "(0.07741385332378936, 2.1139814838153734)\n",
      "1.0956976685695814\n",
      "\n",
      "increased cardiomyocyte apoptosis\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 53\n",
      "number overlapping genes = 4\n",
      "0.016928589096825084\n",
      "(0.2236927276697911, 2.2710372920718473)\n",
      "1.247365009870819\n",
      "\n",
      "decreased cardiac output\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 35\n",
      "number overlapping genes = 0\n",
      "0.7235426492801781\n",
      "(-3.297258896299264, 2.289058039821088)\n",
      "-0.504100428239088\n",
      "\n",
      "increased cardiac output\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 13\n",
      "number overlapping genes = 0\n",
      "0.7353056736663703\n",
      "(-2.338715699078265, 3.3137085040739587)\n",
      "0.48749640249784676\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal fetal cardiomyocyte proliferation\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 61\n",
      "number overlapping genes = 1\n",
      "0.7295441794731865\n",
      "(-2.3275320301928892, 1.6295196407940273)\n",
      "-0.34900619469943095\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "decreased heart right ventricle size\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 29\n",
      "number overlapping genes = 2\n",
      "0.11842913402213588\n",
      "(-0.29272164012996393, 2.5862414856733857)\n",
      "1.146759922771711\n",
      "\n",
      "increased heart left ventricle weight\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 31\n",
      "number overlapping genes = 1\n",
      "0.7340331390454671\n",
      "(-1.648863264765388, 2.340416596465544)\n",
      "0.34577666585007805\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal heart right ventricle morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 166\n",
      "number overlapping genes = 8\n",
      "0.03450446704974214\n",
      "(0.056376145959530666, 1.4898664835750124)\n",
      "0.7731213147672715\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal heart left ventricle morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 366\n",
      "number overlapping genes = 15\n",
      "0.023086518087910388\n",
      "(0.0837034461625048, 1.135221855941187)\n",
      "0.6094626510518459\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal heart right atrium morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 45\n",
      "number overlapping genes = 2\n",
      "0.34790803436623285\n",
      "(-0.7404492994969725, 2.101498871370221)\n",
      "0.6805247859366244\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal heart left atrium morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 24\n",
      "number overlapping genes = 3\n",
      "0.0035310553712639666\n",
      "(0.592703206524049, 3.0196642316653675)\n",
      "1.806183719094708\n",
      "\n",
      "increased heart rate variability\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 36\n",
      "number overlapping genes = 1\n",
      "0.8505155139772607\n",
      "(-1.7986956492077384, 2.1814027660452826)\n",
      "0.191353558418772\n",
      "\n",
      "decreased heart rate variability\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 12\n",
      "number overlapping genes = 0\n",
      "0.6943062389475754\n",
      "(-2.2629717018239894, 3.3981587745901116)\n",
      "0.5675935363830611\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal endocardium morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 28\n",
      "number overlapping genes = 1\n",
      "0.6580086752691361\n",
      "(-1.5469026013950897, 2.4495038063018626)\n",
      "0.45130060245338655\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal cardiac cell glucose uptake\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 17\n",
      "number overlapping genes = 0\n",
      "0.8787523726285018\n",
      "(-2.5948698617219477, 3.0328992245809423)\n",
      "0.21901468142949732\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal myocardium compact layer morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 87\n",
      "number overlapping genes = 4\n",
      "0.1623463072102308\n",
      "(-0.28937569442682154, 1.7263533033381169)\n",
      "0.7184888044556477\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "thin myocardium compact layer\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 59\n",
      "number overlapping genes = 2\n",
      "0.581052442453081\n",
      "(-1.0153220997098777, 1.8111423166161422)\n",
      "0.39791010845313224\n",
      "\n",
      "dilated heart right atrium\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 11\n",
      "number overlapping genes = 0\n",
      "0.650920527384572\n",
      "(-2.1810419024243837, 3.4903605756690164)\n",
      "0.6546593366223163\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal cardiac muscle relaxation\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 33\n",
      "number overlapping genes = 2\n",
      "0.16785618388757684\n",
      "(-0.42469902843641627, 2.441482395959662)\n",
      "1.008391683761623\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal heartbeat\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 459\n",
      "number overlapping genes = 13\n",
      "0.4479009359370941\n",
      "(-0.3430791889598672, 0.7766506037036236)\n",
      "0.2167857073718782\n",
      "\n",
      "absent heartbeat\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 24\n",
      "number overlapping genes = 3\n",
      "0.0035310553712639666\n",
      "(0.592703206524049, 3.0196642316653675)\n",
      "1.806183719094708\n",
      "\n",
      "cardia bifida\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 19\n",
      "number overlapping genes = 0\n",
      "0.9401226419991326\n",
      "(-2.701974643515019, 2.91733496611851)\n",
      "0.10768016130174551\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal myocardial fiber physiology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 145\n",
      "number overlapping genes = 11\n",
      "6.786269386171751e-05\n",
      "(0.6424728779031554, 1.8869535937937953)\n",
      "1.2647132358484754\n",
      "\n",
      "failure of heart looping\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 46\n",
      "number overlapping genes = 1\n",
      "0.952334373393864\n",
      "(-2.0444176767487265, 1.9234057811051715)\n",
      "-0.06050594782177754\n",
      "\n",
      "abnormal direction of heart looping\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 40\n",
      "number overlapping genes = 1\n",
      "0.9348179265232167\n",
      "(-1.904296900545638, 2.070140857038808)\n",
      "0.08292197824658487\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal response of heart to induced stress\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 106\n",
      "number overlapping genes = 6\n",
      "0.02610608990525746\n",
      "(0.11199399376991714, 1.7707202189951596)\n",
      "0.9413571063825383\n",
      "\n",
      "increased response of heart to induced stress\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 57\n",
      "number overlapping genes = 5\n",
      "0.0026816720690196593\n",
      "(0.4905743839518386, 2.335925242940916)\n",
      "1.4132498134463773\n",
      "\n",
      "decreased response of heart to induced stress\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 34\n",
      "number overlapping genes = 3\n",
      "0.019564230837919767\n",
      "(0.22725609084206044, 2.605092489178241)\n",
      "1.4161742900101508\n",
      "\n",
      "enlarged myocardial fiber\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 69\n",
      "number overlapping genes = 6\n",
      "0.0010825943488664874\n",
      "(0.5625594341730259, 2.248268420896774)\n",
      "1.4054139275348998\n",
      "\n",
      "small myocardial fiber\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 14\n",
      "number overlapping genes = 1\n",
      "0.25484061740524955\n",
      "(-0.8532314455962604, 3.2191322239076774)\n",
      "1.1829503891557085\n",
      "\n",
      "myocardial fiber degeneration\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 33\n",
      "number overlapping genes = 1\n",
      "0.782148602703644\n",
      "(-1.7115036075780896, 2.273761972731572)\n",
      "0.28112918257674124\n",
      "\n",
      "decreased myocardial fiber number\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 19\n",
      "number overlapping genes = 0\n",
      "0.9401226419991326\n",
      "(-2.701974643515019, 2.91733496611851)\n",
      "0.10768016130174551\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal cardinal vein morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 33\n",
      "number overlapping genes = 1\n",
      "0.782148602703644\n",
      "(-1.7115036075780896, 2.273761972731572)\n",
      "0.28112918257674124\n",
      "\n",
      "abnormal anterior cardinal vein morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 13\n",
      "number overlapping genes = 1\n",
      "0.2254453944459337\n",
      "(-0.7791721489886974, 3.305267195070543)\n",
      "1.2630475230409228\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal heart weight\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 497\n",
      "number overlapping genes = 15\n",
      "0.28540389038319336\n",
      "(-0.23790968408406354, 0.8079388794464191)\n",
      "0.28501459768117776\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "dilated heart\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 246\n",
      "number overlapping genes = 9\n",
      "0.15885663424863194\n",
      "(-0.18907747292465282, 1.1559694909907967)\n",
      "0.4834460090330719\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "decreased cardiac muscle contractility\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 252\n",
      "number overlapping genes = 8\n",
      "0.3572862674776722\n",
      "(-0.3769238315879514, 1.0445617141722567)\n",
      "0.3338189412921526\n",
      "\n",
      "hemopericardium\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 49\n",
      "number overlapping genes = 1\n",
      "0.901488999510235\n",
      "(-2.1077746754017124, 1.857358574800644)\n",
      "-0.12520805030053417\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal heart ventricle morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 774\n",
      "number overlapping genes = 26\n",
      "0.04771335066257379\n",
      "(0.0041031203919815185, 0.8103715397899549)\n",
      "0.4072373300909682\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "pericardial effusion\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 118\n",
      "number overlapping genes = 4\n",
      "0.4344332070650283\n",
      "(-0.6021656885943816, 1.4010373215814294)\n",
      "0.3994358164935239\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal myocardium layer morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 447\n",
      "number overlapping genes = 15\n",
      "0.13709619174923665\n",
      "(-0.12648306154514521, 0.9211268680503374)\n",
      "0.3973219032525961\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "cardiomyopathy\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 137\n",
      "number overlapping genes = 5\n",
      "0.29755545871074895\n",
      "(-0.42076750271038577, 1.375404670703456)\n",
      "0.47731858399653504\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "decreased heart rate\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 261\n",
      "number overlapping genes = 8\n",
      "0.41235880661781443\n",
      "(-0.41324727862308197, 1.0074502152970233)\n",
      "0.2971014683369706\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal heart size\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 1253\n",
      "number overlapping genes = 37\n",
      "0.11545759201633614\n",
      "(-0.06730194605874296, 0.6164589927036257)\n",
      "0.27457852332244137\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "increased cardiac muscle contractility\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 64\n",
      "number overlapping genes = 1\n",
      "0.6933003932429282\n",
      "(-2.37571530143439, 1.5797951532066958)\n",
      "-0.39796007411384693\n",
      "\n",
      "cardiac interstitial fibrosis\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 61\n",
      "number overlapping genes = 4\n",
      "0.034947344957591914\n",
      "(0.07741385332378936, 2.1139814838153734)\n",
      "1.0956976685695814\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal heart position or orientation\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 63\n",
      "number overlapping genes = 3\n",
      "0.2038752980987566\n",
      "(-0.40922166000536775, 1.9176931237402104)\n",
      "0.7542357318674213\n",
      "\n",
      "myocardium necrosis\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 23\n",
      "number overlapping genes = 0\n",
      "0.9533956257004136\n",
      "(-2.8869833414451063, 2.7197975793044624)\n",
      "-0.08359288107032192\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal heart septum morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 411\n",
      "number overlapping genes = 17\n",
      "0.013890436643283386\n",
      "(0.12633881996145402, 1.1165785932358077)\n",
      "0.6214587065986308\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal cardiac outflow tract development\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 159\n",
      "number overlapping genes = 2\n",
      "0.3841322906433916\n",
      "(-2.0187181123435374, 0.7772077093444332)\n",
      "-0.6207552014995521\n",
      "\n",
      "congestive heart failure\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 87\n",
      "number overlapping genes = 2\n",
      "0.9964183887372254\n",
      "(-1.4085745848640348, 1.4021372208056353)\n",
      "-0.0032186820291997265\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "dilated heart ventricle\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 162\n",
      "number overlapping genes = 6\n",
      "0.23865143743102613\n",
      "(-0.3274037621844027, 1.3146120303232602)\n",
      "0.4936041340694288\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "increased heart atrium size\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 48\n",
      "number overlapping genes = 1\n",
      "0.9180494704834862\n",
      "(-2.087096123579111, 1.878895901547685)\n",
      "-0.10410011101571293\n",
      "\n",
      "increased heart right atrium size\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 11\n",
      "number overlapping genes = 0\n",
      "0.650920527384572\n",
      "(-2.1810419024243837, 3.4903605756690164)\n",
      "0.6546593366223163\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "increased heart ventricle size\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 174\n",
      "number overlapping genes = 8\n",
      "0.04769014140862343\n",
      "(0.007362195217282208, 1.4392169605184584)\n",
      "0.7232895778678703\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal heart ventricle pressure\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 63\n",
      "number overlapping genes = 1\n",
      "0.7051170581540702\n",
      "(-2.3599090013519044, 1.5960986852700099)\n",
      "-0.38190515804094716\n",
      "\n",
      "abnormal fetal cardiomyocyte morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 40\n",
      "number overlapping genes = 1\n",
      "0.9348179265232167\n",
      "(-1.904296900545638, 2.070140857038808)\n",
      "0.08292197824658487\n",
      "\n",
      "abnormal cardiac epithelial to mesenchymal transition\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 28\n",
      "number overlapping genes = 1\n",
      "0.6580086752691361\n",
      "(-1.5469026013950897, 2.4495038063018626)\n",
      "0.45130060245338655\n",
      "\n",
      "ventricular tachycardia\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 24\n",
      "number overlapping genes = 1\n",
      "0.5496390725284372\n",
      "(-1.3925240107009893, 2.616246221876069)\n",
      "0.6118611055875398\n",
      "\n",
      "delayed heart looping\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 11\n",
      "number overlapping genes = 0\n",
      "0.650920527384572\n",
      "(-2.1810419024243837, 3.4903605756690164)\n",
      "0.6546593366223163\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "cardiac valve regurgitation\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 27\n",
      "number overlapping genes = 1\n",
      "0.6316489189504142\n",
      "(-1.510476605063172, 2.488667401364193)\n",
      "0.4890953981505106\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "heart hemorrhage\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 60\n",
      "number overlapping genes = 1\n",
      "0.7421686512094239\n",
      "(-2.3109445581283317, 1.6466555369476636)\n",
      "-0.332144510590334\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal heart ventricle outflow tract morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 105\n",
      "number overlapping genes = 4\n",
      "0.3087962586502744\n",
      "(-0.4825392810833877, 1.524989213432396)\n",
      "0.5212249661745041\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "heart right ventricle hypoplasia\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 16\n",
      "number overlapping genes = 1\n",
      "0.31460277435193573\n",
      "(-0.9867431941643781, 3.066224562572851)\n",
      "1.0397406842042365\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal heart and great vessel attachment\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 225\n",
      "number overlapping genes = 8\n",
      "0.21290159532084385\n",
      "(-0.2595423119992246, 1.164696243944116)\n",
      "0.4525769659724457\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal heart and great artery attachment\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 223\n",
      "number overlapping genes = 8\n",
      "0.20365487975557695\n",
      "(-0.25028851258749074, 1.174181238451207)\n",
      "0.4619463629318581\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal heart right ventricle outflow tract morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 65\n",
      "number overlapping genes = 3\n",
      "0.22394901680403334\n",
      "(-0.44123272408142333, 1.8839062611332336)\n",
      "0.7213367685259051\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal heart left ventricle outflow tract morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 78\n",
      "number overlapping genes = 3\n",
      "0.36942260916965863\n",
      "(-0.6276681181717612, 1.688214790511295)\n",
      "0.5302733361697669\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal heart atrium and ventricle connection\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 91\n",
      "number overlapping genes = 5\n",
      "0.04960227345243174\n",
      "(0.0015793571944470175, 1.8150007149697318)\n",
      "0.9082900360820894\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "heart left ventricle outflow tract stenosis\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 22\n",
      "number overlapping genes = 2\n",
      "0.05150164762601168\n",
      "(-0.009429864471458993, 2.903921318342084)\n",
      "1.4472457269353125\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "heart right ventricle outflow tract stenosis\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 22\n",
      "number overlapping genes = 2\n",
      "0.05150164762601168\n",
      "(-0.009429864471458993, 2.903921318342084)\n",
      "1.4472457269353125\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal atrium myocardium morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 11\n",
      "number overlapping genes = 0\n",
      "0.650920527384572\n",
      "(-2.1810419024243837, 3.4903605756690164)\n",
      "0.6546593366223163\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal ventricle myocardium morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 118\n",
      "number overlapping genes = 7\n",
      "0.01147307426815905\n",
      "(0.22308384067362463, 1.7626865975910448)\n",
      "0.9928852191323347\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "myocardium hypoplasia\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 97\n",
      "number overlapping genes = 5\n",
      "0.0687416165639415\n",
      "(-0.06458202635193111, 1.7456236916986354)\n",
      "0.8405208326733522\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ventricle myocardium hypoplasia\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 81\n",
      "number overlapping genes = 5\n",
      "0.026159853533152123\n",
      "(0.12250439934308088, 1.9423961824335385)\n",
      "1.0324502908883098\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "myocardial trabeculae hypoplasia\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 61\n",
      "number overlapping genes = 2\n",
      "0.6141507368953809\n",
      "(-1.0491089065129666, 1.7757385259689473)\n",
      "0.36331480972799035\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal heart electrocardiography waveform feature\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 341\n",
      "number overlapping genes = 11\n",
      "0.25591637708227777\n",
      "(-0.2557427912180076, 0.9610703067958889)\n",
      "0.35266375778894066\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal heart layer morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 456\n",
      "number overlapping genes = 15\n",
      "0.15909305469086954\n",
      "(-0.14743060183028278, 0.8998330204601531)\n",
      "0.37620120931493517\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "thin ventricle myocardium compact layer\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 26\n",
      "number overlapping genes = 2\n",
      "0.08640911152552311\n",
      "(-0.18093940868396574, 2.7103521140270357)\n",
      "1.264706352671535\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "heart block\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 118\n",
      "number overlapping genes = 4\n",
      "0.4344332070650283\n",
      "(-0.6021656885943816, 1.4010373215814294)\n",
      "0.3994358164935239\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "increased heart right ventricle size\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 55\n",
      "number overlapping genes = 3\n",
      "0.13183152504803763\n",
      "(-0.26990968910436774, 2.065455726113677)\n",
      "0.8977730185046546\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal heart right ventricle size\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 81\n",
      "number overlapping genes = 3\n",
      "0.40568949531342147\n",
      "(-0.6662017444948927, 1.6479792917914982)\n",
      "0.49088877364830275\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal heart left ventricle size\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 161\n",
      "number overlapping genes = 9\n",
      "0.00714056933347083\n",
      "(0.2530699734048618, 1.6115263260158847)\n",
      "0.9322981497103733\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "increased heart left ventricle size\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 131\n",
      "number overlapping genes = 8\n",
      "0.005345569653409248\n",
      "(0.3038953039445289, 1.7470052445156483)\n",
      "1.0254502742300886\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "decreased heart left ventricle size\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 29\n",
      "number overlapping genes = 1\n",
      "0.6838624058265801\n",
      "(-1.5820528483568495, 2.411809823559534)\n",
      "0.41487848760134227\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal cardiac muscle tissue morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 520\n",
      "number overlapping genes = 17\n",
      "0.14006176466209883\n",
      "(-0.12184019227129472, 0.8641257363627145)\n",
      "0.3711427720457099\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "cardiac muscle necrosis\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 30\n",
      "number overlapping genes = 0\n",
      "0.806392931317837\n",
      "(-3.1461084091285785, 2.4467537675130147)\n",
      "-0.3496773208077819\n",
      "\n",
      "sinus bradycardia\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 13\n",
      "number overlapping genes = 0\n",
      "0.7353056736663703\n",
      "(-2.338715699078265, 3.3137085040739587)\n",
      "0.48749640249784676\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal heart left ventricle pressure\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 44\n",
      "number overlapping genes = 1\n",
      "0.98823416348311\n",
      "(-1.999846679896033, 1.969977611132633)\n",
      "-0.014934534381700004\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal heart right ventricle pressure\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 20\n",
      "number overlapping genes = 0\n",
      "0.9686339419527712\n",
      "(-2.7515228169535777, 2.8641876568718305)\n",
      "0.05633241995912641\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "absent heart\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 11\n",
      "number overlapping genes = 1\n",
      "0.1685955456781253\n",
      "(-0.612357220595233, 3.5033130673400317)\n",
      "1.4454779233723993\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal fetal cardiomyocyte physiology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 67\n",
      "number overlapping genes = 1\n",
      "0.6593576798907441\n",
      "(-2.4216982463819123, 1.5324105827803265)\n",
      "-0.4446438318007928\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "decreased fetal cardiomyocyte proliferation\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 22\n",
      "number overlapping genes = 0\n",
      "0.9782093848563553\n",
      "(-2.8438308359708757, 2.7656575106717676)\n",
      "-0.03908666264955407\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "thick heart valve cusps\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 33\n",
      "number overlapping genes = 1\n",
      "0.782148602703644\n",
      "(-1.7115036075780896, 2.273761972731572)\n",
      "0.28112918257674124\n",
      "\n",
      "abnormal heart echocardiography feature\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 58\n",
      "number overlapping genes = 4\n",
      "0.027149274896251428\n",
      "(0.1298079674460333, 2.1700491355480134)\n",
      "1.1499285514970232\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal heart valve physiology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 33\n",
      "number overlapping genes = 1\n",
      "0.782148602703644\n",
      "(-1.7115036075780896, 2.273761972731572)\n",
      "0.28112918257674124\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "increased cardiac stroke volume\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 21\n",
      "number overlapping genes = 0\n",
      "0.9958272910303897\n",
      "(-2.7987382563569554, 2.8137138680969436)\n",
      "0.007487805869994091\n",
      "\n",
      "decreased cardiac stroke volume\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 26\n",
      "number overlapping genes = 1\n",
      "0.6047917153710545\n",
      "(-1.472678679036409, 2.529419831139403)\n",
      "0.528370576051497\n",
      "\n",
      "delayed heart development\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 19\n",
      "number overlapping genes = 1\n",
      "0.4045860266224103\n",
      "(-1.1586693013743776, 2.873180927815657)\n",
      "0.8572558132206396\n",
      "\n",
      "cardiac edema\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 13\n",
      "number overlapping genes = 0\n",
      "0.7353056736663703\n",
      "(-2.338715699078265, 3.3137085040739587)\n",
      "0.48749640249784676\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal pericardial cavity morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 131\n",
      "number overlapping genes = 7\n",
      "0.024381774872861778\n",
      "(0.1139809914002714, 1.6488635033956034)\n",
      "0.8814222473979374\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "prenatal lethality prior to heart atrial septation\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 110\n",
      "number overlapping genes = 2\n",
      "0.7330562872356408\n",
      "(-1.645883773623902, 1.1579713636714268)\n",
      "-0.2439562049762376\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal cardiac muscle glycogen level\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 21\n",
      "number overlapping genes = 0\n",
      "0.9958272910303897\n",
      "(-2.7987382563569554, 2.8137138680969436)\n",
      "0.007487805869994091\n",
      "\n",
      "increased cardiac muscle glycogen level\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 17\n",
      "number overlapping genes = 0\n",
      "0.8787523726285018\n",
      "(-2.5948698617219477, 3.0328992245809423)\n",
      "0.21901468142949732\n",
      "\n",
      "decreased heart left ventricle posterior wall thickness\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 14\n",
      "number overlapping genes = 0\n",
      "0.7740935236234859\n",
      "(-2.409141497019146, 3.2358094993592985)\n",
      "0.41333400117007635\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal heart atrium arrangement\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 17\n",
      "number overlapping genes = 2\n",
      "0.021432662587959284\n",
      "(0.25671157529782196, 3.2136884337472997)\n",
      "1.7352000045225608\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal heart ventricle wall thickness\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 222\n",
      "number overlapping genes = 9\n",
      "0.08534746848083427\n",
      "(-0.08234422826926469, 1.2654152238365421)\n",
      "0.5915354977836387\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal heart left ventricle weight\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 42\n",
      "number overlapping genes = 1\n",
      "0.9741751109367783\n",
      "(-1.9532070421119314, 2.018812126277102)\n",
      "0.03280254208258526\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal cardiac thrombosis\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 34\n",
      "number overlapping genes = 0\n",
      "0.7389226833773714\n",
      "(-3.268794667514915, 2.3186778677001962)\n",
      "-0.47505839990735943\n",
      "\n",
      "abnormal circardian behavior entrainment\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 13\n",
      "number overlapping genes = 1\n",
      "0.2254453944459337\n",
      "(-0.7791721489886974, 3.305267195070543)\n",
      "1.2630475230409228\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal heart left ventricle wall thickness\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 46\n",
      "number overlapping genes = 1\n",
      "0.952334373393864\n",
      "(-2.0444176767487265, 1.9234057811051715)\n",
      "-0.06050594782177754\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal heart right ventricle wall thickness\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 12\n",
      "number overlapping genes = 0\n",
      "0.6943062389475754\n",
      "(-2.2629717018239894, 3.3981587745901116)\n",
      "0.5675935363830611\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "increased heart left ventricle wall thickness\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 17\n",
      "number overlapping genes = 1\n",
      "0.3446682613592619\n",
      "(-1.0473830733439908, 2.9976785292777923)\n",
      "0.9751477279669007\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "decreased heart left ventricle wall thickness\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 30\n",
      "number overlapping genes = 0\n",
      "0.806392931317837\n",
      "(-3.1461084091285785, 2.4467537675130147)\n",
      "-0.3496773208077819\n",
      "\n",
      "increased cardiac cell glucose uptake\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 13\n",
      "number overlapping genes = 0\n",
      "0.7353056736663703\n",
      "(-2.338715699078265, 3.3137085040739587)\n",
      "0.48749640249784676\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "myocardial fiber disarray\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 19\n",
      "number overlapping genes = 1\n",
      "0.4045860266224103\n",
      "(-1.1586693013743776, 2.873180927815657)\n",
      "0.8572558132206396\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal response to myocardial ischemic injury\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 56\n",
      "number overlapping genes = 1\n",
      "0.7955780657546137\n",
      "(-2.241718184343156, 1.7182741473119658)\n",
      "-0.26172201851559507\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "decreased susceptibility to myocardial ischemic injury\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 27\n",
      "number overlapping genes = 0\n",
      "0.8642498026572712\n",
      "(-3.043127200229141, 2.5548204318201906)\n",
      "-0.2441533842044752\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "increased susceptibility to myocardial ischemic injury\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 32\n",
      "number overlapping genes = 1\n",
      "0.7583472123710618\n",
      "(-1.6806718759494526, 2.30653660283605)\n",
      "0.31293236344329856\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "globular heart\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 14\n",
      "number overlapping genes = 1\n",
      "0.25484061740524955\n",
      "(-0.8532314455962604, 3.2191322239076774)\n",
      "1.1829503891557085\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal heart ventricle wall morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 288\n",
      "number overlapping genes = 12\n",
      "0.03695609673913069\n",
      "(0.037755006136902414, 1.209685035687694)\n",
      "0.6237200209122982\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal heart left ventricle wall morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 102\n",
      "number overlapping genes = 6\n",
      "0.020399287072508914\n",
      "(0.15207033729000463, 1.812725309836663)\n",
      "0.9823978235633337\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal heart right ventricle wall morphology\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 54\n",
      "number overlapping genes = 3\n",
      "0.12385645359501248\n",
      "(-0.2510571432196309, 2.0855484359105585)\n",
      "0.9172456463454637\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "abnormal heart ventricle size\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 225\n",
      "number overlapping genes = 9\n",
      "0.09300166773550779\n",
      "(-0.09630971029433688, 1.2510780345300443)\n",
      "0.5773841621178537\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "decreased heart ventricle size\n",
      "number of genes in root node = 434\n",
      "number of genes in focal MPO term = 54\n",
      "number overlapping genes = 3\n",
      "0.12385645359501248\n",
      "(-0.2510571432196309, 2.0855484359105585)\n",
      "0.9172456463454637\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2337: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._hierarchy_to_pandas(), ignore_index=True)\n",
      "/home/hiyoothere/miniconda3/envs/r4/lib/python3.10/site-packages/ddot/Ontology.py:2339: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df = df.append(self._mapping_to_pandas(), ignore_index=True)\n"
     ]
    }
   ],
   "source": [
    "MP_focal_list = MP_focal_heart_list\n",
    "root_KO_heart_df=validation.MPO_enrichment_root(hier_df,MPO,mgi_df,MP_focal_list,G_int,verbose=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "id": "5a335821",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>OR_p</th>\n",
       "      <th>log_OR</th>\n",
       "      <th>log_OR_CI_lower</th>\n",
       "      <th>log_OR_CI_upper</th>\n",
       "      <th>num_genes_in_term</th>\n",
       "      <th>MP_description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>MP:0000267</th>\n",
       "      <td>0.315871</td>\n",
       "      <td>0.298190</td>\n",
       "      <td>-0.284516</td>\n",
       "      <td>0.880897</td>\n",
       "      <td>392</td>\n",
       "      <td>abnormal heart development</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MP:0000269</th>\n",
       "      <td>0.642400</td>\n",
       "      <td>0.236699</td>\n",
       "      <td>-0.762388</td>\n",
       "      <td>1.235787</td>\n",
       "      <td>138</td>\n",
       "      <td>abnormal heart looping</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MP:0000270</th>\n",
       "      <td>0.502891</td>\n",
       "      <td>-0.674758</td>\n",
       "      <td>-2.648795</td>\n",
       "      <td>1.299279</td>\n",
       "      <td>84</td>\n",
       "      <td>abnormal heart tube morphology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MP:0000274</th>\n",
       "      <td>0.186265</td>\n",
       "      <td>0.257807</td>\n",
       "      <td>-0.124495</td>\n",
       "      <td>0.640108</td>\n",
       "      <td>993</td>\n",
       "      <td>enlarged heart</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MP:0000276</th>\n",
       "      <td>0.048468</td>\n",
       "      <td>1.186164</td>\n",
       "      <td>0.007984</td>\n",
       "      <td>2.364344</td>\n",
       "      <td>42</td>\n",
       "      <td>heart right ventricle hypertrophy</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                OR_p    log_OR  log_OR_CI_lower  log_OR_CI_upper  \\\n",
       "MP:0000267  0.315871  0.298190        -0.284516         0.880897   \n",
       "MP:0000269  0.642400  0.236699        -0.762388         1.235787   \n",
       "MP:0000270  0.502891 -0.674758        -2.648795         1.299279   \n",
       "MP:0000274  0.186265  0.257807        -0.124495         0.640108   \n",
       "MP:0000276  0.048468  1.186164         0.007984         2.364344   \n",
       "\n",
       "            num_genes_in_term                     MP_description  \n",
       "MP:0000267                392         abnormal heart development  \n",
       "MP:0000269                138             abnormal heart looping  \n",
       "MP:0000270                 84     abnormal heart tube morphology  \n",
       "MP:0000274                993                     enlarged heart  \n",
       "MP:0000276                 42  heart right ventricle hypertrophy  "
      ]
     },
     "execution_count": 109,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "root_KO_heart_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "id": "263df3e6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MPO_term_type</th>\n",
       "      <th>MP_description</th>\n",
       "      <th>OR_p</th>\n",
       "      <th>log_OR</th>\n",
       "      <th>log_OR_CI_lower</th>\n",
       "      <th>log_OR_CI_upper</th>\n",
       "      <th>num_genes_in_term</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>MP:0004215</th>\n",
       "      <td>heart</td>\n",
       "      <td>abnormal myocardial fiber physiology</td>\n",
       "      <td>0.000068</td>\n",
       "      <td>1.264713</td>\n",
       "      <td>0.642473</td>\n",
       "      <td>1.886954</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MP:0000929</th>\n",
       "      <td>Neural tube</td>\n",
       "      <td>open neural tube</td>\n",
       "      <td>0.000104</td>\n",
       "      <td>1.052151</td>\n",
       "      <td>0.520738</td>\n",
       "      <td>1.583564</td>\n",
       "      <td>242</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MP:0002625</th>\n",
       "      <td>heart</td>\n",
       "      <td>heart left ventricle hypertrophy</td>\n",
       "      <td>0.000842</td>\n",
       "      <td>1.437784</td>\n",
       "      <td>0.593744</td>\n",
       "      <td>2.281824</td>\n",
       "      <td>67</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MP:0004564</th>\n",
       "      <td>heart</td>\n",
       "      <td>enlarged myocardial fiber</td>\n",
       "      <td>0.001083</td>\n",
       "      <td>1.405414</td>\n",
       "      <td>0.562559</td>\n",
       "      <td>2.248268</td>\n",
       "      <td>69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MP:0004188</th>\n",
       "      <td>Neural tube</td>\n",
       "      <td>delayed embryo turning</td>\n",
       "      <td>0.001317</td>\n",
       "      <td>2.017711</td>\n",
       "      <td>0.786623</td>\n",
       "      <td>3.248798</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MP:0001853</th>\n",
       "      <td>heart</td>\n",
       "      <td>heart inflammation</td>\n",
       "      <td>0.988234</td>\n",
       "      <td>-0.014935</td>\n",
       "      <td>-1.999847</td>\n",
       "      <td>1.969978</td>\n",
       "      <td>44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MP:0011951</th>\n",
       "      <td>heart</td>\n",
       "      <td>increased cardiac stroke volume</td>\n",
       "      <td>0.995827</td>\n",
       "      <td>0.007488</td>\n",
       "      <td>-2.798738</td>\n",
       "      <td>2.813714</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MP:0014070</th>\n",
       "      <td>heart</td>\n",
       "      <td>abnormal cardiac muscle glycogen level</td>\n",
       "      <td>0.995827</td>\n",
       "      <td>0.007488</td>\n",
       "      <td>-2.798738</td>\n",
       "      <td>2.813714</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MP:0002795</th>\n",
       "      <td>heart</td>\n",
       "      <td>dilated cardiomyopathy</td>\n",
       "      <td>0.996418</td>\n",
       "      <td>-0.003219</td>\n",
       "      <td>-1.408575</td>\n",
       "      <td>1.402137</td>\n",
       "      <td>87</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MP:0006138</th>\n",
       "      <td>heart</td>\n",
       "      <td>congestive heart failure</td>\n",
       "      <td>0.996418</td>\n",
       "      <td>-0.003219</td>\n",
       "      <td>-1.408575</td>\n",
       "      <td>1.402137</td>\n",
       "      <td>87</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>281 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           MPO_term_type                          MP_description      OR_p  \\\n",
       "MP:0004215         heart    abnormal myocardial fiber physiology  0.000068   \n",
       "MP:0000929   Neural tube                        open neural tube  0.000104   \n",
       "MP:0002625         heart        heart left ventricle hypertrophy  0.000842   \n",
       "MP:0004564         heart               enlarged myocardial fiber  0.001083   \n",
       "MP:0004188   Neural tube                  delayed embryo turning  0.001317   \n",
       "...                  ...                                     ...       ...   \n",
       "MP:0001853         heart                      heart inflammation  0.988234   \n",
       "MP:0011951         heart         increased cardiac stroke volume  0.995827   \n",
       "MP:0014070         heart  abnormal cardiac muscle glycogen level  0.995827   \n",
       "MP:0002795         heart                  dilated cardiomyopathy  0.996418   \n",
       "MP:0006138         heart                congestive heart failure  0.996418   \n",
       "\n",
       "              log_OR  log_OR_CI_lower  log_OR_CI_upper  num_genes_in_term  \n",
       "MP:0004215  1.264713         0.642473         1.886954                145  \n",
       "MP:0000929  1.052151         0.520738         1.583564                242  \n",
       "MP:0002625  1.437784         0.593744         2.281824                 67  \n",
       "MP:0004564  1.405414         0.562559         2.248268                 69  \n",
       "MP:0004188  2.017711         0.786623         3.248798                 20  \n",
       "...              ...              ...              ...                ...  \n",
       "MP:0001853 -0.014935        -1.999847         1.969978                 44  \n",
       "MP:0011951  0.007488        -2.798738         2.813714                 21  \n",
       "MP:0014070  0.007488        -2.798738         2.813714                 21  \n",
       "MP:0002795 -0.003219        -1.408575         1.402137                 87  \n",
       "MP:0006138 -0.003219        -1.408575         1.402137                 87  \n",
       "\n",
       "[281 rows x 7 columns]"
      ]
     },
     "execution_count": 110,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# join brain and heart results together\n",
    "root_KO_brain_df['MPO_term_type']='Neural tube'\n",
    "root_KO_heart_df['MPO_term_type']='heart'\n",
    "root_KO_df = pd.concat([ root_KO_brain_df,root_KO_heart_df], axis=0, sort= True)\n",
    "#root_KO_df = pd.concat([ root_KO_heart_df], axis=0, sort= True)\n",
    "root_KO_df.loc['MP:0002419','MPO_term_type'] = 'neg_ctrl'\n",
    "root_KO_df = root_KO_df.sort_values('OR_p')\n",
    "root_KO_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "id": "6eed4bc7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "heart          170\n",
      "Neural tube    110\n",
      "neg_ctrl         1\n",
      "Name: MPO_term_type, dtype: int64\n"
     ]
    }
   ],
   "source": [
    "pd.set_option('display.max_rows', 500)\n",
    "\n",
    "print (root_KO_df['MPO_term_type'].value_counts())\n",
    "root_KO_df\n",
    "\n",
    "root_KO_df.to_csv(\"/projects/ps-gleesonlab8/User/hiyoothere/NTD/5.Analysis/Network/output/MPO_NTD_terms.txt\", sep = '\\t', header = True, index = True, index_label='MPO_term')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5b5487b4",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "89ce96c5",
   "metadata": {},
   "source": [
    "### Plot top performing brain and heart terms + negative control term"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "id": "cc867494",
   "metadata": {},
   "outputs": [],
   "source": [
    "# plot top performing brain and heart terms + negative control term (MP:0002419) (for terms which have at least 150 genes)\n",
    "\n",
    "#heart_terms_plot = root_KO_heart_df[root_KO_heart_df['num_genes_in_term']>150]\n",
    "#heart_terms_plot = heart_terms_plot.sort_values('OR_p',ascending=True).head(5).index.tolist() #+heart_terms_plot.sort_values('OR_p',ascending=True).tail(5).index.tolist()\n",
    "\n",
    "#brain_terms_plot = root_KO_brain_df[root_KO_brain_df['num_genes_in_term']>150]\n",
    "#brain_terms_plot = brain_terms_plot.sort_values('OR_p',ascending=True).head(20).index.tolist() #+brain_terms_plot.sort_values('OR_p',ascending=True).tail(5).index.tolist()\n",
    "root_KO_df = root_KO_df.loc[(root_KO_df['num_genes_in_term'] >= 30) & (root_KO_df['OR_p'] < 0.01) | (root_KO_df['MPO_term_type'] == 'neg_ctrl')]\n",
    "#root_KO_df.loc['MP:0002419','MPO_term_type'] = 'neg_ctrl'\n",
    "#root_KO_df = root_KO_df.sort_values('OR_p')\n",
    "#print (root_KO_df)\n",
    "#Together\n",
    "\n",
    "brain_terms_plot = root_KO_df[root_KO_df['num_genes_in_term']>30]\n",
    "##brain_terms_plot\n",
    "#brain_terms_plot = root_KO_df[root_KO_df['OR_p']< 0.01]\n",
    "#brain_terms_plot\n",
    "brain_terms_plot = brain_terms_plot.sort_values('OR_p',ascending=True).head(30).index.tolist() #+brain_terms_plot.sort_values('OR_p',ascending=True).tail(5).index.tolist()\n",
    "\n",
    "neg_ctrl_terms_plot=['MP:0002419']\n",
    "\n",
    "terms_plot = brain_terms_plot+neg_ctrl_terms_plot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "id": "4b017689",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['MP:0004215',\n",
       " 'MP:0000929',\n",
       " 'MP:0002625',\n",
       " 'MP:0004564',\n",
       " 'MP:0002085',\n",
       " 'MP:0004485',\n",
       " 'MP:0003105',\n",
       " 'MP:0001672',\n",
       " 'MP:0001698',\n",
       " 'MP:0010579',\n",
       " 'MP:0000285',\n",
       " 'MP:0003720',\n",
       " 'MP:0010578',\n",
       " 'MP:0002419',\n",
       " 'MP:0002419']"
      ]
     },
     "execution_count": 113,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "terms_plot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "id": "723acda9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAh8AAAHvCAYAAAD0J71VAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjUuMywgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/NK7nSAAAACXBIWXMAAAsTAAALEwEAmpwYAADbGElEQVR4nOydd5gV1fnHP1QFlg5SRCmirxI0qCyKnWBNsIBIYvuJMTEx9ogEY8NuTCyxRowRYzeIRlERJIogFsReeBEVlSpFytLL/v54z8Ds7L279267y/J+nuc+u3fmzDnvOTN3znfe854ztQoLC3Ecx3Ecx6kqaufaAMdxHMdxti1cfDiO4ziOU6W4+HAcx3Ecp0px8eE4juM4TpVSN9cGOI7jOGVDRLYD8oF5wMYcm+M4ceoA7YCpqro2udPFh+M4ztZLPjAp10Y4TgkcDExObnTx4TiOs/UyD+Cxxx6jbdu2lV7YqFGjABg4cGCll+Vs3cyfP59TTz0VwjWaxMWH4zjO1stGgLZt29KhQ4dKL6xZs2YAVVKWU2NIORzoAaeO4ziO41QpLj4cx3Ecx6lSfNjFcRzHyYif/OQnuTbBqSG4+HAcx3EyIj8/P9cmODUEH3ZxHMdxMmL9+vWsX78+12Y4NQAXH47jOE5GPPbYYzz22GO5NsOpAbj4cBzHcRynSnHx4TiO4zhOleLiw3Ecx3GcKsXFh+M4juM4VYpPtXUcx3EyokePHrk2wakhuPhwHMdxMsLFR1HWr4cFC+Drr2HhQli3DrbbDtq2hU6doE0bqFMn11ZWT1x8OI7jOBmxatUqABo2bJhjS3LLxo0mOD76CL77DpYvL7r/44+heXMTIHvtZX9r1cqFpdUXFx+O4zhORjz99NMADB48OLeG5JD16+H99+GNN2DlyvTpfvzRPl9/DX36wJ57uhckjgecOo7jOE4GbNxo3o7XXitZeMRZtgxefRW++AIKCyvXvq0J93yUERE5DHgNyFfV93JsTqUjIq8DBaraL9e2VEdEZDgwRFXzwvdOwDfASao6Kot8BgMPAa1VdVEJ6fYB7gX2AhoAnYHhQE9V7Z5NXrmirG1UlWVV9zZ0qpZ582DyZFizJrvjCgrMU7LDDvZxXHw4TmUxD+gNzKik/O8B6gD9gFWhvOuARpVU3tZOZZ8PZxtg5kxYurRsxy5caPEhLj4MFx9bOSLSQFVX59qObYFs2lpV1wJvV6I5ewB3qur/Ytu+qsTyirE1XXtVcD6cGs6SJaCaet/MmaOZOvVmCgrmkpfXnvz8YXTtOqBImsJC+PRT2H13yMurAoOrOS4+UiAi+wF/BvKBpthN/S5VfSBF8tYiMgo4GijAOoQbY3kNB4YA+wH3AT2Bb4ErVPWZRLlnAxcDXYAFwCPA1aq6IewfjLmA98dc7IcAT4rII9gQ0NHAmcAvgBXh2AdE5CzgcqAlMBb4raouD3k2AG4BjgB2BhYCrwKXquqSLNttZKjfhcCtgAAfBpvmAHcDJwBLgZtV9b5wXH9gNLCbqn4Zy68B9sR6S9SmInIwcBOwL7AaeBm4RFXnx47bDrgSOAXYMaqTqg4O+0s9v7FhtZ8DZwDHAFOBw0WkHTbkcRSwDDuvRWLZU7n5ReQ04HdAN8xr8QlwmapOzrCJ43YBXCkiVwITVfWwqP2jYZcYXUTkSeAA4AfgOlV9MJFvL+B6zDsAMA64UFVnJ+ozGLv+TsLOY9c0ds4CxoRjLgZaAOOB36vqvETy7UTk78DpwHrgGeCPqrpGRHYAZgPnJn9/IjIOQFWPFJF6wf5fAW2BH4H3gcGq+kOa81EbGAb8BrtOvgfuUdXbU9UpVm4L7DdzHNAEO49XqOorsTS1sN/cH7Br7H/AzcDkyIZQ5/5AJ1XdFDv2AOBN4EBVnVKSLVVNz549c21Czli40KbVJpk5czSTJg1lwwbT4QUFc5g0aShAMQEyfz4sXuziA1x8pKMj8BYwAnNp7w/cJSL1VfWeRNoRwFPAicCRwA0iskRV/xFLUw94Eut8rwcuwETDrqo6C0BEzgfuxDqyi4C9gWuwG+lZiTKfBB4E/gqsAeqH7fcBD4d9/weMEJFdMOFzAbAD8PeYDWDxAvWwznoB0AG4DBMpvTJsrzhtgTuAG4Jtt4X2+Ra7SQ8EBgH3isg7qvo+8AImMn4dyo4YCOQBIwFEZF9MGE0Gfgk0B24EJojIvqoajcQ+A/ws7HsbaA3E7wLZnN8HsPY+EYjCxZ7FYizOBRZhneuuGbRNJ+BRYCbW5icBr4lIT1X9KIPjwTrU3lhn9jjwT2B5iUdY+z+AXS+nAP8Ukbmq+jJsFh5vYOLg1GDbcOAVEdlLVTfG8roJeCWkK+3+cQLWoZ8LNMM67Gex9o5zA/ASJhz2Dt/nAtcH4fBf7NqIi8OOQF/g5LBpGNbR/wn4DBPafbHrOx1/xX5rN4X69wVuFZE8Vb0u1QEiUgcTvF2xa3U2cDbwoogcoaqRMDwfuBYT4eOBg7BzH2cE9js8AmvTiLOAL5LCQ0SaYe0Yp0MJ9atwundP6trcM3DgwCopp6AAFqWI+lmwYBqbNq0rsm3DhtVMnHgJ06c/Xiz9G29AVc1UHjWqUkOpyoWLjxSo6tPR/+EJZhLWgf0OG2uP85qqDgn/vxKeiq8QkRGxp5n62BPumJDnNKyjPwG4I9zQrgL+o6p/iOW1CbhZRG5Q1a9jZT6Q8K4cFv59RlWvCdvexDrcXwOdVXVl2L4P1uldEOq6BPh9LK+6wOfAByKyTxAH2dAC6KOqn4T8mmHemjdV9fKw7XWsMz8JeF9VN4jIQ8CZInJFrLP7DfCyqs4N3y/HntyPUdV1Ia/pmMD4FTBSRI7APD+nqOoTMbs2/5/l+X0pdn4RkaMwMXekqo4P28Zj4qpEVPX6WD61MSHVA+tsLkhzWDKP5cDb4dqYraqZDCU8ErteXhGRrtj19nLYdgvmoTpOVQuDfe9h8REnU7TT/FRVz8zEVswrcLSqLgt5zgvlHxX3EgDvxa77cSLSGxOeUXuNCNu7qernYduvMe/Gc+H7fsA4Vb03lu/odIaJSCtMINyqqlfEym4M/ElEblfVghSH/gIT5b9Q1ZdCXmOBj4GrMTFZBxND/1bVS2N5N8G8ggCo6mciMgU7/6+EvPIwcT48RdkXhTJyxrJlywBo2rRpLs3ICelmqiSFR2nbHcPFRwpEpDn24z8ee7KIZmevTZH82cT3UdgNuwPwXdi2CXv6AUBVF4vID2x5atkdaIU9ocZ5CvgLcCAQFx/PpzF9XKyMVSIyB/g8Eh6BGUAbEamnqutDfU9ny9N73CG4G/aknQ3zIuERKy9p2/rglt8plu4B7Eny58ALIrIrNqx0QizNwcCTkfAIeb0T8joY85D0xbwZT6YzMMvzm2zr/YHlkfAINqwRkRexTiMtIrIH9lR/ANAmtmthScdVAMlr9BlM1NbBhPFBwKVAHRGJ0swBFOto4+Ij3bWXitci4QGgquNEZDkmFOLi45XEcZ8HmyJexa7/s4BLgnAbjImq6Fp4H7hURK4BXgSmJTw2SfbDPDypfnPnYB6YSSmOOxhYEQmPUK9NIvIf4PLQph2AdmwRRhHPEBMfgRGYh7Klqi7GRPR2wL9TlH0HwQsYo0MaOyuFZ5+1S6k6rfNRVU/3n30G//lP8e1PPNGLgoI5xbbn5e1Iv35FbatbF045Bbp0qSwrtx58nY/UjMTcyrdhQyn52JDJdinS/pD4Ho0KtottWx0C3uKsA7YP/zcPf+cn0kR5tUizPcmPKcpYmmJbLUJdQrzFv4Fp2FDG/ljsCDH7siGVDaSxY3P+YfhpPFuGmM7C6vli7JjmFG8jQrqojVpiAqikGfUjyfz8Jtu6HcXPeap0RQhP1OOweJ4hmLDKx4Z/ytLO2ZDqGq2HCd4WmPi6DYu3iH/2xOKAkseWtdxoW7vEtlTXTPzaKMSGl04PsR1HBrvicSs3YDEVpwHvAAtE5PogBlKR7W8uflyqNojaNI8t9UuKylTt8TQmlk8P338DPK+qxQSpqi5V1VnxDzbs41QBzZtDs2bFt+fnD6Nu3aKje3XrNiA/f1ixtC1bQpMmlWTgVoZ7PhKIyPbY9MVLVPXO2PYz0hySnDgVPdEmg+pKIgrsbJPY3iaxP6Iil6o5CfhIVX8bbQixFblgBBYL0wEL8nw4CrYNLKF4GxG2fRb+Xwy0E5FaqQRIGc5vMo95FD/nkQ0l0Rt7Sj1WVT+Mlds42FyZ7IB5MiLaYOJiEdbJF2JxD0kPCRQXjdlce6naaQey+21EPITFUByHeQfeVdVPo51B3F8DXCMinbHOfDjWOf+jWG5Ff3PJtonvT3VcumtwPRZ0HtWvdSJNsfZQ1dUi8ihwVhi+2y/Uw6lm7LADdO4MH3xQdHsUVFrabBeA3XYzAeK45yMV22HtstlTETqsE9Ok75/4PhALlsvmiUSxp6Sk234QdrPPeDZEGWhA8eGGUyuxvJJ4HuuIH8MCVx9M7J8MnBCefgEQkXwskDNyPb8KNCT9EEi25zfJO0CTEFsSP/4XpRwXPRrFy/0p8JMMyy0PyWv0RMKwRBiSmwJ0U9X3UnxmlqPcPiKyOThARI7E4kDeyTajMJvpecxrdBzmCUmX9htVvRb7DXZLk+xdTCyk+s2tJP1w42SgsYhE3sEobmggMCUM9czGPConJI5Nd42NALpjM6i+p/gwlFMNqFsXfvITe3Fckq5dB3Dyye/y29/O5uST300pPPLyYNdd/R0vEe75SKCqy0RkKnCZiCzGOotLsJkbqegjIn/FhgyOwuI9zo1PncugzI0ici0242IhNvtjb+xJ7yFV/absNSqV8cA9YlOCJwOHYzf3KifEgjyEBetNUtXkglA3YB3lS2GaYnPsif1zQoyHqr4qIi8B/wozfd7BXOgDVfWXZTi/SRvHhuMfFZFhmGj8I9aRlcTb2FPxfSJyE/akHHWQlc3pIrIGG1o7BYs5iYulIVig5ChsBs0SoD3QB3hRVZ8rY7nLgZdF5GbsXN0CTE0Em2bDCGwW1koSMT0i8hwmGD7AppkfjcUUvZoqI1VdJCJ3AkNEZC02tbUPFu9xdSJOKs6LmHB5RET+zJbZLntgv53o93wTFky+EBtuO4gtYqTIvUFVPxGRt7GhuOuyuXc4VcuOO8JPfwpTp2a3VHqdOrDPPva2W8dwz0dqTsECJf+FzX4Yh01jTcXvsGl3z2LjzVcmIu4zQlXvDnkdga2PcD42Te/3JR1XAdwP/C2U8yx2E62auWupiWYoJL0eqOo0rH0aAP/Bpia/AfSNTbMFe8K8E2vPl7F4hvjMhWzObypOwMTEPdgT+KRU9iZsX4C1a0ssEHEoFuT7cRbllpWTsY7xOeAw4OxEwOTbWFBzfaweY7HVUmuzZTirLDwXPvdhQx/vUtwbkA3jsfP4H1Vdkdg3GRtOewT7/RwFnKGqJQXIDsVmj5wRjhmELZGfcpotmLDA1nx5ni1DVR2Afqr6eizpXSHv04H/smW6O9jaMElGY17Of5Vgb87p3bs3vXv3Lj1hDaVBAzjwQBMgtTPsPevWhfx82G8/qF+/9PTbCrUK/U03TjVCRK7CPAntVXVVru1xykaYgTRGVc+rwDwPASZSDRffygQRuQTz/rRT1R8S+/4HbFDVI7PMsxPwzYQJE+jQoUqX/NimWbEC3n3XXha3eHFqL0jt2tC6NXTvDvvuW3Vre1QXZs+eTd++fcGWepiV3O/DLk61QGyOp2DegPtdeDgRItIe8y7ehgWaVnvhEaZVn44NE67BPEt/Ah6LCw8R6YkNyfTBpplXaxaFVbZatWqVY0tyS+PG0KcP7LknfPutTcNdudLeelunDjRtaqKjQwcLMPU4j+K4+HCqC/dj03zHYS5/x4k4G1uB92Ns5d6tgVXYGim/w4Js52HDdFcm0k3FYmNu0rDibHVmzJgxQPVa5yNX1K695S213brB6tWwaZOJj4YNbYjGSY+LD6daoKqH5doGp+JQ1U4VmNdwUq/4WW1R1W8JAailpPNn4hpAo0b2cTLHA04dx3Ecx6lSXHw4juM4jlOluPhwHMdxHKdK8ZgPx3EcJyMOOeSQXJvg1BBcfDiO4zgZ0cVfx+pUED7s4jiO42TE/PnzmT8/1YulHSc73PPhOI7jZMTYsWOBGr7Ox/r1toTp6tW2dGnt2rZoR+PGtla6UyF4SzqO4zjO4sUwezZ8/jksXbplydK6dW0Rj+bN7bW2HTpAs2a5tnarx8WH4ziOs+2ydi3MmAFvvQXz59sypUlWrLB9M2ZA+/ZwwAHQtSvUq1f19tYQPObDcRzH2TYpKICJE2HMGJg7N7XwiLNxI3z/PTz/PLz5pg3NOGXCPR+O4zjOtsfq1SYg3n3XREW2x06ebP8fcADUr1/x9tVwXHw4juM4GRFekV4zmD4dpk3LXnhEbNgA77wDbdrAHntUrG3bAD7s4jjONoOIDBeRglLS9AjpGpaxjJEi8mnZLKze7LTTTuy00065NqP8/PADvP02rFtXvnwi78mPP1aMXdsQLj4cx3GK0gO4GiiT+KjJfP/993z//fe5NqP8zJplAqQimDvX4kCcrHDx4TjOVo2INMi1DdsKEyZMYMKECbk2o3ysXGnTaQsLKya/TZvgs89sfRAnYzzmw3GqGSJyAnAV0A1YBjwDDFXVgrD/MOA14OfAWcDRQAFwp6remMhLgJuAPsB2wJvARar6WSxNITAUaACcE9KNA85R1bT+ZBEZDDwE7A1cH8qYD9ymqvck0vYKaXqHTeOAC1V1dqJO+ar6Xuy4kUBPVe2eKHN/YDhwCPAkcJaIXAScCuwGrAOmAZfE61oasfwBFlrz8a2qdhKR4cAQVc1LHDMLGKOq5yW2HwX8DdgV+Aw4T1XfSqQ5DRgC7A4sBR4HLlPVtZna7GRJQQEsW1ZiktEzZ3Lz1KnMLSigfV4ew/LzGdC1a/oDfvwRli+Hli0r2Niai3s+HKcaISLHAaOBmUB/4BqsQ30uRfIRwCzgROAx4AYR+X0sr07AFKAt8Bvgl0B94DURaZrI6zzgJ8CvgcuAfsBfMzT7cWAicDzwBnC3iBwYs6NX2L421GUwJhBeEZE6GZaR5ElMSB0LPBi27QTcA5wAnBnKmyIi7bLI90VMJIGJut7YeciWdsA/MPHxS2A9Vt8dogQicgEwEvgfcBwmpgYDd5ahPCdTVq0yAZKG0TNnMnTSJOYUFFAIzCkoYOikSYyeOTN9nitX+rTbLHHPh+NUL4YDU1V1ULRBRBYBT4nIYar6eizta6o6JPz/SuhkrxCREaq6KeS1HOirqqtDXq8DXwPns6WTBfhBVX8ZK3MPrCP8TQY236uqd4fjJmLC5SRMHADcAnwIHKeqhSHde8AM4GTg0QzKSPJA0sujqpfE7K+DeVdmhzJuyyRTVV0oIl+Fr9NUdVEZbANoAZykqv8L9rwBfA9cDFwmIo2x9r9VVf8Us3sp8KiI3KSqs+IZikgzoFminA5ltG+rYuDAgRWX2erVsGBB2t3TFixgXWK9j9UbNnDJxIk8Pn166oNq1bL1QrbfvkJMHDVqVIXkU51x8eE41QQRycOCHS9N7HoG2AAcDLwe2/5sIt0orKPtAHwHHAk8DawXkei3vhp4C+iVOHZc4vvnQFMRyYuGe0rglegfVV0vIl8GG6J4jINCneqEYQyAOYAGO8oiPp5PbhCR/YFrgX2AuP97tzLkX16WRcIDQFV/FJH/AfuFTb2BxpiojN+HJwB1sDrMSuR5ERYI65SXWrXSxnwkhUdp2zfn52SFiw/HqT40A2phcRObUdWNIrIYe5qOkwzXjx7n2mHiozVwYfgk+SjxPRnbEc1B3B6LJymJVMdGj4AtsM70NlJ7H74uJe90FHl0FZGdMQH1Pha3MhcbdhkVs6UqWZhi2wIs/gPs3IDFpaRi5xTb7sCGaeJ0ACZlaVuZOfroo6uqqCJUqCdg3jx48sm0cR+9nniCOSmGZXbMy2NUv36p82zVCn75S2jdOvV+pxguPhyn+rAUKATaxDeGIYSWwJJE+h0S36Pj5oW/S7AYhntTlFVVA9RLsTrdRHFPTbQfYE34m1wqMim4IpKPrUcDeUD/eJCsiKQ7viysAVK9zKN5im2peqE2FD03YPE636VIW2zupqouZUt7ARDzJFUJbdu2rdLyKoW8PGjSJK34GJafz9BJk1i9YcPmbQ3q1mVYfn76PJs0sbfeOhnj4sNxqgmqWiAiHwKDKOolGID9VpNPuP0p2qEPxJ74Z4fv44E9gQ9UtYzLOJYPVV0pIlOAbqp6eQlJo862GxYkS4iL6E3Cy5GGBpgg2TzfMQTvlqVHiHt9kjbWF5GuqjozlNEbaJIij6Yi8rNYzEdz4GfAXWH/FGAlsJOqji6DjTnh66/NUdWlS5ccW1IOGjeGXXdNuzZHNKsl49kutWrZCqcVFO+xreDiw3GqF8OB50TkCeBhoBNwMzAhEWwK0EdE/oqJjKOweI9zQ7Ap2HTdqcB4ERmBPXW3AQ4AVFXvq9yqbGYINsNmFDYzZgnQHpua+6KqPqeqc4JIGS4iyzABMITMPTRRfMVIEfkHFudxGamHP0rji/D3fBEZDaxS1U+Al7EhqAdF5EZsFtGlWFBvkiUh3XBsWOrPmDi6A0BVl4nIFcBfRKQDNs14HdAZC9g9N5qGXJ144403gK1cfADssgt88EHalUkHdO1a8tTaOK1bQ8eOFWjctoFPtXWcaoSqPo+54ncH/osFUD5O6umevwO6Yt6P04ArVXXzEIuqfo0FdM7Hpm+Ow6bPtsVESZWgqm8DB2JDKg8CY4HrsPtPfA2OU7FA13+xJb7hf2RAEAdnYAG7L2AzdU6meFxMJnl9gInAUzAPxQth+xJsGm8zbOrzBcBvKR7zAib0/oCtn/IfrO5Hq+pmL46q3oGdt4Ow2JTR2Cyk6WnydCqKdu2gZ0+oU9aZ3oF69WD//WGH5AioUxq1CitqlTfHcaqEdAtyOdseYS2XbyZMmECHDpU/63bkyJEADB48uNLLqnRWrIBx4+CTT8p2fK1asO++8LOfQUNfiT/J7NmzoxcRdk5OGwcfdnEcx3G2RRo3hj597P/PPrNl0jOlbl346U/h4INdeJQRFx+O4zjOtkmLFnDEEdCmDbz3HixdWvoxLVvCfvvBT34CjRpVuok1FRcfjrOVEQJPfVUjp8rpl26di62ZJk3ggAOgUyf48kuYOdPe07JypXlDatfeMj13992hSxeLGfGFxcqFiw/HcRwnI1q1apVrEyqH2rWhQwf77LOPvftl1SoTH3Xq2NBKJECcCsHFh+M4jpMRqgpU/eJmVUrTpvZxKhUXH47jOE5GvPXWW0ANFx9OleDrfDiO4ziOU6W4+HAcx3Ecp0px8eE4juM4TpXi4sNxHMdxnCrFA04dx3GcjOjfP9Urhhwne1x8OI7jOBnR1KegOhWED7s4juM4GfHpp5/y6aef5toMpwbgng/HcRwnI957z16i3L179xxbsnWycqUtnFpYCPXqQbNm2+4q7S4+HMdxHKcSWbgQvvsOPv3UXhtTWAj169trYkTsVTH16+fayqrFxYfjOI7jVAKbNsGMGfD667BggYmOOPPnw0cfQY8e0Lu3vT5mW8HFRw1FREYCPVU15/5RETkBaK+q92aQdiRlsFtE6gP3A/2AVsA1wIeZllvViEgP4ATgFlVdleExs4AxqnpeFuUUApeq6t/KYGY8n8HAQ0BrVV1UnryqChG5CJihqi/loOzXgQJVrYGvgXUy5csv4eWXYdmy9GlWroS33jKhcsgh0KBB1dmXSzzg1KkKTgD+UMllnAmcDlwM9Ab+WUXllpUewNVAwyyO6Q+US0RsY1wE/DzXRjjbJsuWwZtvliw8IjZtgvfeg2+/rXy7qgvu+XAqDRFpoKqrq6i4PYC5qvporPwqKrpyidpRVT/ItS1bA9ledyJSD9ikqhsr0awawaBBg3JtwlbDvHn2iTNz5mimTr2ZgoK55OW1Jz9/GF27DgBg/XqLCenSZduI/3DxUcMRkUOAO4DdAQUuVNU3EmlOA4aENEuBx4HLVHVt2N8GuBHoA7QD5gDPAVfGb/LBxX8Z0BjzRLQWkceAM2L7AR5W1cFZ1KEdcDP2FNsYG04ZoqqTw/5ZQMdkGZmWKyJXYx6TNlGdw/bOwNfAr1T1qbCtF3A95l0BGIe16eywvxPwDXAa0AvzxqwHngH+qKprYkMYAAuDSPpWVTvF9u0PDAcOAZ4Ezko17CIivbEhpv2BWsDnwBWqOr6E9jwauArzvqzBzuUfVXVpumNidAhDY32A+cBtqnpPyLc/MBrYTVW/jJXXAJiHDTHdKCLDsevtUOBe4KfALGx46IVsbBWRw4DXsGvjDOAYYKqIdMWuiXNF5NyQ3ZmqOjJqR+zcng/sDHQSkTnAMOA3wI7A98A9qnp7zJ6MbQ/pBwA3hDI+AH6vqp+Gff8FWqnqgYljTgEeATpG11V1oWHDbBx12y4bN8Jnn5mgiJg5czSTJg1lwwa7ZRYUzGHSpKEAmwXId99ZcOqOO1a5yVWOi4+aTVvgHuAWYBHWmT0nIh1VdQWAiFwA3AbcCQwFumBCoxHwu5BPS0yU/BH4EeiKdQgdgZMSZV4ATAPOBuoDHwGtMWFzakizMNMKiEgz4E2s47kIWAz8FhgvIt1U9RtsOOLPwEHhf4KdmZb7BNY2PweejW0/GSgAng+29ALeAMaHPOuF414Rkb0ST843AC8BvwL2Dt/nYsLlxfD3CuBoYBmwlqI8CTwI/DXUPVXbHAj8D3gH6zCXAj2xji4lIf7mGeDfwHVYfMwNwFPAUemOi/E4Jo7uwNrgbhH5UFXfBF7ARMavMREaMRDIA0bGttUDnsauvW+Ac4HRIrKPqn5SBlsfwNrsRKAQWIK1/2Tg1pDmq1j6AZhouARr+2VYW18E3ISd577ArSKSp6rXZWN7oAd2XV4JbAj5PyMie6jqJmAEMEZERFU1dtxZwCtJ4RF+C80S9e5AFfLhhx8C0KNHj6osNiUDBw7MtQlp2bgRfvgB1sZ+1QsWTGPTpnVF0m3YsJqJEy9h+vTHAZt2O2FC9Y/7GDVqVLnzcPFRs2kB9IndzOdhT199MRHSGOsEb1XVP0UHichS4FERuUlVZ6nq59hNOtr/JiZmnhWRlqq6OFbmMuD4cHON0i/EnuLeLkMdLsLEj6jq/JDfOOBT4HLgN6r6gYgsANbGy8i0XFWdISLTMLGRFB/Pxbw7t2Bel+NUtTCU8R4wI6R9NHbse6oaxZuMCx6KgcD1qrpQRKKOcFqaAM4HVPXGkuwO9szEznEkfMalSywitYDbgVGqemZs+5fAWyJyUORNKoF7VfXucNxELMD3JOBNVd0gIg8BZ4rIFTGbfgO8rKpzY/nUB25Q1X+FvMZh4uDPwMllsPUlVR2SqO9aYEGa878dcKSqFoS0rTAvyK2qekVIMy78Rv4kIrdHaUuzPVZGc2BfVV0Q0oFdX3th19HLmHflLEz4IyJdMK9SUtSD/RauTrG9yqhO4mNrIyk8Stte03HxUbOZl3gS+zz8jZ6WemPDGE+JSPxamADUAfYBZoWO4ELMm9EZ2D6WdlfMGxHxYlx4VABHYm71RSlsPLQCy3kcuD485RaISHegO1s6hQaYZ+VSoE4snmQONpzVi6Li45VE/p+H4zPl+ZJ2ikhDbKjlsixiFXYFOgF/TLTle8ByrA6liY/N9VLV9UEMxJ++H8C8Hj8HXhCRXbGhoxNS5LVZ6AXh8l9MzJTF1hLbKwWvxcQEwH6YR+OpRLqngHMw79WkDG2P+DASHoH47+9DVd0kIg8C54jIn1V1A+Y1WpSmPndQ1HsU5TWpWMptgIp4+q4sNm6E556DT2J33yee6EVBwZxiafPydqRfP6tLkybwq19B+/ZVZGgO8dkuNZsf419UNZLYkXhoHf5Ow+ISos8PYXvkvr8IczG/gHUi+2ExHfG8IhZQsbQGjk/Ytx44jxKGF8rAk9jT8Anh+8lYJxDFTrTABNltKWzZM4UtPya+r6N4W5VEae3YHPv9zi0lXZzofI+meB2akFl7llgvVZ2FtdlZYdNZWF1eTBy3XlWTeS3AYorKYmu2110yffPwd36adC1i20qzPSJVW0HR6+BBbDipn4jUAQYD/1bV9YljUdWlwRO5+QNUq5gQx6hTB7p1s1VMI/Lzh1G3btHxlLp1G5CfP2zz9512glatqsrK3OKej22bJeHvicB3KfZ/H/6eBDyfGJrZKU2ehWm2l5Ul2NP2FSn2VdjsBFWdKyJvsGX45FfAf8LTKFg8RSEWD/BsiiyWVpQtgdLacSmwCcjmGSk63+dhcSJJfkixrSyMAJ4UkQ5YEOjDsXaMqCcizROdeBssZqQstmZ73SXTR+W1wbxZcZvi+6F02zNGVWeLyMuYSFuPBbo+mG0+TvWjfXto2xa+D3fRKKg03WyXevVgzz23jZku4OJjW2cKsBLYSVVHl5CuAcUDIk9NlTAN2T71xxkP/B8wPeEmr4xyHwfuEZGfY4G3j0c7VHWliEwBuqnq5Vnakc42srRvM8Get4D/E5FbMxx6mY4Jyq7RDJVK4nlsKO4xLOg5XWfaH4jiJupiHq4pFWhrNuf/XazzHwS8H9s+CPuNvJ9IX5Lt2TICE7TNgSmq+kUZ83GqEU2bwkEHwUsvbVnro2vXAZvFRpzataFnT+jYsYqNzCEuPrZhVHWZiFwB/CU8pb6G3bA7Y+PX54aI+/HARWFmzHTME9Iji6K+wKaKnorFRywKLuNMuA3zRrwhIn/HZii0xGZ1rFXVayqw3FHA3VjcwrfYLJs4Q4DXRGQUJkyWYJ6HPlisy3MZ1imyDeB8ERkNrErE52TCMGy2ywQRuQdz8++D1fNfycSqWii26udTItIIm266AhvCOAr4u6qm8jJkRYgFeSjYN0lVZ6RItg64PMTSRDNG2mOepYqy9QvgcBE5EjtX3ySCo+M2LxKRO4EhIVD1Tey8ngNcraorM7W9DLyEeU0OZMtwVbXk1FOzeeZwdt0Vjj4aJk5Mvbw6QMOGW5ZXr+6zXCoSj/nYxlHVO7A1KQ7COt/RWNT/dLaMWV+LrTtwNRaAVx8LPs2UB7GpiX8HpmLTUzO1bwkWGDsVmwI8HrgLCwYt7Ukzq3KDG30s1pE8Gc1oie1/G+sg6oe8x2JTQGsDn2Vap5DXB8GeU0I9iq0RkUEek4HDsOGXkdi5648Jp3THjMY6710wz8QYTCQsxoRdRRF50tJ5PdZjQ1uDQ9quwEBV/bgCbf0zNpw4Cjv/x5aSfih2jZ8RyhqErSdzXSJdqbZnQ/BaPY9N6366LHlUFfXq1aNePJDBKZHatWGPPWDAAPjFL6BTJ2jZEpo3tyGZ3r1h0CA49FBo3DjX1lYttQpTSTHHcZxyICJXYevCtNfEu2uihbpUdat7jVZl2C4itbHp2v9T1WxE/eZF7SZMmECHDpW/5MfUqVMByM/Pr/SyaiIFBbBqlXlA6tWDZs1MoNREZs+eTd++fQE6p/I4+7CL4zgVhtgcZMFWjL0/KTycLYi9DPGnWMB3Z2yhv2rNZ5+Zg8/FR9nIy9u23lxbEi4+HMepSO7H1h8Zhw1JOelpjwW6LgIu0LDsuuNsC7j4cBynwlDVwzJIM5ws4n6qExVpe3BF16qIvBxna6OGjjY5juM4jlNdcfHhOI7jOE6V4sMujuM4TkYMHjw41yY4NQT3fDiO4ziOU6W4+HAcx3EyYsqUKUyZUtZV5B1nCy4+HMdxnIyYMWMGM2akWi3fcbLDxYfjOI7jOFWKiw/HcRzHcaoUFx+O4ziO41QpPtXWcRzHyQh/o61TUbj4cBzHcTLi1FNPzbUJWxWFhbB4McyeDd9/D+vXQ4MG0LkztG8PTZrk2sLc4eLDcRzHcSqYFStg2jT44gtYuBA2bdqy7/33oW1b2Htv6N4d6tfPnZ25wsWHU6MQkcOA14B8VX0vx+ZkhIgMB4aoqr9sOwWpzqmIzALGqOp5WeTTCfgGOElVR5WQrgXwIHAo0Bw4M+x6CGitqosyzaumMXHiRAAOPfTQHFtSvVmxAl57DT78sKjoiFi/3jwhP/wAq1bBfvvBtjai5eLDcZytkf7Aj5WU96VAH+AMYAHwVdjeG1haSWVuFXzzzTeAi4+SKCyEqVPTC484a9fC5MnQvDn85CdVYl61wcWH46RARGoB26nqmlzbsi0gIg1UdXWm6VX1g0o0Zw/gY1X9b2L7wkosswgiUg/YpKobq6pMp2JYtMiGWuLCY+bM0UydejMFBXPJy2tPfv4wunYdAMCaNfDRR7DLLrD99jkyOge4+HCqBSLSC7gee7oEGAdcqKqzw/5OmJv7NKAXcDqwHngG+GNJIkFELgJOBXYD1gHTgEtU9bNYmpFAT+CPwM1Ad+As4BEROT5s6wx8ClwEPErC7V9aHUKadsC9wFHAMuA+oFYG7TMYc/v3BG4ADgbmAxcDLwDDgPOA7YGngItUdZ2I7ADMBs5V1QcSeY4DUNUjw/edgb8BRwL1gXeAoao6NXHc/4Vy9wAKgHeBc1T1WxFpA9yIeQ7aAXOA54Ar4+JCRAqBy4DG2LBGa6CeiNQF/oJ5HeoDzwPFhjWSwy4ish/wZyAfaIp5K+5K1rk0gl1F/lfVWrH2b62qi2KHNBKRB4CTgE3Aw6HN1sfyaYddPz8P9f0QG2abnKwP8DVwPrAz0An4Phv7ndzz3XcmQCJmzhzNpElD2bDBLv+CgjlMmjQUYLMAmT0b5s+HTp2q2trc4eLDyTmh034DGI+JhHrAcOAVEdkr8fR3A/AS8Ctg7/B9Ltbpp2Mn4B7gW6AR8FtgiojsrqrzYunaA/8IeX0DzBORvTGB8zJwCbAj8AjQrIx1eBYTMecCi7BOfNdSmijOo8D9wG2YUHoaEzM7YWKpG3ALMBO4TVV/EJH/Ar8GNnfEItIR6AucHL43BiZiQug8YAUwBHhdRPZV1ekh3aUh/38BV2D3kJ9h4uFboCU2NPFHbFikK3AV0BHroONcgAnBszGhAdb2FwDXYqLmBOCuDNqlI/AWMAJYBewP3CUi9VX1ngyOj+iNCbCGwB8ySH8D8CrwS0wYDgfWYmIQEWkGvAmswUTrYuz6Gy8i3VT1m1heA4BZ2HW2FhOnmwl5NUuU3yGjWtUgBg4cmGsTSmThQli5csv3BQumsWnTuiJpNmxYzcSJlzB9+uObt02YAI0bV5WVZWPUqIoLb3Lx4VQHbsGeBo9T1UIAEXkPmIF1jo/G0r6nqlGnME5EegMDKUF8qOol0f8iUgfzSMwOed8WS9oc6KeqU2Lpn8Y61RMiASEiCzERkVUdROQoYD/gSFUdH9KMD/lnyt1RZyoiMzCRdCSwl6puAsaKyM+wjj6q2wisrbqp6udh268xcfBc+H4m1oHvpaqfhvxfxTrDYcBgEWmKda4jVPV3MZs2D0+E/OPt/SYmsp4VkZaqujh23DLg+GA3ItIce+q/RVWvC2leEZEdKaWTVdWnY2XWAiZhguh3mPDMCFV9W0SWAhtU9e0MDvlaVQfHbG0EXCgif1HVHzHB0RIQVZ0f7BuHedAuB34Ty2s77NooSFPWRcDVmdalMmjYsGEui98qSMZ5JIVHuu2lxYfUNFx8ODlFRBoAB2FBfnVEJNo1B1BsiCUuPl5JZPF5OL6kMvbHnqT3wTqCiN0SSRfHhUcgH/hvwvvyPDbkk20d9geWR8IDQFXXiMiLwKCS6hBjXOzYWSKyDng16sADMxL5vYq5888CLhGR2sBg4BFVje6ABwOfRcIj5L9SRMYAh4RNvTGPwIPpjAsd/4WYN6MzNgwUsSv25B/xYsLuvUL+ycerUcCx6coM5TbHhNHxmFCpE3atLem4CiApQkdhw0l7Yp6wI7GZOovCkFLEBGw2TZzXShAeAHcAIxPbOmBCq0oYNCjTy7TyqMin78rg2WcthiPiiSd6UVAwp1i6vLwd6ddvS12OOw722acqLKwe+PLqTq5pgXUUt2EdevyzJzb2HSc5w2EdRTu4IoQ4hnGYW/8cTCTkY96G5HELUmTRjkSgYegw451opnVoB/yQooxU5aYjWf/1FJ+BUaRNgifmn8DpIZDxyGBTXEQ0x2JIUtnWIvwfCbe5Jdh3EdYOL2BDJvuxZapqae3dLvxNtlEm7TMSG+66DatfPnA35k2oTNLZGtWlNSaIktfFeRS/tkusp6ouVdVZ8Q/mwXOqEbvsAnXqbPmenz+MunUbFElTt24D8vOHbf7epAm0bl1VFlYP3PPh5JqlQCFwE8WfIqP95eFoIA/oH9zgwOa1HJIUptg2D+tANhM8B3EPylIyq8M8YIcU+9ukNr1CeQjz/hyHxcu8G/dyAEuA3dPYtiT8Hwmu9qTv9E4CnlfVP0UbRGSnNGmT7R3F3+yAeY3iNqRFRLYH+mFBxHfGtp9R0nEVRPJ8RrZGdVmCeeuuSHFsciZLquuvWvHqq68CcPjhh+fYkupLhw6www4wL1wBUVBputkuADvvbIuObUu4+HBySnDtTwG6qerllVBEA+ymHh8mOQ6bdZAJU4FjReSS2NDLcVhAKZBVHd4BmojIEbGYj+2BX2RcmzKiqvNF5HksiHQf7Mk7zmRgYDwuREQaBtteCmnewoI5z8SCQVPRgOJDHZmuyf0JsBqL4YlPpS0twnA7zIu7udzQridmWG556A/cHvs+EGujT8L38cD/AdNLGVLZKpg92x0tpdGiBey1l814WR/uOl27DigiNuI0bgw9evgiY46TC4YAr4nIKOBx7GmxPTZd80VVfa4cef8v/B0pIv/A4jwuI/M1G27CBMhzInJfsOtyLFgyHq9Qah1UdayITMWCT4cFG/5ITBhVMiOAscBK4MnEvoewmTdjROQKbArtEGx20M0AqrpMRK4B/hICd5/DOv0+wBNh9dHxwEUicgEwHfOE9MjEOFVdIiJ3A0NFZDVbZruUOBIe7JoKXCYiizERcgk2w6Sy6SIiD2Ht2ROL+7k95mW7DQs4fkNE/o4F8LYMadeq6jVVYKNTxfToYSuXvvMOrEsdbwqY8OjTB7p0qTLTqg0e8+HknDCr4EAsLuNBrIO8Drs+Pyvh0Ezy/gRbM6IHFocwGOsMUsVepDr+A+xptis2pHI+NlOkkNhUyCzqcALwNjYD459YsGDaAM4KZjwmKv6jqiviO8L3Q7Gpr/dhnekmoE80zTakuwWr//5Ye4zEBF3UntdiU5GvxtYbqY8Fn2bK5Visxh+xKc7RDJjSOAULtP0X1rbjQj0qm8uxa+FpTKzdG7YBJqiwQN2p2Pon47Gpw92BZHCzU0No0AAOOgiOOcZeIrddIvIoLw+6dYNjj4Wf/hRqb4M9ca3Cwmo/zOg41QoR2Rd4Dxioqs/k2p5MEZFDsLU8Dkwxq8fZCokW35swYQIdOlT+kh8jR44EYPDgwZVeFuvWQUEBbNxoa5bXqmW9dF5e8d68GlNQYAuILVxoVdp+e2jXzmI8avIL5WbPnk3fvn0BOofg6CL4sIvjlIKI3IsN3ywCBHuy/RJbkbLaIyLtMc/NbVigqQsPp0w0qcx3wK9ZYz30jz9ab71ggfXcGzZsER9160KjRhbR2b49NGsGrVpBNV5/JC8Puna1j7MFFx+OUzrNgDuBVtjKn69iy2NX9hoSFcXZwJXAx1jwo+OUiQEDUgdNloslS2x98U8/hTlzii4Pmo6vv7a/DRuaCOne3aaZtGxpIsWp9rj4cJxSUNVTcm1DeVDV4dgCXI5TfVi+3ATHJ5/Yu+U3luEdeqtWwcyZJkZat7ZXw+61l3lEnGqNiw/HcRwnI8aOHQvA0UcfXfZMNm2Cb7+Ft94y4VAR64pv2mTDNAsXwqxZcOCB9pa2+GpfTrXCxYfjOI6TEfPnp1oENwtWrYIPPoB334Vly0pPny2bNpkXZOFC6NkT9t3Xgi6caoeLD8dxHKfyKSiAKVNMeGzYULllrVgBb7xh8SMHHWTrlzvVim1wdrHjOI5TpaxaBW++aatuVbbwiNi4Ed57z0TIihWlp3eqFBcfjuM4TuWxfr2JjqlTyxZUWh42bYL33zePy9qtZXLatoGLD8dxHCcjWrZsScuWLUtPGOerr0x4VJXHI0kkQGbMyE35Tko85sNxHMfJiGOPPTa7AxYvtuGWVasqx6BMWbvWvB9t2tgCZU7Occ+H4ziOU/GsX28eh+ryJtx588wD48Mv1QIXH47jOE5GvPDCC7zwwguZJZ4/Hz7+2JZGry58+inMnZtrKxxcfDiO4zgZsnjxYhYvXpxZ4i+/rH6zTFavhunTK2ZhM6dcuPhwHMdxKpbFi6tvgOdXX9kiZE5OcfHhOI7jVCxz5tj7WqojixfDd9/l2optHhcfjuM4TsXy7bfVd2ijsNDe/1Jd7dtG2GbFh4gcJiKFItIz17ZUBSLyuoiMybENnUKbD8ylHdkgIiNF5NMKyGe4iByQYvssEbm7vPk7WwjX2JAKyivnv5vqRNu2bWnbtm3JiVatgkWLqsagsrJ0qS337uQMX+fDcUrmOqBRBeRzNVAATEls7w/8WAH5O06lk9HbbFesqH6BpklWrDDx4e98yRkuPnKMiDRQ1dW5tqO6k6t2UtWvKjn/Dyoz/5qA/0a2MlatqlCvwuiZM7l56lTmFhTQPi+PYfn5DOjatXyZFhS45yPH1EjxISL7AX8G8oGmwFfAXar6QIrkrUVkFHA09mR6p6reGMtrODAE2A+4D+gJfAtcoarPJMo9G7gY6AIsAB4BrlbVDWH/YOAhYH9gOHAI8KSIPAK8Fmw4E/gFsCIc+4CInAVcDrQExgK/VdXlIc8GwC3AEcDOwELgVeBSVV1ShrbrBVwP9A6bxgEXqurssL8T8A3wfyHNycAm4A5VvU5EjgNuAjoCbwKDVXVeophGIvIAcFI49mFgqKquL6WdWgGtVPXAhM2nYG3dUVVni0htYBjwG2BH4HvgHlW9PXbMcDI4ryIyEuipqt1j23YMdTwKaBKOu09V/56mTaOFDv4qIn8N//dR1ddFZBYwRlXPC2n3AP4W6t4AmAM8qqrXZLj/daBAVfvFyu+EnbOTVHVUbPtpoQ12B5YCjwOXqWraVZhi7XYgcC+wDzAT+B0wFfgLcDqwEbgfGK6qhbHjDw5tty+wGngZuERV5ydsHRzqeFKwrWtUN+BJ7LroALwN/F5VpydMrSUiVwHnANth1/E5qrrZyyQiO4e2PBKoD7yDXYdT09U/kzqENDuG9jki2H8vUC+kyxOR+sBs4AFVvTyR/wjgUFWVkuzIBaNHjwZgwIAB6ROtXw/r1lVMeTNnMnTSJFaHpdnnFBQwdNIks6E8AmTTJl9sLMfUSPGBdXxvASOAVdhN7C4Rqa+q9yTSjgCeAk7EbkI3iMgSVf1HLE097IZ3N9YxX4B1hruq6iwAETkfuBPryC4C9gauAdoCZyXKfBJ4EPgrsAa78RGOfTjs+z9ghIjsgnWQFwA7AH+P2QDWAdUDrsQETwfgMkyk9MqwvQh16AW8AYwHTg35DgdeEZG9VDX+VqgbgGeBQZhoulZEGgOHB1vqhPa4HzguUdQNmED6JdbpDwfWYoIhTrKdmgNjRERUVWPpzgJeiQRSSH8R1kG8AfQFbhWRPFW9LnZcqec1RRu1xK4tgCswYbsrsEuq9IHe4Zi7sA4e4PM0aV/ABORZWKe1S8g/0/0ZISIXALdh52goJphvxIaYflfK4fWAR4E7sHa7HrsWxgDrgFOAw4CrgI+A0aHMfbHzPhk7981DmRNEZF9VXRMr4ybgFew6jN+n9gl1vix8vwG7PndLiKbzMGHya0yU34oJl98EWxoDE4FaIe0KTFS9HmxJipmo3Uqtg4jUAp4H2mPiZzF2Pe4W5aOq60TkYeAMEbkq+m2JSCPgV6FNk2U3A5olNndIZWdlsXz58tITxd7hMnBM+cJlpi1YwLpEYOjqDRu4ZOJEHp+e8hSVyqh+QZfn6l0zDlBDxYeqPh39H24Ek4DW2E01KT5eU9UoOO0VEWkHXCEiI1Q1uurrY0+EY0Ke07CO/gTgDhGpg91o/6Oqf4jltQm4WURuUNWvY2U+kPCuHBb+fSb2BPsmMAC7eXZW1ZVh+z7Y0+AFoa5LgN/H8qqLdWwfiMg+qvp+pu2GeVA+BI6LnlZF5D1gBubheDSW9h1VvTD8P15EBgAXArvEvCQdgVtSuM2/VtXBsXZqBFwoIn+JP5mmaKfamBfjLKzDRES6AH1CmxC8I+cDt6rqFeHQcaGz+ZOI3K6qkb+1xPOapo3+iInA3WMC5X9p0gKgqm+LCMB3qvp2unTB9l2Ai1U1Wkby9Uz3Z0poi+uxNvpTbPtS4FERuSmd+ArUB4ap6ovhuNWY566dqv48pBkvIsdj52V02HY58ANwjKquC8dOx0TCr4CRsTI+VdUzU5TdBvMKfBmO/xj4AvOU3B9L94Oq/jJWtz1Cmt+ETWdiDyl7qeqnIc2rwCxMBA9OU/dM6nAMJpL6qOrrIc14zEMWZwQmeI4CXgrbBmEPFA+nKPsiLHaoelOrVoVllRQepW3Pigq008meGik+RKQ59jR9PPZkUCfsSuVnezbxfRTW0XYAosngmzBvAACqulhEfmDLU8fuQCvMgxLnKcwNfSAQFx/PpzF9XKyMVSIyB/g8Eh6BGUAbEakXG6Y4HRvu2RXIi6XdDchIfIThm4OAS4E6obMEc+sr5kWJi49xFOVLYFnM+xDZWgsb+pgZ256qzS8D9sQ8FRFF2klVN4nIg8A5IvLnMJz1a2BRLO1+2JN5qnNxDuaRmhS2lXZeU9EX+F8pnXNZWYx1UDcFD8sEVf0+i/2Z0htoDDwVxGrEBOy3sg/WCadjE/b0HxGtJpW8JmYAO8W+Hww8GXXaAKr6Thh6Opii4iPdb+TTSHiE41VsNtJ+FBUfSVs+B5oG71dBKO+zSHiEvFaGmS2HpCk70zrkY7+F12Np1orIi5i3JNr2ZRhKOost4uMsbBhuQYqy76BoG4Fdq5OKpcwldetax15YuMXLUEZ6PfEEc1LEZuyYl1fuvKlbI7u/rYaaOtV2JOauvQ0bSsnHXOvbpUibXAkn+tG3i21bnWIcfB2wffi/efg7P5EmyqtFmu1JkrMe1mGu9eS2WoS6iEh/4N/ANOzGtj82DELMvkxogXU8twHrE589Mdd1WWxNZUcmbR7fHudBTOj1Cx6nwcC/IyFGdueitPOaipZApbwcInibjsQ6yruA70TkQxHpm8n+LGgd/k6j6HmOzkvyXCdJtlt0npcm0iXbsjnFzwvYucn0N5Jq5aoFFL92Ul2fUPQ3m6ktcTI5rh02NJYkle0jgGNFpLWI7I49qDyYqmBVXaqqs+IfLG6kerHddrB9Nree9AzLz6dBQiQ0qFuXYfn55cu4Xj1o2LB8eTjlosZJPxHZHuiHBXbdGdt+RppDku9XbhP+JoMkSyIK7GyT2N4msT+iIt+0dBLwkar+NtoQxqWzZSlm100U90xE+yuKTNu8WDuFgNKXsSfE9ZhXJX6zjp+LOSnKyDoIN8FibCy/UlDVGcCg4JHYH7gWeF5EdlbVxaXtp2gMUUSyM43a4ES2ePfilMWbkglLKP4bIWz7LLEt3W8k1fvQ2wDvlcGW3dPkVdI1kkkd5rFF4MVJZftoTEj+HxYfNhcLYK2WdOiQQYhJo0bQuLG9R6WcREGlFT7bpXFjyMsrPZ1TadQ48YF5BGoTG2IJguTENOn7U7SzHYjdALJ5olDsSWcQW8a3Cd8LseC0yqIBxYeTTs02k+ByngJ0S0bfVwL9gdtj3wdigcGfZHj8COycNQemqOoXsX3vYqJkEEWHnAYBK8lwGKoEXgWGhM4+mzWa15OFJyoMKU0Wkeux4ZCOmPApbf/3wFEiUjsWs3RUIvspWFvspKqjqTomAyeIyCWxIcN8oBOZDx10F5GuqjozHC9Ad4rHcmViy0AR6aaqn4e8GmIzzV4q5bjS6jAVG+I5LBbzsV3IuwhhOOZhLBalBfDPRGB3teLwww8vPVHjxvapoOXVB3TtWn6xkaRxYxNJTs6oceJDVZeJyFTgMhFZjHXMl2BPhKnoIzb9cTx2kz4ZODd2486kzI0ici02o2YhNiNhb+yp9CFV/absNSqV8cA9YlMgJ2OzTZKzSzJlCPCa2NTjx7GnvPZYQOeLqvpcua01uojIQ9hMk55YnMntiWDTkngJe7o8kMRMIlVdJCJ3YgJhLTbdtw8W73F1In6mLNyOPaW+ETr+r7CZIrvFgzdT8AXQX0QmYR2/qmqRlZhEZC9s2OupkG8eFlg7F/i8tP0hm6exjuweEXkGG3IcHC8n/EauAP4iIh2wYNF1QGfMa3huInanorgBEz4vicjfMfF4U7D9yQzzWAC8IDaNNspzNsVjIUrjISxOakxoiwLs+m8E3FzOOryMidzHReQytsx2WY/FyyQZgQUyFwL/yrIe1Y/69aFtW3uBW3WlZUsXHzmmpsZ8nIIFu/0LeyIah01jTcXvgK7Yk/RpwJWqem+2Barq3SGvI7Aph+dj0/t+X9JxFcD92FoFv8fqsAfmSciaMBPjQMxt/yA2Xfc67DpJusXLw+XYjfZp7IZ/b9iWqZ0bsYDEgpBHkqHYrIAzsHMxCBiSmGZbJsLQxoGY0PsLJoSGULqn7Nzw92XsyTjV0Nh8TEgMA14E/ol51A4P01BL24+qjsc6smOA/2LBpYNT1OMO7Ho/CAv4HY1ds9OppBVXVXUa9vtoAPwHm+b7BtBXi06zLYn3MXHwF2xtl/nAUSlid0qzZQVwKBb3ch8mHDZhM1TSzuHMpA4hNud4bCjoPuy39BbWxstS5KmYeJmolbyoXXl5+umnefrpVD+5BJ07W+xHdaRuXdg169npTgVTq7CwIsMPHKfyCVNuZ2CzTs7OtT1O1SApFlDbWgjX7PvA96p6bGJfJ8yTdbqqPp7i8JLy7QR8M2HChMziMcrJyJEjARg8eHDJCQsKYNQoe4FbdaNDBxg4EJo1y7UlNZrZs2fTt29fsKUiZiX317hhF6fmIrYq5E+x+J3O2FOn41Q7xFY7roPFgzXBpoTvhQ0BR2laYtPhr8KCo0cVz2krJS8Pune3t9tWtwfcPfZw4VENcPHhbE20xwJKFwEXxNdocJxqxmpseKwTNmz5KTBAVSfE0hyLDQ3PBE6Lrx1SI+jYEVq3rrDA0wqhRQvo0iXXVji4+HC2IoLrzpcl3EZR1cNybUOmqOojWExKSWlGkn2g7NZD69aw334wdqy97yXX1KkD+fnQJtVMaaeqcfHhOI7jZETnzp2zO2CPPWzo5eOPK8egbNh1V9hzT6hdU+dZbF24+HAcx3Ey4tBDD83ugIYNzfsxdy4sWlQ5RmVCs2bQu7cvLFaNcAnoOI7jVB477giHHpq7jr9hQzjkENhpp9LTOlWGiw/HcRwnIx577DEee+yx7A/s1g369q36hb0aNIDDDoO99vLhlmqGD7s4juM4GbG+rIGjdepsEQD/+x8sK7bWWsWTlwd9+li5/gbbaoefEcdxHKfyqVPHAj7z8mDKFPjmG9iU8VssMqdWLdh5ZzjoIJtWW6dOxZfhlBsXH47jOE7VULs27LILtGoFH34I770HK1aUeljGNGoE++wDe+9ta3o41RYXH47jOE7V0rSpeSZ23hk++cS8ID+W45VCzZrZomZ77WV/fZil2uNnyHEcx8mI3XbbreIyq1PHXkC38862CuqsWfDFFxYPUlAAGzemP7Z2bRu+adoURGx4ZYcdXHRsRfiZchzHcTLigAMOqPhM69SBdu3ss+eeJjxWrLB1QebPh7VrTYjUrm1vym3TxlZPbdzYBEjjxhVvk1PpuPhwHMdxqgd5efZp27boa+8LCy2QdCtk0ybTUfPmma6qU8fCUdq2hSZNcm1d7nDx4TiO42TEyJEjARg8eHDVFryVCo8ffoD334eZM2Hx4i0v+K1Xzxw93bvbp2HD3NqZC0pddUVEhotIQVUYs7UhIoUiMiTXdmxriMgtIjJPRDaJyMg0aQaH89Oqis1LZUuP8Dsq9RYjIoeJyJ8rsOxOoR0GVlB+1fJ+ICKzROTucuYxWEROqSibnG2befPgxRfh7bfN8xEJD7D37H33HYwbB5MmwapVubMzV2Sy5Ns/gT6VbYjjZIKIHAVcCvwFOBC4LrcWZUQP4Gogk+ebw4AKEx/APKA38L8KzLOmMhhw8eGUmxUr4LXX7J16JbFhA0ydarOOtzVKHXZR1dnA7CqwBRFpoKqrq6KsXCEi9YBNqlpCKLdTAnuEv3eqaiWsUFRxiEgdIKcrHKnqWuDtXNpQ04jOq6quy7UtTvVk7lybPRxn5szRTJ16MwUFc8nLa09+/jC6dh3Ahg3w6af2AuDmzXNjby4oVXyIyHBgiKrmhe+HAa8BR2BPCscDS4GHgOHxDkFE9gBuwJ7mtge+BG5W1SfC/kLgMqAxcCbQGqgnIrWAC4HfA52BBcD9wI2qWhiOFWA49vTbGpgF/Bv4q6puiNlwKXA2sBNQAHwC/EFVvwj76wNXAqcB7YFvgb+p6ohEO5wZ0rUDpgLnlNZ24bhZwBjga+B8YGegE/C9iJwGDAF2D234OHBZ6DAQkabALUA/oBWwCOtIfqmqG0RkcGj3/YAbgQOAH4DrVPXBhB0nAFcB3YBlwDPAUFUtCPsPI4PzKiI7Ardi3rCmwHxgrKr+PlaWADeFNNsBbwIXqepnpbRVi1Df44Am2Lm6QlVfCftfB6LXam60Yuijqq+XkG2HMDTTJ9h6m6rekyi3F3A95iEAGAdcGIR3lOZG7Dx0AZYDU4CLVfX7WJrXsWvsCexa6QpcA1wbkiwMNn+rqp1S1H845iGJfhsAE1X1sLDtYKxd9wVWAy8Dl6jq/HSVF5FOwDfASao6KmybhV2TnwLDgBbYOTo7UZ92wL3AUdg1cx9QZPA9dg22VtVFse2vAwWq2i+2rbT7Qam/+5DuF8BfsXPxOXBRuvonbD0A+530wO593wJ3qeo/4tdWrO2vUdXhac5rH2CSiByN/a56AGuA54A/qurSkFc97Nr6FdAW+BF4Hxisqj+Utj+TejnVi/XrTUzEV6KfOXM0kyYNZcMGe7YuKJjDpElDAejadQALFsCcOS4+MuV+7MfYHzgS+1HOAB4FEJFdgbcwr8kF2I2/O9b5xrkAmIYJhPph261Y534TdpPfB7uJbwrbwETAl8GG5cCeIU0LzC2PiJyO3WyuCrY0wTroeIzxk9iN5Fqss/sZcJ+IrIjdFI8B/hXq9liox3NZtNUATBxdAqwFlonIBcBtwJ3AUOxGeiPQCPhdOO424OdYB/FNqPPPKT5c9hTwAHZDPgX4p4jMVdWXg/3HAaOBUcDl2I39JmA34PBEXiWeV0zg7ciWc7oTcFB0cOjspgAK/AZYhwms10RkV1VN+VKH8DT5MnZjvwy7bs4GXhSRI1T1NeAPWMd0PluEwuep8ovxONY53gGcCtwtIh+q6puh3F7AG8D4sL8eJmpfEZG9Yh6qHYCbgblAS6zDe1NEJOGt2xc7l9dgYnE6dl1fARyNdeJr09j6T6ADdg5/FrYtD3buC7wKTAZ+CTTHrpcJIrKvqq4ppR2SHIuJ3vMx8X871k7x6+FZ7Fo5N9TlYmBXykCG94NSf/cisif223sV+53vCDwCNCul/MbAiyHfU7BzsDtb7gV/wK7xVdj1CkU9vsnz+k0Q9M9gv4nrsAeEG7Df41HhuGEh7z8Bn2HXTl+gQYb743VolqKeHUqqd0Xzk5/8pCqLA2DgwAoJV6oyNmywQNN1Mb/YggXT2LRpXSLdaiZOvITp0x8H4PXXt07xMWrUqDIdVx7x8ayqXhH+f1VEjgQGsqWTGo51PAeo6vIoXYp8lgHHx56su2BPP+ep6n2x/GsDl4nInaq6Mjztvh6OqYXdlGsD14jI0PCktB/wsareFCvv+eif8LTfH/h51FGHslpgN5MnwrargCmqenr4PlZENmE3y0zYDjgy5mVojD3t3Kqqf4rZsxR4VERuUtVZwf7HVfXhWF5Ppcj/EVW9Mfz/ioh0DTZHdRoOTFXVQbGyFgFPichhCc9Baed1P8w7E7fj0dj/w7EOs2/UKYcnx8jzc30K+wF+AfQCfqGqL4XjxgIfY96A11T1cxH5DkBVMx1KuFdV7w75TcS8FydhT/pgnpYPgeNiXrX3MMF1clQ3Vf1NlGEQSq9jXqZjMGEX0RLYX1W/jaX/Kvw7Le4dSKKqs0VkNjYsl6zf5VF5kbtfRKZjnrBfASNLb4oi1AH6RaJFRHYAbheRZqq6NMTW7Iddt+NDmvGYt6AsDKeE+0Gmv3ssHmYOcGzk4QzX8jOllC9Yxz1MVT8J2yZEO8O1tRzz1qS6toqc13DPmQSMUtUzY/X4EnhLRA5S1clYG45T1XtjecWvl9L2x7mI4BnLFfn5+bksfqugsLBocClQTHik2p48pqZTHvHxSuL759iTdERf7Ie5nJJ5MTF2fzjm2v2PiMTtexV7+hHgfRHZHntCPhV7+q4fS9sGe7J6H/iDiNyOPcW9nRinPRJzc45PUdbvgghZBvTEvBNxRpG5+HgtEh6B3tjT5lOJcidgncI+mKfkfWCwiMzH2vuTuPs5xrOJ788AN4dOsgHmEr40RZoNwMEEERco7by+DwwRkY3AeFX9MpH+SOBpYH2sbquxp95eKWyPOBhYEQkPAFXdJCL/AS4XkTpljJPZXB9VXR86hw5gMUaY1+ZSoE4YEgHr3DTYG3nyjsG8Fz/Bhpsikks+fhwXHhXIwcCT8etXVd8JQygHk734mJjwlkQepA7YcNv+wPJIeITy1ojIi8Agsqe0+0FGv/tg13/jQ6vAf7FruSS+wkTxfSJyF/abzGZYI3led8WGT/+YsPe9UE4v7IHofeBSEbkG87xMS1zHpe2PcwfFz3MHTARVCdFbbevVq1dVRZb5yTpXFBTA00/bbJaIJ57oRUHBnGJp8/J2pF8/q1/fvnDwwVVlZe7JZLZLOpIL8a/DxnEjWmIu6tJYkPjeGrsJLQTWxz5Tw/7ITfsXTBA8iD3N9sKeDonZMRJ7WjgCmAgsEpF7RKRRrKzmiXLWA/+JldUaE2nJG1XS7mzrCDbcFC83KiOq4/mYS/di4CMsTuTCFPmnsq0e5gZuhrVnkbiAcINbjA1TxSntvP4S6xCuA2aIyJci8qtE3S6keJseS/EhtzjNSd2mUV3ySji2JEqqTwtM7N2Wwt49I3tFJB/zmC0AzsDEYz7F2yaytzJoTuIcxsoryxu0UrULbKlPO4pfV1F5ZaG0+0Gmv/tidoVrOa1HKaT5EbsPLAceBuaLyCQR2TtD+9P9hkdT/NppErP3Bmy47jTgHWCBiFwfHgwy2R+vw1JVnRX/UEWTASIee+wxHnvssaoscqsjL6/o+mgA+fnDqFu36Eha3boNyM8fBtj78Dp2rCoLqweVucjYYiyAszSST/JLwraD2HJDjBO5sE8C7o8PqYjIIfGEwaNyJ3BnCJ4biMVFrMDGWpdgN61j0tg2Axsb3oCN+cdpU2KtipKqjgAnAt9RnO+D/csw4XGxiHTDxobvsDADHRtLvwP2tB63bT1Wtwah/CL2hptby5gtGaGq84CzROQ3mIfmT8BjIvKxqn4e8nsRC1RMUtJMpiVJGxN1qYy1JZZibXMTxb1H0X6wobnlWNDmRgARaUlRb1tEZTlPS2qfEgN5y8g8il/zUXlxIu9Jsi1aUPSclXY/yPR3X8yu2LVcIqr6LvDz4DU9FOv0XxSRDlr6zKl0v+HzMNGQ5IdQ5losTuQaEekMnI4NQc0G/lHa/tLq5FRPunSxF/YuCxFuXbsOAEg52wVgp51s1fhticoUH68CA0XkT6qazTuTo3HY1qr63xLSNSAWuBfGYNPO0Q+d5l0ichI24wMsyHAosEFVP0x3rIhMw4TL7bHN5YmCmgKsBHZS1XTju0UIY9IXYQF53YC4+OgPfBD7fiJb3LcFIvIh5iq/LZZmAHb+y+SyDcM/00RkGCYEd8dc9+Mxr8EHWQ6TTMbcz0dHwiqc04FYvE2FT01W1ZUiMgXopqqXl5C0ASaA4h3UqVkUlfQqlJZ2OxGplRhimwycICKXqOp62OyR6UTluN3fAZqEYN8o5mN7LDYnTjQ7phvBsyEiHbFhkriwLu1+kOnv/h3g+NAO0VDL8Zh3LCPCcNMrYjO3HsS8g0tI7clKx3Ss7l01MXuqhHK/Aa4Nwr1btvudrYc2bSA/HyZO3DLrpWvXAZvFRpyWLWG//ey1NdsSlSk+rsGGQ94Ukb9gTyzdgIaqeku6g1R1hojcCfxbRG7FYgXqALsA/VU1isYfj8VlTMdcor/FXNObEZH7safXt7Gby37YmPEloaxXReQ54GUR+SsWeNgA60h7qeovQ1bXAWNE5BEsBqA7GU61TVPHZSJyBfAXEemATXFdh80s6AecG4IP38Qi+z/FOr+TsQ7w9USWp4vIGmwY5xRsRk+8kxgOPCciT2Au507YU98ELXmaahHEpv6Ow2YXKHZezsE8SdHT31WYq3y8iIzAznubYJPGggmTvAi8CzwitsJnNNtlD4rPyKlIopk4o7CZMUuwJ/Q+WDzSc9i1dhFwj4g8gw25/JbUT+ip+CL8PV9ERgOrYkGPqdLWBS4SkclY3IVi7vkpwEsi8nfsWr8JE3xPZl7dzFDVsSIyFQuAHoYNh/wRuw7jvIMFod4hIpdhv5/LKO5RK/F+kMXv/iYsruJ5sRVNd8RicUqMLRObnvsbzMP1HTZscgnwvqpGtn6BxVgdhwmpuaqacqhIVQvDw8BTYRh3DPY72Bmb6fL3EJPzHBbX8UHYfzQWo/ZqsKvE/c7WSd260LOn/f/22xYHkqRWLXu/y89+Zi/33dYoT8xHiYRAxAOwKaL3Ai8AZ5FZtPzFmDt/QDjuCWztiTdiac7DVm28A4vtmIVNC40zJdjwT8xTcCYW7X5XLM0g4C6soxuLdc79iXXwqvoiduM6CAtuGxDSlBlVvQMb5z0IC14djcV4TGfLePybIc3TWIDoHtjMoPcT2Z2MddDPYWsonJ0I3Hwe84bsHuy/Futos63DGiz25NyQz+PYqp1HqeqcUNbXWPzNfGzIaxw21NWWLeP3xQiejWOw2IpoGKQDNiPj9SztzJgws+FAbNjgQewauA77bXwW0ryEeciOxa7Ho7C1SJIdcboyPsAE4CnYNflCCclfwH4vf8I69vtDHtOwmIUGWEzSndjvoa9mP802U07AhPs92G9oEtZGmwnehxOwjvMpTGRchYnTeLpM7gel/u5V9aOwvzN2jZyLxeEkY1iSzMSGT6/DgpD/jon142JpbsE8TA9j1+rZJWUYvJZHYQLpMUyADMOGmGaFZJMx0fVI2H8UcEb4TWay39lK2X5782gMGgQHHWTvcmne3DwdnTvDL34BAwYUjw/ZVqhVuK3N76lBSJoFnhzH2TaQsIjchAkT6NCh8pf8+DCsA96jR49KL6smUVgIS5fa2h+1almAaaNGpR62VTN79mz69u0L0DkERxfB32rrOI7jZISLjrJRq9bWuYBYZVJpwy6O4zhOzWLVqlWs2hZfwepUOO752IpR1ZFkv7iU4zhOmXj66acBGDx4cG4NcbZ63PPhOI7jOE6V4uLDcRzHcZwqxcWH4ziO4zhViosPx3Ecx3GqFA84dRzHcTKiZ7Rsp+OUExcfjuM4TkZ079491yY4NQQXH47jOE5GLAuvaW3atGmOLcmODRtgxQpYtQo2boTate1Fbo0b2zLoTtXj4sNxHMfJiGeffRbYetb5WLYMZs+Gzz6DJUvsBW8bNpj4aNjQxIcIdOpkb6KtVSvXFm87uPhwHMdxahTr18PXX8OUKSY+Nm4snmbVKli0CGbNspe99ewJ3btDXl6Vm7tN4uLDcRzHqTGsWQNTp8Jbb5nAKI3CQhMh48fDnDlw2GEmRpzKxcWH4ziOUyNYvx7efRcmTbL/s2HjRvjkE/t7xBH+IrjKxtf5cBzHcWoEX30Fb7+dvfCI88UX8P77FhviVB4uPqoJInKYiBSKyDYxkV5EXheRMaWkGSkin1aVTSUhIieIyB8qML8eIjJcRBpmccwsEbm7omyoqYR2LaigvLap32Vp9O7dm969e+fajJQsXWoxHuV96W5hoYmPb7+tELOcNLj4cJzMOAGoMPEB9ACuBjIWH0B/4G8VaIPjZIWIICK5NiMl339vMRsVwcqV8Omn7v2oTFx8OEUQkQa5tqE6UR3aI7JBVT9Q1Vk5NqfKEZFaIuKrMVQDFi1axKJFi3JtRjHWrTOxkGpWS1mZNcsCUZ3KwQNOqwAR2Q/4M5APNAW+Au5S1QdSJG8tIqOAo4EC4E5VvTGW13BgCLAfcB/QE/gWuEJVn0mUezZwMdAFWAA8AlytqhvC/sHAQ8D+wHDgEOBJEXkEeC3YcCbwC2BFOPYBETkLuBxoCYwFfquqy0OeDYBbgCOAnYGFwKvApaq6JPvWAxE5BLgD2B1Q4EJVfSOR5rTQLrsDS4HHgctUdW3Y3wa4EegDtAPmAM8BV6rq6lg+hcBlQONQ99Yi8hhwRmw/wMOqOjiNvcdg7d4DaABMB65X1f+G/YOxdgdYGJ4kv1XVTunOCXCWiMwCxqjqeSGf14ECVe0XK7sT8A1wkqqOCttmAWOAGcAl2Hl7Efg1sBN2HfXCrstzVPXNVPVKtNGfQhv9FvPePAJcgF2XdwA/AT4GzlTV6bFjtweuB04GWgFfAjer6qOxNCOx6/qPwM1A91D/77Hr8ufAWaT5jcTy+Qnl/I2kqX+pdYjlfRnQFngXOAf4DPst/E1ELgZuANpGv59wXIdg7+mq+ng6O3LBmDE2Ulrd1vlYsQJ+/LH0dDNnjmbq1JspKJhLXl578vOH0bXrgJRply+3fNu2rWBjHcA9H1VFR+At7EbdD3gCuEtEzk2RdgQwCzgReAy4QUR+n0hTD+uQHsOGA77CREOnKIGInA/cj92sjwPuxTrn+1OU+STwJnAs8GBs+33AF8AAYAIwQkRuBk7DOpqLsY7g+tgxDYJ9V2Kdw2VYxzY2RbmZ0Ba4B7gda5N1wHMi0jhW1wuAkcD/Ql2HA4OBO2P5tMREyR+DXTcBJwH/TlHmBcBewNnAL4HrgJeAr4He4XNdCTZ3wjr307Ghkv8Bz4rIL8L+F9nSZkeH/Pon8kh3TsrK8ZiIPAe7DqJr4ulQ1omYwBydoZfhfKzDPgMTdedgouMBrN1/CTQHnkoc9xhwLnY+jwPeBh4RkdMT6doD/wDuBo4Cpsb25eI3klUdROS4kM/rofz/As8m8nkYuwefnNh+JrAMGF2KHU5g1SpbQKwkZs4czaRJQykomAMUUlAwh0mThjJzZupm3rgRFi+ueFsdwz0fVYCqPh39LyK1gElAa+B3WMca5zVVHRL+f0VE2gFXiMgIVd0UttfHnurHhDynYU9tJwB3iEgd4CrgP6r6h1hem4CbReQGVf06VuYDCe/KYeHfZ1T1mrDtTUyE/BrorKorw/Z9sE78glDXJcDvY3nVBT4HPhCRfVT1/UzbLdAC6KOqn4T85gEfAH3ZIkKuB25V1T/Fyl0KPCoiN6nqLFX9HHvqj/a/CSzCREFLVY3fZpYBx8faGxFZCHRU1bdLM1hV74sdVxvrgHbFzveLqrpQRL4KSaapairn7gOpnubLQS3guJgnaH9MOPwyuj5FZBHWyR8KvFJKfvNV9dTw/ysicjRwHrCfqr4b8tse+I+IiKqqiOyFXUPnquq9sWPbY2LukVj+zYF+qjol2iAi0TNoLn4jkQ2Z1uEKYKKqnpnI+/YoL1VdIiLPYF6c+0P+tbDf2GOquiZRdjOgWcKkDkkbtwYGDhxYofmtWQMLFliwaDoWLJjGpk3rimzbsGE1EydewvTpqR1MkydDs2YVaGhg1KhRFZ/pVoZ7PqoAEWkuIn8P7u/14XMxsFuK5Mmno1HAjhS9yWwCxkdfQsf5QyzN7pg7OPnU+RTWCR2Y2P58GtPHxcpYhQ1VTImER2AG0EZE6kUbROR0EXlfRFZgdf0g7EpV39KYFwmPwOfhb1TX3pj7/ykRqRt9ME9NHWCfYFMtEblIRD4XkdXBruew9tg1UeaLceGRLSLSIczUmc2W892f7Oqf7pyUlYmR8AjMCH/Hpdi2Uwb5jUt8nwEsi4RHmvwODn9TXZcdRSRe7uK48EiQi99IRKl1CMJmb+z6ivMMxRkB5IvInuF7X8xz9s8UaS/ChtTin0lp7HQSJIVHadudysU9H1XDSOxmdi3wKbAce+o8L0XaHxLfF4S/7YDvwv+rEx0J2HBE5C6PlseZnyavFmm2J0mOoq7Dhi6S22oB2wHrRaQ/NpTxT+zpbzH2tDY2Zl82FLFBVdeFGIkor9bh77Q0x+8c/l4E3Ar8FRsG+RHohsVXJO1K1x6lEjwdz2Pn4BosHqAAG37aN4usymxDGlKdS1R1aXIbmZ2nTK+NeH7NgQ0JLxMUvS6/T2xLRS5+IxGZ1GE9dm9dWIrdqOpEEVHM+3ER8BvMG/ZRirLvwO4lcTqwFQqQin7ynz8fnnqq5LiPJ57oFYZcipKXtyP9+qW2Z+BAW3LdqXhcfFQywfXcD7hEVe+MbT8jzSE7JL63CX/nZVFsFNjZJrG9TWJ/RAnOyqw5CfhIVX8bbRCRbDrdbInqciJbOp44UWd2EvB8Ymgm3RN+edqjK/bU219Vn4uVVT/LfDKxYQ02vBAnXadZHVgC1BWRFong41TXZUn1z8VvJH5caXVYCGxgizCOSNod8QBwmYj8BRsWujhVoiAUl8a3SRVPez3kkEOqtLxMycuzl8SVJD7y84cxadJQNmzYHF9O3boNyM8fljJ9w4awlb28d6vCh10qn+2wdt78FBYEyYlp0icDDwcCc4HZWZSp2A1wUGL7IOymPjmLvLKlAbG6Bk5NlbCCmAKsBHZS1fdSfKIn0vLaFX9qLoloam78fLfB3OnJ/Mgwz3R8D+wevC0RR5Ujv8omuu5SXZffqur3ZEYufyOl1kFVN2JDjSck0qT7zT8M5GEBspuwmVrVki5dutClS5dcm1GMvDzo2rXkNF27DuDgg28hL29HoBZ5eTty8MG3pJ3t0qaNL7Fembjno5JR1WUiMhV7slmMdUqXYE+tqegjIn/FxquPwiLhz80mBkFVN4rItdiMmoXAC9jT+LXAQ6r6TdlrVCrjgXvEpgRPBg7HZgRUCqF9rwD+EqYovoZ17J0xj9O5qjo72HVRmBkzHfOE9MiiqC+w6Z6nYh3XojRrbkzHOsFbQuxJAyywcT5Fxf4X4e/5IjIaWJWIbcmEpzE3/T0hcDEfm+VTLVHVj4Odt4mt7PoZJhx+DvxfFlnl7DeSRR2uB/4rIg9hoqIbNiMITGDE81wkIs8CvwIeUdVlmdajqpk/30ap2lbD+ae77GIrky5dmj5N164D0oqNOLVr+xtuKxv3fFQNp2DBd//CZreMw6axpuJ3mOv+WWxK65WxqPqMUdW7Q15HYGs8nI/FPCSnJFY092OrcP4eq8Me2M250lDVO7C2OggLPhyN1Xc6W2ITriWs4YAFB9bHptJmyoNYZ/93bEbI8DS2rMWezFeFcm7Epp6+lEj3QcjjFMx780IWtkR5jMemDh+DTeXsTTUWH4HTsGt/CFbnA7D1LB4p8aii5Po3UmodVPX5kHcf7NychAlFsNlUSaL5nhUxrbrSGDt2LGPHlnXWfIasWmUKYskS+yxbZtNZSqFtW9h7bxMO5aVLF9g1GYbuVCi1Ckuam+Q4jlNNCFPAXwPyVfW9HJuTNSJyIiaOe6rqtMS+fwEHqmpWQRxh3ZJvJkyYQIcOlT/rduTIkUAFLjK2aZMtplFQYH/nzbNlRdets30AdepA/frQujW0awctWliAR4sWUKtWkeyWLYMXX4QZM1KUlSHNm8Nxx0HnzuWol8Ps2bPp27cv2NIMs5L7fdjFcRynghGRFpiX7TVsdlsPbFXgN+LCI0yx/SnmTbmw6i3NEatX2xSVL76Ab74xT0dpr6KdNcv+1q9vCqFrV9htN3N5bLcdYAGiffvaAmFffZU+q3Q0bw5HHAGdOmV/rJMdLj4cx3EqnvXYCrAnY9NzF2NDK0MT6V7AZsU8iq35UbNZvRpmzoSPPrK3wK1eXfoxSdatsxXFFiyADz+EnXaCHj1srKR+fdq0gWOOgXffhY8/zmjEhtq1zdNx4IH2N+FQcSoBFx+O42wVqOrr2Joy1R5VXYEtjV9auk6Vb001Yd48UwSffWYCoiJYuRKmTzevyE9/Cj17QuvWtGoFhx9ueuTjj2H2bHtPSzLKYPvtbVZL9+7mRPGptVWHiw/HcRwnI8IYfnasWWMCYcoU+KHYOmsVw5o18M478P335r7YdVfq16/P7rubJ2PRIntR3MKFlrROHWjVypZOj0JInKrFxYfjOI6TETvtlMnK+zFWrYK33jJhUFHejpKYOxdeeAEOOgjy82G77dhuO9hxR/vssUflm+BkhosPx3EcJyO+/97WgctIhKxcaW9me/ddiwCtKtasgYkTYcMG6N17czCqU73wdT4cx3GcjJgwYQITJkwoPeHq1ebxqGrhEbF+Pbz5JkydWjUeFydrXHw4juM4FUdhIXz6qQ215EJ4RKxfb54X1dzZ4KTFxYfjOI5TccyZA2+/Xfq6HVXBmjXmAVlQ0S+JdsqLiw/HcRynYli50oTH4sW5tmQL8+fb8Mva5HslnVzi4sNxHMepGL74wqbVVjc++QS+/DLXVjgxfLaL4ziOkxFHH310+p3LltnKpRs2VJ1BmbJ2ra2Gussu0KBBrq1xcPHhOI7jZEjbtm3T75w3z4Y4qitz5ph9/sa4aoEPuziO4zgZ8fXXX/P1118X37Func1wqQ5BpulYvdqGhXI5A8fZjHs+HMdxnIx44403AOjSpUvRHT/8AN99lwOLsuTrry0Ydocdcm3JNo97PhzHcZzy8eOP9vKU6s6PP1psipNzXHxso4jIYSJSKCI9c21LVSAir4vImBzb0Cm0+cBc2pEtweYhOSx/uIgU5Kr8qkJELhKRn+fajjJRnWM94mzcWL2mAW/DuPhwHMepHlwEbH3iY/36rWsRrzlzbBVWJ6e4+HByioj4vLcM8HZyqi3Ll1f4kMvomTPp9cQTdHjgAXo98QSjZ86suMx//BEKarwjrdrjAac1EBHZD/gzkA80Bb4C7lLVB1Ikby0io4CjgQLgTlW9MZbXcGAIsB9wH9AT+Ba4QlWfSZR7NnAx0AVYADwCXK2qG8L+wcBDwP7AcOAQ4EkReQR4LdhwJvALYEU49gEROQu4HGgJjAV+q6rLQ54NgFuAI4CdgYXAq8ClqrqkDG3XC7ge6B02jQMuVNXZYX8n4Bvg/0Kak4FNwB2qep2IHAfcBHQE3gQGq+q8RDGNROQB4KRw7MPAUFVdX0o7tQJaqeqBCZtPwdq6o6rOFpHawDDgN8COwPfAPap6ewb1PxO4EmgHTAXOSZPuaOAqoAewBngO+KOqLhWRhsAPwPWqenPiuIeBXqq6R/heP5R3GtAeu7b+pqojSrFzZ+BvwJFAfeAdrA2nxtLMAsZg5+tioAUwHvh9dE7Kez4zsV9ERmK/mz8AdwC7A4pdV2/EbO0InCsi54ZDz1TVkSW1Q1XTr1+/4hvXr6/Q1UNHz5zJ0EmTWB3WC5lTUMDQSZMAGNC1a/kLWLvWXzZXDXDxUTPpCLwFjABWYZ3YXSJSX1XvSaQdATwFnIjdyG8QkSWq+o9YmnrAk8DdWMd8AdYZ7qqqswBE5HzgTkygXATsDVwDtAXOSpT5JPAg8Fes46oftt+HdcQPYp3BCBHZBRM+FwA7AH+P2QDQINh3JSZ4OgCXYSKlV4btRahDL+ANrIM6NeQ7HHhFRPZS1fgcvRuAZ4FBmGi6VkQaA4cHW+qE9rgfOC5R1A2YQPol1ikNB9ZigiFOsp2aA2NERFSLvC3rLOCVSCCF9BdhneYbQF/gVhHJU9XrSqj/McC/gEeBx4DumKhIpjsBeAb4N3Ad0CrU6SngKFVdJSLPYx35zbHjtgdOwERDvI59gGuBT4CfAfeJyApVfSKNnY2BiUAt4DxMqA4BXheRfVU1vsTmCZj4OhdohgnVZ7HfRJyyns9M7W8L3BPKX4Sd8+dEpKOqrgD6Ay8Bk4FbwzFfJerdLNQhTodUbVRZtGrVqvjGTZvsE2PgmLKHV01bsIB1ifxWb9jAJRMn8ngZV08dFRdNmzb5dNtqgIuPGoiqPh39LyK1gElAa+B32A0wzmuqGgUTviIi7YArRGSEqkZ3gPrAZao6JuQ5DevoTwDuEJE62FPwf1T1D7G8NgE3i8gNqhpfHOCBhHflsPDvM6p6Tdj2JjAA+DXQWVVXhu37YB6DC0JdlwC/j+VVF/gc+EBE9lHV9zNtN6xj+BA4TlULQ37vATOwjvTRWNp3VPXC8P94ERkAXAjsEvOSdARuEZEGqro6duzXqjo41k6NgAtF5C+q+mMJ7VQb60jPAoaGbV2wzu+k8L0VcD5wq6peEQ4dFzrSP4nI7aqazud8FTBFVU8P38eGcxh1htH1dDswSlXPjG3/EnhLRA5S1cnA48ALItJNVT8PyX4BNAn7ovPeH/i5qr4c0rwqIi0wUZNSfGDesY7AXqr6acjrVWAWJuAGx9I2AY5W1WUh3TyszY9S1Vdi6bI+n1na3wLoo6qfxOz4ABOGz6nqByKyFligqm+nqfdFwNVp9lUJkeYVkUorIyk8StueNR7vUS1w8VEDEZHm2JPV8diTUZ2wK5Vv9NnE91FYR9sBiCbub8K8AQCo6mIR+YEtT127Y0+/TyXyegr4C3AgEBcfz6cxfVysjFUiMgf4PBIegRlAGxGpFxumOB1zq+8K5MXS7gZkJD7C8M1BwKVAndjNdQ7mIu9FUfExjqJ8CSyLeR8iW2thQx/xQetUbX4ZsCfmqYgo0k6quklEHgTOEZE/h+GsX2NP0lHa/TCPTapzcQ7mkZqU2EcQkD0JoiZh262x77sCnYA/BqEX8R6wHGunycArwGLsWroypDkZeFdVoyf6I4Efsc4+nterwO9EpEWaobODgc8i4QGgqivDbKZDEmlfi4RHSDdORJZj7RQXH2U5n9nYPy8SHoFIkGXjubgDGJnY1oEU57OyeOutt4CE+KhVyz4xRqUansmQXk88wZwUMRk75uWVK9/N1K5tHyen+BmomYzEhg1uw26Q+diQyXYp0v6Q+B6FrbeLbVutqknhsg7YPvzfPPxNzreL8mqRZnuSHxPf1wFLU2yrRaiLiPTH3P/TsGGM/TG3OTH7MqEFJtJuA9YnPnti8SRlsTWVHZm0eXx7nAcxodcvCIbBwL8jIUb25yKiNfYwks62eDqA0RRvpyaEdgr2jAJ+BZuHSn5B8HrE8mqeIp//hP3JNo9onqJ+ka3J+iXrE21LtnVZzmc29hfJX1XTXRtpUdWlqjor/gFml3ZcpVO3LtSvX3q6DBmWn0+DukWfixvUrcuw/PyKKaBePbPZySl+BmoYYVy9H3CJqt4Z235GmkOSS/21CX+TQZIlET3dtUlsb5PYH1GRfs+TgI9U9bfRBhHZtwz5LMXsuoninolof0WRaZsXa6cQUPoyNvSyHnsKfzCWJH4u5qQoI10Q7kJgQwm2JfM/DwvyTBLv7J/APAD5wB7Y8F3cI7ME89ock8amGWm2L8G8bUnaULx+qZay3IHsru90lNX+mkVenn0WLaqQ7KKg0punTmVuQQHt8/IYlp9fMcGmAE2bQuPGFZOXU2ZcfNQ8tsM8Wps9FUGQnJgmfX+KdrYDgblk90SlWOc1CHsijhiEdaCTs8grWxpQfDjp1GwzCW77KUA3Vb28QixLT38sbiJiIBYY/Enq5MUYgZ2z5liMxhexfe9iomQQRYecBgErSTMMpaobQyzPwBS2xZmOxZ10TRG8nOQN7Do6GRMfr6lq3GMxHhvm2aCqH5aSV5zJwMB4PEmYYfMLLGgzTh8RaRqL+TgS89CkEk7ZUlb70xH3Jm49NGgArVvDrFkVluWArl0rTmwkadcO6tQpPZ1Tqbj4qGGo6jIRmQpcJiKLsY75Emy2RCr6iMhfsRvpUVhHcW4s2DSTMjeKyLXYjJqFwAtYbMG1wEOq+k3Za1Qq44F7xKYET8ZmJyRnl2TKEOA1sanHj2NPtu2xgM4XVfW5cltrdBGRh7CZEj2xOJPbE8GmJfES9uR+IImZRKq6SETuBIaEAMY3g/3nYFOXVyYzi3EdNpvmESy+pTuJqbaqWigiFwFPhUDZMdhsk52x6+fvqvpOLO2T2MylFljAczyvV0XkOeDlcA1+iInJ3bHpuL9MY+dDWIzPGBG5ApsiPgRoRGx2TWB5yP9mTKzdAkxNBJuWiXLYn44vgMODQFoCfKOqW8dynO2So1jVlFq1oE3SmefkAo/5qJmcgrl8/4XNbhmHTWNNxe+ArtiT9GnAlap6b7YFqurdIa8jsA7pfCxQ8fclHVcB3I9N3fw9Voc9KP60nhFhlsGB2PDAg9h03euw38lnFWFs4HLMI/Q01mneG7ZlaudGLMC0IOSRZCg2K+IM7FwMAoaUNM025PsitjbIQcB/sdlG/VOkG40JjV2wKbljsFkmi7EZJ3Eex4Y5NmLTc5MMAu4Czsba++FQ5usl2LkCOBSL87kPE3GbsNkkybmYz4XPfcA/MM/QCenyLgNZ218Cf8aCvEdha6wcWzEmVhz9+/enf/9ilwS0bAnbbwVOm8aNfcilmlCr0KcdOc5WRZhyOwP4n6qenWt7qisSFhlT1fNybUtlIWGRtAkTJtChQ5Uu+VGUVatg1Ch7a2x15ic/geOPr9AAWSc1s2fPpm/fvmBLJcxK7vdhF8fZShBbTfOnWPxOZ2zRK8epMj791GY3d+/eveiOhg1hzz0t7qOi1uOoaOrWhe7dXXhUE1x8OM7WQ3ts2GARcEF8nQvHqQree+89IIX4AOjQAVq1gh9SzW6uBrRtC+3b59oKJ+Diw3G2EoLrslZp6RxDVTvl2oZtilatYPfdYeHC6reKaO3a5vVo2jTXljgBFx+O4zhO+alVC3r0sKGX774rLXX2eTdqZMGi9erZ98JCe0lcQYHFnJQkeHbZBbp1q1ibnHLh4sNxHMepGFq0gAMPtAXHVq0qX161a8MOO1ie221nb6L94QdYtgw2bLAYjqZNoUsX+3/tWis36Xlp2hQOOACaNCmfPU6F4uLDcRzHqTi6dIF99oE33yzb8EudOrDzziYaFi+GyZNh7lz48cfUb6OtW9cESocO5nlp1w6WLIHZs81Dkp8PHTuWu1pOxeLiw3Ecx8mIQYMGlZ6oXj3Yd18TAJ9/Xnr6OM2amVCYMwcmTIDvvy9dwGzYYB6RH36Ajz6y4w84wGbfNGoEP/2pv0iuGuLiw3Ecx8mIhg0bZpaweXP42c9MGMzI8BU3nTpZTMcbb5iIKMuU3Y0bba2R776DI4+E/v3tvTNOtcPloOM4jpMRH374IR9++GFmiVu1MgHQrZsNf5TELruYd+LJJ+GDD8q3VkitWlb24sXw8svwxRfVd+2RbRj3fDiO4zgZEQmPHj16ZHZAq1ZwxBEWv/Hhh7B6dfE0nTubOBg92oJJy0P9+hbz0b69BakuWQKvhNf47LFH6SLIqTLc8+E4juNUHtEQzHHHWVBonBYtLC7j+efLLzyaNbN1Rjp2NOERsWwZvPqqxY841Qb3fDiO4ziVS7165nlo3Rrefx9UTRTsvLMFli4ux8t7GzWyfNu0gQYNUqdZsgSmTDGx4zEg1QIXH47jOE7V0KoVHH64zUBZssSm437xRfb51Kljwalt2tj6HQ0blj6k8uWX8MknsP/+PvxSDXDx4TiO41QdtWubaNi40QRIt262INny5bBmjS0WtmnTlim2tWqZ2Khf3zwbjRubt6NRo+xeErdxo8Wd7LqriSAnp7j4cBzHcTLi1FNPrbjMvv7aREeLFvYpLLRVTNets//j4qN2bRMa0dLqZWXhQlt8zMVHzvGAU2ebQkQKRWRIru2obojI6yIyJsc2dArnZ2Au7SgrIjJcRApybUdlUq9ePerVq1f+jJYuLT7cUquWBYo2bmxDKU2b2qdJE4vTqF+//MMlmzbZ0Et5l353yo2LD8dxnIrhn0CfXBtRmUydOpWpU6eWP6Ply8sXZFoeFi2y8p2c4uLDcSoBEakjIlkMSDvlQUTSTHOoOlR1tqpWQM9cffnss8/47LPPyp/R0qUW35ELCgpgxYrclO1sxmM+nBqLiJwJXAm0A6YC56RJdzRwFdADWAM8B/xRVZfG0jQDrgP6A62BucCTqnpZ2P86UAA8Ecrsij0FTyot/9Bx3gIcAewMLAReBS5V1SUxG/qFfLoBG4GvgOtVdXQszWnAEGB3YCnwOHCZqq6NpdkfuAvYE/gGuLzklizSVr2A64HeYdM44EJVnR32dwp5/l9IczKwCbhDVa8TkeOAm4COwJvAYFWdlyimkYg8AJwUjn0YGKqq60MZg4GHgP2B4cAhwJMi0gpopaoHJmw+BXgE6Kiqs0WkNjAM+A2wI/A9cI+q3l5K3fcA/hbKbQDMAR5V1WvC/uHAEFXNC99nhXommaiqh4U0TUJ7ngi0AhS4SlWfK8mWrZ55yVOeGaNnzuTmqVOZW1BA+7w8huXnM6Br1+wy2bjRvB+77lomG5yKwT0fTo1ERI4B/oV1cP2B57FOP5nuBOBF7KZ/InAhcDjwVCzNdsD/gFOxzucYrNNLRq3ti3Xk14Q032SSP9aR1cNEy9HAZUAvYGzMhl2A0cAXoT4nYcKieSzNBcDIYOtxwcbBwJ2xNDtggmET8Cus4/srIMm2SdFWvYA3gLWhLQYDuwGviEidRPIbgPXAoGDTtSJyS7DpSuBMYC/g/hRF3RDa45fArcB5mPBL8iR2fo8FHgRGAAeISLIuZwGvRAIp1Pc6rP2OBZ4BbhWRK0tpgheAFiG/nwM3Y+cuHf0xARZ9foW1yXQAEamHnYsBWLv0A94CnhGRA1PkV3NYujTrQ0bPnMnQSZOYU1BAITCnoIChkyYxeubM7Mv3YZec454Pp6ZyFTBFVU8P38eKyCasMwNARGoBtwOjVPXM2PYvgbdE5CBVnYw9xe8NHKCqb8XKeDhRZktgf1X9Npb/pNLyD96N38f21wU+Bz4QkX1U9f1Qfj3gPFWNfMbjYsc0xoTErar6p9j2pcCjInKTqs4CLgZqAUer6o8hzRfAtFLaE8w78yFwnKoWhmPfA2ZgHo5HY2nfUdULw//jRWQAJrx2iXlJOgK3iEgDVY2vu/21qg4O/78iIo2AC0XkL5HNgQdU9cZYXWtjXoyzgKFhWxfMA3VS+N4KOD+00xXh0HGh/f4kIrerarGg0XDcLsDFqvpC2Px6SY2lqh/Ejm8I/AP4GDsHYAKuJ7C3qn4Sa6uOmIA9PGFDM6BZopjEkqHVh4EDS4gbXrAg9VLrJTBtwQLWJd7RsnrDBi6ZOJHHp0/PzriJE+GBB9LuHjVqVHb5OVnjng+nxhGewnsCyTtI8vuuQCfMZV83+gDvAcsx7wNAX+CLhPBIxceR8Mgyf0TkdBF5X0RWYE/HUce1W5Q3NtTyuIgcFzqiOL2BxsBTibImAHWAfUK6/YH/xTvxIG6+KaliYWjoIMxjUyeW/xzMq9Mrcci4xPcvsTacHds2AxNCOybSPpv4PgpoiA0TxXk+/kVVN2EekP8LtgH8GlgUS7sfJuLinifC90aYyEvFYuBb4CYRGSwiO6VJl46HgDbACTGhdSTwCfBF4pyNB/JT5HERdp7in0lZ2lE9KMOslaTwKG17RZfvVCzu+XBqIq2xa/uHxPYFKdKBDWekYufwtyUW41EaZcpfRPoD/8ZmS1yBdXTNsGGX7QFUdUaI+bgMGyZARMYB56vq17Gy0nkworq0A1L5qZO2J2mBiZjbwifJ14nvPya+r8NiUJLbINQxRrrz1i7N9jgPYsM6/UTkBWxo6N9RvAhbhqnmp8mrRYo8UdVCETkS8y7dBeSJyEfAJao6IdUxESLyZ+B4oE9CfLXG4oDWpzmuWTzuCLgDG8KK04EqFCCDBw/OOG2J3oOXX4Z33smq7F5PPMGcguIzmXfMy2NUv35Z5cURR8CBNXtkq7rj4sOpiSwENgA7JLa3SXyPgjnPA1LdCaNOcDEWn1AahWXM/yTgI1X9bbRDRPZNJlbVsdjwUR721HwrFrewf6ysE4HvUpQVvVVrHsXbBaxtSpr7uBSr300U90xE+yuKdOctGaWYbG9CQOnL2NDLesyr8mAsSdRObTCvTbKMJaRBVWcAg4J3Yn/gWuB5EdlZVVO2XRCM1wG/TuE5W4J5tM5KU2SRnjYIkaWJ/NOZW71pl9SRpTMsP5+hkyaxesOGzdsa1K3LsPxUTqISqF3b3gXj5BQXH06NQ1U3isg0YCAW0xGRHISejnXKXVX1nhKyfBX4pYjsp6rZPK5lmn8DLIgzTtqlJENMwmgR6YG54gGmACuBneKzX1LwDnCuiDSPxXzsA3TG4kzSlblSRKYA3VQ149kxZaQ/xc/bKmyIIhNGYAKpORb3E1/N6l22BMK+H9s+CGu/+LaUqOoGYLKIXI8Na3UkhXATkd2Bx7CZPsn4ILDhlZ8D81R1Tor91Y4pU6YAcMABB5Qvo2bNbEGxtcnLPj3RrJZyz3bJy/OXy1UDXHw4NZXrgDEi8ggWCNmdxFTb4Eq/CIuTaASMAVZgQxRHAX8PYuMR4A/AiyJyLdYJ7ggcoqpnpzMgi/zHA/eEqZqTsUDD4+J5icjvgAOBl7EhoJ2xJ+bxoaxlInIF8BcR6QC8hg1rdMZmUZwbXP63A+diHpQbgTzsCb60YRewKbyvicgozOOyBGiPBXS+WIHTQ7uIyEPYbJaewKXA7Ylg05J4CfOSHEjCq6Cqi0TkTmCIiKzFZsv0wa6Nq1V1ZaoMRWQvbLjpKWyKcx4W1DqX9KLtv9jwzugwvTliuap+jg21/RaYKCJ/w8RqUyy2pZ2qnpthfauMGTNmABUgPpo0sSXVs5xyO6Br1+zFRpIWLax8J6d4wKlTI1HVF7F1HA7COoEB2BN1Mt1oTAjsgj2ljsHWgFgMzApp1mJBp09jMRdjsdkIydiEVHaUmj823fRv2IyXZ4E9KO6l+Rh7kv8bFsx5Q0j761hZdwCnhTqPwmJNzsc6tR9DmgXBnjqhPlcFe0qdLqCqb2Mden1sKGMsJvJqAxWw8tRmLseGVJ7GBM+9ZLEWiapuxAJMC0IeSYYCVwNnYOdjELY+R6rpvBHzMaExDJs6/U9seO9wVU23WtZu4TMZm0Ibfe4Ndq7DrqvRwJ+w83o/cCgwMbPabqW0aAG77VZ6uoqmVi3o3t09H9WAWoWFxYZNHcdxtlrClNsZ2KyetJ6pmkC0qNuECRPo0KHyZ92OHDkSyC7wNC3ffw9PP121q422agWDBsEOqcKenIpk9uzZ9O3bF6BzmOZfBB92cRynRiC2nP1PsaDbzsQWV3OqIW3bQrduWc96KTO1a8Oee/obbasJLj4cx6kptMcCShcBF6jqpzm2p8ZRIW+03ZIZ9OxpHpC5mcxkLyedO0OPHiZCnJzj4sNxnBpBcO366lGVyKmnpp2EVTZat7b1NsaMyXrF06xo0gQOOACaNq28MpyscAnoOI7j5I7ddoNDDrGpt5VBo0bQp495Ppxqg3s+HMdxnIyYONEm4Rx66KEVl2m9erBvWFPvjTcq1gPSuDH87Gew114+3FLN8LPhOI7jZMQ333zDN9+U+BqgslG/vsV/HHss7Jh81U8ZqFULOnWC444z4VEn+dJlJ9e458NxHMfJPfXq2eyX1q1h2jT4/HNYvjz7fJo3t1kte+9t/zvVEhcfjuM4TvWhdWvo29cWA5s5E1RhyZKSl2Jv0MAWLuvWDbp0sXU83NtRrXHx4TiO41Qv6tWDDh3ss9dethDZ0qW2HPvy5bBxI9Sta7NX2ra1d8U0aWJ/na0CFx+O4zhORjRs2LDqC23Rwj4dO8JPf1r15TuVgosPx3EcJyMGDRqUaxOcGoLPdnEcx3Ecp0pxz4fjOI6TEa+++ioAhx9+eI4tqXhWr4YFC2y191WrbFteHuy0k4WV1K+fW/tqGi4+HMdxnIyYPXt2rk2ocFatskk1n34Ks2dvER4RkQDZc0/YZZfKW4h1W8PFh+M4jrNNsnw5vPmmLSuyYUPqNAUF8MUX8NVXsP/+9slF3G1Nw2M+HMdxnG2OVatMeEydml54xFm3ztK/807JS444meHioxojIsNFpCDXdlRHRKRQRIaUkmaWiNxdVTaVhIhcJCI/zzBtmewWkc4i8qqILA/tc1g25VY1InKCiPwhi/SdQr0GZnHMYeGYnmWzMnP897p1MXOmeTw2bcr8mI0b4e234dtvK8+ubQUXH9WbfwJ9cm2EUyFcBFS2CLgR2AU4CegNvF9F5ZaVE4CMxQcwD6vX/yrFmvJT43+vTZo0oUmTJrk2o9ysXg0ff5yZxyPJ2rUWH7J+fcXbtS3hMR/VGFWdDVRJhJeINFDVCnydpANV3q57AJNU9ZVY+VVUdOUhIrWA7VR1DfB2ru1JR1X+XnPFgAEDcm1ChTB/PsyZk3rfzJmjmTr1ZgoK5pKX1578/GF07Vq03t9+Cz/8UDHvwNtWcfFRjRGR4cAQVc0L3w8DXgOOAAYDxwNLgYeA4aq6KXbsHsANwGHA9sCXwM2q+kTYXwhcBjQGzgRaA/XCjf5C4PdAZ2ABcD9wo6oWhmMFGA4cGI6bBfwb+Kuqbn6WEJFLgbOBnYAC4BPgD6r6RdhfH7gSOA1oD3wL/E1VRyTa4cyQrh0wFTgny3b8PTAMaAG8CZytqt/H9pdqh4jsB/wZyAeaAl8Bd6nqA7E0h2Hn5+fAGcAxwFQR6Qp0BM4VkXND8jNVdWQWdegFXI89+QOMAy5U1dki0gmIXjX6UxE5PdSBTMsVkdeAtap6dGL7mdgTfQdVnRe2nQYMAXbHrr/HgctUdW3YPxi7JvcONvcB5gO3qeo9Ic3I0EbRtQjwsKoODvt6An8Ebga6A2eJyKRQz5NUdVTMxv8DLsbEVwHwLnCOqqZ0jmdyjac5bg/gb8D+QANgDvCoql4T9g+n6O91Ftb+SSaq6mEhTZPQRicCrQAFrlLV59LZ4ZSf774z70eSmTNHM2nSUDZssJ0FBXOYNGkoQBEBsmyZCRgXH2XHxcfWyf3AE0B/4Eis45wBPAogIrsCb2FPYRdgN/7uwM6JfC4ApmECIZrFfivWud8ETAH2Aa4BNoVtYCLgy2DDcmDPkKYFcGmw4XRsGOCqYEsT4IDwN+JJrGO6FhMmPwPuE5EVMZF0DPCvULfHQj2ey6KtjsU6yfMxoXU71jHGFyoo1Q6sE3kLGAGswjqgu0SkftShxngg5HkiUAgsAV4CJmPtCyZeMiIIjzeA8cCpQD1M/L0iInuxZTjiMWA6cB0QhcRlWu4TwD0i0lpVF8a2nwy8HhMeFwC3AXcCQ4Eu2HluBPwukefjWFvfEey+W0Q+VNU3g42tsXNzakgfL7c98A+sY/4m1DFV21wK3IJdI1dg97SfhbzTjcxnco2n4oVg41mY6NoF2LWE9P2B+MTMjsAj2DlCROphIrIDdj5nAQOBZ0TkkNBO8bo2A5olyuhQQvkVztixYwE4+uijS0lZMgMHZhy2UyksXmyvi0myYME0Nm1aV2Tbhg2rmTjxEqZPf7zI9smTc/8qmVGjRpWeqJri4mPr5FlVvSL8/6qIHIndtB4N24YD64ADVDV6J/WrKfJZBhwfeUxEpAv2RHieqt4Xy782cJmI3KmqK1X1deD1cMz/t3feYVJW1+P/LL0s0kSKq4KiB1t+iqyGYA22RDRiiyUFYyxfo8YWo4mJJXZjb1FjxFhQRDCKCYrEgooJgh05igKh97YsAgv7++PcF959d2Z3ZpmdWeB8nmeembnvfe89977vzD333HPuW4QNbo2A60TkijB7PAD4RFXjf+YvRR+ClWAg8ENV/Vesrg7YwBQN+n8E3lPVn4bvo0RkPRsH09poDAwIJntEZDvgLhFpp6pLM5VDVYfGZC8CxmID3LlAUvn4p6pWcYYVkdXAPFWty7LBbcBHwHEx69MHmMJ5mqo+BbwvIquABfE6sqh3GHA/5i/yYDi3MzaQnxu+t8GUgTtU9bexOpYCT4nIzao6LVbmg6p6f8jzFjAglP+uqn4tIguAndLI1h67bu/F6ukezyAibbF7/RFVjSs+/0jXyEzv8RTnbYspG5eo6ssh+c109QCo6oex81thytQnmJUGTOnqA+yrqp+GtNEishOmDCV38roYuKamOuubuXPnFrL6eiepeNSUXlRU39Js2bjysXnyauL7JGC32Pf+wLCY4pGOV+JLNdifXRHwvIjE743XsRmhABNFpAW2ZHMGtqQS3/uvM2ZpmQicLyJ3ASOA91U1/gs+EliC/dkm6zo3DP7LsD/nKxJyDyNz5eOtSPEITArvJdjstVY5VHWxiLTHBrofhXOj53WnCrp7KUVanRCRlsCBmEWpccyHYxZmot+fjUpnnQltHIVZOh4MyScDFcAL4XtfzHr0XKKvxmD90RubvUdsuE9Vda2IfEXmM/VFccUjDX2BVsBjGZYJGd7jqeTBLCk3i0hHYEx86S4DHsd+G6UxH6AjMUvbFwlZRmNKd5K7gcGJtBJMEd6sKPSM/e234d8p3JaHDNmfsrLqziDFxdszYEBVmY87Dnr3ri8Jt3w82mXzZEni+xrMryOiIzA7g3LmJb53wv6YFwBrY6/x4Xi0bHMrphA8hs1m9wd+H45FcgzGZmpHAG8BC0XkARFpHaurfaKetcDzsbo6YQry/FrkrolUfRWXMxM5ovacgS05HIn5ftxPVbN6XeSrjQ7YwH5nChn3pvpS2qbwDNBPRHYI308D/qWqS8P3TuF9QkKO6PokZantPq2JTPqwY3jP5F6PyPQer0KwOB2JKa/3Af8TkY9EpH9tFYrI7zCl9YTglBqXZR+qX9c/A9uEZZa4DEtVdVr8xRbu4Fpf7Lhj6o3CSkuvpEmTllXSmjRpSWnplVXS2rWzLdeduuOWjy2TRdiaeW0knesWh7QD2ThIx4n8BU4GHo4vqYjIwfGMwaJyL3CviHTFloVuB1Zgzp+LgYWYU2YqvsSsChXAdoljnWtsVXbUKkew9AwALlPVe6MDIvLzNOekdVqsA0tDeTdjFqRUx3PFS5g/y6kiMhSzLJwaO744vJ8I/C/F+dlYAmojkz5cFN67kfkgnOk9Xg1V/RI4JVgpvov5CL0kIjuq6qJU54jIAGz57heqOi6FLJ9gPiSp8D1D6okuXaCkBL78smp65FRaW7RLjx6wXfJfyckKVz62TF4HThKR36pqCreqtIwJ751UNe26Oebpv2G5IfhAnJ4uc3BWvE9ETgb2CMmjMetJhap+lO5cEZmAKS53xZJz6a1WqxzBt6ARVdvcAhuEMyWbWf8GVHWliLwH7KGqv6/1hE2oV1XLReQfmMWjETb4vRzL8h6wEthBVYfXQZY6y5aGcZiydCYW4ZIJmd7jaQkRXe+IyA2hvJ3YqAhtQER6YU7Ad6vqEymKGo1FRs1R1TSBnw2Ljh071p5pM6BFC/jOd2Dq1Or7dfTseUI1ZSNOy5aw557QxEfPTcK7b8vkOmym/q6I3IpFCuwBtFLV29KdpKpfisi9wN9F5A7sz70x5mg3UFUjB7jRmD/EZMw8fja2dLEBEXkYm5W/j83wDsBmi5eFul4XkReBf4nI7ZhDZUss+mF/Vf1xKOpPwEgReRLzbdiLLENtayITOVR1mYiMxxwSF2FKyGXAt2mKTcUXwOHBOXgxMDXdbDkFlwNviMgwbGlkMTbbPwzz23kxh/U+A4zErEsvxvcoCf1wNXCriJRgYcVrsHDVAcCvEssKtfEFFkJ7Bua/sjDhsFojQZ7rgjyNsSioRli/DFHVD1Kck+k9XoUQVXQn8BxmHSnGlNbZbPQjSvIPzP9puIh8N5a+XFUnYeHpZwNvicifsSiYtthyWldV/VWywEJz7LHHFlqEnNGzJ+y/v+1Yum5dZuc0bQrf+x7slCqA2skK9/nYAlHVr7Cw1qmY8+DLmGk3k02BLwF+C5wQzhuC7SnydizPBdguk3djvhDTsHDfOO8FGf4KjMJmp1eq6n2xPKdg6+fnhDxPYJEnb8ba8grwS8xM/o8g18AM2pENtcqBWXa+xEI6H8BCJB8ic36HLVUMw/wLMv4XD9Eg/TDH3seCjH/Cfr+f57je19i4bPdM8qCq3o3th3JgKHM4FsY8meo+HrXxGDAUuCfIdm2W5xOU6V9giu0I7H7cjep+QnEyuceTzMUUjSuBV7D7egFweMKhOc5u4fUOpuRErweD7Gsw5/DhQZ7XsDD6QzA/KaceadEC+va1V7Nmtedv2RIOPhj69DElxNk0iiorc7k87TiO4+SLEH48dcyYMZSU1P+WHy+/bKtwW5IF5NtvYdo02zJ9+vTq+3+0a2eWjr32gu7dXfHIlJkzZ9K/f3+AHqksmr7s4jiO42TEokWZrhRuPrRoAb16wc4725bpc+bYE28Biouha1dzLnUfj9zi3ek4juNs9TRrZhEweTAgObjPh+M4juM4ecaVD8dxHMdx8oovuziO4zgZ0cW39XRyhCsfjuM4TkZs6tNsHSfCl10cx3Ecx8krrnw4juM4GTF8+HCGD8/FzvrO1o4vuziO4zgZsXz58kKL4GwhuOXDcRzHcZy84pYPx3EcZ6tgzRpYtgxWrrSn2RYV2TNbiouhbdtCS7d14cqH4ziOs0VTXm7PbfnsM5g7F8rKTBEpKoLWrWGbbUAEdt3VtlN36h9XPhzHcZyMyMfD63LNnDkwbhx88YVZO+JUVtqD5FasgFmz4JNP7Cm3e+5pz3xx6g9XPhzHcZyMOPzwwwstQlb8738wahTMnp1Z/oULLf+yZaaEtGxZv/JtzbjDqeM4jrPFsXAhvP565opHxNq1Zin5+GOzjDj1gysfmxEicqiIVIpIn0LLkg9E5E0RGVlLnsEi8lm+ZKoJETleRM7Pc53dwz1xUj7rbciIyDQRuT9HZTWY+6shMHToUIYOHVpoMWqlogImToQZM+p2/tq18J//wMyZuZXL2YgrH46TO44H8qp8OE4+KS8vp7y8vNBi1Mq8efD555tmuViyxBxU16/PnVzORlz5cLJGRHwlNIb3R/3i/etky/Tp5reRjilThjNkyP48+mgJQ4bsz5QpqXdt/eYbWLSonoTcynGH0waCiBwA/A4oBdoCXwP3qeqjKbJ3EpFhwNFAGXCvqt4UK+ta4HLgAOAhoA8wHbhaVV9I1HsOcAmwMzAPeBK4RlUrwvFBwOPAd4FrgYOBZ0XkSeCNIMOZwDHAinDuoyJyFvB7oCMwCjhbVZeHMlsCtwFHADsCC4DXgd+o6uLsew9E5GDgbqAXoMCvVfXtRJ6fhH7pBSwFngGuUtXV4Xhn4CbgMKArMAt4EfiDqq6KlVMJXAW0CW3vJCJPAz+PHQd4QlUHpZD1GqzPO0d1h/QewDfAqar6XJb3RFTG40BfVe2VSD8M+DfwXVX9T0g7GvgjsA/wbWjrpaq6tIbyB2H3Qx/gRuAgYG5oz8vAlcAFQAvgOeBiVV0TO38v4PZw3nrgrVDnV7E81foXaCoig0O9lwF/BnYFPgcuUNVxKWQ9L8jTAXgXOEdVZ8SOd8Duw+OAbYBPsd/Iq+nan0Ub2gL3AgOBtdjv6gvgL0AnVV0oIh8Dn6nqGYnyfwdcAXSN33dOZqxbZ46m6ZgyZThjx15BRYV1bVnZLMaOvQKAnj1PqJJ36VKLhOnUqb6k3Xpx5aPhsBMwDngEKMcG+/tEpJmqPpDI+wj2x34icCRwo4gsVtW/xPI0BZ4F7gduAC7ClIZdVXUagIhciP1BPgRcDOwLXAd0Ac5K1Pks8Bj2p/st0CykPwQ8EY79DHhERHbBFJ+LgO2Ae2IyALQM8v0BU3hKsMFmFLB/hv0VpwvwADaQLMSUpBdFZCdVXRHaehFwZ2jvFZiydRPQGjg3lNMRU0ouBZYAPbHBeSfg5ESdFwETgHNCX3yMDZK9gGgwWZBG3iFBxh8CI2Lpp2HK5Evhezb3RLzsQSKyr6p+mCj765jicTzwAvB34E/Atpgy8RxwVJqy4zwFPIz16aXAUOBBYAfs3tkDux5TQh5EZAdgLKYIDwKKsPttrIjsrarx/kr2b0RXbAC/FrtWVwGvikhPVZ0fy3csdi0uxJSYuzCl6fAgS2PgX9g1vgqYGep6RUSOUNU3UjU6izZEdV2FKY2DgBMSxT0C/FlE2kUKn4gUAb8AnkkqHiLSDmiXKGOziX096aT8uCVVVMD8+baPRyrmzZvA+vVrEues4q23LmPy5Geq5X/7bWjTpj4krc6wYcPyU1EDwJWPBoKqbvDiCn9AY7HB7FxsYI3zhqpeHj6/KiJdgatF5BFVjVYom2Gz+pGhzAnYQH88cHf48/0j8Lyqnh8raz1wi4jcqKrfxOp8NGFdOTR8fEFVrwtp72J/sL8AeqjqypDeGxu8LwptXQycFyurCTAJ+FBEeqvqxEz7LdABOExVPw3lzQE+BPpjSkgbTPm5Q1V/G6t3KfCUiNysqtNUdRI2q46Ov4spMyNEpKOqxg2wy4AfxfobEVkA7KSq79ckrKp+Ga7HaVRXPl6MBp0s74mIMdh1Pi30ASLSFFNUH4yVdRcwTFXPjNXxFTBORA5U1XdqagNwf6QAiciXwFRMEf5O6JNRIvJ97LrfGc65BLsvj4gGaREZh1l7foUpFBHV+jfQAThZVf8dzn8bmBHKviqWrzEwQFW/Dfm2A+6KDfTHYIruMar6z5BnFPAJcA1m1UtFrW0QkT0wi8eZqjo4nDdKRD7ElLOIpzAF7XTCtQEOBXbBlPkkFwfZCkaPHj0KWX1G1ObnkVQ8akv3iJf6wZWPBoKItMf+fH+EzWYah0OrU2Qfkfg+DBtsSoDI4LgeGB1lUNVFIjKfjTOlXths97lEWc8BtwL9sD/UiJdIzWuxOspFZBYwKVI8Al8CnUWkqaquDe39KfZHvitQHMu7G5Ct8jEnUjwCk8J71Na+2Oz3uaDoRIzB+rk3MC0Myr/GZsA9sKWDiF2BuPLxSoqBMRueAW4QkWJVLQum/L0wqwyQ9T0BgKquE5GhwKki8ltVrcQsGR1CnVFbugOXJvrjA2A5NijXpnzEr/s0EVkDvJ7oky+BU2LfDwL+HbdwqOrMoOQdlCg/Xf8uixSPcP4SEfk3ZmmL81akeATi98TSUN+KSPEIZa0XkeeB34tIY1Vdl6L+TNpQGt5fTJw7HFviis5bFq7VWWxUPs4CPlLVCSnqvhsYnEgrwZTSvHDIIYfU+dx8zepXrIDnnksfqTJkyP6Ulc2qll5cvD0DBlSX8YQT4DvfybWUjjucNhwGY+b6O7EZZCm2ZNI8Rd75ie/zwnt8Y+BVcX+CwBo2Dqjtw/vcNGV1SJOeZEmKOpamSCsitEVEBmLm/gnAj7HlhKND3rrsK1hFhpiPQVRWtGI7AVt/j15RP+4Y3i/G+v9lzEJ0AOZzkEqudP2RKc9i/XF8+H4aZmUZHcszmMzviTjPYDPsfrGyP1LVL8L3qD+GU7U/1mK+DztSO8nrvpbU1z3eb+2pfr+B9WWm91uqpax5VL33U8mXvCfap6ljHrYkWJziWHRebW3oCqxN4TuT/N2CLb30FpF9wrLKiaS2eqCqS4OFbsMLWy5yYrRuDe3bpz9eWnolTZpU9WFu0qQlpaVXVsvbooU/86W+cMtHA0BEWgADgMtU9d5Y+s/TnLJd4nvn8D4ni2ojx87OifTOieMRuTQ+ngx8rKpnRwkisl8Oy08SteVENlqG4kROiCcDLyWWZnZIkR82sT9UdXZYMjgNM7+fii2BRY6+2d4T8bLfF5GpwGkiMhGznFwfyxL1xwXAf1IUkWqQzAWLqX6/EdIyvd9Suf51Jrt7vzZZ1mK+N9meF7VhDuYg2y6hgCR/t6jquLCPyFnA5JD8dK3SF4innzbRzjjjjFpyFo5GjaBHDwuTTbVkEjmVjh9/C2Vlsyku7kZp6ZXVnE0BOnSw5744uceVj4ZBc8wKFY98aIENlqkYSNWll5OA2WQ3C1JsFnkKNgOOOAX746/N7L4ptKT60kF9/pu9B6wEdlDV1DF1xqbKlZzp18YzwAMi8kPMATbu7ZbtPZFkCHA21vZW4XvEZEzh6lmD42p98A5wTtx/RkS2B74H3JxhGW1F5Psxn4/2wPeB++ogy29E5GhVHRXKKsJ+S++lWXLJtA0fhPfjqbpMUn10Mx7BnFZnAsNVNWm1aTCsTT4cpYFSUgLbbgsL0rh89+x5QkplI0mvXjVbUZy648pHAyCs/Y4HrhKRRdiAcxkWVZKKw0TkdsxEfxQ2e/5VNj4IwTfgeix6YgG21LAvNkN+XFWn1r1FtTIaG3Svxf7MD8fCHeuF0L9XA7eKSAnmTLgG8+sYgPXdzCDXxSEyZjJmCdkni6q+AM4SkTMw5W5hFFmUhmHYMsqjWPTEuwmZs7knkjyDheneBrwTDzFV1UoRuRjzgWkNjMTCpHfE7qd7oqiYHHMXtoz1mojciClX12JLJJkqQYuBx8K9swRrYyXmD5ENrwD/BZ4Moa1RtMvuhIiYurZBVT8XkRHAvaF/o2iXbUMZyd/pk5if1d7Y0p+ziXTqBL172/bq69KpkbXQrRvsvntu5XI24j4fDYfTMQe9v2F/Yq9hYaypOBcLERwB/ATbh+LBNHnToqr3h7KOwAagC4E7iEWi1BMPY/s0nIe1YXdsxllvqOrdWF8diA36w7H2Tmajf8D1hH1OMMfbZtiAlCmPYSGn9wDjqRq9kUqmJVh4cTfg2eAcGiebeyJZ9udY5EY3qlpUouPDMUVjF8zMPxLbE2MRMC2TOrIlKEAHYxa3v2PhqNOAgxNhtjUxB9tF9grgeewaHa2qWfngBMvGDzBH6pux+7AEi5B5MwdtOBNzOL0FszotwRTNSkzRi5e5FFOIvyF9lI2TBUVFsNdedVceWreGfv18f4/6pKjS44gcx9kMkLDJmKruVWhZ6oKIvAx0V9W9E+nF2IZ2t8bD2TMsszswdcyYMXl53P3gwYMBGDRoUL3XlQsWLYJ//xu++CLzbdK32QYOPRT+3/+Dxo1rze6kYebMmfTv3x9s24VpyeO+7OI4jpNjROREbJO4TzA/oBOxJb6zYnnaYJux/R8WRp1259qGwm677VZoEbKiY0c48kjo3NkeNLd0afq8TZrATjuZxWOnnVzxqG9c+XAcx8k9ZZiz8nXY0tBX2Pbuf4vl2Q9bZpkJDMpi6algfO973yu0CFnTtq0pFLvsYs9q+eYb2wtk7VpbnmnVyqJa9tzTHFXztZvp1o4rH47jbBZoiufkNFTUng9T4zNigm9JUV4E2spp3Bi2395epaWmfFRUmPLRsqUpHI3cAzKvuPLhOI7jZMTm5vORihYt7OUUFtf1HMdxHMfJK658OI7jOI6TV1z5cBzHcRwnr7jPh+M4jrN58u235j367bcbPUibNoXiYnt5vGyDxZUPx3EcJyP23HPPQotgSsb8+TB1KkyZYsrHypUblY/mzS18JYqf7doV2rUrtNROAlc+HMdxnIwoLS0trAALFsCECTB5cvodw9asMYVk9mzL16WLxdeKeJhLA8KVD8dxHCcjoqfaNm3aNL8Vr1sHqvDOO6ZUZEpFBcycaZaS//0P+va1x906BceVD8dxHCcjnn76aSDP+3ysWwcff2wPaSkrq1sZa9aYxWTp0o37rTsFxaNdHMdxnIaL6qYpHnG+/hrGjIElS2rP69Qrrnw4juM4DZP582Hs2NwoHhFffQUffWQWFadguPLhOI7jNDzWroUPPoA5c3JbbmWlLcFMn57bcp2scOXDcRzHaXjMn29LLvVBWRl88ok5pDoFwZUPp06IyKEiUikifQotSz4QkTdFZGSh5cg1mbRLRI4XkfM3oY5pInJ/Xc/fVESke7hXTyqUDFsK++yzD/vss09+KvvmG1i2rP7KnzbNFBynIHi0i+M4tXE80Ad4sMByOAUmb4pHebk5h9YnS5faviHdutVvPU5K3PLhbNaISMtCy9CQEJEiEfGdlBowItJYRJoVWo66UF5eTnl5ef1XVFZmG4VlyPApU9h/yBBKHn2U/YcMYfiUKZmdmGt/Eidj3PLhVENEDgB+B5QCbYGvgftU9dEU2TuJyDDgaKAMuFdVb4qVdS1wOXAA8BA2g54OXK2qLyTqPQe4BNgZmAc8CVyjqhXh+CDgceC7wLXAwcCzIvIk8EaQ4UzgGGBFOPdRETkL+D3QERgFnK2qy0OZLYHbgCOAHYEFwOvAb1R1cZb9Nji073zgbqAXoMCvVfXtRN6fhH7pBSwFngGuUtXV8X5T1eLEedOAkap6QaLOS4FbgL2As8I12eR2hfJ/Hj5XhuQnVHWQiLwJlKnqgFj+7sBU4GRVHZYo6zLs+nYARgPnqeqc2PFmwB+AnwDdsPvkz6r6SAZyHoPds/sCq4GPgUtU9cM0+RsBVwK/BLYHZgAPqOpdsTzbA3cAh2G/g7nAKFU9L9Y3fVR1r0TZlVg//zl8fxP7bQwJ7esZyhwrIkcDfwT2Ab4FXgQuVdWltbW5EAwdOhTIwz4fq1ZlHOEyfMoUrhg7llXBf2NWWRlXjB0LwAk9e9Z88vz55tia703THFc+nJTsBIwDHgHKscH+PhFppqoPJPI+AjwHnAgcCdwoIotV9S+xPE2BZ4H7gRuAizClYVdVnQYgIhcC92IKysXYIHId0AU4K1Hns8BjwO3YH3Y0i3wIeCIc+xnwiIjsgik+FwHbAffEZABoGeT7A6bwlABXYUrK/hn2V5wuwAPYwL8QU5JeFJGdVHVFaOtFwJ2hvVdgytZNQGvg3DrU2Q34S2jXVGBODtv1J6ATpiSdEdIW1EHG47EB/ldAO6x/RmD3VsSz2KB8PfAp8H3gIRFZoapD0hUsIj/GBvZ/BBlXA/0wpSKl8oHdOxcDNwNvA/2BO0SkWFX/FPL8PZRxEaZ47AAcmHmTq7Afdp2vw+6LqSJyPPBCqOdPwLbAjdjv6agU7WyH9V2ckjrKkxdOOqmObjarVsG8eRllnTBvHmvWr696ekUFl731Fs9Mnlzzyc2bw4gRdX4A3bBhw2rP5KTElQ+nGqo6NPosIkXAWGwAOhcbWOO8oaqXh8+vikhX4GoReURVo3+EZtisfmQocwI2IB4P3C0ijbHZ3/Oqen6srPXALSJyo6p+E6vz0YR15dDw8QVVvS6kvQucAPwC6KGqK0N6b+BkgvIRrADnxcpqAkwCPhSR3qo6MdN+C3QADlPVT0N5c7ABsD+mhLTBlIQ7VPW3sXqXAk+JyM2RQpYF7YEBqvpeIn2T26WqX4vIAmAnVX0/S7nibAMcrarLgjxzsGt8lKq+Gq7hQOCHqvqvcM7rItIBG5hTKh/h/vwz8JqqDowd+mc6QURkW+BC7BpcHZJfC9fmtyJyl6qWYUrrVar6XOz0p7Jr9gY6At9V1ekxuccCw1T1zJhsXwHjRORAVX0nUcbFwDV1rH/zo6jIwmJrIal41JbuNAxc+XCqISLtsRn7j7CZVTQtWJ0i+4jE92HAaeG8/4W09ZiZHQBVXSQi89k4a+uFzfrif/KE77dis9i48vFSGtFfi9VRLiKzgEmR4hH4EugsIk1VdW1o70+x5YBdgfgyx25AtsrHnEjxCEwK71Fb+wJtgOeCQhAxBuvn3sC0LOtclELxyHW7NpU3IsUDQFVfE5Hl2AD/KmY1WwKMTvTL68C5ItIhzXKRYH17WRayHIBZhVLdb/+HWd3GYn10uYisA0ar6ldZ1JHkk0jxCOwKdAcuTbT3A2A5Zp1KKh93A4MTaSVB1gZJnS0DM2fCkCH2tNpa2H/IEGalWKLZvriYYQMGpDgjRq9ecPLJdbZ8OHXHHU6dVAzGzNd3YoNCKbZk0jxF3mSsWmQr7RpLWxX5MsRYA0SOke3D+9w0ZXVIk54kuWfyGsyfIplWRGiLiAzEzN4TgB9jywBHh7x1cdysIoOqrkmU1Sm8TwDWxl5RP+5Yhzqr9Uc9tGtTSRXTOJ+N90kn7D5Ym3g9H46n65eO4T2Lp41lfL/9GFN+/gR8KSJficipWdSTquyI6D4YTvU2b0OK9qrqUlWdFn8BM+soT8OmVSto0yajrFeWltKySdV5dMsmTbgykyfwduvmikeBcMuHU4UQKTEAuExV742l/zzNKdslvkdPbMrGjTya0Saf9tQ5cTyidlts5pwMfKyqZ0cJIrJfDstPErXlRDZahuLMCO/fYrPzJO1TpKXqj3y0K+5vE5FUFCOS90mUFt0nizFfiB+kOf/LNOmLwns28ZLx+21WLL3K/RacYc8SkV9iFqnfAk+LyCeqOokU7Q/LRKlIXqNIhguA/6TI3yA3oOjTJ0/b+rRpA+3awdykflidyKn0lvHjmV1WRrfiYq4sLa3d2bRxY3/AXAFx5cNJ0hyziG2wVASF5MQ0+QdSdenlJGwWms2MTDEnxlOwmWDEKdifdtL8nEtaUn056YxUGXPEe8BKYAdVHV5DvhlAMxHpqapTAESkLzYrzoRctitupUrKeJSINIr591RzlAwcJiJtYz4fR2JtiQbe0ZjzbYWqfpSFbIrda2cCQ2vJG/FfzMJwClWXn07Brk2VJSlVrQQmiMiVmFLXC1tOmwHsICJtImdi0rc/yeRwfs8UTtwNlr322qv2TLmgaVPYYw97DksGz2A5oWfP2pWNJJ07w3apdGInH7jy4VRBVZeJyHjgKhFZhA1gl2GzvFQcJiK3Y4PHUZi/x69ig1Emda4TkeuxiJoFwMvYuvv1wOOqOrXuLaqV0cADIbT1HeBw4Lj6qiz079XArSJSgoUIrwF6YBanX6nqTOBfWHjmYyJyExZF8xvMHyATctmuLzALwBnYYL8wmPyHYqGqD4jIC9jy3KA0ZSwH/iUit2DWm9uA8ar6KoCqvi4iL4Y8twMfYQpUL2B/Vf1xqkJVtVJELgeGBBn+jt2zfUP51XZvVdWFInIv5s+xGngXi7L5Pyw8e6WItMV8iJ4MbW4cjq9go8L0AuEeFZGHMV+as8mAIPfFmO9Pa2BkKHtH7Hd0j6qmsogUlGVhx9G2bdvWf2UlJaYc1NdeHL16QftUhkQnH7jPh5OK0zEz99+w6JbXsDDWVJyL7VswAtuf4Q+qmvVOmKp6fyjrCOyP+EJsj4XzajovBzyMRUuch7Vhd8x6U2+o6t1YXx2IOegOx9o7meAzEpwrj8dCK1/EonPOprpfSzpy2a7HMEXjHmA85oyMqo7G9hf5ARbm2pf0yseL4fUQFhb8X6x9cU4B7gPOwUKCn8Asa2/WJFyIRvkRFhb7LBYZcyA1W9+uwCJHfo7db6dg+6pEYbbfYnuF/Cq07RmgFXCUqs4K9U7GruPeIc8pmJ9IRgTL11HALsDTQY4rsaWkaZmWk09GjBjBiBFJH/N6okMH2G+/+tmDo1s32H333JfrZExRZQahTI7jOE7DI9rUbcyYMZSU1P+WH4MHDwbysMlYRHk5jBplD4HLFS1awLHHwp575q5MpxozZ86kf//+YFsdTEsed8uH4ziO0zBp1Qr69YMd6xIEloKmTaFvX8jWP8TJOa58OI7jOA2Xzp3hBz+AnXe2jcfqSvPmcOCBsP/+9tkpKK58OI7jOA2brl3hmGPMatGyDs+S7NrVllrqer6TczzaxXEcx8mIvn37Fq7yjh3hsMOge3fzAZk+veYn3zZqZFYTEQvb9bDaBoUrH47jOE5GiEhhBWjaFHbbzRSQ+fPt4XOzZ8OiRbYfSFGROZR27Qpdupjy0SHdvm9OIXHlw3Ecx8mIhQsXArDtttsWVpBmzWwfkJISC8f99luoqDDlo3lzaOJDW0PHfT4cx3GcjBg5ciQjR1bbt63wtGgBxcXQurUrHpsJrnw4juM4jpNXXPlwHMdxHCevuPLhOI7jOE5e8cUxx3EcZ6tj7Vrbvb2iwqJyW7TwLUDyiSsfjuM4TkYcfPDBhRZhk1m+3B6U+9lnFqEbVz523x122gk6dYLGjQst6ZaNKx+O4zhORuy8886FFqHOrF4Nn34KH38Mc+ea5SPJ9Om2LUjPnlBaCoWOKN6SceXDcRzHyYi5c+cC0KVLlwJLkh2rVsG4cfD++7BmTfp8lZVmDVm0CBYsgCOOsP3KnNzjDqeO4zhORowaNYpRo0YVWoysWLsWPvgA3nuvZsUjyTffwOjREPZVc3KMWz62AETkUOANoFRVPyiwOPWOiLwJlKnqgBryDAb6qOpe+ZKrBlmOB7qp6oMZ5B1MHeQWkWbAw8AAYFvgOuCjTOvNNyKyD3A8cJuqlmd4zjRgpKpekEU9lcBvVPXPdRAzY7a23+DmxKxZZvGoqMj+3G++gYkTzQKyKQ/Udarjlg/HqX+OB86v5zrOBH4KXAL0Bf6ap3rryj7ANUCrLM4ZCNSrErEJTMT6/YtCC+JsZP16+OILWLmy7mV89ZUtwTi5xS0fTs4QkZaquqrQcjQU8twfuwOzVfWpWP15qrp+ifpRVT8stCzpUNXlwPuFlsOpysKF8PXX6Y9PmTKc8eNvoaxsNsXF3SgtvZKePU+oVsaMGf5Q3FzjykcDR0QOAH4HlAJtga+B+1T10RTZO4nIMOBooAy4V1VvipV1LXA5cADwENAHmA5craovJOo9B5tF7wzMA54ErlHVinB8EPA48F3gWuBg4FkReRIzPx+NzcaPAVaEcx8VkbOA3wMdgVHA2eGPGxFpCdwGHAHsCCwAXsfM5ouz7z0QkYOBu4FegAK/VtW3E3l+EvqlF7AUeAa4SlVXh+OdgZuAw4CuwCzgReAPceUimPivAtqEtncSkaeBn8eOAzyhqoOyaENX4Bbgh6Hsj4DLVfWdcHwasFOyjkzrFZFrsGvdOWpzSO8BfAOcqqrPhbT9gRuwWT7Aa1ifzgzHuwNTgZ8A+2PWmLXAC8Clqvpt7N4BWBCUpOmq2j3dfQWclWrZRUT6YktM3wWKgEnY/Ty6hv48GvgjZn35FruWl6rq0hrOaYvdm9Gy1kJM2fixqlYkl13C8tnPU5WlqkWhzCLg18B5QA/sd/YwcJOqVqY618mOuXPNeTQVU6YMZ+zYK6iosJ9wWdksxo69AqCKAlJZadaTvfe259k5ucGVj4bPTsA44BGgHPuTvU9EmqnqA4m8jwDPAScCRwI3ishiVf1LLE9T7M/8fmwQuQhTGnZV1WkAInIhcC+moFwM7Iv9wXcBzkrU+SzwGHA79kce/TwfwgbAx4CfAY+IyC6Y4nMRsB1wT0wGgJZBvj9gf8Ql2GA+ChvIsqUL8AA2aCzEBrMXRWQnVV0R2noRcGdo7xWYsnUT0Bo4N5TTEVNKLgWWAD2xwWsn4OREnRcBE4BzQl98DHTCFJszQp6Mjbgi0g54F+vbi4FFwNnAaBHZQ1WnYssRvwMODJ8JcmZa7xCsb34IjIiln4YpsS8FWfYH3gZGhzKbhvNeFZHvqOq62Lk3Av8ETsXunxuB2dj1fiW8X40pqcuA1VQleV+l6pt+wL+B/wC/xK5RH0xxTUnwv3kB+DvwJ0yRuBH73RyV7jzsHvkhcCWmXHUN39MtXf8JiP/ummEKfNwSdgfwf8DNwHtAb+x3tj6kJWVvB7RLJJfUIHPO6d+/f53PPemkk3IoSWYsXw6L00xb5s2bwPr1VT1QKypW8dZblzF58jNV0ps3hxdeyP/eH8OGDctvhXnElY8GjqoOjT6HmdJYbFA5FxtY47yhqpeHz6+GGfPVIvKIqq4P6c2wWf3IUOYEbKA/HrhbRBpjA+vzqnp+rKz1wC0icqOqfhOr89GEdeXQ8PEFVb0upL0LnAD8AuihqitDem9s8L4otHUxNguMymqCzWQ/FJHeqjox034LdAAOU9VPQ3lzgA+B/pgS0gYbBO9Q1d/G6l0KPCUiN6vqNFWdBFwWO/4upsyMEJGOqhqfWy0DfhTrb0RkAbCTqtbFLH8xpvyIqs4N5b0GfIZZkH6pqh+KyDxgdbyOTOtV1S/DfXAa1ZWPF2PWndswq8tx0cxcRD4Avgx5n4qd+0Hs/nktWChOAm5Q1QUiEhnDJ6hqqniCKvdVGm4DpmDXOFJ8XkuXOfx+7gKGqeqZsfSvgHEicmBkTUrBAcAzqvpELO25dHWp6teYlTKq4xFgG8yqh4jsjFk9LlDVh0K210WkEXCViNwb/U5iXIz5yRSMHXbYoZDVZ01lDfajpOJRU3plZc1lOdnjykcDR0TaY7PLH2GznEj3Ts4UoerAATAMGxRKgP+FtPXYzBUAVV0kIvPZOIPqhc0Gk3+szwG3Av0wU3zES2lE3zAIqGq5iMwCJiX+UL8EOotIU1VdG9obOU3uChTH8u6GOfVlw5xI8QhMCu9RW/tiyxjPBUUnYgzWz72BaTHz+DmYebxFLO+umDUi4pW44pEDjsTM+QtTyHhIDut5BrhBRIpVtUxE9gL2wqxB0ZLYgcBvgMYxf5JZ2HLW/lRVPl5NlD8pnJ8p6e4rgjytMCvgVQmLS03sCnQHLk305QfAcqwN6ZSPicAgEZmLte3TTJdGROR8TPE+RlW/DMmHY8tEzydkeR2zegjV7/e7gcGJtBJsQpIXZsyYAdRNCSnELP6DD2DkyNTHhgzZn7KyWdXSi4u3Z8CAqrLuuCOceiq0ysY92qkRj3Zp+AzGTNx3YgNRKbZk0jxF3vmJ7/PCe3ybnFXxdf3AGjYOqO3D+9w0ZXVIk55kSYo6lqZIKyK0RUQGYubwCcCPscHl6JC3BdlTRQZVjaY0UVmdwvsEzC8hekX9GJnvL8b6/2XMQnQA5tORSq50/VFXOmGK59rE6wJqWF6oA89i1+H48P00zLoTKaodMIXszhSy7J1CllTXP5trWFs/tsf+v2ZnUWZ0vYdTvQ3bUHN/Xojdm5dgS2kzROTXtVUY8zn6jarGFbJO2L2/ICHH+HC8miyqujRY4ja8gJm1yZBLxowZw5gxY/JZ5SbRoYNtm56K0tIradKk6sNcmjRpSWnpldXydumSvhynbrjlowEjIi0wB7fLVPXeWHpKRzbMjyJO5/A+J4tqoxXSzon0zonjEbk0Rp4MfKyqZ0cJIrJfDstPErXlRDZahuLMiMn1UmJpJt3UL9fG2cXYTPvqFMcynfHXiqrOFpG32bh8ciq29BbtjrAUa9vNVLewRcdzSW39uBSz4nXLoszoel+A+YkkSSrvG1DVZZjicYmI7IGFMN8tIqqqKXfdEpGdMOvjU6p6VwpZKjFrUCr7fw0xGk6mdOkC22+fOuIlciqtLdqlZUt75ksjn6rnFFc+GjbNsdldPAKhBTZYpmIgVQeGk7CZYTazI8VmY6dgM8SIU7A/y3Rm6VzQkurLSWekypgj3gNWAjuo6vAa8m2qXNnO+uOMxhx2J6tqWZbnZlvvM8ADIvJDzPF2g9edqq4UkfeAPVT191nKkU42spRvA0GeccDPROSODJdeJmMKZc8UztrZ1D1JRC7GnEX3wByiqxCWhV7EfFLOSx7Hls0AOqnqP+oqi1MzrVpZlMrUqbbnR5KePU+opmwkKSkxJcbJLa58NGBUdZmIjMcc0BZhA+BlpPH+Bw4TkduxAesobBb7q2x8EFR1nYhcj0XULMCWGvYFrgceD9EV9cVobPC7FlNyDgeOq6/KQv9eDdwqIiWYb8UazK9jANZ3M4NcF4fImMmYJWSfLKr6AgsVPQNT7hZGkUUZcCd2Hd8WkXuAaZgDah/MwfS6HNY7DFvSexQLwX43cfxy4I0Qzv0MNnvvhoUgv6KqL2bYpkg2gAtFZDhQnvDPyYQrsWiXMSLyALbU0xtr59+SmVW1MigNz4lIa2AkFga+I/Z7uUdVU1lEIifjFzFH37XYNVkPvJlGtrswv43TgN4xHxlU9f3g5Hsv8HcRuQOLaGsM7AIMVNXDM+8GpyZ23NFe06Zlf26rVrDPPmb9cHKLG5IaPqdjjpl/w6JbXsPCWFNxLhYGOgLbZ+EPWoettVX1/lDWEdgf9IVYWGCqGVwueRjbwfI8rA27Y9abekNV78b66kBs8B2OtXcyG/0Wrifsc4I53jbDnE8z5TFgKBZaPB5zIM5UvsWYY+x4LAR4NHAf5gz6Xi7rVdUl2Cy+G/Bs0qEyRM30w9r/WMj7J+x/5PNM2xTK+jDIc3pox8vZnB/KeAc4FFMCBmPXbiCmOKU7ZzimaOwCPI3d31diTsPTaqjuXew+GYqF6u6ORTWlc4IWzGL2IqZYxF8RlwC/xSLBXsZCngdh4cxOjujQAQ4/PPsHxLVoAQcdBFvIXn0NjqJKjx9yHMfZLIk2dRszZgwlJfW/5cfm+lRbgJkz4d//hunTYV0ti3Tt2sGBB8J3vuMbi9WVmTNnRvvC9EhlcfVlF8dxHCcjNkelI6KkBI47zrZK//RTU0bKY480bNwYOneGXr1gl13MzyPfm4ptTbjy4TiO42TEN9/YFj8777xzgSWpG+3a2WvXXWHePFi2DNasMSWjVSt7fkv79rWV4uQCVz4cx3GcjHj7bXNH2VyVj4gWLWCnnQotxdaNO5w6juM4jpNX3PLhOI7j1Jk1a9awfPlyVq1axdq1awFo1KgRzZs3p7i4mOLiYoqKigospdPQcOXDcRzHyYo1a9Ywd+5cvv76a2bMmMGKFSsoKyujoqKCyspKGjVqRKtWrWjTpg3t27dHRNh+++1p165doUV3GgiufDiO4zgZUVlZyapVqxgxYgQzZsygrCz9prtr1qxh6dKlzJgxg88//5wuXbqw++67s8cee9ChQ/IRUc7WhisfjuM4Tq0sXLiQDh06MGnSJObPT/sYnJSsW7eOWbNmMXv2bL766isOPPBAevToQZMmPgRtrfiVdxzHcWpk2rRpjBkzhpkzZ7IpG1NWVlYyffp0Fi5cyAEHHEBpaSktfe/yrRJXPhzHcZy0fPPNN/zrX/9iwYIFOStz5cqVjB07loqKCr73ve/Rwp9Xv9XhobaO4zhOSmbNmsVrr72WU8UjYu3atYwbN44PPviAdbXtd+5scbjy4TiO41Rj5cqVjBs3bsPzXOqDtWvX8p///IdpdXnkrLNZ48qH4ziOU4XKykomTZrE5MmT672uFStW8O6777Js2bJ6r8tpOLjy4TiO41Rh4cKFfPDBB1RUVOSlvmnTpvHll1/mpS6nYeDKh+M4jlOFGTNmZB1OuymsX7+ezz//nBUrVuStTqewuPLhNDhE5FARqRSRPoWWJR+IyJsiMrKWPNeKSPodnfKAiOwT5GiVwzKnicj9uSqvoRDu38tj3weJyOmFlClTVq5cyaeffrpJIbV1Yfbs2fXqX+I0LFz5cJzNg78ChxVYhn2Aa4CcKR/AQODPOSyvodAXeDr2fRCwWSgfS5YsqZfoltpYs2YNU6dOzXu9TmHwfT4cJw0i0lJVVxVaDgBVnQnMLLQcuUZVPyy0DPWBqr5faBnqyooVK1i5cmVB6p43bx5r1qyhWbNmBanfyR+ufDh5RUQOAH4HlAJtga+B+1T10RTZO4nIMOBooAy4V1VvipV1LXA5cADwENAHmA5craovJOo9B7gE2BmYBzwJXKOqFeH4IOBx4LvAtcDBwLMi8iTwRpDhTOAYYEU491EROQv4PdARGAWcrarLQ5ktgduAI4AdgQXA68BvVHVxlv12LXC5qhaH74cGuY7AZtU/ApaGNlyrquuz6SMR+UHon32AlsBk4AZV/UeifwAWiAjAdFXtHo53BW4Bfgi0AT4K8r5TS7umASNV9YLwfXCQ8dfAHYCEss4EZgH3A8eHtt6iqg/FytqUc98EylR1QCytOzAVOFlVh8XlBT4DrgQ6AO8C56jqjNi5ldh1/nMo+5BYOsB1wDLgRqBLdM+EPCXYNfqpqj5TU//VB/PmzavTksuUKVMYP348ZWVlFBcXU1paSs+ePbMqY8WKFaxYsYKOHTtmXb+zeeHLLk6+2QkYB5wNDACGAPeJyK9S5H0EmAaciJmwbxSR8xJ5mgLPhuPHY8rMs2HgAEBELgQexgbr44AHsQH54RR1PosNJscCj8XSHwK+AE4AxgCPiMgtwE+Ai7CB+4fADbFzWgb5/oApL1cB+2NKSq54GOujgVhf/oHq5v1a+wjoDrwC/DSU9W9ghIgcE46/wsa2HY0tKwwEEJF2WJ+VAheHOmYBo0WkRx3a1AW4G1NmTgM6A88F+WcCJwGvAQ+KSO8cnpspx4ZyLgTOA3qzUTFLxfnAh1gf9Q2vvwJPYP/BpyXyn4kpJsPrKN8msXhxVnoxYIrH2LFjNzxorqysjLFjxzJlypSsylm1ahWrV6/Oun5n88MtH05eUdWh0WcRKQLGAp2Ac4EHEtnfUNXIae/VMLu+WkQeiWb2QDPgKlUdGcqcgFk2jgfuFpHGwB+B51X1/FhZ64FbRORGVf0mVuejCevKoeHjC6p6XUh7F1NCfgH0UNWVIb03cDKmjBCsG+fFymoCTAI+FJHeqjox036rgRGqenX4/LqIHIkNjE/F8tTYR0HWuBWgEfAmsCt2XV5R1QUi8nXIMkFVF8bKvxiz/Iiqzg1lvIZZB34P/DLLNnUADlPVT0NZ7bDB/V1V/X1IexNTSk8GJubo3ExpDAxQ1W9DedsBd4lIO1VdmsysqpNEZDlmWamyHCMiLwBnERTh8Jv4BfB0VH4sbzugXaL4ktqEPemkkzJrVWD+/PmUl5enPNa3b18Axo0bVyV93rx5rF+/vkpaRUUFb731VlZ7hTRq1Ig333yT5s2bZyXzsGHDssrvFB63fDh5RUTai8g9wXy9NrwuAXZLkX1E4vswYHuq/uGuB0ZHX1R1ETA/lqcXsC02+43zHFAE9Eukv5RG9NdidZRjM/v3IsUj8CXQWUSaRgki8lMRmSgiK7C2Rj4OqdpbF15NfJ9E9QGptj5CREpEZLCIzGTjdRmYoZxHYlalhSLSJChZjTAL0f7ZNQeAOZHyEIg2gIhfg7WYxWeHHJ6bKW8lFINJ4b1WRSAFjwClIrJ3+N4fs0L9NUXei7FloPhrbB3qrJGioqKsz0kqHrWl57JuZ/PELR9OvhmMDfjXYzPj5cDPgQtS5E1uNDAvvHcF/hc+r1LVpJ12DRA9qap9eE/G8EVldUiTnmRJijqWpkgrApoDa0VkIPB3bCC5GliEzVxHxeTbVFLJlSy7xj4Klo6XsL66DvgK87G5CtgvAxk6YUsJa1Mcq8u2lanaBKn7O9nWTTk3U9LVkXV5qvqWiChm/bgYsxJNUNWPU2S/G/v9xCmhFgUkW6vAqFGjeP/9mv1lBwwYUOX7kCFDNiy5xCkuLq6WtyY6dOjAj3/8Yzp37pzxOc7miSsfTt4QkRaYn8dlqnpvLP3naU7ZLvE9+keak0W10QJ28t+sc+J4RC43NzgZ+FhVz44SRCSTwTzf9AT2BQaq6otRoohkGnKwGLPAXJ3i2ObyxLBvseWpOEnFtL54FLhKRG7FlsIuSZUpLOksjacFx9+c0qVLl6zPKS0t3fCU2ogmTZpQWlqaVTnFxcUUFxdnXb+z+eHKh5NPmmPm+A2z8KCQnJgm/0CqLr2cBMwmu5BTxaJMTqGqA98pmKJRYzTGJtKSWFsDZ9RjfXWlZXiPX5fO2BJA3PqUboY/GvgZMFlVC7oR2iYwAzhKRBrF/ImOymH5NVlansCiXp7FlsjyHuESp3379jRv3jwrx88oqmVTo1223XZbWrdundU5zuaJKx9O3lDVZSIyHpvlLcIGu8uwWWcqDhOR27HB7SgsKuBXscEhkzrXicj1WETNAuBlbJZ/PfC4qtbnrkajgQdCuOs7wOFYtE1DYzKm0N0W/DVaYk66c6nqF/ZFeL9QRIYD5cG/4k7s2rwtIvdg/hQdsbDX1ZGjbgNnKLbk8UBwAi3FQphzxRfAIBE5DlOgZ6vqbABVXSgiI4BTgSdVtaBPWGvTpg3t27fPerfRnj17Zq1sxGnUqBE77rhjnc93Ni/c4dTJN6djToB/w6JbXsPCWFNxLrYkMAILaf2Dqj6YbYWqen8o6whsj4YLsX0gkmG7ueZhbPfO87A27I5ZbxoUwR9kIFCOOeLeBNwL/DOR70NsD5TTgfcwRS6K6ukLjA/njgbuA/YK+Ro8qjoauBT4AfAPrD2DcljFbZgC+gTWT+ckjkdWuccoMB06dGC33XLlD5052267Ldtvv33e63UKQ1G+9+93HMdxqiIifwP6qWpWThzRRmhjxoyhpKQuwTapmTFjBkOHDs3rg9769evH4Ycf7hEvWwgzZ86kf//+YNsRTEsed8uH4zhOgRCRvUXkJ5hl7+4Ci7OBLl26sMcee+Stvo4dO7LHHnu44rEV4cqH4zhO4XgZW557Ctvzo0HQtGlT+vTpQ7du3eq9riZNmtC3b9+81OU0HNzh1HEcp0BEz8ZpiHTq1Il+/foxcuRIVq2qv+cr7rHHHm712Apxy4fjOI6Tkt12241DDjkk6+3OM2XXXXfl4IMPplWrVvVSvtNwccuH4ziOk5KmTZvSu7c9f+/tt9+u9syXxo0b06ZNG4qLi2nVqhWNGtl8trKykrVr1254Su2331aNpi8qKkJE6N+/P9tuu21+GuM0KFz5cBzHcdLSrFkz+vTpQ7t27Xj11VdZunQpbdu2pXPnzrRq1Yry8nLmzp3Ll19+yZo1a1i/fj1NmjShffv2dOvWjW7dulFZWcnixYuZO3cuzZs3Z7/99mPfffelffv2tQvgbJG48uE4juPUSJMmTejVqxfvvPMOjRo1oqSkhAkTJjB58mQWLVpETVs2tGjRgh49erDvvvvSr18/evbsSdeuXWnSxIefrRm/+o7jOE6trFu3joqKCsrLy3n//fdZuHAha9asoWnTpqxduzalAtKkSRMaNWrEkiVL+Oyzz1i+fDnNmzenXbt2tGnTpgCtcBoKrnw4juM4NVJWVsbEiRNZuHAhFRUVtGzZkh122IHtttuONWvWsHr1asrLy6moqKCyspJGjRrRsmVLWrRoQdOmTWnZsiWNGjWirKyMMWPGMH36dPr16+fbqW/FuPLhOI7jpGX58uWMHTuWiRMnsm5d1YcUN2/enObNm2dlxVi3bh2qypIlSzjqqKPYZZddci2ysxngobaO4zhOSlauXMk777zDhAkTqikem8r8+fMZNWoU06ZNy2m5zuaBKx+O4zhONdavX8/HH3/MhAkTWL8+4wdJZ8WCBQv497//zeLFi+ulfKfh4sqH4ziOU40ZM2bw3//+N+cWj1T1TJw4kYqKinqtx2lYuPLhOI7jVKG8vJxx48axdOnSeq+rsrKSiRMnMn369Hqvy2k4uPLhOI7jVGHOnDlMnTo1b/WVl5fzySefuPVjK8KVD8dxHGcD69at4/PPP2f16tV5rXf69OnMnz8/r3U6hcOVD6fBISKHikiliPQptCz5QETeFJGRteS5VkTK8iVTGhn2CXLk7ClgIjJNRO7PVXkNhXD/Xh77PkhETi+kTJmyaNGigiyBLF26lP/97395r9cpDL7Ph+NsHvwVeKXAMuwDXAPcD5TXnDVjBgJLclRWQ6IvEB/BBwFlwDMFkSYLVqxYwbJlywpS98yZM1m/fv2GB9Q5Wy6ufDhOGkSkpaquKrQcAKo6E5hZaDlyjap+WGgZ6gNVfb/QMtSVRYsW1cn3YsqUKYwfP56ysjKKi4spLS2lZ8+eWZWxZMkSysrK2GabbbKu39m8cOXDySsicgDwO6AUaAt8Ddynqo+myN5JRIYBR2OzxntV9aZYWdcClwMHAA8BfbDZ5tWq+kKi3nOAS4CdgXnAk8A1qloRjg8CHge+C1wLHAw8KyJPAm8EGc4EjgFWhHMfFZGzgN8DHYFRwNmqujyU2RK4DTgC2BFYALwO/EZVs9rYIGqrqhaH74cGuY7AZtU/ApaGNlyrquuz6SMR+UHon32AlsBk4AZV/UeifwAWiAjAdFXtHo53BW4Bfgi0AT4K8r5TS7umASNV9YLwfXCQ8dfAHYCEss4EZmFWl+NDW29R1YdiZW3KuW8CZao6IJbWHZgKnKyqw+LyAp8BVwIdgHeBc1R1RuzcSuw6/zmUfUgsHeA6YBlwI9AlumdCnhLsGv1UVfNuKZk3b17W50yZMoWxY8duUFrKysoYO3YsQFYKyMqVK1m5cqUrH1sBrnw4+WYnYBzwCGa6/y5wn4g0U9UHEnkfAZ4DTgSOBG4UkcWq+pdYnqbAs9jAcgNwEaY07Kqq0wBE5ELgXmzwvRjYF/vz7wKclajzWeAx4HbgW6BZSH8IeCIc+xnwiIjsgg3qFwHbAffEZAAbxJsCf8AUnhLgKkxJ2T/D/qqNh4Eh2PLFkaGuL4GnYnlq7SOgO7ascwewDlNqRojIsar6Sjh2A3A1pogtA1YDiEg7bAD+FuvfRcDZwGgR2UNVsw2b6ALcjQ3M3wJ3YvfBdOBT4CTgFOBBEfmPqk7M0bmZcizQC7gQU7TuwhSzw9PkPx+7HuWYIghmxSoHbgZOw65jxJlY/w6PFxL6uV2i7JLahD3ppJNqy1KF+fPnU16e3aravHnzqm1EVlFRwVtvvcXkyZMzLqdx48aMGTOG5s2bZ1X/sGHDssrvFB5XPpy8oqpDo88iUgSMBToB5wJJ5eMNVY3+rF8Ns+urReSRaGaPKQdXqerIUOYEbKA/HrhbRBoDfwSeV9XzY2WtB24RkRtV9ZtYnY8mrCuHho8vqOp1Ie1d4ATgF0APVV0Z0nsDJxOUj2DdOC9WVhNgEvChiPSu48CXZISqXh0+vy4iR2IDbFz5qLGPgqxxK0Aj4E1gV+y6vKKqC0Tk65BlgqoujJV/MWb5EVWdG8p4DbMO/B74ZZZt6gAcpqqfhrLaYYP7u6r6+5D2JqaUngxMzNG5mdIYGKCq34bytgPuEpF2qro0mVlVJ4nIcsyyUmU5RkRewBTgh8P3Iuy+ejoqP8bFmM9Nwdh1110B+Oqrr6qkp9sBNdudUVM9GdfZMnHlw8krItIeW9b4ETZraxwOpYrrG5H4PgybJZYAkVv8emB0lEFVF4nIfDbOCHsB22Kz3zjPAbcC/YC48vFSGtFfi9VRLiKzgEmR4hH4EugsIk1VdW1o70+x5YxdgeJY3t2o28CX5NXE90mh7Di19VFk6r8Bm713ZWMknGYgw5HYEtDCoGBFjCEsN2TJnEh5CHwZ3uPXYG1YAtkhh+dmylsJxWBSeC/BlnSy4RHgTRHZO8jdH7NC/TVF3ruBwYm0EkyBT0u2VoEXX3yRjz76qMY8YdltA0OGDKGsrHowVnFxMQMGDKiWno5tttmG0047ja5du2Z8jrN54sqHk28GYwP+9djMeDnwc+CCFHmTQf/RYnRXNiofq1Q1qbisAVqEz+3D+9w0ZXVIk54kGZGxhuoDzRqgCGgOrBWRgcDfsYHkamw5oh227NKC3JBKrmTZNfZRsHS8hPXVdcBXmI/NVcB+GcjQCYvuWJviWF3CJlK1CVL3d7Ktm3JupqSrI+vyVPUtEVHM+nExZiWaoKofp8i7lEQ7kkpALthuu+2yPqe0tLSKzwdAkyZNKC0tzaqc1q1b06JFrn4aTkPGlQ8nb4hIC2AAcJmq3htL/3maU5L/gp3D+5wsqo0cOzsn0jsnjkfk0u57MvCxqp4dJYhIJoN5vumJ+cEMVNUXo0QRaZb2jKosxiwwV6c4Vr8PBskdcf+eiKRiWl88ClwlIrdiS2GX5KnelHTq1InGjRtn9UyXyKl0U6Nd2rZt686mWwmufDj5pDlmzt8wCw8KyYlp8g+k6tLLScBssgs5VSzK5BSqOvCdgikaNUZjbCItqb6cdEY91ldXWob3+HXpjC0BxK1P6Wb4ozEn3MmqWtCN0DaBGcBRItIo5k90VA7Lr8nS8gTmIPsstkRW0L1AiouLadOmTdbPdenZs2fWykaSbt260bhx49ozOps9rnw4eUNVl4nIeGyWtwgb7C7DZp2pOExEbscGt6Mwf49fxQaHTOpcJyLXYxE1C4CXsVn+9cDjdYjEyIbRwAMh3PUdzJ/iuHqsr65MxhS624LPRkvMSXcuVXdB/iK8Xygiw4Hy4KdwJ3Zt3haRe4BpmANqH2B15KjbwBmKLXk8EJxAS7EQ5lzxBTBIRI7DFOjZqjobQFUXisgI4FTgSVUtzA5fgY4dO7L99tvn5aFycVq3bk337t3zWqdTOHwbOSffnI45Af4Ni255DQtjTcW52JLACOAnwB9U9cFsK1TV+0NZR2B7NFyIhZSeV9N5OeBh4M+hnhHA7pj1pkER/EEGYqGfzwE3YaHJ/0zk+xBzFj4deA9T5KKonr7A+HDuaOA+YK+Qr8GjqqOBS4EfAP/A2jMoh1XchimgT2D9dE7ieGSVeyyHddaJZs2asddee9GkSX7npjvssAOdOydXR50tlSIPbXIcxyksIvI3oJ+qZuVBGm2ENmbMGEpKat3yI2OWL1/OCy+8kLdnvDRr1ozjjz+ePfbYIy/1OfXPzJkz6d+/P9h2BNOSx33ZxXEcp0CIyN7A/8Mse78usDgb2Gabbfje977HggULst5wrC7stdde7LzzzvVej9Nw8GUXx3GcwvEytjz3FLbnR4Nh5513Zt9996WoqKhe69luu+0oLS31ENutDLd8OI7jFIjo2TgNkaZNm9KnTx8WLVqU1Rbp2dCmTRsOPvhg31RsK8QtH47jOE5K2rdvT//+/etlM7M2bdrQv39/9/PYSnHLh+M4jpOWTp06ceSRR9K6dWs++uijrJ/XkoquXbty0EEH0atXLxo18jnw1ogrH47jOJsvjQHmzk0+PSD37LnnnkyZMoVFixaxenWqRzHVTvPmzdl9993Ze++92WabbZg9e3aOpXQaCrF7MuWuca58OI7jbL50BTjjjIa4ca7jAHaPfp1MdOXDcRxn82U8cBD2vKN8PEcneoruQWT3mIPNiS29jflqX2NM8Rif6qBvMuY4juNkRLSpGWk2jtoS2NLb2FDa554+juM4juPkFVc+HMdxHMfJK658OI7jOI6TV1z5cBzHcTJlKXBdeN9SWcqW3calNID2ucOp4ziO4zh5xS0fjuM4juPkFVc+HMdxHMfJK658OI7jOI6TV3yHU8dxHKdGRGRX4D7gQGAV8CzwW1UtL6hgOUJEegKXA98F9gImq+pehZUqd4jIycAZwH5AB2y784eAh1V1058UWAdc+XAcx3HSIiLtgDeA6cBJwHbAnUAn4NTCSZZT9gSOAf6DrQhsaasCl2HX7zfAPOAw4F5g55CWd1z5cBzHcWriXKA9sI+qLgQQkQrgaRH5k6p+XlDpcsPLqvoPABEZDPQprDg551hVXRD7/oaIFAMXiMjVqlq3xxRvAluaduc4juPklh8CYyLFI/ACsBr4QWFEyi2FWnrIFwnFI+JDoAW2DJN3XPlwHMdxamJ3YFI8IcyUvwZ6FUQiJxccBCwG5heiclc+HMdxnJpoT+rdMJdQoFmzs2mISB/gTOAuVV1XCBlc+XAcx3GcrQQR6YItm/0XuLVQcrjy4TiO49TEEqBdivT2mNne2UwQkbbAv4By4DhVXVsoWVz5cBzHcWriC8zvYwMi0hzYBZhcEImcrBGRFsBLWKj00aq6qJDyuPLhOI7j1MQ/gf4i0jGWNhBoHo45DRwRaQIMBb4D/EBVpxdYJH+qreM4jpOesMnYZ8A04E9s3GRsjKpuEZuMiUgrLKQY4FeYVefS8H18QxisNwUReRg4B7gCGJs4PElVl+dbJt9kzHEcx0mLqi4Vke9jO2IOZ+P26lcUVLDcsh3wfCIt+n4mMDiv0uSeo8L7bSmOHQa8mT9RDLd8OI7jOI6TV9znw3Ecx3GcvOLKh+M4juM4ecWVD8dxHMdx8oorH47jOI7j5BVXPhzHcRzHySuufDiO4ziOk1dc+XAcx9mCEZFBIlIpIt3zUFdrEZkrImfVd12Jem8Xkf/ks05n0/BNxhzHcZxc8WtgLfBk8oCI7Aj8Djga6AosA94FblfV9xJ5uwNTY0mVwFLgP8D1qjouUfydwIUicpyqvpSTluQAEekE3AQciz2cbwpwt6r+NZFvEPB4mmK6qurckK8I+CO2W2lTYAjwG1VdEyurMTABeFZVb8lS3gHAecD+Qd4lWJ8/rqojQp7u2LW5Ktvy47jlw3Ecx9lkRKQpcDHwt/hgGI71Az4FTgOGAedjO6buDbwjIhemKfY54KfYLqMPAd8F3hCRveOZVHUO9tC03+SqPZuKiLTBtjI/HVMsLsW2qH9URNLJeS3W3vhraez4GZgC9xhwB9YvlyXKOA9ojSlkmcpaJCKPAC8DOwD3hXLuBNoCw0Xk9EzLywS3fDiO4zi5YADQCXuA2QZEpD2mcKwC+qnq17FjdwCvAneLyISkBQT4SFWfiuUfiz0S/v8wBSbOc8AwEdlVVb/KVngRqQTOVNXB2Z6bhnMBAX6oqv8KaQ+KyAjgOhEZrKoLEue8qqrv11DmAOBpVf1jkLklcBxwc/i+Lfb8nZ8lFcBauAQ4G7gf+LWqro8du1VEjsYsLTnDlQ/HcZytEBE5EbgK2BMoB14Dfquq/0vkOxm4DtgZWzb4IzbgHaqq3WNZjwfmqOrniarOBbpgA+LX8QOqukpEfg58Fco9uhaxo4ei7ZLi2OvY8szxwO21lJMPDgKWxhSPiKcxGX8E/DV5kohsA6xU1XUpymwJzIp9Xwy0in2/EXhfVUdmKmRQYH4HKHBJQvEAQFVHZVpepviyi+M4zlaGiPwEs0aAKSB/wWbV74bZc5TvGMyiUIENUMOxAXO/FMV+D/ggRfqxwLckLCIRqjoVeAf4fhgIa6J7eF+SopxlwNdAv1rKyBfNMaUuSZTWJ8Wx0ZgvTLmIjBQRSRwfD5wmIt8NS0/nAv8FEJHewM+wpa9s6Ad0BJ5R1Yosz60zbvlwHMfZigi+GX8GvgAOUtVVIX008AZwJXB5yH4zMAdbLlkR8o3BnoI6PVZmE8wa8UqKKvcAVFVX1yDWx8AhQE/MNySiVVCGGgO7stGPYRip+SbU1xBQ4EgR6amqU2Lph4T37WNp5diTc98AlmPK3aXAeyLSW1Wjvr4He0Jt5HD7OXBtcES9H7hPVb/MUs6ovz7J8rxNwpUPx3GcrYs+QGfgxkjxAFDVN0VkAnAMcLmIdMMcQm+LFI+Q7y0R+RTYJlZmB6CIFBYJoA2wIkV6nOj4Non0P4RXPN9lqppO+VgClNZSFyLSiqrLFRHFccsPgKourK28NDyKOW0OFZGLgRnADzB/FbAllKiOoVS1DL0oIq8Cb2Pt/2XIt0JEDgF6Ac2Az1V1rYj8DLMKHSUi22OWrP2wqJdzVXV2DXJGfV7bNcoprnw4juNsXewU3jXFsS+AkxL5pqTINwXonSK9KEXaCkwBqYnoeHIAfAx4FmgBHIotKVTWUE5RLccjrgCuSZF+X3gly4x8I9rGD0QhsKlQ1c9E5MfAw8BbIXkZcCFm5ahxsFfVd8LeJYcn0tcDk6LvIarmVuCKoJz8E7NWHYtZsZ7B+i4dy8N7bdcop7jy4TiO42wqi7BBv32KY18A+4pI8xqWXr6D7Q+SjFKZoqqvh88jRWQtcLuIjFXVVP4l7YFMLBV/x/xM4ozGHFVfS3POj6m+F0cqZWsDqvqiiLyCta85trxUEg5nsjwyA3MIrolrsOWmp0RkB+BAoIeqThORK4BvRKREVWemOf+L8L438GIGMuUEVz4cx3G2LiL/AaH6QNsL24sinq9nijKqpKnqOhH5CuiRIu9IoC9wMvBU8mDYtOog4PX4MlAabsacLG8gdWRMD8wPokZU9RtswI7LATAppuwkeRU4orayU9S1Flv+iOo5MnwcncHpOwPJcNwNiEgv4ALMJ6cyLJUBzE68bw+kUz7ewZarTheRm9JE2eQcj3ZxHMfZuvgAmAecKyItokQROQjzBxkJEPwEPgN+Ekz7Ub5DsFlykndJHcHxMDAfuE1EqignYSnjccyCcH1tgqvqcsyf4SgR2TdRVlvM6TW5V0hOUNU5qvp6/JVtGSLSGfgt8CEwJpbeKUXeH2J+GzWFud4LPKmqkXIzL7z3Cu+7h/ealodWYUpdL+CO4LyalOXIsPtpznDLh+M4zlZEcFD8Dbb0MFZEnsI2B7sI20Pi1lj23wH/wEJwH8e23L4AU0qKE0X/AzhTRPaM7/WhqotE5CQsEuZDEXkUM/V3AQZhVpRfp9hgLB33Yrt6XgWcEks/HFNi/pFhOfWOiHwBPI9ZkUqwbdGbAT9R1bhvynsi8iGmGC7D/Gl+gV2PG9KUPRBzrj0jSgtLLR8Ag0XkMcxR9T+xaJl0/BlTPn4NHCoiz2NWk05YdM33sZ1ac4ZbPhzHcbYyVPVJzLG0CFM2zgf+iZnvF8byvYxtid4Umx2fgG3prdjeHXFewSwcpyTSUdWxmN/D0HD8L9iumpOwcN+kk2dNss/FNuo6UUR2ix06BXhPVVM50haKD4GfAw9iUS6jgf1UdVIi33NYKPHvMIfXo7H9VErD1vFVCBarO4FrU+ySeirmzHoL5kxaq9KgqpWqeha28dlszLH3EUzJKwOOU9UhGbQ3Y4oqKzNxDHYcx3EcQ0Q+Ahao6hGJ9KuwQXaX4OuQL3m6Yg87O1VVX8xXvU7dccuH4ziOkxIRaRo2EIunHQr8P2yjsST3YMsKP6t34apyKfCJKx6bD275cBzHcVISIlFex6JUZmN+Aedhfgl7qeqiwknnbM64w6njOI6TjiVYmOgvMefDlZhvx5WueDibgls+HMdxHMfJK+7z4TiO4zhOXnHlw3Ecx3GcvOLKh+M4juM4ecWVD8dxHMdx8oorH47jOI7j5JX/D2z7ygDCHTi+AAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 216x576 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "plt.figure(figsize=(3,8))\n",
    "\n",
    "plt.errorbar(root_KO_df.loc[terms_plot]['log_OR'],np.arange(len(terms_plot)),\n",
    "            xerr=[np.subtract(root_KO_df.loc[terms_plot]['log_OR'],root_KO_df.loc[terms_plot]['log_OR_CI_lower']),\n",
    "                   np.subtract(root_KO_df.loc[terms_plot]['log_OR_CI_upper'],root_KO_df.loc[terms_plot]['log_OR'])],color='k',fmt='o')\n",
    "\n",
    "color_temp = root_KO_df.loc[terms_plot]['MPO_term_type'].map({'Neural tube':'red','neg_ctrl':'black', 'heart':'blue'})\n",
    "\n",
    "sns.scatterplot(x=root_KO_df.loc[terms_plot]['log_OR'],\n",
    "                y=np.arange(len(terms_plot)),size=root_KO_df.loc[terms_plot]['num_genes_in_term'],sizes=(200, 2000),\n",
    "                alpha=.5,\n",
    "               hue=color_temp.tolist(),palette={'blue':'blue','red':'red','black':'black'},legend=False)\n",
    "\n",
    "plt.yticks(np.arange(len(terms_plot)),root_KO_df.loc[terms_plot]['MP_description'])\n",
    "plt.xticks([0,1,2])\n",
    "plt.xlabel('log(OR) +- 95% CI')\n",
    "\n",
    "plt.plot([0,0],[-.8,len(terms_plot)-.5],'--',color='gray')\n",
    "plt.ylim([-0.8,len(terms_plot)-.5])\n",
    "\n",
    "plt.gca().invert_yaxis()\n",
    "# plt.savefig('ASD_CHD_brain_heart_MPO_root_enrichment_210813.png',dpi=300,bbox_inches='tight')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fc466aa2",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "id": "abae1571",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MPO_term_type</th>\n",
       "      <th>MP_description</th>\n",
       "      <th>OR_p</th>\n",
       "      <th>log_OR</th>\n",
       "      <th>log_OR_CI_lower</th>\n",
       "      <th>log_OR_CI_upper</th>\n",
       "      <th>num_genes_in_term</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>MP:0000929</th>\n",
       "      <td>Neural tube</td>\n",
       "      <td>open neural tube</td>\n",
       "      <td>0.000016</td>\n",
       "      <td>1.106295</td>\n",
       "      <td>0.604486</td>\n",
       "      <td>1.608104</td>\n",
       "      <td>240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MP:0001672</th>\n",
       "      <td>Neural tube</td>\n",
       "      <td>abnormal embryo development</td>\n",
       "      <td>0.000055</td>\n",
       "      <td>0.605600</td>\n",
       "      <td>0.311451</td>\n",
       "      <td>0.899750</td>\n",
       "      <td>1208</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MP:0004215</th>\n",
       "      <td>heart</td>\n",
       "      <td>abnormal myocardial fiber physiology</td>\n",
       "      <td>0.000183</td>\n",
       "      <td>1.187312</td>\n",
       "      <td>0.565353</td>\n",
       "      <td>1.809271</td>\n",
       "      <td>143</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MP:0002625</th>\n",
       "      <td>heart</td>\n",
       "      <td>heart left ventricle hypertrophy</td>\n",
       "      <td>0.000184</td>\n",
       "      <td>1.502543</td>\n",
       "      <td>0.715022</td>\n",
       "      <td>2.290065</td>\n",
       "      <td>68</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MP:0003720</th>\n",
       "      <td>Neural tube</td>\n",
       "      <td>abnormal neural tube closure</td>\n",
       "      <td>0.000290</td>\n",
       "      <td>0.831410</td>\n",
       "      <td>0.381740</td>\n",
       "      <td>1.281079</td>\n",
       "      <td>384</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MP:0002151</th>\n",
       "      <td>Neural tube</td>\n",
       "      <td>abnormal neural tube morphology</td>\n",
       "      <td>0.000349</td>\n",
       "      <td>0.701972</td>\n",
       "      <td>0.317257</td>\n",
       "      <td>1.086686</td>\n",
       "      <td>603</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MP:0004485</th>\n",
       "      <td>heart</td>\n",
       "      <td>increased response of heart to induced stress</td>\n",
       "      <td>0.000375</td>\n",
       "      <td>1.545706</td>\n",
       "      <td>0.693949</td>\n",
       "      <td>2.397463</td>\n",
       "      <td>56</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MP:0002621</th>\n",
       "      <td>Neural tube</td>\n",
       "      <td>delayed neural tube closure</td>\n",
       "      <td>0.001682</td>\n",
       "      <td>1.681851</td>\n",
       "      <td>0.632464</td>\n",
       "      <td>2.731237</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MP:0002085</th>\n",
       "      <td>Neural tube</td>\n",
       "      <td>abnormal embryonic tissue morphology</td>\n",
       "      <td>0.001692</td>\n",
       "      <td>0.512033</td>\n",
       "      <td>0.192373</td>\n",
       "      <td>0.831693</td>\n",
       "      <td>1078</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MP:0000266</th>\n",
       "      <td>heart</td>\n",
       "      <td>abnormal heart morphology</td>\n",
       "      <td>0.002642</td>\n",
       "      <td>0.395950</td>\n",
       "      <td>0.137831</td>\n",
       "      <td>0.654068</td>\n",
       "      <td>1986</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MP:0003222</th>\n",
       "      <td>heart</td>\n",
       "      <td>increased cardiomyocyte apoptosis</td>\n",
       "      <td>0.002986</td>\n",
       "      <td>1.402181</td>\n",
       "      <td>0.476588</td>\n",
       "      <td>2.327774</td>\n",
       "      <td>53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MP:0003920</th>\n",
       "      <td>heart</td>\n",
       "      <td>abnormal heart right ventricle morphology</td>\n",
       "      <td>0.003636</td>\n",
       "      <td>0.914569</td>\n",
       "      <td>0.298179</td>\n",
       "      <td>1.530959</td>\n",
       "      <td>184</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MP:0001698</th>\n",
       "      <td>Neural tube</td>\n",
       "      <td>decreased embryo size</td>\n",
       "      <td>0.003839</td>\n",
       "      <td>0.585446</td>\n",
       "      <td>0.188556</td>\n",
       "      <td>0.982335</td>\n",
       "      <td>624</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MP:0010563</th>\n",
       "      <td>heart</td>\n",
       "      <td>increased heart right ventricle size</td>\n",
       "      <td>0.004494</td>\n",
       "      <td>1.123248</td>\n",
       "      <td>0.348407</td>\n",
       "      <td>1.898089</td>\n",
       "      <td>96</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MP:0011925</th>\n",
       "      <td>heart</td>\n",
       "      <td>abnormal heart echocardiography feature</td>\n",
       "      <td>0.005589</td>\n",
       "      <td>1.302817</td>\n",
       "      <td>0.381311</td>\n",
       "      <td>2.224323</td>\n",
       "      <td>58</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MP:0014134</th>\n",
       "      <td>Neural tube</td>\n",
       "      <td>abnormal embryo morphology</td>\n",
       "      <td>0.007420</td>\n",
       "      <td>0.387647</td>\n",
       "      <td>0.103871</td>\n",
       "      <td>0.671422</td>\n",
       "      <td>1585</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MP:0000276</th>\n",
       "      <td>heart</td>\n",
       "      <td>heart right ventricle hypertrophy</td>\n",
       "      <td>0.007499</td>\n",
       "      <td>1.411069</td>\n",
       "      <td>0.376739</td>\n",
       "      <td>2.445399</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MP:0003221</th>\n",
       "      <td>heart</td>\n",
       "      <td>abnormal cardiomyocyte apoptosis</td>\n",
       "      <td>0.007823</td>\n",
       "      <td>1.247593</td>\n",
       "      <td>0.328197</td>\n",
       "      <td>2.166990</td>\n",
       "      <td>61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MP:0004484</th>\n",
       "      <td>heart</td>\n",
       "      <td>abnormal response of heart to induced stress</td>\n",
       "      <td>0.009189</td>\n",
       "      <td>1.026424</td>\n",
       "      <td>0.254145</td>\n",
       "      <td>1.798703</td>\n",
       "      <td>105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MP:0002419</th>\n",
       "      <td>neg_ctrl</td>\n",
       "      <td>abnormal innate immunity</td>\n",
       "      <td>0.466837</td>\n",
       "      <td>0.177502</td>\n",
       "      <td>-0.300619</td>\n",
       "      <td>0.655622</td>\n",
       "      <td>605</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           MPO_term_type                                 MP_description  \\\n",
       "MP:0000929   Neural tube                               open neural tube   \n",
       "MP:0001672   Neural tube                    abnormal embryo development   \n",
       "MP:0004215         heart           abnormal myocardial fiber physiology   \n",
       "MP:0002625         heart               heart left ventricle hypertrophy   \n",
       "MP:0003720   Neural tube                   abnormal neural tube closure   \n",
       "MP:0002151   Neural tube                abnormal neural tube morphology   \n",
       "MP:0004485         heart  increased response of heart to induced stress   \n",
       "MP:0002621   Neural tube                    delayed neural tube closure   \n",
       "MP:0002085   Neural tube           abnormal embryonic tissue morphology   \n",
       "MP:0000266         heart                      abnormal heart morphology   \n",
       "MP:0003222         heart              increased cardiomyocyte apoptosis   \n",
       "MP:0003920         heart      abnormal heart right ventricle morphology   \n",
       "MP:0001698   Neural tube                          decreased embryo size   \n",
       "MP:0010563         heart           increased heart right ventricle size   \n",
       "MP:0011925         heart        abnormal heart echocardiography feature   \n",
       "MP:0014134   Neural tube                     abnormal embryo morphology   \n",
       "MP:0000276         heart              heart right ventricle hypertrophy   \n",
       "MP:0003221         heart               abnormal cardiomyocyte apoptosis   \n",
       "MP:0004484         heart   abnormal response of heart to induced stress   \n",
       "MP:0002419      neg_ctrl                       abnormal innate immunity   \n",
       "\n",
       "                OR_p    log_OR  log_OR_CI_lower  log_OR_CI_upper  \\\n",
       "MP:0000929  0.000016  1.106295         0.604486         1.608104   \n",
       "MP:0001672  0.000055  0.605600         0.311451         0.899750   \n",
       "MP:0004215  0.000183  1.187312         0.565353         1.809271   \n",
       "MP:0002625  0.000184  1.502543         0.715022         2.290065   \n",
       "MP:0003720  0.000290  0.831410         0.381740         1.281079   \n",
       "MP:0002151  0.000349  0.701972         0.317257         1.086686   \n",
       "MP:0004485  0.000375  1.545706         0.693949         2.397463   \n",
       "MP:0002621  0.001682  1.681851         0.632464         2.731237   \n",
       "MP:0002085  0.001692  0.512033         0.192373         0.831693   \n",
       "MP:0000266  0.002642  0.395950         0.137831         0.654068   \n",
       "MP:0003222  0.002986  1.402181         0.476588         2.327774   \n",
       "MP:0003920  0.003636  0.914569         0.298179         1.530959   \n",
       "MP:0001698  0.003839  0.585446         0.188556         0.982335   \n",
       "MP:0010563  0.004494  1.123248         0.348407         1.898089   \n",
       "MP:0011925  0.005589  1.302817         0.381311         2.224323   \n",
       "MP:0014134  0.007420  0.387647         0.103871         0.671422   \n",
       "MP:0000276  0.007499  1.411069         0.376739         2.445399   \n",
       "MP:0003221  0.007823  1.247593         0.328197         2.166990   \n",
       "MP:0004484  0.009189  1.026424         0.254145         1.798703   \n",
       "MP:0002419  0.466837  0.177502        -0.300619         0.655622   \n",
       "\n",
       "            num_genes_in_term  \n",
       "MP:0000929                240  \n",
       "MP:0001672               1208  \n",
       "MP:0004215                143  \n",
       "MP:0002625                 68  \n",
       "MP:0003720                384  \n",
       "MP:0002151                603  \n",
       "MP:0004485                 56  \n",
       "MP:0002621                 33  \n",
       "MP:0002085               1078  \n",
       "MP:0000266               1986  \n",
       "MP:0003222                 53  \n",
       "MP:0003920                184  \n",
       "MP:0001698                624  \n",
       "MP:0010563                 96  \n",
       "MP:0011925                 58  \n",
       "MP:0014134               1585  \n",
       "MP:0000276                 42  \n",
       "MP:0003221                 61  \n",
       "MP:0004484                105  \n",
       "MP:0002419                605  "
      ]
     },
     "execution_count": 100,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "root_KO_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "05034267",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "adfa9906",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e87b8441",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7a3fbff4",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ec7fed7d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "95e7f6f3",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6c049def",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "18684113",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f70753b6",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9b3acfc6",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d0462c2e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4a097b89",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "03bd4c71",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "17dc49fb",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0eaf3f62",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5167c7c2",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4fabed98",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e6ef3ecb",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "114a23d0",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
