# Codes and notebooks for the Gleesonlab Meningomyelocele de novo mutation project

This repository contains codes, scripts and notebooks for the Meningomyelocele de novo mutation project.

-------------------------------------

# Pipelines for data processing

The pipeline to perform processing of the WXS sequencing data can be found in [https://github.com/Gleeson-Lab/wxs_pipeline](https://github.com/Gleeson-Lab/wxs_pipeline).

Quality control for kinship analysis can be found in [https://github.com/Gleeson-Lab/MM_DNM/tree/main/DNM/Kindship](https://github.com/Gleeson-Lab/MM_DNM/tree/main/DNM/Kindship).

-------------------------------------

# Pipeline for de novo variant calling

De novo SNV detection from HaplotypeCaller to VQSR, variant annotation can be found in [https://github.com/Gleeson-Lab/MM_DNM/tree/main/DNM](https://github.com/Gleeson-Lab/MM_DNM/tree/main/DNM).

Pipelines for computing the consensus regions can be found in [https://github.com/Gleeson-Lab/MM_DNM/tree/main/Consensus_region](https://github.com/Gleeson-Lab/MM_DNM/tree/main/Consensus_region).

De novo variant calling from WXS and WGS can be found in [https://github.com/Gleeson-Lab/WGS_DNM](https://github.com/Gleeson-Lab/WGS_DNM).

-------------------------------------

# Pipeline for spatial transcriptomic

Notebooks and scripts for spatial transcriptomic analysis with MERFISH can be found in [https://github.com/Gleeson-Lab/MM_DNM/tree/main/MERFISH](https://github.com/Gleeson-Lab/MM_DNM/tree/main/MERFISH).

-------------------------------------

# Codes for statistical modeling and plotting

Code to conduct the DNM burden analysis can be found in [https://github.com/Gleeson-Lab/MM_DNM/tree/main/DNM_burden_analysis](https://github.com/Gleeson-Lab/MM_DNM/tree/main/DNM_burden_analysis).

Code to conduct the network colocalization can be foudn in [https://github.com/Gleeson-Lab/MM_DNM/tree/main/Network](https://github.com/Gleeson-Lab/MM_DNM/tree/main/Network), and network propagation analysis can be found in [https://github.com/Gleeson-Lab/MM_DNM/tree/main/Network%20propagation](https://github.com/Gleeson-Lab/MM_DNM/tree/main/Network%20propagation)

Code to plot MERFISH data can be found in [https://github.com/Gleeson-Lab/MM_DNM/blob/main/MERFISH/plot_cell_comp.r](https://github.com/Gleeson-Lab/MM_DNM/blob/main/MERFISH/plot_cell_comp.r).

Code to plot functional experiments can be found in [https://github.com/Gleeson-Lab/MM_DNM/tree/main/Functional%20experiment](https://github.com/Gleeson-Lab/MM_DNM/tree/main/Functional%20experiment).

-------------------------------------

# Contact

:email: Joseph Gleeson: [jogleeson@health.ucsd.edu](mailto:jogleeson@health.ucsd.edu) 
