{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "b5e5bcdb",
   "metadata": {},
   "outputs": [],
   "source": [
    "library(\"tidyverse\")\n",
    "library(reshape)\n",
    "library(ggbeeswarm)\n",
    "library(\"tidyr\")\n",
    "library(\"ggplot2\")\n",
    "library(\"reshape2\")\n",
    "library(\"dplyr\")\n",
    "library(\"ggpubr\")\n",
    "theme_set(theme_bw(base_size = 10)) \n",
    "\n",
    "\n",
    "SE <- function(values){\n",
    "    se = sd(values) / sqrt(length(values))\n",
    "    return (se)\n",
    "}\n",
    "\n",
    "CI <- function(values){\n",
    "    t = qt((1-0.05)/2 + 0.5, length(values)-1)\n",
    "    ci = t*SE(values)\n",
    "    return (as.numeric(ci))\n",
    "}\n",
    "options(repr.matrix.max.rows=200, repr.matrix.max.cols=200)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "890ebffb",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "percent= '70'\n",
    "cutoff = 7\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bc98e488",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#Ped (trio) that are used for analysis, after sample filtration (kindships, contamination, and Poission)\n",
    "ped = read.delim('NTD_WES_DNM.FixedSample.PoiFil.ped',sep='\\t')\n",
    "\n",
    "\n",
    "# PLI scores\n",
    "oe = read.delim('/projects/ps-gleesonlab8/User/hiyoothere/NTD/WorkFlow/Others/gnomad.v2.1.1.lof_metrics.by_gene.txt')\n",
    "colnames(oe)[1] = \"SYMBOL\"\n",
    "dim(oe)\n",
    "oe = oe[!duplicated(oe$SYMBOL), ]\n",
    "head(oe)\n",
    "oe$gene = factor(oe$SYMBOL, levels=oe$SYMBOL)\n",
    "oe = oe[, c(1,10,21,22,23,24,29,30)]\n",
    "oe_unique <- unique(oe)\n",
    "\n",
    "\n",
    "#DNM all \n",
    "data.all = read.delim(\"NTD_WES_DNM.FixedSample.PoiFil.den_input\", header = TRUE)\n",
    "paste(\"Total DNM number:\", dim(data.all)[1])\n",
    "#DNM in consensus region\n",
    "data.con = read.delim(paste(\"NTD_WES_DNM.FixedSample.C\",percent,\".PoiFil.den_input\", sep=\"\"), header = TRUE)\n",
    "paste(\"DNMs in consensus:\", dim(data.con)[1])\n",
    "\n",
    "#FIL\n",
    "data.con <- left_join(data.con, batch, by = \"pro\")\n",
    "dim(data.con)\n",
    "\n",
    "##add pli\n",
    "dim(data.con)\n",
    "data.con <- left_join(data.con, oe_unique, by= 'SYMBOL')\n",
    "\n",
    "data.all <- left_join(data.all, fa, by = \"family\")\n",
    "data.con <- left_join(data.con, fa, by = \"family\")\n",
    "head(data.all)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6a2ebc42",
   "metadata": {},
   "source": [
    "## Clean the vep annotation terms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "aea74043",
   "metadata": {},
   "outputs": [],
   "source": [
    "data.fil <- data.con\n",
    "\n",
    "data.fil$Consequence[grepl(\"stop_gain\", data.fil$Consequence)] <- \"stop_gain\"\n",
    "data.fil$Consequence[grepl(\"stop_lost\", data.fil$Consequence)] <- \"stop_lost\"\n",
    "data.fil$Consequence[grepl(\"start_lost\", data.fil$Consequence)] <- \"start_lost\"\n",
    "data.fil$Consequence[grepl(\"inframe_deletion\", data.fil$Consequence)] <- \"inframe_deletion\"\n",
    "data.fil$Consequence[grepl(\"inframe_insertion\", data.fil$Consequence)] <- \"inframe_insertion\"\n",
    "data.fil$Consequence[grepl(\"splice_acceptor_variant\", data.fil$Consequence)] <- \"splice_acceptor_variant\"\n",
    "data.fil$Consequence[grepl(\"splice_donor_variant\", data.fil$Consequence)] <- \"splice_donor_variant\"\n",
    "data.fil$Consequence[grepl(\"missense\", data.fil$Consequence)] <- \"missense_variant\"\n",
    "data.fil$Consequence[grepl(\"synonymous\", data.fil$Consequence)] <- \"synonymous_variant\"\n",
    "\n",
    "\n",
    "coding_term = c('frameshift_variant', 'inframe_deletion', 'inframe_insertion', \n",
    "                'synonymous_variant', 'missense_variant', \n",
    "                'splice_acceptor_variant', 'splice_donor_variant', \n",
    "                'start_lost', \n",
    "                'stop_gained', 'stop_lost', 'stop_retained_variant', \"protein_altering_variant\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "79df1445",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 142</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>locus</th><th scope=col>ref</th><th scope=col>alt</th><th scope=col>family</th><th scope=col>pro</th><th scope=col>dad</th><th scope=col>mom</th><th scope=col>pro_sex</th><th scope=col>pro_DP</th><th scope=col>pro_ref</th><th scope=col>pro_alt</th><th scope=col>pro_GT</th><th scope=col>pro_GQ</th><th scope=col>dad_DP</th><th scope=col>dad_ref</th><th scope=col>dad_alt</th><th scope=col>dad_GT</th><th scope=col>dad_GQ</th><th scope=col>mom_DP</th><th scope=col>mom_ref</th><th scope=col>mom_alt</th><th scope=col>mom_GT</th><th scope=col>mom_GQ</th><th scope=col>p_denovo</th><th scope=col>confidence</th><th scope=col>MQ</th><th scope=col>QD</th><th scope=col>filter</th><th scope=col>Allele</th><th scope=col>Consequence</th><th scope=col>IMPACT</th><th scope=col>SYMBOL</th><th scope=col>Gene</th><th scope=col>Feature_type</th><th scope=col>Feature</th><th scope=col>BIOTYPE</th><th scope=col>EXON</th><th scope=col>INTRON</th><th scope=col>HGVSc</th><th scope=col>HGVSp</th><th scope=col>cDNA_position</th><th scope=col>CDS_position</th><th scope=col>Protein_position</th><th scope=col>Amino_acids</th><th scope=col>Codons</th><th scope=col>Existing_variation</th><th scope=col>DISTANCE</th><th scope=col>STRAND</th><th scope=col>FLAGS</th><th scope=col>VARIANT_CLASS</th><th scope=col>SYMBOL_SOURCE</th><th scope=col>HGNC_ID</th><th scope=col>CANONICAL</th><th scope=col>MANE_SELECT</th><th scope=col>MANE_PLUS_CLINICAL</th><th scope=col>TSL</th><th scope=col>APPRIS</th><th scope=col>CCDS</th><th scope=col>ENSP</th><th scope=col>SWISSPROT</th><th scope=col>TREMBL</th><th scope=col>UNIPARC</th><th scope=col>UNIPROT_ISOFORM</th><th scope=col>GENE_PHENO</th><th scope=col>SIFT</th><th scope=col>PolyPhen</th><th scope=col>DOMAINS</th><th scope=col>miRNA</th><th scope=col>HGVS_OFFSET</th><th scope=col>AF</th><th scope=col>AFR_AF</th><th scope=col>AMR_AF</th><th scope=col>EAS_AF</th><th scope=col>EUR_AF</th><th scope=col>SAS_AF</th><th scope=col>AA_AF</th><th scope=col>EA_AF</th><th scope=col>gnomAD_AF</th><th scope=col>gnomAD_AFR_AF</th><th scope=col>gnomAD_AMR_AF</th><th scope=col>gnomAD_ASJ_AF</th><th scope=col>gnomAD_EAS_AF</th><th scope=col>gnomAD_FIN_AF</th><th scope=col>gnomAD_NFE_AF</th><th scope=col>gnomAD_OTH_AF</th><th scope=col>gnomAD_SAS_AF</th><th scope=col>MAX_AF</th><th scope=col>MAX_AF_POPS</th><th scope=col>CLIN_SIG</th><th scope=col>SOMATIC</th><th scope=col>PHENO</th><th scope=col>PUBMED</th><th scope=col>MOTIF_NAME</th><th scope=col>MOTIF_POS</th><th scope=col>HIGH_INF_POS</th><th scope=col>MOTIF_SCORE_CHANGE</th><th scope=col>TRANSCRIPTION_FACTORS</th><th scope=col>CADD_phred</th><th scope=col>DisGeNET</th><th scope=col>MPC</th><th scope=col>MTR</th><th scope=col>Mastermind</th><th scope=col>MetaSVM_pred</th><th scope=col>MetaSVM_rankscore</th><th scope=col>Phenotypes</th><th scope=col>Polyphen2_HDIV_score</th><th scope=col>SIFT_pred</th><th scope=col>SplicAI</th><th scope=col>SpliceRegion</th><th scope=col>gnomAD_exomes_AC</th><th scope=col>gnomAD_exomes_AF</th><th scope=col>gnomAD_exomes_AN</th><th scope=col>pLI.x</th><th scope=col>pLI_values</th><th scope=col>occurence</th><th scope=col>probands</th><th scope=col>DP_filter12</th><th scope=col>pval</th><th scope=col>Inspected</th><th scope=col>batch.x</th><th scope=col>oe_mis_pphen.x</th><th scope=col>pLI.y</th><th scope=col>pNull.x</th><th scope=col>pRec.x</th><th scope=col>oe_lof.x</th><th scope=col>oe_lof_lower.x</th><th scope=col>oe_lof_upper.x</th><th scope=col>batch.y</th><th scope=col>batch</th><th scope=col>oe_mis_pphen.y</th><th scope=col>pLI</th><th scope=col>pNull.y</th><th scope=col>pRec.y</th><th scope=col>oe_lof.y</th><th scope=col>oe_lof_lower.y</th><th scope=col>oe_lof_upper.y</th><th scope=col>pro_dob</th><th scope=col>fa_AoC</th><th scope=col>mo_AoC</th><th scope=col>fa_dob</th><th scope=col>mo_dob</th><th scope=col>pro_FA</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>chr1:1290349 </td><td>G</td><td>A</td><td>12111</td><td>12111.s1  </td><td>12111.fa  </td><td>12111.mo  </td><td>2</td><td> 33</td><td>19</td><td>14</td><td>0/1</td><td>99</td><td>40</td><td>40</td><td>0</td><td>0/0</td><td>76</td><td>40</td><td>40</td><td>0</td><td>0/0</td><td>67</td><td>.</td><td>hiConfDeNovo</td><td>60.00</td><td>11.4 </td><td>PASS</td><td>A</td><td>downstream_gene_variant           </td><td>MODIFIER</td><td>ACAP3 </td><td>ENSG00000131584</td><td>Transcript</td><td>ENST00000492936</td><td>retained_intron     </td><td>     </td><td></td><td>                           </td><td>                             </td><td>    </td><td>    </td><td>   </td><td>   </td><td>       </td><td>            </td><td>2041</td><td>-1</td><td>          </td><td>SNV</td><td>HGNC</td><td>HGNC:16754</td><td>   </td><td>           </td><td></td><td> 1</td><td>  </td><td>        </td><td>               </td><td>          </td><td>         </td><td>             </td><td>        </td><td>NA</td><td>              </td><td>                        </td><td>                                 </td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>       NA</td><td>       NA</td><td>       NA</td><td>NA</td><td>       NA</td><td>NA</td><td>       NA</td><td>NA</td><td>NA</td><td>       NA</td><td>          </td><td></td><td> </td><td> </td><td></td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>  NA</td><td>             </td><td>             </td><td>             </td><td>             </td><td> </td><td>     NA</td><td>             </td><td>                   </td><td>       </td><td>             </td><td>             </td><td>NA</td><td>NA</td><td>NA</td><td>             </td><td>             </td><td>1</td><td>12111.s1             </td><td>TRUE</td><td>0.4868502</td><td>NA</td><td>SSC     </td><td>0.53448</td><td>4.4135e-06</td><td>1.4907e-05</td><td>0.99998000</td><td>0.39045</td><td>0.266</td><td>0.586</td><td>SSC     </td><td>SSC     </td><td>0.53448</td><td>4.4135e-06</td><td>1.4907e-05</td><td>0.99998000</td><td>0.39045</td><td>0.266</td><td>0.586</td><td>2000</td><td>29</td><td>26</td><td>1971</td><td>1974</td><td>None</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>chr1:1486163 </td><td>G</td><td>C</td><td>7004 </td><td>4926035813</td><td>2917406449</td><td>3160220249</td><td>1</td><td> 64</td><td>30</td><td>34</td><td>0/1</td><td>99</td><td>43</td><td>43</td><td>0</td><td>0/0</td><td>88</td><td>44</td><td>44</td><td>0</td><td>0/0</td><td>91</td><td>.</td><td>hiConfDeNovo</td><td>59.68</td><td>16.27</td><td>PASS</td><td>C</td><td><span style=white-space:pre-wrap>missense_variant                  </span></td><td>MODERATE</td><td>ATAD3B</td><td>ENSG00000160072</td><td>Transcript</td><td>ENST00000673477</td><td><span style=white-space:pre-wrap>protein_coding      </span></td><td>10/16</td><td></td><td>ENST00000673477.1:c.1017G&gt;C</td><td>ENSP00000500094.1:p.Lys339Asn</td><td>1137</td><td>1017</td><td>339</td><td>K/N</td><td>aaG/aaC</td><td>COSV58019127</td><td><span style=white-space:pre-wrap>  NA</span></td><td> 1</td><td><span style=white-space:pre-wrap>          </span></td><td>SNV</td><td>HGNC</td><td>HGNC:24007</td><td>YES</td><td>NM_031921.6</td><td></td><td>NA</td><td>P1</td><td>CCDS30.1</td><td>ENSP00000500094</td><td>Q5T9A4.154</td><td><span style=white-space:pre-wrap>         </span></td><td>UPI000013E044</td><td>Q5T9A4-1</td><td>NA</td><td>deleterious(0)</td><td>possibly_damaging(0.824)</td><td>AFDB-ENSP_mappings:AF-Q5T9A4-F1.A</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td><span style=white-space:pre-wrap>       NA</span></td><td><span style=white-space:pre-wrap>       NA</span></td><td><span style=white-space:pre-wrap>       NA</span></td><td>NA</td><td><span style=white-space:pre-wrap>       NA</span></td><td>NA</td><td><span style=white-space:pre-wrap>       NA</span></td><td>NA</td><td>NA</td><td><span style=white-space:pre-wrap>       NA</span></td><td><span style=white-space:pre-wrap>          </span></td><td></td><td>1</td><td>1</td><td></td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>23.7</td><td>invalid_field</td><td>invalid_field</td><td>invalid_field</td><td>invalid_field</td><td>D</td><td>0.94306</td><td>invalid_field</td><td><span style=white-space:pre-wrap>0.955              </span></td><td><span style=white-space:pre-wrap>D      </span></td><td>invalid_field</td><td>invalid_field</td><td>NA</td><td>NA</td><td>NA</td><td>invalid_field</td><td>invalid_field</td><td>1</td><td><span style=white-space:pre-wrap>4926035813           </span></td><td>TRUE</td><td>0.7079808</td><td>NA</td><td><span style=white-space:pre-wrap>UCI     </span></td><td>1.34320</td><td>1.0032e-22</td><td>9.9972e-01</td><td>0.00027792</td><td>1.09950</td><td>0.829</td><td>1.477</td><td><span style=white-space:pre-wrap>UCI     </span></td><td><span style=white-space:pre-wrap>UCI     </span></td><td>1.34320</td><td>1.0032e-22</td><td>9.9972e-01</td><td>0.00027792</td><td>1.09950</td><td>0.829</td><td>1.477</td><td><span style=white-space:pre-wrap>  NA</span></td><td>NA</td><td>NA</td><td><span style=white-space:pre-wrap>  NA</span></td><td><span style=white-space:pre-wrap>  NA</span></td><td><span style=white-space:pre-wrap>NA  </span></td></tr>\n",
       "\t<tr><th scope=row>3</th><td>chr1:9356429 </td><td>G</td><td>A</td><td>6144 </td><td>7350647467</td><td>2898643437</td><td>4401829860</td><td>1</td><td>154</td><td>76</td><td>78</td><td>0/1</td><td>99</td><td>33</td><td>33</td><td>0</td><td>0/0</td><td>76</td><td>33</td><td>33</td><td>0</td><td>0/0</td><td>76</td><td>.</td><td>hiConfDeNovo</td><td>60.00</td><td>9.72 </td><td>PASS</td><td>A</td><td><span style=white-space:pre-wrap>missense_variant                  </span></td><td>MODERATE</td><td>SPSB1 </td><td>ENSG00000171621</td><td>Transcript</td><td>ENST00000450402</td><td><span style=white-space:pre-wrap>protein_coding      </span></td><td><span style=white-space:pre-wrap>2/2  </span></td><td></td><td>ENST00000450402.1:c.538G&gt;A </td><td>ENSP00000409235.1:p.Ala180Thr</td><td>754 </td><td>538 </td><td>180</td><td>A/T</td><td>Gcc/Acc</td><td>COSV60164629</td><td><span style=white-space:pre-wrap>  NA</span></td><td> 1</td><td>cds_end_NF</td><td>SNV</td><td>HGNC</td><td>HGNC:30628</td><td><span style=white-space:pre-wrap>   </span></td><td><span style=white-space:pre-wrap>           </span></td><td></td><td> 5</td><td><span style=white-space:pre-wrap>  </span></td><td><span style=white-space:pre-wrap>        </span></td><td>ENSP00000409235</td><td><span style=white-space:pre-wrap>          </span></td><td>A2A276.84</td><td>UPI0000EDA26A</td><td><span style=white-space:pre-wrap>        </span></td><td>NA</td><td>deleterious(0)</td><td><span style=white-space:pre-wrap>benign(0.031)           </span></td><td><span style=white-space:pre-wrap>                                 </span></td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td><span style=white-space:pre-wrap>       NA</span></td><td><span style=white-space:pre-wrap>       NA</span></td><td><span style=white-space:pre-wrap>       NA</span></td><td>NA</td><td><span style=white-space:pre-wrap>       NA</span></td><td>NA</td><td><span style=white-space:pre-wrap>       NA</span></td><td>NA</td><td>NA</td><td><span style=white-space:pre-wrap>       NA</span></td><td><span style=white-space:pre-wrap>          </span></td><td></td><td>1</td><td>1</td><td></td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>24.4</td><td>invalid_field</td><td>invalid_field</td><td>invalid_field</td><td>invalid_field</td><td>T</td><td>0.56603</td><td>invalid_field</td><td>0.027&amp;.&amp;0.027&amp;0.027</td><td>D&amp;D&amp;D&amp;D</td><td>invalid_field</td><td>invalid_field</td><td>NA</td><td>NA</td><td>NA</td><td>invalid_field</td><td>invalid_field</td><td>1</td><td><span style=white-space:pre-wrap>7350647467           </span></td><td>TRUE</td><td>0.9358089</td><td>NA</td><td>Macrogen</td><td>0.30535</td><td>5.1982e-01</td><td>3.2925e-03</td><td>0.47688000</td><td>0.18868</td><td>0.076</td><td>0.594</td><td>Macrogen</td><td>Macrogen</td><td>0.30535</td><td>5.1982e-01</td><td>3.2925e-03</td><td>0.47688000</td><td>0.18868</td><td>0.076</td><td>0.594</td><td><span style=white-space:pre-wrap>  NA</span></td><td>NA</td><td>NA</td><td><span style=white-space:pre-wrap>  NA</span></td><td><span style=white-space:pre-wrap>  NA</span></td><td><span style=white-space:pre-wrap>NA  </span></td></tr>\n",
       "\t<tr><th scope=row>4</th><td>chr1:10418892</td><td>C</td><td>T</td><td>6137 </td><td>1065817357</td><td>9484178706</td><td>6071727996</td><td>1</td><td> 21</td><td>11</td><td>10</td><td>0/1</td><td>99</td><td>34</td><td>34</td><td>0</td><td>0/0</td><td>99</td><td>35</td><td>35</td><td>0</td><td>0/0</td><td>99</td><td>.</td><td>hiConfDeNovo</td><td>60.00</td><td>9.32 </td><td>PASS</td><td>T</td><td>non_coding_transcript_exon_variant</td><td>MODIFIER</td><td><span style=white-space:pre-wrap>PGD   </span></td><td>ENSG00000142657</td><td>Transcript</td><td>ENST00000498356</td><td>processed_transcript</td><td><span style=white-space:pre-wrap>4/6  </span></td><td></td><td>ENST00000498356.1:n.433C&gt;T </td><td><span style=white-space:pre-wrap>                             </span></td><td>433 </td><td><span style=white-space:pre-wrap>    </span></td><td><span style=white-space:pre-wrap>   </span></td><td><span style=white-space:pre-wrap>   </span></td><td><span style=white-space:pre-wrap>       </span></td><td>rs766247086 </td><td><span style=white-space:pre-wrap>  NA</span></td><td> 1</td><td><span style=white-space:pre-wrap>          </span></td><td>SNV</td><td>HGNC</td><td>HGNC:8891 </td><td><span style=white-space:pre-wrap>   </span></td><td><span style=white-space:pre-wrap>           </span></td><td></td><td> 3</td><td><span style=white-space:pre-wrap>  </span></td><td><span style=white-space:pre-wrap>        </span></td><td><span style=white-space:pre-wrap>               </span></td><td><span style=white-space:pre-wrap>          </span></td><td><span style=white-space:pre-wrap>         </span></td><td><span style=white-space:pre-wrap>             </span></td><td><span style=white-space:pre-wrap>        </span></td><td>NA</td><td><span style=white-space:pre-wrap>              </span></td><td><span style=white-space:pre-wrap>                        </span></td><td><span style=white-space:pre-wrap>                                 </span></td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>3.182e-05</td><td>0.0001846</td><td>2.892e-05</td><td> 0</td><td>5.437e-05</td><td> 0</td><td>2.637e-05</td><td> 0</td><td> 0</td><td>0.0001846</td><td>gnomAD_AFR</td><td></td><td> </td><td> </td><td></td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td><span style=white-space:pre-wrap>  NA</span></td><td><span style=white-space:pre-wrap>             </span></td><td><span style=white-space:pre-wrap>             </span></td><td><span style=white-space:pre-wrap>             </span></td><td><span style=white-space:pre-wrap>             </span></td><td> </td><td><span style=white-space:pre-wrap>     NA</span></td><td><span style=white-space:pre-wrap>             </span></td><td><span style=white-space:pre-wrap>                   </span></td><td><span style=white-space:pre-wrap>       </span></td><td><span style=white-space:pre-wrap>             </span></td><td><span style=white-space:pre-wrap>             </span></td><td>NA</td><td>NA</td><td>NA</td><td><span style=white-space:pre-wrap>             </span></td><td><span style=white-space:pre-wrap>             </span></td><td>2</td><td>1065817357/6578846017</td><td>TRUE</td><td>1.0000000</td><td>NA</td><td><span style=white-space:pre-wrap>Murat   </span></td><td>0.55188</td><td>9.8788e-01</td><td>3.4517e-08</td><td>0.01211700</td><td>0.11301</td><td>0.050</td><td>0.292</td><td><span style=white-space:pre-wrap>Murat   </span></td><td><span style=white-space:pre-wrap>Murat   </span></td><td>0.55188</td><td>9.8788e-01</td><td>3.4517e-08</td><td>0.01211700</td><td>0.11301</td><td>0.050</td><td>0.292</td><td><span style=white-space:pre-wrap>  NA</span></td><td>NA</td><td>NA</td><td><span style=white-space:pre-wrap>  NA</span></td><td><span style=white-space:pre-wrap>  NA</span></td><td><span style=white-space:pre-wrap>NA  </span></td></tr>\n",
       "\t<tr><th scope=row>5</th><td>chr1:10418892</td><td>C</td><td>T</td><td>6137 </td><td>6578846017</td><td>9484178706</td><td>6071727996</td><td>1</td><td>137</td><td>71</td><td>66</td><td>0/1</td><td>99</td><td>34</td><td>34</td><td>0</td><td>0/0</td><td>99</td><td>35</td><td>35</td><td>0</td><td>0/0</td><td>99</td><td>.</td><td>hiConfDeNovo</td><td>60.00</td><td>9.32 </td><td>PASS</td><td>T</td><td>non_coding_transcript_exon_variant</td><td>MODIFIER</td><td><span style=white-space:pre-wrap>PGD   </span></td><td>ENSG00000142657</td><td>Transcript</td><td>ENST00000498356</td><td>processed_transcript</td><td><span style=white-space:pre-wrap>4/6  </span></td><td></td><td>ENST00000498356.1:n.433C&gt;T </td><td><span style=white-space:pre-wrap>                             </span></td><td>433 </td><td><span style=white-space:pre-wrap>    </span></td><td><span style=white-space:pre-wrap>   </span></td><td><span style=white-space:pre-wrap>   </span></td><td><span style=white-space:pre-wrap>       </span></td><td>rs766247086 </td><td><span style=white-space:pre-wrap>  NA</span></td><td> 1</td><td><span style=white-space:pre-wrap>          </span></td><td>SNV</td><td>HGNC</td><td>HGNC:8891 </td><td><span style=white-space:pre-wrap>   </span></td><td><span style=white-space:pre-wrap>           </span></td><td></td><td> 3</td><td><span style=white-space:pre-wrap>  </span></td><td><span style=white-space:pre-wrap>        </span></td><td><span style=white-space:pre-wrap>               </span></td><td><span style=white-space:pre-wrap>          </span></td><td><span style=white-space:pre-wrap>         </span></td><td><span style=white-space:pre-wrap>             </span></td><td><span style=white-space:pre-wrap>        </span></td><td>NA</td><td><span style=white-space:pre-wrap>              </span></td><td><span style=white-space:pre-wrap>                        </span></td><td><span style=white-space:pre-wrap>                                 </span></td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>3.182e-05</td><td>0.0001846</td><td>2.892e-05</td><td> 0</td><td>5.437e-05</td><td> 0</td><td>2.637e-05</td><td> 0</td><td> 0</td><td>0.0001846</td><td>gnomAD_AFR</td><td></td><td> </td><td> </td><td></td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td><span style=white-space:pre-wrap>  NA</span></td><td><span style=white-space:pre-wrap>             </span></td><td><span style=white-space:pre-wrap>             </span></td><td><span style=white-space:pre-wrap>             </span></td><td><span style=white-space:pre-wrap>             </span></td><td> </td><td><span style=white-space:pre-wrap>     NA</span></td><td><span style=white-space:pre-wrap>             </span></td><td><span style=white-space:pre-wrap>                   </span></td><td><span style=white-space:pre-wrap>       </span></td><td><span style=white-space:pre-wrap>             </span></td><td><span style=white-space:pre-wrap>             </span></td><td>NA</td><td>NA</td><td>NA</td><td><span style=white-space:pre-wrap>             </span></td><td><span style=white-space:pre-wrap>             </span></td><td>2</td><td>1065817357/6578846017</td><td>TRUE</td><td>0.7326841</td><td>NA</td><td>Macrogen</td><td>0.55188</td><td>9.8788e-01</td><td>3.4517e-08</td><td>0.01211700</td><td>0.11301</td><td>0.050</td><td>0.292</td><td>Macrogen</td><td>Macrogen</td><td>0.55188</td><td>9.8788e-01</td><td>3.4517e-08</td><td>0.01211700</td><td>0.11301</td><td>0.050</td><td>0.292</td><td><span style=white-space:pre-wrap>  NA</span></td><td>NA</td><td>NA</td><td><span style=white-space:pre-wrap>  NA</span></td><td><span style=white-space:pre-wrap>  NA</span></td><td><span style=white-space:pre-wrap>NA  </span></td></tr>\n",
       "\t<tr><th scope=row>6</th><td>chr1:10418892</td><td>C</td><td>T</td><td>6137 </td><td>1065817357</td><td>9484178706</td><td>6071727996</td><td>1</td><td> 21</td><td>11</td><td>10</td><td>0/1</td><td>99</td><td>34</td><td>34</td><td>0</td><td>0/0</td><td>99</td><td>35</td><td>35</td><td>0</td><td>0/0</td><td>99</td><td>.</td><td>hiConfDeNovo</td><td>60.00</td><td>9.32 </td><td>PASS</td><td>T</td><td>non_coding_transcript_exon_variant</td><td>MODIFIER</td><td><span style=white-space:pre-wrap>PGD   </span></td><td>ENSG00000142657</td><td>Transcript</td><td>ENST00000498356</td><td>processed_transcript</td><td><span style=white-space:pre-wrap>4/6  </span></td><td></td><td>ENST00000498356.1:n.433C&gt;T </td><td><span style=white-space:pre-wrap>                             </span></td><td>433 </td><td><span style=white-space:pre-wrap>    </span></td><td><span style=white-space:pre-wrap>   </span></td><td><span style=white-space:pre-wrap>   </span></td><td><span style=white-space:pre-wrap>       </span></td><td>rs766247086 </td><td><span style=white-space:pre-wrap>  NA</span></td><td> 1</td><td><span style=white-space:pre-wrap>          </span></td><td>SNV</td><td>HGNC</td><td>HGNC:8891 </td><td><span style=white-space:pre-wrap>   </span></td><td><span style=white-space:pre-wrap>           </span></td><td></td><td> 3</td><td><span style=white-space:pre-wrap>  </span></td><td><span style=white-space:pre-wrap>        </span></td><td><span style=white-space:pre-wrap>               </span></td><td><span style=white-space:pre-wrap>          </span></td><td><span style=white-space:pre-wrap>         </span></td><td><span style=white-space:pre-wrap>             </span></td><td><span style=white-space:pre-wrap>        </span></td><td>NA</td><td><span style=white-space:pre-wrap>              </span></td><td><span style=white-space:pre-wrap>                        </span></td><td><span style=white-space:pre-wrap>                                 </span></td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>3.182e-05</td><td>0.0001846</td><td>2.892e-05</td><td> 0</td><td>5.437e-05</td><td> 0</td><td>2.637e-05</td><td> 0</td><td> 0</td><td>0.0001846</td><td>gnomAD_AFR</td><td></td><td> </td><td> </td><td></td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td><span style=white-space:pre-wrap>  NA</span></td><td><span style=white-space:pre-wrap>             </span></td><td><span style=white-space:pre-wrap>             </span></td><td><span style=white-space:pre-wrap>             </span></td><td><span style=white-space:pre-wrap>             </span></td><td> </td><td><span style=white-space:pre-wrap>     NA</span></td><td><span style=white-space:pre-wrap>             </span></td><td><span style=white-space:pre-wrap>                   </span></td><td><span style=white-space:pre-wrap>       </span></td><td><span style=white-space:pre-wrap>             </span></td><td><span style=white-space:pre-wrap>             </span></td><td>NA</td><td>NA</td><td>NA</td><td><span style=white-space:pre-wrap>             </span></td><td><span style=white-space:pre-wrap>             </span></td><td>2</td><td>1065817357/6578846017</td><td>TRUE</td><td>1.0000000</td><td>NA</td><td><span style=white-space:pre-wrap>Murat   </span></td><td>0.55188</td><td>9.8788e-01</td><td>3.4517e-08</td><td>0.01211700</td><td>0.11301</td><td>0.050</td><td>0.292</td><td><span style=white-space:pre-wrap>Murat   </span></td><td><span style=white-space:pre-wrap>Murat   </span></td><td>0.55188</td><td>9.8788e-01</td><td>3.4517e-08</td><td>0.01211700</td><td>0.11301</td><td>0.050</td><td>0.292</td><td><span style=white-space:pre-wrap>  NA</span></td><td>NA</td><td>NA</td><td><span style=white-space:pre-wrap>  NA</span></td><td><span style=white-space:pre-wrap>  NA</span></td><td><span style=white-space:pre-wrap>NA  </span></td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 142\n",
       "\\begin{tabular}{r|llllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll}\n",
       "  & locus & ref & alt & family & pro & dad & mom & pro\\_sex & pro\\_DP & pro\\_ref & pro\\_alt & pro\\_GT & pro\\_GQ & dad\\_DP & dad\\_ref & dad\\_alt & dad\\_GT & dad\\_GQ & mom\\_DP & mom\\_ref & mom\\_alt & mom\\_GT & mom\\_GQ & p\\_denovo & confidence & MQ & QD & filter & Allele & Consequence & IMPACT & SYMBOL & Gene & Feature\\_type & Feature & BIOTYPE & EXON & INTRON & HGVSc & HGVSp & cDNA\\_position & CDS\\_position & Protein\\_position & Amino\\_acids & Codons & Existing\\_variation & DISTANCE & STRAND & FLAGS & VARIANT\\_CLASS & SYMBOL\\_SOURCE & HGNC\\_ID & CANONICAL & MANE\\_SELECT & MANE\\_PLUS\\_CLINICAL & TSL & APPRIS & CCDS & ENSP & SWISSPROT & TREMBL & UNIPARC & UNIPROT\\_ISOFORM & GENE\\_PHENO & SIFT & PolyPhen & DOMAINS & miRNA & HGVS\\_OFFSET & AF & AFR\\_AF & AMR\\_AF & EAS\\_AF & EUR\\_AF & SAS\\_AF & AA\\_AF & EA\\_AF & gnomAD\\_AF & gnomAD\\_AFR\\_AF & gnomAD\\_AMR\\_AF & gnomAD\\_ASJ\\_AF & gnomAD\\_EAS\\_AF & gnomAD\\_FIN\\_AF & gnomAD\\_NFE\\_AF & gnomAD\\_OTH\\_AF & gnomAD\\_SAS\\_AF & MAX\\_AF & MAX\\_AF\\_POPS & CLIN\\_SIG & SOMATIC & PHENO & PUBMED & MOTIF\\_NAME & MOTIF\\_POS & HIGH\\_INF\\_POS & MOTIF\\_SCORE\\_CHANGE & TRANSCRIPTION\\_FACTORS & CADD\\_phred & DisGeNET & MPC & MTR & Mastermind & MetaSVM\\_pred & MetaSVM\\_rankscore & Phenotypes & Polyphen2\\_HDIV\\_score & SIFT\\_pred & SplicAI & SpliceRegion & gnomAD\\_exomes\\_AC & gnomAD\\_exomes\\_AF & gnomAD\\_exomes\\_AN & pLI.x & pLI\\_values & occurence & probands & DP\\_filter12 & pval & Inspected & batch.x & oe\\_mis\\_pphen.x & pLI.y & pNull.x & pRec.x & oe\\_lof.x & oe\\_lof\\_lower.x & oe\\_lof\\_upper.x & batch.y & batch & oe\\_mis\\_pphen.y & pLI & pNull.y & pRec.y & oe\\_lof.y & oe\\_lof\\_lower.y & oe\\_lof\\_upper.y & pro\\_dob & fa\\_AoC & mo\\_AoC & fa\\_dob & mo\\_dob & pro\\_FA\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <int> & <int> & <int> & <chr> & <int> & <int> & <int> & <int> & <chr> & <int> & <int> & <int> & <int> & <chr> & <int> & <chr> & <chr> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <chr> & <chr> & <chr> & <lgl> & <int> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <lgl> & <lgl> & <lgl> & <lgl> & <lgl> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <dbl> & <int> & <chr> & <chr> & <int> & <chr> & <lgl> & <dbl> & <int> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <int> & <int> & <int> & <int> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & chr1:1290349  & G & A & 12111 & 12111.s1   & 12111.fa   & 12111.mo   & 2 &  33 & 19 & 14 & 0/1 & 99 & 40 & 40 & 0 & 0/0 & 76 & 40 & 40 & 0 & 0/0 & 67 & . & hiConfDeNovo & 60.00 & 11.4  & PASS & A & downstream\\_gene\\_variant            & MODIFIER & ACAP3  & ENSG00000131584 & Transcript & ENST00000492936 & retained\\_intron      &       &  &                             &                               &      &      &     &     &         &              & 2041 & -1 &            & SNV & HGNC & HGNC:16754 &     &             &  &  1 &    &          &                 &            &           &               &          & NA &                &                          &                                   & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA &        NA &        NA &        NA & NA &        NA & NA &        NA & NA & NA &        NA &            &  &   &   &  & NA & NA & NA & NA & NA &   NA &               &               &               &               &   &      NA &               &                     &         &               &               & NA & NA & NA &               &               & 1 & 12111.s1              & TRUE & 0.4868502 & NA & SSC      & 0.53448 & 4.4135e-06 & 1.4907e-05 & 0.99998000 & 0.39045 & 0.266 & 0.586 & SSC      & SSC      & 0.53448 & 4.4135e-06 & 1.4907e-05 & 0.99998000 & 0.39045 & 0.266 & 0.586 & 2000 & 29 & 26 & 1971 & 1974 & None\\\\\n",
       "\t2 & chr1:1486163  & G & C & 7004  & 4926035813 & 2917406449 & 3160220249 & 1 &  64 & 30 & 34 & 0/1 & 99 & 43 & 43 & 0 & 0/0 & 88 & 44 & 44 & 0 & 0/0 & 91 & . & hiConfDeNovo & 59.68 & 16.27 & PASS & C & missense\\_variant                   & MODERATE & ATAD3B & ENSG00000160072 & Transcript & ENST00000673477 & protein\\_coding       & 10/16 &  & ENST00000673477.1:c.1017G>C & ENSP00000500094.1:p.Lys339Asn & 1137 & 1017 & 339 & K/N & aaG/aaC & COSV58019127 &   NA &  1 &            & SNV & HGNC & HGNC:24007 & YES & NM\\_031921.6 &  & NA & P1 & CCDS30.1 & ENSP00000500094 & Q5T9A4.154 &           & UPI000013E044 & Q5T9A4-1 & NA & deleterious(0) & possibly\\_damaging(0.824) & AFDB-ENSP\\_mappings:AF-Q5T9A4-F1.A & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA &        NA &        NA &        NA & NA &        NA & NA &        NA & NA & NA &        NA &            &  & 1 & 1 &  & NA & NA & NA & NA & NA & 23.7 & invalid\\_field & invalid\\_field & invalid\\_field & invalid\\_field & D & 0.94306 & invalid\\_field & 0.955               & D       & invalid\\_field & invalid\\_field & NA & NA & NA & invalid\\_field & invalid\\_field & 1 & 4926035813            & TRUE & 0.7079808 & NA & UCI      & 1.34320 & 1.0032e-22 & 9.9972e-01 & 0.00027792 & 1.09950 & 0.829 & 1.477 & UCI      & UCI      & 1.34320 & 1.0032e-22 & 9.9972e-01 & 0.00027792 & 1.09950 & 0.829 & 1.477 &   NA & NA & NA &   NA &   NA & NA  \\\\\n",
       "\t3 & chr1:9356429  & G & A & 6144  & 7350647467 & 2898643437 & 4401829860 & 1 & 154 & 76 & 78 & 0/1 & 99 & 33 & 33 & 0 & 0/0 & 76 & 33 & 33 & 0 & 0/0 & 76 & . & hiConfDeNovo & 60.00 & 9.72  & PASS & A & missense\\_variant                   & MODERATE & SPSB1  & ENSG00000171621 & Transcript & ENST00000450402 & protein\\_coding       & 2/2   &  & ENST00000450402.1:c.538G>A  & ENSP00000409235.1:p.Ala180Thr & 754  & 538  & 180 & A/T & Gcc/Acc & COSV60164629 &   NA &  1 & cds\\_end\\_NF & SNV & HGNC & HGNC:30628 &     &             &  &  5 &    &          & ENSP00000409235 &            & A2A276.84 & UPI0000EDA26A &          & NA & deleterious(0) & benign(0.031)            &                                   & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA &        NA &        NA &        NA & NA &        NA & NA &        NA & NA & NA &        NA &            &  & 1 & 1 &  & NA & NA & NA & NA & NA & 24.4 & invalid\\_field & invalid\\_field & invalid\\_field & invalid\\_field & T & 0.56603 & invalid\\_field & 0.027\\&.\\&0.027\\&0.027 & D\\&D\\&D\\&D & invalid\\_field & invalid\\_field & NA & NA & NA & invalid\\_field & invalid\\_field & 1 & 7350647467            & TRUE & 0.9358089 & NA & Macrogen & 0.30535 & 5.1982e-01 & 3.2925e-03 & 0.47688000 & 0.18868 & 0.076 & 0.594 & Macrogen & Macrogen & 0.30535 & 5.1982e-01 & 3.2925e-03 & 0.47688000 & 0.18868 & 0.076 & 0.594 &   NA & NA & NA &   NA &   NA & NA  \\\\\n",
       "\t4 & chr1:10418892 & C & T & 6137  & 1065817357 & 9484178706 & 6071727996 & 1 &  21 & 11 & 10 & 0/1 & 99 & 34 & 34 & 0 & 0/0 & 99 & 35 & 35 & 0 & 0/0 & 99 & . & hiConfDeNovo & 60.00 & 9.32  & PASS & T & non\\_coding\\_transcript\\_exon\\_variant & MODIFIER & PGD    & ENSG00000142657 & Transcript & ENST00000498356 & processed\\_transcript & 4/6   &  & ENST00000498356.1:n.433C>T  &                               & 433  &      &     &     &         & rs766247086  &   NA &  1 &            & SNV & HGNC & HGNC:8891  &     &             &  &  3 &    &          &                 &            &           &               &          & NA &                &                          &                                   & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & 3.182e-05 & 0.0001846 & 2.892e-05 &  0 & 5.437e-05 &  0 & 2.637e-05 &  0 &  0 & 0.0001846 & gnomAD\\_AFR &  &   &   &  & NA & NA & NA & NA & NA &   NA &               &               &               &               &   &      NA &               &                     &         &               &               & NA & NA & NA &               &               & 2 & 1065817357/6578846017 & TRUE & 1.0000000 & NA & Murat    & 0.55188 & 9.8788e-01 & 3.4517e-08 & 0.01211700 & 0.11301 & 0.050 & 0.292 & Murat    & Murat    & 0.55188 & 9.8788e-01 & 3.4517e-08 & 0.01211700 & 0.11301 & 0.050 & 0.292 &   NA & NA & NA &   NA &   NA & NA  \\\\\n",
       "\t5 & chr1:10418892 & C & T & 6137  & 6578846017 & 9484178706 & 6071727996 & 1 & 137 & 71 & 66 & 0/1 & 99 & 34 & 34 & 0 & 0/0 & 99 & 35 & 35 & 0 & 0/0 & 99 & . & hiConfDeNovo & 60.00 & 9.32  & PASS & T & non\\_coding\\_transcript\\_exon\\_variant & MODIFIER & PGD    & ENSG00000142657 & Transcript & ENST00000498356 & processed\\_transcript & 4/6   &  & ENST00000498356.1:n.433C>T  &                               & 433  &      &     &     &         & rs766247086  &   NA &  1 &            & SNV & HGNC & HGNC:8891  &     &             &  &  3 &    &          &                 &            &           &               &          & NA &                &                          &                                   & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & 3.182e-05 & 0.0001846 & 2.892e-05 &  0 & 5.437e-05 &  0 & 2.637e-05 &  0 &  0 & 0.0001846 & gnomAD\\_AFR &  &   &   &  & NA & NA & NA & NA & NA &   NA &               &               &               &               &   &      NA &               &                     &         &               &               & NA & NA & NA &               &               & 2 & 1065817357/6578846017 & TRUE & 0.7326841 & NA & Macrogen & 0.55188 & 9.8788e-01 & 3.4517e-08 & 0.01211700 & 0.11301 & 0.050 & 0.292 & Macrogen & Macrogen & 0.55188 & 9.8788e-01 & 3.4517e-08 & 0.01211700 & 0.11301 & 0.050 & 0.292 &   NA & NA & NA &   NA &   NA & NA  \\\\\n",
       "\t6 & chr1:10418892 & C & T & 6137  & 1065817357 & 9484178706 & 6071727996 & 1 &  21 & 11 & 10 & 0/1 & 99 & 34 & 34 & 0 & 0/0 & 99 & 35 & 35 & 0 & 0/0 & 99 & . & hiConfDeNovo & 60.00 & 9.32  & PASS & T & non\\_coding\\_transcript\\_exon\\_variant & MODIFIER & PGD    & ENSG00000142657 & Transcript & ENST00000498356 & processed\\_transcript & 4/6   &  & ENST00000498356.1:n.433C>T  &                               & 433  &      &     &     &         & rs766247086  &   NA &  1 &            & SNV & HGNC & HGNC:8891  &     &             &  &  3 &    &          &                 &            &           &               &          & NA &                &                          &                                   & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & 3.182e-05 & 0.0001846 & 2.892e-05 &  0 & 5.437e-05 &  0 & 2.637e-05 &  0 &  0 & 0.0001846 & gnomAD\\_AFR &  &   &   &  & NA & NA & NA & NA & NA &   NA &               &               &               &               &   &      NA &               &                     &         &               &               & NA & NA & NA &               &               & 2 & 1065817357/6578846017 & TRUE & 1.0000000 & NA & Murat    & 0.55188 & 9.8788e-01 & 3.4517e-08 & 0.01211700 & 0.11301 & 0.050 & 0.292 & Murat    & Murat    & 0.55188 & 9.8788e-01 & 3.4517e-08 & 0.01211700 & 0.11301 & 0.050 & 0.292 &   NA & NA & NA &   NA &   NA & NA  \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 142\n",
       "\n",
       "| <!--/--> | locus &lt;chr&gt; | ref &lt;chr&gt; | alt &lt;chr&gt; | family &lt;chr&gt; | pro &lt;chr&gt; | dad &lt;chr&gt; | mom &lt;chr&gt; | pro_sex &lt;int&gt; | pro_DP &lt;int&gt; | pro_ref &lt;int&gt; | pro_alt &lt;int&gt; | pro_GT &lt;chr&gt; | pro_GQ &lt;int&gt; | dad_DP &lt;int&gt; | dad_ref &lt;int&gt; | dad_alt &lt;int&gt; | dad_GT &lt;chr&gt; | dad_GQ &lt;int&gt; | mom_DP &lt;int&gt; | mom_ref &lt;int&gt; | mom_alt &lt;int&gt; | mom_GT &lt;chr&gt; | mom_GQ &lt;int&gt; | p_denovo &lt;chr&gt; | confidence &lt;chr&gt; | MQ &lt;dbl&gt; | QD &lt;chr&gt; | filter &lt;chr&gt; | Allele &lt;chr&gt; | Consequence &lt;chr&gt; | IMPACT &lt;chr&gt; | SYMBOL &lt;chr&gt; | Gene &lt;chr&gt; | Feature_type &lt;chr&gt; | Feature &lt;chr&gt; | BIOTYPE &lt;chr&gt; | EXON &lt;chr&gt; | INTRON &lt;chr&gt; | HGVSc &lt;chr&gt; | HGVSp &lt;chr&gt; | cDNA_position &lt;chr&gt; | CDS_position &lt;chr&gt; | Protein_position &lt;chr&gt; | Amino_acids &lt;chr&gt; | Codons &lt;chr&gt; | Existing_variation &lt;chr&gt; | DISTANCE &lt;int&gt; | STRAND &lt;int&gt; | FLAGS &lt;chr&gt; | VARIANT_CLASS &lt;chr&gt; | SYMBOL_SOURCE &lt;chr&gt; | HGNC_ID &lt;chr&gt; | CANONICAL &lt;chr&gt; | MANE_SELECT &lt;chr&gt; | MANE_PLUS_CLINICAL &lt;chr&gt; | TSL &lt;int&gt; | APPRIS &lt;chr&gt; | CCDS &lt;chr&gt; | ENSP &lt;chr&gt; | SWISSPROT &lt;chr&gt; | TREMBL &lt;chr&gt; | UNIPARC &lt;chr&gt; | UNIPROT_ISOFORM &lt;chr&gt; | GENE_PHENO &lt;int&gt; | SIFT &lt;chr&gt; | PolyPhen &lt;chr&gt; | DOMAINS &lt;chr&gt; | miRNA &lt;lgl&gt; | HGVS_OFFSET &lt;int&gt; | AF &lt;dbl&gt; | AFR_AF &lt;dbl&gt; | AMR_AF &lt;dbl&gt; | EAS_AF &lt;dbl&gt; | EUR_AF &lt;dbl&gt; | SAS_AF &lt;dbl&gt; | AA_AF &lt;dbl&gt; | EA_AF &lt;dbl&gt; | gnomAD_AF &lt;dbl&gt; | gnomAD_AFR_AF &lt;dbl&gt; | gnomAD_AMR_AF &lt;dbl&gt; | gnomAD_ASJ_AF &lt;dbl&gt; | gnomAD_EAS_AF &lt;dbl&gt; | gnomAD_FIN_AF &lt;dbl&gt; | gnomAD_NFE_AF &lt;dbl&gt; | gnomAD_OTH_AF &lt;dbl&gt; | gnomAD_SAS_AF &lt;dbl&gt; | MAX_AF &lt;dbl&gt; | MAX_AF_POPS &lt;chr&gt; | CLIN_SIG &lt;chr&gt; | SOMATIC &lt;chr&gt; | PHENO &lt;chr&gt; | PUBMED &lt;chr&gt; | MOTIF_NAME &lt;lgl&gt; | MOTIF_POS &lt;lgl&gt; | HIGH_INF_POS &lt;lgl&gt; | MOTIF_SCORE_CHANGE &lt;lgl&gt; | TRANSCRIPTION_FACTORS &lt;lgl&gt; | CADD_phred &lt;dbl&gt; | DisGeNET &lt;chr&gt; | MPC &lt;chr&gt; | MTR &lt;chr&gt; | Mastermind &lt;chr&gt; | MetaSVM_pred &lt;chr&gt; | MetaSVM_rankscore &lt;dbl&gt; | Phenotypes &lt;chr&gt; | Polyphen2_HDIV_score &lt;chr&gt; | SIFT_pred &lt;chr&gt; | SplicAI &lt;chr&gt; | SpliceRegion &lt;chr&gt; | gnomAD_exomes_AC &lt;int&gt; | gnomAD_exomes_AF &lt;dbl&gt; | gnomAD_exomes_AN &lt;int&gt; | pLI.x &lt;chr&gt; | pLI_values &lt;chr&gt; | occurence &lt;int&gt; | probands &lt;chr&gt; | DP_filter12 &lt;lgl&gt; | pval &lt;dbl&gt; | Inspected &lt;int&gt; | batch.x &lt;chr&gt; | oe_mis_pphen.x &lt;dbl&gt; | pLI.y &lt;dbl&gt; | pNull.x &lt;dbl&gt; | pRec.x &lt;dbl&gt; | oe_lof.x &lt;dbl&gt; | oe_lof_lower.x &lt;dbl&gt; | oe_lof_upper.x &lt;dbl&gt; | batch.y &lt;chr&gt; | batch &lt;chr&gt; | oe_mis_pphen.y &lt;dbl&gt; | pLI &lt;dbl&gt; | pNull.y &lt;dbl&gt; | pRec.y &lt;dbl&gt; | oe_lof.y &lt;dbl&gt; | oe_lof_lower.y &lt;dbl&gt; | oe_lof_upper.y &lt;dbl&gt; | pro_dob &lt;int&gt; | fa_AoC &lt;int&gt; | mo_AoC &lt;int&gt; | fa_dob &lt;int&gt; | mo_dob &lt;int&gt; | pro_FA &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | chr1:1290349  | G | A | 12111 | 12111.s1   | 12111.fa   | 12111.mo   | 2 |  33 | 19 | 14 | 0/1 | 99 | 40 | 40 | 0 | 0/0 | 76 | 40 | 40 | 0 | 0/0 | 67 | . | hiConfDeNovo | 60.00 | 11.4  | PASS | A | downstream_gene_variant            | MODIFIER | ACAP3  | ENSG00000131584 | Transcript | ENST00000492936 | retained_intron      | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> | 2041 | -1 | <!----> | SNV | HGNC | HGNC:16754 | <!----> | <!----> | <!----> |  1 | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> | NA | <!----> | <!----> | <!----> | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |        NA |        NA |        NA | NA |        NA | NA |        NA | NA | NA |        NA | <!----> | <!----> | <!----> | <!----> | <!----> | NA | NA | NA | NA | NA |   NA | <!----> | <!----> | <!----> | <!----> | <!----> |      NA | <!----> | <!----> | <!----> | <!----> | <!----> | NA | NA | NA | <!----> | <!----> | 1 | 12111.s1              | TRUE | 0.4868502 | NA | SSC      | 0.53448 | 4.4135e-06 | 1.4907e-05 | 0.99998000 | 0.39045 | 0.266 | 0.586 | SSC      | SSC      | 0.53448 | 4.4135e-06 | 1.4907e-05 | 0.99998000 | 0.39045 | 0.266 | 0.586 | 2000 | 29 | 26 | 1971 | 1974 | None |\n",
       "| 2 | chr1:1486163  | G | C | 7004  | 4926035813 | 2917406449 | 3160220249 | 1 |  64 | 30 | 34 | 0/1 | 99 | 43 | 43 | 0 | 0/0 | 88 | 44 | 44 | 0 | 0/0 | 91 | . | hiConfDeNovo | 59.68 | 16.27 | PASS | C | missense_variant                   | MODERATE | ATAD3B | ENSG00000160072 | Transcript | ENST00000673477 | protein_coding       | 10/16 | <!----> | ENST00000673477.1:c.1017G&gt;C | ENSP00000500094.1:p.Lys339Asn | 1137 | 1017 | 339 | K/N | aaG/aaC | COSV58019127 |   NA |  1 | <!----> | SNV | HGNC | HGNC:24007 | YES | NM_031921.6 | <!----> | NA | P1 | CCDS30.1 | ENSP00000500094 | Q5T9A4.154 | <!----> | UPI000013E044 | Q5T9A4-1 | NA | deleterious(0) | possibly_damaging(0.824) | AFDB-ENSP_mappings:AF-Q5T9A4-F1.A | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |        NA |        NA |        NA | NA |        NA | NA |        NA | NA | NA |        NA | <!----> | <!----> | 1 | 1 | <!----> | NA | NA | NA | NA | NA | 23.7 | invalid_field | invalid_field | invalid_field | invalid_field | D | 0.94306 | invalid_field | 0.955               | D       | invalid_field | invalid_field | NA | NA | NA | invalid_field | invalid_field | 1 | 4926035813            | TRUE | 0.7079808 | NA | UCI      | 1.34320 | 1.0032e-22 | 9.9972e-01 | 0.00027792 | 1.09950 | 0.829 | 1.477 | UCI      | UCI      | 1.34320 | 1.0032e-22 | 9.9972e-01 | 0.00027792 | 1.09950 | 0.829 | 1.477 |   NA | NA | NA |   NA |   NA | NA   |\n",
       "| 3 | chr1:9356429  | G | A | 6144  | 7350647467 | 2898643437 | 4401829860 | 1 | 154 | 76 | 78 | 0/1 | 99 | 33 | 33 | 0 | 0/0 | 76 | 33 | 33 | 0 | 0/0 | 76 | . | hiConfDeNovo | 60.00 | 9.72  | PASS | A | missense_variant                   | MODERATE | SPSB1  | ENSG00000171621 | Transcript | ENST00000450402 | protein_coding       | 2/2   | <!----> | ENST00000450402.1:c.538G&gt;A  | ENSP00000409235.1:p.Ala180Thr | 754  | 538  | 180 | A/T | Gcc/Acc | COSV60164629 |   NA |  1 | cds_end_NF | SNV | HGNC | HGNC:30628 | <!----> | <!----> | <!----> |  5 | <!----> | <!----> | ENSP00000409235 | <!----> | A2A276.84 | UPI0000EDA26A | <!----> | NA | deleterious(0) | benign(0.031)            | <!----> | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |        NA |        NA |        NA | NA |        NA | NA |        NA | NA | NA |        NA | <!----> | <!----> | 1 | 1 | <!----> | NA | NA | NA | NA | NA | 24.4 | invalid_field | invalid_field | invalid_field | invalid_field | T | 0.56603 | invalid_field | 0.027&amp;.&amp;0.027&amp;0.027 | D&amp;D&amp;D&amp;D | invalid_field | invalid_field | NA | NA | NA | invalid_field | invalid_field | 1 | 7350647467            | TRUE | 0.9358089 | NA | Macrogen | 0.30535 | 5.1982e-01 | 3.2925e-03 | 0.47688000 | 0.18868 | 0.076 | 0.594 | Macrogen | Macrogen | 0.30535 | 5.1982e-01 | 3.2925e-03 | 0.47688000 | 0.18868 | 0.076 | 0.594 |   NA | NA | NA |   NA |   NA | NA   |\n",
       "| 4 | chr1:10418892 | C | T | 6137  | 1065817357 | 9484178706 | 6071727996 | 1 |  21 | 11 | 10 | 0/1 | 99 | 34 | 34 | 0 | 0/0 | 99 | 35 | 35 | 0 | 0/0 | 99 | . | hiConfDeNovo | 60.00 | 9.32  | PASS | T | non_coding_transcript_exon_variant | MODIFIER | PGD    | ENSG00000142657 | Transcript | ENST00000498356 | processed_transcript | 4/6   | <!----> | ENST00000498356.1:n.433C&gt;T  | <!----> | 433  | <!----> | <!----> | <!----> | <!----> | rs766247086  |   NA |  1 | <!----> | SNV | HGNC | HGNC:8891  | <!----> | <!----> | <!----> |  3 | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> | NA | <!----> | <!----> | <!----> | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | 3.182e-05 | 0.0001846 | 2.892e-05 |  0 | 5.437e-05 |  0 | 2.637e-05 |  0 |  0 | 0.0001846 | gnomAD_AFR | <!----> | <!----> | <!----> | <!----> | NA | NA | NA | NA | NA |   NA | <!----> | <!----> | <!----> | <!----> | <!----> |      NA | <!----> | <!----> | <!----> | <!----> | <!----> | NA | NA | NA | <!----> | <!----> | 2 | 1065817357/6578846017 | TRUE | 1.0000000 | NA | Murat    | 0.55188 | 9.8788e-01 | 3.4517e-08 | 0.01211700 | 0.11301 | 0.050 | 0.292 | Murat    | Murat    | 0.55188 | 9.8788e-01 | 3.4517e-08 | 0.01211700 | 0.11301 | 0.050 | 0.292 |   NA | NA | NA |   NA |   NA | NA   |\n",
       "| 5 | chr1:10418892 | C | T | 6137  | 6578846017 | 9484178706 | 6071727996 | 1 | 137 | 71 | 66 | 0/1 | 99 | 34 | 34 | 0 | 0/0 | 99 | 35 | 35 | 0 | 0/0 | 99 | . | hiConfDeNovo | 60.00 | 9.32  | PASS | T | non_coding_transcript_exon_variant | MODIFIER | PGD    | ENSG00000142657 | Transcript | ENST00000498356 | processed_transcript | 4/6   | <!----> | ENST00000498356.1:n.433C&gt;T  | <!----> | 433  | <!----> | <!----> | <!----> | <!----> | rs766247086  |   NA |  1 | <!----> | SNV | HGNC | HGNC:8891  | <!----> | <!----> | <!----> |  3 | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> | NA | <!----> | <!----> | <!----> | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | 3.182e-05 | 0.0001846 | 2.892e-05 |  0 | 5.437e-05 |  0 | 2.637e-05 |  0 |  0 | 0.0001846 | gnomAD_AFR | <!----> | <!----> | <!----> | <!----> | NA | NA | NA | NA | NA |   NA | <!----> | <!----> | <!----> | <!----> | <!----> |      NA | <!----> | <!----> | <!----> | <!----> | <!----> | NA | NA | NA | <!----> | <!----> | 2 | 1065817357/6578846017 | TRUE | 0.7326841 | NA | Macrogen | 0.55188 | 9.8788e-01 | 3.4517e-08 | 0.01211700 | 0.11301 | 0.050 | 0.292 | Macrogen | Macrogen | 0.55188 | 9.8788e-01 | 3.4517e-08 | 0.01211700 | 0.11301 | 0.050 | 0.292 |   NA | NA | NA |   NA |   NA | NA   |\n",
       "| 6 | chr1:10418892 | C | T | 6137  | 1065817357 | 9484178706 | 6071727996 | 1 |  21 | 11 | 10 | 0/1 | 99 | 34 | 34 | 0 | 0/0 | 99 | 35 | 35 | 0 | 0/0 | 99 | . | hiConfDeNovo | 60.00 | 9.32  | PASS | T | non_coding_transcript_exon_variant | MODIFIER | PGD    | ENSG00000142657 | Transcript | ENST00000498356 | processed_transcript | 4/6   | <!----> | ENST00000498356.1:n.433C&gt;T  | <!----> | 433  | <!----> | <!----> | <!----> | <!----> | rs766247086  |   NA |  1 | <!----> | SNV | HGNC | HGNC:8891  | <!----> | <!----> | <!----> |  3 | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> | NA | <!----> | <!----> | <!----> | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | 3.182e-05 | 0.0001846 | 2.892e-05 |  0 | 5.437e-05 |  0 | 2.637e-05 |  0 |  0 | 0.0001846 | gnomAD_AFR | <!----> | <!----> | <!----> | <!----> | NA | NA | NA | NA | NA |   NA | <!----> | <!----> | <!----> | <!----> | <!----> |      NA | <!----> | <!----> | <!----> | <!----> | <!----> | NA | NA | NA | <!----> | <!----> | 2 | 1065817357/6578846017 | TRUE | 1.0000000 | NA | Murat    | 0.55188 | 9.8788e-01 | 3.4517e-08 | 0.01211700 | 0.11301 | 0.050 | 0.292 | Murat    | Murat    | 0.55188 | 9.8788e-01 | 3.4517e-08 | 0.01211700 | 0.11301 | 0.050 | 0.292 |   NA | NA | NA |   NA |   NA | NA   |\n",
       "\n"
      ],
      "text/plain": [
       "  locus         ref alt family pro        dad        mom        pro_sex pro_DP\n",
       "1 chr1:1290349  G   A   12111  12111.s1   12111.fa   12111.mo   2        33   \n",
       "2 chr1:1486163  G   C   7004   4926035813 2917406449 3160220249 1        64   \n",
       "3 chr1:9356429  G   A   6144   7350647467 2898643437 4401829860 1       154   \n",
       "4 chr1:10418892 C   T   6137   1065817357 9484178706 6071727996 1        21   \n",
       "5 chr1:10418892 C   T   6137   6578846017 9484178706 6071727996 1       137   \n",
       "6 chr1:10418892 C   T   6137   1065817357 9484178706 6071727996 1        21   \n",
       "  pro_ref pro_alt pro_GT pro_GQ dad_DP dad_ref dad_alt dad_GT dad_GQ mom_DP\n",
       "1 19      14      0/1    99     40     40      0       0/0    76     40    \n",
       "2 30      34      0/1    99     43     43      0       0/0    88     44    \n",
       "3 76      78      0/1    99     33     33      0       0/0    76     33    \n",
       "4 11      10      0/1    99     34     34      0       0/0    99     35    \n",
       "5 71      66      0/1    99     34     34      0       0/0    99     35    \n",
       "6 11      10      0/1    99     34     34      0       0/0    99     35    \n",
       "  mom_ref mom_alt mom_GT mom_GQ p_denovo confidence   MQ    QD    filter Allele\n",
       "1 40      0       0/0    67     .        hiConfDeNovo 60.00 11.4  PASS   A     \n",
       "2 44      0       0/0    91     .        hiConfDeNovo 59.68 16.27 PASS   C     \n",
       "3 33      0       0/0    76     .        hiConfDeNovo 60.00 9.72  PASS   A     \n",
       "4 35      0       0/0    99     .        hiConfDeNovo 60.00 9.32  PASS   T     \n",
       "5 35      0       0/0    99     .        hiConfDeNovo 60.00 9.32  PASS   T     \n",
       "6 35      0       0/0    99     .        hiConfDeNovo 60.00 9.32  PASS   T     \n",
       "  Consequence                        IMPACT   SYMBOL Gene           \n",
       "1 downstream_gene_variant            MODIFIER ACAP3  ENSG00000131584\n",
       "2 missense_variant                   MODERATE ATAD3B ENSG00000160072\n",
       "3 missense_variant                   MODERATE SPSB1  ENSG00000171621\n",
       "4 non_coding_transcript_exon_variant MODIFIER PGD    ENSG00000142657\n",
       "5 non_coding_transcript_exon_variant MODIFIER PGD    ENSG00000142657\n",
       "6 non_coding_transcript_exon_variant MODIFIER PGD    ENSG00000142657\n",
       "  Feature_type Feature         BIOTYPE              EXON  INTRON\n",
       "1 Transcript   ENST00000492936 retained_intron                  \n",
       "2 Transcript   ENST00000673477 protein_coding       10/16       \n",
       "3 Transcript   ENST00000450402 protein_coding       2/2         \n",
       "4 Transcript   ENST00000498356 processed_transcript 4/6         \n",
       "5 Transcript   ENST00000498356 processed_transcript 4/6         \n",
       "6 Transcript   ENST00000498356 processed_transcript 4/6         \n",
       "  HGVSc                       HGVSp                         cDNA_position\n",
       "1                                                                        \n",
       "2 ENST00000673477.1:c.1017G>C ENSP00000500094.1:p.Lys339Asn 1137         \n",
       "3 ENST00000450402.1:c.538G>A  ENSP00000409235.1:p.Ala180Thr 754          \n",
       "4 ENST00000498356.1:n.433C>T                                433          \n",
       "5 ENST00000498356.1:n.433C>T                                433          \n",
       "6 ENST00000498356.1:n.433C>T                                433          \n",
       "  CDS_position Protein_position Amino_acids Codons  Existing_variation DISTANCE\n",
       "1                                                                      2041    \n",
       "2 1017         339              K/N         aaG/aaC COSV58019127         NA    \n",
       "3 538          180              A/T         Gcc/Acc COSV60164629         NA    \n",
       "4                                                   rs766247086          NA    \n",
       "5                                                   rs766247086          NA    \n",
       "6                                                   rs766247086          NA    \n",
       "  STRAND FLAGS      VARIANT_CLASS SYMBOL_SOURCE HGNC_ID    CANONICAL\n",
       "1 -1                SNV           HGNC          HGNC:16754          \n",
       "2  1                SNV           HGNC          HGNC:24007 YES      \n",
       "3  1     cds_end_NF SNV           HGNC          HGNC:30628          \n",
       "4  1                SNV           HGNC          HGNC:8891           \n",
       "5  1                SNV           HGNC          HGNC:8891           \n",
       "6  1                SNV           HGNC          HGNC:8891           \n",
       "  MANE_SELECT MANE_PLUS_CLINICAL TSL APPRIS CCDS     ENSP            SWISSPROT \n",
       "1                                 1                                            \n",
       "2 NM_031921.6                    NA  P1     CCDS30.1 ENSP00000500094 Q5T9A4.154\n",
       "3                                 5                  ENSP00000409235           \n",
       "4                                 3                                            \n",
       "5                                 3                                            \n",
       "6                                 3                                            \n",
       "  TREMBL    UNIPARC       UNIPROT_ISOFORM GENE_PHENO SIFT          \n",
       "1                                         NA                       \n",
       "2           UPI000013E044 Q5T9A4-1        NA         deleterious(0)\n",
       "3 A2A276.84 UPI0000EDA26A                 NA         deleterious(0)\n",
       "4                                         NA                       \n",
       "5                                         NA                       \n",
       "6                                         NA                       \n",
       "  PolyPhen                 DOMAINS                           miRNA HGVS_OFFSET\n",
       "1                                                            NA    NA         \n",
       "2 possibly_damaging(0.824) AFDB-ENSP_mappings:AF-Q5T9A4-F1.A NA    NA         \n",
       "3 benign(0.031)                                              NA    NA         \n",
       "4                                                            NA    NA         \n",
       "5                                                            NA    NA         \n",
       "6                                                            NA    NA         \n",
       "  AF AFR_AF AMR_AF EAS_AF EUR_AF SAS_AF AA_AF EA_AF gnomAD_AF gnomAD_AFR_AF\n",
       "1 NA NA     NA     NA     NA     NA     NA    NA           NA        NA    \n",
       "2 NA NA     NA     NA     NA     NA     NA    NA           NA        NA    \n",
       "3 NA NA     NA     NA     NA     NA     NA    NA           NA        NA    \n",
       "4 NA NA     NA     NA     NA     NA     NA    NA    3.182e-05 0.0001846    \n",
       "5 NA NA     NA     NA     NA     NA     NA    NA    3.182e-05 0.0001846    \n",
       "6 NA NA     NA     NA     NA     NA     NA    NA    3.182e-05 0.0001846    \n",
       "  gnomAD_AMR_AF gnomAD_ASJ_AF gnomAD_EAS_AF gnomAD_FIN_AF gnomAD_NFE_AF\n",
       "1        NA     NA                   NA     NA                   NA    \n",
       "2        NA     NA                   NA     NA                   NA    \n",
       "3        NA     NA                   NA     NA                   NA    \n",
       "4 2.892e-05      0            5.437e-05      0            2.637e-05    \n",
       "5 2.892e-05      0            5.437e-05      0            2.637e-05    \n",
       "6 2.892e-05      0            5.437e-05      0            2.637e-05    \n",
       "  gnomAD_OTH_AF gnomAD_SAS_AF MAX_AF    MAX_AF_POPS CLIN_SIG SOMATIC PHENO\n",
       "1 NA            NA                   NA                                   \n",
       "2 NA            NA                   NA                      1       1    \n",
       "3 NA            NA                   NA                      1       1    \n",
       "4  0             0            0.0001846 gnomAD_AFR                        \n",
       "5  0             0            0.0001846 gnomAD_AFR                        \n",
       "6  0             0            0.0001846 gnomAD_AFR                        \n",
       "  PUBMED MOTIF_NAME MOTIF_POS HIGH_INF_POS MOTIF_SCORE_CHANGE\n",
       "1        NA         NA        NA           NA                \n",
       "2        NA         NA        NA           NA                \n",
       "3        NA         NA        NA           NA                \n",
       "4        NA         NA        NA           NA                \n",
       "5        NA         NA        NA           NA                \n",
       "6        NA         NA        NA           NA                \n",
       "  TRANSCRIPTION_FACTORS CADD_phred DisGeNET      MPC           MTR          \n",
       "1 NA                      NA                                                \n",
       "2 NA                    23.7       invalid_field invalid_field invalid_field\n",
       "3 NA                    24.4       invalid_field invalid_field invalid_field\n",
       "4 NA                      NA                                                \n",
       "5 NA                      NA                                                \n",
       "6 NA                      NA                                                \n",
       "  Mastermind    MetaSVM_pred MetaSVM_rankscore Phenotypes   \n",
       "1                                 NA                        \n",
       "2 invalid_field D            0.94306           invalid_field\n",
       "3 invalid_field T            0.56603           invalid_field\n",
       "4                                 NA                        \n",
       "5                                 NA                        \n",
       "6                                 NA                        \n",
       "  Polyphen2_HDIV_score SIFT_pred SplicAI       SpliceRegion  gnomAD_exomes_AC\n",
       "1                                                            NA              \n",
       "2 0.955                D         invalid_field invalid_field NA              \n",
       "3 0.027&.&0.027&0.027  D&D&D&D   invalid_field invalid_field NA              \n",
       "4                                                            NA              \n",
       "5                                                            NA              \n",
       "6                                                            NA              \n",
       "  gnomAD_exomes_AF gnomAD_exomes_AN pLI.x         pLI_values    occurence\n",
       "1 NA               NA                                           1        \n",
       "2 NA               NA               invalid_field invalid_field 1        \n",
       "3 NA               NA               invalid_field invalid_field 1        \n",
       "4 NA               NA                                           2        \n",
       "5 NA               NA                                           2        \n",
       "6 NA               NA                                           2        \n",
       "  probands              DP_filter12 pval      Inspected batch.x  oe_mis_pphen.x\n",
       "1 12111.s1              TRUE        0.4868502 NA        SSC      0.53448       \n",
       "2 4926035813            TRUE        0.7079808 NA        UCI      1.34320       \n",
       "3 7350647467            TRUE        0.9358089 NA        Macrogen 0.30535       \n",
       "4 1065817357/6578846017 TRUE        1.0000000 NA        Murat    0.55188       \n",
       "5 1065817357/6578846017 TRUE        0.7326841 NA        Macrogen 0.55188       \n",
       "6 1065817357/6578846017 TRUE        1.0000000 NA        Murat    0.55188       \n",
       "  pLI.y      pNull.x    pRec.x     oe_lof.x oe_lof_lower.x oe_lof_upper.x\n",
       "1 4.4135e-06 1.4907e-05 0.99998000 0.39045  0.266          0.586         \n",
       "2 1.0032e-22 9.9972e-01 0.00027792 1.09950  0.829          1.477         \n",
       "3 5.1982e-01 3.2925e-03 0.47688000 0.18868  0.076          0.594         \n",
       "4 9.8788e-01 3.4517e-08 0.01211700 0.11301  0.050          0.292         \n",
       "5 9.8788e-01 3.4517e-08 0.01211700 0.11301  0.050          0.292         \n",
       "6 9.8788e-01 3.4517e-08 0.01211700 0.11301  0.050          0.292         \n",
       "  batch.y  batch    oe_mis_pphen.y pLI        pNull.y    pRec.y     oe_lof.y\n",
       "1 SSC      SSC      0.53448        4.4135e-06 1.4907e-05 0.99998000 0.39045 \n",
       "2 UCI      UCI      1.34320        1.0032e-22 9.9972e-01 0.00027792 1.09950 \n",
       "3 Macrogen Macrogen 0.30535        5.1982e-01 3.2925e-03 0.47688000 0.18868 \n",
       "4 Murat    Murat    0.55188        9.8788e-01 3.4517e-08 0.01211700 0.11301 \n",
       "5 Macrogen Macrogen 0.55188        9.8788e-01 3.4517e-08 0.01211700 0.11301 \n",
       "6 Murat    Murat    0.55188        9.8788e-01 3.4517e-08 0.01211700 0.11301 \n",
       "  oe_lof_lower.y oe_lof_upper.y pro_dob fa_AoC mo_AoC fa_dob mo_dob pro_FA\n",
       "1 0.266          0.586          2000    29     26     1971   1974   None  \n",
       "2 0.829          1.477            NA    NA     NA       NA     NA   NA    \n",
       "3 0.076          0.594            NA    NA     NA       NA     NA   NA    \n",
       "4 0.050          0.292            NA    NA     NA       NA     NA   NA    \n",
       "5 0.050          0.292            NA    NA     NA       NA     NA   NA    \n",
       "6 0.050          0.292            NA    NA     NA       NA     NA   NA    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(data.fil)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "8ea480f6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "'callable'"
      ],
      "text/latex": [
       "'callable'"
      ],
      "text/markdown": [
       "'callable'"
      ],
      "text/plain": [
       "[1] \"callable\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>family_dirty</th><th scope=col>callable</th><th scope=col>batch</th><th scope=col>affected</th><th scope=col>family</th><th scope=col>pro</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>101       </td><td>65162766</td><td>Kibar</td><td>affected  </td><td>101  </td><td>S12816  </td></tr>\n",
       "\t<tr><th scope=row>2</th><td>103_S14383</td><td>78872610</td><td>Kibar</td><td>affected  </td><td>101  </td><td>S14383  </td></tr>\n",
       "\t<tr><th scope=row>3</th><td>103_S14386</td><td>79120866</td><td>Kibar</td><td>affected  </td><td>101  </td><td>S14386  </td></tr>\n",
       "\t<tr><th scope=row>4</th><td>11000     </td><td>47881923</td><td>SSC  </td><td>unaffected</td><td>11000</td><td>11000.s1</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>11002     </td><td>55431363</td><td>SSC  </td><td>unaffected</td><td>11002</td><td>11002.s1</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>11003     </td><td>62732391</td><td>SSC  </td><td>unaffected</td><td>11003</td><td>11003.s1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 6\n",
       "\\begin{tabular}{r|llllll}\n",
       "  & family\\_dirty & callable & batch & affected & family & pro\\\\\n",
       "  & <chr> & <int> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & 101        & 65162766 & Kibar & affected   & 101   & S12816  \\\\\n",
       "\t2 & 103\\_S14383 & 78872610 & Kibar & affected   & 101   & S14383  \\\\\n",
       "\t3 & 103\\_S14386 & 79120866 & Kibar & affected   & 101   & S14386  \\\\\n",
       "\t4 & 11000      & 47881923 & SSC   & unaffected & 11000 & 11000.s1\\\\\n",
       "\t5 & 11002      & 55431363 & SSC   & unaffected & 11002 & 11002.s1\\\\\n",
       "\t6 & 11003      & 62732391 & SSC   & unaffected & 11003 & 11003.s1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 6\n",
       "\n",
       "| <!--/--> | family_dirty &lt;chr&gt; | callable &lt;int&gt; | batch &lt;chr&gt; | affected &lt;chr&gt; | family &lt;chr&gt; | pro &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 1 | 101        | 65162766 | Kibar | affected   | 101   | S12816   |\n",
       "| 2 | 103_S14383 | 78872610 | Kibar | affected   | 101   | S14383   |\n",
       "| 3 | 103_S14386 | 79120866 | Kibar | affected   | 101   | S14386   |\n",
       "| 4 | 11000      | 47881923 | SSC   | unaffected | 11000 | 11000.s1 |\n",
       "| 5 | 11002      | 55431363 | SSC   | unaffected | 11002 | 11002.s1 |\n",
       "| 6 | 11003      | 62732391 | SSC   | unaffected | 11003 | 11003.s1 |\n",
       "\n"
      ],
      "text/plain": [
       "  family_dirty callable batch affected   family pro     \n",
       "1 101          65162766 Kibar affected   101    S12816  \n",
       "2 103_S14383   78872610 Kibar affected   101    S14383  \n",
       "3 103_S14386   79120866 Kibar affected   101    S14386  \n",
       "4 11000        47881923 SSC   unaffected 11000  11000.s1\n",
       "5 11002        55431363 SSC   unaffected 11002  11002.s1\n",
       "6 11003        62732391 SSC   unaffected 11003  11003.s1"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 142</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>locus</th><th scope=col>ref</th><th scope=col>alt</th><th scope=col>family</th><th scope=col>pro</th><th scope=col>dad</th><th scope=col>mom</th><th scope=col>pro_sex</th><th scope=col>pro_DP</th><th scope=col>pro_ref</th><th scope=col>pro_alt</th><th scope=col>pro_GT</th><th scope=col>pro_GQ</th><th scope=col>dad_DP</th><th scope=col>dad_ref</th><th scope=col>dad_alt</th><th scope=col>dad_GT</th><th scope=col>dad_GQ</th><th scope=col>mom_DP</th><th scope=col>mom_ref</th><th scope=col>mom_alt</th><th scope=col>mom_GT</th><th scope=col>mom_GQ</th><th scope=col>p_denovo</th><th scope=col>confidence</th><th scope=col>MQ</th><th scope=col>QD</th><th scope=col>filter</th><th scope=col>Allele</th><th scope=col>Consequence</th><th scope=col>IMPACT</th><th scope=col>SYMBOL</th><th scope=col>Gene</th><th scope=col>Feature_type</th><th scope=col>Feature</th><th scope=col>BIOTYPE</th><th scope=col>EXON</th><th scope=col>INTRON</th><th scope=col>HGVSc</th><th scope=col>HGVSp</th><th scope=col>cDNA_position</th><th scope=col>CDS_position</th><th scope=col>Protein_position</th><th scope=col>Amino_acids</th><th scope=col>Codons</th><th scope=col>Existing_variation</th><th scope=col>DISTANCE</th><th scope=col>STRAND</th><th scope=col>FLAGS</th><th scope=col>VARIANT_CLASS</th><th scope=col>SYMBOL_SOURCE</th><th scope=col>HGNC_ID</th><th scope=col>CANONICAL</th><th scope=col>MANE_SELECT</th><th scope=col>MANE_PLUS_CLINICAL</th><th scope=col>TSL</th><th scope=col>APPRIS</th><th scope=col>CCDS</th><th scope=col>ENSP</th><th scope=col>SWISSPROT</th><th scope=col>TREMBL</th><th scope=col>UNIPARC</th><th scope=col>UNIPROT_ISOFORM</th><th scope=col>GENE_PHENO</th><th scope=col>SIFT</th><th scope=col>PolyPhen</th><th scope=col>DOMAINS</th><th scope=col>miRNA</th><th scope=col>HGVS_OFFSET</th><th scope=col>AF</th><th scope=col>AFR_AF</th><th scope=col>AMR_AF</th><th scope=col>EAS_AF</th><th scope=col>EUR_AF</th><th scope=col>SAS_AF</th><th scope=col>AA_AF</th><th scope=col>EA_AF</th><th scope=col>gnomAD_AF</th><th scope=col>gnomAD_AFR_AF</th><th scope=col>gnomAD_AMR_AF</th><th scope=col>gnomAD_ASJ_AF</th><th scope=col>gnomAD_EAS_AF</th><th scope=col>gnomAD_FIN_AF</th><th scope=col>gnomAD_NFE_AF</th><th scope=col>gnomAD_OTH_AF</th><th scope=col>gnomAD_SAS_AF</th><th scope=col>MAX_AF</th><th scope=col>MAX_AF_POPS</th><th scope=col>CLIN_SIG</th><th scope=col>SOMATIC</th><th scope=col>PHENO</th><th scope=col>PUBMED</th><th scope=col>MOTIF_NAME</th><th scope=col>MOTIF_POS</th><th scope=col>HIGH_INF_POS</th><th scope=col>MOTIF_SCORE_CHANGE</th><th scope=col>TRANSCRIPTION_FACTORS</th><th scope=col>CADD_phred</th><th scope=col>DisGeNET</th><th scope=col>MPC</th><th scope=col>MTR</th><th scope=col>Mastermind</th><th scope=col>MetaSVM_pred</th><th scope=col>MetaSVM_rankscore</th><th scope=col>Phenotypes</th><th scope=col>Polyphen2_HDIV_score</th><th scope=col>SIFT_pred</th><th scope=col>SplicAI</th><th scope=col>SpliceRegion</th><th scope=col>gnomAD_exomes_AC</th><th scope=col>gnomAD_exomes_AF</th><th scope=col>gnomAD_exomes_AN</th><th scope=col>pLI.x</th><th scope=col>pLI_values</th><th scope=col>occurence</th><th scope=col>probands</th><th scope=col>DP_filter12</th><th scope=col>pval</th><th scope=col>Inspected</th><th scope=col>batch.x</th><th scope=col>oe_mis_pphen.x</th><th scope=col>pLI.y</th><th scope=col>pNull.x</th><th scope=col>pRec.x</th><th scope=col>oe_lof.x</th><th scope=col>oe_lof_lower.x</th><th scope=col>oe_lof_upper.x</th><th scope=col>batch.y</th><th scope=col>batch</th><th scope=col>oe_mis_pphen.y</th><th scope=col>pLI</th><th scope=col>pNull.y</th><th scope=col>pRec.y</th><th scope=col>oe_lof.y</th><th scope=col>oe_lof_lower.y</th><th scope=col>oe_lof_upper.y</th><th scope=col>pro_dob</th><th scope=col>fa_AoC</th><th scope=col>mo_AoC</th><th scope=col>fa_dob</th><th scope=col>mo_dob</th><th scope=col>pro_FA</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>chr1:1290349 </td><td>G</td><td>A</td><td>12111</td><td>12111.s1  </td><td>12111.fa  </td><td>12111.mo  </td><td>2</td><td> 33</td><td>19</td><td>14</td><td>0/1</td><td>99</td><td>40</td><td>40</td><td>0</td><td>0/0</td><td>76</td><td>40</td><td>40</td><td>0</td><td>0/0</td><td>67</td><td>.</td><td>hiConfDeNovo</td><td>60.00</td><td>11.4 </td><td>PASS</td><td>A</td><td>downstream_gene_variant           </td><td>MODIFIER</td><td>ACAP3 </td><td>ENSG00000131584</td><td>Transcript</td><td>ENST00000492936</td><td>retained_intron     </td><td>     </td><td></td><td>                           </td><td>                             </td><td>    </td><td>    </td><td>   </td><td>   </td><td>       </td><td>            </td><td>2041</td><td>-1</td><td>          </td><td>SNV</td><td>HGNC</td><td>HGNC:16754</td><td>   </td><td>           </td><td></td><td> 1</td><td>  </td><td>        </td><td>               </td><td>          </td><td>         </td><td>             </td><td>        </td><td>NA</td><td>              </td><td>                        </td><td>                                 </td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>       NA</td><td>       NA</td><td>       NA</td><td>NA</td><td>       NA</td><td>NA</td><td>       NA</td><td>NA</td><td>NA</td><td>       NA</td><td>          </td><td></td><td> </td><td> </td><td></td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>  NA</td><td>             </td><td>             </td><td>             </td><td>             </td><td> </td><td>     NA</td><td>             </td><td>                   </td><td>       </td><td>             </td><td>             </td><td>NA</td><td>NA</td><td>NA</td><td>             </td><td>             </td><td>1</td><td>12111.s1             </td><td>TRUE</td><td>0.4868502</td><td>NA</td><td>SSC     </td><td>0.53448</td><td>4.4135e-06</td><td>1.4907e-05</td><td>0.99998000</td><td>0.39045</td><td>0.266</td><td>0.586</td><td>SSC     </td><td>SSC     </td><td>0.53448</td><td>4.4135e-06</td><td>1.4907e-05</td><td>0.99998000</td><td>0.39045</td><td>0.266</td><td>0.586</td><td>2000</td><td>29</td><td>26</td><td>1971</td><td>1974</td><td>None</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>chr1:1486163 </td><td>G</td><td>C</td><td>7004 </td><td>4926035813</td><td>2917406449</td><td>3160220249</td><td>1</td><td> 64</td><td>30</td><td>34</td><td>0/1</td><td>99</td><td>43</td><td>43</td><td>0</td><td>0/0</td><td>88</td><td>44</td><td>44</td><td>0</td><td>0/0</td><td>91</td><td>.</td><td>hiConfDeNovo</td><td>59.68</td><td>16.27</td><td>PASS</td><td>C</td><td><span style=white-space:pre-wrap>missense_variant                  </span></td><td>MODERATE</td><td>ATAD3B</td><td>ENSG00000160072</td><td>Transcript</td><td>ENST00000673477</td><td><span style=white-space:pre-wrap>protein_coding      </span></td><td>10/16</td><td></td><td>ENST00000673477.1:c.1017G&gt;C</td><td>ENSP00000500094.1:p.Lys339Asn</td><td>1137</td><td>1017</td><td>339</td><td>K/N</td><td>aaG/aaC</td><td>COSV58019127</td><td><span style=white-space:pre-wrap>  NA</span></td><td> 1</td><td><span style=white-space:pre-wrap>          </span></td><td>SNV</td><td>HGNC</td><td>HGNC:24007</td><td>YES</td><td>NM_031921.6</td><td></td><td>NA</td><td>P1</td><td>CCDS30.1</td><td>ENSP00000500094</td><td>Q5T9A4.154</td><td><span style=white-space:pre-wrap>         </span></td><td>UPI000013E044</td><td>Q5T9A4-1</td><td>NA</td><td>deleterious(0)</td><td>possibly_damaging(0.824)</td><td>AFDB-ENSP_mappings:AF-Q5T9A4-F1.A</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td><span style=white-space:pre-wrap>       NA</span></td><td><span style=white-space:pre-wrap>       NA</span></td><td><span style=white-space:pre-wrap>       NA</span></td><td>NA</td><td><span style=white-space:pre-wrap>       NA</span></td><td>NA</td><td><span style=white-space:pre-wrap>       NA</span></td><td>NA</td><td>NA</td><td><span style=white-space:pre-wrap>       NA</span></td><td><span style=white-space:pre-wrap>          </span></td><td></td><td>1</td><td>1</td><td></td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>23.7</td><td>invalid_field</td><td>invalid_field</td><td>invalid_field</td><td>invalid_field</td><td>D</td><td>0.94306</td><td>invalid_field</td><td><span style=white-space:pre-wrap>0.955              </span></td><td><span style=white-space:pre-wrap>D      </span></td><td>invalid_field</td><td>invalid_field</td><td>NA</td><td>NA</td><td>NA</td><td>invalid_field</td><td>invalid_field</td><td>1</td><td><span style=white-space:pre-wrap>4926035813           </span></td><td>TRUE</td><td>0.7079808</td><td>NA</td><td><span style=white-space:pre-wrap>UCI     </span></td><td>1.34320</td><td>1.0032e-22</td><td>9.9972e-01</td><td>0.00027792</td><td>1.09950</td><td>0.829</td><td>1.477</td><td><span style=white-space:pre-wrap>UCI     </span></td><td><span style=white-space:pre-wrap>UCI     </span></td><td>1.34320</td><td>1.0032e-22</td><td>9.9972e-01</td><td>0.00027792</td><td>1.09950</td><td>0.829</td><td>1.477</td><td><span style=white-space:pre-wrap>  NA</span></td><td>NA</td><td>NA</td><td><span style=white-space:pre-wrap>  NA</span></td><td><span style=white-space:pre-wrap>  NA</span></td><td><span style=white-space:pre-wrap>NA  </span></td></tr>\n",
       "\t<tr><th scope=row>3</th><td>chr1:9356429 </td><td>G</td><td>A</td><td>6144 </td><td>7350647467</td><td>2898643437</td><td>4401829860</td><td>1</td><td>154</td><td>76</td><td>78</td><td>0/1</td><td>99</td><td>33</td><td>33</td><td>0</td><td>0/0</td><td>76</td><td>33</td><td>33</td><td>0</td><td>0/0</td><td>76</td><td>.</td><td>hiConfDeNovo</td><td>60.00</td><td>9.72 </td><td>PASS</td><td>A</td><td><span style=white-space:pre-wrap>missense_variant                  </span></td><td>MODERATE</td><td>SPSB1 </td><td>ENSG00000171621</td><td>Transcript</td><td>ENST00000450402</td><td><span style=white-space:pre-wrap>protein_coding      </span></td><td><span style=white-space:pre-wrap>2/2  </span></td><td></td><td>ENST00000450402.1:c.538G&gt;A </td><td>ENSP00000409235.1:p.Ala180Thr</td><td>754 </td><td>538 </td><td>180</td><td>A/T</td><td>Gcc/Acc</td><td>COSV60164629</td><td><span style=white-space:pre-wrap>  NA</span></td><td> 1</td><td>cds_end_NF</td><td>SNV</td><td>HGNC</td><td>HGNC:30628</td><td><span style=white-space:pre-wrap>   </span></td><td><span style=white-space:pre-wrap>           </span></td><td></td><td> 5</td><td><span style=white-space:pre-wrap>  </span></td><td><span style=white-space:pre-wrap>        </span></td><td>ENSP00000409235</td><td><span style=white-space:pre-wrap>          </span></td><td>A2A276.84</td><td>UPI0000EDA26A</td><td><span style=white-space:pre-wrap>        </span></td><td>NA</td><td>deleterious(0)</td><td><span style=white-space:pre-wrap>benign(0.031)           </span></td><td><span style=white-space:pre-wrap>                                 </span></td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td><span style=white-space:pre-wrap>       NA</span></td><td><span style=white-space:pre-wrap>       NA</span></td><td><span style=white-space:pre-wrap>       NA</span></td><td>NA</td><td><span style=white-space:pre-wrap>       NA</span></td><td>NA</td><td><span style=white-space:pre-wrap>       NA</span></td><td>NA</td><td>NA</td><td><span style=white-space:pre-wrap>       NA</span></td><td><span style=white-space:pre-wrap>          </span></td><td></td><td>1</td><td>1</td><td></td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>24.4</td><td>invalid_field</td><td>invalid_field</td><td>invalid_field</td><td>invalid_field</td><td>T</td><td>0.56603</td><td>invalid_field</td><td>0.027&amp;.&amp;0.027&amp;0.027</td><td>D&amp;D&amp;D&amp;D</td><td>invalid_field</td><td>invalid_field</td><td>NA</td><td>NA</td><td>NA</td><td>invalid_field</td><td>invalid_field</td><td>1</td><td><span style=white-space:pre-wrap>7350647467           </span></td><td>TRUE</td><td>0.9358089</td><td>NA</td><td>Macrogen</td><td>0.30535</td><td>5.1982e-01</td><td>3.2925e-03</td><td>0.47688000</td><td>0.18868</td><td>0.076</td><td>0.594</td><td>Macrogen</td><td>Macrogen</td><td>0.30535</td><td>5.1982e-01</td><td>3.2925e-03</td><td>0.47688000</td><td>0.18868</td><td>0.076</td><td>0.594</td><td><span style=white-space:pre-wrap>  NA</span></td><td>NA</td><td>NA</td><td><span style=white-space:pre-wrap>  NA</span></td><td><span style=white-space:pre-wrap>  NA</span></td><td><span style=white-space:pre-wrap>NA  </span></td></tr>\n",
       "\t<tr><th scope=row>4</th><td>chr1:10418892</td><td>C</td><td>T</td><td>6137 </td><td>1065817357</td><td>9484178706</td><td>6071727996</td><td>1</td><td> 21</td><td>11</td><td>10</td><td>0/1</td><td>99</td><td>34</td><td>34</td><td>0</td><td>0/0</td><td>99</td><td>35</td><td>35</td><td>0</td><td>0/0</td><td>99</td><td>.</td><td>hiConfDeNovo</td><td>60.00</td><td>9.32 </td><td>PASS</td><td>T</td><td>non_coding_transcript_exon_variant</td><td>MODIFIER</td><td><span style=white-space:pre-wrap>PGD   </span></td><td>ENSG00000142657</td><td>Transcript</td><td>ENST00000498356</td><td>processed_transcript</td><td><span style=white-space:pre-wrap>4/6  </span></td><td></td><td>ENST00000498356.1:n.433C&gt;T </td><td><span style=white-space:pre-wrap>                             </span></td><td>433 </td><td><span style=white-space:pre-wrap>    </span></td><td><span style=white-space:pre-wrap>   </span></td><td><span style=white-space:pre-wrap>   </span></td><td><span style=white-space:pre-wrap>       </span></td><td>rs766247086 </td><td><span style=white-space:pre-wrap>  NA</span></td><td> 1</td><td><span style=white-space:pre-wrap>          </span></td><td>SNV</td><td>HGNC</td><td>HGNC:8891 </td><td><span style=white-space:pre-wrap>   </span></td><td><span style=white-space:pre-wrap>           </span></td><td></td><td> 3</td><td><span style=white-space:pre-wrap>  </span></td><td><span style=white-space:pre-wrap>        </span></td><td><span style=white-space:pre-wrap>               </span></td><td><span style=white-space:pre-wrap>          </span></td><td><span style=white-space:pre-wrap>         </span></td><td><span style=white-space:pre-wrap>             </span></td><td><span style=white-space:pre-wrap>        </span></td><td>NA</td><td><span style=white-space:pre-wrap>              </span></td><td><span style=white-space:pre-wrap>                        </span></td><td><span style=white-space:pre-wrap>                                 </span></td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>3.182e-05</td><td>0.0001846</td><td>2.892e-05</td><td> 0</td><td>5.437e-05</td><td> 0</td><td>2.637e-05</td><td> 0</td><td> 0</td><td>0.0001846</td><td>gnomAD_AFR</td><td></td><td> </td><td> </td><td></td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td><span style=white-space:pre-wrap>  NA</span></td><td><span style=white-space:pre-wrap>             </span></td><td><span style=white-space:pre-wrap>             </span></td><td><span style=white-space:pre-wrap>             </span></td><td><span style=white-space:pre-wrap>             </span></td><td> </td><td><span style=white-space:pre-wrap>     NA</span></td><td><span style=white-space:pre-wrap>             </span></td><td><span style=white-space:pre-wrap>                   </span></td><td><span style=white-space:pre-wrap>       </span></td><td><span style=white-space:pre-wrap>             </span></td><td><span style=white-space:pre-wrap>             </span></td><td>NA</td><td>NA</td><td>NA</td><td><span style=white-space:pre-wrap>             </span></td><td><span style=white-space:pre-wrap>             </span></td><td>2</td><td>1065817357/6578846017</td><td>TRUE</td><td>1.0000000</td><td>NA</td><td><span style=white-space:pre-wrap>Murat   </span></td><td>0.55188</td><td>9.8788e-01</td><td>3.4517e-08</td><td>0.01211700</td><td>0.11301</td><td>0.050</td><td>0.292</td><td><span style=white-space:pre-wrap>Murat   </span></td><td><span style=white-space:pre-wrap>Murat   </span></td><td>0.55188</td><td>9.8788e-01</td><td>3.4517e-08</td><td>0.01211700</td><td>0.11301</td><td>0.050</td><td>0.292</td><td><span style=white-space:pre-wrap>  NA</span></td><td>NA</td><td>NA</td><td><span style=white-space:pre-wrap>  NA</span></td><td><span style=white-space:pre-wrap>  NA</span></td><td><span style=white-space:pre-wrap>NA  </span></td></tr>\n",
       "\t<tr><th scope=row>5</th><td>chr1:10418892</td><td>C</td><td>T</td><td>6137 </td><td>6578846017</td><td>9484178706</td><td>6071727996</td><td>1</td><td>137</td><td>71</td><td>66</td><td>0/1</td><td>99</td><td>34</td><td>34</td><td>0</td><td>0/0</td><td>99</td><td>35</td><td>35</td><td>0</td><td>0/0</td><td>99</td><td>.</td><td>hiConfDeNovo</td><td>60.00</td><td>9.32 </td><td>PASS</td><td>T</td><td>non_coding_transcript_exon_variant</td><td>MODIFIER</td><td><span style=white-space:pre-wrap>PGD   </span></td><td>ENSG00000142657</td><td>Transcript</td><td>ENST00000498356</td><td>processed_transcript</td><td><span style=white-space:pre-wrap>4/6  </span></td><td></td><td>ENST00000498356.1:n.433C&gt;T </td><td><span style=white-space:pre-wrap>                             </span></td><td>433 </td><td><span style=white-space:pre-wrap>    </span></td><td><span style=white-space:pre-wrap>   </span></td><td><span style=white-space:pre-wrap>   </span></td><td><span style=white-space:pre-wrap>       </span></td><td>rs766247086 </td><td><span style=white-space:pre-wrap>  NA</span></td><td> 1</td><td><span style=white-space:pre-wrap>          </span></td><td>SNV</td><td>HGNC</td><td>HGNC:8891 </td><td><span style=white-space:pre-wrap>   </span></td><td><span style=white-space:pre-wrap>           </span></td><td></td><td> 3</td><td><span style=white-space:pre-wrap>  </span></td><td><span style=white-space:pre-wrap>        </span></td><td><span style=white-space:pre-wrap>               </span></td><td><span style=white-space:pre-wrap>          </span></td><td><span style=white-space:pre-wrap>         </span></td><td><span style=white-space:pre-wrap>             </span></td><td><span style=white-space:pre-wrap>        </span></td><td>NA</td><td><span style=white-space:pre-wrap>              </span></td><td><span style=white-space:pre-wrap>                        </span></td><td><span style=white-space:pre-wrap>                                 </span></td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>3.182e-05</td><td>0.0001846</td><td>2.892e-05</td><td> 0</td><td>5.437e-05</td><td> 0</td><td>2.637e-05</td><td> 0</td><td> 0</td><td>0.0001846</td><td>gnomAD_AFR</td><td></td><td> </td><td> </td><td></td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td><span style=white-space:pre-wrap>  NA</span></td><td><span style=white-space:pre-wrap>             </span></td><td><span style=white-space:pre-wrap>             </span></td><td><span style=white-space:pre-wrap>             </span></td><td><span style=white-space:pre-wrap>             </span></td><td> </td><td><span style=white-space:pre-wrap>     NA</span></td><td><span style=white-space:pre-wrap>             </span></td><td><span style=white-space:pre-wrap>                   </span></td><td><span style=white-space:pre-wrap>       </span></td><td><span style=white-space:pre-wrap>             </span></td><td><span style=white-space:pre-wrap>             </span></td><td>NA</td><td>NA</td><td>NA</td><td><span style=white-space:pre-wrap>             </span></td><td><span style=white-space:pre-wrap>             </span></td><td>2</td><td>1065817357/6578846017</td><td>TRUE</td><td>0.7326841</td><td>NA</td><td>Macrogen</td><td>0.55188</td><td>9.8788e-01</td><td>3.4517e-08</td><td>0.01211700</td><td>0.11301</td><td>0.050</td><td>0.292</td><td>Macrogen</td><td>Macrogen</td><td>0.55188</td><td>9.8788e-01</td><td>3.4517e-08</td><td>0.01211700</td><td>0.11301</td><td>0.050</td><td>0.292</td><td><span style=white-space:pre-wrap>  NA</span></td><td>NA</td><td>NA</td><td><span style=white-space:pre-wrap>  NA</span></td><td><span style=white-space:pre-wrap>  NA</span></td><td><span style=white-space:pre-wrap>NA  </span></td></tr>\n",
       "\t<tr><th scope=row>6</th><td>chr1:10418892</td><td>C</td><td>T</td><td>6137 </td><td>1065817357</td><td>9484178706</td><td>6071727996</td><td>1</td><td> 21</td><td>11</td><td>10</td><td>0/1</td><td>99</td><td>34</td><td>34</td><td>0</td><td>0/0</td><td>99</td><td>35</td><td>35</td><td>0</td><td>0/0</td><td>99</td><td>.</td><td>hiConfDeNovo</td><td>60.00</td><td>9.32 </td><td>PASS</td><td>T</td><td>non_coding_transcript_exon_variant</td><td>MODIFIER</td><td><span style=white-space:pre-wrap>PGD   </span></td><td>ENSG00000142657</td><td>Transcript</td><td>ENST00000498356</td><td>processed_transcript</td><td><span style=white-space:pre-wrap>4/6  </span></td><td></td><td>ENST00000498356.1:n.433C&gt;T </td><td><span style=white-space:pre-wrap>                             </span></td><td>433 </td><td><span style=white-space:pre-wrap>    </span></td><td><span style=white-space:pre-wrap>   </span></td><td><span style=white-space:pre-wrap>   </span></td><td><span style=white-space:pre-wrap>       </span></td><td>rs766247086 </td><td><span style=white-space:pre-wrap>  NA</span></td><td> 1</td><td><span style=white-space:pre-wrap>          </span></td><td>SNV</td><td>HGNC</td><td>HGNC:8891 </td><td><span style=white-space:pre-wrap>   </span></td><td><span style=white-space:pre-wrap>           </span></td><td></td><td> 3</td><td><span style=white-space:pre-wrap>  </span></td><td><span style=white-space:pre-wrap>        </span></td><td><span style=white-space:pre-wrap>               </span></td><td><span style=white-space:pre-wrap>          </span></td><td><span style=white-space:pre-wrap>         </span></td><td><span style=white-space:pre-wrap>             </span></td><td><span style=white-space:pre-wrap>        </span></td><td>NA</td><td><span style=white-space:pre-wrap>              </span></td><td><span style=white-space:pre-wrap>                        </span></td><td><span style=white-space:pre-wrap>                                 </span></td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>3.182e-05</td><td>0.0001846</td><td>2.892e-05</td><td> 0</td><td>5.437e-05</td><td> 0</td><td>2.637e-05</td><td> 0</td><td> 0</td><td>0.0001846</td><td>gnomAD_AFR</td><td></td><td> </td><td> </td><td></td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td><span style=white-space:pre-wrap>  NA</span></td><td><span style=white-space:pre-wrap>             </span></td><td><span style=white-space:pre-wrap>             </span></td><td><span style=white-space:pre-wrap>             </span></td><td><span style=white-space:pre-wrap>             </span></td><td> </td><td><span style=white-space:pre-wrap>     NA</span></td><td><span style=white-space:pre-wrap>             </span></td><td><span style=white-space:pre-wrap>                   </span></td><td><span style=white-space:pre-wrap>       </span></td><td><span style=white-space:pre-wrap>             </span></td><td><span style=white-space:pre-wrap>             </span></td><td>NA</td><td>NA</td><td>NA</td><td><span style=white-space:pre-wrap>             </span></td><td><span style=white-space:pre-wrap>             </span></td><td>2</td><td>1065817357/6578846017</td><td>TRUE</td><td>1.0000000</td><td>NA</td><td><span style=white-space:pre-wrap>Murat   </span></td><td>0.55188</td><td>9.8788e-01</td><td>3.4517e-08</td><td>0.01211700</td><td>0.11301</td><td>0.050</td><td>0.292</td><td><span style=white-space:pre-wrap>Murat   </span></td><td><span style=white-space:pre-wrap>Murat   </span></td><td>0.55188</td><td>9.8788e-01</td><td>3.4517e-08</td><td>0.01211700</td><td>0.11301</td><td>0.050</td><td>0.292</td><td><span style=white-space:pre-wrap>  NA</span></td><td>NA</td><td>NA</td><td><span style=white-space:pre-wrap>  NA</span></td><td><span style=white-space:pre-wrap>  NA</span></td><td><span style=white-space:pre-wrap>NA  </span></td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 142\n",
       "\\begin{tabular}{r|llllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll}\n",
       "  & locus & ref & alt & family & pro & dad & mom & pro\\_sex & pro\\_DP & pro\\_ref & pro\\_alt & pro\\_GT & pro\\_GQ & dad\\_DP & dad\\_ref & dad\\_alt & dad\\_GT & dad\\_GQ & mom\\_DP & mom\\_ref & mom\\_alt & mom\\_GT & mom\\_GQ & p\\_denovo & confidence & MQ & QD & filter & Allele & Consequence & IMPACT & SYMBOL & Gene & Feature\\_type & Feature & BIOTYPE & EXON & INTRON & HGVSc & HGVSp & cDNA\\_position & CDS\\_position & Protein\\_position & Amino\\_acids & Codons & Existing\\_variation & DISTANCE & STRAND & FLAGS & VARIANT\\_CLASS & SYMBOL\\_SOURCE & HGNC\\_ID & CANONICAL & MANE\\_SELECT & MANE\\_PLUS\\_CLINICAL & TSL & APPRIS & CCDS & ENSP & SWISSPROT & TREMBL & UNIPARC & UNIPROT\\_ISOFORM & GENE\\_PHENO & SIFT & PolyPhen & DOMAINS & miRNA & HGVS\\_OFFSET & AF & AFR\\_AF & AMR\\_AF & EAS\\_AF & EUR\\_AF & SAS\\_AF & AA\\_AF & EA\\_AF & gnomAD\\_AF & gnomAD\\_AFR\\_AF & gnomAD\\_AMR\\_AF & gnomAD\\_ASJ\\_AF & gnomAD\\_EAS\\_AF & gnomAD\\_FIN\\_AF & gnomAD\\_NFE\\_AF & gnomAD\\_OTH\\_AF & gnomAD\\_SAS\\_AF & MAX\\_AF & MAX\\_AF\\_POPS & CLIN\\_SIG & SOMATIC & PHENO & PUBMED & MOTIF\\_NAME & MOTIF\\_POS & HIGH\\_INF\\_POS & MOTIF\\_SCORE\\_CHANGE & TRANSCRIPTION\\_FACTORS & CADD\\_phred & DisGeNET & MPC & MTR & Mastermind & MetaSVM\\_pred & MetaSVM\\_rankscore & Phenotypes & Polyphen2\\_HDIV\\_score & SIFT\\_pred & SplicAI & SpliceRegion & gnomAD\\_exomes\\_AC & gnomAD\\_exomes\\_AF & gnomAD\\_exomes\\_AN & pLI.x & pLI\\_values & occurence & probands & DP\\_filter12 & pval & Inspected & batch.x & oe\\_mis\\_pphen.x & pLI.y & pNull.x & pRec.x & oe\\_lof.x & oe\\_lof\\_lower.x & oe\\_lof\\_upper.x & batch.y & batch & oe\\_mis\\_pphen.y & pLI & pNull.y & pRec.y & oe\\_lof.y & oe\\_lof\\_lower.y & oe\\_lof\\_upper.y & pro\\_dob & fa\\_AoC & mo\\_AoC & fa\\_dob & mo\\_dob & pro\\_FA\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <int> & <int> & <int> & <chr> & <int> & <int> & <int> & <int> & <chr> & <int> & <int> & <int> & <int> & <chr> & <int> & <chr> & <chr> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <chr> & <chr> & <chr> & <lgl> & <int> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <lgl> & <lgl> & <lgl> & <lgl> & <lgl> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <dbl> & <int> & <chr> & <chr> & <int> & <chr> & <lgl> & <dbl> & <int> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <int> & <int> & <int> & <int> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & chr1:1290349  & G & A & 12111 & 12111.s1   & 12111.fa   & 12111.mo   & 2 &  33 & 19 & 14 & 0/1 & 99 & 40 & 40 & 0 & 0/0 & 76 & 40 & 40 & 0 & 0/0 & 67 & . & hiConfDeNovo & 60.00 & 11.4  & PASS & A & downstream\\_gene\\_variant            & MODIFIER & ACAP3  & ENSG00000131584 & Transcript & ENST00000492936 & retained\\_intron      &       &  &                             &                               &      &      &     &     &         &              & 2041 & -1 &            & SNV & HGNC & HGNC:16754 &     &             &  &  1 &    &          &                 &            &           &               &          & NA &                &                          &                                   & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA &        NA &        NA &        NA & NA &        NA & NA &        NA & NA & NA &        NA &            &  &   &   &  & NA & NA & NA & NA & NA &   NA &               &               &               &               &   &      NA &               &                     &         &               &               & NA & NA & NA &               &               & 1 & 12111.s1              & TRUE & 0.4868502 & NA & SSC      & 0.53448 & 4.4135e-06 & 1.4907e-05 & 0.99998000 & 0.39045 & 0.266 & 0.586 & SSC      & SSC      & 0.53448 & 4.4135e-06 & 1.4907e-05 & 0.99998000 & 0.39045 & 0.266 & 0.586 & 2000 & 29 & 26 & 1971 & 1974 & None\\\\\n",
       "\t2 & chr1:1486163  & G & C & 7004  & 4926035813 & 2917406449 & 3160220249 & 1 &  64 & 30 & 34 & 0/1 & 99 & 43 & 43 & 0 & 0/0 & 88 & 44 & 44 & 0 & 0/0 & 91 & . & hiConfDeNovo & 59.68 & 16.27 & PASS & C & missense\\_variant                   & MODERATE & ATAD3B & ENSG00000160072 & Transcript & ENST00000673477 & protein\\_coding       & 10/16 &  & ENST00000673477.1:c.1017G>C & ENSP00000500094.1:p.Lys339Asn & 1137 & 1017 & 339 & K/N & aaG/aaC & COSV58019127 &   NA &  1 &            & SNV & HGNC & HGNC:24007 & YES & NM\\_031921.6 &  & NA & P1 & CCDS30.1 & ENSP00000500094 & Q5T9A4.154 &           & UPI000013E044 & Q5T9A4-1 & NA & deleterious(0) & possibly\\_damaging(0.824) & AFDB-ENSP\\_mappings:AF-Q5T9A4-F1.A & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA &        NA &        NA &        NA & NA &        NA & NA &        NA & NA & NA &        NA &            &  & 1 & 1 &  & NA & NA & NA & NA & NA & 23.7 & invalid\\_field & invalid\\_field & invalid\\_field & invalid\\_field & D & 0.94306 & invalid\\_field & 0.955               & D       & invalid\\_field & invalid\\_field & NA & NA & NA & invalid\\_field & invalid\\_field & 1 & 4926035813            & TRUE & 0.7079808 & NA & UCI      & 1.34320 & 1.0032e-22 & 9.9972e-01 & 0.00027792 & 1.09950 & 0.829 & 1.477 & UCI      & UCI      & 1.34320 & 1.0032e-22 & 9.9972e-01 & 0.00027792 & 1.09950 & 0.829 & 1.477 &   NA & NA & NA &   NA &   NA & NA  \\\\\n",
       "\t3 & chr1:9356429  & G & A & 6144  & 7350647467 & 2898643437 & 4401829860 & 1 & 154 & 76 & 78 & 0/1 & 99 & 33 & 33 & 0 & 0/0 & 76 & 33 & 33 & 0 & 0/0 & 76 & . & hiConfDeNovo & 60.00 & 9.72  & PASS & A & missense\\_variant                   & MODERATE & SPSB1  & ENSG00000171621 & Transcript & ENST00000450402 & protein\\_coding       & 2/2   &  & ENST00000450402.1:c.538G>A  & ENSP00000409235.1:p.Ala180Thr & 754  & 538  & 180 & A/T & Gcc/Acc & COSV60164629 &   NA &  1 & cds\\_end\\_NF & SNV & HGNC & HGNC:30628 &     &             &  &  5 &    &          & ENSP00000409235 &            & A2A276.84 & UPI0000EDA26A &          & NA & deleterious(0) & benign(0.031)            &                                   & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA &        NA &        NA &        NA & NA &        NA & NA &        NA & NA & NA &        NA &            &  & 1 & 1 &  & NA & NA & NA & NA & NA & 24.4 & invalid\\_field & invalid\\_field & invalid\\_field & invalid\\_field & T & 0.56603 & invalid\\_field & 0.027\\&.\\&0.027\\&0.027 & D\\&D\\&D\\&D & invalid\\_field & invalid\\_field & NA & NA & NA & invalid\\_field & invalid\\_field & 1 & 7350647467            & TRUE & 0.9358089 & NA & Macrogen & 0.30535 & 5.1982e-01 & 3.2925e-03 & 0.47688000 & 0.18868 & 0.076 & 0.594 & Macrogen & Macrogen & 0.30535 & 5.1982e-01 & 3.2925e-03 & 0.47688000 & 0.18868 & 0.076 & 0.594 &   NA & NA & NA &   NA &   NA & NA  \\\\\n",
       "\t4 & chr1:10418892 & C & T & 6137  & 1065817357 & 9484178706 & 6071727996 & 1 &  21 & 11 & 10 & 0/1 & 99 & 34 & 34 & 0 & 0/0 & 99 & 35 & 35 & 0 & 0/0 & 99 & . & hiConfDeNovo & 60.00 & 9.32  & PASS & T & non\\_coding\\_transcript\\_exon\\_variant & MODIFIER & PGD    & ENSG00000142657 & Transcript & ENST00000498356 & processed\\_transcript & 4/6   &  & ENST00000498356.1:n.433C>T  &                               & 433  &      &     &     &         & rs766247086  &   NA &  1 &            & SNV & HGNC & HGNC:8891  &     &             &  &  3 &    &          &                 &            &           &               &          & NA &                &                          &                                   & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & 3.182e-05 & 0.0001846 & 2.892e-05 &  0 & 5.437e-05 &  0 & 2.637e-05 &  0 &  0 & 0.0001846 & gnomAD\\_AFR &  &   &   &  & NA & NA & NA & NA & NA &   NA &               &               &               &               &   &      NA &               &                     &         &               &               & NA & NA & NA &               &               & 2 & 1065817357/6578846017 & TRUE & 1.0000000 & NA & Murat    & 0.55188 & 9.8788e-01 & 3.4517e-08 & 0.01211700 & 0.11301 & 0.050 & 0.292 & Murat    & Murat    & 0.55188 & 9.8788e-01 & 3.4517e-08 & 0.01211700 & 0.11301 & 0.050 & 0.292 &   NA & NA & NA &   NA &   NA & NA  \\\\\n",
       "\t5 & chr1:10418892 & C & T & 6137  & 6578846017 & 9484178706 & 6071727996 & 1 & 137 & 71 & 66 & 0/1 & 99 & 34 & 34 & 0 & 0/0 & 99 & 35 & 35 & 0 & 0/0 & 99 & . & hiConfDeNovo & 60.00 & 9.32  & PASS & T & non\\_coding\\_transcript\\_exon\\_variant & MODIFIER & PGD    & ENSG00000142657 & Transcript & ENST00000498356 & processed\\_transcript & 4/6   &  & ENST00000498356.1:n.433C>T  &                               & 433  &      &     &     &         & rs766247086  &   NA &  1 &            & SNV & HGNC & HGNC:8891  &     &             &  &  3 &    &          &                 &            &           &               &          & NA &                &                          &                                   & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & 3.182e-05 & 0.0001846 & 2.892e-05 &  0 & 5.437e-05 &  0 & 2.637e-05 &  0 &  0 & 0.0001846 & gnomAD\\_AFR &  &   &   &  & NA & NA & NA & NA & NA &   NA &               &               &               &               &   &      NA &               &                     &         &               &               & NA & NA & NA &               &               & 2 & 1065817357/6578846017 & TRUE & 0.7326841 & NA & Macrogen & 0.55188 & 9.8788e-01 & 3.4517e-08 & 0.01211700 & 0.11301 & 0.050 & 0.292 & Macrogen & Macrogen & 0.55188 & 9.8788e-01 & 3.4517e-08 & 0.01211700 & 0.11301 & 0.050 & 0.292 &   NA & NA & NA &   NA &   NA & NA  \\\\\n",
       "\t6 & chr1:10418892 & C & T & 6137  & 1065817357 & 9484178706 & 6071727996 & 1 &  21 & 11 & 10 & 0/1 & 99 & 34 & 34 & 0 & 0/0 & 99 & 35 & 35 & 0 & 0/0 & 99 & . & hiConfDeNovo & 60.00 & 9.32  & PASS & T & non\\_coding\\_transcript\\_exon\\_variant & MODIFIER & PGD    & ENSG00000142657 & Transcript & ENST00000498356 & processed\\_transcript & 4/6   &  & ENST00000498356.1:n.433C>T  &                               & 433  &      &     &     &         & rs766247086  &   NA &  1 &            & SNV & HGNC & HGNC:8891  &     &             &  &  3 &    &          &                 &            &           &               &          & NA &                &                          &                                   & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & 3.182e-05 & 0.0001846 & 2.892e-05 &  0 & 5.437e-05 &  0 & 2.637e-05 &  0 &  0 & 0.0001846 & gnomAD\\_AFR &  &   &   &  & NA & NA & NA & NA & NA &   NA &               &               &               &               &   &      NA &               &                     &         &               &               & NA & NA & NA &               &               & 2 & 1065817357/6578846017 & TRUE & 1.0000000 & NA & Murat    & 0.55188 & 9.8788e-01 & 3.4517e-08 & 0.01211700 & 0.11301 & 0.050 & 0.292 & Murat    & Murat    & 0.55188 & 9.8788e-01 & 3.4517e-08 & 0.01211700 & 0.11301 & 0.050 & 0.292 &   NA & NA & NA &   NA &   NA & NA  \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 142\n",
       "\n",
       "| <!--/--> | locus &lt;chr&gt; | ref &lt;chr&gt; | alt &lt;chr&gt; | family &lt;chr&gt; | pro &lt;chr&gt; | dad &lt;chr&gt; | mom &lt;chr&gt; | pro_sex &lt;int&gt; | pro_DP &lt;int&gt; | pro_ref &lt;int&gt; | pro_alt &lt;int&gt; | pro_GT &lt;chr&gt; | pro_GQ &lt;int&gt; | dad_DP &lt;int&gt; | dad_ref &lt;int&gt; | dad_alt &lt;int&gt; | dad_GT &lt;chr&gt; | dad_GQ &lt;int&gt; | mom_DP &lt;int&gt; | mom_ref &lt;int&gt; | mom_alt &lt;int&gt; | mom_GT &lt;chr&gt; | mom_GQ &lt;int&gt; | p_denovo &lt;chr&gt; | confidence &lt;chr&gt; | MQ &lt;dbl&gt; | QD &lt;chr&gt; | filter &lt;chr&gt; | Allele &lt;chr&gt; | Consequence &lt;chr&gt; | IMPACT &lt;chr&gt; | SYMBOL &lt;chr&gt; | Gene &lt;chr&gt; | Feature_type &lt;chr&gt; | Feature &lt;chr&gt; | BIOTYPE &lt;chr&gt; | EXON &lt;chr&gt; | INTRON &lt;chr&gt; | HGVSc &lt;chr&gt; | HGVSp &lt;chr&gt; | cDNA_position &lt;chr&gt; | CDS_position &lt;chr&gt; | Protein_position &lt;chr&gt; | Amino_acids &lt;chr&gt; | Codons &lt;chr&gt; | Existing_variation &lt;chr&gt; | DISTANCE &lt;int&gt; | STRAND &lt;int&gt; | FLAGS &lt;chr&gt; | VARIANT_CLASS &lt;chr&gt; | SYMBOL_SOURCE &lt;chr&gt; | HGNC_ID &lt;chr&gt; | CANONICAL &lt;chr&gt; | MANE_SELECT &lt;chr&gt; | MANE_PLUS_CLINICAL &lt;chr&gt; | TSL &lt;int&gt; | APPRIS &lt;chr&gt; | CCDS &lt;chr&gt; | ENSP &lt;chr&gt; | SWISSPROT &lt;chr&gt; | TREMBL &lt;chr&gt; | UNIPARC &lt;chr&gt; | UNIPROT_ISOFORM &lt;chr&gt; | GENE_PHENO &lt;int&gt; | SIFT &lt;chr&gt; | PolyPhen &lt;chr&gt; | DOMAINS &lt;chr&gt; | miRNA &lt;lgl&gt; | HGVS_OFFSET &lt;int&gt; | AF &lt;dbl&gt; | AFR_AF &lt;dbl&gt; | AMR_AF &lt;dbl&gt; | EAS_AF &lt;dbl&gt; | EUR_AF &lt;dbl&gt; | SAS_AF &lt;dbl&gt; | AA_AF &lt;dbl&gt; | EA_AF &lt;dbl&gt; | gnomAD_AF &lt;dbl&gt; | gnomAD_AFR_AF &lt;dbl&gt; | gnomAD_AMR_AF &lt;dbl&gt; | gnomAD_ASJ_AF &lt;dbl&gt; | gnomAD_EAS_AF &lt;dbl&gt; | gnomAD_FIN_AF &lt;dbl&gt; | gnomAD_NFE_AF &lt;dbl&gt; | gnomAD_OTH_AF &lt;dbl&gt; | gnomAD_SAS_AF &lt;dbl&gt; | MAX_AF &lt;dbl&gt; | MAX_AF_POPS &lt;chr&gt; | CLIN_SIG &lt;chr&gt; | SOMATIC &lt;chr&gt; | PHENO &lt;chr&gt; | PUBMED &lt;chr&gt; | MOTIF_NAME &lt;lgl&gt; | MOTIF_POS &lt;lgl&gt; | HIGH_INF_POS &lt;lgl&gt; | MOTIF_SCORE_CHANGE &lt;lgl&gt; | TRANSCRIPTION_FACTORS &lt;lgl&gt; | CADD_phred &lt;dbl&gt; | DisGeNET &lt;chr&gt; | MPC &lt;chr&gt; | MTR &lt;chr&gt; | Mastermind &lt;chr&gt; | MetaSVM_pred &lt;chr&gt; | MetaSVM_rankscore &lt;dbl&gt; | Phenotypes &lt;chr&gt; | Polyphen2_HDIV_score &lt;chr&gt; | SIFT_pred &lt;chr&gt; | SplicAI &lt;chr&gt; | SpliceRegion &lt;chr&gt; | gnomAD_exomes_AC &lt;int&gt; | gnomAD_exomes_AF &lt;dbl&gt; | gnomAD_exomes_AN &lt;int&gt; | pLI.x &lt;chr&gt; | pLI_values &lt;chr&gt; | occurence &lt;int&gt; | probands &lt;chr&gt; | DP_filter12 &lt;lgl&gt; | pval &lt;dbl&gt; | Inspected &lt;int&gt; | batch.x &lt;chr&gt; | oe_mis_pphen.x &lt;dbl&gt; | pLI.y &lt;dbl&gt; | pNull.x &lt;dbl&gt; | pRec.x &lt;dbl&gt; | oe_lof.x &lt;dbl&gt; | oe_lof_lower.x &lt;dbl&gt; | oe_lof_upper.x &lt;dbl&gt; | batch.y &lt;chr&gt; | batch &lt;chr&gt; | oe_mis_pphen.y &lt;dbl&gt; | pLI &lt;dbl&gt; | pNull.y &lt;dbl&gt; | pRec.y &lt;dbl&gt; | oe_lof.y &lt;dbl&gt; | oe_lof_lower.y &lt;dbl&gt; | oe_lof_upper.y &lt;dbl&gt; | pro_dob &lt;int&gt; | fa_AoC &lt;int&gt; | mo_AoC &lt;int&gt; | fa_dob &lt;int&gt; | mo_dob &lt;int&gt; | pro_FA &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | chr1:1290349  | G | A | 12111 | 12111.s1   | 12111.fa   | 12111.mo   | 2 |  33 | 19 | 14 | 0/1 | 99 | 40 | 40 | 0 | 0/0 | 76 | 40 | 40 | 0 | 0/0 | 67 | . | hiConfDeNovo | 60.00 | 11.4  | PASS | A | downstream_gene_variant            | MODIFIER | ACAP3  | ENSG00000131584 | Transcript | ENST00000492936 | retained_intron      | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> | 2041 | -1 | <!----> | SNV | HGNC | HGNC:16754 | <!----> | <!----> | <!----> |  1 | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> | NA | <!----> | <!----> | <!----> | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |        NA |        NA |        NA | NA |        NA | NA |        NA | NA | NA |        NA | <!----> | <!----> | <!----> | <!----> | <!----> | NA | NA | NA | NA | NA |   NA | <!----> | <!----> | <!----> | <!----> | <!----> |      NA | <!----> | <!----> | <!----> | <!----> | <!----> | NA | NA | NA | <!----> | <!----> | 1 | 12111.s1              | TRUE | 0.4868502 | NA | SSC      | 0.53448 | 4.4135e-06 | 1.4907e-05 | 0.99998000 | 0.39045 | 0.266 | 0.586 | SSC      | SSC      | 0.53448 | 4.4135e-06 | 1.4907e-05 | 0.99998000 | 0.39045 | 0.266 | 0.586 | 2000 | 29 | 26 | 1971 | 1974 | None |\n",
       "| 2 | chr1:1486163  | G | C | 7004  | 4926035813 | 2917406449 | 3160220249 | 1 |  64 | 30 | 34 | 0/1 | 99 | 43 | 43 | 0 | 0/0 | 88 | 44 | 44 | 0 | 0/0 | 91 | . | hiConfDeNovo | 59.68 | 16.27 | PASS | C | missense_variant                   | MODERATE | ATAD3B | ENSG00000160072 | Transcript | ENST00000673477 | protein_coding       | 10/16 | <!----> | ENST00000673477.1:c.1017G&gt;C | ENSP00000500094.1:p.Lys339Asn | 1137 | 1017 | 339 | K/N | aaG/aaC | COSV58019127 |   NA |  1 | <!----> | SNV | HGNC | HGNC:24007 | YES | NM_031921.6 | <!----> | NA | P1 | CCDS30.1 | ENSP00000500094 | Q5T9A4.154 | <!----> | UPI000013E044 | Q5T9A4-1 | NA | deleterious(0) | possibly_damaging(0.824) | AFDB-ENSP_mappings:AF-Q5T9A4-F1.A | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |        NA |        NA |        NA | NA |        NA | NA |        NA | NA | NA |        NA | <!----> | <!----> | 1 | 1 | <!----> | NA | NA | NA | NA | NA | 23.7 | invalid_field | invalid_field | invalid_field | invalid_field | D | 0.94306 | invalid_field | 0.955               | D       | invalid_field | invalid_field | NA | NA | NA | invalid_field | invalid_field | 1 | 4926035813            | TRUE | 0.7079808 | NA | UCI      | 1.34320 | 1.0032e-22 | 9.9972e-01 | 0.00027792 | 1.09950 | 0.829 | 1.477 | UCI      | UCI      | 1.34320 | 1.0032e-22 | 9.9972e-01 | 0.00027792 | 1.09950 | 0.829 | 1.477 |   NA | NA | NA |   NA |   NA | NA   |\n",
       "| 3 | chr1:9356429  | G | A | 6144  | 7350647467 | 2898643437 | 4401829860 | 1 | 154 | 76 | 78 | 0/1 | 99 | 33 | 33 | 0 | 0/0 | 76 | 33 | 33 | 0 | 0/0 | 76 | . | hiConfDeNovo | 60.00 | 9.72  | PASS | A | missense_variant                   | MODERATE | SPSB1  | ENSG00000171621 | Transcript | ENST00000450402 | protein_coding       | 2/2   | <!----> | ENST00000450402.1:c.538G&gt;A  | ENSP00000409235.1:p.Ala180Thr | 754  | 538  | 180 | A/T | Gcc/Acc | COSV60164629 |   NA |  1 | cds_end_NF | SNV | HGNC | HGNC:30628 | <!----> | <!----> | <!----> |  5 | <!----> | <!----> | ENSP00000409235 | <!----> | A2A276.84 | UPI0000EDA26A | <!----> | NA | deleterious(0) | benign(0.031)            | <!----> | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |        NA |        NA |        NA | NA |        NA | NA |        NA | NA | NA |        NA | <!----> | <!----> | 1 | 1 | <!----> | NA | NA | NA | NA | NA | 24.4 | invalid_field | invalid_field | invalid_field | invalid_field | T | 0.56603 | invalid_field | 0.027&amp;.&amp;0.027&amp;0.027 | D&amp;D&amp;D&amp;D | invalid_field | invalid_field | NA | NA | NA | invalid_field | invalid_field | 1 | 7350647467            | TRUE | 0.9358089 | NA | Macrogen | 0.30535 | 5.1982e-01 | 3.2925e-03 | 0.47688000 | 0.18868 | 0.076 | 0.594 | Macrogen | Macrogen | 0.30535 | 5.1982e-01 | 3.2925e-03 | 0.47688000 | 0.18868 | 0.076 | 0.594 |   NA | NA | NA |   NA |   NA | NA   |\n",
       "| 4 | chr1:10418892 | C | T | 6137  | 1065817357 | 9484178706 | 6071727996 | 1 |  21 | 11 | 10 | 0/1 | 99 | 34 | 34 | 0 | 0/0 | 99 | 35 | 35 | 0 | 0/0 | 99 | . | hiConfDeNovo | 60.00 | 9.32  | PASS | T | non_coding_transcript_exon_variant | MODIFIER | PGD    | ENSG00000142657 | Transcript | ENST00000498356 | processed_transcript | 4/6   | <!----> | ENST00000498356.1:n.433C&gt;T  | <!----> | 433  | <!----> | <!----> | <!----> | <!----> | rs766247086  |   NA |  1 | <!----> | SNV | HGNC | HGNC:8891  | <!----> | <!----> | <!----> |  3 | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> | NA | <!----> | <!----> | <!----> | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | 3.182e-05 | 0.0001846 | 2.892e-05 |  0 | 5.437e-05 |  0 | 2.637e-05 |  0 |  0 | 0.0001846 | gnomAD_AFR | <!----> | <!----> | <!----> | <!----> | NA | NA | NA | NA | NA |   NA | <!----> | <!----> | <!----> | <!----> | <!----> |      NA | <!----> | <!----> | <!----> | <!----> | <!----> | NA | NA | NA | <!----> | <!----> | 2 | 1065817357/6578846017 | TRUE | 1.0000000 | NA | Murat    | 0.55188 | 9.8788e-01 | 3.4517e-08 | 0.01211700 | 0.11301 | 0.050 | 0.292 | Murat    | Murat    | 0.55188 | 9.8788e-01 | 3.4517e-08 | 0.01211700 | 0.11301 | 0.050 | 0.292 |   NA | NA | NA |   NA |   NA | NA   |\n",
       "| 5 | chr1:10418892 | C | T | 6137  | 6578846017 | 9484178706 | 6071727996 | 1 | 137 | 71 | 66 | 0/1 | 99 | 34 | 34 | 0 | 0/0 | 99 | 35 | 35 | 0 | 0/0 | 99 | . | hiConfDeNovo | 60.00 | 9.32  | PASS | T | non_coding_transcript_exon_variant | MODIFIER | PGD    | ENSG00000142657 | Transcript | ENST00000498356 | processed_transcript | 4/6   | <!----> | ENST00000498356.1:n.433C&gt;T  | <!----> | 433  | <!----> | <!----> | <!----> | <!----> | rs766247086  |   NA |  1 | <!----> | SNV | HGNC | HGNC:8891  | <!----> | <!----> | <!----> |  3 | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> | NA | <!----> | <!----> | <!----> | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | 3.182e-05 | 0.0001846 | 2.892e-05 |  0 | 5.437e-05 |  0 | 2.637e-05 |  0 |  0 | 0.0001846 | gnomAD_AFR | <!----> | <!----> | <!----> | <!----> | NA | NA | NA | NA | NA |   NA | <!----> | <!----> | <!----> | <!----> | <!----> |      NA | <!----> | <!----> | <!----> | <!----> | <!----> | NA | NA | NA | <!----> | <!----> | 2 | 1065817357/6578846017 | TRUE | 0.7326841 | NA | Macrogen | 0.55188 | 9.8788e-01 | 3.4517e-08 | 0.01211700 | 0.11301 | 0.050 | 0.292 | Macrogen | Macrogen | 0.55188 | 9.8788e-01 | 3.4517e-08 | 0.01211700 | 0.11301 | 0.050 | 0.292 |   NA | NA | NA |   NA |   NA | NA   |\n",
       "| 6 | chr1:10418892 | C | T | 6137  | 1065817357 | 9484178706 | 6071727996 | 1 |  21 | 11 | 10 | 0/1 | 99 | 34 | 34 | 0 | 0/0 | 99 | 35 | 35 | 0 | 0/0 | 99 | . | hiConfDeNovo | 60.00 | 9.32  | PASS | T | non_coding_transcript_exon_variant | MODIFIER | PGD    | ENSG00000142657 | Transcript | ENST00000498356 | processed_transcript | 4/6   | <!----> | ENST00000498356.1:n.433C&gt;T  | <!----> | 433  | <!----> | <!----> | <!----> | <!----> | rs766247086  |   NA |  1 | <!----> | SNV | HGNC | HGNC:8891  | <!----> | <!----> | <!----> |  3 | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> | NA | <!----> | <!----> | <!----> | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | 3.182e-05 | 0.0001846 | 2.892e-05 |  0 | 5.437e-05 |  0 | 2.637e-05 |  0 |  0 | 0.0001846 | gnomAD_AFR | <!----> | <!----> | <!----> | <!----> | NA | NA | NA | NA | NA |   NA | <!----> | <!----> | <!----> | <!----> | <!----> |      NA | <!----> | <!----> | <!----> | <!----> | <!----> | NA | NA | NA | <!----> | <!----> | 2 | 1065817357/6578846017 | TRUE | 1.0000000 | NA | Murat    | 0.55188 | 9.8788e-01 | 3.4517e-08 | 0.01211700 | 0.11301 | 0.050 | 0.292 | Murat    | Murat    | 0.55188 | 9.8788e-01 | 3.4517e-08 | 0.01211700 | 0.11301 | 0.050 | 0.292 |   NA | NA | NA |   NA |   NA | NA   |\n",
       "\n"
      ],
      "text/plain": [
       "  locus         ref alt family pro        dad        mom        pro_sex pro_DP\n",
       "1 chr1:1290349  G   A   12111  12111.s1   12111.fa   12111.mo   2        33   \n",
       "2 chr1:1486163  G   C   7004   4926035813 2917406449 3160220249 1        64   \n",
       "3 chr1:9356429  G   A   6144   7350647467 2898643437 4401829860 1       154   \n",
       "4 chr1:10418892 C   T   6137   1065817357 9484178706 6071727996 1        21   \n",
       "5 chr1:10418892 C   T   6137   6578846017 9484178706 6071727996 1       137   \n",
       "6 chr1:10418892 C   T   6137   1065817357 9484178706 6071727996 1        21   \n",
       "  pro_ref pro_alt pro_GT pro_GQ dad_DP dad_ref dad_alt dad_GT dad_GQ mom_DP\n",
       "1 19      14      0/1    99     40     40      0       0/0    76     40    \n",
       "2 30      34      0/1    99     43     43      0       0/0    88     44    \n",
       "3 76      78      0/1    99     33     33      0       0/0    76     33    \n",
       "4 11      10      0/1    99     34     34      0       0/0    99     35    \n",
       "5 71      66      0/1    99     34     34      0       0/0    99     35    \n",
       "6 11      10      0/1    99     34     34      0       0/0    99     35    \n",
       "  mom_ref mom_alt mom_GT mom_GQ p_denovo confidence   MQ    QD    filter Allele\n",
       "1 40      0       0/0    67     .        hiConfDeNovo 60.00 11.4  PASS   A     \n",
       "2 44      0       0/0    91     .        hiConfDeNovo 59.68 16.27 PASS   C     \n",
       "3 33      0       0/0    76     .        hiConfDeNovo 60.00 9.72  PASS   A     \n",
       "4 35      0       0/0    99     .        hiConfDeNovo 60.00 9.32  PASS   T     \n",
       "5 35      0       0/0    99     .        hiConfDeNovo 60.00 9.32  PASS   T     \n",
       "6 35      0       0/0    99     .        hiConfDeNovo 60.00 9.32  PASS   T     \n",
       "  Consequence                        IMPACT   SYMBOL Gene           \n",
       "1 downstream_gene_variant            MODIFIER ACAP3  ENSG00000131584\n",
       "2 missense_variant                   MODERATE ATAD3B ENSG00000160072\n",
       "3 missense_variant                   MODERATE SPSB1  ENSG00000171621\n",
       "4 non_coding_transcript_exon_variant MODIFIER PGD    ENSG00000142657\n",
       "5 non_coding_transcript_exon_variant MODIFIER PGD    ENSG00000142657\n",
       "6 non_coding_transcript_exon_variant MODIFIER PGD    ENSG00000142657\n",
       "  Feature_type Feature         BIOTYPE              EXON  INTRON\n",
       "1 Transcript   ENST00000492936 retained_intron                  \n",
       "2 Transcript   ENST00000673477 protein_coding       10/16       \n",
       "3 Transcript   ENST00000450402 protein_coding       2/2         \n",
       "4 Transcript   ENST00000498356 processed_transcript 4/6         \n",
       "5 Transcript   ENST00000498356 processed_transcript 4/6         \n",
       "6 Transcript   ENST00000498356 processed_transcript 4/6         \n",
       "  HGVSc                       HGVSp                         cDNA_position\n",
       "1                                                                        \n",
       "2 ENST00000673477.1:c.1017G>C ENSP00000500094.1:p.Lys339Asn 1137         \n",
       "3 ENST00000450402.1:c.538G>A  ENSP00000409235.1:p.Ala180Thr 754          \n",
       "4 ENST00000498356.1:n.433C>T                                433          \n",
       "5 ENST00000498356.1:n.433C>T                                433          \n",
       "6 ENST00000498356.1:n.433C>T                                433          \n",
       "  CDS_position Protein_position Amino_acids Codons  Existing_variation DISTANCE\n",
       "1                                                                      2041    \n",
       "2 1017         339              K/N         aaG/aaC COSV58019127         NA    \n",
       "3 538          180              A/T         Gcc/Acc COSV60164629         NA    \n",
       "4                                                   rs766247086          NA    \n",
       "5                                                   rs766247086          NA    \n",
       "6                                                   rs766247086          NA    \n",
       "  STRAND FLAGS      VARIANT_CLASS SYMBOL_SOURCE HGNC_ID    CANONICAL\n",
       "1 -1                SNV           HGNC          HGNC:16754          \n",
       "2  1                SNV           HGNC          HGNC:24007 YES      \n",
       "3  1     cds_end_NF SNV           HGNC          HGNC:30628          \n",
       "4  1                SNV           HGNC          HGNC:8891           \n",
       "5  1                SNV           HGNC          HGNC:8891           \n",
       "6  1                SNV           HGNC          HGNC:8891           \n",
       "  MANE_SELECT MANE_PLUS_CLINICAL TSL APPRIS CCDS     ENSP            SWISSPROT \n",
       "1                                 1                                            \n",
       "2 NM_031921.6                    NA  P1     CCDS30.1 ENSP00000500094 Q5T9A4.154\n",
       "3                                 5                  ENSP00000409235           \n",
       "4                                 3                                            \n",
       "5                                 3                                            \n",
       "6                                 3                                            \n",
       "  TREMBL    UNIPARC       UNIPROT_ISOFORM GENE_PHENO SIFT          \n",
       "1                                         NA                       \n",
       "2           UPI000013E044 Q5T9A4-1        NA         deleterious(0)\n",
       "3 A2A276.84 UPI0000EDA26A                 NA         deleterious(0)\n",
       "4                                         NA                       \n",
       "5                                         NA                       \n",
       "6                                         NA                       \n",
       "  PolyPhen                 DOMAINS                           miRNA HGVS_OFFSET\n",
       "1                                                            NA    NA         \n",
       "2 possibly_damaging(0.824) AFDB-ENSP_mappings:AF-Q5T9A4-F1.A NA    NA         \n",
       "3 benign(0.031)                                              NA    NA         \n",
       "4                                                            NA    NA         \n",
       "5                                                            NA    NA         \n",
       "6                                                            NA    NA         \n",
       "  AF AFR_AF AMR_AF EAS_AF EUR_AF SAS_AF AA_AF EA_AF gnomAD_AF gnomAD_AFR_AF\n",
       "1 NA NA     NA     NA     NA     NA     NA    NA           NA        NA    \n",
       "2 NA NA     NA     NA     NA     NA     NA    NA           NA        NA    \n",
       "3 NA NA     NA     NA     NA     NA     NA    NA           NA        NA    \n",
       "4 NA NA     NA     NA     NA     NA     NA    NA    3.182e-05 0.0001846    \n",
       "5 NA NA     NA     NA     NA     NA     NA    NA    3.182e-05 0.0001846    \n",
       "6 NA NA     NA     NA     NA     NA     NA    NA    3.182e-05 0.0001846    \n",
       "  gnomAD_AMR_AF gnomAD_ASJ_AF gnomAD_EAS_AF gnomAD_FIN_AF gnomAD_NFE_AF\n",
       "1        NA     NA                   NA     NA                   NA    \n",
       "2        NA     NA                   NA     NA                   NA    \n",
       "3        NA     NA                   NA     NA                   NA    \n",
       "4 2.892e-05      0            5.437e-05      0            2.637e-05    \n",
       "5 2.892e-05      0            5.437e-05      0            2.637e-05    \n",
       "6 2.892e-05      0            5.437e-05      0            2.637e-05    \n",
       "  gnomAD_OTH_AF gnomAD_SAS_AF MAX_AF    MAX_AF_POPS CLIN_SIG SOMATIC PHENO\n",
       "1 NA            NA                   NA                                   \n",
       "2 NA            NA                   NA                      1       1    \n",
       "3 NA            NA                   NA                      1       1    \n",
       "4  0             0            0.0001846 gnomAD_AFR                        \n",
       "5  0             0            0.0001846 gnomAD_AFR                        \n",
       "6  0             0            0.0001846 gnomAD_AFR                        \n",
       "  PUBMED MOTIF_NAME MOTIF_POS HIGH_INF_POS MOTIF_SCORE_CHANGE\n",
       "1        NA         NA        NA           NA                \n",
       "2        NA         NA        NA           NA                \n",
       "3        NA         NA        NA           NA                \n",
       "4        NA         NA        NA           NA                \n",
       "5        NA         NA        NA           NA                \n",
       "6        NA         NA        NA           NA                \n",
       "  TRANSCRIPTION_FACTORS CADD_phred DisGeNET      MPC           MTR          \n",
       "1 NA                      NA                                                \n",
       "2 NA                    23.7       invalid_field invalid_field invalid_field\n",
       "3 NA                    24.4       invalid_field invalid_field invalid_field\n",
       "4 NA                      NA                                                \n",
       "5 NA                      NA                                                \n",
       "6 NA                      NA                                                \n",
       "  Mastermind    MetaSVM_pred MetaSVM_rankscore Phenotypes   \n",
       "1                                 NA                        \n",
       "2 invalid_field D            0.94306           invalid_field\n",
       "3 invalid_field T            0.56603           invalid_field\n",
       "4                                 NA                        \n",
       "5                                 NA                        \n",
       "6                                 NA                        \n",
       "  Polyphen2_HDIV_score SIFT_pred SplicAI       SpliceRegion  gnomAD_exomes_AC\n",
       "1                                                            NA              \n",
       "2 0.955                D         invalid_field invalid_field NA              \n",
       "3 0.027&.&0.027&0.027  D&D&D&D   invalid_field invalid_field NA              \n",
       "4                                                            NA              \n",
       "5                                                            NA              \n",
       "6                                                            NA              \n",
       "  gnomAD_exomes_AF gnomAD_exomes_AN pLI.x         pLI_values    occurence\n",
       "1 NA               NA                                           1        \n",
       "2 NA               NA               invalid_field invalid_field 1        \n",
       "3 NA               NA               invalid_field invalid_field 1        \n",
       "4 NA               NA                                           2        \n",
       "5 NA               NA                                           2        \n",
       "6 NA               NA                                           2        \n",
       "  probands              DP_filter12 pval      Inspected batch.x  oe_mis_pphen.x\n",
       "1 12111.s1              TRUE        0.4868502 NA        SSC      0.53448       \n",
       "2 4926035813            TRUE        0.7079808 NA        UCI      1.34320       \n",
       "3 7350647467            TRUE        0.9358089 NA        Macrogen 0.30535       \n",
       "4 1065817357/6578846017 TRUE        1.0000000 NA        Murat    0.55188       \n",
       "5 1065817357/6578846017 TRUE        0.7326841 NA        Macrogen 0.55188       \n",
       "6 1065817357/6578846017 TRUE        1.0000000 NA        Murat    0.55188       \n",
       "  pLI.y      pNull.x    pRec.x     oe_lof.x oe_lof_lower.x oe_lof_upper.x\n",
       "1 4.4135e-06 1.4907e-05 0.99998000 0.39045  0.266          0.586         \n",
       "2 1.0032e-22 9.9972e-01 0.00027792 1.09950  0.829          1.477         \n",
       "3 5.1982e-01 3.2925e-03 0.47688000 0.18868  0.076          0.594         \n",
       "4 9.8788e-01 3.4517e-08 0.01211700 0.11301  0.050          0.292         \n",
       "5 9.8788e-01 3.4517e-08 0.01211700 0.11301  0.050          0.292         \n",
       "6 9.8788e-01 3.4517e-08 0.01211700 0.11301  0.050          0.292         \n",
       "  batch.y  batch    oe_mis_pphen.y pLI        pNull.y    pRec.y     oe_lof.y\n",
       "1 SSC      SSC      0.53448        4.4135e-06 1.4907e-05 0.99998000 0.39045 \n",
       "2 UCI      UCI      1.34320        1.0032e-22 9.9972e-01 0.00027792 1.09950 \n",
       "3 Macrogen Macrogen 0.30535        5.1982e-01 3.2925e-03 0.47688000 0.18868 \n",
       "4 Murat    Murat    0.55188        9.8788e-01 3.4517e-08 0.01211700 0.11301 \n",
       "5 Macrogen Macrogen 0.55188        9.8788e-01 3.4517e-08 0.01211700 0.11301 \n",
       "6 Murat    Murat    0.55188        9.8788e-01 3.4517e-08 0.01211700 0.11301 \n",
       "  oe_lof_lower.y oe_lof_upper.y pro_dob fa_AoC mo_AoC fa_dob mo_dob pro_FA\n",
       "1 0.266          0.586          2000    29     26     1971   1974   None  \n",
       "2 0.829          1.477            NA    NA     NA       NA     NA   NA    \n",
       "3 0.076          0.594            NA    NA     NA       NA     NA   NA    \n",
       "4 0.050          0.292            NA    NA     NA       NA     NA   NA    \n",
       "5 0.050          0.292            NA    NA     NA       NA     NA   NA    \n",
       "6 0.050          0.292            NA    NA     NA       NA     NA   NA    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "\n",
    "\n",
    "callable= read.delim(\"callable_family_DP12_220826.clean.txt\", header = FALSE)\n",
    "colnames(callable) = c(\"family_dirty\", \"callable\", \"batch\", \"affected\", \"family\", \"pro\")\n",
    "paste(\"callable\")\n",
    "head(callable)\n",
    "\n",
    "\n",
    "head(data.fil)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6859b71d",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "\n",
    "\n",
    "\"After removing poi-failed samples\"\n",
    "dim(ped)\n",
    "colnames(ped) = c('family', 'sample', 'dad', 'mom', 'sex', 'V6')\n",
    "ped.pro = ped[ped[, 3] != 0 & ped[, 4] != 0, ] # proband only\n",
    "dim(ped.pro)\n",
    "\n",
    "ped.small = ped.pro[ , c(\"family\", \"sample\")]\n",
    "head(ped.small)\n",
    "\n",
    "#Merge with callableregion file \n",
    "head(callable)\n",
    "callable <- left_join(ped.small,callable, by = 'family')\n",
    "colnames(callable)[6] = 'sample'\n",
    "head(callable)\n",
    "callable <- callable[!duplicated(callable$sample),]\n",
    "colnames(callable) <- gsub(\".x\", \"\", colnames(callable))\n",
    "callable$cohort = ifelse(grepl('SSC', callable$batch), 'SSC', 'MM')\n",
    "callable = callable[ , c(\"sample\", \"callable\")]\n",
    "head(callable)\n",
    "colnames(data.fil)[5] = \"sample\"\n",
    "colnames(ped.pro)[2] = \"sample\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "afd225e2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 7</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>family</th><th scope=col>sample</th><th scope=col>dad</th><th scope=col>mom</th><th scope=col>sex</th><th scope=col>V6</th><th scope=col>NA</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>101  </td><td>S12816  </td><td>S12818  </td><td>S12817  </td><td>1</td><td>2</td><td>Kibar</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>11000</td><td>11000.s1</td><td>11000.fa</td><td>11000.mo</td><td>2</td><td>2</td><td>SSC  </td></tr>\n",
       "\t<tr><th scope=row>3</th><td>11002</td><td>11002.s1</td><td>11002.fa</td><td>11002.mo</td><td>1</td><td>2</td><td>SSC  </td></tr>\n",
       "\t<tr><th scope=row>4</th><td>11003</td><td>11003.s1</td><td>11003.fa</td><td>11003.mo</td><td>2</td><td>2</td><td>SSC  </td></tr>\n",
       "\t<tr><th scope=row>5</th><td>11004</td><td>11004.s1</td><td>11004.fa</td><td>11004.mo</td><td>1</td><td>2</td><td>SSC  </td></tr>\n",
       "\t<tr><th scope=row>6</th><td>11005</td><td>11005.s1</td><td>11005.fa</td><td>11005.mo</td><td>1</td><td>2</td><td>SSC  </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 7\n",
       "\\begin{tabular}{r|lllllll}\n",
       "  & family & sample & dad & mom & sex & V6 & NA\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <int> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & 101   & S12816   & S12818   & S12817   & 1 & 2 & Kibar\\\\\n",
       "\t2 & 11000 & 11000.s1 & 11000.fa & 11000.mo & 2 & 2 & SSC  \\\\\n",
       "\t3 & 11002 & 11002.s1 & 11002.fa & 11002.mo & 1 & 2 & SSC  \\\\\n",
       "\t4 & 11003 & 11003.s1 & 11003.fa & 11003.mo & 2 & 2 & SSC  \\\\\n",
       "\t5 & 11004 & 11004.s1 & 11004.fa & 11004.mo & 1 & 2 & SSC  \\\\\n",
       "\t6 & 11005 & 11005.s1 & 11005.fa & 11005.mo & 1 & 2 & SSC  \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 7\n",
       "\n",
       "| <!--/--> | family &lt;chr&gt; | sample &lt;chr&gt; | dad &lt;chr&gt; | mom &lt;chr&gt; | sex &lt;int&gt; | V6 &lt;int&gt; | NA &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|\n",
       "| 1 | 101   | S12816   | S12818   | S12817   | 1 | 2 | Kibar |\n",
       "| 2 | 11000 | 11000.s1 | 11000.fa | 11000.mo | 2 | 2 | SSC   |\n",
       "| 3 | 11002 | 11002.s1 | 11002.fa | 11002.mo | 1 | 2 | SSC   |\n",
       "| 4 | 11003 | 11003.s1 | 11003.fa | 11003.mo | 2 | 2 | SSC   |\n",
       "| 5 | 11004 | 11004.s1 | 11004.fa | 11004.mo | 1 | 2 | SSC   |\n",
       "| 6 | 11005 | 11005.s1 | 11005.fa | 11005.mo | 1 | 2 | SSC   |\n",
       "\n"
      ],
      "text/plain": [
       "  family sample   dad      mom      sex V6 <NA> \n",
       "1 101    S12816   S12818   S12817   1   2  Kibar\n",
       "2 11000  11000.s1 11000.fa 11000.mo 2   2  SSC  \n",
       "3 11002  11002.s1 11002.fa 11002.mo 1   2  SSC  \n",
       "4 11003  11003.s1 11003.fa 11003.mo 2   2  SSC  \n",
       "5 11004  11004.s1 11004.fa 11004.mo 1   2  SSC  \n",
       "6 11005  11005.s1 11005.fa 11005.mo 1   2  SSC  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "\n",
    "head(ped.pro)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "414b9c9d",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Make new dataframe\n",
    "\n",
    "df = data.frame(sample = factor(ped.pro$sample))\n",
    "dim(df)\n",
    "#Add familhy\n",
    "df = merge(df, ped.pro[, c('sample', 'family', 'sex')], by.x= 'sample',by.y = 'sample', sort = F, all.x = TRUE)\n",
    "head(df)\n",
    "#Add Cohort\n",
    "df$cohort = ifelse(grepl('s', df$sample), 'SSC', 'MM')\n",
    "head(df)\n",
    "dim(df)\n",
    "#Add Callable region\n",
    "df = merge(df, callable[, c('sample', 'callable')], by= 'sample', sort = F, all.x=TRUE)\n",
    "df$callable = ifelse(is.na(df$callable), 'NA', df$callable)\n",
    "dim(df)\n",
    "head(df)\n",
    "\n",
    "samplecnt.mm = as.numeric(table(df$cohort)[1])\n",
    "samplecnt.mm\n",
    "samplecnt.ssc = as.numeric(table(df$cohort)[2])\n",
    "samplecnt.ssc\n",
    "\n",
    "head(data.all)\n",
    "colnames(data.all)[30] = \"term\"\n",
    "colnames(data.fil)[30] = \"term\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "69c12025",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Count DNMs in categories\n",
    "head(df)\n",
    "head(data.fil)\n",
    "All = as.data.frame(table(data.fil$sample))\n",
    "Coding = as.data.frame(table(data.fil[data.fil$term %in% coding_term,]$sample))\n",
    "\n",
    "Noncoding = as.data.frame(table(data.fil[!data.fil$term %in% coding_term,]$sample))\n",
    "SNV = as.data.frame(table(data.fil[data.fil$VARIANT_CLASS == 'SNV' | data.fil$VARIANT_CLASS == 'substitution',]$sample))\n",
    "INDEL= as.data.frame(table(data.fil[data.fil$VARIANT_CLASS == 'insertion' | data.fil$VARIANT_CLASS == 'deletion' | data.fil$VARIANT_CLASS == 'indel',]$sample))\n",
    "Inframe = as.data.frame(table(data.fil[data.fil$term == 'inframe_deletion' | data.fil$term == 'inframe_insertion',]$sample))\n",
    "INDEL_Coding = as.data.frame(table(data.fil[(data.fil$VARIANT_CLASS == 'insertion' | data.fil$VARIANT_CLASS == 'deletion' | data.fil$VARIANT_CLASS == 'indel') & (data.fil$term %in% coding_term) ,]$sample))\n",
    "INDEL_Noncoding = as.data.frame(table(data.fil[(data.fil$VARIANT_CLASS == 'insertion' | data.fil$VARIANT_CLASS == 'deletion' | data.fil$VARIANT_CLASS == 'indel') & (!data.fil$term %in% coding_term) ,]$sample))\n",
    "\n",
    "\n",
    "LGD = as.data.frame(table(data.fil[data.fil$IMPACT =='HIGH',]$sample))\n",
    "LGD_FS = as.data.frame(table(data.fil[data.fil$term == 'frameshift_variant',]$sample))\n",
    "LGD_SP_A = as.data.frame(table(data.fil[data.fil$term == 'splice_acceptor_variant',]$sample))\n",
    "LGD_SP_D = as.data.frame(table(data.fil[data.fil$term == 'splice_donor_variant',]$sample))\n",
    "LGD_STOP = as.data.frame(table(data.fil[data.fil$term == 'stop_gain',]$sample))\n",
    "\n",
    "#D-Mis-HC\n",
    "DMisS = as.data.frame(table(data.fil[data.fil$MetaSVM_pred == 'D',]$sample))\n",
    "DamagingS = as.data.frame(table(data.fil[data.fil$IMPACT =='HIGH' | (data.fil$MetaSVM_pred == 'D' ),]$sample))\n",
    "\n",
    "#D-Mis\n",
    "DMisMC2 = as.data.frame(table(data.fil[((grepl(\"damaging\",data.fil$PolyPhen) & data.fil$CADD_phred >= 20) | \n",
    "                                        \n",
    "                                      (grepl(\"deleterious\", data.fil$SIFT) & data.fil$CADD_phred >= 20) |\n",
    "                                      data.fil$MetaSVM_pred == 'D' ),]$sample))\n",
    "\n",
    "DamagingMC2 = as.data.frame(table(data.fil[((grepl(\"damaging\",data.fil$PolyPhen) & data.fil$CADD_phred >= 20) | \n",
    "                                        (data.fil$IMPACT =='HIGH') | \n",
    "                                      (grepl(\"deleterious\", data.fil$SIFT) & data.fil$CADD_phred >= 20) |\n",
    "                                      data.fil$MetaSVM_pred == 'D' ),]$sample))\n",
    "\n",
    "Synonymous = as.data.frame(table(data.fil[data.fil$term == 'synonymous_variant',]$sample))\n",
    "Nonsynonymous = as.data.frame(table(data.fil[data.fil$term != 'synonymous_variant' &data.fil$term %in% coding_term ,]$sample))\n",
    "Missense = as.data.frame(table(data.fil[data.fil$term == 'missense_variant',]$sample))\n",
    "OtherMissense = as.data.frame(table(data.fil[(data.fil$term == 'missense_variant') & (data.fil$MetaSVM_pred != 'D') & !(grepl(\"damaging\",data.fil$PolyPhen)),]$sample))\n",
    "\n",
    "\n",
    "\n",
    "colnames(All) = c(\"sample\", \"All\")\n",
    "colnames(Coding) = c(\"sample\", \"Coding\")\n",
    "colnames(Noncoding) = c(\"sample\", \"Noncoding\")\n",
    "colnames(SNV) = c(\"sample\", \"SNV\")\n",
    "colnames(INDEL) = c(\"sample\", \"INDEL\")\n",
    "colnames(Inframe) = c(\"sample\", \"Inframe\")\n",
    "colnames(INDEL_Coding) = c(\"sample\", \"INDEL_Coding\")\n",
    "colnames(INDEL_Noncoding) = c(\"sample\", \"INDEL_Noncoding\")\n",
    "\n",
    "colnames(LGD) = c(\"sample\", \"LGD\")\n",
    "colnames(LGD_FS) = c(\"sample\", \"LGD_FS\")\n",
    "colnames(LGD_SP_A) = c(\"sample\", \"LGD_SP_A\")\n",
    "colnames(LGD_SP_D) = c(\"sample\", \"LGD_SP_D\")\n",
    "colnames(LGD_STOP) = c(\"sample\", \"LGD_STOP\")\n",
    "\n",
    "colnames(DMisS) = c(\"sample\", \"DMis\")\n",
    "colnames(DamagingS) = c(\"sample\", \"DamagingS\")\n",
    "\n",
    "colnames(DMisMC2) = c(\"sample\", \"DMisMC2\")\n",
    "colnames(DamagingMC2) = c(\"sample\", \"DamagingMC2\") #MC = (polypehn & cadd) or (sift & cadd)\n",
    "                                          \n",
    "colnames(Synonymous) = c(\"sample\", \"Synonymous\")\n",
    "colnames(Nonsynonymous) = c(\"sample\", \"Nonsynonymous\")\n",
    "colnames(Missense) = c(\"sample\", \"Missense\")\n",
    "colnames(OtherMissense) = c(\"sample\", \"OtherMissense\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a5bd0b5d",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "\n",
    "\n",
    "df  = left_join(df, All, by = 'sample')\n",
    "df = left_join(df, Coding, by = 'sample')\n",
    "head(df)\n",
    "df$Noncoding2 = df$All - df$Coding\n",
    "df = left_join(df, Noncoding, by = 'sample')\n",
    "df = left_join(df, SNV, by = 'sample')\n",
    "df = left_join(df, INDEL, by = 'sample')\n",
    "df = left_join(df, Inframe, by = 'sample')\n",
    "df = left_join(df, INDEL_Coding, by = 'sample')\n",
    "df = left_join(df, INDEL_Noncoding, by = 'sample')\n",
    "\n",
    "\n",
    "df = left_join(df, LGD, by = 'sample')\n",
    "df = left_join(df, LGD_FS, by = 'sample')\n",
    "df = left_join(df, LGD_SP_A, by = 'sample')\n",
    "df = left_join(df, LGD_SP_D, by = 'sample')\n",
    "df = left_join(df, LGD_STOP, by = 'sample')\n",
    "\n",
    "df = left_join(df, DMisS, by = 'sample')\n",
    "df = left_join(df, DamagingS, by = 'sample')\n",
    "df = left_join(df, DMisMC2, by = 'sample')\n",
    "df = left_join(df, DamagingMC2, by = 'sample')\n",
    "\n",
    "df = left_join(df, Synonymous, by = 'sample')\n",
    "df = left_join(df, Nonsynonymous, by = 'sample')\n",
    "df = left_join(df, Missense, by = 'sample')\n",
    "df = left_join(df, OtherMissense, by = 'sample')\n",
    "df[is.na(df)] <- 0\n",
    "\n",
    "\n",
    "head(df)\n",
    "df.mm = df[df$cohort == 'MM', ]\n",
    "df.ss = df[df$cohort == 'SSC', ]\n",
    "head(df.mm)\n",
    "\n",
    "dim(df.mm)\n",
    "dim(df.ss)\n",
    "\n",
    "#Make melted one\n",
    "df.melt <- melt(df, id.vars=c('sample', 'family', 'cohort', 'callable'))\n",
    "\n",
    "\n",
    "colnames(df.melt)[5] = 'class'\n",
    "colnames(df.melt)[6] = 'variant'\n",
    "head(df.melt)\n",
    "dim(df)\n",
    "head(df)\n",
    "df.mm = df[df$cohort == 'MM', ]\n",
    "df.ss = df[df$cohort == 'SSC', ]\n",
    "\n",
    "sum(df.mm$LGD,na.rm = T)\n",
    "df.mm[is.na(df.mm)] <- 0\n",
    "df.ss[is.na(df.ss)] <- 0\n",
    "df.melt[is.na(df.melt)] <- 0\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f1e949f2",
   "metadata": {},
   "source": [
    "# Burden in Big categories"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "3793757b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<span style=white-space:pre-wrap>'consensus size is  36553428'</span>"
      ],
      "text/latex": [
       "'consensus size is  36553428'"
      ],
      "text/markdown": [
       "<span style=white-space:pre-wrap>'consensus size is  36553428'</span>"
      ],
      "text/plain": [
       "[1] \"consensus size is  36553428\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\tWilcoxon rank sum test with continuity correction\n",
      "\n",
      "data:  l_rate_mm and l_rate_ssc\n",
      "W = 291117, p-value = 0.0006177\n",
      "alternative hypothesis: true location shift is not equal to 0\n",
      "\n",
      "\n",
      "\tWelch Two Sample t-test\n",
      "\n",
      "data:  l_rate_mm and l_rate_ssc\n",
      "t = 3.3437, df = 1392.5, p-value = 0.0008486\n",
      "alternative hypothesis: true difference in means is not equal to 0\n",
      "95 percent confidence interval:\n",
      " 0.02495668 0.09580257\n",
      "sample estimates:\n",
      "mean of x mean of y \n",
      "0.1151696 0.0547900 \n",
      "\n",
      "\n",
      "\tWilcoxon rank sum test with continuity correction\n",
      "\n",
      "data:  l_rate_mm and l_rate_ssc\n",
      "W = 283314, p-value = 0.1536\n",
      "alternative hypothesis: true location shift is not equal to 0\n",
      "\n",
      "\n",
      "\tWelch Two Sample t-test\n",
      "\n",
      "data:  l_rate_mm and l_rate_ssc\n",
      "t = 1.5273, df = 1454.8, p-value = 0.1269\n",
      "alternative hypothesis: true difference in means is not equal to 0\n",
      "95 percent confidence interval:\n",
      " -0.005848614  0.046982721\n",
      "sample estimates:\n",
      " mean of x  mean of y \n",
      "0.06024257 0.03967552 \n",
      "\n",
      "\n",
      "\tWilcoxon rank sum test with continuity correction\n",
      "\n",
      "data:  l_rate_mm and l_rate_ssc\n",
      "W = 295590, p-value = 0.001461\n",
      "alternative hypothesis: true location shift is not equal to 0\n",
      "\n",
      "\n",
      "\tWelch Two Sample t-test\n",
      "\n",
      "data:  l_rate_mm and l_rate_ssc\n",
      "t = 3.0132, df = 1484, p-value = 0.002629\n",
      "alternative hypothesis: true difference in means is not equal to 0\n",
      "95 percent confidence interval:\n",
      " 0.02854865 0.13504802\n",
      "sample estimates:\n",
      "mean of x mean of y \n",
      "0.2480577 0.1662593 \n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>cohort</th><th scope=col>category</th><th scope=col>mean</th><th scope=col>lower</th><th scope=col>upper</th><th scope=col>mean_diff</th><th scope=col>mean_diff_lower</th><th scope=col>mean_diff_upper</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>MM </td><td>LGD       </td><td>0.115169624025016 </td><td>0.0865689793513251</td><td>0.143770268698707 </td><td>0.0603796209933105</td><td>0.0249566763471755</td><td>0.0958025656394456</td></tr>\n",
       "\t<tr><td>SSC</td><td>LGD       </td><td>0.0547900030317054</td><td>0.0338462364518557</td><td>0.0757337696115552</td><td>0.0603796209933105</td><td>0.0249566763471755</td><td>0.0958025656394456</td></tr>\n",
       "\t<tr><td>MM </td><td>DMis      </td><td>0.0602425725669314</td><td>0.039798740838269 </td><td>0.0806864042955939</td><td>NA                </td><td>NA                </td><td>NA                </td></tr>\n",
       "\t<tr><td>SSC</td><td>DMis      </td><td>0.0396755194367522</td><td>0.0229146241248542</td><td>0.0564364147486503</td><td>NA                </td><td>NA                </td><td>NA                </td></tr>\n",
       "\t<tr><td>MM </td><td>DamagingMC</td><td>0.248057651746188 </td><td>0.20828111335987  </td><td>0.287834190132507 </td><td>0.0817983322017028</td><td>0.0285486462435022</td><td>0.135048018159903 </td></tr>\n",
       "\t<tr><td>SSC</td><td>DamagingMC</td><td>0.166259319544485 </td><td>0.13079237519488  </td><td>0.201726263894091 </td><td>0.0817983322017028</td><td>0.0285486462435022</td><td>0.135048018159903 </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 8\n",
       "\\begin{tabular}{llllllll}\n",
       " cohort & category & mean & lower & upper & mean\\_diff & mean\\_diff\\_lower & mean\\_diff\\_upper\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t MM  & LGD        & 0.115169624025016  & 0.0865689793513251 & 0.143770268698707  & 0.0603796209933105 & 0.0249566763471755 & 0.0958025656394456\\\\\n",
       "\t SSC & LGD        & 0.0547900030317054 & 0.0338462364518557 & 0.0757337696115552 & 0.0603796209933105 & 0.0249566763471755 & 0.0958025656394456\\\\\n",
       "\t MM  & DMis       & 0.0602425725669314 & 0.039798740838269  & 0.0806864042955939 & NA                 & NA                 & NA                \\\\\n",
       "\t SSC & DMis       & 0.0396755194367522 & 0.0229146241248542 & 0.0564364147486503 & NA                 & NA                 & NA                \\\\\n",
       "\t MM  & DamagingMC & 0.248057651746188  & 0.20828111335987   & 0.287834190132507  & 0.0817983322017028 & 0.0285486462435022 & 0.135048018159903 \\\\\n",
       "\t SSC & DamagingMC & 0.166259319544485  & 0.13079237519488   & 0.201726263894091  & 0.0817983322017028 & 0.0285486462435022 & 0.135048018159903 \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 8\n",
       "\n",
       "| cohort &lt;chr&gt; | category &lt;chr&gt; | mean &lt;chr&gt; | lower &lt;chr&gt; | upper &lt;chr&gt; | mean_diff &lt;chr&gt; | mean_diff_lower &lt;chr&gt; | mean_diff_upper &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|\n",
       "| MM  | LGD        | 0.115169624025016  | 0.0865689793513251 | 0.143770268698707  | 0.0603796209933105 | 0.0249566763471755 | 0.0958025656394456 |\n",
       "| SSC | LGD        | 0.0547900030317054 | 0.0338462364518557 | 0.0757337696115552 | 0.0603796209933105 | 0.0249566763471755 | 0.0958025656394456 |\n",
       "| MM  | DMis       | 0.0602425725669314 | 0.039798740838269  | 0.0806864042955939 | NA                 | NA                 | NA                 |\n",
       "| SSC | DMis       | 0.0396755194367522 | 0.0229146241248542 | 0.0564364147486503 | NA                 | NA                 | NA                 |\n",
       "| MM  | DamagingMC | 0.248057651746188  | 0.20828111335987   | 0.287834190132507  | 0.0817983322017028 | 0.0285486462435022 | 0.135048018159903  |\n",
       "| SSC | DamagingMC | 0.166259319544485  | 0.13079237519488   | 0.201726263894091  | 0.0817983322017028 | 0.0285486462435022 | 0.135048018159903  |\n",
       "\n"
      ],
      "text/plain": [
       "  cohort category   mean               lower              upper             \n",
       "1 MM     LGD        0.115169624025016  0.0865689793513251 0.143770268698707 \n",
       "2 SSC    LGD        0.0547900030317054 0.0338462364518557 0.0757337696115552\n",
       "3 MM     DMis       0.0602425725669314 0.039798740838269  0.0806864042955939\n",
       "4 SSC    DMis       0.0396755194367522 0.0229146241248542 0.0564364147486503\n",
       "5 MM     DamagingMC 0.248057651746188  0.20828111335987   0.287834190132507 \n",
       "6 SSC    DamagingMC 0.166259319544485  0.13079237519488   0.201726263894091 \n",
       "  mean_diff          mean_diff_lower    mean_diff_upper   \n",
       "1 0.0603796209933105 0.0249566763471755 0.0958025656394456\n",
       "2 0.0603796209933105 0.0249566763471755 0.0958025656394456\n",
       "3 NA                 NA                 NA                \n",
       "4 NA                 NA                 NA                \n",
       "5 0.0817983322017028 0.0285486462435022 0.135048018159903 \n",
       "6 0.0817983322017028 0.0285486462435022 0.135048018159903 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "total_size = 59281518 #size of coding region (hg38)\n",
    "consensus = read.delim(paste(\"all_batches_\",percent, \".txt\", sep=\"\"), header = FALSE)#all_batches_70.bed\n",
    "paste(\"consensus size is \", consensus[1,])\n",
    "consensus_size = consensus[1,]\n",
    "\n",
    "Get_summary <- function(input, category){\n",
    "    \n",
    "    tmp_data = subset(df.melt, class == category)\n",
    "    head(tmp_data)\n",
    "    tmp_mm = tmp_data[tmp_data$cohort == 'MM', ]\n",
    "    tmp_ssc = tmp_data[tmp_data$cohort == 'SSC', ]\n",
    "    #Rate in consensus\n",
    "    l_rate_mm = c(as.numeric(tmp_mm$variant) * 100000000 / consensus_size / 2)\n",
    "    l_rate_ssc = c(as.numeric(tmp_ssc$variant) * 100000000 / consensus_size / 2)\n",
    "    mean_mm = mean(l_rate_mm)\n",
    "    mean_ssc = mean(l_rate_ssc)\n",
    "    ##one sample t test\n",
    "    ttest_mm = t.test(l_rate_mm, mu=mean_mm)\n",
    "    ttest_ssc = t.test(l_rate_ssc, mu=mean_ssc)\n",
    "    ##two sample t test for difference of means\n",
    "    #print (var.test(l_rate_mm, l_rate_ssc)) they have same distribution p=0.3057\n",
    "    wcox = wilcox.test(l_rate_mm, l_rate_ssc)\n",
    "    print (wcox)\n",
    "    tt = t.test(l_rate_mm, l_rate_ssc)\n",
    "    print (tt)\n",
    "    tt_pval = tt$p.value\n",
    "    ttmean_mm = tt$estimate[1]\n",
    "    ttmean_ssc = tt$estimate[2]\n",
    "    diff_mean = as.numeric(ttmean_mm - ttmean_ssc)\n",
    "    diff_mean_lower = as.numeric(tt$conf.int[1])\n",
    "    diff_mean_upper = as.numeric(tt$conf.int[2])\n",
    "    #difference for only with significant p value\n",
    "    diff_mean.final = ifelse(tt_pval < 0.05, diff_mean, NA)\n",
    "    diff_mean_lower.final = ifelse(tt_pval < 0.05, diff_mean_lower, NA)\n",
    "    diff_mean_upper.final = ifelse(tt_pval < 0.05, diff_mean_upper, NA)\n",
    "    df_tmp <- rbind(df_tmp,c('MM', category,  mean_mm, ttest_mm$conf[1], ttest_mm$conf[2], diff_mean.final, diff_mean_lower.final, diff_mean_upper.final))\n",
    "    df_tmp <- rbind(df_tmp,c('SSC', category,  mean_ssc, ttest_ssc$conf[1], ttest_ssc$conf[2], diff_mean.final, diff_mean_lower.final, diff_mean_upper.final))                  \n",
    "\n",
    "    colnames(df_tmp) = c('cohort', 'category','mean', 'lower', 'upper','mean_diff', 'mean_diff_lower', 'mean_diff_upper')\n",
    "    #df_for_diff\n",
    "    return (df_tmp)\n",
    "}\n",
    "\n",
    "df_tmp <- data.frame()\n",
    "df_for_diff = data.frame()\n",
    "\n",
    "#Test\n",
    "df_tmp <- Get_summary(df.melt, 'LGD')\n",
    "\n",
    "df_tmp <- Get_summary(df.melt, 'DMis')\n",
    "\n",
    "df_tmp <- Get_summary(df.melt, 'DamagingMC')\n",
    "#df_tmp$mean\n",
    "#df_tmp$diff\n",
    "df_tmp\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7df2b2c3",
   "metadata": {},
   "source": [
    "## For all , Coding, Noncoding rate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "7b4a0c74",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\tWilcoxon rank sum test with continuity correction\n",
      "\n",
      "data:  l_rate_mm and l_rate_ssc\n",
      "W = 284003, p-value = 0.5629\n",
      "alternative hypothesis: true location shift is not equal to 0\n",
      "\n",
      "\n",
      "\tWelch Two Sample t-test\n",
      "\n",
      "data:  l_rate_mm and l_rate_ssc\n",
      "t = 0.27979, df = 1478.7, p-value = 0.7797\n",
      "alternative hypothesis: true difference in means is not equal to 0\n",
      "95 percent confidence interval:\n",
      " -0.1169188  0.1558210\n",
      "sample estimates:\n",
      "mean of x mean of y \n",
      " 1.273953  1.254502 \n",
      "\n",
      "\n",
      "\tWilcoxon rank sum test with continuity correction\n",
      "\n",
      "data:  l_rate_mm and l_rate_ssc\n",
      "W = 284940, p-value = 0.398\n",
      "alternative hypothesis: true location shift is not equal to 0\n",
      "\n",
      "\n",
      "\tWelch Two Sample t-test\n",
      "\n",
      "data:  l_rate_mm and l_rate_ssc\n",
      "t = 1.1008, df = 1494, p-value = 0.2712\n",
      "alternative hypothesis: true difference in means is not equal to 0\n",
      "95 percent confidence interval:\n",
      " -0.03401133  0.12099793\n",
      "sample estimates:\n",
      "mean of x mean of y \n",
      "0.4553630 0.4118697 \n",
      "\n",
      "\n",
      "\tWilcoxon rank sum test with continuity correction\n",
      "\n",
      "data:  l_rate_mm and l_rate_ssc\n",
      "W = 279257, p-value = 0.9779\n",
      "alternative hypothesis: true location shift is not equal to 0\n",
      "\n",
      "\n",
      "\tWelch Two Sample t-test\n",
      "\n",
      "data:  l_rate_mm and l_rate_ssc\n",
      "t = -0.42007, df = 1467.3, p-value = 0.6745\n",
      "alternative hypothesis: true difference in means is not equal to 0\n",
      "95 percent confidence interval:\n",
      " -0.13630975  0.08822533\n",
      "sample estimates:\n",
      "mean of x mean of y \n",
      "0.8185903 0.8426325 \n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "<0 x 0 matrix>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>cohort</th><th scope=col>category</th><th scope=col>mean</th><th scope=col>lower</th><th scope=col>upper</th><th scope=col>mean_diff</th><th scope=col>mean_diff_lower</th><th scope=col>mean_diff_upper</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>NTD    </td><td>All      </td><td>1.27395322575364 </td><td>1.18086404770263 </td><td>1.36704240380464 </td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>Control</td><td>All      </td><td>1.25450213838112 </td><td>1.15469614261994 </td><td>1.3543081341423  </td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>NTD    </td><td>Coding   </td><td>0.455362974991217</td><td>0.39951877348784 </td><td>0.511207176494594</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>Control</td><td>Coding   </td><td>0.411869677962475</td><td>0.358035280663547</td><td>0.465704075261403</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>NTD    </td><td>Noncoding</td><td>0.818590250762421</td><td>0.743335196110104</td><td>0.893845305414739</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>Control</td><td>Noncoding</td><td>0.842632460418642</td><td>0.759200177207099</td><td>0.926064743630186</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 8\n",
       "\\begin{tabular}{llllllll}\n",
       " cohort & category & mean & lower & upper & mean\\_diff & mean\\_diff\\_lower & mean\\_diff\\_upper\\\\\n",
       " <fct> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t NTD     & All       & 1.27395322575364  & 1.18086404770263  & 1.36704240380464  & NA & NA & NA\\\\\n",
       "\t Control & All       & 1.25450213838112  & 1.15469614261994  & 1.3543081341423   & NA & NA & NA\\\\\n",
       "\t NTD     & Coding    & 0.455362974991217 & 0.39951877348784  & 0.511207176494594 & NA & NA & NA\\\\\n",
       "\t Control & Coding    & 0.411869677962475 & 0.358035280663547 & 0.465704075261403 & NA & NA & NA\\\\\n",
       "\t NTD     & Noncoding & 0.818590250762421 & 0.743335196110104 & 0.893845305414739 & NA & NA & NA\\\\\n",
       "\t Control & Noncoding & 0.842632460418642 & 0.759200177207099 & 0.926064743630186 & NA & NA & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 8\n",
       "\n",
       "| cohort &lt;fct&gt; | category &lt;chr&gt; | mean &lt;chr&gt; | lower &lt;chr&gt; | upper &lt;chr&gt; | mean_diff &lt;chr&gt; | mean_diff_lower &lt;chr&gt; | mean_diff_upper &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|\n",
       "| NTD     | All       | 1.27395322575364  | 1.18086404770263  | 1.36704240380464  | NA | NA | NA |\n",
       "| Control | All       | 1.25450213838112  | 1.15469614261994  | 1.3543081341423   | NA | NA | NA |\n",
       "| NTD     | Coding    | 0.455362974991217 | 0.39951877348784  | 0.511207176494594 | NA | NA | NA |\n",
       "| Control | Coding    | 0.411869677962475 | 0.358035280663547 | 0.465704075261403 | NA | NA | NA |\n",
       "| NTD     | Noncoding | 0.818590250762421 | 0.743335196110104 | 0.893845305414739 | NA | NA | NA |\n",
       "| Control | Noncoding | 0.842632460418642 | 0.759200177207099 | 0.926064743630186 | NA | NA | NA |\n",
       "\n"
      ],
      "text/plain": [
       "  cohort  category  mean              lower             upper            \n",
       "1 NTD     All       1.27395322575364  1.18086404770263  1.36704240380464 \n",
       "2 Control All       1.25450213838112  1.15469614261994  1.3543081341423  \n",
       "3 NTD     Coding    0.455362974991217 0.39951877348784  0.511207176494594\n",
       "4 Control Coding    0.411869677962475 0.358035280663547 0.465704075261403\n",
       "5 NTD     Noncoding 0.818590250762421 0.743335196110104 0.893845305414739\n",
       "6 Control Noncoding 0.842632460418642 0.759200177207099 0.926064743630186\n",
       "  mean_diff mean_diff_lower mean_diff_upper\n",
       "1 NA        NA              NA             \n",
       "2 NA        NA              NA             \n",
       "3 NA        NA              NA             \n",
       "4 NA        NA              NA             \n",
       "5 NA        NA              NA             \n",
       "6 NA        NA              NA             "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "<0 x 0 matrix>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "'===All==='"
      ],
      "text/latex": [
       "'===All==='"
      ],
      "text/markdown": [
       "'===All==='"
      ],
      "text/plain": [
       "[1] \"===All===\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "\tWilcoxon rank sum test with continuity correction\n",
       "\n",
       "data:  df.mm$All and df.ss$All\n",
       "W = 284003, p-value = 0.2814\n",
       "alternative hypothesis: true location shift is greater than 0\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<span style=white-space:pre-wrap>'Mean of mm dnm:  0.931347150259067'</span>"
      ],
      "text/latex": [
       "'Mean of mm dnm:  0.931347150259067'"
      ],
      "text/markdown": [
       "<span style=white-space:pre-wrap>'Mean of mm dnm:  0.931347150259067'</span>"
      ],
      "text/plain": [
       "[1] \"Mean of mm dnm:  0.931347150259067\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<span style=white-space:pre-wrap>'Mean of control dnm:  0.917127071823204'</span>"
      ],
      "text/latex": [
       "'Mean of control dnm:  0.917127071823204'"
      ],
      "text/markdown": [
       "<span style=white-space:pre-wrap>'Mean of control dnm:  0.917127071823204'</span>"
      ],
      "text/plain": [
       "[1] \"Mean of control dnm:  0.917127071823204\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "'===Coding==='"
      ],
      "text/latex": [
       "'===Coding==='"
      ],
      "text/markdown": [
       "'===Coding==='"
      ],
      "text/plain": [
       "[1] \"===Coding===\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "\tWilcoxon rank sum test with continuity correction\n",
       "\n",
       "data:  df.mm$Coding and df.ss$Coding\n",
       "W = 284940, p-value = 0.199\n",
       "alternative hypothesis: true location shift is greater than 0\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<span style=white-space:pre-wrap>'Mean of mm dnm:  0.332901554404145'</span>"
      ],
      "text/latex": [
       "'Mean of mm dnm:  0.332901554404145'"
      ],
      "text/markdown": [
       "<span style=white-space:pre-wrap>'Mean of mm dnm:  0.332901554404145'</span>"
      ],
      "text/plain": [
       "[1] \"Mean of mm dnm:  0.332901554404145\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<span style=white-space:pre-wrap>'Mean of control dnm:  0.301104972375691'</span>"
      ],
      "text/latex": [
       "'Mean of control dnm:  0.301104972375691'"
      ],
      "text/markdown": [
       "<span style=white-space:pre-wrap>'Mean of control dnm:  0.301104972375691'</span>"
      ],
      "text/plain": [
       "[1] \"Mean of control dnm:  0.301104972375691\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "'===Noncoding==='"
      ],
      "text/latex": [
       "'===Noncoding==='"
      ],
      "text/markdown": [
       "'===Noncoding==='"
      ],
      "text/plain": [
       "[1] \"===Noncoding===\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "\tWilcoxon rank sum test with continuity correction\n",
       "\n",
       "data:  df.mm$Noncoding and df.ss$Noncoding\n",
       "W = 279257, p-value = 0.5111\n",
       "alternative hypothesis: true location shift is greater than 0\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<span style=white-space:pre-wrap>'Mean of mm dnm:  0.598445595854922'</span>"
      ],
      "text/latex": [
       "'Mean of mm dnm:  0.598445595854922'"
      ],
      "text/markdown": [
       "<span style=white-space:pre-wrap>'Mean of mm dnm:  0.598445595854922'</span>"
      ],
      "text/plain": [
       "[1] \"Mean of mm dnm:  0.598445595854922\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<span style=white-space:pre-wrap>'Mean of control dnm:  0.616022099447514'</span>"
      ],
      "text/latex": [
       "'Mean of control dnm:  0.616022099447514'"
      ],
      "text/markdown": [
       "<span style=white-space:pre-wrap>'Mean of control dnm:  0.616022099447514'</span>"
      ],
      "text/plain": [
       "[1] \"Mean of control dnm:  0.616022099447514\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "\n",
    "\n",
    "df_rr = data.frame()\n",
    "\n",
    "\n",
    "\n",
    "df_tmp <- data.frame()\n",
    "df_tmp <- Get_summary(df.melt, 'All')\n",
    "df_tmp <- Get_summary(df.melt, 'Coding')\n",
    "df_tmp <- Get_summary(df.melt, 'Noncoding')\n",
    "df_tmp$cohort <- factor(df_tmp$cohort, levels = c('MM', 'SSC'), labels = c(\"NTD\", \"Control\"))\n",
    "df_rr\n",
    "df_tmp\n",
    "\n",
    "\n",
    "\n",
    "df_rr\n",
    "\"===All===\"\n",
    "wilcox.test(df.mm$All, df.ss$All, alternative= \"greater\")\n",
    "paste(\"Mean of mm dnm: \", mean(df.mm$All))\n",
    "paste(\"Mean of control dnm: \", mean(df.ss$All))\n",
    "\n",
    "\"===Coding===\"\n",
    "wilcox.test(df.mm$Coding, df.ss$Coding, alternative= \"greater\")\n",
    "paste(\"Mean of mm dnm: \", mean(df.mm$Coding))\n",
    "paste(\"Mean of control dnm: \", mean(df.ss$Coding))\n",
    "\n",
    "\"===Noncoding===\"\n",
    "wilcox.test(df.mm$Noncoding, df.ss$Noncoding, alternative= \"greater\")\n",
    "paste(\"Mean of mm dnm: \", mean(df.mm$Noncoding))\n",
    "paste(\"Mean of control dnm: \", mean(df.ss$Noncoding))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "557ec564",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\tWilcoxon rank sum test with continuity correction\n",
      "\n",
      "data:  l_rate_mm and l_rate_ssc\n",
      "W = 284003, p-value = 0.5629\n",
      "alternative hypothesis: true location shift is not equal to 0\n",
      "\n",
      "\n",
      "\tWelch Two Sample t-test\n",
      "\n",
      "data:  l_rate_mm and l_rate_ssc\n",
      "t = 0.27979, df = 1478.7, p-value = 0.7797\n",
      "alternative hypothesis: true difference in means is not equal to 0\n",
      "95 percent confidence interval:\n",
      " -0.1169188  0.1558210\n",
      "sample estimates:\n",
      "mean of x mean of y \n",
      " 1.273953  1.254502 \n",
      "\n",
      "\n",
      "\tWilcoxon rank sum test with continuity correction\n",
      "\n",
      "data:  l_rate_mm and l_rate_ssc\n",
      "W = 278652, p-value = 0.9174\n",
      "alternative hypothesis: true location shift is not equal to 0\n",
      "\n",
      "\n",
      "\tWelch Two Sample t-test\n",
      "\n",
      "data:  l_rate_mm and l_rate_ssc\n",
      "t = -0.2914, df = 1478.8, p-value = 0.7708\n",
      "alternative hypothesis: true difference in means is not equal to 0\n",
      "95 percent confidence interval:\n",
      " -0.1537925  0.1140093\n",
      "sample estimates:\n",
      "mean of x mean of y \n",
      " 1.211939  1.231830 \n",
      "\n",
      "\n",
      "\tWilcoxon rank sum test with continuity correction\n",
      "\n",
      "data:  l_rate_mm and l_rate_ssc\n",
      "W = 286790, p-value = 0.003024\n",
      "alternative hypothesis: true location shift is not equal to 0\n",
      "\n",
      "\n",
      "\tWelch Two Sample t-test\n",
      "\n",
      "data:  l_rate_mm and l_rate_ssc\n",
      "t = 3.1124, df = 1251.6, p-value = 0.001898\n",
      "alternative hypothesis: true difference in means is not equal to 0\n",
      "95 percent confidence interval:\n",
      " 0.01454315 0.06414222\n",
      "sample estimates:\n",
      " mean of x  mean of y \n",
      "0.06201441 0.02267173 \n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "<0 x 0 matrix>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>cohort</th><th scope=col>category</th><th scope=col>mean</th><th scope=col>lower</th><th scope=col>upper</th><th scope=col>mean_diff</th><th scope=col>mean_diff_lower</th><th scope=col>mean_diff_upper</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>NTD    </td><td>All  </td><td>1.27395322575364  </td><td>1.18086404770263   </td><td>1.36704240380464  </td><td>NA                </td><td>NA                </td><td>NA                </td></tr>\n",
       "\t<tr><td>Control</td><td>All  </td><td>1.25450213838112  </td><td>1.15469614261994   </td><td>1.3543081341423   </td><td>NA                </td><td>NA                </td><td>NA                </td></tr>\n",
       "\t<tr><td>NTD    </td><td>SNV  </td><td>1.21193881281709  </td><td>1.1205167100218    </td><td>1.30336091561238  </td><td>NA                </td><td>NA                </td><td>NA                </td></tr>\n",
       "\t<tr><td>Control</td><td>SNV  </td><td>1.23183041298869  </td><td>1.1338485587561    </td><td>1.32981226722128  </td><td>NA                </td><td>NA                </td><td>NA                </td></tr>\n",
       "\t<tr><td>NTD    </td><td>INDEL</td><td>0.0620144129365471</td><td>0.0407257226065329 </td><td>0.0833031032665613</td><td>0.0393426875441172</td><td>0.0145431549242424</td><td>0.0641422201639921</td></tr>\n",
       "\t<tr><td>Control</td><td>INDEL</td><td>0.0226717253924298</td><td>0.00992083467921379</td><td>0.0354226161056459</td><td>0.0393426875441172</td><td>0.0145431549242424</td><td>0.0641422201639921</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 8\n",
       "\\begin{tabular}{llllllll}\n",
       " cohort & category & mean & lower & upper & mean\\_diff & mean\\_diff\\_lower & mean\\_diff\\_upper\\\\\n",
       " <fct> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t NTD     & All   & 1.27395322575364   & 1.18086404770263    & 1.36704240380464   & NA                 & NA                 & NA                \\\\\n",
       "\t Control & All   & 1.25450213838112   & 1.15469614261994    & 1.3543081341423    & NA                 & NA                 & NA                \\\\\n",
       "\t NTD     & SNV   & 1.21193881281709   & 1.1205167100218     & 1.30336091561238   & NA                 & NA                 & NA                \\\\\n",
       "\t Control & SNV   & 1.23183041298869   & 1.1338485587561     & 1.32981226722128   & NA                 & NA                 & NA                \\\\\n",
       "\t NTD     & INDEL & 0.0620144129365471 & 0.0407257226065329  & 0.0833031032665613 & 0.0393426875441172 & 0.0145431549242424 & 0.0641422201639921\\\\\n",
       "\t Control & INDEL & 0.0226717253924298 & 0.00992083467921379 & 0.0354226161056459 & 0.0393426875441172 & 0.0145431549242424 & 0.0641422201639921\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 8\n",
       "\n",
       "| cohort &lt;fct&gt; | category &lt;chr&gt; | mean &lt;chr&gt; | lower &lt;chr&gt; | upper &lt;chr&gt; | mean_diff &lt;chr&gt; | mean_diff_lower &lt;chr&gt; | mean_diff_upper &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|\n",
       "| NTD     | All   | 1.27395322575364   | 1.18086404770263    | 1.36704240380464   | NA                 | NA                 | NA                 |\n",
       "| Control | All   | 1.25450213838112   | 1.15469614261994    | 1.3543081341423    | NA                 | NA                 | NA                 |\n",
       "| NTD     | SNV   | 1.21193881281709   | 1.1205167100218     | 1.30336091561238   | NA                 | NA                 | NA                 |\n",
       "| Control | SNV   | 1.23183041298869   | 1.1338485587561     | 1.32981226722128   | NA                 | NA                 | NA                 |\n",
       "| NTD     | INDEL | 0.0620144129365471 | 0.0407257226065329  | 0.0833031032665613 | 0.0393426875441172 | 0.0145431549242424 | 0.0641422201639921 |\n",
       "| Control | INDEL | 0.0226717253924298 | 0.00992083467921379 | 0.0354226161056459 | 0.0393426875441172 | 0.0145431549242424 | 0.0641422201639921 |\n",
       "\n"
      ],
      "text/plain": [
       "  cohort  category mean               lower               upper             \n",
       "1 NTD     All      1.27395322575364   1.18086404770263    1.36704240380464  \n",
       "2 Control All      1.25450213838112   1.15469614261994    1.3543081341423   \n",
       "3 NTD     SNV      1.21193881281709   1.1205167100218     1.30336091561238  \n",
       "4 Control SNV      1.23183041298869   1.1338485587561     1.32981226722128  \n",
       "5 NTD     INDEL    0.0620144129365471 0.0407257226065329  0.0833031032665613\n",
       "6 Control INDEL    0.0226717253924298 0.00992083467921379 0.0354226161056459\n",
       "  mean_diff          mean_diff_lower    mean_diff_upper   \n",
       "1 NA                 NA                 NA                \n",
       "2 NA                 NA                 NA                \n",
       "3 NA                 NA                 NA                \n",
       "4 NA                 NA                 NA                \n",
       "5 0.0393426875441172 0.0145431549242424 0.0641422201639921\n",
       "6 0.0393426875441172 0.0145431549242424 0.0641422201639921"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "<0 x 0 matrix>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "'===All==='"
      ],
      "text/latex": [
       "'===All==='"
      ],
      "text/markdown": [
       "'===All==='"
      ],
      "text/plain": [
       "[1] \"===All===\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "\tWilcoxon rank sum test with continuity correction\n",
       "\n",
       "data:  df.mm$All and df.ss$All\n",
       "W = 284003, p-value = 0.2814\n",
       "alternative hypothesis: true location shift is greater than 0\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<span style=white-space:pre-wrap>'Mean of mm dnm:  0.931347150259067'</span>"
      ],
      "text/latex": [
       "'Mean of mm dnm:  0.931347150259067'"
      ],
      "text/markdown": [
       "<span style=white-space:pre-wrap>'Mean of mm dnm:  0.931347150259067'</span>"
      ],
      "text/plain": [
       "[1] \"Mean of mm dnm:  0.931347150259067\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<span style=white-space:pre-wrap>'Mean of control dnm:  0.917127071823204'</span>"
      ],
      "text/latex": [
       "'Mean of control dnm:  0.917127071823204'"
      ],
      "text/markdown": [
       "<span style=white-space:pre-wrap>'Mean of control dnm:  0.917127071823204'</span>"
      ],
      "text/plain": [
       "[1] \"Mean of control dnm:  0.917127071823204\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "'===SNV==='"
      ],
      "text/latex": [
       "'===SNV==='"
      ],
      "text/markdown": [
       "'===SNV==='"
      ],
      "text/plain": [
       "[1] \"===SNV===\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "\tWilcoxon rank sum test with continuity correction\n",
       "\n",
       "data:  df.mm$SNV and df.ss$SNV\n",
       "W = 278652, p-value = 0.5413\n",
       "alternative hypothesis: true location shift is greater than 0\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<span style=white-space:pre-wrap>'Mean of mm dnm:  0.886010362694301'</span>"
      ],
      "text/latex": [
       "'Mean of mm dnm:  0.886010362694301'"
      ],
      "text/markdown": [
       "<span style=white-space:pre-wrap>'Mean of mm dnm:  0.886010362694301'</span>"
      ],
      "text/plain": [
       "[1] \"Mean of mm dnm:  0.886010362694301\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<span style=white-space:pre-wrap>'Mean of control dnm:  0.900552486187845'</span>"
      ],
      "text/latex": [
       "'Mean of control dnm:  0.900552486187845'"
      ],
      "text/markdown": [
       "<span style=white-space:pre-wrap>'Mean of control dnm:  0.900552486187845'</span>"
      ],
      "text/plain": [
       "[1] \"Mean of control dnm:  0.900552486187845\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "'===INDEL==='"
      ],
      "text/latex": [
       "'===INDEL==='"
      ],
      "text/markdown": [
       "'===INDEL==='"
      ],
      "text/plain": [
       "[1] \"===INDEL===\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "\tWilcoxon rank sum test with continuity correction\n",
       "\n",
       "data:  df.mm$INDEL and df.ss$INDEL\n",
       "W = 286790, p-value = 0.001512\n",
       "alternative hypothesis: true location shift is greater than 0\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<span style=white-space:pre-wrap>'Mean of mm dnm:  0.0453367875647668'</span>"
      ],
      "text/latex": [
       "'Mean of mm dnm:  0.0453367875647668'"
      ],
      "text/markdown": [
       "<span style=white-space:pre-wrap>'Mean of mm dnm:  0.0453367875647668'</span>"
      ],
      "text/plain": [
       "[1] \"Mean of mm dnm:  0.0453367875647668\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<span style=white-space:pre-wrap>'Mean of control dnm:  0.0165745856353591'</span>"
      ],
      "text/latex": [
       "'Mean of control dnm:  0.0165745856353591'"
      ],
      "text/markdown": [
       "<span style=white-space:pre-wrap>'Mean of control dnm:  0.0165745856353591'</span>"
      ],
      "text/plain": [
       "[1] \"Mean of control dnm:  0.0165745856353591\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "\n",
    "options(repr.plot.width=6, repr.plot.height=8)\n",
    "\n",
    "df_rr = data.frame()\n",
    "\n",
    "\n",
    "df_tmp <- data.frame()\n",
    "df_tmp <- Get_summary(df.melt, 'All')\n",
    "df_tmp <- Get_summary(df.melt, 'SNV')\n",
    "df_tmp <- Get_summary(df.melt, 'INDEL')\n",
    "df_tmp$cohort <- factor(df_tmp$cohort, levels = c('MM', 'SSC'), labels = c(\"NTD\", \"Control\"))\n",
    "df_rr\n",
    "df_tmp\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "df_rr\n",
    "\"===All===\"\n",
    "wilcox.test(df.mm$All, df.ss$All, alternative= \"greater\")\n",
    "paste(\"Mean of mm dnm: \", mean(df.mm$All))\n",
    "paste(\"Mean of control dnm: \", mean(df.ss$All))\n",
    "\n",
    "\"===SNV===\"\n",
    "wilcox.test(df.mm$SNV, df.ss$SNV, alternative= \"greater\")\n",
    "paste(\"Mean of mm dnm: \", mean(df.mm$SNV))\n",
    "paste(\"Mean of control dnm: \", mean(df.ss$SNV))\n",
    "\n",
    "\"===INDEL===\"\n",
    "wilcox.test(df.mm$INDEL, df.ss$INDEL, alternative= \"greater\")\n",
    "paste(\"Mean of mm dnm: \", mean(df.mm$INDEL))\n",
    "paste(\"Mean of control dnm: \", mean(df.ss$INDEL))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "181ca280",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<span style=white-space:pre-wrap>'consensus size is  36553428'</span>"
      ],
      "text/latex": [
       "'consensus size is  36553428'"
      ],
      "text/markdown": [
       "<span style=white-space:pre-wrap>'consensus size is  36553428'</span>"
      ],
      "text/plain": [
       "[1] \"consensus size is  36553428\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"All: 0.398   Rate ratio: 1.01550502528248\"\n",
      "\n",
      "\tWilcoxon rank sum test with continuity correction\n",
      "\n",
      "data:  l_rate_mm and l_rate_ssc\n",
      "W = 284003, p-value = 0.5629\n",
      "alternative hypothesis: true location shift is not equal to 0\n",
      "\n",
      "\n",
      "\tWelch Two Sample t-test\n",
      "\n",
      "data:  l_rate_mm and l_rate_ssc\n",
      "t = 0.27979, df = 1478.7, p-value = 0.7797\n",
      "alternative hypothesis: true difference in means is not equal to 0\n",
      "95 percent confidence interval:\n",
      " -0.1169188  0.1558210\n",
      "sample estimates:\n",
      "mean of x mean of y \n",
      " 1.273953  1.254502 \n",
      "\n",
      "[1] \"Coding: 0.148   Rate ratio: 1.10559965774588\"\n",
      "\n",
      "\tWilcoxon rank sum test with continuity correction\n",
      "\n",
      "data:  l_rate_mm and l_rate_ssc\n",
      "W = 284940, p-value = 0.398\n",
      "alternative hypothesis: true location shift is not equal to 0\n",
      "\n",
      "\n",
      "\tWelch Two Sample t-test\n",
      "\n",
      "data:  l_rate_mm and l_rate_ssc\n",
      "t = 1.1008, df = 1494, p-value = 0.2712\n",
      "alternative hypothesis: true difference in means is not equal to 0\n",
      "95 percent confidence interval:\n",
      " -0.03401133  0.12099793\n",
      "sample estimates:\n",
      "mean of x mean of y \n",
      "0.4553630 0.4118697 \n",
      "\n",
      "[1] \"Noncoding: 0.681   Rate ratio: 0.971467738562699\"\n",
      "\n",
      "\tWilcoxon rank sum test with continuity correction\n",
      "\n",
      "data:  l_rate_mm and l_rate_ssc\n",
      "W = 279257, p-value = 0.9779\n",
      "alternative hypothesis: true location shift is not equal to 0\n",
      "\n",
      "\n",
      "\tWelch Two Sample t-test\n",
      "\n",
      "data:  l_rate_mm and l_rate_ssc\n",
      "t = -0.42007, df = 1467.3, p-value = 0.6745\n",
      "alternative hypothesis: true difference in means is not equal to 0\n",
      "95 percent confidence interval:\n",
      " -0.13630975  0.08822533\n",
      "sample estimates:\n",
      "mean of x mean of y \n",
      "0.8185903 0.8426325 \n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cohort</th><th scope=col>category</th><th scope=col>mean</th><th scope=col>lower</th><th scope=col>upper</th><th scope=col>mean_diff</th><th scope=col>mean_diff_lower</th><th scope=col>mean_diff_upper</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>MM </td><td>All      </td><td>1.27395322575364 </td><td>1.18086404770263 </td><td>1.36704240380464 </td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>SSC</td><td>All      </td><td>1.25450213838112 </td><td>1.15469614261994 </td><td>1.3543081341423  </td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>MM </td><td>Coding   </td><td>0.455362974991217</td><td>0.39951877348784 </td><td>0.511207176494594</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>SSC</td><td>Coding   </td><td>0.411869677962475</td><td>0.358035280663547</td><td>0.465704075261403</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>MM </td><td>Noncoding</td><td>0.818590250762421</td><td>0.743335196110104</td><td>0.893845305414739</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>SSC</td><td>Noncoding</td><td>0.842632460418642</td><td>0.759200177207099</td><td>0.926064743630186</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cohort & category & mean & lower & upper & mean\\_diff & mean\\_diff\\_lower & mean\\_diff\\_upper\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & MM  & All       & 1.27395322575364  & 1.18086404770263  & 1.36704240380464  & NA & NA & NA\\\\\n",
       "\t2 & SSC & All       & 1.25450213838112  & 1.15469614261994  & 1.3543081341423   & NA & NA & NA\\\\\n",
       "\t3 & MM  & Coding    & 0.455362974991217 & 0.39951877348784  & 0.511207176494594 & NA & NA & NA\\\\\n",
       "\t4 & SSC & Coding    & 0.411869677962475 & 0.358035280663547 & 0.465704075261403 & NA & NA & NA\\\\\n",
       "\t5 & MM  & Noncoding & 0.818590250762421 & 0.743335196110104 & 0.893845305414739 & NA & NA & NA\\\\\n",
       "\t6 & SSC & Noncoding & 0.842632460418642 & 0.759200177207099 & 0.926064743630186 & NA & NA & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 8\n",
       "\n",
       "| <!--/--> | cohort &lt;chr&gt; | category &lt;chr&gt; | mean &lt;chr&gt; | lower &lt;chr&gt; | upper &lt;chr&gt; | mean_diff &lt;chr&gt; | mean_diff_lower &lt;chr&gt; | mean_diff_upper &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | MM  | All       | 1.27395322575364  | 1.18086404770263  | 1.36704240380464  | NA | NA | NA |\n",
       "| 2 | SSC | All       | 1.25450213838112  | 1.15469614261994  | 1.3543081341423   | NA | NA | NA |\n",
       "| 3 | MM  | Coding    | 0.455362974991217 | 0.39951877348784  | 0.511207176494594 | NA | NA | NA |\n",
       "| 4 | SSC | Coding    | 0.411869677962475 | 0.358035280663547 | 0.465704075261403 | NA | NA | NA |\n",
       "| 5 | MM  | Noncoding | 0.818590250762421 | 0.743335196110104 | 0.893845305414739 | NA | NA | NA |\n",
       "| 6 | SSC | Noncoding | 0.842632460418642 | 0.759200177207099 | 0.926064743630186 | NA | NA | NA |\n",
       "\n"
      ],
      "text/plain": [
       "  cohort category  mean              lower             upper            \n",
       "1 MM     All       1.27395322575364  1.18086404770263  1.36704240380464 \n",
       "2 SSC    All       1.25450213838112  1.15469614261994  1.3543081341423  \n",
       "3 MM     Coding    0.455362974991217 0.39951877348784  0.511207176494594\n",
       "4 SSC    Coding    0.411869677962475 0.358035280663547 0.465704075261403\n",
       "5 MM     Noncoding 0.818590250762421 0.743335196110104 0.893845305414739\n",
       "6 SSC    Noncoding 0.842632460418642 0.759200177207099 0.926064743630186\n",
       "  mean_diff mean_diff_lower mean_diff_upper\n",
       "1 NA        NA              NA             \n",
       "2 NA        NA              NA             \n",
       "3 NA        NA              NA             \n",
       "4 NA        NA              NA             \n",
       "5 NA        NA              NA             \n",
       "6 NA        NA              NA             "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 3 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>category</th><th scope=col>RR</th><th scope=col>pvalue</th><th scope=col>MM</th><th scope=col>SSC</th><th scope=col>Gene</th><th scope=col>NA</th><th scope=col>NA</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>All      </td><td>1.016</td><td>0.398</td><td>MM</td><td>SSC</td><td>All</td><td>0.928155522682364</td><td>Inf</td></tr>\n",
       "\t<tr><td>Coding   </td><td>1.106</td><td>0.148</td><td>MM</td><td>SSC</td><td>All</td><td>0.946411717378104</td><td>Inf</td></tr>\n",
       "\t<tr><td>Noncoding</td><td>0.971</td><td>0.681</td><td>MM</td><td>SSC</td><td>All</td><td>0.869084835019194</td><td>Inf</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 3 × 8\n",
       "\\begin{tabular}{llllllll}\n",
       " category & RR & pvalue & MM & SSC & Gene & NA & NA\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t All       & 1.016 & 0.398 & MM & SSC & All & 0.928155522682364 & Inf\\\\\n",
       "\t Coding    & 1.106 & 0.148 & MM & SSC & All & 0.946411717378104 & Inf\\\\\n",
       "\t Noncoding & 0.971 & 0.681 & MM & SSC & All & 0.869084835019194 & Inf\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 3 × 8\n",
       "\n",
       "| category &lt;chr&gt; | RR &lt;chr&gt; | pvalue &lt;chr&gt; | MM &lt;chr&gt; | SSC &lt;chr&gt; | Gene &lt;chr&gt; | NA &lt;chr&gt; | NA &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|\n",
       "| All       | 1.016 | 0.398 | MM | SSC | All | 0.928155522682364 | Inf |\n",
       "| Coding    | 1.106 | 0.148 | MM | SSC | All | 0.946411717378104 | Inf |\n",
       "| Noncoding | 0.971 | 0.681 | MM | SSC | All | 0.869084835019194 | Inf |\n",
       "\n"
      ],
      "text/plain": [
       "  category  RR    pvalue MM SSC Gene <NA>              <NA>\n",
       "1 All       1.016 0.398  MM SSC All  0.928155522682364 Inf \n",
       "2 Coding    1.106 0.148  MM SSC All  0.946411717378104 Inf \n",
       "3 Noncoding 0.971 0.681  MM SSC All  0.869084835019194 Inf "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mSaving 6.67 x 6.67 in image\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAtAAAAPACAIAAACTlNnkAAAACXBIWXMAABJ0AAASdAHeZh94\nAAAgAElEQVR4nOzdZ0ATSR8G8H8SQuhdioAgxYYgCoqKXeyiqNgbntj19LXX86zYznL2E3sX\nez0bnl1seDYOEQEFUXqvSfb9gCIlgQRZ6vP7ZHZmNgOM4WF3ZpbDMAwBAAAAsIlb3h0AAACA\nqg+BAwAAAFiHwAEAAACsQ+AAAAAA1iFwAAAAAOsQOAAAAIB1CBwAAADAOgQOAAAAYB0CBwAA\nALAOgQMAAABYh8ABAAAArEPgAAAAANYhcAAAAADrEDgAAACAdQgcAAAAwDoEDgAAAGAdAgcA\nAACwDoEDAAAAWIfAAQAAAKxD4AAAAADWIXAAAAAA6xA4AAAAgHUIHAAAAMA6BA4AAABgHQIH\nAAAAsA6BAwAAAFiHwAEAAACsQ+AAAAAA1iFwAAAAAOsQOAAAAIB1CBwAAADAOgQOAAAAYB0C\nBwAAALAOgQMAAABYh8ABAAAArEPgAAAAANYhcAAAAADrEDgAAACAdQgcAAAAwDoEDgAAAGAd\nAgcAAACwDoEDAAAAWIfAAQAAAKxD4AAAAADWIXAAAAAA6xA4AAAAgHUIHAAAAMA6BA4AAABg\nHQIHAAAAsA6BAwAAAFiHwAEAAACsQ+AAAAAA1iFwAAAAAOsQOAAAAIB1CBwAAADAOgQOAAAA\nYB0CBwAAALAOgQMAAABYh8ABAAAArEPgAAAAANYhcAAAAADrEDgAAACAdQgcAAAAwDoEDgAA\nAGAdAgcAAACwDoEDAAAAWIfAAQAAAKxD4AAAAADWIXAAAAAA6xA4AAAAgHUIHAAAAMA6BA4A\nAABgHQIHAAAAsA6BAwAAAFiHwAEAAACsQ+AAAAAA1iFwAAAAAOsQOAAAAIB1CBwAAADAOgQO\nAAAAYB0CBwAAALAOgQMAAABYh8ABAAAArEPgAAAAANYhcAAAAADrEDgAAACAdQgcAAAAwDoE\nDgAAAGAdAgcAAACwDoEDAAAAWIfAAQAAAKxD4AAAAADWIXAAAAAA6xA4AAAAgHUIHAAAAMA6\nBA4AAABgHQIHAAAAsA6BAwAAAFiHwAEAAACsQ+AAAAAA1iFwAAAAAOsQOAAAAIB1CBwAAADA\nOgQOAAAAYB0CBwAAALAOgQMAAABYh8ABAAAArEPgAAAAANYhcAAAAADrEDgAAACAdQgcAAAA\nwDoEDgAAAGAdAgcAAACwDoEDAAAAWIfAAQAAAKxD4AAAAADWIXAAAAAA6xA4AAAAgHUIHAAA\nAMA6BA4AAABgHQIHAAAAsA6BAwAAAFiHwAEAAACsQ+AAAAAA1iFwAAAAAOsQOAAAAIB1CBwA\nAADAOgQOAAAAYB0CBwAAALAOgQMAAABYh8ABAAAArEPgAAAAANYhcAAAAADrEDgAAACAdQgc\nAAAAwDoEDgAAAGAdAgcAAACwDoEDAAAAWIfAAQAAAKxD4AAAAADWIXAAAAAA6xA4AAAAgHUI\nHAAAAMC6qhU4Mg715PzAE6hp1TCu4+jiPnbRrmvvksRSKltMfZBV4ET/jNfjcDjuJwtXVnDe\nHF74jd8uapBTrORxsfS/LAAAgMquagWOHIr1eo4bN27cOM8R7j3a2eunv768a/nYLvUsnaec\n/iCUUD9k+8wt72U6s4KCguiB996AAodF93ftC+AoKFTFbyYAAEBpqIq/I5Wdp+7YsWPHjp27\n9h32uXTvzefI/84t7lwz/tGWAS7j/47LX1nXykor++HyeafiZTizcfcejejlHu8nTN6jWX97\nHwznd+jRSbkUvwgAAICqpCoGjoK4mnV7/X7pppezsihk99Q1T0R5C9V6L13QnB9/cu6Kh9nF\nn8p6lGdbXugB7xt57sEknfH2iVXr5TlIv0Dl9OAr68Z0sreqqaUsUNUxMm/Yym3cWt/oUviK\nAAAAKpnqEDiIiEih7tQlI/WJ3h058jR/ifWUdePM6f2WmTtCiz+N8TDPrkoxR73PpX4/8uXI\nroupugM9+6jlrxlxYGDT7rMOvObb9/KcPmfaKLd2dbivjnpfiyiFLwYAAKCSUSjvDpQdxdYd\nWyvuOPXp8eNIcjLKUyBwXryy74Ehp5fOPz3iSF/Nos+i5T7G/ddLJ7xPxPYfpUtEIfu8fYW1\nfvXsLIg5lrde5ImdF+J5bbc+uzWxJif3aHq8LHduAAAAqppqc4WDiBRNTGoQUXR0obsaeoNW\nzXPkxxybu+ppsfdVlHp4DjXKuuF9IIyImBe7dz9jbEZ5Npf4jeTxFRU4eQ8oa2tjogcAAFRD\n1SlwEMMwRMThcAqVcKynrRtnygRtnLnzU3FnUWg7ZqS1+MHuPQEk+mfX3vdcJ89fGhaqZdRr\ncFuVrBsTbBr3nbx0x8lbr7+kMxJOBgAAUC1Up8CRGR4eQ0Q1atSQUKjUdvGKXhoZt5csvJBU\n3InsPD2bct7s3X3roveRz4qdPIfXklDJcvKFR4fmuVknXt++eEL/DrZGOqYtR266H/PTXwYA\nAEDlU40CR9bdm3eziGo5ORlKLNcbtnpeY4WYQ3NWvRBJrPCD5UjPdvxP+0dMPp2g6jZ6oK7k\nWuq2Q1f6PPgQHx/69O8DXuNbcJ4emNZt8O5iL6EAAABUOdUmcAgDNy4+EE1Ub+gQBylVOPWm\nrfU0FQdsnOUtYTPRfAwGe/ZUjQkPz9Ab6Nlbvei6ChpmDl2Gz91+4+yMupR886xvSkn6DwAA\nUJlVh1UqTNK7SxumjF/2IF3BcuymWQ7SQ5ZSxyUruh8ecXnxktDMos+p3u+Pv1VGxHBqOroo\nSqzw+fH1GMv2dro/vsHpMTEpRHwVlerwPQcAAMinKv7yS7+/afz4k0QkykxJig578+zZ2y/p\nDLdGy6l/HV7XWbvItvrD1szecHWRf7FbnQtqt3KrXUT5uz2D2+9VsG/f1t7KzESHl/TB78KZ\nWxECm9nTeirJ+fUAAABUelUxcGT9d3Hnf0TEVVRR09AyqNWgm+eYbv2HD3Cx1ii8PqUgjs30\ntaN2uHj/7P5c9YcuXyC+eufR06uPL8am83WNzeoPWL56xqT+DbEuFgAAqh9OzlJRAAAAAPZU\nm0mjAAAAUH4QOAAAAIB1CBwAAADAOgQOAAAAYB0CBwAAALAOgQMAAABYh8ABAAAArEPgAAAA\nANYhcAAAAADrEDgAAACAdQgcAAAAwDoEDgAAAGBd1Xla7LVr1yIjI8u7F5WbWCwWiUQKCgoc\nTvHP1YVqSCQSicViPp9f3h2BCkooFBKRgkLV+c1SXgYNGiQQCMq7F6Ws6gyLM2fOdO/eXVkZ\nT38vuaysrMzMTFVVVS4Xl75AgvT0dKFQqK6uXt4dgQoqNTWViFRVVcu7I5Wbt7e3m5sbAkeF\n1rZtWw0NjfLuRSWWnp6empqqra3N4/HKuy9QESUnJ2dmZurp6ZV3R6CCSkhIICItLa3y7kjl\ndvr06fLuAivwhywAAACwDoEDAAAAWIfAAQAAAKxD4AAAAADWIXAAAAAA6xA4AAAAgHUIHAAA\nAMA6BA4AAABgHQIHAAAAsA6BAwAAAFiHwAEAAACsQ+AAAAAA1iFwAAAAAOsQOCqqFwutOLm4\nfGUt4/qtB847+DKpRNXkE3plzdxxA7s0ta6hxOVwOF33pcjYMO2/4/P7NqutoyJQ0TZzdJt1\n+E2+lmmhtw94TenXppGloYayWo3aDVv2m73/eazoZ/pafVW+ESJLqxKPPcivPIcHFftRIBET\n93TXtN6OtXVVFAUahvXaeXhd+5Sdt8J/y+05kjmuC/3JDkNZqFKPp696FB1/WTGwPpE4Pfaj\n/5WjJ1eNuOQb/ujBvIa8klST2b/7568+rW5m07hZw5TLz77I2EocuNW1xWRfsh86eUVz3YRn\nh7f+Mazlg6/3/5nekJ9T4/nGoSP/TLRo0an9kJ5GgpTQh+d81nqcPXz20KMzg01L1tXqrpxG\nyIsSjRBZWpXszCBZOQ2P4j8KJIi/PqF5j51BKo0Hjf2teU1h2O1DO+d3vf70qN+pgSbfqtRo\nN3nZsvxjIvb2lo03vtbv1s28ZF2FssVUFePHj09MTCzvXpQe/wWWRKojL/w4Inq/zllApPHL\n5Wy5q8kmLS0tOjpaKIz5FByTIWYYJnpzWyLqsjdZhsZR+7upEqfejMdp3w5kvlhsxyOlDtsi\nvlf5cP3QjQ+ped/wyW8OikT6466L5O0rlMcISUpKio6OZpj4kowQmVqV7MxQSHkMD4Zh4uPj\n44O2F/tRUNizedZE/GYrXuW+a+q96VYc0h1yOkl6qw9rnTjEbbk+tARdrcgmTJiQkJBQ3r0o\nfbilUnlwLXt0rUeUFBhY5N99MlYripaJha6AI1+bqFMHrqZy2v86q6nytyOKjf43tbNCxq0D\nPuHfjtR2GdqxtkqeRsqOEzyaEkW9fh1V0r7CDxV7hMjWqmRnBhmU1fCIvnCs2I+CQsJ9fYOI\nWniMaZh71V3FedzwhhR7cs95qfd3Xu3y9mMEnceOMCtZV6GM4ZZKJZL5/n04EUdXV1vuakxy\nZPDX1CKbKemZ6QpK3rknfk/EZOXsbJDnmKazc0O68tzvsZBMpIy0qKhoIiUjo6K/JJAN6yNE\ns4TX2KEiYH14mGjxicj/6XP5PwrS09OJSFVVNe9BNTU1oqyHD5/T0HYS2ghv/7UvkDSHjB2g\nW2TPoMJA4KjQmJSo0NBQInFG3Ef/i2vnHIwlbfcRPVXlrpZ6apT1qKtFvpeDV9DdqcYl7Wli\neHgSkYOJSb6jJiYmRC/Cw6OIakpq9eXgb9veUc1x43r+RNSp1sp4hNyaZFBkFahYynh4PJ1r\nRZQU8TmZyFHOjwIzGxtVevnE1zd5RE/1b8eirt94SUSxHz+mEakUapJ+4a9DkWQ4aYyrcqEy\nqJgQOCq0tFOja5/KfcUzbDFm17b1/Qr9fVJ8NeX2C318PIt8Lx1bo5/paVoaESkpKeU7qqys\nnFtWSMqj33qNO5dgPenvNS4FPwFBRmU+QsQ/32coM+XxAZKeLvdHAREp9pw6pY7PqgNjepn8\nsXSwk5Ho491dc2fczFYkykpLkxQ44k78dSqB6swb2w6/xSoN/KgqNKX2C47+6kjijLgPfof/\n3HLvcwxXWcK1gOKr8cxauRd/mzPnsmaJqKioEFFmZma+T4acE+aU5Zfst7x712X+RkMP39jk\nolHSd4UyHiHJycml1XMoA2U8PIiISFlZvo+CbxSdll87KRw5afOKoW1WEBHxTTrO3TL50tj1\nzzU0JHxChO3/61omx2mMp51s3YKKAIGjQuPVau7m1pOIiAaN7N+wRyPPcX0W2b5Y1VRRzmpM\nYnhgZNEL4ZX1LfVLfmlS08REgyg8PJwozx9GERERRIomJvr5KyfcWdi154rnhh7HfHf3M8HE\n5Z9QxiNER/qqRqiAynh4mOnwiTSMa6rL/FGQrxdmvdf+0/O3j69fBcdmq5vZNbZUOzNoHZFl\ngwaKhSq/3e39QKzYaayHRbHfBKhAynuZTKmp+stiGeb9+pZ84rf4I0juasl7uxQ3Ehy8gr4v\nixV+bybH0sQv2zpwidNxe1SeY4l7uvOJnDZ8zFsx+uq0xiokqD/2fKRYpu8ESFYeI+T7sthc\nJVu8KksrLIv9OeUxPBiGiY+PD/yjjWwfBcVJ9OmnTmT66wNhwRLh3anGRBoDfVLkOV8lUlWX\nxeIKR2ViOWHpyLUu3l6/XxxzKHdmlUzVSn0OBxMT6Pc+XsnYzt405yKpQb/hnWb6Xt+87pnH\nagclIqKsV5v+vJotaDe8f+6mXkzk+QkuA3Z+qDvl7I1NXWtg9WNpY3+EyDyHo9AIgXJXNh8g\n+q4DOy26U9xHQeHhkZWZqSjIvZMj/Hhs3IxTyWqd1k1vUXBtVMalvw5GkMGEsb0x+atyQeCo\nVJQ6LpjTav+0o4s2zuuxyEbq7+vC1eSbw5H94qjXxWAiSvMLI6IPF9YsD1ckorp9F/RvkPOu\nmX/PaDH8kuUc//er7HMa6w/f4LW/xdQ1nVp+njyiuW7C88Nb975QdlqzcWzu0pcni9q77wwU\nGXfuqP9054qnP95Yq8XoyR1/Zs4qfMPyCPk+hyP7xdHVco8QWVrJVAdKiuXh8U2NgRu8fIr+\nKJA0PC7/Yjw7smO7RlaGqumf/C6cvPE+q+7o4/vHFnrfhJN/nYwjqzlj2+MOX2VT3pdYSk11\nuKXCMAyTdn20IZFm3xNx8lSTzfdbKgm7JF8/7X00dxfA9IM9iMhyjn++E6S8OTLHzcFMS4mv\npGnaxHX6gZf5Ngn0GShlGwezGU/k6CYwDFM+I+T7LZXkEo0QGVrJVAdkUB7Dg8nZaTQ+vtiP\nAknDI+DA+N7N69bUUuYLNAzqtRm08MgriR/on/5swyNO09Xv5etZpVJVb6lwGIYpJpJUEhMm\nTFi9erXE+cwgo/T09NTUVG1tbR4PGzyBBMnJyZmZmXp6euXdEaigEhISiEhLS6u8O1K5TZw4\n0cvLS1NTs7w7UsqwQgAAAABYh8ABAAAArEPgAAAAANYhcAAAAADrEDgAAACAdVVnHw6xWJya\nmsrlIkKVnEgkIqK0tDQOBzsegARCoZCIUlKK3uQaqi+xWEwYIT9NJBJVmQWkeVWdwMHhcAQC\nQcFnFII8MjMzs7OzBQIBchtIJBaLRSIR/peBNNnZ2VT4WbEgJy6XWyX/6qtSgUNBQUFBoep8\nRWUv58OCx+NhHw6QKOdDEP/LQBqMkFJRJdMGYQ4HAAAAlAEEDgAAAGAdAgcAAACwDoEDAAAA\nWIfAAQAAAKxD4Khu0v47Pr9vs9o6KgIVbTNHt1mH3xS3Yj76n3UTB7k41DGpoaEkUDeobdNq\nwLyDT6NF+WsxcU93TevtWFtXRVGgYVivnYfXtU/ZBc7EJLw8unhwK5va+mpKqrqmdZq6Ttvq\n+ymrNL88+GnyjxAKvbJm7riBXZpa11DicjicrvuKbpH976JGfA6Hw2m18Uv+EoyQCq8Ew4OI\niFLfnVo6skNDE20VgYqOcb0Wvcfte5W3XIYfvZzDDCog1h58X9bGjx+fmJhY3r2o4ET/bemg\nRaRlP3Th+s0bfvNw1OWQRss/XmXlFKelpUVHRwuFwnyNXixxNG/s4j5q4v/mLVo4a8owFyt1\nDvFMRpz5+qNO3LVx1nwizcaDZnlt3LDsf251VYhTs++xT3nOk/5wto2AODpNBs/02rRl06o5\nQxxrcIkM+h/7ykAFUcwISUpKio6OLtTqzEAe8bXM7Nt3dzAkoi57k4t4C+G/Sx34ampqROS8\nITJvCUZIRVfM8GAYJj4+Pj4+vmC7qKuTbJWJ1Ot2HOg5ff6CWb+Odm9bb/zpHxVk+tHLNcwq\ntwkTJiQkJJR3L0ofAkd1ErW/mypx6s14nPbtQOaLxXY8UuqwLYJhGGmBo5DMt8ubcIiarPzw\n/cizedZE/GYrXmV/P5J6b7oVh3SHnE76fiTDZ6CAyHzK7YzcE2U9n12PiNrtKPwrDMpFcSNE\nSuCI/xQckyFmGCZ6c9tifhMI365yFKh13bisS6HAgRFS0RU3PBjJgSNyX09t4jWYeOljFiOZ\nbD96OYZZZVdVAwduqVRgJwdxOJxBR0LO/jbQyUJXRaCsa9lyxOpbX8UlPF/UqQNXUzntf53V\nVPnbEcVG/5vaWSHj1gGfcDnOo2htV0+ZKD4+/tuBcF/fIKIWHmMa5m73o+I8bnhDij2553zS\ntyMZiYmZRPqmpoLcE/Fr1TIk4qmqKhOUREUZIVomFroCWXYqEgetH/37W6cV238xKVyIEVLK\nKsbwYPw3L7sYbzZ+6/rupryMhOiY5KxCe3bL9qOXeZhBRYX94Cq6u7PbPrIft+n4sobqcc8O\nL5wyt/Oj8CvPNruo5xQzyZHBX1OLPIOSnpmJFp+Invg9EZOVs7NBnkJNZ+eGdOW532MhmRQ5\nGNJjP35NFgnT4j69+nvzghNpZNaju833svR0IlJVVc3bQE1NjSjr4cPnNLQdEZFmu87N+H8/\nXu0x13jZ8ObmKukf/Y7+vvQfdcfFC9xVCUqsooyQ4jHvN3sufm639N/J5pwDhYsxQthQ7sMj\n4MKFYFIf1ThzhVuDzRcDEkSkqGvTddyKTb/3Nud/Pwt+9PJgMqICnjx4+OjZm5CIL1HxWQoa\n+mYNHDu6uXeqqyFDIMv64nfq8Nk7/36ISmVU9MxtW7sO7d/aWFB8w1JQ3pdYSk0VvKXiM5CI\nqNbkez8uRKZe9NAnqrvohfjbgeS9XYr7ETt4BTEMwzAJO1yIqP3O2HxvkrKvJxG13hTBFH1L\n5cro3P/4HHWLDpMP/JeWW5Z5arAqkd7IC0k/6n/d102ViMh1f+r3Q9khp//XxiTP36oq9Ubs\neZ3+09+n6qvMR4iUWyq5irrWLQ7Z1k5VwX7JC2FurwrM4cAIKV1lPjwYSbdUTg7iE2nq6fFV\nbQYu3nboyJ51k9oacIlTc5BP1I9acv3oq/stlehzs11dXV37DBk/87cVq7yWzJ041M3V1bXX\nyGXXv4ilNfomO/Tc3EGurr0GT172519/bV05bVhvV9cBM48FZ5bylyARrnBUdPVHjnHm575S\n6TFmaM19G06febO0UUMiIuX2C318PIs8g46tERERpaWlUeHHKikrK+eWFaXJ5GNnemZlxEUE\nPjhx5J4oIyU1m+jb54Niz6lT6visOjCml8kfSwc7GYk+3t01d8bNbEWirLS0NCIVIiJS0DCy\nbmDbWLlHrwEdrNVTQm4fXL/T0yU65e/TUxrhWU8lVlFGSDHC/xo3926t6Y/mNpL+nB6MkNJX\n3sMjIzk5mygxhu/x98O9XdSJiAaP7GXSwmbesdnrZ7p7OeRUw49eDnw9u14TBnVv26imyrdJ\nERmf/l47f9uTxzv33G05r42K9KZfzm/e9yZV3Wnqhnkd9blExAy+/8f01XeO/nmmxYaBtdi+\nYYXAUdFZWlrme21hUZvoWXAwUUMiIp5ZK3cz2c6koqJCRJmZmd8DABF9vx2SU1YUffuebvZE\nRPTLlLHH+zQa1CaEAm5MMCUiIkWn5ddOCkdO2rxiaJsVRER8k45zt0y+NHb9cw0NjZwTZD9b\n3KbNSqXFL/wW2OT8yuk/ys2ite2M6aO3dHs600q2LwIKqSgjpEgR+8bPua435c7vjopS62CE\nsKG8h4dAWZlDxNQaPLaL+vdjXOuxo9vNe3zd1zeEHGoTfvRy0mw5tEBGVDLt6tn72pMD71+/\nDqI2jaS2DL525Z2QLHr/kpM2iIij7TzKvf7dnQFXrwYOGFOP5cSBwFHRZWZmEqkWeP3jWYJM\nYnhgZNHL0ZX1Lc10+ESaJiYaROHh4UTaP0ojIiKIFE1M9OXok1F/jy6jzx7yPhYyYVbtnEM8\ns95r/+n528fXr4Jjs9XN7Bpbqp0ZtI7IskGDnN8wzNXNG95k11/R3+bHH7gcy4H9HWbcv+d7\nJ3GmlaYcHYA8KuIIKSD1wuzpl7iu20YaRrx/T0REaVFpRJQRE/r+fYqaoZWhGkYIS8p7eHBM\nTU2IPhkZGeU9qmNkJPg277w2fvSlQkVVhYj4/KJ+p0e9fPmVqGYTh3w/DF0Hh9oU8OHly89U\nz5jdTiJwVHQBr1+LqU3uaqLM16/fE1lYWHx7nXrGs/6oq0WewcEr6OlcKyJybObI3Xfr/v1o\nsq3xvTDp/v3XRE2aNZVrJGQlJWUSJSQk5D/MU6/VqGWtbyc+eexyOpn26GH7raNfviQTCYXC\nfC2ys7OJKDs7/1GQR4UcIfklh4XFU/z5iY3P5z/+bEUL6xXU7yhzchBGCEvKfXjYt2ihvOHT\nx7AwIvPcg18/fswkMjTMmX+KH70EwcHBOberiMjMzKyYS4zMl5u+b4g0mzWrW0St8IgIIjI2\nKRArDIxr8uhDxOcIIgSOai58/+qDc1uNrMklIhKGbF93LIHqTO7zfYGIHLdgyaDf8E4zfa9v\nXvfMY7WDEhFR1qtNf17NFrQb3t/0e20mLtDvXZKKqZ29ac4Aj373hqlto//jNjCTeH/JH1eF\npNmyZf3cg1mZmYqC3HnOwo/Hxs04lazWad30Ft/+ZlGzsalFV/87svv+rD+cv/03yny288Az\nIksnJ90SfW+AqFxGyKP38UrGuSOkeNrdlvv4xOQ9kvHPiuFbX9Tz2LWsh5ZJcyLCCGFJmQ+P\n2KBnISK9urnDQ6WHR/8aJw4c3nhsUZtB+hwiooxn67ffIa59j245v9/wo5dg5syZSUnfNhXY\ns2ePnZ2d9LqZQcfWHgkUabYYM7hRUb/T01LTiLiqKgVmxXBUVVWIklNSs4ik3/IsDQgcFZ2N\n9ddJTp2eTuhnpxH75PhW73tCywmbZzX6fkVUjluwRPrDN3jtbzF1TaeWnyePaK6b8Pzw1r0v\nlJ3WbBz7I9dm3ZzlPPGy5Rz/96typmz8u75dj3O1XNo3tqxlpCGK+/Tm9vmrbxLIuP+e5T1+\njNvLvxjPjuzYrpGVoWr6J78LJ2+8z6o7+vj+sT/61mLaip4Hh19c36lR4OjhHeqopQbfOux9\nIVBkMmT1r/Y/+U2q1sp+hMxoMfFSnhFClP3i6OqLwUSU5hdGRB8urFkerkhEdfsu6N+AQySw\nbOeefyZBSoo3Eek26unubvj9GEYIG8p+eCzsNO5a3uGh0s1r+1DfAYeHO3S8PbpHXaWv94/s\nOPlG0X7e1inW3xrJ9qMvbphVLRMnTuRyv12ZMjGRsG/Nd1lhl7yWHg3iWrrPn9pGW3q9IjDE\nEBGnDL6HZbEUpkxU1WWxA0/E3F3v0cpKV0VRoF27+dCVNz6LfuakKW+OzHFzMNNS4itpmjZx\nnX7g5Y+1rGlpadHh27oTkeUc/+8HI26unzrYxd6ypo4KX0FRVc+sUceh8/c/icnfi4AD43s3\nr1tTS5kv0DCo12bQwiOvCv8wMkP/Xj26S6NaOsp8Hl9Fx7xxtwkbb0UWs68pSFfmIyQpKSk6\nfFuP/COEYZJ3SV5b2ftotoQ3YBhpy2IxQkpVmQ8PhmHiv+zsXHB4MAyTHX5z3VKlc+4AACAA\nSURBVC8dGhioKfKVtMwcek3f+7zA54MMP3r5h1mlJfNOo+nvzywc4ura/397X8uwSPjZ5kGu\nrq5LbxZYbiy+t9rN1bXPBr8S9VUeCBwVWM7nhU/ZvaGMW5tDRVHmI6S4fTigIinz4cFIe5YK\nyEm2wJH69sisga6uA2cdCkgtri7DMAzz9fQMV1fXcfvf5z8ceWKqq6vr5OPhJeqrPLC1OQAA\nQOWS9GL3/N+O/sezHbFi6dB6sk2n0rezMyD6/Px5vqc0xz17FkKka2dXk5WO5oXAAQAAUHkw\ncY+2zlt2LkS5yZiVi90tpe6Mlhhw8/z58zcCEr8fsOzcrQ6PPpzb6xv97Yk6TMKDfafeMlyL\nrp3rsj+HA5NGAQAAKo3PZ1auvvpJxNGtqRZ8ZtvGfGXmnSa52XxfUxj7+Lj3qS/6/Rq41P+2\nlYlhrykeT+bsfrhp6rSHzrYGnJi3D/3eJyrVGTqlj1mJ8oY4MyVFpKSuoiBTawSOCsz9GMMc\nK+9OQAWGEQJFwPCootKSk0RExMS+veP7tkBZkwbjfwQOCRTMei/ZaHjq0Jnb/96+8oxU9Mxa\n9B87tH9rE9nXw2ZEPDh34sL1W7fuPA74GJ2UKSYOX1Xb0NLeuV37Dl37undpoC3t1gmHYQo9\nKrhymjBhwurVq3M30oYSSE9PT01N1dbW5vGkP+8CqrHk5OTMzEw9Pb3y7ghUUDm7AWppaZV3\nRyq3iRMnenl5aWpWrC1WRTGP961ZtWnP+VexIiLi8FW1dXV1dNR46QmxsXFxSRliIiJls7aD\nJ86cN6WnlXKhM2AOBwAAABTlzrrhja2ae669Em/tPnvj4b+fBMelpsRGhgW9efPfh4joxLTk\nyNf/nN61bHxblZf757ja1HWZeeK/9AInQeAAAACAovw596pynzWX/vsS9vDY6qlDujhaaOW7\nc8NRMbRp28dz4fYrbyPDHx74n23YjmXHggucBHM4AAAAoCheAR+srdVkqiowbD581aXB04M+\nCQqUIHAAAABAUWRNG7kU9K1rFzyGWyoAAADAOgQOAAAAYB1uqQAAAEBRJpqbX5a1bvftodu6\nSSpA4AAAAICiZKSkpOR9LUpLSEgXERHxlFUVs1K//1tLS4WXIZRyEtxSAQAAgKLsickj9OLU\nhqqaTUb+cenVl+SMtJS0jOQvry79MbKxpmrDqZdC97hKOQmucAAAAICMRPcXDfrt88C7r7a0\n+v6UWgU1g4bdp+9r12yybeuBizoEb3CWuFk1rnAAAACAjJ74+ISZ9x2emzZ+UGk1vK952MmT\nT6W0ROAAAAAAGUVHR5O0p7AxDEPR0dFSWiJwAAAAgIzMzc0p7NT+26mFSlJv7zsZRrVrF9rx\n6xsEDgAAAJCR7ajxLQWhO3o7D1tz9vmnuHQxidPjPj0/u2aYc++dYQLncSNtpLTEpFEAAACQ\nldXU44cDuo7YdXhOn8NziHgKPJFQREREqjZjjxz/1UpaQwQOAAAAkBnXpN9fzxwH79+x/9wd\n/6DIRKGCppF14za9PSaMbFer4BPb8kDgAAAAALkIzNqP9Wo/Vq42mMMBAAAArEPgAAAAANbh\nlgoAAAAUZY2b2wNZ67acc3Z2C0kFCBwAAABQlMfnzp2Tta7CMGkFpdQZAAAAqJr++vRpo6x1\nVXSlFCBwAAAAQFF0TEx0fvokmDQKAAAArEPgAAAAANbhlgoAAAAUZY2b2wPSdvXaO7r+wzVu\nq4tcsYJVKgAAAFAij8+dO0cG5jOJKKK4FStYpQIAAAAl8tenTxuJp65PRD2KW7GCVSoAAABQ\nInlWqSiXeMUKJo0CAAAA6xA4AAAAgHUIHAAAACA7UeTtzRN7OVkbaqkI+AoFDTwtpRnmcAAA\nAICsYi+OadZ7b7iYo6hpaGptpqbAyV9eS0NKQwQOAAAAkFHIrt/3hnPq/HL49PrBNpqc4hvk\nQuAAAAAAGb15/ZoEfZZuH2KjKGdLzOEAAAAAGamqqpKmvr68aYMQOAAAAEBmzVxc1KP8/ELk\nb8neLZXMsIdX7r38EBwcHBwaEZ8h5raec3aWswwNww5NmHIiotBhrR6rDoxrUPr9BAAAABmp\nuq/Z4ur066glrU4sbKXPk6Mle4Ej8YnPnuPviaeiZ6KrkRCRIF9rjmGj9g3ybY+qWlerNLsH\nAAAAMtnq4fEkz0sNG72rv7e22uvYzN7aWEsx/8zRppP3TXKUdBL2AoeG06il9noWZoYa/Kcb\n+y3zFcnVWqFer2nTmrLUNQAAAJDZrf37TxU6mBX29GbY00KHU3qWeeBQMrW1Z+3kAAAAUEbW\n+fsvlLWuVm0pBRV1Waw48rHPvqeRSSKBhoF5g2YtHc3UMb8VAACgHJjbl8IVhIoaOESBVw8G\nfn9x+uBus06T501sU1Oe6SkAAABQUVTAwKFq2XHYNOumNrUMtFXFiV8+PL9y+NDlgOvrf1fR\n3TI6/0YjPj4+aWlpOf/OyMjIyMjg8/nl0ecqIjs7m4gyMjK4XFxPAglEIhERpaenl3dHoIIS\ni8WEEfLTxGIxwzDl3Qupvp6YNHB75tiD3kNM8heEHx49zFt1is+f/fQkNqyAgUOvhfuA3Bc1\natl1GVffQnXm7BMhF0/eGWzjopKn6t69e6OionL+bWVllZqait+UPw8fFlC01NTU8u4CVGgY\nIT8pJ9lXVHFnd+28nbx8r0mhEhOn2gnDluw6u6Sfp7aklhUwcBTGt+7U1vxESPC7d8HkYpun\nYM+ePbk/mCVLlmhpaWloSHtqDBQvIyMjPT1dQ0ODx8PNK5AgNTU1KytLW1vihwkAJSUlERE+\nh3+SgoIChyPPQ0rK1NtXr0S6bo6SZoZaOTpqCa+8+o+ohaSWlSJw5Gyl+v2Kfx6Ghoa5/+Z9\nV7Y9q1Jyrg/h2wjS5HwIYniANBghpaICpw0iiouLI11dXYll+vo1KCYmRkrLynEDIu31mw9E\nZGRoVN49AQAAqMZ0dXXpU1BQpoSijPfvw0lLS9ouneUeOBIDbp4/f/5GQOK316lBfo9DE/Pc\nv2KS3l9av+OfFOLVbd8GgQMAAKD8NGnVSiX9/IY//ytwy4GyA/5cfz5dydnZQUpLFm+pRD88\ncNgvjogo9q2QiAIvb9z4hIhIw2HIL631v9WKfXzc+9QX/X4NXOprEhGlvvZZvjdY3dDMxEBP\nR0Wc8DUsOCQ6g+HqNhv/P1fkDQAAgHKk3Gf6JMvTa+c6t3j3+7KJ3RpbG6ulRAT5X9m26Pc9\nz7ItZsx0V5HSksXAkRz80Nc3z0PYol75+hIRkb5Gzx+BoyCNRj3cuzwK+O/D5w+f36WJ+Kra\nNRu1b9qhV++2lmoV+a4WAABANaDYYvnZrQFdJ1/0/rW7d94CnrHr1nMrW0p9cD2LgcNi2Pbz\nw4qvNfKv8yPzvFayaD9iUnv2egUAAAA/QbHh+POvWvps23bsql9AREKWopZJ/eZdBk+c6G6r\nVcSVgUqySgUAAAAqCo623YAFOwYskKdNuU8aBQAAgKoPgQMAAABYh8ABAAAArEPgAAAAANYh\ncAAAAADrEDgAAABARnFBjx49evEpTf6WCBwAAAAgo38WtGzRdvn9EjygD4EDAAAAZKSrq0sK\nKipS9xOVDoEDAAAAZOTYrp1q2uOHL0XFVy0AgQMAAABkpOq+8o9OMRs8pp39IOc8DmxtDgAA\nADJ6umPFwxp2xre39Kl3rG5T+3q1amoJ8j1ApenkfZMcJbVE4AAAAAAZhd7av/9Uzj9jAh/c\nCHxQsEJKTwQOAAAA+Dmd1vn7LyyqglZtKQUIHAAAACAjTXN7+5K1xKRRAAAAkJcwIez104d+\nwYmyNkDgAAAAADnEP9o0qJG+nrlt05Zd1j75djDq2Gj7hk4Lb0tdL4vAAQAAALISBazv7jLt\n+AedNr2b18xzXL9nV8vQx3t87jJSGiJwAAAAgIzSzy9f+ijbafmDl77eI63zlqj16NWeF3nv\nXrCUlggcAAAAIKNnN28mKveZOctWpVCRSr16phQeHi6lJQIHAAAAyCg2NpZqmptLfJYKj8ej\n5ORkKS0ROAAAAEBG2traFB0dLako3t8/lIyMjKS0ROAAAAAAGTk4OyslXTp1I7NgQbzvwtXX\nRVrt2jWS0hKBAwAAAGSk6j5zkkX0nhF9Vlx5GyskEqdHBz25umtGh4ZdtwUr2M6Y3p0vpSV2\nGgUAAABZCZxXntsa1G3Kwu5XiIju99LflVPAN+vvfWqendQLGQgcAAAAIDvFhuPPvm59dueO\nI38/fPMxNkNB07he827DpkzqZ6PJkd4MgQMAAADkwtG06TN7c5/Z8rTBHA4AAABgHa5wAAAA\ngJzSwu5euHjbPygyMZuvaWTduG1P19ZmhTcDy0OOwMEI05IzeGpqAlwVAbFIvKv/OIlFwuxs\nEokVlAQSSwdsWqJtWlNiEQAAVAai0DMzB4/981GMOO/RX/Wa/7rr6Do3c56UZsUFjrSwW8f2\nHb9045/7/sFRqUKGeEpaJvWd2nTo7DZ8RK9GerhCUj1xONSodxeJRe9uPUiKipFWqqSuxma/\nAACAXSm+0zr03xLCNWg+cszg1g3MDFTSvoa9vXt015FHG907iq7/+2d7yZ/z0gNDWvC5tXMX\nbDr9Jl5B17qxU7sB7fR1NZWEyXExkR9e+u6cdfCP+catPRauWTqmub60OANVFYfLbT7SXWJR\nRlJy1PtQaaUAAFCZRe5evC1E0Wn5/ZsLGqv+ODz617lTl3d0XrR18Z757X81lNRSauD4X90G\nO/mtRs4+tHu4m5OxcsFiUULAjeMH9u5a1LrOue0vr3jWKo0vA4rHCJOY7AR5W3F4yhzFGmz0\nBwAAqhP/Z8/EOkN+n5s3bRARkWrjeb8P2dDt6DN/om6SWkoNHC3n35jt2dpI2oZhPK36XcZ5\ndRk3P+D0qXBc4Cg72ZEnsj5skFwmziCugEjCKmiebhvlhtvZ7RkAAFR9AoGA9I2MJP3i5xkZ\n6ZNAIHkGXxGBo/+E1rK8s3r9vh71ZakIpUPR1FPR1FNiUcpde5UmJ7mqVmXcJQAAqDYcW7dW\n2n///ldqZFCw6Ov9+8FKreY5SGmJFScAAAAgI80hq1Y1f77Afe65oJQ8h1OCzs11X/C8xapV\nQzSltCx6lUlWXEhInLJJbUNVHhFR6rsbZ2++ieGbNOvWs4WxtIsmAAAAUDU93bnG39hW/eRq\nt3o7LRwb189ZpRLg//RDAte0lbv/Gk+PH5Wb7ts3KfeF9MCR9HBZrz5Lbn8VkWr9Ubuverd9\nNKL5kMNhQiIiUq4/8ey9rZ112PqCAAAAoOIJvbV//6mcfyZ8eHzrw+MfReJP947tv5e3copM\ngePFGs/f7nHbei5opxtyfqfnNKW+989pemy54OGoGXV32+yF2yZ5DQ9Y2xz7cAAAAFQbndb5\n+y+UtbJW3hdSA4OPz9s6s576ejlwiWa2HGHW+0Dtle92TbImInJqaRrzvOm5S4Frm9uUvNMA\nAABQuWia29uXrKXUSaOfPgmat3TIKVZr1aoRaTdzsv5eyHNo0UwxLCysZO8JAAAA1YzUwKGp\nmRkV9X2Dqa9fv1JydHRGbmlCVFSWtrY2270DAACA8paWJm8LUVpaVoFDUgOHk5PqtdVTDz8P\n/xJ6b/2Mrcn1rG6u/e1BHENEoq/X56+/oWhnV1feDgAAAEBl42npPGHLrYiCEUIicfyr4wv7\nNOy49l2BAqmBY9Bvcxp+OjDMwdSodusZDxosPOU9MnF961rG9W3rGVt02RFoOmHeMCxSAQAA\nqPK6dEg+MqWDeU27XlP+8Ln/LiaTKVRFmBj69OLO+UOczY3tBq19YtStbcEHg0udNKrQZNHd\nZ7a7j9z7wrfoMHxUFwvl7Otn1eeuOfciqqbTkInTl81rg404AAAAqr6Rh/27Td61eL7X/i0z\nL2yZSYo6ljYNzA11dbTVuBkJcbHRn/57Ffg1nSHi6dj2Wbxp5ew+1ioFT1LUslaNBm7/W+6W\n+5Jv3nPpsZ5L2fhSAAAAoOLi6bcYv/3WmBWvL+7zPnrhxj+PXtwLznOZQ0HTvGmPDl37jfQc\n3MZUSfIp5NlHQxTzzj9CuV4jU8lPugcAAIAqjKfTsPf0jb2nE2XGfwr99CU6NlmkrF1Dv6aZ\nmYFqcQ9ylSdwJJ+d2HRhw7tfNrb6me4CAABA5SbQNq2rbSrX2hE8vA0AAABYh8ABpYZhmE/+\nr8Nfvo3+8DH4/lOxUFTePQIAgIoCz0KB0vHy/LVLSzbEhobnvNzea5Sank6n2eNb/jKIw+GU\nb98AAKDcyRM41Pvu9G+hZF18Rahurnptub5uJxFD9CNbpMTEnZm9MuzJv4O3reRwcS0NAKBa\nkzVwZMcG+vm9+hjDb1jbxpgYsUhMXB4Xf7gC0Yszf19ft5OI8qaNXM99LhnUtez4vzFl3CsA\ngCoqM+zhlXsvPwQHBweHRsRniLmt55yd5SxDw7BDE6aciCh0WKvHqgPjGpR+PwuSJXBE3Vg0\nzGPt9YhMIrJZFtDbTvPzxtamR3sHPZljyXb/oIJjxOLLSzcUXcd3g3cLjwEq2ppl0yUAgCot\n8YnPnuPviaeiZ6KrkRCRUHyLvDiGjdo30M17RLWulrTKpar4wPF+k7vrqpA2s/buc0v5s/UO\nIiIy7tWryczLl7/OmWLAdgehYvv4/HXcx89F18lMTQu4fsdhgGvZdAkAoErTcBq11F7PwsxQ\ng/90Y79lvvLNz1eo12vatKal0A1hQth/7z+najdwspTtz8li76y/2rvtbu0ZZy6u9HBpZq7x\n/ZK5RZ06vMDAoJ/pKpQScXpo5rvfU/06Mdlx6f+OyHg7TZTgV2bvHvXugyzVvgYGs90TAIDq\nQcnU1t7KSINffrMa4h9tGtRIX8/ctmnLLmuffDsYdWy0fUOnhbel5p9ir3CEhISoObd35Oc/\nylVXV4mLi/upDkMpyPq4KzN4JYmzc16K0z+K0z9mR57kG/VXqreKuKw/70YsFMpWDUtkAQAq\nAHHkY599TyOTRAINA/MGzVo6mqnLN6lfFLC+u8uMRxzL9r3rBp4LyD2u37OrpeeePT53l7Vt\nJzELFRs4tLS0UoI/xhHlezTsl8DAJCMHI7n6CKUt69PuzKAlEouyI30YcYZyw20SJ3KWIt3a\nprJVq8VqNwAAqpKXL18KBN/+YmzQoIGaWuk9UkQUePVg4PcXpw/uNus0ed7ENjWL25c8V/r5\n5UsfZTstf+q7wOhAu3Nzf5So9ejVnnf63r1gamclqWWxgaNp585afVfNuNhzV8/cfBF3a8mf\ndwy7LbSXtX9Q+sQZnzLfryiigvDrBaF+TwX9Hqx2o7ZTExVtzbT4xCLqcBV4DTq3YbUbAABV\nyeLFi5OSknL+vWfPHjs7u9I4q6plx2HTrJva1DLQVhUnfvnw/MrhQ5cDrq//XUV3y2gbRdlO\n8uzmzUTlPjNn2apQTIEilXr1TOlheDhRyQKHcq+la12ajulV/0GH7mbhwqjQpUMu3z31MMv1\n8CIXfnGNgT3Z4QdJnFV0nayP3mwHDp4iv+P0sRcWrS2iTvMR7po1Mb0YAEBWEydO5H7fvsjE\nxKSUzqrXwn1A7osatey6jKtvoTpz9omQiyfvDLZxKfQ8eYliY2Opprm5xHjC4/EoOTlZSsvi\nV6lwLTzPP9JdOnf5vrM3IkVM4OmbjbpOP/PHEjdjmboGLBElPCq+TtJzEmcRV8bcWkJtxg8L\ne/Lvy/PXJJbWcrBzXTqT1Q4AAFQxnTp10tQsg60E+Nad2pqfCAl+9y6YXGxlaqKtrU3R0dGS\niuL9/UPJyE3adAuZpoooWfdZeerZ55SM5ISkjLSvL856uVmy+zsMisVkFbyYJamSiMmOZbsn\nHC53mPfazrMn8JXyzVHlKvBajBow4dxuvrIS230AAICSUFVVJaLs7GxZGzg4OyslXTp1I7Ng\nQbzvwtXXRVrt2jWS0lKerc25imqaFTdniMXitLQ0Hk/miS+VHMPVkKEWJzWTzxGmynhOoVBI\nROnp6SV4+onz5JH2g3u/u3nP3+dSenxisxH9rDu01DIxyhKLslJl7QBUcDkjJBU/UJBCLBYT\nRshPE4lEDMOUzXulvX7zgYiMDGVeBaLqPnPSb23Wj+hTe/dcEhKJ06ODnjz/59jq3zff+qxg\nu2x6d2nTLYoLHKLEIL+7T14Hf45LzmCU1HVrWjVs1rq5lUbFezIGh8Ph8/mKihU3EpUuoaaD\nMPV10XU4avUFynJclGMYJjs7m8/nc0v06BNFI32nYX0zE5Ojg8OcPQeX4AxQwQmFQpFIVH3+\nl4G8srKyiAgj5CdxudzSeOZlYsDN20GpKtZtXeprEhGlBvm94ddzMNf8/nc5k/T+8p87/kkh\nXt32bWRfdipwXnlua1C3KQu7XyEiut9Lf1dOAd+sv/epeXZSf30UETgy3h6eNW7WX/ciC85M\nFBi3Gbdu55pB9Vjf5EEeOYGDz68uM1l5pkOFn/cXXUdgMlyub0jO368KCgo/c6GIx+Nxudzq\n84OoVnKSKH64IE3Or0mMkJ9UbNqIfnjgsF8cEVHsWyERBV7euPEJEZGGw5BfWut/qxX7+Lj3\nqS/6/Rp8DxyvfZbvDVY3NDMx0NNRESd8DQsOic5guLrNxv/PVa5tLhQbjj/7uvXZnTuO/P3w\nzcfYDAVN43rNuw2bMqmfjWYRXZcaOL4cG9Z22Fme09DfFvVwtrM00lFTpKyUuMjgV/cv7du5\ndXCbL9xXxwdg6UG54ao1UKw1LuvjTmkVeFpO/Jq4zAAAUNUkBz/09c3zELaoV76+RESkr9Hz\nR+AoSKNRD/cujwL++/D5w+d3aSK+qnbNRu2bdujVu62lmtzXUziaNn1mb+4zW542UgOH96pT\nvP5H/z0+yCBfP2wdnDsPGDex2wC7wV7eKwcswNPbypHAaj4jTs8OP0BU4MnwxNNyUrbzJo48\nc3QAAKAysBi2/fyw4muN/Ov8yDyvlSzaj5jU/qff/OPdQ3fEzkPa1pZw6yTp9eXzL9TbDGst\ncatHqfda/vtP0GX4QAOJqYdjMHBEV0FAQICkQig7HJ5S3ZUqjY8p6HYgnlLOEZ5GY6X661Sa\nnODwtcu7fwAAUMU83jR8+Ib7kneB+nx2/vDhmx5LaSn1L2AtrcznQR+JzCSWfgwKytR2wC+0\nioCn00pZpxURpdxtpNL4BFetbnn3CAAAqiORSEQcqYsOpF7h6NfP5OFvrmN23fuUlm9xDpP2\n6d6uMa6/PTR1d3cq1Y7CT+MQp7qsCgYAgAom7e3bj6SjoyOlWOoVjvYrTyx53WvJ2NbekzRN\nrSyMddT4lJ0SFxHy/lNCNrdGu+UXVrTFBAEAAIBqwG/joA2PiIjC/YiYrSMGXcx/wUKUkfDR\n/87jJOVezk2knEJ6ZtBs8ds/79zO7N1/xtfv9fvPH6OySFFN18x50ACXvh6jejcsau0LAAAA\nVB2f7h0/fir31SOf4xKeriEwaj52w4Yh0nalLPIiBUfbru/0P/pO/5k+AgAAQCXXbUtQ0Coi\nor+nWE8RrnuzvXe+7d04PEU1XSMDjaK2YMFdEQAAACiaqqFVzjPnO43z8hJ3rWtlJfeUwZIG\njuy0hFRS1VLBfnIAAADVRl23uXNL1rKkgePcL9r9hT7MSfcStgcAAIDKShj37uG9J4ERcamZ\novwPmrPuOa2HlaQmuKUCAAAAsou/5zVi2NKLYRkSS/uZyBs4/tm3L7SIt3saQmQqVwcBAACg\nsos5NsZ1/kVq6rHMLfPggtM1Z2zpox380vfoId9I0/6LF/Rq3UxKQ6mBY8uoUaeklX2DwAEA\nAFCtfDm89XSCwS/Xbu/ulLjl7wXnLV08f+1KtGDxvMOj24/0fjp+ioeUllIDh4Y66Xfdcnyi\njeTi20vb//7T3QYAAIDK5OW//zJaA/p3UiZK4nBILBbnHFeyHLpt2TEDzxWHFneYUENSS6mB\no3Fj2pfAt2/XTkticcwOImEp9BwAAAAqjaysLKqhr09EpKioSBmJiVlEOXtyaNrb1xZufvYv\nkYukllIDh4ODKbP/mT+R5GfZKqpqagoVJRZBNcAwTMRLyU8LToz8mhqXEP7vW4mlBnUt+UoC\nNrsGAADsMTQ0pJiYGCIiLRMTNebhq9c0OGc386zQ0MgiWkoNHM0W3H71C9eEIZK4hXmv3QkJ\nP9NjqNwYMXPwlxkSi0TZQoYRSysd47NDz0LyI4gBAKDCq9eokWL827dfqLMhp2WHdoIDO6ev\n7LhngpNW5LVFC08k8pzspEzFkB44FHRrN9RlqbtQ6XF53HnPrpR3LwAAoIypdevRljv++PHw\naVNNNAcvnru+9ZIFLpYLcgp5FlOXehpJaYl9OAAAAEBWemOvicZ++7ei4++37pmv2XDiQWi6\ntoWj65R5o5uqSGuIwAEAAAAlpOPgseqQR+7L7MjPiUY19STV5Eo6CAAAACAHcdyLg3N61LWc\n+o+UCrjCAQAAALIQJ4U8vv8sJIlfo26z1vZG35ccpr0/v2Hh/LUn3iQypDJCVUpjBA4AAAAo\nDvP57JSeHtv8E3Oe1aZo7PrHFZ/JDRN8lw0fseJ6RBYp1XKZunDZvFHNpZyg2MAhFAoVFCTV\nYuLiEnR0tH+i81AC2ZEnsj/tk1wmzs54NYG4/MIlPC1HQZ2lrHYMAACqsC/7xgzb6p/K07Jo\n3MiUPr7wD7nw69AVqj1Ojl8VINJvOf43r4Vj2hgXtT9XsYHj/JT2T9yOLe9izMt7NO7R+pED\nn44MO4LH05cxnnYLrkDamiOpOHwdNjoDAADVQ5TPviup3AZTb97Z0FaXQ0zU9UmtumxfPvo1\nGfbYdPHglCbaEjftyqvYwFGDH7y2W+P78w8eXZITOpj4h+s9Bs47n2S3clmpfBUgD66SKSnh\nsXkAAFCWXr96xVC7yb+11eUQEXH0Oy39tc32Kbd1ftnt82sTZVlOUewqGamZaQAAIABJREFU\nldYbX9xYbBfk1a1xx4XXPkc/Wte7cZuZ/xiM83n+YJ79z34BAAAAUPElJiaSZu3aea6W61la\nahA1bdNGprRBskwa5eq3W3ztRaulQ4Ys61p7LS9LudGUk77r+lngQSoAAADVg0gkpgIzOvl8\nPpGqqrRFKYXItg8HV0ldW1OZx2RlCQUGdZvUM0DaAAAAANnJEDgSn/3Zr3HraTd0PY+/eOLt\nzj0zqlnTUfvepLHfOQAAAKgg4va6aeXhtjeO6PxIrQJGnpfSvNhbKo/mNGm3Jrr+hONXNwyw\nFhAdeurUdnz/qaOaPbm999LeAXjuJwAAQHXAZKUmZhU4lp2WmJj/SGrBKt8VGzgiojVGHv97\n0wBrpW8HVBuOOfi0edsJ/X898QSBAwAAoOrrezQ9XSRTTZ60SRfFBo4e2x72U1IqcFDV1vPA\nU6dXMTK9NwAAAFRqXL6SkoRtJeVRbOBQKpQ2vlGztVX7ufcGAACAakLqpNHMTBnPwGRmZpdS\nZwAAAKBqkho4Rtfp+D/ve5+Lih3ihLfnVw5p7LI6kIWOAQAAQNUh9ZbKpJlmk+e03TyrTqf+\nA3q0b+nUpH4tfV1NJVFyXExk8MvHj+5eP3P8/KMYw67TNgyxKsseAwAAQKUjNXC0mLLnybBp\np7Zv+nPn2im70gsW87Rte47acHOKRwdzKXM8AAAAAL4patIoV9uu//zd/edt+fL64Z2H/wZ/\njopP52noGdaq79iqtaOFZvHbogMAAACQLM9SIY6yoW2HAbYd2O8MAAAAVGQf7x66I3Ye0ra2\nhDmgSa8vn3+h3mZY61qSWsr2LBUAAAAAerxp+PAN9yXvJvr57Pzhwzc9ltISgQMAAABKgUgk\nIg5XWrBA4AAAAICfl/b27UfS0dGRUoyJnwAAAFA0v42DNjwiIgr3I2K2jhh0Mf8FC1FGwkf/\nO4+TlHs5N5FyCgQOAAAAKNqne8ePn8p99cjn+KPCdQRGzcdu2DBEQ8opZA0c2bGBfn6vPsbw\nG/bubafJiEVi4vK4HLm7DAAAAJVNty1BQauIiP6eYj1FuO7N9t75HgrL4Smq6RoZaBT1fDdZ\nAkfUjUXDPNZej8gkIptlAb3tND9vbG16tHfQkzmWP9F7AAAAqBRUDa1ydhXvNM7LS9y1rpUV\nT95TFD9p9P0md9dVATbT91732+z6LdAY9+rV5Nnly1/lfTcAAACoxOq6zZ3b10butEEyBI5X\ne7fdrT3jzMWVHi7NzDW+30OxqFOHFxgYVII3BAAAgOqn2MAREhKi5tzescBtGa66ukpcXBxb\nvQIAAICKSRR5e/PEXk7WhloqAr5CQQNPS2lW7BwOLS2tlOCPcUT5FtZ+CQxMMnIwKpWeAwAA\nQCURe3FMs957w8UcRU1DU2szNYUC60dqlXiVStPOnbX6rppxseeunrn5Iu7Wkj/vGHZbaP9T\nfQYAAIDKJWTX73vDOXV+OXx6/WAbTXkWqxYbOJR7LV3r0nRMr/oPOnQ3CxdGhS4dcvnuqYdZ\nrocXuRS1/AUAAACqmjevX5Ogz9LtQ2wUi6+cT/GrVLgWnucfnZ7bUe3t2RuBouj7p2+mOUw/\n88BniHHJ+goAAACVlKqqKmnq68ubNkjGjb+UrPusPNVnpTgrJTmTr64uwANYAAAAqqNmLi7q\nf/n5hVD72nK2LDY7pH55H54gzKmrqKb5PW1kxoa9/5Iqd0cBAACg8lJ1X7PFNXzVqCX3okTy\ntSz2CseVydarmoc8nWme//CrtW2bvl/HnHSX7+0AAACg8nq6bbGvho3e1d9bW+11bGZvbayl\nmH/maNPJ+yY5SmpZ0oe3CYVCLq8kO40BAABAZRV6a//+nKe4ZYU9vRn2tFCFlJ7yBg5xVnpa\nlogoQ0TirLSUlJQfRaL0aP9zd74YNDf42X4DAABAJdJpnb//wqIqaEmb2yE1cLxd2dR2yZuc\nf5+1UV9QsJxff+6fzeToIgAAAFR2mub2JdyES2rgMOo8c61aDNGrA7N8zeZPbaudW8LhCTSM\nbTt0b2dZ5P2YzLCHV+69/BAcHBwcGhGfIea2nnN2lrOM3cr64nfq8Nk7/36ISmVU9MxtW7sO\n7d/aWCBjawAAAKhYpGYG3ZYeM1sS0Sv9L03MZuYNHDJKfOKz5/h74qnomehqJEQkyN5SGHZ+\n8VzvN2lqZk1bdTXgxbx9+ODE2mf/fvZaMdCiBCt/AQAAoFQJ4949vPckMCIuNVPE5Cux7jmt\nh5WkJsVOGrUdsc62RJ3RcBq11F7PwsxQg/90Y79lvjIvn/lyfvO+N6nqTlM3zOuozyUiZvD9\nP6avvnP0zzMtNgysJc8+qgAAAFC64u95jRi29GJYhsTSfiYlDRxERKLoRwf/Ov048EN4XIb4\nx+FW8y7OlX6PRMnUtkS3eYKvXXknJIvev+SkDSLiaDuPcq9/d2fA1auBA8bUQ+IAAAAoJzHH\n/s/efQZEcfRhAP9fBY4uSFeQoiiKREUExYpGYsWSaDTWaHxjjMaoUcGuUWMsUWNi7C12FGPv\nRgW7QaVIs9GkisBxXH0/YAG84w5hOcrz+wQ7O3d/dIGH2dmZcX1mHyfPUYv6F+4KDLb5cX2A\nafyDi3t3X0xpMHheYF9fVfM71QeOgts/d+k+56HAkp+aLWjSiJP+NCmzQG5k7+HiVli5XwQR\nEaU9ePCSyKZVya1ozVq3bkRRCQ8eJJMrllQHAADQjtQ9vwe/shxz9sqW7jnrTwcec/L7+vue\nRIHzZu0Z22Xk5jsTJo1S0VPtSqOZu4PmR3r9HvXir158p/8diXyekZ1wfkXveuS5aFbXyv0q\niIgoMSmJiGztSsUKS1sbDlFSchIDbwkAAAAaeRAerjDpPbi7HhGLxSK5/M2ND12nYRsW+Sdv\nXLI7XUVPtSMcd8PCFP5bxjbknGWxFAoFEenYd5t2YEuCy+zNL/ynNai8L6KIMF9IxNYX6JY8\nzNLXFxDl5uWLid5PHD148KBQKCz6WCQSiUQiHg9b2H48iURCRCKRiM3GfjmghEwmI6KCggJt\nFwLVVNEvH1whFSSXy4t+3VZLYrGY6ltYEBHx+XwS5eS8+71s7OHRSLrubjiRn7KeagOHRCIx\nNDPjEeno6GRlZb05qufj4/Hw3COiSg8cqihIQUSskhM4tm3blpaWVvSxs7Nzfn4+flNWHH5Y\nQNny87GNEpQFV0gFFSX76srKyooyMjKIiEzs7AwUYQ8f0dBWREQkfvo0pYyeagOHjY1NzpMM\nKZGzs3PqvvMRsvZuHCLJ/fsR+vr6lVN9CQJ9AVF+vlBEVHyQQyHMLyDi6gtKPBe7du3aoj/K\niWj16tXGxsZGRkYMFFVXFBYWFhQUGBkZIbeBUkKhUCwWm5iYaLsQqKZyc3OJyNDQUNuF1Gxc\nLpfFqrbPR7i2bMnPjoxMpR5WLJ+unXV2bpz6c7et//MySTk7J+hADsfL3U1FT/WPxXbsaLzv\nxi363GfQl95zp/t1jPvCxyzt8p79hX2Pt67kL4OIyM7WligmKTGJyKnY4ZfJyTIie5uSUzuc\nnd8/esPhcLhcLpf7sbvDwNtbKhwOh4N9ckCZoh+C+C4DVXCFVIpqnDaIyMC/Vyf2hP37E6dM\ntjMeOm/mKt8FgX5Ob5Yj5zhOXvi1tYqeai8LbregIysTLQqJnCfv3Z84bsb2P9eTlUfv1cdW\n+jMxwmHh7m5JMcn37qWOcLJ6dzTr7t0nRGbu7jYMvCUAAABoxnz8Wdn4Nx/z28y/dM3hl9UH\nQp8WmDq26TNp1lhPgaqO6nOorlOnQUVjDRz7gFVnA1ZVSsHv5ERduBKbL3Dp5NfUmIiInHr4\nNz6yPSZk28XOP3WtzyYixavQ7YcjFWzHnj2aVOfYBwAAUMfUaz1q2e5RmpypLnDkB49ttc71\n0KXp5V9uND1s556bWUREmZFSInp8cs2a20RERq2/HONr8easzFv7Nx9OtRjY7G3gIKu+k0bd\n/mlL2G+Tp4S1b2HJyogMuxmXo9t42KQAe+QNAACAmkhd4NA3UKTksT9qImZufNjFi8XWzUh7\nePEiERFZGPV+HziU1WTfb8Eaq8O7j1wJv3LqLgnM7b0Hjx822NcOG6kAAADUTGpvqbT29o4/\ncTXzR3uz8r604/A/jg1Xf9bIv46N/OAo38pr6DSvoeV9SwAAAKhsv/TvH0qmfZZuG9s07Jf+\ny0PLOtfnp6MzvJU1qA0c9b5aOmv7572nSJZO6NHS3tKw2LpabA6XjXscAAAAtdutkJAQsnSY\nRkRJt0JCQso6l6tqqEFt4Dg8vPWMUKLQMV1+K9008KDi0CDNagUAAIAa6q8XL9YQx9CCiHr9\n9eLFmrLOFai6IaI2cDQfumiRil1fmzZX1xkAAABqunp2dvXefFiQmZiYqWvr7tFA5fOvKqgN\nHK4Dg4LKXxsAAADUPpcDfQaGDNj7+tCQ8vbECtYAAACgITMzM+IKBB/x2CgCBwAAAGioTefO\n+sJbYQ/Kv8EcAgcAAABoSH/Qzyu7Z6weNeVogrB8PbHFDgAAAGjozp9Lwuq7215ZH+C6r4mn\nh2tDGxOdEitkeH63fWIbZT0ROAAAAEBDTy/t2HG46MOMx6HnH3+wCFhe748NHPmpcdm6DnYm\npU4szHz2QmLubMXEhrEAAABQLXX/9f79Mh9eNWmkokFt4Dj1ncuydk/uTHMoefjhik6ecb9i\n4S8AAIA6xNjBQ8XiXOp87KRRqVTK5nA+sjMAAADULSpHOOTiAqFYRiSSkVwszMvLe98kK0i/\nH/JvqmU7y6qoEAAAKo008wrJcsvbi230CVvXlol6oIaSZsWEXbv9OCkrv1CmKNHi0ntKL2dl\nXVQGjsifPVssiCj6+KibYWDpdl7TmWvbVqhaAACoatK043JR4ofHFdLXClEK26CJ0l46vHqE\nwAFvZF9bOmL4wuPPREpbB9qVN3BY95i2wiCD6OHO6RftZ0/uZPquhcXRMbJt0fWzzk54xAUA\noGbRbbpC6XFp5hVxwi+CT/ZWcT1Q42TsG9dn9nHyHLWof+GuwGCbH9cHmMY/uLh398WUBoPn\nBfb1VTUYoTIzmPmMmuZDRA8tUlvZTyseOAAAAKBuSt3ze/AryzFnr2zpnrP+dOAxJ7+vv+9J\nFDhv1p6xXUZuvjNh0igVPdVOGm0x4lekDQAAACCiB+HhCpPeg7vrEbFYLJLL5UXHdZ2GbVjk\nn7xxye50FT01visizU9LSs4uKD43xMjW1cawAlUDAABAjSIWi6m+hQUREZ/PJ1FOjpioaCs3\nYw+PRtJ1d8OJ/JT11CBwvLr526SJi/bdzZSWahh4EOtwAAAA1CFWVlaUkZFBRGRiZ2egCHv4\niIa2IiIi8dOnKWX0VBs48o5P6T/1tN1Xs9Z0cLEw4BVfML1BuwoVDQAAADWLa8uW/OzIyFTq\nYcXy6dpZZ+fGqT932/o/L5OUs3OCDuRwvNzdVPRUGzjunzrFGrf33z/99Cq5ZgAAAKhhDPx7\ndWJP2L8/ccpkO+Oh82au8l0Q6Of0Zu0MjuPkhV9bq+ipfi+V/HwnzxZIGwAAUGmEWa+yXiSX\ntxdPT9eysSMT9YA6sZeOCN37tDTjkvn4s7Lxb47y28y/dM3hl9UHQp8WmDq26TNp1lhPgaqX\nUBs4PH194y5cz5k4wLjSygYAgLot7tqtEwtWK23Kz36la6DP4fE+bLJydRm9Zy3DpYFS4b8P\nGHzcuu2A0ePGfT20S6P3O7fWaz1q2e5RmryE2sBRb/jymbu+6B8kWzGuq5utcfErgM3hslkq\nOwIAACjn3reHe98eSptW+g7ov2y2U3ulO5yDtjj6dHc5e/7W3p9v7V061cXvy3Hjx43q17q+\nklioktp1OA4PaznlSvTlJZ97OpgLeCV8frgCtQMAAEAN0Wrq2ZiUhEubg4a1s5HEnts4Y3Ab\nuwatP5+58XxcrkJ9dyINRjiaD120SMVOtE2bl6dYAAAAqLn0HTqPXdR57Px1Ead2b9q0adfJ\neweXTzj4y4+OXYZ8PX786IC2VvyyuqsNHK4Dg4Iqr1oAAACoyTimbr0nrek9aXnKreBtmzZt\n2Xf54pbZF7fMNW/Zd+S4ceOG92hirPTuidpbKgAAAACl6Vi3HTp708W4lNjzm2YO8aqfGx68\n8jt/16+PqDhfo6XNZek3dv0VfOtxQmKWSP7+cIdZx2e2r4SaAQAAoGZiGTj59g7IzHyZEL/9\nVoaCFKqmdKgPHAW3f+7Sfc5DgSU/NVvQpBEn/WlSZoHcyN7Dxa2wcosGAACAGkOeHXV69+bN\nm3f+8yBDSkT6jTp9MXacl4qz1QaOzN1B8yO9fo86aTdbf6brkUdBroXPLqz77uv11otmda3c\nygEAoMrJC55JX4ZIMy/JhQmiqOlcs87c+p8SS+OtPaGqFT4LO3XtQUJ8fHz806RskZzt+9PR\n6ZrebxCn3jy85+i/4Qlp+QqBuUML3z7DBvva6pSzBOGzKwe2btq87fD1FyIi4tVvNXDkuHHj\nh/VwMVS9WobaS+puWJjCf8vYhpyzLJZCoSAiHftu0w5sSXCZvfmF/7QG5SwSAACqC4WsMH65\n+MUmkkuKDkiS90qS97L1nfXcfmcbqtoTA7Qr5/bBrfvjiCMwtzMzepX0SvOe0mfH5s3cHCE0\nsPfs0NOSkxEZFnpgxd3w5KVLvnAs8wGTtyQv7x3bsWnT5r/Pxb6WE7EMnbqN+Hr8+NH9W1uq\n7682cEgkEkMzMx6Rjo5OVlbWm6N6Pj4eD889IkLgAAComRSiyB8kqcEfNsjz44R3BwhaByNz\nVEtGXqMXepg72lsZ8e6sGbjookzTjqnH1m2PyDf0mrx6VjcLNhEphl5fOXX5v3vXHvFe/UXD\nMlfyfHltw8rVm3f8cz9NQkR8S8/PR40bP25oVycDjRcAVfuUio2NTU5GhpTI2dk59dL5iKIv\nTHL/foS+vr6avgAAUE1JUo8oTRtFFLL8goiJpJBWZUmgGd0GLTycrY145V3qO/7sqRgpOfYb\nU5Q2iIhl2n70oKYsecKZM4/VrN51dc3EFcH3MwRNevxvxaH/EhNv7V82rls50gZpMMLRomNH\n4303btHnPoO+9J473a9j3Bc+ZmmX9+wv7Hu8dTneCAAAqhHx0/VlnyDPj5OmneRa9q2aeoBp\naQ8evCSyadW6xG6uZq1bN6KohAcPksnVtozeZq2+DOw17usvOjuo3JxNHbWBg9st6MjKRItC\nIufJe/cnjpux/c/1ZOXRe/Wxlf4Y4QAAqIkUomR5foza06SZlxE4tOLBgwc6Om9mcjZr1szA\nwKDir5mYlEREtnalYoWlrQ2HEpKSk4jKChxdZu/pUsEC1M9D1nXqNMiJiIg49gGrzgasquA7\nAgCAlskLNdoaXi5KYroSUGrevHmvX78u+njr1q3u7u4Vf01hvpCIrS/QLXmYpa8vIMrNyxcT\naTRx9KPhwScAgDqHxdFoXFzD0ypF1rOkq3/tib0clhabsHP0Dw5tP/Ec2s/ts64sVl3clXz6\n9Ok83pudWO3t7Rl9LwUpiKgK/plVBg7hq1dijq6RIUf0Kl+s/BS+vomgPDvTAgBAtcAWOLI4\negpZgZrTDFtUTT03dhw6OmuptPDNb5v8zFcRpy5FnLrUpGv74ZtX6BkbVk0Z1YePj4+xsXHl\nvqZAX0CUny8UERUf5FAI8wuIuPoCZoc3qIynVEaYmpp6L4uhkBGmqowIYbo6AABgAluXa9FL\nzTksLs+qXxXUcmdfyKGpC96ljeIeX7y+ddh3MgkelqkEdra2RJSUWOo22cvkZBmRrU1ZEzgq\nh8oRjq/Xrets1taKzL5et66z8lOcP2GoKAAAYBjfcZo047xConLZKH6DMWyBE9Nl5GVkH525\nrIwTnoTdC926z/eb4UxXUutZuLtbUkzyvXupI5ys3h3Nunv3CZGZu7sN4wWoDBw9v/uu1AcA\nAFBrsHXt9Ny3FDwYqzRzcC376DjProIy7u4/JsrNK/uca3/tQeAov5yoC1di8wUunfyaFt2c\ncerh3/jI9piQbRc7/9S1PpuIFK9Ctx+OVLAde/ZoosU5HG/kB49ttc710KXpVXQjDwAAqgjH\nxEvQ9rQ4frkk7RTJRUUH2QJHvsN3POvBRFUxWzM+9I7aczKfJr5KSjWxtVJ7Zh2RHrZzz80s\nIqLMSCkRPT65Zs1tIiKj1l+O8bV4c1bmrf2bD6daDGz2NnCQVd9Jo27/tCXst8lTwtq3sGRl\nRIbdjMvRbTxsUoB9Ffxnqwsc+gaKlDy2EfOFAABAlWPr2um6rdNxFUqSD0gSt+t57GTrNazK\nAvIzstSfRJSXkYXA8U5ufNjFi8WmYqQ9vHiRiIgsjHq/DxxKcO37LVhjdXj3kSvhV07dJYG5\nvffg8cMG+9qpmzD6S//+oZpW5/PT0RneSt9dXc/W3t7xJ65m/mhvpul7AQBAjcLiCNiCRiyu\nfhWnDSISmGr0LIbA1ITpSmoQx+F/HFN/i8lx5F/HRn5wlG/lNXSa19ByvuOtkBCNHxPhqipN\nbeCo99XSWds/7z1FsnRCj5b2lobFnoNlc7jsuvh8NAAAVBIHr0+izl0t+xwTWyvTBtZlnwOM\n+uvFizWanitQNT6hNnAcHt56RihR6Jguv5VuGnhQcWiQphUAAACU1mZIv/O/bpSICss4p92o\nwXVz+a/qo56dXb0Kv4jawNF86KJFHsqbmjav8NsDAEBdZmxt8dncKSGzl6s6waaFa6dvP7wz\nADWP2sDhOjAoqCoKAQCAOsn3m+GF+cLTS9Z92NSwVYtRu3/j6epUfVVQNmlWTNi124+TsvIL\nZSV3tnfpPaWXs7Iu2EsFAAC0zG/q+KZ+vpd/3x5zKSw/M5vN4zZo2czzy4C2wwLYXI62q4NS\nsq8tHTF84fFnIqWtA+0qEjhk6Td2/RV863FCYpZI/v5wh1nHZ7b/iFIBAABKsnVvOmzjciJa\n2SGg//JAp/ZttF0RKJexb1yf2cfJc9Si/oW7AoNtflwfYBr/4OLe3RdTGgyeF9jXt62KjuoD\nR8Htn7t0n/NQYMlPzRY0acRJf5qUWSA3svdwcStrkg8AAMBHwPzQai11z+/BryzHnL2ypXvO\n+tOBx5z8vv6+J1HgvFl7xnYZufnOhEmjVPRUuXnbW5m7g+ZHev0e9eKvXnyn/x2JfJ6RnXB+\nRe965LloVtfK/SoAAACgWnsQHq4w6T24ux4Ri8UiufzNjQ9dp2EbFvknb1yyO11FT7WB425Y\nmML/67ENOSwWS6FQEJGOfbdpB7Z8dnLB5heV9xUAAABAtScWi6m+hQUREZ/PJ1FOzrttfo09\nPBpJ794NV9FTbeCQSCSGZmY8Ih0dnayst0vQ6vn4eDx8+KjilQMAAECNYWVlRRkZGUREJnZ2\nBopiWUD89GlKGT3VBg4bG5ucjAwpkbOzc+ql8xEyIiKS3L8foa+vX9G6AQAAoAZxbdmSnx0Z\nmUpELJ+unXViN079+UJCdl5WZPAPQQdyOO7ubip6qg0cLTp2NL554xaRw6AvvSMX+nX8asr0\nKV/69F1T2KdP68r9KgAAAKBaM/Dv1Ykdun9/IhEZD5030114JdDPqZ6hmdvADY9kjt8t/FrV\nKvQqn1J5+VJsackn4nYLOrIy0aKQyHny3v2J42Zs/3M9WXn0Xn1spT9GOAAAAOoU8/FnZePf\nfMxvM//SNYdfVh8IfVpg6timz6RZYz0FqjqqDBwT7Wzjeg8fO3bsMP9Og5yIiIhjH7DqbMCq\nyq0cAAAAaqp6rUct2z1KkzNVBo5eXzZbcOi374+umW7dtv/IMWPHDO3mYqT2BgwAAFRnspy7\nJBN+eFye+4ikubIs5Ru3sg1cWfz6DJcGtZzKwDF6x5VR6+PP79u6bdvOI8sm7F821d530Kix\nY0YP7mgvwKosAAA1kvjJKrnwiZIGhYwUElH0T0p76TRewDXvzmxlUEO8PDDxiz8Kx+/a/KVd\nyYbEPWOHb9afdHDtQHOlHctaaZRl6NR93JLu4xa9ij7799atW3ftXzBq58JJzn5DxowZOzLA\nywbb6QAA1Cx6Hnu0XQLUaFlHN228krt4m90HLXZejV4NX7Dp6IKBX5sq66nJXipsE9ee3/7S\n89ufsx6e2rN169bd22cP3TSnntvMY48WV5+9VORyuVAo5HCwzc/Hk0qlRFRQUMDC0sKgTNEV\nkp+fr+1CoJoqWnWygleIXC4XiUR1+TKTyWRF62xWS5EPH8rM+rdppKTJuU0bE+mph9FE3sp6\nlme3WG69Fn0mLWvt1arR1ClrrmdFRJe1wEeVY7FYPB6Pz+dru5AaTKFQSCQSHo/HZmO6Digh\nlUplMhm+y0AVsVhMRetPVgCLxeJyuXX5MmOz2dX4r76srCwyMzNT2mZhUf/tomBKaBw4xGl3\n/9m1bdu2v09HZMu45i37/TDm++ozvEFvAwePx9N2ITVY0d+vXC4XA0WgVFESxXcZqFL0a1KT\nK+TxxeuX129X2pSXkX1myVodAyULL9R3sh+wIqhiNdYA1ThtEJGZmRm9iI0tpE8+mFYhiotL\nJJPuJip6qg0c0syHp3Zv3bp194kHGRK2cZMe45aPHTuybxvLuhs+AQCgYuo7N2o3cnB5ewlM\njJgoBsqjVYcOgvXHVq+NDpjuWiJZSqLWrjpWoPtZe1WLgqoMHLJXUWf+3rZ1665/7qaKWfqN\nOg5bOHbs6EEd7PQqtXIAAKh76jW0qdfQRttVwEfQC5g60Sl4xcz23jHzF33r/4mLrUFeUuz9\nUxvmzN96V+L447RBqlb+Uhk4Rlk32y0iXRuvgbMWjh0zpKuzYXUe4gEAAIAqwPdefPT3qJ7f\nHd/8/WebizdwbPv8HvKzj8r7HyoDh1PPH9aNHTvM380Ud/MBAADgDX7zCcce+hzcsGHfmZtR\nSa/EfBO7pu0+Hfrtt4NamJQxNqEycMw/gjXMAQAA4EMsU/fPA//5gZNbAAAgAElEQVT8PLA8\nffD0IwAAADAOgQMAAAAYV56FvwAAAKDu+aV//1Ay7bN029imYb/0Xx5a1rk+Px2dUeGVRgEA\nAKDuuRUSEkKWDtOIKOlWSEhIWedyh6tqqPy6AAAAoBb568WLNcQxtCCiXn+9eLGmrHMFypc9\nR+AAAACAstWzs6v35kO9Yh+XDyaNAgAAgIaeX929+8oTudK2149O7t599bmKnggcAAAAoKFb\nv3311errYqVtyUdnf/XVb7dU9ETgAAAAgEogk8mIxVYVLBA4AAAAoOKEkZHPqV49VTM8MGkU\nAAAAynZzzZDVN4iIEm8SKX4fMeR4yQELmejV8/v/3nqt17d9KxUvgcABAAAAZXtxbf/+w+8+\nu3Fw/40Pz9Gxbjd+9eovjVS8BAIHAAAAlM1/fWzsMiKi05NcJkl/jfijX4lt6FkcvoGZtaUR\nr4yXQOAAAACAsulbOTsTEVH3b5Yulfds4uzMKe9LIHAAAACAhpr0nzmTiIikr55FxyXnmzbz\ncjLWqCeeUgEAAIByyL7x25CWFuYOLTx9Pl1x+83BtH1jPZp7BV2RqeqFwAEAAACakkWt+sxv\nyv6Eeh37tbMpdtyid0+np7e2HryqUNERgQMAAAA0VHBs8cIbEq/FoQ8ubh7pUrzFoFffLpyU\na9fiVfRE4AAAAAAN3b1wIUcvYNr0FoIPmgSurg0oMTFRRU8EDgAAANBQZmYm2Tg48JW1cTgc\nys3NVdETgQMAAAA0ZGpqSunp6cqasu/ff0rW1tYqeiJwAAAAgIZat2+v+/rE4fOFpRuyLwYt\nPycz6dy5pYqeCBwAAACgIf1B0yY6pm8dEbDkVGSmlEhekB57+8ymH7s277khntvix6mfqVpt\nFAt/AQAAgKZ02v8c8nus/6Sgz04REV3va7GpqIFnP3jz4VnuKgcyEDgAAABAc/zmE44+8j26\n8c+/T4dFPM8UcY1tXdv5D580caCbMUt1NwQOAAAA0FBW7I2YTF1bd4+AGesCZpSnJ+ZwAAAA\ngIYuB/p4d1p8vdxbtyFwAAAAgMbMzMyIKxAoXYejbAgcAAAAoKE2nTvrC2+FPVC5R5tKCBwA\nAACgIf1BP6/snrF61JSjCcLy9cSkUQAAANDQnT+XhNV3t72yPsB1XxNPD9eGNiY6JR5N8fxu\n+8Q2ynoicAAAAICGnl7aseNw0YcZj0PPPw4tfUJebwQOAAAAqJjuv96/H1TWCSaNVDQgcAAA\nAICGjB08PD6uJwIHAAAAfAS56FV6eo6Yb1y/vomu2odQ8JQKAAAAlIf4yfElIzo41jM0tWro\n0NDK1LCeo++In08+k5TVCYEDAAAANCYKX+7Xqk/QrutPC/RtXFq29nCxMSh4em1XYK9Puv/6\n8INt699B4AAAAABNRa0cPftqbkP/ecER6dlJMf/duR+TlJUeeWS+f8OcKzNHr45W1RGBAwAA\nADQUvX/vfXmzGf8cmx/Q1OjtAhwsI9f+844d+6mp7O6e/Y9V9ETgAAAAAA3FxcVRo1793T98\n5oTbsl8vB4qLi1PRE4EDAAAANMTj8UgoVL6qeX6+kHg8noqeCBwAAACgIXd3d3p5aOPRzA9a\n0oM3Hk6jli3dVfRE4AAAAAANWQ//tr/Ry31fevWateNSxLP0/ML89GcRl7b/5O817ECa8cCJ\nw6xU9MTCXwAAAKCp+sM2Hbj3bPCqk8tGnVxWosWozfRDfw0xV9URgQMAAAA0Z/7pyhuR/XZs\n2Bpy5X5sSo6EZ2zt0qpz/zHfjvC15avuhsABAAAA5cK36zju547jytUHczgAAACgvKSvnj26\nE3YzPkfTDggcAAAAUA7ZN34b0tLC3KGFp8+nK26/OZi2b6xHc6+gKzJVvRA4AAAAQFOyqFWf\n+U3Zn1CvY792NsWOW/Tu6fT01taDVxUqOiJwAAAAgIYKji1eeEPitTj0wcXNI12Ktxj06tuF\nk3LtWryKnggcAAAAoKG7Fy7k6AVMm95C8EGTwNW1ASUmJqroicABAAAAGsrMzCQbBwelj79y\nOBzKzc1V0ROBAwAAADRkampK6enpypqy799/StbW1ip6InAAAACAhlq3b6/7+sTh84WlG7Iv\nBi0/JzPp3Lmlip4IHAAAAKAh/UHTJjqmbx0RsORUZKaUSF6QHnv7zKYfuzbvuSGe2+LHqZ+p\n2i0WK40CAACApnTa/xzye6z/pKDPThERXe9rsamogWc/ePPhWe4qBzKYDRzi1JuH9xz9Nzwh\nLV8hMHdo4dtn2GBfW52yOz3b/b9JB5I+OGzSa9nOb5oxVCgAAABohN98wtFHvkc3/vn36bCI\n55kirrGtazv/4ZMmDnQzZqnuxmDgkD47Nm/m5gihgb1nh56WnIzIsNADK+6GJy9d8oVjGbu7\nFGFZtezSzKz4Ef0mJszVCgAAAJpiGbsFzFgXMKM8fZgLHKnH1m2PyDf0mrx6VjcLNhEphl5f\nOXX5v3vXHvFe/UXDMkIQERHXte+UKZ6MFQcAAABVibHAEX/2VIyUHPuNKUobRMQybT96UNOr\nG6POnHn8+ThXNYkDAGqwwrz853cflrsbi+XS0YuBcgBA+5gKHGkPHrwksmnVusTzuGatWzei\nqIQHD5LJ1bbM/vKUWwe330l5LdMxsnRo1tanjb0hHqgBqDFyUtJOLFittCn3ZSZXj69nZPhh\nE5vDcTn3N8OlAUAFyVKubFi0cve5W4+TsvPFslJ7pww8IN0/QFk3pgJHYlISEdnalYoVlrY2\nHEpISk4iKjtwyB6f2fX47SfBu7bYd/9u1rcdbThM1AoAlc3CpdGUi/uVNu3936wGnzTvMH5Y\nFZcEAJUi8/i4tv22JcpZfGOrBi72BtxSNywaGqnoyFTgEOYLidj6At2Sh1n6+gKi3Lx8MZGq\niaP6Tt2GT3HxdGtoaaovz0lNuHdqz+6TUedWzReYrR/rVqLXqVOnCgoKij4uLCwsLCwUiUQM\nfDV1hVQqJaLCwkI2G+NJoIRMJiOiCn6XyWQyiUSCb9VaSaFQKBQK/OdWkFwuVyhUbbmqdU82\nzd+WyGo8Zk/wqqFlPpTygapeh0NBCiJilVWiufegz999Ur+h+6ffNHXUnzbjwJPjh/4d6uZX\nfLuYdevWpaWlFX3s7Oycl5fHKvOlQRNCoVDbJUC1lpeXV5HuUqlULBZX8EWgOsN/bgUVJfvq\nKuLRI9IJWPjHl25qnzcthanAIdAXEOXnC0VExQc5FML8AiKuvqBcdfJcundyOPAkPiYmnvxa\nFGtYu3atRCIp+nj16tXGxsZGRqrGckC9wsLCgoICIyMjjHCAUkKhUCwWm5hU6BF1Pp+vp6dX\nwReB6qlo3y5DQyUTdEBzXC63Gv/xrK+vT8YWFuVNG8Rc4LCztSWKSUpMInIqdvhlcrKMyN6m\n7AkcH9LX1yeid+HiLWdn53cfczgcLpfL5WLt1I9X9A/M4XA4HMyWASWKfghW8LuMxWKx2Wx8\nq9ZKlXKFQDVOG0TU1s/P8K+bN59Ql0bl7MnUH7IW7u6WRMn37qUWP5p19+4TIjN3d5vyvZrw\nUUQCEVlbqdqDDgAAAJinP+iX9X0Sl41ecC2tnHd+GMuhTj38Gx/ZHhOy7WLnn7rWZxOR4lXo\n9sORCrZjzx5N3qe3nKgLV2LzBS6d/JoaExHlx96M4Lm2djB++ye24nXcybV/Xs4jTpMuHRE4\nAAAAqtjvo0bdLvapkZv5mfm+ztvatPVwsTXhlxyQ8fxu+8Q2yl6EuYEvq76TRt3+aUvYb5On\nhLVvYcnKiAy7GZej23jYpAD7YsVl3tq/+XCqxcBmbwPHo4OLt8UbWtnbWZrXE8hfvXwW/yRd\npGCbtZ3wQx/kDQAAgKp2aceOwx8cFD+7c+HZnQ8O5/Wu8sBBXPt+C9ZYHd595Er4lVN3SWBu\n7z14/LDBvnZlTjUxatlr0Kc3oqITkhOSY4Qynr6pTcsunl379uvkZFCd72oBgDoFObkxl8Ne\nRseLhQXmTvbOHdpydT5i5hkAVLVf798P0vRcE1VzO5id2sO38ho6zWtoWac4jvzr2Mhin+s6\ndhkxsQujVQFAFZOJJWeW/371z90SUSERJT6Ienj8gmF9M/+g79sOV7omIQBUIw4e1v+dvp9h\n29qvRf2PfhE8/QgAzJIWijcNnnBxzZaitPFObnrmgcnz/pmzQluFAYDGri729++/8mZFXgIP\nLwHUKtKMs/L8uPL24hi14pi2Y6IeIjo+b2XctVuqWq9s2Gnn4fbJwM8YencAqCYQOABqFYU4\nU17wTEmDTCRJO8mzVn7/gi1wZKieV4kpYdsPlH3OqcW/eQT0ZGG5OYBaDYEDoFbh2Qzl2SiZ\nN6UQp0vSjum6Lq/ieiJOX5ZJpGWfk/U8OfFBVAMPt6opCQC0AoEDABiUHvdUo9NinyJwAFRz\nioJXqamp6s7SNbUy0VHWgMABANpXjffGBIA3hAe+slZzg5SIBh5UHBqkrAGBAwAYVN/ZQZPT\nLFw0Og0AtIilb9HQXE/dWRb6KhoQOACAQc0+7XQs6Jeyp3GYNrCxa9msykoCgI8jGLTl6fbe\nH90dgQMAGGTawKbdiMHXt+wt45zPgibjERWAchGn3jy85+i/4Qlp+QqBuUML3z7DBvvaKp06\n8d6z3f+bdCDpg8MmvZbt/KYKIj8CBwAwq8+iH1OjY+Ovf7jnAhGR7zfDPxmERTgAykH67Ni8\nmZsjhAb2nh16WnIyIsNCD6y4G568dMkXjmp3C2BZtezSzKz4Ef0mJszVWgwCB0AdoJDLRS+I\nFHLRC7Zugyp+c66OzvhDf51euv7qxj3SwveLjRqYm/oHTfb6amAV1wNQw6UeW7c9It/Qa/Lq\nWd0s2ESkGHp95dTl/+5de8R79RcN1Ww7xnXtO2WKZ5UUWvqdtfGmAFBFFLI88bM/JEl/K8Tp\nRJR/3Zut15BnN5rfYDSxqu7bn8Pn9Zr3Q5fvx8RcDru4ZnO9BjZeIwa5dPLi6qgZAgaA0uLP\nnoqRkmO/MUVpg4hYpu1HD2p6dWPUmTOPPx/nWl03OkXgAKi1FKJk4X/D5fkxxQ/KC54Xxi6Q\npp/Wa7mdxTWsynoEpsYeAT2jzl5p8Enzpj06VuVbA9QaaQ8evCSyadXauvhRs9atG1FUwoMH\nyeRqW2Z/ecqtg9vvpLyW6RhZOjRr69PG3lCTGVT9dmZnS/gGFakcgQOglpKLheEjS6WNd2Sv\nbooeTdTz2FnFRQGAJkJDQ3k8XtHHnp6exsbG75oSk5KIyNauVKywtLXhUEJSchJR2YFD9vjM\nrsdvPwnetcW++3ezvu1ow1FTEU9gUtGZHggcALWTOGmXPC+qjBOkmRelGee45t2rrCQA0NCK\nFStev35d9PHWrVvd3d3fNQnzhURsfYFuyR4sfX0BUW5evphI1cRRfaduw6e4eLo1tDTVl+ek\nJtw7tWf3yahzq+YLzNaPdVM73bSiEDgAaidpyiG150hSDiJwAFRD06dPfzfCYW9vr0kXBSmI\niFXWBA5z70Gfv/ukfkP3T79p6qg/bcaBJ8cP/TvUzU9QgYI1gcABUBspZLK8SLVnyV8/rIJa\nAKC8fHx8it9GKU6gLyDKzxeKiIoPciiE+QVEXH1BuQYqeC7dOzkceBIfExNPfi0qUrEGsNgO\nQC2kkBeSQqb+NFl+FRQDAJXIztaWiJISSy3g9TI5WUZka1P2BI4P6evrE5FEIqmk8sqAwAFQ\nC7E4AhbXSP1pujZVUAwAVCILd3dLouR790ps25p19+4TIjN393J+TwsfRSQQkbWVtdpTKwyB\nA6B24ph1VnsO16wT84UAQKVy6uHfmEMJIdsupsuLjihehW4/HKlgO/bs0eT9HI6cqAvHjh07\nH5Xz5vP82Ju3nuYUG/lUvI47serPy3nEadKlYxUEDszhAKid+PYTpC+PlXECiyPg241h6N3T\nYp/8/c1MpU25aRlPb4Xf2aekNjaH8/25vxkqCaC2sOo7adTtn7aE/TZ5Slj7FpasjMiwm3E5\nuo2HTQqwLzZnNPPW/s2HUy0GNvNrakxElP/o4OJt8YZW9naW5vUE8lcvn8U/SRcp2GZtJ/zQ\npwryBgIHQC3FMXTXcZxemLBC1Qk6rstZOpYMvbuxtUWveT+Uu1uZM+wBoAjXvt+CNVaHdx+5\nEn7l1F0SmNt7Dx4/bLCvXZkTRo1a9hr06Y2o6ITkhOQYoYynb2rTsotn1779OjkZVMl3HgIH\nQK3FbzSZuIbi+KUKWUHx4yyeqa7rMq5FL+beWsdA36VTO+ZeH6CO41t5DZ3mNbSsUxxH/nVs\nZLHPdR27jJjYheG6yoLAAVCb8RuM4Vn0kqQelmWFSrOucC16ckx9eFYDNZlSCgBQiRA4AGo5\nlo4l3/5bhfXgvOtt9Vps0nY5AFBH4SkVAAAAYBwCBwAAADAOgQMAAAAYh8ABAAAAjEPgAAAA\nAMYhcAAAAADj8FgsQK2ikAlJoWTfR4X0NREppDkfNhERi61DbF2lTQAAlQKBA6BWKYyaIXl5\nVFVr3hU3pcf5DhN1nGYxVhQAAAIHQO2i23y9bvP12q4CAKA0zOEAAAAAxiFwAAAAAOMQOAAA\nAIBxCBwAAADAOAQOAAAAYBwCBwAAADAOgQMAAAAYh8ABAAAAjEPgAAAAAMYhcAAAAADjEDgA\nAACAcQgcAAAAwDgEDgAAAGAcAgcAAAAwDoEDAAAAGIfAAQAAAIxD4AAAAADGIXAAAAAA4xA4\nAAAAgHEIHAAAAMA4BA4AAABgHAIHAAAAMA6BAwAAABiHwAEAAACMQ+AAAAAAxiFwAAAAAOMQ\nOAAAAIBxCBwAAADAOK62C6g0crlcJBLxeDxtF1KDSSQSIhKJRGw2kigoIZPJiKigoEDbhUA1\nJZfLCVdIhcnlcoVCoe0qKl/tCRwsFovFYuE3ZUWwWCwiYrPZ+GeEMuDyAFXe/QzRdiE1XtG/\nZC1TqwKHjo6Ojo6OtgupweRyeWFhIZ/P53A42q4FqiOxWCyVSvFdBqoUjW3gCqmg2prYaudX\nBQAAANUKAgcAAAAwDoEDAAAAGIfAAQAAAIyrPZNGiWjChAmY7VgRcrlcLpdzOJxaOUEaKk4m\nkykUCi63Vv3cgEpU9OA0fg5X0PPnz7VdAiNYtfJhX/g4O3fuXLt27aFDhxwcHLRdC1RHc+fO\nPXfuXFhYmLYLgWpqyJAhfD5/586d2i4EqiPcUgEAAADGIXAAAAAA4xA4AAAAgHGYwwHv5ebm\nvn792tLSErMCQamsrCyRSGRjY6PtQqCaSktLIyILCwttFwLVEQIHAAAAMA63VAAAAIBxCBwA\nAFB5pCKRVNs1QLWEwAEAAJVEmnRm2f++XXY2WabtSqD6QeAAAIDKIE06syxww63MjFu/z/4t\nLFfb5UB1g8ABAOWikMnk2q4Bqp83aSOLWCwihVRHV1/bFUF1g6cfQTl57pPQ66mO3b1tsCsC\nEBFJ0/47HXLy0s1HT9PzpBz9+s4tfbr1H9Dd1QR/tcC7tMG27tzD9u6pO3Ivb3dcGFAKAgeU\nVvgy/HxI8NFz918WsurdmfjzrB7IHHWd6MmJtb9su5YkfntAmp8WHXo0OvTsmT5TA8e2NcPv\nlrrsfdroOmNxt/9+vEyCLt4t8VMDSkPggHfkOfHXTwQHn7genysnjp4enwqyHt5NEPWwwdho\nXZb/aHvQouD4AuKYf9Lni4Buno0tdQvT426fO7j3n/C4f5ZMz5yy7KcuFthguG4qnjaWTG6X\ntvnPVyTo7O2BvAEfQOAAIhKl3jt/5MjRC+FpYiKOaZPuAV90lh2euyPC8tMAH6SNukyW9M/P\nS4LjC1imrcbOmdHXWVB0WNDAvccYd2+vvYsW740OXbtop83KkU342i0Vql7JtOFtzooOuZFN\nup28W5X41SKMDj6U2G6YH8ZK6zgEjrpNlhNz9fiR4JOhT3MVRDwzN/8BXwz81MOC9eiPbyJk\nnKb9+7pisLwOUyT/s3b7w3zSbzF+cVCvBqV/XBi6DZ0/R/RT4JFnwWv2+awf4YLfJ3XJB2mD\nKC40LJ102nu3KhY+hdF75s7bHyM6J2+4flRjXCJ1GQJHHaUQJd89czQ45OKjDDER8S1a9Bg4\nZKBfCzMeEVHGyb3nMsiwy0A/cy3XCdoke3D4cJSEBG3HT/swbRQRNBs5fVjElJ0xx/Zc7D+/\nu1EVVwjaoiRtEMWH3kgjvrd3a523p71JGwW6jgETByFt1HUIHHWONPvxlX+Cj5y68Tz/zTY6\nPJfhq5Z/3vDdtSCNPHz4oZTsegV46qh6FagLIsNu5BDV7/55Z1PVJ7EbBozzPzPjOo9TWHWV\ngVYpTRtET27cSCVuO+82b35uFEsbcxePbm6gvYKhesB4eZ3z5NyWvw6FPc9nGTRo02tApwY8\nksQeWLUj/N0qPVnn955NJ36rgF4OyuYBinJzsYZg3ZCbnJxLxHV1dSl7QiinSb/v5y/5qWv9\nKqoLtOx16M5tH6QNosSw0ETieni3FRAhbYASGOGoc1w+n7/Q7PxL2/bermY8olfuBoFLTiSE\nLJpDcxaNbWkoe3z4ULiETP0HdDb+sLMobv/8eacMRiyZ+aktrp3ajsflEpFMKpUTlT0YbtHi\nk6opCaoDo46T5+c453gP8jIvFkWTQsNeEKeVd1sB0gYohxGOOkjQpFvfjq5FszXIpNU3SwJ7\nNeCJE0IWzdkS/vzi3jNpxHLuE+DOK91PFLd//pw9kXlVXjBoh26jRlZEiod374vVnwx1icC1\nz+ASaYMoJSzsGbHdvb30kTZABQQOKJ45Fv7w5z0x6foE+FuVOulN2hDWa/sthjfqCsfOXexZ\nlPfvzn2xiBxQpvSwsHhiu/m0SEbaAFUQOIDofeaQSCRElp8GtC+5+IaqtIH5HLVcg4Bxn1mw\n5E8PL1l5Qc32nwqFooqKgmooIyw0jlgO9eJWqUwbuEIAf6jCGyafdG1teuJFGqd5/36NiwdR\nlWkD8zlqP133sUGjn83c+ihs7U/z0qZPH+JuXGoCqSzj/rG9h87eiE7OUxjYNu/Qe/gw/8ZG\nWHa0bskMC4tVkCLhyhmi0mkDVwi8xZk/f762a4BqQfZg/5rjcYWGnSb80M2mRKpQlTbm7InM\n17P17NbB2QgjZbUW28S1fSuDZ7f/S3j64NLZ0KcigbV9g3q6bCIiRU749jmz/7jwOC1XLCeS\ni1+nxt25cO25ZVufRgb4jVKHZMacu3gvQ/pB2sAVAsUhcECRwrBdv114Lmk4YOrYT0ze/CRQ\nkzYwn6OO4Jo28e3Wql5BUkJMXGySzNm3KGKKHu8KmhccJ9J39h8/fcbUCV/1btdQ9vTB4+cJ\nt/+TtP7UAzu61R3GTdq3oEd3hL6zi49t4AqBkli4rwZERCSK2T93zkHB+C3zuxU9Dou0AaVJ\nMqITpM5NrLhEJI3aMnFmSIqe28jlCwbav1vJWhSxcfLsEymcNt/vmOtnqMVaocqJxWI+//2a\n5rhCoDQkTCii2/iL+YsXje6EtAEq8cxdi9IGkejm0dMpCnbjoT8U+11CRLpu/T91IZJGRD7W\nTo2gNcXTBq4QUAKBA94RuDS15xLSBmjgRWxcIVFDT0+L0i18HewaC7hCQAkEDigt89qeYKQN\nKBuPxyOi169ySt+SzQwNfUzEadrURRtlQbWBKwQ+gMABpZn5TV8wbSbSBpTFvtUn9YiyLv59\nIqnY+hyiuEO/7nwoIYseAzu+WRlfJi6UYLGWOkjjK0SRHR2RJNVOkVC18JQKfIhnbt/g/ZOu\nSBvwIVZ9J7NnF0OfPL93PbLQyMKIJ0qNOL9z5dpjcSK+4+dB07vXf3U/ZMv639b8vmXP/uCT\nNxMKLZo2sxYU+wtHLhbLORz8yVNbqb1CrLhEpMi+syFwwV+nn9p19LUXaLtmYBieUoEyIW2A\nSqLY4CULd4QXHzNnGbkNmTV7iF3C9oVLj8QKiYjFZivkciJimfnOXDXdu2ije3n61TWB24WD\nFs3qYVP2vnBQg6m8Qoa6GdKbtLH0TKLc1m/Gkkne9bAwR22HwAFlyTy/4Nu19/SQNkA5WXbk\n2UNHL91PyJTqWbq06do/oJsLP3bnzNmHEiTGbgHfjAto52ggen7zwPr1R6PzjP3mbv2+DU+e\nfnVN4MrLqXzHAfOWjHLTV/82UGMpu0KMWPQ+bUiI9EwtBSwysm/ZdfCwvs1NkTtqLQQOKJsw\n+t9wQx9vpA3QzJvVFwxb/W/FHH/rd6MXBQ9+nxR0Rt731839Xr5JG9jcq+56lzYEjr7+HZuZ\ny5NunTl7/6WYbebzwy8/daqPzFE7IXAAQOURXV/61fIwectJWxZ1NynRkrDrf1MuOw5wiz36\nYdoQ5ebyDA1xb6VueH8npduPC77rYFH0/14Qu2fOrP0xYl3PKVvmdMWKYLUSpmwBQOV5EhVd\nSOTUpo1J6RaFQkHpV4OVpI24/fMnTFp6Bk8q1AUl5m18/zZtEJGey5cT+loRie7deYS/gmsp\nBA4AqDxFQ6avsrNLHRdHnv83mT7cSpRI9jLqv+d5VVgiaE3Zs0RZBgYCItLR0dFOdcA4BA4A\nqDyOzZsLiFJP7zyV/H71BVnqxdW/nkz7IG2IXiSkSDn2featWLsMs5LrgJTQI/+qfCYl786B\nk0+IzDp1cccUjloKczgAoBIpkk/NnfpHuJBl3OyzIf3aNeRnx1wNOXwpPk9HyZ2UOX+nuE1Z\nEdjFAr9h6ghhzM1Y87YtP0gbBdF7583bG11o9em8VRM/wVTiWgqBAwAqV+Gz85t+23EhLuf9\nGMcHYxtY3wXeeZM2RObek3+e0dUKk4drLQQOAGCAVJj5Mv7q1tXbb2comyWKtAFFiqWNJTO6\nvnuQWpYdcTb42KV78ekFXNOGzX38B/bxssbkjhoOgQMAGJEf9uu4pf9KkTZAJeVpQ5HzYMei\npcEx+cVP5dl2njz3+47WuGJqMAQOAGCIMPr0JVmHXm6apZfxCrEAABoQSURBVA1FdnSk0NkN\nGaTOkDzaOH72iaxSYxuFMbt+mn0wQSxw7P7ViF6t7XRfJ9wI2bXv6gsR27bvsjVfu2Kco8ZC\n4ACAKqEmbdzZELj0bH6baatm+ZppqUKoasLoo0dSvYZ0fr8krTxu16QfD77QbTZi+cJBDvy3\nh0Wxu2bPOhgnsf9y7boh9topFioMj8UCAPM0SBtnEuU2rTu71dNShaAFAtf+w4qlDSLFozNn\nXijIYdCkge/TBhHpugyfPrwJR/Hs6vUXVV4lVBYEDgBgXOa1PcFq0oaEiC98+NfU8ZPnrQ95\nlI2R1zop+9mz10Smbs1tSz84y7Lu0bOt/acje9pppTCoDLhdCgCMM/ObvoCvZBdA5Zt4nd0S\nfjcKm3jVRUXLjOamp4uoqW6pNoHvjyu78vm4Jmouzvz587VdAwDUejxz+wZGJUdUi23iNf2X\nOUPbuzVu3Kx1l+6fyB9cfpT49H6KVc9OjTBBsG7hG2TdPh6ekZrKbevXwrTUCDyHwylKGwqF\ngsVC8Kh5cEsFALQAm3iBMpafDe9ejyVL2LdoacjjvFIXgCzj/pF1gf8bNqh//wHDvp37x8mY\n17hEahKMcABAVVOziZc4/sI//2XrufgNameNv2PrGJ61R3Pd6Bv/PU24d/HSf8l5LLNGjmY6\nRIqc8O1zZv9x4XFarlhOJBe/To27c+Hac8u2Po0McJXUDBjhAICqhk28QDW+c8DC3xYMb+9s\n9DouNsfI3ICIRI93zVt0JEao7+w/celfew8e2Lbyx96N9WVpoWsX7YqVartk0AzW4QCAqodN\nvEAtuTCvUM9Aj0XSqC0TZ4ak6LmNXL5goP37xTkiNk6efSKF0+b7HXP9DLVZKWgGIxwAUPUE\njb1Upg2Ruff3CycUSxsKiUgokuJPo7qGLTDQYxGR6ObR0ykKduOhPxRLG0Sk69b/UxciaUTk\nY22VCOWCx2IBoDooua1G0Zahsox7IX8fPB0anSqUcQztPXuNHv9FK3PsJlrHvIiNKyRy8PS0\nKN3C1+Er6wDVFAIHAGidkk28FDnh2xcuPRIrJGJxdfjswtxnN/YtiE2auWq6t6m264WqxOPx\niOj1qxwF2ZQYFcsMDX1MxGna1EVLlUH54JYKAGiZ5NHO5aW2DC2aJBhbYNJi0Mz1ew4dPPj3\nhlkBrgaKzKsbdt2RaLtgqFL2rT6pR5R18e8TSbL3R0Vxh37d+VBCFj0GdjR+d1QmLpTIlLwG\nVAeYNAoAWldqEy9JxKaJs/9JNWz97Yqgnu+32ih4uGFS4Gl53183DzDOMrI0xwBtXZF9ddmk\nFaGvWSYtAkYO6djYqPDF7ZN/H7jyXMR3/HzxsuGuuiTLuH9s76EzoZHJ+XKecaM2n40Y93mJ\nu29ysVjB5+N2nFYhcABANSO8snjEylvUavKW+d2MS7Q8+3vipAsOA9xiQ8LtJi4N9LPBL5A6\nQhQbvGThjvCcYr+vWEZuQ2bNHupmWOzuG7HYbIVcTkQsM9/3d9/k6VfXBG4XDlo0qwcuGS3C\nnwgAUM3ERUaKiZp7ehqXblEoiNKvBl8mXUdvKyP86qg7dF0GzF/revbQ0Uv3EzKlepYubbr2\nD+jmYsQiUcyueYuOJEiM3QZ8My6gnaOB6PnNA+vXH42+umFXlzbft+HJ06+uCVx5OZXvmJwt\nIht9bX8pdRgCBwBUMwqFnIiyM7OIjIofF0efu/KCiHQdA+YuHt0cq3TULRzTZv7jmvmXPCiN\n2rPycILYqNX/fpnjX3T3zaChz5gFBgWTgs78998Tuf3LN2kjYO7iUW5IG1qFSaMAUF1IEh8n\n5BO5uLfUJUo6tf1U8vv5f7LUS6t/OZ6KtAHFFC3RwWs5coq/dfEBLz13/062xMoOfZ82il0z\notxcTCzVBgQOAKgWFCkXlgbO/vtmPgnajxjVUkD59/78KWjTqdsR0Q+uH107/cc11zN0kDag\nuCdR0YVETm3amJRuUSgUlH41WEnaiNs/f8KkpWeSsCB6lcMtFQCoFliStNRsScrh/Y/bj2ny\nWdBy/qbfdlyI+OePiH/enICxDSit6KmHV9nZRCUWZxFHnv83mZRdM7KXUf89z6NmVVsnEBF2\niwWA6sLYUh5++k585P0ki7btXOyc237at2+Pji2MXz56mCLE2AYoYSiOPXHtefazNBMfXxfD\nNyP2stSLqxbtjhaWThuiFwnp+mZ2TTu08+nQs7UN/tyucggcAFBN6Dd2rxdz6WZ8/M3zoU/y\nJcK0uNsndm4/Hp2j13jQ3IWjkDagNK5dI/2oi3cSn9+9+vC1jj6vMC362oF1a/aFvyqdUEVx\n++fP/vVojFV7XxdrEwPMJtAGrMMBANWI6Omp1Us2h718t5qonkO3cdMm+DXU0WZVUI0VPju/\n6bcdF+Jy3k8E/WBsI27//Dl7IoX12n67ZOanthjb0BIEDgCoZsRp4Zcu3Y7LkBnaNe/QpZ0j\nFtwAdaTCzJfxV7eu3n47Q9ksUaSNagGBAwAAarz8sF/HLf1XqlnaUGRHRwqd3ZA+qhYCBwAA\n1ALC6NOXZB16ualPG3c2BC49m99m2qpZvmZaKrZOQr4DAIBaQODas9f7z8pMG2cS5bZ+nd3q\naaXOugsjHAAAULuoSRsSIj1TSwGLjOxbdh08rG9zU5ZWy60rEDgAAKBWyTy/4Nu19/RUpA2B\no69/x2bm8qRbZ87efylmm/n88MtPneojczAOgQMAAGoZYfS/4YY+3krupHT7ccF3HSyKHnwq\niN0zZ9b+GLGu55Qtc7oaaq/cOgKrnwAAQC0jcO2oLG34zVjy/du0QUR6Ll9O6GtFJLp35xH+\n9GYeAgcAANRaJdLGJO96Je+csAwMBESko4OF5aoAAgcAANRaKaFH/lWRNojy7hw4+YTIrFMX\nd0zhYB7mcAAAQC0mjLkZa9625QdpoyB677x5e6MLrT6dt2riJ9iph3kIHAAAUNe8SRsic+/J\nP8/oaoXV86sCbqkAAECdUixtLCmVNhQSkVAkxd/hjMBKowAAUHeUTBvWb9OGLONeyN8HT4dG\npwplHEN7z16jx3/RyhwjH5UJt1QAAKCukDzaOH72iaySaUORE7594dIjsUIiFleHJy8Uy4lY\nZr4zV033NtVuvbUKAgcAANQdwuijR1K9hnR+N7YherxzZuChBIlJi4ETvhnQrqG+KPHG/rXr\njkTnGfvN3fp9G55Wy61NEDgAAKDOkkRsmjj7n1TD1t+uCOr5LoRQwcMNkwJPy/v+unmAcZaR\npTmmH1QCTBoFAIC6Shh65Eyqgt9q9JRiaYOI9Fr06tqAWNmha2ZO+PHn88kyrVVYiyBwAABA\nXRUXGSkmauzpaVy6RaEgSr8afDmVZ9rAygizRysBAgcAANRVCoWciLIzs0odF0efu/KCiHQd\nA+YuHt0cy4JVBgQOAACocySJjxPyiVzcW+oSJZ3afqrYXRNZ6qXVvxxPRdqoZJg0CgAAdYsi\n5cKimRvYI3cGdRUknZz745/hQpaxW+8vB/g00nkVffnwwQtxuTpIG5UNgQMAAOqY53u//W5v\nSoP+y1aNaaJT+Oz8pt92XIjLeT/GgbENJiBwAABAXZP2z8xvN0VKzH2+XzKtmzWXSCrMTHl8\naeu63Xcz+EgbzEDgAACAOkeRdn7BD2vv5ZJug3afdW9tw3557+zJ0OdCQeNBc+aPcEPaYAAC\nBwAA1EWip6dWL9kc9lLy9oCeQ7dx0yb4NdTRZlW1GAIHAADUVeK08EuXbsdlyAztmnfo0s4R\nC24wCIEDAAAAGId1OAAAAIBxCBwAAADAOAQOAAAAYBwCBwAAADAOgQMAAAAYh8ABAAAAjEPg\nAAAAAMYhcAAAAADjEDgAAACAcQgcAAAAwDgEDgAAAGAcAgcAAAAwDoEDAAAAGIfAAQAAAIxD\n4AAAAADGIXAAAAAA4xA4AAAAgHEIHAAAAMA4BA4AAABgHAIHAAAAMA6BAwAAABiHwAEAAACM\nQ+AAAAAAxiFwAAAAAOMQOAAAAIBxCBwAAADAOAQOAAAAYBwCBwAAADAOgQMAAAAYh8ABAAAA\njEPgAAAAAMYhcAAAAADjEDgAAACAcQgcAAAAwDgEDgAAAGAcAgcAAAAwDoEDAAAAGIfAAQAA\nAIxD4AAAAADGIXAAAAAA4xA4AAAAgHEIHAAAAMA4BA4AAABgHAIHAAAAMA6BAwAAABiHwAEA\nAACMQ+AAAAAAxiFwAAAAAOMQOAAAAIBxCBwAAADAOAQOAAAAYBwCBwAAADAOgQMAAAAYh8AB\nAAAAjONquwAAqAC5WPbqplyYQCwOW+DMMWlDLHxTExEJs3Pir9/OSX6pa2Rg697MupmLtisC\nqOvwswlAvejFHk3nOOyVHB1Sed8xr0I3rzmr23/2cA/+x72AQvxii/jJGoXkFSmIWERELB0L\nHcfpPJuhlVZlVctY37n+7t5Pbkxz+OiXEL3OO7Fg1a09R2QS6buDdi2b9V82y6GtRyXUCAAf\nBbdUALTjVejmBQt2/yf+qM4KuejRpMKY+QrJK6I3aYOIFIVpoqjphY/nVKQw8dPTP4/u5mZj\nrKdrbOvq1X/q5hsvJeV/mdQ1HVgd1iRWpJKPkJeRva7n8LDtB4unDSJKDI/8s//Y8KNnPupV\npbv7s1gsky8OZr87dPprE+6QQyTa3ZullHPQf0W9WCwWi6trWN/BvcvQGVtupMkr9gUC1FwI\nHABVTSIs+Ihf4MWJn/0heXlUZWviNkny3o97ZenjPz5r89mKh45fr953IezK4bXfe2b9OXbJ\n9Y+tVCmZWCyr1Bd8b8/4GS8fxyttkhaK900MfBmT8HGvzNctPDxvRXipvKDTa939N/4cZEJe\ngf+++eyf75oQEQn6/xEVFfXw9uXg36d2ZZ35oX3LQTueKj6uAoAaDoED6pbChBOLRnZpZmOs\nqyMws3Pr/OWKazlvmrJub/y2ZwtbY12+oJ5j20GBwTGFJfvKUy4s/sKzgZGeodX/27v3cBvr\nRIHjv2VfI9eNtqI2ox5EcTgTQtFFokFlOlPq6DLhREZjVNKYwjBdOFOG6XRXMY+pM3S3a0RD\ndKppdCOmNiW5pJCy2bZ1/sCmib3X3p3fM0f783nWH971vr/fepfn4f2u129tJ3S+fMobm7+x\nt5Thy8e1TiT63PvXh4Z2b3V0jazsNuNmjchr/ItXQ5h7efU9n4FbT/x7qm8huevLnavuLuNt\nfnBb2F2R2xL3XTN8/jGjFyy+d/hFPTq2ad3+7EtuemjJy2NahxDCV0vvuapLk9rZWdUanNz3\nl/lr91x7V01slzj9V7PG9T3pmJq16p9w5vXPfZIMYXb/vOGLwqLhjRKJRCJ3yMI9h42ePqpn\niwbVq3W888NDzPadrJi/eOWCJaUcUFS4Y+6E31Vs8tr9BvXbcNfNj278xrOJ2o1b7/WDnLRw\nZMNWe7ea5x4RQgiJmg2bNWvWvFXbzj0H3DxjSf51dWZfO+KJrRU7BTi8CQ4qkV3vT+3Rrtet\nL1Y558apf3zqT/dNHNghWVCwLYQQCl8b07XzoOlrWg+b+sc5j912UZ0lv7mgwwUPrj5g9NJb\nzh24+LgBt0+f8bshJ6+dMbTH0Dnb9u1LYfiCkRdOy+w/9YW3P1j2+JVn3bTwL6Nah9Dl9ncL\nCgoKCgqeHZyX6rso3vRSsnhb6cckd24s3vxqqjOW+OzJWfOKeo4YeVLGgc+m5+TUCmH73J+d\nO3TBsTfM+etbC6f23DjlR/825aN9R7wyZVrx8PyCzetfueHI+y8Z+fRXoc+jqyafGk6d/HEy\nmUyum9IphBDCorun7rr2+Q+2bJ0/tH4ps1XY0jn5ZR6zLH9BUeGOMg87iBq9b/lFi+dumfA/\nu8o+9hCOaH/doA5bn31yfsWngMOXRaNUHptnjhz50s4z730r/6qGe1Y9dO/bf8+ujQ+PmvhW\n9vmzXn6kX50QQujRu1O1di1vvmH8vEv/q9vePyXrcq9+95mhTUIIoc95x61/9fTfP/L8g70v\nTE9x+I4Oo+fceWluydl8XTsrhCPqHpuXd2QZ5138xaLirUv3b37+l1Te7c41D6R9+VbJZlqt\nH6bVbFfGmA9Wrkw2vLjZwU5oy6z/fGjnZU/dd3WX7BBa/Hr6B/lNJ09749oJbUMIod5PfnVz\nl9wqITS9fFCvYVcteDuc1/4gc9S8cMz47sdmhBCKZ5YyW+pemzF728bPSzZXLlhc5pCiwh3P\njb+7et06Jc+0vahXjdz6qbzcCUPH9p/cZ/QDI/KvPrp8J1qiQYsWtba/v2pDCBWdAQ5b7nBQ\naex++bn8r7J7/8cVDRP/uKt44byXd1brPeDCkstQlWZXDOgYNsyb907JQW379Wuy79fpbdue\nFHatXv1J6sNP7t59f22US7Loi93bV+9/FG0ue0wIu3d+fuCo5K4U7+MnEt/67QkhhL8vX76r\nTceO2Xs3m3TocNTq998v3LvVtOnev0oSdevmbNiw4eBTN23ePCOF2VK3ee36Tas/LnkUbU/p\n1sXmNZ9+Y1Rhygt3q3Yfc+O/zh837uUK3SIJIYSQTFrBQWXlDgeVxpcbNxaGoxs2PEhlf7lp\n086Q26DBgZfa3KOPToS3N20qeaJu3br792ZnZ4ewffv21IcfddRRFTzx9Pq90uv3KtksWvdE\n4bvDyhyV1XhYek7X8r3SD5o2TXy8bNm20K6suy7flJaWdsDW7t2HWI+RnZ198B0VddaIgQdu\n7tpR9Pof5pQ+JC0j/aIpY7OqVa3YK+YNGnflHWeNvmfkqIqNX/vee1uqNs1L6YYKfM+4w0Gl\nUb1eveywds2ag1wNq+fkZIb1n3564IfPdWvXJkNOTk4qM3+34eWUXuf0UKWMn92RSK+RXrtD\nuaeu2/vHXTOeufOOd7+xxqB406YtoWmzZul/W7Jk30f7DxcvXp/XrFkpAZGZmRmKiw/xbZRy\nz5aSlj27lXnM8ae1r3BthBBCVpfRo7u8NuHW/K8rMHj74km/X1Lz3N6n+6RHZSQ4qDSqdOlx\ndrXCOVMf+PbPhkjrdMZpmduemv7fJf9YkVzx0PRXQv1u3VqmMHOFhmdlZe2/R1IeicyczEZX\nln5MZt7QUKUC1+/cn065o/NHt3buOPC3s+Yu/tvSV/MfGz+gfZdb3gw1fzz88oyHh/70voXL\nVrw5e9Rlv36n088G/0spU9XJy6uxYuHc99Zu/Ozzr/5xkWS5Z0vJieec3qj1iaUf0/2Ga77j\nqxxzxbhBVR+Z9qevUjk4uWXN8uXLl73z5qJnHx7bv8M5k7/oc9ft59f4jqcAhyXBQeVR6ye3\n3dY188XBp5z587tnPP3CC0/OnDq6/zWPfhJCqHfZ+BtO+vrxAaf9++2PPf387PtH9Tzjl2/U\n7Dnxpm4pfRatyPDcVq3qhTdmTHpiwZLXX3/9vXXlWRaQ1WRkWu1TD7U3vf65mccOPNTe0qU3\nH/L8609d13zFPcP6dT2lc9+hv32tzsAHR3cK4YizJz17V+eC8ee1adlp8DN1hzz5hyGNSp2q\n+4jJZ310a4fGR9Vrcf23vqta7tlSkahS5dIHJ5WyArTvb0Y1apNKQpYq45Qbx/RIK0zpmyZf\nzx7cvHnzlm07/2jwHfOKz560aOnjl+UddJEMfP8loTLZvnLOmIs7HV+vWkZG1ToNW3S95I6F\nW/bu2vTqtEFnt2xQPTM9u9Zx7c6/8fH3C/eNWjb25BB6zyzaP0/RzAtCOHHssv3PlGt4MplM\nJtc+fX2PE3OrZ1QJIZw8YWU530nxzsKV47bOa7z1xWNKHl++dPyODycnd+8q51zfK1s+XX//\nxUN+XqflgY+xJ5319jN//mefGlRqiaQ103DYSu7cuOuzP+/eviok0qpUbZKec0Yio9Y/+6T+\nX1i/4sOV8xdv/mRd1pHVGrVp2bTzD9OzKvif1gD/JwQHABCdNRwAQHSCAwCITnAAANEJDgAg\nOsEBAEQnOACA6AQHABCd4AAAohMcAEB0ggMAiE5wAADRCQ4AIDrBAQBEJzgAgOgEBwAQneAA\nAKITHABAdIIDAIhOcAAA0QkOACA6wQEARCc4AIDoBAcAEJ3gAACiExwAQHSCAwCITnAAANEJ\nDgAgOsEBAEQnOACA6AQHABCd4AAAohMcAEB0ggMAiE5wAADRCQ4AIDrBAQBEJzgAgOgEBwAQ\nneAAAKITHABAdIIDAIhOcAAA0QkOACA6wQEARCc4AIDoBAcAEJ3gAACiExwAQHSCAwCITnAA\nANEJDgAgOsEBAEQnOACA6AQHABCd4AAAohMcAEB0ggMAiE5wAADRCQ4AIDrBAQBEJzgAgOgE\nBwAQneAAAKITHABAdIIDAIhOcAAA0QkOACA6wQEARCc4AIDoBAcAEJ3gAACiExwAQHSCAwCI\nTnAAANEJDgAgOsEBAEQnOACA6AQHABCd4AAAohMcAEB0ggMAiE5wAADRCQ4AIDrBAQBEJzgA\ngOgEBwAQneAAAKITHABAdIIDAIhOcAAA0QkOACA6wQEARCc4AIDoBAcAEJ3gAACiExwAQHSC\nAwCITnAAANEJDgAgOsEBAEQnOACA6AQHABCd4AAAohMcAEB0ggMAiE5wAADRCQ4AIDrBAQBE\nJzgAgOgEBwAQneAAAKITHABAdIIDAIhOcAAA0QkOACA6wQEARCc4AIDoBAcAEJ3gAACiExwA\nQHSCAwCITnAAANEJDgAgOsEBAEQnOACA6AQHABCd4AAAohMcAEB0ggMAiE5wAADRCQ4AIDrB\nAQBEJzgAgOgEBwAQneAAAKITHABAdIIDAIhOcAAA0QkOACA6wQEARCc4AIDoBAcAEJ3gAACi\nExwAQHSCAwCITnAAANEJDgAgOsEBAEQnOACA6AQHABCd4AAAohMcAEB0ggMAiE5wAADRCQ4A\nIDrBAQBEJzgAgOgEBwAQ3f8Cw08WBlAlXWsAAAAASUVORK5CYII=",
      "text/plain": [
       "plot without title"
      ]
     },
     "metadata": {
      "image/png": {
       "height": 480,
       "width": 360
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "figure_a = c('All','Coding','Noncoding')\n",
    "\n",
    "\n",
    "df_tmp <- data.frame()\n",
    "df_rr <- data.frame()\n",
    "\n",
    "paste(\"consensus size is \", consensus[1,])\n",
    "consensus_size = consensus[1,]\n",
    "\n",
    "\n",
    "for (eff in figure_a) {\n",
    "    c = c(sum(df.mm[, eff]), sum(df.ss[, eff]))\n",
    "    T = c(consensus_size*samplecnt.mm, consensus_size*samplecnt.ssc)\n",
    "    pt = poisson.test(c, T, alternative='greater', conf.level = 0.95)\n",
    "    show(paste0(eff, ': ', round(pt$p.value, 3), '   ', 'Rate ratio: ', pt$estimate))\n",
    "    df_rr <- rbind(df_rr, c(eff, round(pt$estimate,3), round(pt$p.value, 3), \"MM\", \"SSC\", \"All\", pt$conf.int))\n",
    "    df_tmp <- Get_summary(df.melt, eff)\n",
    "}\n",
    "head(df_tmp)\n",
    "\n",
    "df_tmp$class <- factor(df_tmp$category, levels=figure_a)\n",
    "\n",
    "df_tmp$cohort <- factor(df_tmp$cohort, levels = c('SSC', 'MM'), labels = c(\"Control\", \"NTD\"))\n",
    "df_tmp$Gene <- \"All genes\"\n",
    "df_tmp_all <- df_tmp\n",
    "df_tmp$mean <- as.numeric(df_tmp$mean)\n",
    "df_tmp$lower <- as.numeric(df_tmp$lower)\n",
    "df_tmp$upper <- as.numeric(df_tmp$upper)\n",
    "df.melt$variant <- as.numeric(df.melt$variant)\n",
    "\n",
    "colnames(df_rr) <- c(\"category\", \"RR\", \"pvalue\", \"MM\", \"SSC\", \"Gene\")\n",
    "df_rr\n",
    "\n",
    "write.table(df_tmp, \"/projects/ps-gleesonlab8/User/hiyoothere/NTD/5.Analysis/DNM/Burden/Figure_broad/All_rate.txt\", sep='\\t', quote=F, row.names=F)\n",
    "\n",
    "p = ggplot() +\n",
    "geom_errorbar(data = df_tmp, aes(x = as.factor(category), \n",
    "                        ymin = lower, \n",
    "                        ymax = upper,group = cohort,color = cohort),\n",
    "              position = position_dodge(0.5), width = 0.2,size = .3) + \n",
    "geom_point(data = df_tmp, aes(category, as.numeric(mean), group = cohort, color = cohort),\n",
    "        size =3,\n",
    "          position =position_dodge(0.5)) + \n",
    "    scale_y_continuous(limits = c(0.3, 1.8), sec.axis = sec_axis(~ . / (100000000 / 2) * total_size, name = 'Theoretical Variant Rate (per child)')) +\n",
    "    labs(fill='Cohort', x='', y='Variant rate (x 10e-8)') +\n",
    "    theme(legend.position='bottom', \n",
    "          aspect.ratio = 0.8,\n",
    "          axis.title.y.right = element_text( angle = 90,size = 12),\n",
    "         axis.text.x = element_text(angle = -45, hjust = 0.01,size = 12),\n",
    "          axis.text.y = element_text(size = 12))+ \n",
    "scale_color_manual(values =c(\"#F2BC1B\", \"#73122C\")) +\n",
    "geom_text(data =subset(df_rr, Gene == 'All'), aes(x = category, y = 1.75,label = paste(paste(\"RR=\",round(as.numeric(RR),2), sep=\"\"), paste(\"p=\",pvalue,sep=\"\" ), sep='\\n'))) + \n",
    "    ggtitle(\"DNMs\") \n",
    "ggsave(\"/projects/ps-gleesonlab8/User/hiyoothere/NTD/5.Analysis/DNM/Burden/Figure_broad/Coding.pdf\", plot= p)\n",
    "\n",
    "p"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a32c9fd8",
   "metadata": {},
   "outputs": [],
   "source": [
    "figure_a = c('All', 'SNV', 'INDEL')\n",
    "df_tmp <- data.frame()\n",
    "df_rr <- data.frame()\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "for (eff in figure_a) {\n",
    "    c = c(sum(df.mm[, eff]), sum(df.ss[, eff]))\n",
    "    T = c(consensus_size*samplecnt.mm, consensus_size*samplecnt.ssc)\n",
    "    pt = poisson.test(c, T, alternative='greater', conf.level = 0.95)\n",
    "    show(paste0(eff, ': ', round(pt$p.value, 3), '   ', 'Rate ratio: ', pt$estimate))\n",
    "    cnt_mm <- sum(df.mm[,eff])\n",
    "    cnt_ssc <- sum(df.ss[,eff])\n",
    "    df_rr <- rbind(df_rr, c(eff, round(pt$estimate,3), round(pt$p.value, 3), \"MM\", \"SSC\", \"All\", pt$conf.int, cnt_mm, cnt_ssc))\n",
    "    df_tmp <- Get_summary(df.melt, eff)\n",
    "\n",
    "}\n",
    "head(df_tmp)\n",
    "df_tmp$class <- factor(df_tmp$category, levels=figure_a)\n",
    "df_tmp$mean <- as.numeric(df_tmp$mean)\n",
    "df_tmp$lower <- as.numeric(df_tmp$lower)\n",
    "df_tmp$upper <- as.numeric(df_tmp$upper)\n",
    "df_tmp$cohort <- factor(df_tmp$cohort, levels = c('SSC', 'MM'), labels = c(\"Control\", \"NTD\"))\n",
    "df_tmp$Gene <- \"All genes\"\n",
    "df_tmp_all <- df_tmp\n",
    "\n",
    "df.melt$variant <- as.numeric(df.melt$variant)\n",
    "\n",
    "colnames(df_rr) <- c(\"category\", \"RR\", \"pvalue\", \"MM\", \"SSC\", \"Gene\", \"Conf\", \"cnt_mm\", \"cnt_ssc\")\n",
    "df_rr\n",
    "\n",
    "\n",
    "df_tmp$category = factor(df_tmp$category, levels = c('All', \"SNV\", \"INDEL\"))\n",
    "p = ggplot() +\n",
    "geom_errorbar(data = df_tmp, aes(x = as.factor(category), \n",
    "                        ymin = lower, \n",
    "                        ymax = upper,group = cohort,color = cohort),\n",
    "              position = position_dodge(0.5), width = 0.2,size = .3) + \n",
    "geom_point(data = df_tmp, aes(category, as.numeric(mean), group = cohort, color = cohort),\n",
    "        size =3,\n",
    "          position =position_dodge(0.5)) + \n",
    "    scale_y_continuous(limits = c(-0.01, 2), sec.axis = sec_axis(~ . / (100000000 / 2) * total_size, name = 'Theoretical Variant Rate (per child)')) +\n",
    "    labs(fill='Cohort', x='', y='Variant rate (x 10e-8)') +\n",
    "    theme(aspect.ratio = 0.8,\n",
    "          legend.position='bottom', \n",
    "          axis.title.y.right = element_text( angle = 90,size = 12),\n",
    "         axis.text.x = element_text(angle = -45, hjust = 0.01,size = 12),\n",
    "          axis.text.y = element_text(size = 12))+ \n",
    "scale_color_manual(values =c(\"#F2BC1B\", \"#73122C\")) + #=c(\"#F2BC1B\", \"#73122C\")) +\n",
    "geom_text(data =subset(df_rr, Gene == 'All'), aes(x = category, y = 1.9,label = paste(paste(\"RR=\",round(as.numeric(RR),2), sep=\"\"), paste(\"p=\",pvalue,sep=\"\" ), sep='\\n'))) + \n",
    "    ggtitle(\"DNMs\") \n",
    "\n",
    "ggsave(\"/projects/ps-gleesonlab8/User/hiyoothere/NTD/5.Analysis/DNM/Burden/Figure_broad/SNV_INDEL.pdf\", plot= p)\n",
    "p"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5a2cfafa",
   "metadata": {},
   "source": [
    "## Calculate burden in all catergories"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "eee3b803",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "figure_all = c('All', 'Nonsynonymous', 'Missense', 'LGD', 'DMisL','DMis',\"DMisL2\", \"DMisM\", \"DMisMC\",\"DMisMC2\", 'DamagingL','DamagingS',\"DamagingL2\", \"DamagingM\", \"DamagingMC\",\"DamagingMC2\", 'OtherMissense','Synonymous')\n",
    "\n",
    "typeof(figure_a)\n",
    "figure_now = figure_a\n",
    "df_all_stat = data.frame()\n",
    "print (percent)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "45b4971b",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "\n",
    "\n",
    "\n",
    "df_tmp <- data.frame()\n",
    "df_rr <- data.frame()\n",
    "\n",
    "paste(\"consensus size is \", consensus[1,])\n",
    "consensus_size = consensus[1,]\n",
    "\n",
    "paste(\"consensus ratio \", consensus[1,]/total_size)\n",
    "\n",
    "paste(\"sample ntd \", samplecnt.mm)\n",
    "paste(\"sample ssc \", samplecnt.ssc)\n",
    "\n",
    "for (eff in figure_all) {\n",
    "    ratio_consen = consensus_size/total_size\n",
    "    mm = round((total_size*sum(df.mm[, eff])/consensus_size),0)\n",
    "    ssc = round((total_size*sum(df.ss[, eff])/consensus_size),0)\n",
    "    show(paste(eff, \"raw count\", sum(df.mm[, eff]), sum(df.ss[, eff])))\n",
    "    show(paste(eff, \"modi count\", mm, ssc))\n",
    "    c = c(mm,ssc)\n",
    "    #c=c(sum(df.mm[, eff]), sum(df.ss[, eff]))\n",
    "    T = c(samplecnt.mm, samplecnt.ssc)\n",
    "    pt = poisson.test(c, T, alternative='greater', conf.level = 0.95)\n",
    "    show(paste0(eff, ': ', round(pt$p.value, 8), '   ', 'Rate ratio: ', pt$estimate))\n",
    "    cnt_mm <- sum(df.mm[,eff])\n",
    "    cnt_ssc <- sum(df.ss[,eff])\n",
    "    df_rr <- rbind(df_rr, c(eff, round(pt$estimate,3), round(pt$p.value, 8), \"MM\", \"SSC\", \"All\", pt$conf.int, cnt_mm, cnt_ssc))\n",
    "    df_tmp <- Get_summary(df.melt, eff)\n",
    "}\n",
    "head(df_tmp)\n",
    "df_tmp$mean <- as.numeric(df_tmp$mean)\n",
    "df_tmp$lower <- as.numeric(df_tmp$lower)\n",
    "df_tmp$upper <- as.numeric(df_tmp$upper)\n",
    "df_tmp$cohort <- factor(df_tmp$cohort, levels = c('SSC', 'MM'), labels = c(\"SSC\", \"MM\"))\n",
    "df_tmp$Gene <- \"All genes\"\n",
    "df_tmp_all <- df_tmp\n",
    "df_all_stat = df_tmp_all\n",
    "df_all_stat_backup = df_all_stat\n",
    "\n",
    "df.melt$variant <- as.numeric(df.melt$variant)\n",
    "head(df.melt)\n",
    "df_tmp\n",
    "options(repr.plot.width=14, repr.plot.height=8)\n",
    "df.mm = df[df$cohort == 'MM', ]\n",
    "df.ss = df[df$cohort == 'SSC', ]\n",
    "head(df.mm)\n",
    "\n",
    "\n",
    "dim(df.mm)\n",
    "dim(df.ss)\n",
    "\n",
    "dim(df.mm$All)\n",
    "dim(df.ss$All)\n",
    "df.melt <- melt(df, id.vars=c('sample', 'family', 'cohort', 'callable'))\n",
    "\n",
    "print (dim(subset(df.mm,df.mm$All == 0)))\n",
    "print (dim(subset(df.ss, df.ss$All == 0)))\n",
    "colnames(df.melt)[5] = 'class'\n",
    "colnames(df.melt)[6] = 'variant'\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "head(df.melt)#ggsave(\"/projects/ps-gleesonlab8/User/hiyoothere/NTD/15.Final/221020_DNM_v9/figure/vplot_all.pdf\", plot= p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "779d9a3f",
   "metadata": {},
   "outputs": [],
   "source": [
    "table(df.melt$class)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8e2b509b",
   "metadata": {},
   "outputs": [],
   "source": [
    "dim(data.all)\n",
    "data.raw <- data.all\n",
    "\n",
    "NTD_LGD <- subset(data.raw, data.raw$batch.x != \"SSC\" &  data.raw$IMPACT == \"HIGH\" )\n",
    "#D-Mis\n",
    "NTD_DMis <- subset(data.raw, data.raw$batch.x != \"SSC\" &  data.raw$MetaSVM_pred== 'D' )\n",
    "#Damaging\n",
    "NTD_Dam <- subset(data.raw, data.raw$batch.x != \"SSC\" & ( data.raw$IMPACT == \"HIGH\" |  data.raw$MetaSVM_pred== 'D') ) \n",
    "#D-Mis-HC\n",
    "NTD_DamMC <- subset(data.raw, data.raw$batch.x != \"SSC\" & ( data.raw$IMPACT == \"HIGH\" |\n",
    "                                                           data.raw$MetaSVM_pred== 'D' |\n",
    "                                        (grepl(\"damaging\",data.raw$PolyPhen) &\n",
    "                                      grepl(\"deleterious\", data.raw$SIFT) &\n",
    "                                          data.raw$CADD_phred >= 20 )) ) \n",
    "#Damaging HC\n",
    "NTD_DamMC2 <- subset(data.raw, data.raw$batch.x != \"SSC\" & ( data.raw$IMPACT == \"HIGH\" |\n",
    "                                                           data.raw$MetaSVM_pred== 'D' |\n",
    "                                        (grepl(\"damaging\",data.raw$PolyPhen) &data.raw$CADD_phred >= 20 ) |\n",
    "                                      (grepl(\"deleterious\", data.raw$SIFT) & data.raw$CADD_phred >= 20 )) ) \n",
    "\n",
    "\n",
    "\n",
    "write.table(NTD_Dam,\"/projects/ps-gleesonlab8/User/hiyoothere/NTD/5.Analysis/DNM/Burden/NTD_DamS.den_input\", sep='\\t', quote = F, row.names = F )\n",
    "write.table(NTD_DamMC2,\"/projects/ps-gleesonlab8/User/hiyoothere/NTD/5.Analysis/DNM/Burden/NTD_DamMC2.den_input\", sep='\\t', quote = F, row.names = F )\n",
    "\n",
    "SSC_Dam <- subset(data.raw, data.raw$batch.x == \"SSC\" & (( data.raw$IMPACT == \"HIGH\" | data.raw$MetaSVM_pred== 'D') )) \n",
    "\n",
    "\n",
    "SSC_DamMC2 <- subset(data.raw, data.raw$batch.x == \"SSC\" & ( data.raw$IMPACT == \"HIGH\" |\n",
    "                                                           data.raw$MetaSVM_pred== 'D' |\n",
    "                                        (grepl(\"damaging\",data.raw$PolyPhen) &data.raw$CADD_phred >= 20 ) |\n",
    "                                      (grepl(\"deleterious\", data.raw$SIFT) & data.raw$CADD_phred >= 20 )) ) \n",
    "\n",
    "\n",
    "write.table(SSC_Dam,\"/projects/ps-gleesonlab8/User/hiyoothere/NTD/5.Analysis/DNM/Burden/SSC_DamS.den_input\", sep='\\t', quote = F, row.names = F )\n",
    "write.table(SSC_DamMC2,\"/projects/ps-gleesonlab8/User/hiyoothere/NTD/5.Analysis/DNM/Burden/SSC_DamMC2.den_input\", sep='\\t', quote = F, row.names = F )\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "29a6cdde",
   "metadata": {},
   "source": [
    "# Intolerant genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 245,
   "id": "5004a5ea",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>1383</li><li>142</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 1383\n",
       "\\item 142\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 1383\n",
       "2. 142\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 1383  142"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>348</li><li>142</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 348\n",
       "\\item 142\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 348\n",
       "2. 142\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 348 142"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dim(data.fil)\n",
    "data.fil.intol = data.fil\n",
    "data.fil.intol = subset(data.fil.intol, as.numeric(data.fil.intol$pLI.y) >= 0.9)\n",
    "dim(data.fil.intol)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 246,
   "id": "79680bb8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>1496</li><li>1</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 1496\n",
       "\\item 1\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 1496\n",
       "2. 1\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 1496    1"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>sample</th><th scope=col>family</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>S12816  </td><td>101  </td></tr>\n",
       "\t<tr><th scope=row>2</th><td>11000.s1</td><td>11000</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>11002.s1</td><td>11002</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>11003.s1</td><td>11003</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>11004.s1</td><td>11004</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>11005.s1</td><td>11005</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 2\n",
       "\\begin{tabular}{r|ll}\n",
       "  & sample & family\\\\\n",
       "  & <fct> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & S12816   & 101  \\\\\n",
       "\t2 & 11000.s1 & 11000\\\\\n",
       "\t3 & 11002.s1 & 11002\\\\\n",
       "\t4 & 11003.s1 & 11003\\\\\n",
       "\t5 & 11004.s1 & 11004\\\\\n",
       "\t6 & 11005.s1 & 11005\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 2\n",
       "\n",
       "| <!--/--> | sample &lt;fct&gt; | family &lt;chr&gt; |\n",
       "|---|---|---|\n",
       "| 1 | S12816   | 101   |\n",
       "| 2 | 11000.s1 | 11000 |\n",
       "| 3 | 11002.s1 | 11002 |\n",
       "| 4 | 11003.s1 | 11003 |\n",
       "| 5 | 11004.s1 | 11004 |\n",
       "| 6 | 11005.s1 | 11005 |\n",
       "\n"
      ],
      "text/plain": [
       "  sample   family\n",
       "1 S12816   101   \n",
       "2 11000.s1 11000 \n",
       "3 11002.s1 11002 \n",
       "4 11003.s1 11003 \n",
       "5 11004.s1 11004 \n",
       "6 11005.s1 11005 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>sample</th><th scope=col>family</th><th scope=col>cohort</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>S12816  </td><td>101  </td><td>MM </td></tr>\n",
       "\t<tr><th scope=row>2</th><td>11000.s1</td><td>11000</td><td>SSC</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>11002.s1</td><td>11002</td><td>SSC</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>11003.s1</td><td>11003</td><td>SSC</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>11004.s1</td><td>11004</td><td>SSC</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>11005.s1</td><td>11005</td><td>SSC</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 3\n",
       "\\begin{tabular}{r|lll}\n",
       "  & sample & family & cohort\\\\\n",
       "  & <fct> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & S12816   & 101   & MM \\\\\n",
       "\t2 & 11000.s1 & 11000 & SSC\\\\\n",
       "\t3 & 11002.s1 & 11002 & SSC\\\\\n",
       "\t4 & 11003.s1 & 11003 & SSC\\\\\n",
       "\t5 & 11004.s1 & 11004 & SSC\\\\\n",
       "\t6 & 11005.s1 & 11005 & SSC\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 3\n",
       "\n",
       "| <!--/--> | sample &lt;fct&gt; | family &lt;chr&gt; | cohort &lt;chr&gt; |\n",
       "|---|---|---|---|\n",
       "| 1 | S12816   | 101   | MM  |\n",
       "| 2 | 11000.s1 | 11000 | SSC |\n",
       "| 3 | 11002.s1 | 11002 | SSC |\n",
       "| 4 | 11003.s1 | 11003 | SSC |\n",
       "| 5 | 11004.s1 | 11004 | SSC |\n",
       "| 6 | 11005.s1 | 11005 | SSC |\n",
       "\n"
      ],
      "text/plain": [
       "  sample   family cohort\n",
       "1 S12816   101    MM    \n",
       "2 11000.s1 11000  SSC   \n",
       "3 11002.s1 11002  SSC   \n",
       "4 11003.s1 11003  SSC   \n",
       "5 11004.s1 11004  SSC   \n",
       "6 11005.s1 11005  SSC   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>1496</li><li>3</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 1496\n",
       "\\item 3\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 1496\n",
       "2. 3\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 1496    3"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>1496</li><li>4</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 1496\n",
       "\\item 4\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 1496\n",
       "2. 4\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 1496    4"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>sample</th><th scope=col>family</th><th scope=col>cohort</th><th scope=col>callable</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>S12816  </td><td>101  </td><td>MM </td><td>65162766</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>11000.s1</td><td>11000</td><td>SSC</td><td>47881923</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>11002.s1</td><td>11002</td><td>SSC</td><td>55431363</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>11003.s1</td><td>11003</td><td>SSC</td><td>62732391</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>11004.s1</td><td>11004</td><td>SSC</td><td>64331070</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>11005.s1</td><td>11005</td><td>SSC</td><td>53838822</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 4\n",
       "\\begin{tabular}{r|llll}\n",
       "  & sample & family & cohort & callable\\\\\n",
       "  & <fct> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & S12816   & 101   & MM  & 65162766\\\\\n",
       "\t2 & 11000.s1 & 11000 & SSC & 47881923\\\\\n",
       "\t3 & 11002.s1 & 11002 & SSC & 55431363\\\\\n",
       "\t4 & 11003.s1 & 11003 & SSC & 62732391\\\\\n",
       "\t5 & 11004.s1 & 11004 & SSC & 64331070\\\\\n",
       "\t6 & 11005.s1 & 11005 & SSC & 53838822\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 4\n",
       "\n",
       "| <!--/--> | sample &lt;fct&gt; | family &lt;chr&gt; | cohort &lt;chr&gt; | callable &lt;chr&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 1 | S12816   | 101   | MM  | 65162766 |\n",
       "| 2 | 11000.s1 | 11000 | SSC | 47881923 |\n",
       "| 3 | 11002.s1 | 11002 | SSC | 55431363 |\n",
       "| 4 | 11003.s1 | 11003 | SSC | 62732391 |\n",
       "| 5 | 11004.s1 | 11004 | SSC | 64331070 |\n",
       "| 6 | 11005.s1 | 11005 | SSC | 53838822 |\n",
       "\n"
      ],
      "text/plain": [
       "  sample   family cohort callable\n",
       "1 S12816   101    MM     65162766\n",
       "2 11000.s1 11000  SSC    47881923\n",
       "3 11002.s1 11002  SSC    55431363\n",
       "4 11003.s1 11003  SSC    62732391\n",
       "5 11004.s1 11004  SSC    64331070\n",
       "6 11005.s1 11005  SSC    53838822"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "'total probands:'"
      ],
      "text/latex": [
       "'total probands:'"
      ],
      "text/markdown": [
       "'total probands:'"
      ],
      "text/plain": [
       "[1] \"total probands:\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "772"
      ],
      "text/latex": [
       "772"
      ],
      "text/markdown": [
       "772"
      ],
      "text/plain": [
       "[1] 772"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "724"
      ],
      "text/latex": [
       "724"
      ],
      "text/markdown": [
       "724"
      ],
      "text/plain": [
       "[1] 724"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Make new df for intolerant genes\n",
    "#With 771 and 723 trios (1492)\n",
    "df = data.frame(sample = factor(ped.pro$sample)) #, levels=levels(ped.pro$sample)))\n",
    "dim(df)\n",
    "#Add familhy\n",
    "df = merge(df, ped.pro[, c('sample', 'family')], by.x= 'sample',by.y = 'sample', sort = F, all.x = TRUE)\n",
    "head(df)\n",
    "#Add Cohort\n",
    "df$cohort = ifelse(grepl('s', df$sample), 'SSC', 'MM')\n",
    "head(df)\n",
    "dim(df)\n",
    "#Add Callable region\n",
    "df = merge(df, callable[, c('sample', 'callable')], by= 'sample', sort = F, all.x=TRUE)\n",
    "df$callable = ifelse(is.na(df$callable), 'NA', df$callable)\n",
    "dim(df)\n",
    "head(df)\n",
    "#Add FA information\n",
    "#df = merge(df, fa, by= 'family', sort = F, all.x=TRUE)\n",
    "#df$FA = ifelse(is.na(df$FA), 'NA', df$FA)\n",
    "#df <- df %>% distinct()\n",
    "\"total probands:\"\n",
    "#dim(df)\n",
    "\n",
    "samplecnt.mm = as.numeric(table(df$cohort)[1])\n",
    "samplecnt.mm\n",
    "samplecnt.ssc = as.numeric(table(df$cohort)[2])\n",
    "samplecnt.ssc\n",
    "\n",
    "colnames(data.fil.intol)[30] = \"term\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 247,
   "id": "279666e1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>sample</th><th scope=col>Inframe</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>12683.s1  </td><td>1</td></tr>\n",
       "\t<tr><td>7035178644</td><td>1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 2\n",
       "\\begin{tabular}{ll}\n",
       " sample & Inframe\\\\\n",
       " <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t 12683.s1   & 1\\\\\n",
       "\t 7035178644 & 1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 2\n",
       "\n",
       "| sample &lt;fct&gt; | Inframe &lt;int&gt; |\n",
       "|---|---|\n",
       "| 12683.s1   | 1 |\n",
       "| 7035178644 | 1 |\n",
       "\n"
      ],
      "text/plain": [
       "  sample     Inframe\n",
       "1 12683.s1   1      \n",
       "2 7035178644 1      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "Inframe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 248,
   "id": "8fbb31cf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 32</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>sample</th><th scope=col>family</th><th scope=col>cohort</th><th scope=col>callable</th><th scope=col>All</th><th scope=col>Coding</th><th scope=col>Noncoding</th><th scope=col>SNV</th><th scope=col>INDEL</th><th scope=col>INDEL_Coding</th><th scope=col>INDEL_Noncoding</th><th scope=col>LGD</th><th scope=col>LGD_FS</th><th scope=col>LGD_SP_A</th><th scope=col>LGD_SP_D</th><th scope=col>LGD_STOP</th><th scope=col>DMis</th><th scope=col>DamagingS</th><th scope=col>DMisL</th><th scope=col>DamagingL</th><th scope=col>DMisM</th><th scope=col>DamagingM</th><th scope=col>DMisL2</th><th scope=col>DamagingL2</th><th scope=col>DMisMC</th><th scope=col>DamagingMC</th><th scope=col>DMisMC2</th><th scope=col>DamagingMC2</th><th scope=col>Synonymous</th><th scope=col>Nonsynonymous</th><th scope=col>Missense</th><th scope=col>OtherMissense</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>S12816  </td><td>101  </td><td>MM </td><td>65162766</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>11000.s1</td><td>11000</td><td>SSC</td><td>47881923</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>11002.s1</td><td>11002</td><td>SSC</td><td>55431363</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>11003.s1</td><td>11003</td><td>SSC</td><td>62732391</td><td>1</td><td>0</td><td>1</td><td>1</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>11004.s1</td><td>11004</td><td>SSC</td><td>64331070</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>11005.s1</td><td>11005</td><td>SSC</td><td>53838822</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 32\n",
       "\\begin{tabular}{r|llllllllllllllllllllllllllllllll}\n",
       "  & sample & family & cohort & callable & All & Coding & Noncoding & SNV & INDEL & INDEL\\_Coding & INDEL\\_Noncoding & LGD & LGD\\_FS & LGD\\_SP\\_A & LGD\\_SP\\_D & LGD\\_STOP & DMis & DamagingS & DMisL & DamagingL & DMisM & DamagingM & DMisL2 & DamagingL2 & DMisMC & DamagingMC & DMisMC2 & DamagingMC2 & Synonymous & Nonsynonymous & Missense & OtherMissense\\\\\n",
       "  & <fct> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & S12816   & 101   & MM  & 65162766 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\t2 & 11000.s1 & 11000 & SSC & 47881923 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\t3 & 11002.s1 & 11002 & SSC & 55431363 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\t4 & 11003.s1 & 11003 & SSC & 62732391 & 1 & 0 & 1 & 1 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\t5 & 11004.s1 & 11004 & SSC & 64331070 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\t6 & 11005.s1 & 11005 & SSC & 53838822 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 32\n",
       "\n",
       "| <!--/--> | sample &lt;fct&gt; | family &lt;chr&gt; | cohort &lt;chr&gt; | callable &lt;chr&gt; | All &lt;dbl&gt; | Coding &lt;dbl&gt; | Noncoding &lt;dbl&gt; | SNV &lt;dbl&gt; | INDEL &lt;dbl&gt; | INDEL_Coding &lt;dbl&gt; | INDEL_Noncoding &lt;dbl&gt; | LGD &lt;dbl&gt; | LGD_FS &lt;dbl&gt; | LGD_SP_A &lt;dbl&gt; | LGD_SP_D &lt;dbl&gt; | LGD_STOP &lt;dbl&gt; | DMis &lt;dbl&gt; | DamagingS &lt;dbl&gt; | DMisL &lt;dbl&gt; | DamagingL &lt;dbl&gt; | DMisM &lt;dbl&gt; | DamagingM &lt;dbl&gt; | DMisL2 &lt;dbl&gt; | DamagingL2 &lt;dbl&gt; | DMisMC &lt;dbl&gt; | DamagingMC &lt;dbl&gt; | DMisMC2 &lt;dbl&gt; | DamagingMC2 &lt;dbl&gt; | Synonymous &lt;dbl&gt; | Nonsynonymous &lt;dbl&gt; | Missense &lt;dbl&gt; | OtherMissense &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | S12816   | 101   | MM  | 65162766 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "| 2 | 11000.s1 | 11000 | SSC | 47881923 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "| 3 | 11002.s1 | 11002 | SSC | 55431363 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "| 4 | 11003.s1 | 11003 | SSC | 62732391 | 1 | 0 | 1 | 1 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "| 5 | 11004.s1 | 11004 | SSC | 64331070 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "| 6 | 11005.s1 | 11005 | SSC | 53838822 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "  sample   family cohort callable All Coding Noncoding SNV INDEL INDEL_Coding\n",
       "1 S12816   101    MM     65162766 0   0      0         0   0     0           \n",
       "2 11000.s1 11000  SSC    47881923 0   0      0         0   0     0           \n",
       "3 11002.s1 11002  SSC    55431363 0   0      0         0   0     0           \n",
       "4 11003.s1 11003  SSC    62732391 1   0      1         1   0     0           \n",
       "5 11004.s1 11004  SSC    64331070 0   0      0         0   0     0           \n",
       "6 11005.s1 11005  SSC    53838822 0   0      0         0   0     0           \n",
       "  INDEL_Noncoding LGD LGD_FS LGD_SP_A LGD_SP_D LGD_STOP DMis DamagingS DMisL\n",
       "1 0               0   0      0        0        0        0    0         0    \n",
       "2 0               0   0      0        0        0        0    0         0    \n",
       "3 0               0   0      0        0        0        0    0         0    \n",
       "4 0               0   0      0        0        0        0    0         0    \n",
       "5 0               0   0      0        0        0        0    0         0    \n",
       "6 0               0   0      0        0        0        0    0         0    \n",
       "  DamagingL DMisM DamagingM DMisL2 DamagingL2 DMisMC DamagingMC DMisMC2\n",
       "1 0         0     0         0      0          0      0          0      \n",
       "2 0         0     0         0      0          0      0          0      \n",
       "3 0         0     0         0      0          0      0          0      \n",
       "4 0         0     0         0      0          0      0          0      \n",
       "5 0         0     0         0      0          0      0          0      \n",
       "6 0         0     0         0      0          0      0          0      \n",
       "  DamagingMC2 Synonymous Nonsynonymous Missense OtherMissense\n",
       "1 0           0          0             0        0            \n",
       "2 0           0          0             0        0            \n",
       "3 0           0          0             0        0            \n",
       "4 0           0          0             0        0            \n",
       "5 0           0          0             0        0            \n",
       "6 0           0          0             0        0            "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 32</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>sample</th><th scope=col>family</th><th scope=col>cohort</th><th scope=col>callable</th><th scope=col>All</th><th scope=col>Coding</th><th scope=col>Noncoding</th><th scope=col>SNV</th><th scope=col>INDEL</th><th scope=col>INDEL_Coding</th><th scope=col>INDEL_Noncoding</th><th scope=col>LGD</th><th scope=col>LGD_FS</th><th scope=col>LGD_SP_A</th><th scope=col>LGD_SP_D</th><th scope=col>LGD_STOP</th><th scope=col>DMis</th><th scope=col>DamagingS</th><th scope=col>DMisL</th><th scope=col>DamagingL</th><th scope=col>DMisM</th><th scope=col>DamagingM</th><th scope=col>DMisL2</th><th scope=col>DamagingL2</th><th scope=col>DMisMC</th><th scope=col>DamagingMC</th><th scope=col>DMisMC2</th><th scope=col>DamagingMC2</th><th scope=col>Synonymous</th><th scope=col>Nonsynonymous</th><th scope=col>Missense</th><th scope=col>OtherMissense</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>S12816</td><td>101</td><td>MM</td><td>65162766</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>498</th><td>S13324</td><td>125</td><td>MM</td><td>77840952</td><td>1</td><td>1</td><td>0</td><td>1</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>1</td><td>1</td><td>0</td><td>0</td><td>0</td><td>0</td><td>1</td><td>1</td><td>0</td><td>1</td><td>1</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>688</th><td>S16116</td><td>134</td><td>MM</td><td>88045008</td><td>1</td><td>1</td><td>0</td><td>1</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>1</td><td>1</td><td>1</td><td>1</td><td>1</td><td>1</td><td>1</td><td>1</td><td>1</td><td>1</td><td>0</td><td>1</td><td>1</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>728</th><td>S12819</td><td>138</td><td>MM</td><td>71003984</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>729</th><td>S12168</td><td>191</td><td>MM</td><td>81651983</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>730</th><td>S10546</td><td>20 </td><td>MM</td><td>81251130</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 32\n",
       "\\begin{tabular}{r|llllllllllllllllllllllllllllllll}\n",
       "  & sample & family & cohort & callable & All & Coding & Noncoding & SNV & INDEL & INDEL\\_Coding & INDEL\\_Noncoding & LGD & LGD\\_FS & LGD\\_SP\\_A & LGD\\_SP\\_D & LGD\\_STOP & DMis & DamagingS & DMisL & DamagingL & DMisM & DamagingM & DMisL2 & DamagingL2 & DMisMC & DamagingMC & DMisMC2 & DamagingMC2 & Synonymous & Nonsynonymous & Missense & OtherMissense\\\\\n",
       "  & <fct> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & S12816 & 101 & MM & 65162766 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\t498 & S13324 & 125 & MM & 77840952 & 1 & 1 & 0 & 1 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 1 & 1 & 0 & 0 & 0 & 0 & 1 & 1 & 0 & 1 & 1 & 1\\\\\n",
       "\t688 & S16116 & 134 & MM & 88045008 & 1 & 1 & 0 & 1 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 1 & 1 & 1 & 1 & 1 & 1 & 1 & 1 & 1 & 1 & 0 & 1 & 1 & 1\\\\\n",
       "\t728 & S12819 & 138 & MM & 71003984 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\t729 & S12168 & 191 & MM & 81651983 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\t730 & S10546 & 20  & MM & 81251130 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 32\n",
       "\n",
       "| <!--/--> | sample &lt;fct&gt; | family &lt;chr&gt; | cohort &lt;chr&gt; | callable &lt;chr&gt; | All &lt;dbl&gt; | Coding &lt;dbl&gt; | Noncoding &lt;dbl&gt; | SNV &lt;dbl&gt; | INDEL &lt;dbl&gt; | INDEL_Coding &lt;dbl&gt; | INDEL_Noncoding &lt;dbl&gt; | LGD &lt;dbl&gt; | LGD_FS &lt;dbl&gt; | LGD_SP_A &lt;dbl&gt; | LGD_SP_D &lt;dbl&gt; | LGD_STOP &lt;dbl&gt; | DMis &lt;dbl&gt; | DamagingS &lt;dbl&gt; | DMisL &lt;dbl&gt; | DamagingL &lt;dbl&gt; | DMisM &lt;dbl&gt; | DamagingM &lt;dbl&gt; | DMisL2 &lt;dbl&gt; | DamagingL2 &lt;dbl&gt; | DMisMC &lt;dbl&gt; | DamagingMC &lt;dbl&gt; | DMisMC2 &lt;dbl&gt; | DamagingMC2 &lt;dbl&gt; | Synonymous &lt;dbl&gt; | Nonsynonymous &lt;dbl&gt; | Missense &lt;dbl&gt; | OtherMissense &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | S12816 | 101 | MM | 65162766 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "| 498 | S13324 | 125 | MM | 77840952 | 1 | 1 | 0 | 1 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 1 | 1 | 0 | 0 | 0 | 0 | 1 | 1 | 0 | 1 | 1 | 1 |\n",
       "| 688 | S16116 | 134 | MM | 88045008 | 1 | 1 | 0 | 1 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 1 | 1 | 1 | 1 | 1 | 1 | 1 | 1 | 1 | 1 | 0 | 1 | 1 | 1 |\n",
       "| 728 | S12819 | 138 | MM | 71003984 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "| 729 | S12168 | 191 | MM | 81651983 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "| 730 | S10546 | 20  | MM | 81251130 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "    sample family cohort callable All Coding Noncoding SNV INDEL INDEL_Coding\n",
       "1   S12816 101    MM     65162766 0   0      0         0   0     0           \n",
       "498 S13324 125    MM     77840952 1   1      0         1   0     0           \n",
       "688 S16116 134    MM     88045008 1   1      0         1   0     0           \n",
       "728 S12819 138    MM     71003984 0   0      0         0   0     0           \n",
       "729 S12168 191    MM     81651983 0   0      0         0   0     0           \n",
       "730 S10546 20     MM     81251130 0   0      0         0   0     0           \n",
       "    INDEL_Noncoding LGD LGD_FS LGD_SP_A LGD_SP_D LGD_STOP DMis DamagingS DMisL\n",
       "1   0               0   0      0        0        0        0    0         0    \n",
       "498 0               0   0      0        0        0        0    0         0    \n",
       "688 0               0   0      0        0        0        0    0         1    \n",
       "728 0               0   0      0        0        0        0    0         0    \n",
       "729 0               0   0      0        0        0        0    0         0    \n",
       "730 0               0   0      0        0        0        0    0         0    \n",
       "    DamagingL DMisM DamagingM DMisL2 DamagingL2 DMisMC DamagingMC DMisMC2\n",
       "1   0         0     0         0      0          0      0          0      \n",
       "498 0         1     1         0      0          0      0          1      \n",
       "688 1         1     1         1      1          1      1          1      \n",
       "728 0         0     0         0      0          0      0          0      \n",
       "729 0         0     0         0      0          0      0          0      \n",
       "730 0         0     0         0      0          0      0          0      \n",
       "    DamagingMC2 Synonymous Nonsynonymous Missense OtherMissense\n",
       "1   0           0          0             0        0            \n",
       "498 1           0          1             1        1            \n",
       "688 1           0          1             1        1            \n",
       "728 0           0          0             0        0            \n",
       "729 0           0          0             0        0            \n",
       "730 0           0          0             0        0            "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>772</li><li>32</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 772\n",
       "\\item 32\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 772\n",
       "2. 32\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 772  32"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>724</li><li>32</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 724\n",
       "\\item 32\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 724\n",
       "2. 32\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 724  32"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>sample</th><th scope=col>family</th><th scope=col>cohort</th><th scope=col>callable</th><th scope=col>class</th><th scope=col>variant</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>S12816  </td><td>101  </td><td>MM </td><td>65162766</td><td>All</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>11000.s1</td><td>11000</td><td>SSC</td><td>47881923</td><td>All</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>11002.s1</td><td>11002</td><td>SSC</td><td>55431363</td><td>All</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>11003.s1</td><td>11003</td><td>SSC</td><td>62732391</td><td>All</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>11004.s1</td><td>11004</td><td>SSC</td><td>64331070</td><td>All</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>11005.s1</td><td>11005</td><td>SSC</td><td>53838822</td><td>All</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 6\n",
       "\\begin{tabular}{r|llllll}\n",
       "  & sample & family & cohort & callable & class & variant\\\\\n",
       "  & <fct> & <chr> & <chr> & <chr> & <fct> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & S12816   & 101   & MM  & 65162766 & All & 0\\\\\n",
       "\t2 & 11000.s1 & 11000 & SSC & 47881923 & All & 0\\\\\n",
       "\t3 & 11002.s1 & 11002 & SSC & 55431363 & All & 0\\\\\n",
       "\t4 & 11003.s1 & 11003 & SSC & 62732391 & All & 1\\\\\n",
       "\t5 & 11004.s1 & 11004 & SSC & 64331070 & All & 0\\\\\n",
       "\t6 & 11005.s1 & 11005 & SSC & 53838822 & All & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 6\n",
       "\n",
       "| <!--/--> | sample &lt;fct&gt; | family &lt;chr&gt; | cohort &lt;chr&gt; | callable &lt;chr&gt; | class &lt;fct&gt; | variant &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 1 | S12816   | 101   | MM  | 65162766 | All | 0 |\n",
       "| 2 | 11000.s1 | 11000 | SSC | 47881923 | All | 0 |\n",
       "| 3 | 11002.s1 | 11002 | SSC | 55431363 | All | 0 |\n",
       "| 4 | 11003.s1 | 11003 | SSC | 62732391 | All | 1 |\n",
       "| 5 | 11004.s1 | 11004 | SSC | 64331070 | All | 0 |\n",
       "| 6 | 11005.s1 | 11005 | SSC | 53838822 | All | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "  sample   family cohort callable class variant\n",
       "1 S12816   101    MM     65162766 All   0      \n",
       "2 11000.s1 11000  SSC    47881923 All   0      \n",
       "3 11002.s1 11002  SSC    55431363 All   0      \n",
       "4 11003.s1 11003  SSC    62732391 All   1      \n",
       "5 11004.s1 11004  SSC    64331070 All   0      \n",
       "6 11005.s1 11005  SSC    53838822 All   0      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>1496</li><li>32</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 1496\n",
       "\\item 32\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 1496\n",
       "2. 32\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 1496   32"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 32</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>sample</th><th scope=col>family</th><th scope=col>cohort</th><th scope=col>callable</th><th scope=col>All</th><th scope=col>Coding</th><th scope=col>Noncoding</th><th scope=col>SNV</th><th scope=col>INDEL</th><th scope=col>INDEL_Coding</th><th scope=col>INDEL_Noncoding</th><th scope=col>LGD</th><th scope=col>LGD_FS</th><th scope=col>LGD_SP_A</th><th scope=col>LGD_SP_D</th><th scope=col>LGD_STOP</th><th scope=col>DMis</th><th scope=col>DamagingS</th><th scope=col>DMisL</th><th scope=col>DamagingL</th><th scope=col>DMisM</th><th scope=col>DamagingM</th><th scope=col>DMisL2</th><th scope=col>DamagingL2</th><th scope=col>DMisMC</th><th scope=col>DamagingMC</th><th scope=col>DMisMC2</th><th scope=col>DamagingMC2</th><th scope=col>Synonymous</th><th scope=col>Nonsynonymous</th><th scope=col>Missense</th><th scope=col>OtherMissense</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>S12816  </td><td>101  </td><td>MM </td><td>65162766</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>11000.s1</td><td>11000</td><td>SSC</td><td>47881923</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>11002.s1</td><td>11002</td><td>SSC</td><td>55431363</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>11003.s1</td><td>11003</td><td>SSC</td><td>62732391</td><td>1</td><td>0</td><td>1</td><td>1</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>11004.s1</td><td>11004</td><td>SSC</td><td>64331070</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>11005.s1</td><td>11005</td><td>SSC</td><td>53838822</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 32\n",
       "\\begin{tabular}{r|llllllllllllllllllllllllllllllll}\n",
       "  & sample & family & cohort & callable & All & Coding & Noncoding & SNV & INDEL & INDEL\\_Coding & INDEL\\_Noncoding & LGD & LGD\\_FS & LGD\\_SP\\_A & LGD\\_SP\\_D & LGD\\_STOP & DMis & DamagingS & DMisL & DamagingL & DMisM & DamagingM & DMisL2 & DamagingL2 & DMisMC & DamagingMC & DMisMC2 & DamagingMC2 & Synonymous & Nonsynonymous & Missense & OtherMissense\\\\\n",
       "  & <fct> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & S12816   & 101   & MM  & 65162766 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\t2 & 11000.s1 & 11000 & SSC & 47881923 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\t3 & 11002.s1 & 11002 & SSC & 55431363 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\t4 & 11003.s1 & 11003 & SSC & 62732391 & 1 & 0 & 1 & 1 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\t5 & 11004.s1 & 11004 & SSC & 64331070 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\t6 & 11005.s1 & 11005 & SSC & 53838822 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 32\n",
       "\n",
       "| <!--/--> | sample &lt;fct&gt; | family &lt;chr&gt; | cohort &lt;chr&gt; | callable &lt;chr&gt; | All &lt;dbl&gt; | Coding &lt;dbl&gt; | Noncoding &lt;dbl&gt; | SNV &lt;dbl&gt; | INDEL &lt;dbl&gt; | INDEL_Coding &lt;dbl&gt; | INDEL_Noncoding &lt;dbl&gt; | LGD &lt;dbl&gt; | LGD_FS &lt;dbl&gt; | LGD_SP_A &lt;dbl&gt; | LGD_SP_D &lt;dbl&gt; | LGD_STOP &lt;dbl&gt; | DMis &lt;dbl&gt; | DamagingS &lt;dbl&gt; | DMisL &lt;dbl&gt; | DamagingL &lt;dbl&gt; | DMisM &lt;dbl&gt; | DamagingM &lt;dbl&gt; | DMisL2 &lt;dbl&gt; | DamagingL2 &lt;dbl&gt; | DMisMC &lt;dbl&gt; | DamagingMC &lt;dbl&gt; | DMisMC2 &lt;dbl&gt; | DamagingMC2 &lt;dbl&gt; | Synonymous &lt;dbl&gt; | Nonsynonymous &lt;dbl&gt; | Missense &lt;dbl&gt; | OtherMissense &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | S12816   | 101   | MM  | 65162766 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "| 2 | 11000.s1 | 11000 | SSC | 47881923 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "| 3 | 11002.s1 | 11002 | SSC | 55431363 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "| 4 | 11003.s1 | 11003 | SSC | 62732391 | 1 | 0 | 1 | 1 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "| 5 | 11004.s1 | 11004 | SSC | 64331070 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "| 6 | 11005.s1 | 11005 | SSC | 53838822 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "  sample   family cohort callable All Coding Noncoding SNV INDEL INDEL_Coding\n",
       "1 S12816   101    MM     65162766 0   0      0         0   0     0           \n",
       "2 11000.s1 11000  SSC    47881923 0   0      0         0   0     0           \n",
       "3 11002.s1 11002  SSC    55431363 0   0      0         0   0     0           \n",
       "4 11003.s1 11003  SSC    62732391 1   0      1         1   0     0           \n",
       "5 11004.s1 11004  SSC    64331070 0   0      0         0   0     0           \n",
       "6 11005.s1 11005  SSC    53838822 0   0      0         0   0     0           \n",
       "  INDEL_Noncoding LGD LGD_FS LGD_SP_A LGD_SP_D LGD_STOP DMis DamagingS DMisL\n",
       "1 0               0   0      0        0        0        0    0         0    \n",
       "2 0               0   0      0        0        0        0    0         0    \n",
       "3 0               0   0      0        0        0        0    0         0    \n",
       "4 0               0   0      0        0        0        0    0         0    \n",
       "5 0               0   0      0        0        0        0    0         0    \n",
       "6 0               0   0      0        0        0        0    0         0    \n",
       "  DamagingL DMisM DamagingM DMisL2 DamagingL2 DMisMC DamagingMC DMisMC2\n",
       "1 0         0     0         0      0          0      0          0      \n",
       "2 0         0     0         0      0          0      0          0      \n",
       "3 0         0     0         0      0          0      0          0      \n",
       "4 0         0     0         0      0          0      0          0      \n",
       "5 0         0     0         0      0          0      0          0      \n",
       "6 0         0     0         0      0          0      0          0      \n",
       "  DamagingMC2 Synonymous Nonsynonymous Missense OtherMissense\n",
       "1 0           0          0             0        0            \n",
       "2 0           0          0             0        0            \n",
       "3 0           0          0             0        0            \n",
       "4 0           0          0             0        0            \n",
       "5 0           0          0             0        0            \n",
       "6 0           0          0             0        0            "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "18"
      ],
      "text/latex": [
       "18"
      ],
      "text/markdown": [
       "18"
      ],
      "text/plain": [
       "[1] 18"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "data.fil.backup <- data.fil\n",
    "data.fil <- data.fil.intol\n",
    "\n",
    "All = as.data.frame(table(data.fil$sample))\n",
    "Coding = as.data.frame(table(data.fil[data.fil$term %in% coding_term,]$sample))\n",
    "Noncoding = as.data.frame(table(data.fil[!data.fil$term %in% coding_term,]$sample))\n",
    "SNV = as.data.frame(table(data.fil[data.fil$VARIANT_CLASS == 'SNV' | data.fil$VARIANT_CLASS == 'substitution',]$sample))\n",
    "INDEL= as.data.frame(table(data.fil[data.fil$VARIANT_CLASS == 'insertion' | data.fil$VARIANT_CLASS == 'deletion' | data.fil$VARIANT_CLASS == 'indel',]$sample))\n",
    "#Inframe = as.data.frame(table(data.fil[data.fil$term == 'inframe_deletion' | data.fil$term == 'inframe_insertion',]$sample))\n",
    "INDEL_Coding = as.data.frame(table(data.fil[(data.fil$VARIANT_CLASS == 'insertion' | data.fil$VARIANT_CLASS == 'deletion' | data.fil$VARIANT_CLASS == 'indel') & (data.fil$term %in% coding_term) ,]$sample))\n",
    "INDEL_Noncoding = as.data.frame(table(data.fil[(data.fil$VARIANT_CLASS == 'insertion' | data.fil$VARIANT_CLASS == 'deletion' | data.fil$VARIANT_CLASS == 'indel') & (!data.fil$term %in% coding_term) ,]$sample))\n",
    "\n",
    "\n",
    "LGD = as.data.frame(table(data.fil[data.fil$IMPACT =='HIGH',]$sample))\n",
    "LGD_FS = as.data.frame(table(data.fil[data.fil$term == 'frameshift_variant',]$sample))\n",
    "LGD_SP_A = as.data.frame(table(data.fil[data.fil$term == 'splice_acceptor_variant',]$sample))\n",
    "LGD_SP_D = as.data.frame(table(data.fil[data.fil$term == 'splice_donor_variant',]$sample))\n",
    "LGD_STOP = as.data.frame(table(data.fil[data.fil$term == 'stop_gain',]$sample))\n",
    "\n",
    "\n",
    "\n",
    "DMisL2 = as.data.frame(table(data.fil[((grepl(\"damaging\",data.fil$PolyPhen) & data.fil$CADD_phred >=20 ) | data.fil$MetaSVM_pred == 'D' ),]$sample))\n",
    "DamagingL = as.data.frame(table(data.fil[data.fil$IMPACT =='HIGH' | (data.fil$MetaSVM_pred == 'D') | (grepl(\"damaging\",data.fil$PolyPhen)),]$sample))\n",
    "DamagingL2 = as.data.frame(table(data.fil[data.fil$IMPACT =='HIGH' | \n",
    "                                          (data.fil$MetaSVM_pred == 'D') | ((grepl(\"damaging\",data.fil$PolyPhen)& data.fil$CADD_phred >=20 )),]$sample))\n",
    "\n",
    "DMisM = as.data.frame(table(data.fil[(grepl(\"damaging\",data.fil$PolyPhen) | \n",
    "                                      grepl(\"deleterious\", data.fil$SIFT) |\n",
    "                                      data.fil$MetaSVM_pred == 'D' ),]$sample))\n",
    "DamagingM = as.data.frame(table(data.fil[data.fil$IMPACT =='HIGH' |\n",
    "                                     grepl(\"damaging\",data.fil$PolyPhen) | \n",
    "                                      grepl(\"deleterious\", data.fil$SIFT) |\n",
    "                                      data.fil$MetaSVM_pred == 'D' ,]$sample))\n",
    "DMisMC = as.data.frame(table(data.fil[((grepl(\"damaging\",data.fil$PolyPhen) & \n",
    "                                      grepl(\"deleterious\", data.fil$SIFT) &\n",
    "                                       data.fil$CADD_phred >= 20) |\n",
    "                                      data.fil$MetaSVM_pred == 'D' ),]$sample))\n",
    "DamagingMC = as.data.frame(table(data.fil[data.fil$IMPACT =='HIGH' |\n",
    "                                     (grepl(\"damaging\",data.fil$PolyPhen) &\n",
    "                                      grepl(\"deleterious\", data.fil$SIFT) &\n",
    "                                          data.fil$CADD_phred >= 20 ) |\n",
    "                                      data.fil$MetaSVM_pred == 'D' ,]$sample))\n",
    "DMisMC2 = as.data.frame(table(data.fil[((grepl(\"damaging\",data.fil$PolyPhen) & data.fil$CADD_phred >= 20) | \n",
    "                                        \n",
    "                                      (grepl(\"deleterious\", data.fil$SIFT) & data.fil$CADD_phred >= 20) |\n",
    "                                      data.fil$MetaSVM_pred == 'D' ),]$sample))\n",
    "\n",
    "DamagingMC2 = as.data.frame(table(data.fil[((grepl(\"damaging\",data.fil$PolyPhen) & data.fil$CADD_phred >= 20) | \n",
    "                                        (data.fil$IMPACT =='HIGH') | \n",
    "                                      (grepl(\"deleterious\", data.fil$SIFT) & data.fil$CADD_phred >= 20) |\n",
    "                                      (data.fil$MetaSVM_pred == 'D') ),]$sample))\n",
    "\n",
    "\n",
    "DMisS = as.data.frame(table(data.fil[data.fil$MetaSVM_pred == 'D',]$sample))\n",
    "DamagingS = as.data.frame(table(data.fil[data.fil$IMPACT =='HIGH' | (data.fil$MetaSVM_pred == 'D' ),]$sample))\n",
    "DMisL = as.data.frame(table(data.fil[(grepl(\"damaging\",data.fil$PolyPhen) | data.fil$MetaSVM_pred == 'D' ),]$sample))\n",
    "DamagingL = as.data.frame(table(data.fil[data.fil$IMPACT =='HIGH' | (data.fil$MetaSVM_pred == 'D') | (grepl(\"damaging\",data.fil$PolyPhen)),]$sample))\n",
    "\n",
    "\n",
    "Synonymous = as.data.frame(table(data.fil[data.fil$term == 'synonymous_variant',]$sample))\n",
    "Nonsynonymous = as.data.frame(table(data.fil[data.fil$term != 'synonymous_variant' &data.fil$term %in% coding_term ,]$sample))\n",
    "Missense = as.data.frame(table(data.fil[data.fil$term == 'missense_variant',]$sample))\n",
    "OtherMissense = as.data.frame(table(data.fil[(data.fil$term == 'missense_variant') & (data.fil$MetaSVM_pred != 'D'),]$sample))\n",
    "\n",
    "\n",
    "\n",
    "colnames(All) = c(\"sample\", \"All\")\n",
    "colnames(Coding) = c(\"sample\", \"Coding\")\n",
    "colnames(Noncoding) = c(\"sample\", \"Noncoding\")\n",
    "colnames(SNV) = c(\"sample\", \"SNV\")\n",
    "colnames(INDEL) = c(\"sample\", \"INDEL\")\n",
    "#colnames(Inframe) = c(\"sample\", \"Inframe\")\n",
    "colnames(INDEL_Coding) = c(\"sample\", \"INDEL_Coding\")\n",
    "colnames(INDEL_Noncoding) = c(\"sample\", \"INDEL_Noncoding\")\n",
    "\n",
    "colnames(LGD) = c(\"sample\", \"LGD\")\n",
    "colnames(LGD_FS) = c(\"sample\", \"LGD_FS\")\n",
    "colnames(LGD_SP_A) = c(\"sample\", \"LGD_SP_A\")\n",
    "colnames(LGD_SP_D) = c(\"sample\", \"LGD_SP_D\")\n",
    "colnames(LGD_STOP) = c(\"sample\", \"LGD_STOP\")\n",
    "\n",
    "colnames(DMisS) = c(\"sample\", \"DMis\")\n",
    "colnames(DamagingS) = c(\"sample\", \"DamagingS\")\n",
    "colnames(DMisL) = c(\"sample\", \"DMisL\")\n",
    "colnames(DamagingL) = c(\"sample\", \"DamagingL\")\n",
    "colnames(DMisM) = c(\"sample\", \"DMisM\")\n",
    "colnames(DamagingM) = c(\"sample\", \"DamagingM\")\n",
    "colnames(DMisL2) = c(\"sample\", \"DMisL2\")\n",
    "colnames(DamagingL2) = c(\"sample\", \"DamagingL2\")\n",
    "colnames(DMisMC) = c(\"sample\", \"DMisMC\")\n",
    "colnames(DamagingMC) = c(\"sample\", \"DamagingMC\")\n",
    "colnames(DMisMC2) = c(\"sample\", \"DMisMC2\")\n",
    "colnames(DamagingMC2) = c(\"sample\", \"DamagingMC2\")\n",
    "                                          \n",
    "colnames(Synonymous) = c(\"sample\", \"Synonymous\")\n",
    "colnames(Nonsynonymous) = c(\"sample\", \"Nonsynonymous\")\n",
    "colnames(Missense) = c(\"sample\", \"Missense\")\n",
    "colnames(OtherMissense) = c(\"sample\", \"OtherMissense\")\n",
    "\n",
    "df  = left_join(df, All, by = 'sample')\n",
    "df = left_join(df, Coding, by = 'sample')\n",
    "df = left_join(df, Noncoding, by = 'sample')\n",
    "df = left_join(df, SNV, by = 'sample')\n",
    "df = left_join(df, INDEL, by = 'sample')\n",
    "#df = left_join(df, Inframe, by = 'sample')\n",
    "df = left_join(df, INDEL_Coding, by = 'sample')\n",
    "df = left_join(df, INDEL_Noncoding, by = 'sample')\n",
    "\n",
    "\n",
    "df = left_join(df, LGD, by = 'sample')\n",
    "df = left_join(df, LGD_FS, by = 'sample')\n",
    "df = left_join(df, LGD_SP_A, by = 'sample')\n",
    "df = left_join(df, LGD_SP_D, by = 'sample')\n",
    "df = left_join(df, LGD_STOP, by = 'sample')\n",
    "\n",
    "df = left_join(df, DMisS, by = 'sample')\n",
    "df = left_join(df, DamagingS, by = 'sample')\n",
    "df = left_join(df, DMisL, by = 'sample')\n",
    "df = left_join(df, DamagingL, by = 'sample')\n",
    "df = left_join(df, DMisM, by = 'sample')\n",
    "df = left_join(df, DamagingM, by = 'sample')\n",
    "df = left_join(df, DMisL2, by = 'sample')\n",
    "df = left_join(df, DamagingL2, by = 'sample')\n",
    "df = left_join(df, DMisMC, by = 'sample')\n",
    "df = left_join(df, DamagingMC, by = 'sample')\n",
    "df = left_join(df, DMisMC2, by = 'sample')\n",
    "df = left_join(df, DamagingMC2, by = 'sample')\n",
    "\n",
    "df = left_join(df, Synonymous, by = 'sample')\n",
    "df = left_join(df, Nonsynonymous, by = 'sample')\n",
    "df = left_join(df, Missense, by = 'sample')\n",
    "df = left_join(df, OtherMissense, by = 'sample')\n",
    "df[is.na(df)] <- 0\n",
    "\n",
    "\n",
    "head(df)\n",
    "df.mm = df[df$cohort == 'MM', ]\n",
    "df.ss = df[df$cohort == 'SSC', ]\n",
    "head(df.mm)\n",
    "\n",
    "dim(df.mm)\n",
    "dim(df.ss)\n",
    "\n",
    "#Make melted one\n",
    "df.melt <- melt(df, id.vars=c('sample', 'family', 'cohort', 'callable'))\n",
    "\n",
    "\n",
    "colnames(df.melt)[5] = 'class'\n",
    "colnames(df.melt)[6] = 'variant'\n",
    "head(df.melt)\n",
    "dim(df)\n",
    "head(df)\n",
    "df.mm = df[df$cohort == 'MM', ]\n",
    "df.ss = df[df$cohort == 'SSC', ]\n",
    "\n",
    "sum(df.mm$LGD,na.rm = T)\n",
    "df.mm[is.na(df.mm)] <- 0\n",
    "df.ss[is.na(df.ss)] <- 0\n",
    "df.melt[is.na(df.melt)] <- 0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 249,
   "id": "c141a138",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(Intercept)         All \n",
      "   1.023969    1.191348 \n",
      "  (Intercept) Nonsynonymous \n",
      "     1.027299      1.981618 \n",
      "(Intercept)    Missense \n",
      "    1.03709     1.88655 \n",
      "(Intercept)         LGD \n",
      "   1.055991    1.755413 \n",
      "(Intercept)       DMisL \n",
      "   1.044881    2.349115 \n",
      "(Intercept)        DMis \n",
      "   1.055325    8.528178 \n",
      "(Intercept)      DMisL2 \n",
      "   1.044881    2.349115 \n",
      "(Intercept)       DMisM \n",
      "   1.045007    2.134693 \n",
      "(Intercept)      DMisMC \n",
      "   1.041899    3.119303 \n",
      "(Intercept)   DamagingL \n",
      "   1.034466    2.092356 \n",
      "(Intercept)   DamagingS \n",
      "   1.045398    2.391978 \n",
      "(Intercept)  DamagingL2 \n",
      "   1.034466    2.092356 \n",
      "(Intercept)   DamagingM \n",
      "   1.033088    2.087804 \n",
      "(Intercept)  DamagingMC \n",
      "   1.031701    2.395067 \n",
      "  (Intercept) OtherMissense \n",
      "     1.048433      1.560770 \n",
      "(Intercept)  Synonymous \n",
      "   1.064195    1.064966 \n"
     ]
    },
    {
     "ename": "ERROR",
     "evalue": "Error in table.write(df_odds, \"\"): could not find function \"table.write\"\n",
     "output_type": "error",
     "traceback": [
      "Error in table.write(df_odds, \"\"): could not find function \"table.write\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "df$MM = ifelse(df$cohort == 'MM', 1, 0)\n",
    "\n",
    "tol_type = \"intolerant genes\"\n",
    "df_odds_intol = data.frame()\n",
    "\n",
    "for (eff in figure_all) {\n",
    "    state <- paste(\"MM ~ \", eff)\n",
    "    x <- glm(as.formula(state),\n",
    "             family = binomial,\n",
    "             data = df)\n",
    "    suppressMessages(a<- confint(x))\n",
    "\n",
    "    coef <- exp(coef(x))\n",
    "    print (coef)[2]\n",
    "    result =  c(eff, coef[2], exp(a)[2,1], exp(a)[2,2], summary(x)$coefficient[2,4], tol_type) \n",
    "    #print (result)\n",
    "    df_odds_intol = rbind(df_odds_intol, result)\n",
    "}\n",
    "\n",
    "#df_odds\n",
    "colnames(df_odds_intol) <- c(\"Type\",\"OR\", \"lower\", \"upper\", \"p\", \"tol\")\n",
    "\n",
    "df_odds = rbind(df_odds, df_odds_intol)\n",
    "\n",
    "table.write(df_odds, \"\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 250,
   "id": "99518e4e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 195\n",
      "[1] 153\n",
      "[1] 82\n",
      "[1] 50\n",
      "[1] 113\n",
      "[1] 103\n",
      "[1] 25\n",
      "[1] 22\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "47"
      ],
      "text/latex": [
       "47"
      ],
      "text/markdown": [
       "47"
      ],
      "text/plain": [
       "[1] 47"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 29\n",
      "[1] 13\n",
      "[1] 26\n",
      "[1] 8\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'frameshift_variant'</li><li>'inframe_deletion'</li><li>'inframe_insertion'</li><li>'synonymous_variant'</li><li>'missense_variant'</li><li>'splice_acceptor_variant'</li><li>'splice_donor_variant'</li><li>'start_lost'</li><li>'stop_gained'</li><li>'stop_lost'</li><li>'stop_retained_variant'</li><li>'protein_altering_variant'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'frameshift\\_variant'\n",
       "\\item 'inframe\\_deletion'\n",
       "\\item 'inframe\\_insertion'\n",
       "\\item 'synonymous\\_variant'\n",
       "\\item 'missense\\_variant'\n",
       "\\item 'splice\\_acceptor\\_variant'\n",
       "\\item 'splice\\_donor\\_variant'\n",
       "\\item 'start\\_lost'\n",
       "\\item 'stop\\_gained'\n",
       "\\item 'stop\\_lost'\n",
       "\\item 'stop\\_retained\\_variant'\n",
       "\\item 'protein\\_altering\\_variant'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'frameshift_variant'\n",
       "2. 'inframe_deletion'\n",
       "3. 'inframe_insertion'\n",
       "4. 'synonymous_variant'\n",
       "5. 'missense_variant'\n",
       "6. 'splice_acceptor_variant'\n",
       "7. 'splice_donor_variant'\n",
       "8. 'start_lost'\n",
       "9. 'stop_gained'\n",
       "10. 'stop_lost'\n",
       "11. 'stop_retained_variant'\n",
       "12. 'protein_altering_variant'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"frameshift_variant\"       \"inframe_deletion\"        \n",
       " [3] \"inframe_insertion\"        \"synonymous_variant\"      \n",
       " [5] \"missense_variant\"         \"splice_acceptor_variant\" \n",
       " [7] \"splice_donor_variant\"     \"start_lost\"              \n",
       " [9] \"stop_gained\"              \"stop_lost\"               \n",
       "[11] \"stop_retained_variant\"    \"protein_altering_variant\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "\n",
    "print (sum(df.mm$All))\n",
    "print (sum(df.ss$All))\n",
    "\n",
    "print (sum(df.mm$Coding))\n",
    "print (sum(df.ss$Coding))\n",
    "\n",
    "print (sum(df.mm$Noncoding))\n",
    "print (sum(df.ss$Noncoding))\n",
    "\n",
    "print (sum(df.mm$Synonymous))\n",
    "print (sum(df.ss$Synonymous))\n",
    "sum(df$Synonymous)\n",
    "\n",
    "print (sum(df.mm$DMisM))\n",
    "print (sum(df.ss$DMisM))\n",
    "\n",
    "print (sum(df.mm$DMisMC))\n",
    "print (sum(df.ss$DMisMC))\n",
    "\n",
    "coding_term"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 251,
   "id": "dce45179",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"All: 0.02   Rate ratio: 1.19496907905733\"\n",
      "\n",
      "\tWilcoxon rank sum test with continuity correction\n",
      "\n",
      "data:  l_rate_mm and l_rate_ssc\n",
      "W = 289484, p-value = 0.08683\n",
      "alternative hypothesis: true location shift is not equal to 0\n",
      "\n",
      "[1] \"Nonsynonymous: 0   Rate ratio: 1.91732872769142\"\n",
      "\n",
      "\tWilcoxon rank sum test with continuity correction\n",
      "\n",
      "data:  l_rate_mm and l_rate_ssc\n",
      "W = 289290, p-value = 0.003339\n",
      "alternative hypothesis: true location shift is not equal to 0\n",
      "\n",
      "[1] \"Missense: 0.001   Rate ratio: 1.85030107828035\"\n",
      "\n",
      "\tWilcoxon rank sum test with continuity correction\n",
      "\n",
      "data:  l_rate_mm and l_rate_ssc\n",
      "W = 286876, p-value = 0.01389\n",
      "alternative hypothesis: true location shift is not equal to 0\n",
      "\n",
      "[1] \"LGD: 0.039   Rate ratio: 1.81312607944732\"\n",
      "\n",
      "\tWilcoxon rank sum test with continuity correction\n",
      "\n",
      "data:  l_rate_mm and l_rate_ssc\n",
      "W = 282526, p-value = 0.09873\n",
      "alternative hypothesis: true location shift is not equal to 0\n",
      "\n",
      "[1] \"DMisL: 0.002   Rate ratio: 2.29245826137018\"\n",
      "\n",
      "\tWilcoxon rank sum test with continuity correction\n",
      "\n",
      "data:  l_rate_mm and l_rate_ssc\n",
      "W = 284992, p-value = 0.01514\n",
      "alternative hypothesis: true location shift is not equal to 0\n",
      "\n",
      "[1] \"DMis: 0.002   Rate ratio: 7.03367875647668\"\n",
      "\n",
      "\tWilcoxon rank sum test with continuity correction\n",
      "\n",
      "data:  l_rate_mm and l_rate_ssc\n",
      "W = 282336, p-value = 0.01483\n",
      "alternative hypothesis: true location shift is not equal to 0\n",
      "\n",
      "[1] \"DMisL2: 0.002   Rate ratio: 2.29245826137018\"\n",
      "\n",
      "\tWilcoxon rank sum test with continuity correction\n",
      "\n",
      "data:  l_rate_mm and l_rate_ssc\n",
      "W = 284992, p-value = 0.01514\n",
      "alternative hypothesis: true location shift is not equal to 0\n",
      "\n",
      "[1] \"DMisM: 0.003   Rate ratio: 2.09893905748828\"\n",
      "\n",
      "\tWilcoxon rank sum test with continuity correction\n",
      "\n",
      "data:  l_rate_mm and l_rate_ssc\n",
      "W = 284944, p-value = 0.02182\n",
      "alternative hypothesis: true location shift is not equal to 0\n",
      "\n",
      "[1] \"DMisMC: 0   Rate ratio: 3.02989238740534\"\n",
      "\n",
      "\tWilcoxon rank sum test with continuity correction\n",
      "\n",
      "data:  l_rate_mm and l_rate_ssc\n",
      "W = 285788, p-value = 0.003349\n",
      "alternative hypothesis: true location shift is not equal to 0\n",
      "\n",
      "[1] \"DamagingL: 0   Rate ratio: 2.13941062176166\"\n",
      "\n",
      "\tWilcoxon rank sum test with continuity correction\n",
      "\n",
      "data:  l_rate_mm and l_rate_ssc\n",
      "W = 288046, p-value = 0.003137\n",
      "alternative hypothesis: true location shift is not equal to 0\n",
      "\n",
      "[1] \"DamagingS: 0   Rate ratio: 2.57901554404145\"\n",
      "\n",
      "\tWilcoxon rank sum test with continuity correction\n",
      "\n",
      "data:  l_rate_mm and l_rate_ssc\n",
      "W = 285394, p-value = 0.006688\n",
      "alternative hypothesis: true location shift is not equal to 0\n",
      "\n",
      "[1] \"DamagingL2: 0   Rate ratio: 2.13941062176166\"\n",
      "\n",
      "\tWilcoxon rank sum test with continuity correction\n",
      "\n",
      "data:  l_rate_mm and l_rate_ssc\n",
      "W = 288046, p-value = 0.003137\n",
      "alternative hypothesis: true location shift is not equal to 0\n",
      "\n",
      "[1] \"DamagingM: 0   Rate ratio: 2.09631209996952\"\n",
      "\n",
      "\tWilcoxon rank sum test with continuity correction\n",
      "\n",
      "data:  l_rate_mm and l_rate_ssc\n",
      "W = 288383, p-value = 0.002676\n",
      "alternative hypothesis: true location shift is not equal to 0\n",
      "\n",
      "[1] \"DamagingMC: 0   Rate ratio: 2.3780532938564\"\n",
      "\n",
      "\tWilcoxon rank sum test with continuity correction\n",
      "\n",
      "data:  l_rate_mm and l_rate_ssc\n",
      "W = 288840, p-value = 0.0008665\n",
      "alternative hypothesis: true location shift is not equal to 0\n",
      "\n",
      "[1] \"OtherMissense: 0.031   Rate ratio: 1.51093839953944\"\n",
      "\n",
      "\tWilcoxon rank sum test with continuity correction\n",
      "\n",
      "data:  l_rate_mm and l_rate_ssc\n",
      "W = 284004, p-value = 0.104\n",
      "alternative hypothesis: true location shift is not equal to 0\n",
      "\n",
      "[1] \"Synonymous: 0.431   Rate ratio: 1.06807714450201\"\n",
      "\n",
      "\tWilcoxon rank sum test with continuity correction\n",
      "\n",
      "data:  l_rate_mm and l_rate_ssc\n",
      "W = 279671, p-value = 0.9341\n",
      "alternative hypothesis: true location shift is not equal to 0\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 32 × 9</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>cohort</th><th scope=col>category</th><th scope=col>mean</th><th scope=col>lower</th><th scope=col>upper</th><th scope=col>mean_diff</th><th scope=col>mean_diff_lower</th><th scope=col>mean_diff_upper</th><th scope=col>Gene</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>MM </td><td>All          </td><td>0.34550887</td><td> 0.297063590</td><td>0.39395415</td><td>NA                </td><td>NA                 </td><td>NA                </td><td>Constrained genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>All          </td><td>0.28906450</td><td> 0.242016333</td><td>0.33611266</td><td>NA                </td><td>NA                 </td><td>NA                </td><td>Constrained genes</td></tr>\n",
       "\t<tr><td>MM </td><td>Nonsynonymous</td><td>0.10099490</td><td> 0.075706682</td><td>0.12628312</td><td>0.0480942084857547</td><td>0.0163324233344647 </td><td>0.0798559936370446</td><td>Constrained genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>Nonsynonymous</td><td>0.05290069</td><td> 0.033643475</td><td>0.07215791</td><td>0.0480942084857547</td><td>0.0163324233344647 </td><td>0.0798559936370446</td><td>Constrained genes</td></tr>\n",
       "\t<tr><td>MM </td><td>Missense     </td><td>0.07973282</td><td> 0.057075877</td><td>0.10238976</td><td>0.0362786762972129</td><td>0.00766228506990043</td><td>0.0648950675245253</td><td>Constrained genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>Missense     </td><td>0.04345414</td><td> 0.025938225</td><td>0.06097006</td><td>0.0362786762972129</td><td>0.00766228506990043</td><td>0.0648950675245253</td><td>Constrained genes</td></tr>\n",
       "\t<tr><td>MM </td><td>LGD          </td><td>0.03189313</td><td> 0.016492222</td><td>0.04729403</td><td>NA                </td><td>NA                 </td><td>NA                </td><td>Constrained genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>LGD          </td><td>0.01700379</td><td> 0.004756062</td><td>0.02925153</td><td>NA                </td><td>NA                 </td><td>NA                </td><td>Constrained genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DMisL        </td><td>0.04783969</td><td> 0.030073765</td><td>0.06560561</td><td>0.0270572750365613</td><td>0.00551159951987933</td><td>0.0486029505532434</td><td>Constrained genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DMisL        </td><td>0.02078241</td><td> 0.008565796</td><td>0.03299903</td><td>0.0270572750365613</td><td>0.00551159951987933</td><td>0.0486029505532434</td><td>Constrained genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DMis         </td><td>0.01594656</td><td> 0.005566230</td><td>0.02632690</td><td>0.0140572528771715</td><td>0.00303771598682774</td><td>0.0250767897675153</td><td>Constrained genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DMis         </td><td>0.00188931</td><td>-0.001819879</td><td>0.00559850</td><td>0.0140572528771715</td><td>0.00303771598682774</td><td>0.0250767897675153</td><td>Constrained genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DMisL2       </td><td>0.04783969</td><td> 0.030073765</td><td>0.06560561</td><td>0.0270572750365613</td><td>0.00551159951987933</td><td>0.0486029505532434</td><td>Constrained genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DMisL2       </td><td>0.02078241</td><td> 0.008565796</td><td>0.03299903</td><td>0.0270572750365613</td><td>0.00551159951987933</td><td>0.0486029505532434</td><td>Constrained genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DMisM        </td><td>0.05138337</td><td> 0.032995934</td><td>0.06977081</td><td>0.0268223348770543</td><td>0.00416767740103035</td><td>0.0494769923530782</td><td>Constrained genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DMisM        </td><td>0.02456104</td><td> 0.011298811</td><td>0.03782326</td><td>0.0268223348770543</td><td>0.00416767740103035</td><td>0.0494769923530782</td><td>Constrained genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DMisMC       </td><td>0.04606785</td><td> 0.028622331</td><td>0.06351337</td><td>0.0309533660150532</td><td>0.0106352895413586 </td><td>0.0512714424887477</td><td>Constrained genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DMisMC       </td><td>0.01511448</td><td> 0.004674221</td><td>0.02555475</td><td>0.0309533660150532</td><td>0.0106352895413586 </td><td>0.0512714424887477</td><td>Constrained genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DamagingL    </td><td>0.07973282</td><td> 0.056547388</td><td>0.10291825</td><td>0.0419466076453203</td><td>0.0131054656748938 </td><td>0.0707877496157468</td><td>Constrained genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DamagingL    </td><td>0.03778621</td><td> 0.020596534</td><td>0.05497588</td><td>0.0419466076453203</td><td>0.0131054656748938 </td><td>0.0707877496157468</td><td>Constrained genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DamagingS    </td><td>0.04783969</td><td> 0.029404526</td><td>0.06627485</td><td>0.0289465854859305</td><td>0.00652870274988775</td><td>0.0513644682219732</td><td>Constrained genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DamagingS    </td><td>0.01889310</td><td> 0.006109424</td><td>0.03167679</td><td>0.0289465854859305</td><td>0.00652870274988775</td><td>0.0513644682219732</td><td>Constrained genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DamagingL2   </td><td>0.07973282</td><td> 0.056547388</td><td>0.10291825</td><td>0.0419466076453203</td><td>0.0131054656748938 </td><td>0.0707877496157468</td><td>Constrained genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DamagingL2   </td><td>0.03778621</td><td> 0.020596534</td><td>0.05497588</td><td>0.0419466076453203</td><td>0.0131054656748938 </td><td>0.0707877496157468</td><td>Constrained genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DamagingM    </td><td>0.08327650</td><td> 0.059635344</td><td>0.10691765</td><td>0.0436009779351824</td><td>0.0141714714898231 </td><td>0.0730304843805418</td><td>Constrained genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DamagingM    </td><td>0.03967552</td><td> 0.022111868</td><td>0.05723917</td><td>0.0436009779351824</td><td>0.0141714714898231 </td><td>0.0730304843805418</td><td>Constrained genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DamagingMC   </td><td>0.07796098</td><td> 0.055007827</td><td>0.10091413</td><td>0.0458426986238121</td><td>0.0178785460470397 </td><td>0.0738068512005846</td><td>Constrained genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DamagingMC   </td><td>0.03211828</td><td> 0.016109973</td><td>0.04812658</td><td>0.0458426986238121</td><td>0.0178785460470397 </td><td>0.0738068512005846</td><td>Constrained genes</td></tr>\n",
       "\t<tr><td>MM </td><td>OtherMissense</td><td>0.06378625</td><td> 0.043396219</td><td>0.08417629</td><td>NA                </td><td>NA                 </td><td>NA                </td><td>Constrained genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>OtherMissense</td><td>0.04156483</td><td> 0.024421712</td><td>0.05870795</td><td>NA                </td><td>NA                 </td><td>NA                </td><td>Constrained genes</td></tr>\n",
       "\t<tr><td>MM </td><td>Synonymous   </td><td>0.04429601</td><td> 0.026484217</td><td>0.06210780</td><td>NA                </td><td>NA                 </td><td>NA                </td><td>Constrained genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>Synonymous   </td><td>0.04156483</td><td> 0.024421712</td><td>0.05870795</td><td>NA                </td><td>NA                 </td><td>NA                </td><td>Constrained genes</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 32 × 9\n",
       "\\begin{tabular}{lllllllll}\n",
       " cohort & category & mean & lower & upper & mean\\_diff & mean\\_diff\\_lower & mean\\_diff\\_upper & Gene\\\\\n",
       " <chr> & <chr> & <dbl> & <dbl> & <dbl> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t MM  & All           & 0.34550887 &  0.297063590 & 0.39395415 & NA                 & NA                  & NA                 & Constrained genes\\\\\n",
       "\t SSC & All           & 0.28906450 &  0.242016333 & 0.33611266 & NA                 & NA                  & NA                 & Constrained genes\\\\\n",
       "\t MM  & Nonsynonymous & 0.10099490 &  0.075706682 & 0.12628312 & 0.0480942084857547 & 0.0163324233344647  & 0.0798559936370446 & Constrained genes\\\\\n",
       "\t SSC & Nonsynonymous & 0.05290069 &  0.033643475 & 0.07215791 & 0.0480942084857547 & 0.0163324233344647  & 0.0798559936370446 & Constrained genes\\\\\n",
       "\t MM  & Missense      & 0.07973282 &  0.057075877 & 0.10238976 & 0.0362786762972129 & 0.00766228506990043 & 0.0648950675245253 & Constrained genes\\\\\n",
       "\t SSC & Missense      & 0.04345414 &  0.025938225 & 0.06097006 & 0.0362786762972129 & 0.00766228506990043 & 0.0648950675245253 & Constrained genes\\\\\n",
       "\t MM  & LGD           & 0.03189313 &  0.016492222 & 0.04729403 & NA                 & NA                  & NA                 & Constrained genes\\\\\n",
       "\t SSC & LGD           & 0.01700379 &  0.004756062 & 0.02925153 & NA                 & NA                  & NA                 & Constrained genes\\\\\n",
       "\t MM  & DMisL         & 0.04783969 &  0.030073765 & 0.06560561 & 0.0270572750365613 & 0.00551159951987933 & 0.0486029505532434 & Constrained genes\\\\\n",
       "\t SSC & DMisL         & 0.02078241 &  0.008565796 & 0.03299903 & 0.0270572750365613 & 0.00551159951987933 & 0.0486029505532434 & Constrained genes\\\\\n",
       "\t MM  & DMis          & 0.01594656 &  0.005566230 & 0.02632690 & 0.0140572528771715 & 0.00303771598682774 & 0.0250767897675153 & Constrained genes\\\\\n",
       "\t SSC & DMis          & 0.00188931 & -0.001819879 & 0.00559850 & 0.0140572528771715 & 0.00303771598682774 & 0.0250767897675153 & Constrained genes\\\\\n",
       "\t MM  & DMisL2        & 0.04783969 &  0.030073765 & 0.06560561 & 0.0270572750365613 & 0.00551159951987933 & 0.0486029505532434 & Constrained genes\\\\\n",
       "\t SSC & DMisL2        & 0.02078241 &  0.008565796 & 0.03299903 & 0.0270572750365613 & 0.00551159951987933 & 0.0486029505532434 & Constrained genes\\\\\n",
       "\t MM  & DMisM         & 0.05138337 &  0.032995934 & 0.06977081 & 0.0268223348770543 & 0.00416767740103035 & 0.0494769923530782 & Constrained genes\\\\\n",
       "\t SSC & DMisM         & 0.02456104 &  0.011298811 & 0.03782326 & 0.0268223348770543 & 0.00416767740103035 & 0.0494769923530782 & Constrained genes\\\\\n",
       "\t MM  & DMisMC        & 0.04606785 &  0.028622331 & 0.06351337 & 0.0309533660150532 & 0.0106352895413586  & 0.0512714424887477 & Constrained genes\\\\\n",
       "\t SSC & DMisMC        & 0.01511448 &  0.004674221 & 0.02555475 & 0.0309533660150532 & 0.0106352895413586  & 0.0512714424887477 & Constrained genes\\\\\n",
       "\t MM  & DamagingL     & 0.07973282 &  0.056547388 & 0.10291825 & 0.0419466076453203 & 0.0131054656748938  & 0.0707877496157468 & Constrained genes\\\\\n",
       "\t SSC & DamagingL     & 0.03778621 &  0.020596534 & 0.05497588 & 0.0419466076453203 & 0.0131054656748938  & 0.0707877496157468 & Constrained genes\\\\\n",
       "\t MM  & DamagingS     & 0.04783969 &  0.029404526 & 0.06627485 & 0.0289465854859305 & 0.00652870274988775 & 0.0513644682219732 & Constrained genes\\\\\n",
       "\t SSC & DamagingS     & 0.01889310 &  0.006109424 & 0.03167679 & 0.0289465854859305 & 0.00652870274988775 & 0.0513644682219732 & Constrained genes\\\\\n",
       "\t MM  & DamagingL2    & 0.07973282 &  0.056547388 & 0.10291825 & 0.0419466076453203 & 0.0131054656748938  & 0.0707877496157468 & Constrained genes\\\\\n",
       "\t SSC & DamagingL2    & 0.03778621 &  0.020596534 & 0.05497588 & 0.0419466076453203 & 0.0131054656748938  & 0.0707877496157468 & Constrained genes\\\\\n",
       "\t MM  & DamagingM     & 0.08327650 &  0.059635344 & 0.10691765 & 0.0436009779351824 & 0.0141714714898231  & 0.0730304843805418 & Constrained genes\\\\\n",
       "\t SSC & DamagingM     & 0.03967552 &  0.022111868 & 0.05723917 & 0.0436009779351824 & 0.0141714714898231  & 0.0730304843805418 & Constrained genes\\\\\n",
       "\t MM  & DamagingMC    & 0.07796098 &  0.055007827 & 0.10091413 & 0.0458426986238121 & 0.0178785460470397  & 0.0738068512005846 & Constrained genes\\\\\n",
       "\t SSC & DamagingMC    & 0.03211828 &  0.016109973 & 0.04812658 & 0.0458426986238121 & 0.0178785460470397  & 0.0738068512005846 & Constrained genes\\\\\n",
       "\t MM  & OtherMissense & 0.06378625 &  0.043396219 & 0.08417629 & NA                 & NA                  & NA                 & Constrained genes\\\\\n",
       "\t SSC & OtherMissense & 0.04156483 &  0.024421712 & 0.05870795 & NA                 & NA                  & NA                 & Constrained genes\\\\\n",
       "\t MM  & Synonymous    & 0.04429601 &  0.026484217 & 0.06210780 & NA                 & NA                  & NA                 & Constrained genes\\\\\n",
       "\t SSC & Synonymous    & 0.04156483 &  0.024421712 & 0.05870795 & NA                 & NA                  & NA                 & Constrained genes\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 32 × 9\n",
       "\n",
       "| cohort &lt;chr&gt; | category &lt;chr&gt; | mean &lt;dbl&gt; | lower &lt;dbl&gt; | upper &lt;dbl&gt; | mean_diff &lt;chr&gt; | mean_diff_lower &lt;chr&gt; | mean_diff_upper &lt;chr&gt; | Gene &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| MM  | All           | 0.34550887 |  0.297063590 | 0.39395415 | NA                 | NA                  | NA                 | Constrained genes |\n",
       "| SSC | All           | 0.28906450 |  0.242016333 | 0.33611266 | NA                 | NA                  | NA                 | Constrained genes |\n",
       "| MM  | Nonsynonymous | 0.10099490 |  0.075706682 | 0.12628312 | 0.0480942084857547 | 0.0163324233344647  | 0.0798559936370446 | Constrained genes |\n",
       "| SSC | Nonsynonymous | 0.05290069 |  0.033643475 | 0.07215791 | 0.0480942084857547 | 0.0163324233344647  | 0.0798559936370446 | Constrained genes |\n",
       "| MM  | Missense      | 0.07973282 |  0.057075877 | 0.10238976 | 0.0362786762972129 | 0.00766228506990043 | 0.0648950675245253 | Constrained genes |\n",
       "| SSC | Missense      | 0.04345414 |  0.025938225 | 0.06097006 | 0.0362786762972129 | 0.00766228506990043 | 0.0648950675245253 | Constrained genes |\n",
       "| MM  | LGD           | 0.03189313 |  0.016492222 | 0.04729403 | NA                 | NA                  | NA                 | Constrained genes |\n",
       "| SSC | LGD           | 0.01700379 |  0.004756062 | 0.02925153 | NA                 | NA                  | NA                 | Constrained genes |\n",
       "| MM  | DMisL         | 0.04783969 |  0.030073765 | 0.06560561 | 0.0270572750365613 | 0.00551159951987933 | 0.0486029505532434 | Constrained genes |\n",
       "| SSC | DMisL         | 0.02078241 |  0.008565796 | 0.03299903 | 0.0270572750365613 | 0.00551159951987933 | 0.0486029505532434 | Constrained genes |\n",
       "| MM  | DMis          | 0.01594656 |  0.005566230 | 0.02632690 | 0.0140572528771715 | 0.00303771598682774 | 0.0250767897675153 | Constrained genes |\n",
       "| SSC | DMis          | 0.00188931 | -0.001819879 | 0.00559850 | 0.0140572528771715 | 0.00303771598682774 | 0.0250767897675153 | Constrained genes |\n",
       "| MM  | DMisL2        | 0.04783969 |  0.030073765 | 0.06560561 | 0.0270572750365613 | 0.00551159951987933 | 0.0486029505532434 | Constrained genes |\n",
       "| SSC | DMisL2        | 0.02078241 |  0.008565796 | 0.03299903 | 0.0270572750365613 | 0.00551159951987933 | 0.0486029505532434 | Constrained genes |\n",
       "| MM  | DMisM         | 0.05138337 |  0.032995934 | 0.06977081 | 0.0268223348770543 | 0.00416767740103035 | 0.0494769923530782 | Constrained genes |\n",
       "| SSC | DMisM         | 0.02456104 |  0.011298811 | 0.03782326 | 0.0268223348770543 | 0.00416767740103035 | 0.0494769923530782 | Constrained genes |\n",
       "| MM  | DMisMC        | 0.04606785 |  0.028622331 | 0.06351337 | 0.0309533660150532 | 0.0106352895413586  | 0.0512714424887477 | Constrained genes |\n",
       "| SSC | DMisMC        | 0.01511448 |  0.004674221 | 0.02555475 | 0.0309533660150532 | 0.0106352895413586  | 0.0512714424887477 | Constrained genes |\n",
       "| MM  | DamagingL     | 0.07973282 |  0.056547388 | 0.10291825 | 0.0419466076453203 | 0.0131054656748938  | 0.0707877496157468 | Constrained genes |\n",
       "| SSC | DamagingL     | 0.03778621 |  0.020596534 | 0.05497588 | 0.0419466076453203 | 0.0131054656748938  | 0.0707877496157468 | Constrained genes |\n",
       "| MM  | DamagingS     | 0.04783969 |  0.029404526 | 0.06627485 | 0.0289465854859305 | 0.00652870274988775 | 0.0513644682219732 | Constrained genes |\n",
       "| SSC | DamagingS     | 0.01889310 |  0.006109424 | 0.03167679 | 0.0289465854859305 | 0.00652870274988775 | 0.0513644682219732 | Constrained genes |\n",
       "| MM  | DamagingL2    | 0.07973282 |  0.056547388 | 0.10291825 | 0.0419466076453203 | 0.0131054656748938  | 0.0707877496157468 | Constrained genes |\n",
       "| SSC | DamagingL2    | 0.03778621 |  0.020596534 | 0.05497588 | 0.0419466076453203 | 0.0131054656748938  | 0.0707877496157468 | Constrained genes |\n",
       "| MM  | DamagingM     | 0.08327650 |  0.059635344 | 0.10691765 | 0.0436009779351824 | 0.0141714714898231  | 0.0730304843805418 | Constrained genes |\n",
       "| SSC | DamagingM     | 0.03967552 |  0.022111868 | 0.05723917 | 0.0436009779351824 | 0.0141714714898231  | 0.0730304843805418 | Constrained genes |\n",
       "| MM  | DamagingMC    | 0.07796098 |  0.055007827 | 0.10091413 | 0.0458426986238121 | 0.0178785460470397  | 0.0738068512005846 | Constrained genes |\n",
       "| SSC | DamagingMC    | 0.03211828 |  0.016109973 | 0.04812658 | 0.0458426986238121 | 0.0178785460470397  | 0.0738068512005846 | Constrained genes |\n",
       "| MM  | OtherMissense | 0.06378625 |  0.043396219 | 0.08417629 | NA                 | NA                  | NA                 | Constrained genes |\n",
       "| SSC | OtherMissense | 0.04156483 |  0.024421712 | 0.05870795 | NA                 | NA                  | NA                 | Constrained genes |\n",
       "| MM  | Synonymous    | 0.04429601 |  0.026484217 | 0.06210780 | NA                 | NA                  | NA                 | Constrained genes |\n",
       "| SSC | Synonymous    | 0.04156483 |  0.024421712 | 0.05870795 | NA                 | NA                  | NA                 | Constrained genes |\n",
       "\n"
      ],
      "text/plain": [
       "   cohort category      mean       lower        upper      mean_diff         \n",
       "1  MM     All           0.34550887  0.297063590 0.39395415 NA                \n",
       "2  SSC    All           0.28906450  0.242016333 0.33611266 NA                \n",
       "3  MM     Nonsynonymous 0.10099490  0.075706682 0.12628312 0.0480942084857547\n",
       "4  SSC    Nonsynonymous 0.05290069  0.033643475 0.07215791 0.0480942084857547\n",
       "5  MM     Missense      0.07973282  0.057075877 0.10238976 0.0362786762972129\n",
       "6  SSC    Missense      0.04345414  0.025938225 0.06097006 0.0362786762972129\n",
       "7  MM     LGD           0.03189313  0.016492222 0.04729403 NA                \n",
       "8  SSC    LGD           0.01700379  0.004756062 0.02925153 NA                \n",
       "9  MM     DMisL         0.04783969  0.030073765 0.06560561 0.0270572750365613\n",
       "10 SSC    DMisL         0.02078241  0.008565796 0.03299903 0.0270572750365613\n",
       "11 MM     DMis          0.01594656  0.005566230 0.02632690 0.0140572528771715\n",
       "12 SSC    DMis          0.00188931 -0.001819879 0.00559850 0.0140572528771715\n",
       "13 MM     DMisL2        0.04783969  0.030073765 0.06560561 0.0270572750365613\n",
       "14 SSC    DMisL2        0.02078241  0.008565796 0.03299903 0.0270572750365613\n",
       "15 MM     DMisM         0.05138337  0.032995934 0.06977081 0.0268223348770543\n",
       "16 SSC    DMisM         0.02456104  0.011298811 0.03782326 0.0268223348770543\n",
       "17 MM     DMisMC        0.04606785  0.028622331 0.06351337 0.0309533660150532\n",
       "18 SSC    DMisMC        0.01511448  0.004674221 0.02555475 0.0309533660150532\n",
       "19 MM     DamagingL     0.07973282  0.056547388 0.10291825 0.0419466076453203\n",
       "20 SSC    DamagingL     0.03778621  0.020596534 0.05497588 0.0419466076453203\n",
       "21 MM     DamagingS     0.04783969  0.029404526 0.06627485 0.0289465854859305\n",
       "22 SSC    DamagingS     0.01889310  0.006109424 0.03167679 0.0289465854859305\n",
       "23 MM     DamagingL2    0.07973282  0.056547388 0.10291825 0.0419466076453203\n",
       "24 SSC    DamagingL2    0.03778621  0.020596534 0.05497588 0.0419466076453203\n",
       "25 MM     DamagingM     0.08327650  0.059635344 0.10691765 0.0436009779351824\n",
       "26 SSC    DamagingM     0.03967552  0.022111868 0.05723917 0.0436009779351824\n",
       "27 MM     DamagingMC    0.07796098  0.055007827 0.10091413 0.0458426986238121\n",
       "28 SSC    DamagingMC    0.03211828  0.016109973 0.04812658 0.0458426986238121\n",
       "29 MM     OtherMissense 0.06378625  0.043396219 0.08417629 NA                \n",
       "30 SSC    OtherMissense 0.04156483  0.024421712 0.05870795 NA                \n",
       "31 MM     Synonymous    0.04429601  0.026484217 0.06210780 NA                \n",
       "32 SSC    Synonymous    0.04156483  0.024421712 0.05870795 NA                \n",
       "   mean_diff_lower     mean_diff_upper    Gene             \n",
       "1  NA                  NA                 Constrained genes\n",
       "2  NA                  NA                 Constrained genes\n",
       "3  0.0163324233344647  0.0798559936370446 Constrained genes\n",
       "4  0.0163324233344647  0.0798559936370446 Constrained genes\n",
       "5  0.00766228506990043 0.0648950675245253 Constrained genes\n",
       "6  0.00766228506990043 0.0648950675245253 Constrained genes\n",
       "7  NA                  NA                 Constrained genes\n",
       "8  NA                  NA                 Constrained genes\n",
       "9  0.00551159951987933 0.0486029505532434 Constrained genes\n",
       "10 0.00551159951987933 0.0486029505532434 Constrained genes\n",
       "11 0.00303771598682774 0.0250767897675153 Constrained genes\n",
       "12 0.00303771598682774 0.0250767897675153 Constrained genes\n",
       "13 0.00551159951987933 0.0486029505532434 Constrained genes\n",
       "14 0.00551159951987933 0.0486029505532434 Constrained genes\n",
       "15 0.00416767740103035 0.0494769923530782 Constrained genes\n",
       "16 0.00416767740103035 0.0494769923530782 Constrained genes\n",
       "17 0.0106352895413586  0.0512714424887477 Constrained genes\n",
       "18 0.0106352895413586  0.0512714424887477 Constrained genes\n",
       "19 0.0131054656748938  0.0707877496157468 Constrained genes\n",
       "20 0.0131054656748938  0.0707877496157468 Constrained genes\n",
       "21 0.00652870274988775 0.0513644682219732 Constrained genes\n",
       "22 0.00652870274988775 0.0513644682219732 Constrained genes\n",
       "23 0.0131054656748938  0.0707877496157468 Constrained genes\n",
       "24 0.0131054656748938  0.0707877496157468 Constrained genes\n",
       "25 0.0141714714898231  0.0730304843805418 Constrained genes\n",
       "26 0.0141714714898231  0.0730304843805418 Constrained genes\n",
       "27 0.0178785460470397  0.0738068512005846 Constrained genes\n",
       "28 0.0178785460470397  0.0738068512005846 Constrained genes\n",
       "29 NA                  NA                 Constrained genes\n",
       "30 NA                  NA                 Constrained genes\n",
       "31 NA                  NA                 Constrained genes\n",
       "32 NA                  NA                 Constrained genes"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "\n",
    "#df_tmp$cohort <- factor(df_tmp$cohort, levels = c('SSC', 'MM'), labels = c(\"Control\", \"NTD\"))\n",
    "df_tmp = data.frame()\n",
    "for (eff in figure_all) {\n",
    "    ratio_consen = consensus_size/total_size\n",
    "    mm = round((total_size/consensus_size)*sum(df.mm[, eff]),0)\n",
    "    ssc = round((total_size/consensus_size)*sum(df.ss[, eff]),0)\n",
    "    c = c(mm, ssc)\n",
    "    #c = c(sum(df.mm[, eff]),sum(df.ss[, eff]))\n",
    "    T = c(samplecnt.mm, samplecnt.ssc)\n",
    "    pt = poisson.test(c, T, alternative='greater', conf.level = 0.95)\n",
    "    show(paste0(eff, ': ', round(pt$p.value, 3), '   ', 'Rate ratio: ', pt$estimate))\n",
    "    df_tmp <- Get_summary(df.melt, eff)\n",
    "    cnt_mm <- sum(df.mm[,eff])\n",
    "    cnt_ssc <- sum(df.ss[,eff])\n",
    "    df_rr <- rbind(df_rr, c(eff, round(pt$estimate,3), round(pt$p.value, 3), \"MM\", \"SSC\", \"Constrained\", pt$conf.int, cnt_mm, cnt_ssc))\n",
    "}\n",
    "\n",
    "\n",
    "\n",
    "df_tmp$mean <- as.numeric(df_tmp$mean)\n",
    "df_tmp$lower <- as.numeric(df_tmp$lower)\n",
    "df_tmp$upper <- as.numeric(df_tmp$upper)\n",
    "df_tmp$Gene <- \"Constrained genes\"\n",
    "df_tmp_con <- df_tmp\n",
    "#df_tmp$cohort <- factor(df_tmp$cohort, levels = c('SSC', 'MM'), labels = c(\"Control\", \"NTD\"))\n",
    "\n",
    "\n",
    "\n",
    "df.melt$variant <- as.numeric(df.melt$variant)\n",
    "df_tmp\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 252,
   "id": "79f95eb2",
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in rbind(deparse.level, ...): numbers of columns of arguments do not match\n",
     "output_type": "error",
     "traceback": [
      "Error in rbind(deparse.level, ...): numbers of columns of arguments do not match\nTraceback:\n",
      "1. rbind(df_all_stat, df_tmp_con)",
      "2. rbind(deparse.level, ...)",
      "3. stop(\"numbers of columns of arguments do not match\")"
     ]
    }
   ],
   "source": [
    "#df_tmp_all\n",
    "#df_tmp_con\n",
    "df_all_stat <- df_tmp_all\n",
    "df_all_stat <- rbind(df_all_stat, df_tmp_con)\n",
    "df_all_stat\n",
    "df_rr\n",
    "colnames(df_rr) <- c(\"category\",\"RR\", \"pvalue\", 'MM', \"SSC\", \"Gene\",\"Lower\",\"Upper\", \"MM\", \"SSC\")\n",
    "df_rr_backup <- df_rr\n",
    "df_odds\n",
    "write.table(df_rr, \"/projects/ps-gleesonlab8/User/hiyoothere/NTD/5.Analysis/DNM/Burden/ForPlot/RR_total_v2.txt\", quote = F, row.names = F, sep = '\\t')\n",
    "write.table(df_all_stat, \"/projects/ps-gleesonlab8/User/hiyoothere/NTD/5.Analysis/DNM/Burden/ForPlot/Stat_v2.txt\", quote = F, row.names = F, sep = '\\t')\n",
    "write.table(df_odds, \"/projects/ps-gleesonlab8/User/hiyoothere/NTD/5.Analysis/DNM/Burden/ForPlot/OR_total_v2.txt\", quote = F, row.names = F, sep = '\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b944cde8",
   "metadata": {},
   "source": [
    "## Burden with FA information"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "18c95bf2",
   "metadata": {},
   "outputs": [],
   "source": [
    "dim(df.mm)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3d88d78a",
   "metadata": {},
   "outputs": [],
   "source": [
    "#df.mm <- df.mm.backup\n",
    "dim(df.mm.backup)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5dd4da82",
   "metadata": {},
   "outputs": [],
   "source": [
    "head(df.mm)\n",
    "dim(df.mm)\n",
    "table(df.mm$FA)\n",
    "\n",
    "df.mm.fa = subset(df.mm, df.mm$FA == 'YES')\n",
    "df.mm.nfa = subset(df.mm, df.mm$FA == 'NO')\n",
    "\n",
    "samplecnt.mm = 319\n",
    "\n",
    "as.numeric(dim(df.mm.fa$cohort)[1])\n",
    "dim(df.mm.fa)\n",
    "dim(df.mm.nfa)\n",
    "\n",
    "df.mm.backup <- df.mm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8370bd4f",
   "metadata": {},
   "outputs": [],
   "source": [
    "df.mm <- df.mm.fa\n",
    "\n",
    "total_size = 59281518 #size of coding region (hg38)\n",
    "\n",
    "consensus = read.delim(paste(\"/projects/ps-gleesonlab8/User/hiyoothere/NTD/10.Poisson/consensus_region/all_batches_\",percent, '.txt', sep=\"\"), header = FALSE)#all_batches_70.bed\n",
    "consensus\n",
    "Get_summary <- function(input, category){\n",
    "    \n",
    "    tmp_data = subset(df.melt, class == category)\n",
    "    head(tmp_data)\n",
    "    tmp_mm = tmp_data[tmp_data$cohort == 'MM', ]\n",
    "    tmp_ssc = tmp_data[tmp_data$cohort == 'SSC', ]\n",
    "#mean(tmp_mm$variant)\n",
    "   \n",
    "    df_tmp <- rbind(df_tmp,c('MM', category,  mean(as.numeric(tmp_mm$variant) * 100000000 / consensus_size / 2), \n",
    "                             sd(as.numeric(tmp_mm$variant) * 100000000 / consensus_size / 2), \n",
    "                             SE(as.numeric(tmp_mm$variant) * 100000000 / consensus_size / 2), \n",
    "                             CI(as.numeric(tmp_mm$variant) * 100000000 / consensus_size / 2)))\n",
    "    df_tmp <- rbind(df_tmp, c('SSC',category, mean(as.numeric(tmp_ssc$variant) * 100000000 / consensus_size / 2), \n",
    "                              sd(as.numeric(tmp_ssc$variant) * 100000000 / consensus_size / 2),\n",
    "                              SE(as.numeric(tmp_ssc$variant) * 100000000 / consensus_size / 2), \n",
    "                              CI(as.numeric(tmp_ssc$variant) * 100000000 / consensus_size / 2)))\n",
    "    colnames(df_tmp) = c('cohort', 'category','mean', 'sd', 'se', 'ci')\n",
    "    return (df_tmp)\n",
    "}\n",
    "\n",
    "df_tmp <- data.frame()\n",
    "df_rr <- data.frame()\n",
    "\n",
    "paste(\"consensus size is \", consensus[1,])\n",
    "consensus_size = consensus[1,]\n",
    "\n",
    "paste(\"consensus ratio \", consensus[1,]/total_size)\n",
    "\n",
    "paste(\"sample ntd \", samplecnt.mm)\n",
    "paste(\"sample ssc \", samplecnt.ssc)\n",
    "\n",
    "for (eff in figure_all) {\n",
    "    ratio_consen = consensus_size/total_size\n",
    "    mm = round((total_size*sum(df.mm[, eff])/consensus_size),0)\n",
    "    ssc = round((total_size*sum(df.ss[, eff])/consensus_size),0)\n",
    "    show(paste(eff, \"raw count\", sum(df.mm[, eff]), sum(df.ss[, eff])))\n",
    "    show(paste(eff, \"modi count\", mm, ssc))\n",
    "    c = c(mm,ssc)\n",
    "    #c=c(sum(df.mm[, eff]), sum(df.ss[, eff]))\n",
    "    T = c(samplecnt.mm, samplecnt.ssc)\n",
    "    pt = poisson.test(c, T, alternative='two.sided', conf.level = 0.95)\n",
    "    show(paste0(eff, ': ', round(pt$p.value, 8), '   ', 'Rate ratio: ', pt$estimate))\n",
    "    cnt_mm <- sum(df.mm[,eff])\n",
    "    cnt_ssc <- sum(df.ss[,eff])\n",
    "    df_rr <- rbind(df_rr, c(eff, round(pt$estimate,3), round(pt$p.value, 8), \"MM\", \"SSC\", \"All\", pt$conf.int, cnt_mm, cnt_ssc))\n",
    "    df_tmp <- Get_summary(df.melt, eff)\n",
    "}\n",
    "head(df_tmp)\n",
    "df_tmp$mean <- as.numeric(df_tmp$mean)\n",
    "df_tmp$ci <- as.numeric(df_tmp$ci)\n",
    "df_tmp$cohort <- factor(df_tmp$cohort, levels = c('SSC', 'MM'), labels = c(\"SSC\", \"MM\"))\n",
    "df_tmp$Gene <- \"All genes\"\n",
    "df_tmp_all <- df_tmp\n",
    "df_all_stat = df_tmp_all\n",
    "df_all_stat_backup = df_all_stat\n",
    "\n",
    "df.melt$variant <- as.numeric(df.melt$variant)\n",
    "head(df.melt)\n",
    "df_tmp\n",
    "options(repr.plot.width=14, repr.plot.height=8)\n",
    "df.mm = df[df$cohort == 'MM', ]\n",
    "df.ss = df[df$cohort == 'SSC', ]\n",
    "head(df.mm)\n",
    "\n",
    "\n",
    "dim(df.mm)\n",
    "dim(df.ss)\n",
    "\n",
    "dim(df.mm$All)\n",
    "dim(df.ss$All)\n",
    "df.melt <- melt(df, id.vars=c('sample', 'family', 'cohort', 'callable'))\n",
    "\n",
    "print (dim(subset(df.mm,df.mm$All == 0)))\n",
    "print (dim(subset(df.ss, df.ss$All == 0)))\n",
    "colnames(df.melt)[5] = 'class'\n",
    "colnames(df.melt)[6] = 'variant'\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "head(df.melt)#ggsave(\"/projects/ps-gleesonlab8/User/hiyoothere/NTD/15.Final/221020_DNM_v9/figure/vplot_all.pdf\", plot= p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 191,
   "id": "f3fc316a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "40.5444312910953"
      ],
      "text/latex": [
       "40.5444312910953"
      ],
      "text/markdown": [
       "40.5444312910953"
      ],
      "text/plain": [
       "[1] 40.54443"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "41"
      ],
      "text/latex": [
       "41"
      ],
      "text/markdown": [
       "41"
      ],
      "text/plain": [
       "[1] 41"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "(total_size/consensus_size)*sum(df.mm[, eff])\n",
    "round((total_size/consensus_size)*sum(df.mm[, eff]),0)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 192,
   "id": "f1e59eac",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 36 × 9</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>cohort</th><th scope=col>category</th><th scope=col>mean</th><th scope=col>lower</th><th scope=col>upper</th><th scope=col>mean_diff</th><th scope=col>mean_diff_lower</th><th scope=col>mean_diff_upper</th><th scope=col>Gene</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>MM </td><td>All          </td><td>1.27395323</td><td>1.18086405</td><td>1.36704240</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>All          </td><td>1.25450214</td><td>1.15469614</td><td>1.35430813</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>Nonsynonymous</td><td>0.33133415</td><td>0.28467739</td><td>0.37799091</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>Nonsynonymous</td><td>0.28906450</td><td>0.24442039</td><td>0.33370860</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>Missense     </td><td>0.27463526</td><td>0.23174229</td><td>0.31752822</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>Missense     </td><td>0.26072484</td><td>0.21812915</td><td>0.30332054</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>LGD          </td><td>0.11516962</td><td>0.08656898</td><td>0.14377027</td><td>0.0603796209933105</td><td>0.0249566763471755</td><td>0.0958025656394456</td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>LGD          </td><td>0.05479000</td><td>0.03384624</td><td>0.07573377</td><td>0.0603796209933105</td><td>0.0249566763471755</td><td>0.0958025656394456</td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DMisL        </td><td>0.15415011</td><td>0.12278819</td><td>0.18551203</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DMisL        </td><td>0.13980897</td><td>0.10865755</td><td>0.17096040</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DMis         </td><td>0.06024257</td><td>0.03979874</td><td>0.08068640</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DMis         </td><td>0.03967552</td><td>0.02291462</td><td>0.05643641</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DMisL2       </td><td>0.15237827</td><td>0.12116656</td><td>0.18358999</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DMisL2       </td><td>0.13225173</td><td>0.10181715</td><td>0.16268631</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DMisM        </td><td>0.18427140</td><td>0.14981844</td><td>0.21872436</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DMisM        </td><td>0.17003794</td><td>0.13545440</td><td>0.20462148</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DMisMC       </td><td>0.13288803</td><td>0.10382803</td><td>0.16194803</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DMisMC       </td><td>0.11146932</td><td>0.08315469</td><td>0.13978394</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DMisMC2      </td><td>0.17186852</td><td>0.13870134</td><td>0.20503569</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DMisMC2      </td><td>0.15303415</td><td>0.11984811</td><td>0.18622018</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DamagingL    </td><td>0.26931974</td><td>0.22696776</td><td>0.31167171</td><td>0.0747207598965531</td><td>0.017680361675465 </td><td>0.131761158117641 </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DamagingL    </td><td>0.19459898</td><td>0.15632169</td><td>0.23287626</td><td>0.0747207598965531</td><td>0.017680361675465 </td><td>0.131761158117641 </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DamagingS    </td><td>0.17541220</td><td>0.14125679</td><td>0.20956761</td><td>0.0809466741234898</td><td>0.0374997132181427</td><td>0.124393635028837 </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DamagingS    </td><td>0.09446552</td><td>0.06755928</td><td>0.12137176</td><td>0.0809466741234898</td><td>0.0374997132181427</td><td>0.124393635028837 </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DamagingL2   </td><td>0.26754790</td><td>0.22528289</td><td>0.30981290</td><td>0.080506161324414 </td><td>0.0238792822303671</td><td>0.137133040418461 </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DamagingL2   </td><td>0.18704173</td><td>0.14928703</td><td>0.22479644</td><td>0.080506161324414 </td><td>0.0238792822303671</td><td>0.137133040418461 </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DamagingM    </td><td>0.29944102</td><td>0.25514149</td><td>0.34374056</td><td>0.0765023894394815</td><td>0.0163323040955573</td><td>0.136672474783406 </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DamagingM    </td><td>0.22293863</td><td>0.18214851</td><td>0.26372876</td><td>0.0765023894394815</td><td>0.0163323040955573</td><td>0.136672474783406 </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DamagingMC   </td><td>0.24805765</td><td>0.20828111</td><td>0.28783419</td><td>0.0817983322017028</td><td>0.0285486462435022</td><td>0.135048018159903 </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DamagingMC   </td><td>0.16625932</td><td>0.13079238</td><td>0.20172626</td><td>0.0817983322017028</td><td>0.0285486462435022</td><td>0.135048018159903 </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DamagingMC2  </td><td>0.28703814</td><td>0.24356651</td><td>0.33050977</td><td>0.0811033008964945</td><td>0.0222567442572848</td><td>0.139949857535704 </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DamagingMC2  </td><td>0.20593484</td><td>0.16620177</td><td>0.24566791</td><td>0.0811033008964945</td><td>0.0222567442572848</td><td>0.139949857535704 </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>OtherMissense</td><td>0.12048515</td><td>0.09136339</td><td>0.14960690</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>OtherMissense</td><td>0.12091587</td><td>0.09068329</td><td>0.15114845</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>Synonymous   </td><td>0.12402883</td><td>0.09335873</td><td>0.15469892</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>Synonymous   </td><td>0.12280518</td><td>0.09330564</td><td>0.15230472</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 36 × 9\n",
       "\\begin{tabular}{lllllllll}\n",
       " cohort & category & mean & lower & upper & mean\\_diff & mean\\_diff\\_lower & mean\\_diff\\_upper & Gene\\\\\n",
       " <fct> & <chr> & <dbl> & <dbl> & <dbl> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t MM  & All           & 1.27395323 & 1.18086405 & 1.36704240 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t SSC & All           & 1.25450214 & 1.15469614 & 1.35430813 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t MM  & Nonsynonymous & 0.33133415 & 0.28467739 & 0.37799091 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t SSC & Nonsynonymous & 0.28906450 & 0.24442039 & 0.33370860 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t MM  & Missense      & 0.27463526 & 0.23174229 & 0.31752822 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t SSC & Missense      & 0.26072484 & 0.21812915 & 0.30332054 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t MM  & LGD           & 0.11516962 & 0.08656898 & 0.14377027 & 0.0603796209933105 & 0.0249566763471755 & 0.0958025656394456 & All genes\\\\\n",
       "\t SSC & LGD           & 0.05479000 & 0.03384624 & 0.07573377 & 0.0603796209933105 & 0.0249566763471755 & 0.0958025656394456 & All genes\\\\\n",
       "\t MM  & DMisL         & 0.15415011 & 0.12278819 & 0.18551203 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t SSC & DMisL         & 0.13980897 & 0.10865755 & 0.17096040 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t MM  & DMis          & 0.06024257 & 0.03979874 & 0.08068640 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t SSC & DMis          & 0.03967552 & 0.02291462 & 0.05643641 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t MM  & DMisL2        & 0.15237827 & 0.12116656 & 0.18358999 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t SSC & DMisL2        & 0.13225173 & 0.10181715 & 0.16268631 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t MM  & DMisM         & 0.18427140 & 0.14981844 & 0.21872436 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t SSC & DMisM         & 0.17003794 & 0.13545440 & 0.20462148 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t MM  & DMisMC        & 0.13288803 & 0.10382803 & 0.16194803 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t SSC & DMisMC        & 0.11146932 & 0.08315469 & 0.13978394 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t MM  & DMisMC2       & 0.17186852 & 0.13870134 & 0.20503569 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t SSC & DMisMC2       & 0.15303415 & 0.11984811 & 0.18622018 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t MM  & DamagingL     & 0.26931974 & 0.22696776 & 0.31167171 & 0.0747207598965531 & 0.017680361675465  & 0.131761158117641  & All genes\\\\\n",
       "\t SSC & DamagingL     & 0.19459898 & 0.15632169 & 0.23287626 & 0.0747207598965531 & 0.017680361675465  & 0.131761158117641  & All genes\\\\\n",
       "\t MM  & DamagingS     & 0.17541220 & 0.14125679 & 0.20956761 & 0.0809466741234898 & 0.0374997132181427 & 0.124393635028837  & All genes\\\\\n",
       "\t SSC & DamagingS     & 0.09446552 & 0.06755928 & 0.12137176 & 0.0809466741234898 & 0.0374997132181427 & 0.124393635028837  & All genes\\\\\n",
       "\t MM  & DamagingL2    & 0.26754790 & 0.22528289 & 0.30981290 & 0.080506161324414  & 0.0238792822303671 & 0.137133040418461  & All genes\\\\\n",
       "\t SSC & DamagingL2    & 0.18704173 & 0.14928703 & 0.22479644 & 0.080506161324414  & 0.0238792822303671 & 0.137133040418461  & All genes\\\\\n",
       "\t MM  & DamagingM     & 0.29944102 & 0.25514149 & 0.34374056 & 0.0765023894394815 & 0.0163323040955573 & 0.136672474783406  & All genes\\\\\n",
       "\t SSC & DamagingM     & 0.22293863 & 0.18214851 & 0.26372876 & 0.0765023894394815 & 0.0163323040955573 & 0.136672474783406  & All genes\\\\\n",
       "\t MM  & DamagingMC    & 0.24805765 & 0.20828111 & 0.28783419 & 0.0817983322017028 & 0.0285486462435022 & 0.135048018159903  & All genes\\\\\n",
       "\t SSC & DamagingMC    & 0.16625932 & 0.13079238 & 0.20172626 & 0.0817983322017028 & 0.0285486462435022 & 0.135048018159903  & All genes\\\\\n",
       "\t MM  & DamagingMC2   & 0.28703814 & 0.24356651 & 0.33050977 & 0.0811033008964945 & 0.0222567442572848 & 0.139949857535704  & All genes\\\\\n",
       "\t SSC & DamagingMC2   & 0.20593484 & 0.16620177 & 0.24566791 & 0.0811033008964945 & 0.0222567442572848 & 0.139949857535704  & All genes\\\\\n",
       "\t MM  & OtherMissense & 0.12048515 & 0.09136339 & 0.14960690 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t SSC & OtherMissense & 0.12091587 & 0.09068329 & 0.15114845 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t MM  & Synonymous    & 0.12402883 & 0.09335873 & 0.15469892 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t SSC & Synonymous    & 0.12280518 & 0.09330564 & 0.15230472 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 36 × 9\n",
       "\n",
       "| cohort &lt;fct&gt; | category &lt;chr&gt; | mean &lt;dbl&gt; | lower &lt;dbl&gt; | upper &lt;dbl&gt; | mean_diff &lt;chr&gt; | mean_diff_lower &lt;chr&gt; | mean_diff_upper &lt;chr&gt; | Gene &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| MM  | All           | 1.27395323 | 1.18086405 | 1.36704240 | NA                 | NA                 | NA                 | All genes |\n",
       "| SSC | All           | 1.25450214 | 1.15469614 | 1.35430813 | NA                 | NA                 | NA                 | All genes |\n",
       "| MM  | Nonsynonymous | 0.33133415 | 0.28467739 | 0.37799091 | NA                 | NA                 | NA                 | All genes |\n",
       "| SSC | Nonsynonymous | 0.28906450 | 0.24442039 | 0.33370860 | NA                 | NA                 | NA                 | All genes |\n",
       "| MM  | Missense      | 0.27463526 | 0.23174229 | 0.31752822 | NA                 | NA                 | NA                 | All genes |\n",
       "| SSC | Missense      | 0.26072484 | 0.21812915 | 0.30332054 | NA                 | NA                 | NA                 | All genes |\n",
       "| MM  | LGD           | 0.11516962 | 0.08656898 | 0.14377027 | 0.0603796209933105 | 0.0249566763471755 | 0.0958025656394456 | All genes |\n",
       "| SSC | LGD           | 0.05479000 | 0.03384624 | 0.07573377 | 0.0603796209933105 | 0.0249566763471755 | 0.0958025656394456 | All genes |\n",
       "| MM  | DMisL         | 0.15415011 | 0.12278819 | 0.18551203 | NA                 | NA                 | NA                 | All genes |\n",
       "| SSC | DMisL         | 0.13980897 | 0.10865755 | 0.17096040 | NA                 | NA                 | NA                 | All genes |\n",
       "| MM  | DMis          | 0.06024257 | 0.03979874 | 0.08068640 | NA                 | NA                 | NA                 | All genes |\n",
       "| SSC | DMis          | 0.03967552 | 0.02291462 | 0.05643641 | NA                 | NA                 | NA                 | All genes |\n",
       "| MM  | DMisL2        | 0.15237827 | 0.12116656 | 0.18358999 | NA                 | NA                 | NA                 | All genes |\n",
       "| SSC | DMisL2        | 0.13225173 | 0.10181715 | 0.16268631 | NA                 | NA                 | NA                 | All genes |\n",
       "| MM  | DMisM         | 0.18427140 | 0.14981844 | 0.21872436 | NA                 | NA                 | NA                 | All genes |\n",
       "| SSC | DMisM         | 0.17003794 | 0.13545440 | 0.20462148 | NA                 | NA                 | NA                 | All genes |\n",
       "| MM  | DMisMC        | 0.13288803 | 0.10382803 | 0.16194803 | NA                 | NA                 | NA                 | All genes |\n",
       "| SSC | DMisMC        | 0.11146932 | 0.08315469 | 0.13978394 | NA                 | NA                 | NA                 | All genes |\n",
       "| MM  | DMisMC2       | 0.17186852 | 0.13870134 | 0.20503569 | NA                 | NA                 | NA                 | All genes |\n",
       "| SSC | DMisMC2       | 0.15303415 | 0.11984811 | 0.18622018 | NA                 | NA                 | NA                 | All genes |\n",
       "| MM  | DamagingL     | 0.26931974 | 0.22696776 | 0.31167171 | 0.0747207598965531 | 0.017680361675465  | 0.131761158117641  | All genes |\n",
       "| SSC | DamagingL     | 0.19459898 | 0.15632169 | 0.23287626 | 0.0747207598965531 | 0.017680361675465  | 0.131761158117641  | All genes |\n",
       "| MM  | DamagingS     | 0.17541220 | 0.14125679 | 0.20956761 | 0.0809466741234898 | 0.0374997132181427 | 0.124393635028837  | All genes |\n",
       "| SSC | DamagingS     | 0.09446552 | 0.06755928 | 0.12137176 | 0.0809466741234898 | 0.0374997132181427 | 0.124393635028837  | All genes |\n",
       "| MM  | DamagingL2    | 0.26754790 | 0.22528289 | 0.30981290 | 0.080506161324414  | 0.0238792822303671 | 0.137133040418461  | All genes |\n",
       "| SSC | DamagingL2    | 0.18704173 | 0.14928703 | 0.22479644 | 0.080506161324414  | 0.0238792822303671 | 0.137133040418461  | All genes |\n",
       "| MM  | DamagingM     | 0.29944102 | 0.25514149 | 0.34374056 | 0.0765023894394815 | 0.0163323040955573 | 0.136672474783406  | All genes |\n",
       "| SSC | DamagingM     | 0.22293863 | 0.18214851 | 0.26372876 | 0.0765023894394815 | 0.0163323040955573 | 0.136672474783406  | All genes |\n",
       "| MM  | DamagingMC    | 0.24805765 | 0.20828111 | 0.28783419 | 0.0817983322017028 | 0.0285486462435022 | 0.135048018159903  | All genes |\n",
       "| SSC | DamagingMC    | 0.16625932 | 0.13079238 | 0.20172626 | 0.0817983322017028 | 0.0285486462435022 | 0.135048018159903  | All genes |\n",
       "| MM  | DamagingMC2   | 0.28703814 | 0.24356651 | 0.33050977 | 0.0811033008964945 | 0.0222567442572848 | 0.139949857535704  | All genes |\n",
       "| SSC | DamagingMC2   | 0.20593484 | 0.16620177 | 0.24566791 | 0.0811033008964945 | 0.0222567442572848 | 0.139949857535704  | All genes |\n",
       "| MM  | OtherMissense | 0.12048515 | 0.09136339 | 0.14960690 | NA                 | NA                 | NA                 | All genes |\n",
       "| SSC | OtherMissense | 0.12091587 | 0.09068329 | 0.15114845 | NA                 | NA                 | NA                 | All genes |\n",
       "| MM  | Synonymous    | 0.12402883 | 0.09335873 | 0.15469892 | NA                 | NA                 | NA                 | All genes |\n",
       "| SSC | Synonymous    | 0.12280518 | 0.09330564 | 0.15230472 | NA                 | NA                 | NA                 | All genes |\n",
       "\n"
      ],
      "text/plain": [
       "   cohort category      mean       lower      upper      mean_diff         \n",
       "1  MM     All           1.27395323 1.18086405 1.36704240 NA                \n",
       "2  SSC    All           1.25450214 1.15469614 1.35430813 NA                \n",
       "3  MM     Nonsynonymous 0.33133415 0.28467739 0.37799091 NA                \n",
       "4  SSC    Nonsynonymous 0.28906450 0.24442039 0.33370860 NA                \n",
       "5  MM     Missense      0.27463526 0.23174229 0.31752822 NA                \n",
       "6  SSC    Missense      0.26072484 0.21812915 0.30332054 NA                \n",
       "7  MM     LGD           0.11516962 0.08656898 0.14377027 0.0603796209933105\n",
       "8  SSC    LGD           0.05479000 0.03384624 0.07573377 0.0603796209933105\n",
       "9  MM     DMisL         0.15415011 0.12278819 0.18551203 NA                \n",
       "10 SSC    DMisL         0.13980897 0.10865755 0.17096040 NA                \n",
       "11 MM     DMis          0.06024257 0.03979874 0.08068640 NA                \n",
       "12 SSC    DMis          0.03967552 0.02291462 0.05643641 NA                \n",
       "13 MM     DMisL2        0.15237827 0.12116656 0.18358999 NA                \n",
       "14 SSC    DMisL2        0.13225173 0.10181715 0.16268631 NA                \n",
       "15 MM     DMisM         0.18427140 0.14981844 0.21872436 NA                \n",
       "16 SSC    DMisM         0.17003794 0.13545440 0.20462148 NA                \n",
       "17 MM     DMisMC        0.13288803 0.10382803 0.16194803 NA                \n",
       "18 SSC    DMisMC        0.11146932 0.08315469 0.13978394 NA                \n",
       "19 MM     DMisMC2       0.17186852 0.13870134 0.20503569 NA                \n",
       "20 SSC    DMisMC2       0.15303415 0.11984811 0.18622018 NA                \n",
       "21 MM     DamagingL     0.26931974 0.22696776 0.31167171 0.0747207598965531\n",
       "22 SSC    DamagingL     0.19459898 0.15632169 0.23287626 0.0747207598965531\n",
       "23 MM     DamagingS     0.17541220 0.14125679 0.20956761 0.0809466741234898\n",
       "24 SSC    DamagingS     0.09446552 0.06755928 0.12137176 0.0809466741234898\n",
       "25 MM     DamagingL2    0.26754790 0.22528289 0.30981290 0.080506161324414 \n",
       "26 SSC    DamagingL2    0.18704173 0.14928703 0.22479644 0.080506161324414 \n",
       "27 MM     DamagingM     0.29944102 0.25514149 0.34374056 0.0765023894394815\n",
       "28 SSC    DamagingM     0.22293863 0.18214851 0.26372876 0.0765023894394815\n",
       "29 MM     DamagingMC    0.24805765 0.20828111 0.28783419 0.0817983322017028\n",
       "30 SSC    DamagingMC    0.16625932 0.13079238 0.20172626 0.0817983322017028\n",
       "31 MM     DamagingMC2   0.28703814 0.24356651 0.33050977 0.0811033008964945\n",
       "32 SSC    DamagingMC2   0.20593484 0.16620177 0.24566791 0.0811033008964945\n",
       "33 MM     OtherMissense 0.12048515 0.09136339 0.14960690 NA                \n",
       "34 SSC    OtherMissense 0.12091587 0.09068329 0.15114845 NA                \n",
       "35 MM     Synonymous    0.12402883 0.09335873 0.15469892 NA                \n",
       "36 SSC    Synonymous    0.12280518 0.09330564 0.15230472 NA                \n",
       "   mean_diff_lower    mean_diff_upper    Gene     \n",
       "1  NA                 NA                 All genes\n",
       "2  NA                 NA                 All genes\n",
       "3  NA                 NA                 All genes\n",
       "4  NA                 NA                 All genes\n",
       "5  NA                 NA                 All genes\n",
       "6  NA                 NA                 All genes\n",
       "7  0.0249566763471755 0.0958025656394456 All genes\n",
       "8  0.0249566763471755 0.0958025656394456 All genes\n",
       "9  NA                 NA                 All genes\n",
       "10 NA                 NA                 All genes\n",
       "11 NA                 NA                 All genes\n",
       "12 NA                 NA                 All genes\n",
       "13 NA                 NA                 All genes\n",
       "14 NA                 NA                 All genes\n",
       "15 NA                 NA                 All genes\n",
       "16 NA                 NA                 All genes\n",
       "17 NA                 NA                 All genes\n",
       "18 NA                 NA                 All genes\n",
       "19 NA                 NA                 All genes\n",
       "20 NA                 NA                 All genes\n",
       "21 0.017680361675465  0.131761158117641  All genes\n",
       "22 0.017680361675465  0.131761158117641  All genes\n",
       "23 0.0374997132181427 0.124393635028837  All genes\n",
       "24 0.0374997132181427 0.124393635028837  All genes\n",
       "25 0.0238792822303671 0.137133040418461  All genes\n",
       "26 0.0238792822303671 0.137133040418461  All genes\n",
       "27 0.0163323040955573 0.136672474783406  All genes\n",
       "28 0.0163323040955573 0.136672474783406  All genes\n",
       "29 0.0285486462435022 0.135048018159903  All genes\n",
       "30 0.0285486462435022 0.135048018159903  All genes\n",
       "31 0.0222567442572848 0.139949857535704  All genes\n",
       "32 0.0222567442572848 0.139949857535704  All genes\n",
       "33 NA                 NA                 All genes\n",
       "34 NA                 NA                 All genes\n",
       "35 NA                 NA                 All genes\n",
       "36 NA                 NA                 All genes"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 18 × 10</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>category</th><th scope=col>RR</th><th scope=col>pvalue</th><th scope=col>MM</th><th scope=col>SSC</th><th scope=col>Gene</th><th scope=col>Lower</th><th scope=col>Upper</th><th scope=col>MM</th><th scope=col>SSC</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>All          </td><td>1.015</td><td>0.36750219</td><td>MM</td><td>SSC</td><td>All</td><td>0.946314210007404</td><td>Inf</td><td>719</td><td>664</td></tr>\n",
       "\t<tr><td>Nonsynonymous</td><td>1.146</td><td>0.06068071</td><td>MM</td><td>SSC</td><td>All</td><td>0.991864762164627</td><td>Inf</td><td>187</td><td>153</td></tr>\n",
       "\t<tr><td>Missense     </td><td>1.051</td><td>0.3108332 </td><td>MM</td><td>SSC</td><td>All</td><td>0.899748854658688</td><td>Inf</td><td>155</td><td>138</td></tr>\n",
       "\t<tr><td>LGD          </td><td>2.095</td><td>9.07e-06  </td><td>MM</td><td>SSC</td><td>All</td><td>1.55232208946951 </td><td>Inf</td><td>65 </td><td>29 </td></tr>\n",
       "\t<tr><td>DMisL        </td><td>1.102</td><td>0.23590122</td><td>MM</td><td>SSC</td><td>All</td><td>0.89180362641801 </td><td>Inf</td><td>87 </td><td>74 </td></tr>\n",
       "\t<tr><td>DMis         </td><td>1.517</td><td>0.03400915</td><td>MM</td><td>SSC</td><td>All</td><td>1.03904139279005 </td><td>Inf</td><td>34 </td><td>21 </td></tr>\n",
       "\t<tr><td>DMisL2       </td><td>1.143</td><td>0.15889109</td><td>MM</td><td>SSC</td><td>All</td><td>0.922002711183897</td><td>Inf</td><td>86 </td><td>70 </td></tr>\n",
       "\t<tr><td>DMisM        </td><td>1.086</td><td>0.25142529</td><td>MM</td><td>SSC</td><td>All</td><td>0.895992573688703</td><td>Inf</td><td>104</td><td>90 </td></tr>\n",
       "\t<tr><td>DMisMC       </td><td>1.192</td><td>0.1111023 </td><td>MM</td><td>SSC</td><td>All</td><td>0.944142160308373</td><td>Inf</td><td>75 </td><td>59 </td></tr>\n",
       "\t<tr><td>DMisMC2      </td><td>1.124</td><td>0.17644896</td><td>MM</td><td>SSC</td><td>All</td><td>0.919095701303745</td><td>Inf</td><td>97 </td><td>81 </td></tr>\n",
       "\t<tr><td>DamagingL    </td><td>1.387</td><td>0.00059199</td><td>MM</td><td>SSC</td><td>All</td><td>1.17120452064468 </td><td>Inf</td><td>152</td><td>103</td></tr>\n",
       "\t<tr><td>DamagingS    </td><td>1.864</td><td>1.84e-06  </td><td>MM</td><td>SSC</td><td>All</td><td>1.47922093198684 </td><td>Inf</td><td>99 </td><td>50 </td></tr>\n",
       "\t<tr><td>DamagingL2   </td><td>1.427</td><td>0.00024288</td><td>MM</td><td>SSC</td><td>All</td><td>1.20242128398786 </td><td>Inf</td><td>151</td><td>99 </td></tr>\n",
       "\t<tr><td>DamagingM    </td><td>1.345</td><td>0.00089926</td><td>MM</td><td>SSC</td><td>All</td><td>1.14761833255911 </td><td>Inf</td><td>169</td><td>118</td></tr>\n",
       "\t<tr><td>DamagingMC   </td><td>1.489</td><td>0.00010054</td><td>MM</td><td>SSC</td><td>All</td><td>1.24291430217307 </td><td>Inf</td><td>140</td><td>88 </td></tr>\n",
       "\t<tr><td>DamagingMC2  </td><td>1.393</td><td>0.00034555</td><td>MM</td><td>SSC</td><td>All</td><td>1.1826688880035  </td><td>Inf</td><td>162</td><td>109</td></tr>\n",
       "\t<tr><td>OtherMissense</td><td>0.992</td><td>0.5510505 </td><td>MM</td><td>SSC</td><td>All</td><td>0.784862516565366</td><td>Inf</td><td>68 </td><td>64 </td></tr>\n",
       "\t<tr><td>Synonymous   </td><td>1.018</td><td>0.4740654 </td><td>MM</td><td>SSC</td><td>All</td><td>0.807887116097002</td><td>Inf</td><td>70 </td><td>65 </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 18 × 10\n",
       "\\begin{tabular}{llllllllll}\n",
       " category & RR & pvalue & MM & SSC & Gene & Lower & Upper & MM & SSC\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t All           & 1.015 & 0.36750219 & MM & SSC & All & 0.946314210007404 & Inf & 719 & 664\\\\\n",
       "\t Nonsynonymous & 1.146 & 0.06068071 & MM & SSC & All & 0.991864762164627 & Inf & 187 & 153\\\\\n",
       "\t Missense      & 1.051 & 0.3108332  & MM & SSC & All & 0.899748854658688 & Inf & 155 & 138\\\\\n",
       "\t LGD           & 2.095 & 9.07e-06   & MM & SSC & All & 1.55232208946951  & Inf & 65  & 29 \\\\\n",
       "\t DMisL         & 1.102 & 0.23590122 & MM & SSC & All & 0.89180362641801  & Inf & 87  & 74 \\\\\n",
       "\t DMis          & 1.517 & 0.03400915 & MM & SSC & All & 1.03904139279005  & Inf & 34  & 21 \\\\\n",
       "\t DMisL2        & 1.143 & 0.15889109 & MM & SSC & All & 0.922002711183897 & Inf & 86  & 70 \\\\\n",
       "\t DMisM         & 1.086 & 0.25142529 & MM & SSC & All & 0.895992573688703 & Inf & 104 & 90 \\\\\n",
       "\t DMisMC        & 1.192 & 0.1111023  & MM & SSC & All & 0.944142160308373 & Inf & 75  & 59 \\\\\n",
       "\t DMisMC2       & 1.124 & 0.17644896 & MM & SSC & All & 0.919095701303745 & Inf & 97  & 81 \\\\\n",
       "\t DamagingL     & 1.387 & 0.00059199 & MM & SSC & All & 1.17120452064468  & Inf & 152 & 103\\\\\n",
       "\t DamagingS     & 1.864 & 1.84e-06   & MM & SSC & All & 1.47922093198684  & Inf & 99  & 50 \\\\\n",
       "\t DamagingL2    & 1.427 & 0.00024288 & MM & SSC & All & 1.20242128398786  & Inf & 151 & 99 \\\\\n",
       "\t DamagingM     & 1.345 & 0.00089926 & MM & SSC & All & 1.14761833255911  & Inf & 169 & 118\\\\\n",
       "\t DamagingMC    & 1.489 & 0.00010054 & MM & SSC & All & 1.24291430217307  & Inf & 140 & 88 \\\\\n",
       "\t DamagingMC2   & 1.393 & 0.00034555 & MM & SSC & All & 1.1826688880035   & Inf & 162 & 109\\\\\n",
       "\t OtherMissense & 0.992 & 0.5510505  & MM & SSC & All & 0.784862516565366 & Inf & 68  & 64 \\\\\n",
       "\t Synonymous    & 1.018 & 0.4740654  & MM & SSC & All & 0.807887116097002 & Inf & 70  & 65 \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 18 × 10\n",
       "\n",
       "| category &lt;chr&gt; | RR &lt;chr&gt; | pvalue &lt;chr&gt; | MM &lt;chr&gt; | SSC &lt;chr&gt; | Gene &lt;chr&gt; | Lower &lt;chr&gt; | Upper &lt;chr&gt; | MM &lt;chr&gt; | SSC &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|\n",
       "| All           | 1.015 | 0.36750219 | MM | SSC | All | 0.946314210007404 | Inf | 719 | 664 |\n",
       "| Nonsynonymous | 1.146 | 0.06068071 | MM | SSC | All | 0.991864762164627 | Inf | 187 | 153 |\n",
       "| Missense      | 1.051 | 0.3108332  | MM | SSC | All | 0.899748854658688 | Inf | 155 | 138 |\n",
       "| LGD           | 2.095 | 9.07e-06   | MM | SSC | All | 1.55232208946951  | Inf | 65  | 29  |\n",
       "| DMisL         | 1.102 | 0.23590122 | MM | SSC | All | 0.89180362641801  | Inf | 87  | 74  |\n",
       "| DMis          | 1.517 | 0.03400915 | MM | SSC | All | 1.03904139279005  | Inf | 34  | 21  |\n",
       "| DMisL2        | 1.143 | 0.15889109 | MM | SSC | All | 0.922002711183897 | Inf | 86  | 70  |\n",
       "| DMisM         | 1.086 | 0.25142529 | MM | SSC | All | 0.895992573688703 | Inf | 104 | 90  |\n",
       "| DMisMC        | 1.192 | 0.1111023  | MM | SSC | All | 0.944142160308373 | Inf | 75  | 59  |\n",
       "| DMisMC2       | 1.124 | 0.17644896 | MM | SSC | All | 0.919095701303745 | Inf | 97  | 81  |\n",
       "| DamagingL     | 1.387 | 0.00059199 | MM | SSC | All | 1.17120452064468  | Inf | 152 | 103 |\n",
       "| DamagingS     | 1.864 | 1.84e-06   | MM | SSC | All | 1.47922093198684  | Inf | 99  | 50  |\n",
       "| DamagingL2    | 1.427 | 0.00024288 | MM | SSC | All | 1.20242128398786  | Inf | 151 | 99  |\n",
       "| DamagingM     | 1.345 | 0.00089926 | MM | SSC | All | 1.14761833255911  | Inf | 169 | 118 |\n",
       "| DamagingMC    | 1.489 | 0.00010054 | MM | SSC | All | 1.24291430217307  | Inf | 140 | 88  |\n",
       "| DamagingMC2   | 1.393 | 0.00034555 | MM | SSC | All | 1.1826688880035   | Inf | 162 | 109 |\n",
       "| OtherMissense | 0.992 | 0.5510505  | MM | SSC | All | 0.784862516565366 | Inf | 68  | 64  |\n",
       "| Synonymous    | 1.018 | 0.4740654  | MM | SSC | All | 0.807887116097002 | Inf | 70  | 65  |\n",
       "\n"
      ],
      "text/plain": [
       "   category      RR    pvalue     MM SSC Gene Lower             Upper MM  SSC\n",
       "1  All           1.015 0.36750219 MM SSC All  0.946314210007404 Inf   719 664\n",
       "2  Nonsynonymous 1.146 0.06068071 MM SSC All  0.991864762164627 Inf   187 153\n",
       "3  Missense      1.051 0.3108332  MM SSC All  0.899748854658688 Inf   155 138\n",
       "4  LGD           2.095 9.07e-06   MM SSC All  1.55232208946951  Inf   65  29 \n",
       "5  DMisL         1.102 0.23590122 MM SSC All  0.89180362641801  Inf   87  74 \n",
       "6  DMis          1.517 0.03400915 MM SSC All  1.03904139279005  Inf   34  21 \n",
       "7  DMisL2        1.143 0.15889109 MM SSC All  0.922002711183897 Inf   86  70 \n",
       "8  DMisM         1.086 0.25142529 MM SSC All  0.895992573688703 Inf   104 90 \n",
       "9  DMisMC        1.192 0.1111023  MM SSC All  0.944142160308373 Inf   75  59 \n",
       "10 DMisMC2       1.124 0.17644896 MM SSC All  0.919095701303745 Inf   97  81 \n",
       "11 DamagingL     1.387 0.00059199 MM SSC All  1.17120452064468  Inf   152 103\n",
       "12 DamagingS     1.864 1.84e-06   MM SSC All  1.47922093198684  Inf   99  50 \n",
       "13 DamagingL2    1.427 0.00024288 MM SSC All  1.20242128398786  Inf   151 99 \n",
       "14 DamagingM     1.345 0.00089926 MM SSC All  1.14761833255911  Inf   169 118\n",
       "15 DamagingMC    1.489 0.00010054 MM SSC All  1.24291430217307  Inf   140 88 \n",
       "16 DamagingMC2   1.393 0.00034555 MM SSC All  1.1826688880035   Inf   162 109\n",
       "17 OtherMissense 0.992 0.5510505  MM SSC All  0.784862516565366 Inf   68  64 \n",
       "18 Synonymous    1.018 0.4740654  MM SSC All  0.807887116097002 Inf   70  65 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 36 × 9</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>cohort</th><th scope=col>category</th><th scope=col>mean</th><th scope=col>lower</th><th scope=col>upper</th><th scope=col>mean_diff</th><th scope=col>mean_diff_lower</th><th scope=col>mean_diff_upper</th><th scope=col>Gene</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>MM </td><td>All          </td><td>1.27395323</td><td>1.18086405</td><td>1.36704240</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>All          </td><td>1.25450214</td><td>1.15469614</td><td>1.35430813</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>Nonsynonymous</td><td>0.33133415</td><td>0.28467739</td><td>0.37799091</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>Nonsynonymous</td><td>0.28906450</td><td>0.24442039</td><td>0.33370860</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>Missense     </td><td>0.27463526</td><td>0.23174229</td><td>0.31752822</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>Missense     </td><td>0.26072484</td><td>0.21812915</td><td>0.30332054</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>LGD          </td><td>0.11516962</td><td>0.08656898</td><td>0.14377027</td><td>0.0603796209933105</td><td>0.0249566763471755</td><td>0.0958025656394456</td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>LGD          </td><td>0.05479000</td><td>0.03384624</td><td>0.07573377</td><td>0.0603796209933105</td><td>0.0249566763471755</td><td>0.0958025656394456</td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DMisL        </td><td>0.15415011</td><td>0.12278819</td><td>0.18551203</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DMisL        </td><td>0.13980897</td><td>0.10865755</td><td>0.17096040</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DMis         </td><td>0.06024257</td><td>0.03979874</td><td>0.08068640</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DMis         </td><td>0.03967552</td><td>0.02291462</td><td>0.05643641</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DMisL2       </td><td>0.15237827</td><td>0.12116656</td><td>0.18358999</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DMisL2       </td><td>0.13225173</td><td>0.10181715</td><td>0.16268631</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DMisM        </td><td>0.18427140</td><td>0.14981844</td><td>0.21872436</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DMisM        </td><td>0.17003794</td><td>0.13545440</td><td>0.20462148</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DMisMC       </td><td>0.13288803</td><td>0.10382803</td><td>0.16194803</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DMisMC       </td><td>0.11146932</td><td>0.08315469</td><td>0.13978394</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DMisMC2      </td><td>0.17186852</td><td>0.13870134</td><td>0.20503569</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DMisMC2      </td><td>0.15303415</td><td>0.11984811</td><td>0.18622018</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DamagingL    </td><td>0.26931974</td><td>0.22696776</td><td>0.31167171</td><td>0.0747207598965531</td><td>0.017680361675465 </td><td>0.131761158117641 </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DamagingL    </td><td>0.19459898</td><td>0.15632169</td><td>0.23287626</td><td>0.0747207598965531</td><td>0.017680361675465 </td><td>0.131761158117641 </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DamagingS    </td><td>0.17541220</td><td>0.14125679</td><td>0.20956761</td><td>0.0809466741234898</td><td>0.0374997132181427</td><td>0.124393635028837 </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DamagingS    </td><td>0.09446552</td><td>0.06755928</td><td>0.12137176</td><td>0.0809466741234898</td><td>0.0374997132181427</td><td>0.124393635028837 </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DamagingL2   </td><td>0.26754790</td><td>0.22528289</td><td>0.30981290</td><td>0.080506161324414 </td><td>0.0238792822303671</td><td>0.137133040418461 </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DamagingL2   </td><td>0.18704173</td><td>0.14928703</td><td>0.22479644</td><td>0.080506161324414 </td><td>0.0238792822303671</td><td>0.137133040418461 </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DamagingM    </td><td>0.29944102</td><td>0.25514149</td><td>0.34374056</td><td>0.0765023894394815</td><td>0.0163323040955573</td><td>0.136672474783406 </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DamagingM    </td><td>0.22293863</td><td>0.18214851</td><td>0.26372876</td><td>0.0765023894394815</td><td>0.0163323040955573</td><td>0.136672474783406 </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DamagingMC   </td><td>0.24805765</td><td>0.20828111</td><td>0.28783419</td><td>0.0817983322017028</td><td>0.0285486462435022</td><td>0.135048018159903 </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DamagingMC   </td><td>0.16625932</td><td>0.13079238</td><td>0.20172626</td><td>0.0817983322017028</td><td>0.0285486462435022</td><td>0.135048018159903 </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DamagingMC2  </td><td>0.28703814</td><td>0.24356651</td><td>0.33050977</td><td>0.0811033008964945</td><td>0.0222567442572848</td><td>0.139949857535704 </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DamagingMC2  </td><td>0.20593484</td><td>0.16620177</td><td>0.24566791</td><td>0.0811033008964945</td><td>0.0222567442572848</td><td>0.139949857535704 </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>OtherMissense</td><td>0.12048515</td><td>0.09136339</td><td>0.14960690</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>OtherMissense</td><td>0.12091587</td><td>0.09068329</td><td>0.15114845</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>Synonymous   </td><td>0.12402883</td><td>0.09335873</td><td>0.15469892</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>Synonymous   </td><td>0.12280518</td><td>0.09330564</td><td>0.15230472</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 36 × 9\n",
       "\\begin{tabular}{lllllllll}\n",
       " cohort & category & mean & lower & upper & mean\\_diff & mean\\_diff\\_lower & mean\\_diff\\_upper & Gene\\\\\n",
       " <fct> & <chr> & <dbl> & <dbl> & <dbl> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t MM  & All           & 1.27395323 & 1.18086405 & 1.36704240 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t SSC & All           & 1.25450214 & 1.15469614 & 1.35430813 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t MM  & Nonsynonymous & 0.33133415 & 0.28467739 & 0.37799091 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t SSC & Nonsynonymous & 0.28906450 & 0.24442039 & 0.33370860 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t MM  & Missense      & 0.27463526 & 0.23174229 & 0.31752822 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t SSC & Missense      & 0.26072484 & 0.21812915 & 0.30332054 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t MM  & LGD           & 0.11516962 & 0.08656898 & 0.14377027 & 0.0603796209933105 & 0.0249566763471755 & 0.0958025656394456 & All genes\\\\\n",
       "\t SSC & LGD           & 0.05479000 & 0.03384624 & 0.07573377 & 0.0603796209933105 & 0.0249566763471755 & 0.0958025656394456 & All genes\\\\\n",
       "\t MM  & DMisL         & 0.15415011 & 0.12278819 & 0.18551203 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t SSC & DMisL         & 0.13980897 & 0.10865755 & 0.17096040 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t MM  & DMis          & 0.06024257 & 0.03979874 & 0.08068640 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t SSC & DMis          & 0.03967552 & 0.02291462 & 0.05643641 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t MM  & DMisL2        & 0.15237827 & 0.12116656 & 0.18358999 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t SSC & DMisL2        & 0.13225173 & 0.10181715 & 0.16268631 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t MM  & DMisM         & 0.18427140 & 0.14981844 & 0.21872436 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t SSC & DMisM         & 0.17003794 & 0.13545440 & 0.20462148 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t MM  & DMisMC        & 0.13288803 & 0.10382803 & 0.16194803 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t SSC & DMisMC        & 0.11146932 & 0.08315469 & 0.13978394 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t MM  & DMisMC2       & 0.17186852 & 0.13870134 & 0.20503569 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t SSC & DMisMC2       & 0.15303415 & 0.11984811 & 0.18622018 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t MM  & DamagingL     & 0.26931974 & 0.22696776 & 0.31167171 & 0.0747207598965531 & 0.017680361675465  & 0.131761158117641  & All genes\\\\\n",
       "\t SSC & DamagingL     & 0.19459898 & 0.15632169 & 0.23287626 & 0.0747207598965531 & 0.017680361675465  & 0.131761158117641  & All genes\\\\\n",
       "\t MM  & DamagingS     & 0.17541220 & 0.14125679 & 0.20956761 & 0.0809466741234898 & 0.0374997132181427 & 0.124393635028837  & All genes\\\\\n",
       "\t SSC & DamagingS     & 0.09446552 & 0.06755928 & 0.12137176 & 0.0809466741234898 & 0.0374997132181427 & 0.124393635028837  & All genes\\\\\n",
       "\t MM  & DamagingL2    & 0.26754790 & 0.22528289 & 0.30981290 & 0.080506161324414  & 0.0238792822303671 & 0.137133040418461  & All genes\\\\\n",
       "\t SSC & DamagingL2    & 0.18704173 & 0.14928703 & 0.22479644 & 0.080506161324414  & 0.0238792822303671 & 0.137133040418461  & All genes\\\\\n",
       "\t MM  & DamagingM     & 0.29944102 & 0.25514149 & 0.34374056 & 0.0765023894394815 & 0.0163323040955573 & 0.136672474783406  & All genes\\\\\n",
       "\t SSC & DamagingM     & 0.22293863 & 0.18214851 & 0.26372876 & 0.0765023894394815 & 0.0163323040955573 & 0.136672474783406  & All genes\\\\\n",
       "\t MM  & DamagingMC    & 0.24805765 & 0.20828111 & 0.28783419 & 0.0817983322017028 & 0.0285486462435022 & 0.135048018159903  & All genes\\\\\n",
       "\t SSC & DamagingMC    & 0.16625932 & 0.13079238 & 0.20172626 & 0.0817983322017028 & 0.0285486462435022 & 0.135048018159903  & All genes\\\\\n",
       "\t MM  & DamagingMC2   & 0.28703814 & 0.24356651 & 0.33050977 & 0.0811033008964945 & 0.0222567442572848 & 0.139949857535704  & All genes\\\\\n",
       "\t SSC & DamagingMC2   & 0.20593484 & 0.16620177 & 0.24566791 & 0.0811033008964945 & 0.0222567442572848 & 0.139949857535704  & All genes\\\\\n",
       "\t MM  & OtherMissense & 0.12048515 & 0.09136339 & 0.14960690 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t SSC & OtherMissense & 0.12091587 & 0.09068329 & 0.15114845 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t MM  & Synonymous    & 0.12402883 & 0.09335873 & 0.15469892 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t SSC & Synonymous    & 0.12280518 & 0.09330564 & 0.15230472 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 36 × 9\n",
       "\n",
       "| cohort &lt;fct&gt; | category &lt;chr&gt; | mean &lt;dbl&gt; | lower &lt;dbl&gt; | upper &lt;dbl&gt; | mean_diff &lt;chr&gt; | mean_diff_lower &lt;chr&gt; | mean_diff_upper &lt;chr&gt; | Gene &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| MM  | All           | 1.27395323 | 1.18086405 | 1.36704240 | NA                 | NA                 | NA                 | All genes |\n",
       "| SSC | All           | 1.25450214 | 1.15469614 | 1.35430813 | NA                 | NA                 | NA                 | All genes |\n",
       "| MM  | Nonsynonymous | 0.33133415 | 0.28467739 | 0.37799091 | NA                 | NA                 | NA                 | All genes |\n",
       "| SSC | Nonsynonymous | 0.28906450 | 0.24442039 | 0.33370860 | NA                 | NA                 | NA                 | All genes |\n",
       "| MM  | Missense      | 0.27463526 | 0.23174229 | 0.31752822 | NA                 | NA                 | NA                 | All genes |\n",
       "| SSC | Missense      | 0.26072484 | 0.21812915 | 0.30332054 | NA                 | NA                 | NA                 | All genes |\n",
       "| MM  | LGD           | 0.11516962 | 0.08656898 | 0.14377027 | 0.0603796209933105 | 0.0249566763471755 | 0.0958025656394456 | All genes |\n",
       "| SSC | LGD           | 0.05479000 | 0.03384624 | 0.07573377 | 0.0603796209933105 | 0.0249566763471755 | 0.0958025656394456 | All genes |\n",
       "| MM  | DMisL         | 0.15415011 | 0.12278819 | 0.18551203 | NA                 | NA                 | NA                 | All genes |\n",
       "| SSC | DMisL         | 0.13980897 | 0.10865755 | 0.17096040 | NA                 | NA                 | NA                 | All genes |\n",
       "| MM  | DMis          | 0.06024257 | 0.03979874 | 0.08068640 | NA                 | NA                 | NA                 | All genes |\n",
       "| SSC | DMis          | 0.03967552 | 0.02291462 | 0.05643641 | NA                 | NA                 | NA                 | All genes |\n",
       "| MM  | DMisL2        | 0.15237827 | 0.12116656 | 0.18358999 | NA                 | NA                 | NA                 | All genes |\n",
       "| SSC | DMisL2        | 0.13225173 | 0.10181715 | 0.16268631 | NA                 | NA                 | NA                 | All genes |\n",
       "| MM  | DMisM         | 0.18427140 | 0.14981844 | 0.21872436 | NA                 | NA                 | NA                 | All genes |\n",
       "| SSC | DMisM         | 0.17003794 | 0.13545440 | 0.20462148 | NA                 | NA                 | NA                 | All genes |\n",
       "| MM  | DMisMC        | 0.13288803 | 0.10382803 | 0.16194803 | NA                 | NA                 | NA                 | All genes |\n",
       "| SSC | DMisMC        | 0.11146932 | 0.08315469 | 0.13978394 | NA                 | NA                 | NA                 | All genes |\n",
       "| MM  | DMisMC2       | 0.17186852 | 0.13870134 | 0.20503569 | NA                 | NA                 | NA                 | All genes |\n",
       "| SSC | DMisMC2       | 0.15303415 | 0.11984811 | 0.18622018 | NA                 | NA                 | NA                 | All genes |\n",
       "| MM  | DamagingL     | 0.26931974 | 0.22696776 | 0.31167171 | 0.0747207598965531 | 0.017680361675465  | 0.131761158117641  | All genes |\n",
       "| SSC | DamagingL     | 0.19459898 | 0.15632169 | 0.23287626 | 0.0747207598965531 | 0.017680361675465  | 0.131761158117641  | All genes |\n",
       "| MM  | DamagingS     | 0.17541220 | 0.14125679 | 0.20956761 | 0.0809466741234898 | 0.0374997132181427 | 0.124393635028837  | All genes |\n",
       "| SSC | DamagingS     | 0.09446552 | 0.06755928 | 0.12137176 | 0.0809466741234898 | 0.0374997132181427 | 0.124393635028837  | All genes |\n",
       "| MM  | DamagingL2    | 0.26754790 | 0.22528289 | 0.30981290 | 0.080506161324414  | 0.0238792822303671 | 0.137133040418461  | All genes |\n",
       "| SSC | DamagingL2    | 0.18704173 | 0.14928703 | 0.22479644 | 0.080506161324414  | 0.0238792822303671 | 0.137133040418461  | All genes |\n",
       "| MM  | DamagingM     | 0.29944102 | 0.25514149 | 0.34374056 | 0.0765023894394815 | 0.0163323040955573 | 0.136672474783406  | All genes |\n",
       "| SSC | DamagingM     | 0.22293863 | 0.18214851 | 0.26372876 | 0.0765023894394815 | 0.0163323040955573 | 0.136672474783406  | All genes |\n",
       "| MM  | DamagingMC    | 0.24805765 | 0.20828111 | 0.28783419 | 0.0817983322017028 | 0.0285486462435022 | 0.135048018159903  | All genes |\n",
       "| SSC | DamagingMC    | 0.16625932 | 0.13079238 | 0.20172626 | 0.0817983322017028 | 0.0285486462435022 | 0.135048018159903  | All genes |\n",
       "| MM  | DamagingMC2   | 0.28703814 | 0.24356651 | 0.33050977 | 0.0811033008964945 | 0.0222567442572848 | 0.139949857535704  | All genes |\n",
       "| SSC | DamagingMC2   | 0.20593484 | 0.16620177 | 0.24566791 | 0.0811033008964945 | 0.0222567442572848 | 0.139949857535704  | All genes |\n",
       "| MM  | OtherMissense | 0.12048515 | 0.09136339 | 0.14960690 | NA                 | NA                 | NA                 | All genes |\n",
       "| SSC | OtherMissense | 0.12091587 | 0.09068329 | 0.15114845 | NA                 | NA                 | NA                 | All genes |\n",
       "| MM  | Synonymous    | 0.12402883 | 0.09335873 | 0.15469892 | NA                 | NA                 | NA                 | All genes |\n",
       "| SSC | Synonymous    | 0.12280518 | 0.09330564 | 0.15230472 | NA                 | NA                 | NA                 | All genes |\n",
       "\n"
      ],
      "text/plain": [
       "   cohort category      mean       lower      upper      mean_diff         \n",
       "1  MM     All           1.27395323 1.18086405 1.36704240 NA                \n",
       "2  SSC    All           1.25450214 1.15469614 1.35430813 NA                \n",
       "3  MM     Nonsynonymous 0.33133415 0.28467739 0.37799091 NA                \n",
       "4  SSC    Nonsynonymous 0.28906450 0.24442039 0.33370860 NA                \n",
       "5  MM     Missense      0.27463526 0.23174229 0.31752822 NA                \n",
       "6  SSC    Missense      0.26072484 0.21812915 0.30332054 NA                \n",
       "7  MM     LGD           0.11516962 0.08656898 0.14377027 0.0603796209933105\n",
       "8  SSC    LGD           0.05479000 0.03384624 0.07573377 0.0603796209933105\n",
       "9  MM     DMisL         0.15415011 0.12278819 0.18551203 NA                \n",
       "10 SSC    DMisL         0.13980897 0.10865755 0.17096040 NA                \n",
       "11 MM     DMis          0.06024257 0.03979874 0.08068640 NA                \n",
       "12 SSC    DMis          0.03967552 0.02291462 0.05643641 NA                \n",
       "13 MM     DMisL2        0.15237827 0.12116656 0.18358999 NA                \n",
       "14 SSC    DMisL2        0.13225173 0.10181715 0.16268631 NA                \n",
       "15 MM     DMisM         0.18427140 0.14981844 0.21872436 NA                \n",
       "16 SSC    DMisM         0.17003794 0.13545440 0.20462148 NA                \n",
       "17 MM     DMisMC        0.13288803 0.10382803 0.16194803 NA                \n",
       "18 SSC    DMisMC        0.11146932 0.08315469 0.13978394 NA                \n",
       "19 MM     DMisMC2       0.17186852 0.13870134 0.20503569 NA                \n",
       "20 SSC    DMisMC2       0.15303415 0.11984811 0.18622018 NA                \n",
       "21 MM     DamagingL     0.26931974 0.22696776 0.31167171 0.0747207598965531\n",
       "22 SSC    DamagingL     0.19459898 0.15632169 0.23287626 0.0747207598965531\n",
       "23 MM     DamagingS     0.17541220 0.14125679 0.20956761 0.0809466741234898\n",
       "24 SSC    DamagingS     0.09446552 0.06755928 0.12137176 0.0809466741234898\n",
       "25 MM     DamagingL2    0.26754790 0.22528289 0.30981290 0.080506161324414 \n",
       "26 SSC    DamagingL2    0.18704173 0.14928703 0.22479644 0.080506161324414 \n",
       "27 MM     DamagingM     0.29944102 0.25514149 0.34374056 0.0765023894394815\n",
       "28 SSC    DamagingM     0.22293863 0.18214851 0.26372876 0.0765023894394815\n",
       "29 MM     DamagingMC    0.24805765 0.20828111 0.28783419 0.0817983322017028\n",
       "30 SSC    DamagingMC    0.16625932 0.13079238 0.20172626 0.0817983322017028\n",
       "31 MM     DamagingMC2   0.28703814 0.24356651 0.33050977 0.0811033008964945\n",
       "32 SSC    DamagingMC2   0.20593484 0.16620177 0.24566791 0.0811033008964945\n",
       "33 MM     OtherMissense 0.12048515 0.09136339 0.14960690 NA                \n",
       "34 SSC    OtherMissense 0.12091587 0.09068329 0.15114845 NA                \n",
       "35 MM     Synonymous    0.12402883 0.09335873 0.15469892 NA                \n",
       "36 SSC    Synonymous    0.12280518 0.09330564 0.15230472 NA                \n",
       "   mean_diff_lower    mean_diff_upper    Gene     \n",
       "1  NA                 NA                 All genes\n",
       "2  NA                 NA                 All genes\n",
       "3  NA                 NA                 All genes\n",
       "4  NA                 NA                 All genes\n",
       "5  NA                 NA                 All genes\n",
       "6  NA                 NA                 All genes\n",
       "7  0.0249566763471755 0.0958025656394456 All genes\n",
       "8  0.0249566763471755 0.0958025656394456 All genes\n",
       "9  NA                 NA                 All genes\n",
       "10 NA                 NA                 All genes\n",
       "11 NA                 NA                 All genes\n",
       "12 NA                 NA                 All genes\n",
       "13 NA                 NA                 All genes\n",
       "14 NA                 NA                 All genes\n",
       "15 NA                 NA                 All genes\n",
       "16 NA                 NA                 All genes\n",
       "17 NA                 NA                 All genes\n",
       "18 NA                 NA                 All genes\n",
       "19 NA                 NA                 All genes\n",
       "20 NA                 NA                 All genes\n",
       "21 0.017680361675465  0.131761158117641  All genes\n",
       "22 0.017680361675465  0.131761158117641  All genes\n",
       "23 0.0374997132181427 0.124393635028837  All genes\n",
       "24 0.0374997132181427 0.124393635028837  All genes\n",
       "25 0.0238792822303671 0.137133040418461  All genes\n",
       "26 0.0238792822303671 0.137133040418461  All genes\n",
       "27 0.0163323040955573 0.136672474783406  All genes\n",
       "28 0.0163323040955573 0.136672474783406  All genes\n",
       "29 0.0285486462435022 0.135048018159903  All genes\n",
       "30 0.0285486462435022 0.135048018159903  All genes\n",
       "31 0.0222567442572848 0.139949857535704  All genes\n",
       "32 0.0222567442572848 0.139949857535704  All genes\n",
       "33 NA                 NA                 All genes\n",
       "34 NA                 NA                 All genes\n",
       "35 NA                 NA                 All genes\n",
       "36 NA                 NA                 All genes"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "\n",
    "df_tmp\n",
    "colnames(df_rr) <- c(\"category\",\"RR\", \"pvalue\", 'MM', \"SSC\", \"Gene\",\"Lower\",\"Upper\", \"MM\", \"SSC\")\n",
    "df_rr\n",
    "df_all_stat"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 193,
   "id": "8ec5b73e",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ERROR while rich displaying an object: \u001b[1m\u001b[33mError\u001b[39m in `geom_errorbar()`:\u001b[22m\n",
      "\u001b[1m\u001b[22m\u001b[33m!\u001b[39m Problem while computing aesthetics.\n",
      "\u001b[36mi\u001b[39m Error occurred in the 1st layer.\n",
      "\u001b[1mCaused by error in `FUN()`:\u001b[22m\n",
      "\u001b[33m!\u001b[39m object 'ci' not found\n",
      "\n",
      "Traceback:\n",
      "1. tryCatch(withCallingHandlers({\n",
      " .     if (!mime %in% names(repr::mime2repr)) \n",
      " .         stop(\"No repr_* for mimetype \", mime, \" in repr::mime2repr\")\n",
      " .     rpr <- repr::mime2repr[[mime]](obj)\n",
      " .     if (is.null(rpr)) \n",
      " .         return(NULL)\n",
      " .     prepare_content(is.raw(rpr), rpr)\n",
      " . }, error = error_handler), error = outer_handler)\n",
      "2. tryCatchList(expr, classes, parentenv, handlers)\n",
      "3. tryCatchOne(expr, names, parentenv, handlers[[1L]])\n",
      "4. doTryCatch(return(expr), name, parentenv, handler)\n",
      "5. withCallingHandlers({\n",
      " .     if (!mime %in% names(repr::mime2repr)) \n",
      " .         stop(\"No repr_* for mimetype \", mime, \" in repr::mime2repr\")\n",
      " .     rpr <- repr::mime2repr[[mime]](obj)\n",
      " .     if (is.null(rpr)) \n",
      " .         return(NULL)\n",
      " .     prepare_content(is.raw(rpr), rpr)\n",
      " . }, error = error_handler)\n",
      "6. repr::mime2repr[[mime]](obj)\n",
      "7. repr_text.default(obj)\n",
      "8. paste(capture.output(print(obj)), collapse = \"\\n\")\n",
      "9. capture.output(print(obj))\n",
      "10. withVisible(...elt(i))\n",
      "11. print(obj)\n",
      "12. print.ggplot(obj)\n",
      "13. ggplot_build(x)\n",
      "14. ggplot_build.ggplot(x)\n",
      "15. by_layer(function(l, d) l$compute_aesthetics(d, plot), layers, \n",
      "  .     data, \"computing aesthetics\")\n",
      "16. try_fetch(for (i in seq_along(data)) {\n",
      "  .     out[[i]] <- f(l = layers[[i]], d = data[[i]])\n",
      "  . }, error = function(cnd) {\n",
      "  .     cli::cli_abort(c(\"Problem while {step}.\", i = \"Error occurred in the {ordinal(i)} layer.\"), \n",
      "  .         call = layers[[i]]$constructor, parent = cnd)\n",
      "  . })\n",
      "17. tryCatch(withCallingHandlers(expr, error = function(cnd) {\n",
      "  .     {\n",
      "  .         .__handler_frame__. <- TRUE\n",
      "  .         .__setup_frame__. <- frame\n",
      "  .     }\n",
      "  .     out <- handlers[[1L]](cnd)\n",
      "  .     if (!inherits(out, \"rlang_zap\")) \n",
      "  .         throw(out)\n",
      "  . }), stackOverflowError = handlers[[1L]])\n",
      "18. tryCatchList(expr, classes, parentenv, handlers)\n",
      "19. tryCatchOne(expr, names, parentenv, handlers[[1L]])\n",
      "20. doTryCatch(return(expr), name, parentenv, handler)\n",
      "21. withCallingHandlers(expr, error = function(cnd) {\n",
      "  .     {\n",
      "  .         .__handler_frame__. <- TRUE\n",
      "  .         .__setup_frame__. <- frame\n",
      "  .     }\n",
      "  .     out <- handlers[[1L]](cnd)\n",
      "  .     if (!inherits(out, \"rlang_zap\")) \n",
      "  .         throw(out)\n",
      "  . })\n",
      "22. f(l = layers[[i]], d = data[[i]])\n",
      "23. l$compute_aesthetics(d, plot)\n",
      "24. compute_aesthetics(..., self = self)\n",
      "25. lapply(aesthetics, eval_tidy, data = data, env = env)\n",
      "26. FUN(X[[i]], ...)\n",
      "27. .handleSimpleError(function (cnd) \n",
      "  . {\n",
      "  .     {\n",
      "  .         .__handler_frame__. <- TRUE\n",
      "  .         .__setup_frame__. <- frame\n",
      "  .     }\n",
      "  .     out <- handlers[[1L]](cnd)\n",
      "  .     if (!inherits(out, \"rlang_zap\")) \n",
      "  .         throw(out)\n",
      "  . }, \"object 'ci' not found\", base::quote(FUN(X[[i]], ...)))\n",
      "28. h(simpleError(msg, call))\n",
      "29. handlers[[1L]](cnd)\n",
      "30. cli::cli_abort(c(\"Problem while {step}.\", i = \"Error occurred in the {ordinal(i)} layer.\"), \n",
      "  .     call = layers[[i]]$constructor, parent = cnd)\n",
      "31. rlang::abort(message, ..., call = call, use_cli_format = TRUE, \n",
      "  .     .frame = .frame)\n",
      "32. signal_abort(cnd, .file)\n"
     ]
    }
   ],
   "source": [
    "figure_a = c('LGD','DMis','DamagingS','OtherMissense', 'Synonymous')\n",
    "label_a = c('LGD','DMis','LGD + DMis','Other missense', 'Synonymous')\n",
    "\n",
    "df_all_stat$category <- factor(df_all_stat$category, levels = figure_a)\n",
    "df_all_stat$cohort <- factor(df_all_stat$cohort, levels = c(\"SSC\", \"MM\"))\n",
    "\n",
    "p_fa <- ggplot(df_all_stat, aes(category, as.numeric(mean))) +\n",
    "  geom_errorbar(aes(x = category, ymin=mean-ci, ymax = mean+ci,group = cohort,color = category),\n",
    "                position = position_dodge(0.5), width = 0.2,size = .3) + \n",
    "  geom_point(data = df_all_stat,\n",
    "             aes(category, as.numeric(mean), group = cohort, color = category, shape = cohort),\n",
    "             size =2.5,\n",
    "             position =position_dodge(0.5)) + \n",
    "  scale_color_manual(values = c(\"#642E44\", \"#A65D7C\",  \"#E5A3B1\", \"#D94929\", \"#A62F14\", \"#C2C0A6\", \"#889C9B\", \"#B2BEBF\")) +\n",
    "  scale_x_discrete(breaks=figure_a,labels=label_a) + \n",
    "  scale_y_continuous(sec.axis = sec_axis(~ . / (100000000 / 2) * total_size, name = 'Theoretical Variant Rate (per child)'), limits= c(0,0.5)) +\n",
    "  labs(fill='Cohort', x='', y='Variant rate (x 10e-8)') +\n",
    "  theme_classic() +\n",
    "  theme(legend.position='bottom', \n",
    "        aspect.ratio = 0.6,\n",
    "        axis.title.y.right = element_text( angle = 90,size = 12),\n",
    "        axis.text.x = element_text(angle = -45, hjust = 0.01,size = 10, color = \"black\", face = \"bold\"),\n",
    "        axis.text.y = element_text(size = 12))+ \n",
    "  #geom_text(data =df_rr, aes(x = category, y = 0.13,\n",
    "     #label = paste(paste(\"RR=\",round(as.numeric(RR),2), sep=\"\"), \n",
    "     #paste(\"p=\",ifelse(pvalue>0.01, round(pvalue,2), formatC(pvalue, format = \"e\", digits = 1)),sep=\"\" ), sep='\\n')), size = 3) + \n",
    "  #geom_text(data =df_rr, \n",
    "            #aes(x = category, y =0.1, label = ifelse(pvalue > 0.05, \"NS\", ifelse(pvalue>0.01, \"*\", ifelse(pvalue > 0.001, \"**\", ifelse(pvalue<= 0.001, \"***\",\"\"))))), size =3) + \n",
    "  ggtitle(\"With FA\")\n",
    "\n",
    "p_fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "780110c2",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "d715b66a",
   "metadata": {},
   "source": [
    "## Burden with FA information - wo FA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 194,
   "id": "51c71fef",
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'df.mm.backup' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'df.mm.backup' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "df.mm <- df.mm.backup\n",
    "\n",
    "head(df.mm)\n",
    "dim(df.mm)\n",
    "table(df.mm$FA)\n",
    "\n",
    "df.mm.nfa = subset(df.mm, df.mm$FA == 'NO')\n",
    "samplecnt.mm = 438\n",
    "\n",
    "dim(df.mm.nfa)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 195,
   "id": "9a86154c",
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'df.mm.nfa' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'df.mm.nfa' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "df.mm <- df.mm.nfa\n",
    "\n",
    "total_size = 59281518 #size of coding region (hg38)\n",
    "\n",
    "consensus = read.delim(paste(\"/projects/ps-gleesonlab8/User/hiyoothere/NTD/10.Poisson/consensus_region/all_batches_\",percent, '.txt', sep=\"\"), header = FALSE)#all_batches_70.bed\n",
    "consensus\n",
    "Get_summary <- function(input, category){\n",
    "    \n",
    "    tmp_data = subset(df.melt, class == category)\n",
    "    head(tmp_data)\n",
    "    tmp_mm = tmp_data[tmp_data$cohort == 'MM', ]\n",
    "    tmp_ssc = tmp_data[tmp_data$cohort == 'SSC', ]\n",
    "#mean(tmp_mm$variant)\n",
    "   \n",
    "    df_tmp <- rbind(df_tmp,c('MM', category,  mean(as.numeric(tmp_mm$variant) * 100000000 / consensus_size / 2), \n",
    "                             sd(as.numeric(tmp_mm$variant) * 100000000 / consensus_size / 2), \n",
    "                             SE(as.numeric(tmp_mm$variant) * 100000000 / consensus_size / 2), \n",
    "                             CI(as.numeric(tmp_mm$variant) * 100000000 / consensus_size / 2)))\n",
    "    df_tmp <- rbind(df_tmp, c('SSC',category, mean(as.numeric(tmp_ssc$variant) * 100000000 / consensus_size / 2), \n",
    "                              sd(as.numeric(tmp_ssc$variant) * 100000000 / consensus_size / 2),\n",
    "                              SE(as.numeric(tmp_ssc$variant) * 100000000 / consensus_size / 2), \n",
    "                              CI(as.numeric(tmp_ssc$variant) * 100000000 / consensus_size / 2)))\n",
    "    colnames(df_tmp) = c('cohort', 'category','mean', 'sd', 'se', 'ci')\n",
    "    return (df_tmp)\n",
    "}\n",
    "\n",
    "df_tmp <- data.frame()\n",
    "df_rr <- data.frame()\n",
    "\n",
    "paste(\"consensus size is \", consensus[1,])\n",
    "consensus_size = consensus[1,]\n",
    "\n",
    "paste(\"consensus ratio \", consensus[1,]/total_size)\n",
    "\n",
    "paste(\"sample ntd \", samplecnt.mm)\n",
    "paste(\"sample ssc \", samplecnt.ssc)\n",
    "\n",
    "for (eff in figure_all) {\n",
    "    ratio_consen = consensus_size/total_size\n",
    "    mm = round((total_size*sum(df.mm[, eff])/consensus_size),0)\n",
    "    ssc = round((total_size*sum(df.ss[, eff])/consensus_size),0)\n",
    "    show(paste(eff, \"raw count\", sum(df.mm[, eff]), sum(df.ss[, eff])))\n",
    "    show(paste(eff, \"modi count\", mm, ssc))\n",
    "    c = c(mm,ssc)\n",
    "    #c=c(sum(df.mm[, eff]), sum(df.ss[, eff]))\n",
    "    T = c(samplecnt.mm, samplecnt.ssc)\n",
    "    pt = poisson.test(c, T, alternative='two.sided', conf.level = 0.95)\n",
    "    show(paste0(eff, ': ', round(pt$p.value, 8), '   ', 'Rate ratio: ', pt$estimate))\n",
    "    cnt_mm <- sum(df.mm[,eff])\n",
    "    cnt_ssc <- sum(df.ss[,eff])\n",
    "    df_rr <- rbind(df_rr, c(eff, round(pt$estimate,3), round(pt$p.value, 8), \"MM\", \"SSC\", \"All\", pt$conf.int, cnt_mm, cnt_ssc))\n",
    "    df_tmp <- Get_summary(df.melt, eff)\n",
    "}\n",
    "head(df_tmp)\n",
    "df_tmp$mean <- as.numeric(df_tmp$mean)\n",
    "df_tmp$ci <- as.numeric(df_tmp$ci)\n",
    "df_tmp$cohort <- factor(df_tmp$cohort, levels = c('SSC', 'MM'), labels = c(\"SSC\", \"MM\"))\n",
    "df_tmp$Gene <- \"All genes\"\n",
    "df_tmp_all <- df_tmp\n",
    "df_all_stat = df_tmp_all\n",
    "df_all_stat_backup = df_all_stat\n",
    "\n",
    "df.melt$variant <- as.numeric(df.melt$variant)\n",
    "head(df.melt)\n",
    "df_tmp\n",
    "options(repr.plot.width=14, repr.plot.height=8)\n",
    "df.mm = df[df$cohort == 'MM', ]\n",
    "df.ss = df[df$cohort == 'SSC', ]\n",
    "head(df.mm)\n",
    "\n",
    "\n",
    "dim(df.mm)\n",
    "dim(df.ss)\n",
    "\n",
    "dim(df.mm$All)\n",
    "dim(df.ss$All)\n",
    "df.melt <- melt(df, id.vars=c('sample', 'family', 'cohort', 'callable'))\n",
    "\n",
    "print (dim(subset(df.mm,df.mm$All == 0)))\n",
    "print (dim(subset(df.ss, df.ss$All == 0)))\n",
    "colnames(df.melt)[5] = 'class'\n",
    "colnames(df.melt)[6] = 'variant'\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "head(df.melt)#ggsave(\"/projects/ps-gleesonlab8/User/hiyoothere/NTD/15.Final/221020_DNM_v9/figure/vplot_all.pdf\", plot= p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 196,
   "id": "20a1c0b8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "40.5444312910953"
      ],
      "text/latex": [
       "40.5444312910953"
      ],
      "text/markdown": [
       "40.5444312910953"
      ],
      "text/plain": [
       "[1] 40.54443"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "41"
      ],
      "text/latex": [
       "41"
      ],
      "text/markdown": [
       "41"
      ],
      "text/plain": [
       "[1] 41"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "(total_size/consensus_size)*sum(df.mm[, eff])\n",
    "round((total_size/consensus_size)*sum(df.mm[, eff]),0)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 197,
   "id": "946f23b8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 36 × 9</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>cohort</th><th scope=col>category</th><th scope=col>mean</th><th scope=col>lower</th><th scope=col>upper</th><th scope=col>mean_diff</th><th scope=col>mean_diff_lower</th><th scope=col>mean_diff_upper</th><th scope=col>Gene</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>MM </td><td>All          </td><td>1.27395323</td><td>1.18086405</td><td>1.36704240</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>All          </td><td>1.25450214</td><td>1.15469614</td><td>1.35430813</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>Nonsynonymous</td><td>0.33133415</td><td>0.28467739</td><td>0.37799091</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>Nonsynonymous</td><td>0.28906450</td><td>0.24442039</td><td>0.33370860</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>Missense     </td><td>0.27463526</td><td>0.23174229</td><td>0.31752822</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>Missense     </td><td>0.26072484</td><td>0.21812915</td><td>0.30332054</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>LGD          </td><td>0.11516962</td><td>0.08656898</td><td>0.14377027</td><td>0.0603796209933105</td><td>0.0249566763471755</td><td>0.0958025656394456</td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>LGD          </td><td>0.05479000</td><td>0.03384624</td><td>0.07573377</td><td>0.0603796209933105</td><td>0.0249566763471755</td><td>0.0958025656394456</td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DMisL        </td><td>0.15415011</td><td>0.12278819</td><td>0.18551203</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DMisL        </td><td>0.13980897</td><td>0.10865755</td><td>0.17096040</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DMis         </td><td>0.06024257</td><td>0.03979874</td><td>0.08068640</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DMis         </td><td>0.03967552</td><td>0.02291462</td><td>0.05643641</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DMisL2       </td><td>0.15237827</td><td>0.12116656</td><td>0.18358999</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DMisL2       </td><td>0.13225173</td><td>0.10181715</td><td>0.16268631</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DMisM        </td><td>0.18427140</td><td>0.14981844</td><td>0.21872436</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DMisM        </td><td>0.17003794</td><td>0.13545440</td><td>0.20462148</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DMisMC       </td><td>0.13288803</td><td>0.10382803</td><td>0.16194803</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DMisMC       </td><td>0.11146932</td><td>0.08315469</td><td>0.13978394</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DMisMC2      </td><td>0.17186852</td><td>0.13870134</td><td>0.20503569</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DMisMC2      </td><td>0.15303415</td><td>0.11984811</td><td>0.18622018</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DamagingL    </td><td>0.26931974</td><td>0.22696776</td><td>0.31167171</td><td>0.0747207598965531</td><td>0.017680361675465 </td><td>0.131761158117641 </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DamagingL    </td><td>0.19459898</td><td>0.15632169</td><td>0.23287626</td><td>0.0747207598965531</td><td>0.017680361675465 </td><td>0.131761158117641 </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DamagingS    </td><td>0.17541220</td><td>0.14125679</td><td>0.20956761</td><td>0.0809466741234898</td><td>0.0374997132181427</td><td>0.124393635028837 </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DamagingS    </td><td>0.09446552</td><td>0.06755928</td><td>0.12137176</td><td>0.0809466741234898</td><td>0.0374997132181427</td><td>0.124393635028837 </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DamagingL2   </td><td>0.26754790</td><td>0.22528289</td><td>0.30981290</td><td>0.080506161324414 </td><td>0.0238792822303671</td><td>0.137133040418461 </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DamagingL2   </td><td>0.18704173</td><td>0.14928703</td><td>0.22479644</td><td>0.080506161324414 </td><td>0.0238792822303671</td><td>0.137133040418461 </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DamagingM    </td><td>0.29944102</td><td>0.25514149</td><td>0.34374056</td><td>0.0765023894394815</td><td>0.0163323040955573</td><td>0.136672474783406 </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DamagingM    </td><td>0.22293863</td><td>0.18214851</td><td>0.26372876</td><td>0.0765023894394815</td><td>0.0163323040955573</td><td>0.136672474783406 </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DamagingMC   </td><td>0.24805765</td><td>0.20828111</td><td>0.28783419</td><td>0.0817983322017028</td><td>0.0285486462435022</td><td>0.135048018159903 </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DamagingMC   </td><td>0.16625932</td><td>0.13079238</td><td>0.20172626</td><td>0.0817983322017028</td><td>0.0285486462435022</td><td>0.135048018159903 </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DamagingMC2  </td><td>0.28703814</td><td>0.24356651</td><td>0.33050977</td><td>0.0811033008964945</td><td>0.0222567442572848</td><td>0.139949857535704 </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DamagingMC2  </td><td>0.20593484</td><td>0.16620177</td><td>0.24566791</td><td>0.0811033008964945</td><td>0.0222567442572848</td><td>0.139949857535704 </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>OtherMissense</td><td>0.12048515</td><td>0.09136339</td><td>0.14960690</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>OtherMissense</td><td>0.12091587</td><td>0.09068329</td><td>0.15114845</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>Synonymous   </td><td>0.12402883</td><td>0.09335873</td><td>0.15469892</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>Synonymous   </td><td>0.12280518</td><td>0.09330564</td><td>0.15230472</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 36 × 9\n",
       "\\begin{tabular}{lllllllll}\n",
       " cohort & category & mean & lower & upper & mean\\_diff & mean\\_diff\\_lower & mean\\_diff\\_upper & Gene\\\\\n",
       " <fct> & <chr> & <dbl> & <dbl> & <dbl> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t MM  & All           & 1.27395323 & 1.18086405 & 1.36704240 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t SSC & All           & 1.25450214 & 1.15469614 & 1.35430813 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t MM  & Nonsynonymous & 0.33133415 & 0.28467739 & 0.37799091 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t SSC & Nonsynonymous & 0.28906450 & 0.24442039 & 0.33370860 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t MM  & Missense      & 0.27463526 & 0.23174229 & 0.31752822 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t SSC & Missense      & 0.26072484 & 0.21812915 & 0.30332054 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t MM  & LGD           & 0.11516962 & 0.08656898 & 0.14377027 & 0.0603796209933105 & 0.0249566763471755 & 0.0958025656394456 & All genes\\\\\n",
       "\t SSC & LGD           & 0.05479000 & 0.03384624 & 0.07573377 & 0.0603796209933105 & 0.0249566763471755 & 0.0958025656394456 & All genes\\\\\n",
       "\t MM  & DMisL         & 0.15415011 & 0.12278819 & 0.18551203 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t SSC & DMisL         & 0.13980897 & 0.10865755 & 0.17096040 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t MM  & DMis          & 0.06024257 & 0.03979874 & 0.08068640 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t SSC & DMis          & 0.03967552 & 0.02291462 & 0.05643641 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t MM  & DMisL2        & 0.15237827 & 0.12116656 & 0.18358999 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t SSC & DMisL2        & 0.13225173 & 0.10181715 & 0.16268631 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t MM  & DMisM         & 0.18427140 & 0.14981844 & 0.21872436 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t SSC & DMisM         & 0.17003794 & 0.13545440 & 0.20462148 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t MM  & DMisMC        & 0.13288803 & 0.10382803 & 0.16194803 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t SSC & DMisMC        & 0.11146932 & 0.08315469 & 0.13978394 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t MM  & DMisMC2       & 0.17186852 & 0.13870134 & 0.20503569 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t SSC & DMisMC2       & 0.15303415 & 0.11984811 & 0.18622018 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t MM  & DamagingL     & 0.26931974 & 0.22696776 & 0.31167171 & 0.0747207598965531 & 0.017680361675465  & 0.131761158117641  & All genes\\\\\n",
       "\t SSC & DamagingL     & 0.19459898 & 0.15632169 & 0.23287626 & 0.0747207598965531 & 0.017680361675465  & 0.131761158117641  & All genes\\\\\n",
       "\t MM  & DamagingS     & 0.17541220 & 0.14125679 & 0.20956761 & 0.0809466741234898 & 0.0374997132181427 & 0.124393635028837  & All genes\\\\\n",
       "\t SSC & DamagingS     & 0.09446552 & 0.06755928 & 0.12137176 & 0.0809466741234898 & 0.0374997132181427 & 0.124393635028837  & All genes\\\\\n",
       "\t MM  & DamagingL2    & 0.26754790 & 0.22528289 & 0.30981290 & 0.080506161324414  & 0.0238792822303671 & 0.137133040418461  & All genes\\\\\n",
       "\t SSC & DamagingL2    & 0.18704173 & 0.14928703 & 0.22479644 & 0.080506161324414  & 0.0238792822303671 & 0.137133040418461  & All genes\\\\\n",
       "\t MM  & DamagingM     & 0.29944102 & 0.25514149 & 0.34374056 & 0.0765023894394815 & 0.0163323040955573 & 0.136672474783406  & All genes\\\\\n",
       "\t SSC & DamagingM     & 0.22293863 & 0.18214851 & 0.26372876 & 0.0765023894394815 & 0.0163323040955573 & 0.136672474783406  & All genes\\\\\n",
       "\t MM  & DamagingMC    & 0.24805765 & 0.20828111 & 0.28783419 & 0.0817983322017028 & 0.0285486462435022 & 0.135048018159903  & All genes\\\\\n",
       "\t SSC & DamagingMC    & 0.16625932 & 0.13079238 & 0.20172626 & 0.0817983322017028 & 0.0285486462435022 & 0.135048018159903  & All genes\\\\\n",
       "\t MM  & DamagingMC2   & 0.28703814 & 0.24356651 & 0.33050977 & 0.0811033008964945 & 0.0222567442572848 & 0.139949857535704  & All genes\\\\\n",
       "\t SSC & DamagingMC2   & 0.20593484 & 0.16620177 & 0.24566791 & 0.0811033008964945 & 0.0222567442572848 & 0.139949857535704  & All genes\\\\\n",
       "\t MM  & OtherMissense & 0.12048515 & 0.09136339 & 0.14960690 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t SSC & OtherMissense & 0.12091587 & 0.09068329 & 0.15114845 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t MM  & Synonymous    & 0.12402883 & 0.09335873 & 0.15469892 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t SSC & Synonymous    & 0.12280518 & 0.09330564 & 0.15230472 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 36 × 9\n",
       "\n",
       "| cohort &lt;fct&gt; | category &lt;chr&gt; | mean &lt;dbl&gt; | lower &lt;dbl&gt; | upper &lt;dbl&gt; | mean_diff &lt;chr&gt; | mean_diff_lower &lt;chr&gt; | mean_diff_upper &lt;chr&gt; | Gene &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| MM  | All           | 1.27395323 | 1.18086405 | 1.36704240 | NA                 | NA                 | NA                 | All genes |\n",
       "| SSC | All           | 1.25450214 | 1.15469614 | 1.35430813 | NA                 | NA                 | NA                 | All genes |\n",
       "| MM  | Nonsynonymous | 0.33133415 | 0.28467739 | 0.37799091 | NA                 | NA                 | NA                 | All genes |\n",
       "| SSC | Nonsynonymous | 0.28906450 | 0.24442039 | 0.33370860 | NA                 | NA                 | NA                 | All genes |\n",
       "| MM  | Missense      | 0.27463526 | 0.23174229 | 0.31752822 | NA                 | NA                 | NA                 | All genes |\n",
       "| SSC | Missense      | 0.26072484 | 0.21812915 | 0.30332054 | NA                 | NA                 | NA                 | All genes |\n",
       "| MM  | LGD           | 0.11516962 | 0.08656898 | 0.14377027 | 0.0603796209933105 | 0.0249566763471755 | 0.0958025656394456 | All genes |\n",
       "| SSC | LGD           | 0.05479000 | 0.03384624 | 0.07573377 | 0.0603796209933105 | 0.0249566763471755 | 0.0958025656394456 | All genes |\n",
       "| MM  | DMisL         | 0.15415011 | 0.12278819 | 0.18551203 | NA                 | NA                 | NA                 | All genes |\n",
       "| SSC | DMisL         | 0.13980897 | 0.10865755 | 0.17096040 | NA                 | NA                 | NA                 | All genes |\n",
       "| MM  | DMis          | 0.06024257 | 0.03979874 | 0.08068640 | NA                 | NA                 | NA                 | All genes |\n",
       "| SSC | DMis          | 0.03967552 | 0.02291462 | 0.05643641 | NA                 | NA                 | NA                 | All genes |\n",
       "| MM  | DMisL2        | 0.15237827 | 0.12116656 | 0.18358999 | NA                 | NA                 | NA                 | All genes |\n",
       "| SSC | DMisL2        | 0.13225173 | 0.10181715 | 0.16268631 | NA                 | NA                 | NA                 | All genes |\n",
       "| MM  | DMisM         | 0.18427140 | 0.14981844 | 0.21872436 | NA                 | NA                 | NA                 | All genes |\n",
       "| SSC | DMisM         | 0.17003794 | 0.13545440 | 0.20462148 | NA                 | NA                 | NA                 | All genes |\n",
       "| MM  | DMisMC        | 0.13288803 | 0.10382803 | 0.16194803 | NA                 | NA                 | NA                 | All genes |\n",
       "| SSC | DMisMC        | 0.11146932 | 0.08315469 | 0.13978394 | NA                 | NA                 | NA                 | All genes |\n",
       "| MM  | DMisMC2       | 0.17186852 | 0.13870134 | 0.20503569 | NA                 | NA                 | NA                 | All genes |\n",
       "| SSC | DMisMC2       | 0.15303415 | 0.11984811 | 0.18622018 | NA                 | NA                 | NA                 | All genes |\n",
       "| MM  | DamagingL     | 0.26931974 | 0.22696776 | 0.31167171 | 0.0747207598965531 | 0.017680361675465  | 0.131761158117641  | All genes |\n",
       "| SSC | DamagingL     | 0.19459898 | 0.15632169 | 0.23287626 | 0.0747207598965531 | 0.017680361675465  | 0.131761158117641  | All genes |\n",
       "| MM  | DamagingS     | 0.17541220 | 0.14125679 | 0.20956761 | 0.0809466741234898 | 0.0374997132181427 | 0.124393635028837  | All genes |\n",
       "| SSC | DamagingS     | 0.09446552 | 0.06755928 | 0.12137176 | 0.0809466741234898 | 0.0374997132181427 | 0.124393635028837  | All genes |\n",
       "| MM  | DamagingL2    | 0.26754790 | 0.22528289 | 0.30981290 | 0.080506161324414  | 0.0238792822303671 | 0.137133040418461  | All genes |\n",
       "| SSC | DamagingL2    | 0.18704173 | 0.14928703 | 0.22479644 | 0.080506161324414  | 0.0238792822303671 | 0.137133040418461  | All genes |\n",
       "| MM  | DamagingM     | 0.29944102 | 0.25514149 | 0.34374056 | 0.0765023894394815 | 0.0163323040955573 | 0.136672474783406  | All genes |\n",
       "| SSC | DamagingM     | 0.22293863 | 0.18214851 | 0.26372876 | 0.0765023894394815 | 0.0163323040955573 | 0.136672474783406  | All genes |\n",
       "| MM  | DamagingMC    | 0.24805765 | 0.20828111 | 0.28783419 | 0.0817983322017028 | 0.0285486462435022 | 0.135048018159903  | All genes |\n",
       "| SSC | DamagingMC    | 0.16625932 | 0.13079238 | 0.20172626 | 0.0817983322017028 | 0.0285486462435022 | 0.135048018159903  | All genes |\n",
       "| MM  | DamagingMC2   | 0.28703814 | 0.24356651 | 0.33050977 | 0.0811033008964945 | 0.0222567442572848 | 0.139949857535704  | All genes |\n",
       "| SSC | DamagingMC2   | 0.20593484 | 0.16620177 | 0.24566791 | 0.0811033008964945 | 0.0222567442572848 | 0.139949857535704  | All genes |\n",
       "| MM  | OtherMissense | 0.12048515 | 0.09136339 | 0.14960690 | NA                 | NA                 | NA                 | All genes |\n",
       "| SSC | OtherMissense | 0.12091587 | 0.09068329 | 0.15114845 | NA                 | NA                 | NA                 | All genes |\n",
       "| MM  | Synonymous    | 0.12402883 | 0.09335873 | 0.15469892 | NA                 | NA                 | NA                 | All genes |\n",
       "| SSC | Synonymous    | 0.12280518 | 0.09330564 | 0.15230472 | NA                 | NA                 | NA                 | All genes |\n",
       "\n"
      ],
      "text/plain": [
       "   cohort category      mean       lower      upper      mean_diff         \n",
       "1  MM     All           1.27395323 1.18086405 1.36704240 NA                \n",
       "2  SSC    All           1.25450214 1.15469614 1.35430813 NA                \n",
       "3  MM     Nonsynonymous 0.33133415 0.28467739 0.37799091 NA                \n",
       "4  SSC    Nonsynonymous 0.28906450 0.24442039 0.33370860 NA                \n",
       "5  MM     Missense      0.27463526 0.23174229 0.31752822 NA                \n",
       "6  SSC    Missense      0.26072484 0.21812915 0.30332054 NA                \n",
       "7  MM     LGD           0.11516962 0.08656898 0.14377027 0.0603796209933105\n",
       "8  SSC    LGD           0.05479000 0.03384624 0.07573377 0.0603796209933105\n",
       "9  MM     DMisL         0.15415011 0.12278819 0.18551203 NA                \n",
       "10 SSC    DMisL         0.13980897 0.10865755 0.17096040 NA                \n",
       "11 MM     DMis          0.06024257 0.03979874 0.08068640 NA                \n",
       "12 SSC    DMis          0.03967552 0.02291462 0.05643641 NA                \n",
       "13 MM     DMisL2        0.15237827 0.12116656 0.18358999 NA                \n",
       "14 SSC    DMisL2        0.13225173 0.10181715 0.16268631 NA                \n",
       "15 MM     DMisM         0.18427140 0.14981844 0.21872436 NA                \n",
       "16 SSC    DMisM         0.17003794 0.13545440 0.20462148 NA                \n",
       "17 MM     DMisMC        0.13288803 0.10382803 0.16194803 NA                \n",
       "18 SSC    DMisMC        0.11146932 0.08315469 0.13978394 NA                \n",
       "19 MM     DMisMC2       0.17186852 0.13870134 0.20503569 NA                \n",
       "20 SSC    DMisMC2       0.15303415 0.11984811 0.18622018 NA                \n",
       "21 MM     DamagingL     0.26931974 0.22696776 0.31167171 0.0747207598965531\n",
       "22 SSC    DamagingL     0.19459898 0.15632169 0.23287626 0.0747207598965531\n",
       "23 MM     DamagingS     0.17541220 0.14125679 0.20956761 0.0809466741234898\n",
       "24 SSC    DamagingS     0.09446552 0.06755928 0.12137176 0.0809466741234898\n",
       "25 MM     DamagingL2    0.26754790 0.22528289 0.30981290 0.080506161324414 \n",
       "26 SSC    DamagingL2    0.18704173 0.14928703 0.22479644 0.080506161324414 \n",
       "27 MM     DamagingM     0.29944102 0.25514149 0.34374056 0.0765023894394815\n",
       "28 SSC    DamagingM     0.22293863 0.18214851 0.26372876 0.0765023894394815\n",
       "29 MM     DamagingMC    0.24805765 0.20828111 0.28783419 0.0817983322017028\n",
       "30 SSC    DamagingMC    0.16625932 0.13079238 0.20172626 0.0817983322017028\n",
       "31 MM     DamagingMC2   0.28703814 0.24356651 0.33050977 0.0811033008964945\n",
       "32 SSC    DamagingMC2   0.20593484 0.16620177 0.24566791 0.0811033008964945\n",
       "33 MM     OtherMissense 0.12048515 0.09136339 0.14960690 NA                \n",
       "34 SSC    OtherMissense 0.12091587 0.09068329 0.15114845 NA                \n",
       "35 MM     Synonymous    0.12402883 0.09335873 0.15469892 NA                \n",
       "36 SSC    Synonymous    0.12280518 0.09330564 0.15230472 NA                \n",
       "   mean_diff_lower    mean_diff_upper    Gene     \n",
       "1  NA                 NA                 All genes\n",
       "2  NA                 NA                 All genes\n",
       "3  NA                 NA                 All genes\n",
       "4  NA                 NA                 All genes\n",
       "5  NA                 NA                 All genes\n",
       "6  NA                 NA                 All genes\n",
       "7  0.0249566763471755 0.0958025656394456 All genes\n",
       "8  0.0249566763471755 0.0958025656394456 All genes\n",
       "9  NA                 NA                 All genes\n",
       "10 NA                 NA                 All genes\n",
       "11 NA                 NA                 All genes\n",
       "12 NA                 NA                 All genes\n",
       "13 NA                 NA                 All genes\n",
       "14 NA                 NA                 All genes\n",
       "15 NA                 NA                 All genes\n",
       "16 NA                 NA                 All genes\n",
       "17 NA                 NA                 All genes\n",
       "18 NA                 NA                 All genes\n",
       "19 NA                 NA                 All genes\n",
       "20 NA                 NA                 All genes\n",
       "21 0.017680361675465  0.131761158117641  All genes\n",
       "22 0.017680361675465  0.131761158117641  All genes\n",
       "23 0.0374997132181427 0.124393635028837  All genes\n",
       "24 0.0374997132181427 0.124393635028837  All genes\n",
       "25 0.0238792822303671 0.137133040418461  All genes\n",
       "26 0.0238792822303671 0.137133040418461  All genes\n",
       "27 0.0163323040955573 0.136672474783406  All genes\n",
       "28 0.0163323040955573 0.136672474783406  All genes\n",
       "29 0.0285486462435022 0.135048018159903  All genes\n",
       "30 0.0285486462435022 0.135048018159903  All genes\n",
       "31 0.0222567442572848 0.139949857535704  All genes\n",
       "32 0.0222567442572848 0.139949857535704  All genes\n",
       "33 NA                 NA                 All genes\n",
       "34 NA                 NA                 All genes\n",
       "35 NA                 NA                 All genes\n",
       "36 NA                 NA                 All genes"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 18 × 10</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>category</th><th scope=col>RR</th><th scope=col>pvalue</th><th scope=col>MM</th><th scope=col>SSC</th><th scope=col>Gene</th><th scope=col>Lower</th><th scope=col>Upper</th><th scope=col>MM</th><th scope=col>SSC</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>All          </td><td>1.015</td><td>0.36750219</td><td>MM</td><td>SSC</td><td>All</td><td>0.946314210007404</td><td>Inf</td><td>719</td><td>664</td></tr>\n",
       "\t<tr><td>Nonsynonymous</td><td>1.146</td><td>0.06068071</td><td>MM</td><td>SSC</td><td>All</td><td>0.991864762164627</td><td>Inf</td><td>187</td><td>153</td></tr>\n",
       "\t<tr><td>Missense     </td><td>1.051</td><td>0.3108332 </td><td>MM</td><td>SSC</td><td>All</td><td>0.899748854658688</td><td>Inf</td><td>155</td><td>138</td></tr>\n",
       "\t<tr><td>LGD          </td><td>2.095</td><td>9.07e-06  </td><td>MM</td><td>SSC</td><td>All</td><td>1.55232208946951 </td><td>Inf</td><td>65 </td><td>29 </td></tr>\n",
       "\t<tr><td>DMisL        </td><td>1.102</td><td>0.23590122</td><td>MM</td><td>SSC</td><td>All</td><td>0.89180362641801 </td><td>Inf</td><td>87 </td><td>74 </td></tr>\n",
       "\t<tr><td>DMis         </td><td>1.517</td><td>0.03400915</td><td>MM</td><td>SSC</td><td>All</td><td>1.03904139279005 </td><td>Inf</td><td>34 </td><td>21 </td></tr>\n",
       "\t<tr><td>DMisL2       </td><td>1.143</td><td>0.15889109</td><td>MM</td><td>SSC</td><td>All</td><td>0.922002711183897</td><td>Inf</td><td>86 </td><td>70 </td></tr>\n",
       "\t<tr><td>DMisM        </td><td>1.086</td><td>0.25142529</td><td>MM</td><td>SSC</td><td>All</td><td>0.895992573688703</td><td>Inf</td><td>104</td><td>90 </td></tr>\n",
       "\t<tr><td>DMisMC       </td><td>1.192</td><td>0.1111023 </td><td>MM</td><td>SSC</td><td>All</td><td>0.944142160308373</td><td>Inf</td><td>75 </td><td>59 </td></tr>\n",
       "\t<tr><td>DMisMC2      </td><td>1.124</td><td>0.17644896</td><td>MM</td><td>SSC</td><td>All</td><td>0.919095701303745</td><td>Inf</td><td>97 </td><td>81 </td></tr>\n",
       "\t<tr><td>DamagingL    </td><td>1.387</td><td>0.00059199</td><td>MM</td><td>SSC</td><td>All</td><td>1.17120452064468 </td><td>Inf</td><td>152</td><td>103</td></tr>\n",
       "\t<tr><td>DamagingS    </td><td>1.864</td><td>1.84e-06  </td><td>MM</td><td>SSC</td><td>All</td><td>1.47922093198684 </td><td>Inf</td><td>99 </td><td>50 </td></tr>\n",
       "\t<tr><td>DamagingL2   </td><td>1.427</td><td>0.00024288</td><td>MM</td><td>SSC</td><td>All</td><td>1.20242128398786 </td><td>Inf</td><td>151</td><td>99 </td></tr>\n",
       "\t<tr><td>DamagingM    </td><td>1.345</td><td>0.00089926</td><td>MM</td><td>SSC</td><td>All</td><td>1.14761833255911 </td><td>Inf</td><td>169</td><td>118</td></tr>\n",
       "\t<tr><td>DamagingMC   </td><td>1.489</td><td>0.00010054</td><td>MM</td><td>SSC</td><td>All</td><td>1.24291430217307 </td><td>Inf</td><td>140</td><td>88 </td></tr>\n",
       "\t<tr><td>DamagingMC2  </td><td>1.393</td><td>0.00034555</td><td>MM</td><td>SSC</td><td>All</td><td>1.1826688880035  </td><td>Inf</td><td>162</td><td>109</td></tr>\n",
       "\t<tr><td>OtherMissense</td><td>0.992</td><td>0.5510505 </td><td>MM</td><td>SSC</td><td>All</td><td>0.784862516565366</td><td>Inf</td><td>68 </td><td>64 </td></tr>\n",
       "\t<tr><td>Synonymous   </td><td>1.018</td><td>0.4740654 </td><td>MM</td><td>SSC</td><td>All</td><td>0.807887116097002</td><td>Inf</td><td>70 </td><td>65 </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 18 × 10\n",
       "\\begin{tabular}{llllllllll}\n",
       " category & RR & pvalue & MM & SSC & Gene & Lower & Upper & MM & SSC\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t All           & 1.015 & 0.36750219 & MM & SSC & All & 0.946314210007404 & Inf & 719 & 664\\\\\n",
       "\t Nonsynonymous & 1.146 & 0.06068071 & MM & SSC & All & 0.991864762164627 & Inf & 187 & 153\\\\\n",
       "\t Missense      & 1.051 & 0.3108332  & MM & SSC & All & 0.899748854658688 & Inf & 155 & 138\\\\\n",
       "\t LGD           & 2.095 & 9.07e-06   & MM & SSC & All & 1.55232208946951  & Inf & 65  & 29 \\\\\n",
       "\t DMisL         & 1.102 & 0.23590122 & MM & SSC & All & 0.89180362641801  & Inf & 87  & 74 \\\\\n",
       "\t DMis          & 1.517 & 0.03400915 & MM & SSC & All & 1.03904139279005  & Inf & 34  & 21 \\\\\n",
       "\t DMisL2        & 1.143 & 0.15889109 & MM & SSC & All & 0.922002711183897 & Inf & 86  & 70 \\\\\n",
       "\t DMisM         & 1.086 & 0.25142529 & MM & SSC & All & 0.895992573688703 & Inf & 104 & 90 \\\\\n",
       "\t DMisMC        & 1.192 & 0.1111023  & MM & SSC & All & 0.944142160308373 & Inf & 75  & 59 \\\\\n",
       "\t DMisMC2       & 1.124 & 0.17644896 & MM & SSC & All & 0.919095701303745 & Inf & 97  & 81 \\\\\n",
       "\t DamagingL     & 1.387 & 0.00059199 & MM & SSC & All & 1.17120452064468  & Inf & 152 & 103\\\\\n",
       "\t DamagingS     & 1.864 & 1.84e-06   & MM & SSC & All & 1.47922093198684  & Inf & 99  & 50 \\\\\n",
       "\t DamagingL2    & 1.427 & 0.00024288 & MM & SSC & All & 1.20242128398786  & Inf & 151 & 99 \\\\\n",
       "\t DamagingM     & 1.345 & 0.00089926 & MM & SSC & All & 1.14761833255911  & Inf & 169 & 118\\\\\n",
       "\t DamagingMC    & 1.489 & 0.00010054 & MM & SSC & All & 1.24291430217307  & Inf & 140 & 88 \\\\\n",
       "\t DamagingMC2   & 1.393 & 0.00034555 & MM & SSC & All & 1.1826688880035   & Inf & 162 & 109\\\\\n",
       "\t OtherMissense & 0.992 & 0.5510505  & MM & SSC & All & 0.784862516565366 & Inf & 68  & 64 \\\\\n",
       "\t Synonymous    & 1.018 & 0.4740654  & MM & SSC & All & 0.807887116097002 & Inf & 70  & 65 \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 18 × 10\n",
       "\n",
       "| category &lt;chr&gt; | RR &lt;chr&gt; | pvalue &lt;chr&gt; | MM &lt;chr&gt; | SSC &lt;chr&gt; | Gene &lt;chr&gt; | Lower &lt;chr&gt; | Upper &lt;chr&gt; | MM &lt;chr&gt; | SSC &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|\n",
       "| All           | 1.015 | 0.36750219 | MM | SSC | All | 0.946314210007404 | Inf | 719 | 664 |\n",
       "| Nonsynonymous | 1.146 | 0.06068071 | MM | SSC | All | 0.991864762164627 | Inf | 187 | 153 |\n",
       "| Missense      | 1.051 | 0.3108332  | MM | SSC | All | 0.899748854658688 | Inf | 155 | 138 |\n",
       "| LGD           | 2.095 | 9.07e-06   | MM | SSC | All | 1.55232208946951  | Inf | 65  | 29  |\n",
       "| DMisL         | 1.102 | 0.23590122 | MM | SSC | All | 0.89180362641801  | Inf | 87  | 74  |\n",
       "| DMis          | 1.517 | 0.03400915 | MM | SSC | All | 1.03904139279005  | Inf | 34  | 21  |\n",
       "| DMisL2        | 1.143 | 0.15889109 | MM | SSC | All | 0.922002711183897 | Inf | 86  | 70  |\n",
       "| DMisM         | 1.086 | 0.25142529 | MM | SSC | All | 0.895992573688703 | Inf | 104 | 90  |\n",
       "| DMisMC        | 1.192 | 0.1111023  | MM | SSC | All | 0.944142160308373 | Inf | 75  | 59  |\n",
       "| DMisMC2       | 1.124 | 0.17644896 | MM | SSC | All | 0.919095701303745 | Inf | 97  | 81  |\n",
       "| DamagingL     | 1.387 | 0.00059199 | MM | SSC | All | 1.17120452064468  | Inf | 152 | 103 |\n",
       "| DamagingS     | 1.864 | 1.84e-06   | MM | SSC | All | 1.47922093198684  | Inf | 99  | 50  |\n",
       "| DamagingL2    | 1.427 | 0.00024288 | MM | SSC | All | 1.20242128398786  | Inf | 151 | 99  |\n",
       "| DamagingM     | 1.345 | 0.00089926 | MM | SSC | All | 1.14761833255911  | Inf | 169 | 118 |\n",
       "| DamagingMC    | 1.489 | 0.00010054 | MM | SSC | All | 1.24291430217307  | Inf | 140 | 88  |\n",
       "| DamagingMC2   | 1.393 | 0.00034555 | MM | SSC | All | 1.1826688880035   | Inf | 162 | 109 |\n",
       "| OtherMissense | 0.992 | 0.5510505  | MM | SSC | All | 0.784862516565366 | Inf | 68  | 64  |\n",
       "| Synonymous    | 1.018 | 0.4740654  | MM | SSC | All | 0.807887116097002 | Inf | 70  | 65  |\n",
       "\n"
      ],
      "text/plain": [
       "   category      RR    pvalue     MM SSC Gene Lower             Upper MM  SSC\n",
       "1  All           1.015 0.36750219 MM SSC All  0.946314210007404 Inf   719 664\n",
       "2  Nonsynonymous 1.146 0.06068071 MM SSC All  0.991864762164627 Inf   187 153\n",
       "3  Missense      1.051 0.3108332  MM SSC All  0.899748854658688 Inf   155 138\n",
       "4  LGD           2.095 9.07e-06   MM SSC All  1.55232208946951  Inf   65  29 \n",
       "5  DMisL         1.102 0.23590122 MM SSC All  0.89180362641801  Inf   87  74 \n",
       "6  DMis          1.517 0.03400915 MM SSC All  1.03904139279005  Inf   34  21 \n",
       "7  DMisL2        1.143 0.15889109 MM SSC All  0.922002711183897 Inf   86  70 \n",
       "8  DMisM         1.086 0.25142529 MM SSC All  0.895992573688703 Inf   104 90 \n",
       "9  DMisMC        1.192 0.1111023  MM SSC All  0.944142160308373 Inf   75  59 \n",
       "10 DMisMC2       1.124 0.17644896 MM SSC All  0.919095701303745 Inf   97  81 \n",
       "11 DamagingL     1.387 0.00059199 MM SSC All  1.17120452064468  Inf   152 103\n",
       "12 DamagingS     1.864 1.84e-06   MM SSC All  1.47922093198684  Inf   99  50 \n",
       "13 DamagingL2    1.427 0.00024288 MM SSC All  1.20242128398786  Inf   151 99 \n",
       "14 DamagingM     1.345 0.00089926 MM SSC All  1.14761833255911  Inf   169 118\n",
       "15 DamagingMC    1.489 0.00010054 MM SSC All  1.24291430217307  Inf   140 88 \n",
       "16 DamagingMC2   1.393 0.00034555 MM SSC All  1.1826688880035   Inf   162 109\n",
       "17 OtherMissense 0.992 0.5510505  MM SSC All  0.784862516565366 Inf   68  64 \n",
       "18 Synonymous    1.018 0.4740654  MM SSC All  0.807887116097002 Inf   70  65 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 36 × 9</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>cohort</th><th scope=col>category</th><th scope=col>mean</th><th scope=col>lower</th><th scope=col>upper</th><th scope=col>mean_diff</th><th scope=col>mean_diff_lower</th><th scope=col>mean_diff_upper</th><th scope=col>Gene</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>MM </td><td>NA           </td><td>1.27395323</td><td>1.18086405</td><td>1.36704240</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>NA           </td><td>1.25450214</td><td>1.15469614</td><td>1.35430813</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>NA           </td><td>0.33133415</td><td>0.28467739</td><td>0.37799091</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>NA           </td><td>0.28906450</td><td>0.24442039</td><td>0.33370860</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>NA           </td><td>0.27463526</td><td>0.23174229</td><td>0.31752822</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>NA           </td><td>0.26072484</td><td>0.21812915</td><td>0.30332054</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>LGD          </td><td>0.11516962</td><td>0.08656898</td><td>0.14377027</td><td>0.0603796209933105</td><td>0.0249566763471755</td><td>0.0958025656394456</td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>LGD          </td><td>0.05479000</td><td>0.03384624</td><td>0.07573377</td><td>0.0603796209933105</td><td>0.0249566763471755</td><td>0.0958025656394456</td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>NA           </td><td>0.15415011</td><td>0.12278819</td><td>0.18551203</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>NA           </td><td>0.13980897</td><td>0.10865755</td><td>0.17096040</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DMis         </td><td>0.06024257</td><td>0.03979874</td><td>0.08068640</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DMis         </td><td>0.03967552</td><td>0.02291462</td><td>0.05643641</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>NA           </td><td>0.15237827</td><td>0.12116656</td><td>0.18358999</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>NA           </td><td>0.13225173</td><td>0.10181715</td><td>0.16268631</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>NA           </td><td>0.18427140</td><td>0.14981844</td><td>0.21872436</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>NA           </td><td>0.17003794</td><td>0.13545440</td><td>0.20462148</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>NA           </td><td>0.13288803</td><td>0.10382803</td><td>0.16194803</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>NA           </td><td>0.11146932</td><td>0.08315469</td><td>0.13978394</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>NA           </td><td>0.17186852</td><td>0.13870134</td><td>0.20503569</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>NA           </td><td>0.15303415</td><td>0.11984811</td><td>0.18622018</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>NA           </td><td>0.26931974</td><td>0.22696776</td><td>0.31167171</td><td>0.0747207598965531</td><td>0.017680361675465 </td><td>0.131761158117641 </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>NA           </td><td>0.19459898</td><td>0.15632169</td><td>0.23287626</td><td>0.0747207598965531</td><td>0.017680361675465 </td><td>0.131761158117641 </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>DamagingS    </td><td>0.17541220</td><td>0.14125679</td><td>0.20956761</td><td>0.0809466741234898</td><td>0.0374997132181427</td><td>0.124393635028837 </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>DamagingS    </td><td>0.09446552</td><td>0.06755928</td><td>0.12137176</td><td>0.0809466741234898</td><td>0.0374997132181427</td><td>0.124393635028837 </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>NA           </td><td>0.26754790</td><td>0.22528289</td><td>0.30981290</td><td>0.080506161324414 </td><td>0.0238792822303671</td><td>0.137133040418461 </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>NA           </td><td>0.18704173</td><td>0.14928703</td><td>0.22479644</td><td>0.080506161324414 </td><td>0.0238792822303671</td><td>0.137133040418461 </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>NA           </td><td>0.29944102</td><td>0.25514149</td><td>0.34374056</td><td>0.0765023894394815</td><td>0.0163323040955573</td><td>0.136672474783406 </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>NA           </td><td>0.22293863</td><td>0.18214851</td><td>0.26372876</td><td>0.0765023894394815</td><td>0.0163323040955573</td><td>0.136672474783406 </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>NA           </td><td>0.24805765</td><td>0.20828111</td><td>0.28783419</td><td>0.0817983322017028</td><td>0.0285486462435022</td><td>0.135048018159903 </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>NA           </td><td>0.16625932</td><td>0.13079238</td><td>0.20172626</td><td>0.0817983322017028</td><td>0.0285486462435022</td><td>0.135048018159903 </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>NA           </td><td>0.28703814</td><td>0.24356651</td><td>0.33050977</td><td>0.0811033008964945</td><td>0.0222567442572848</td><td>0.139949857535704 </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>NA           </td><td>0.20593484</td><td>0.16620177</td><td>0.24566791</td><td>0.0811033008964945</td><td>0.0222567442572848</td><td>0.139949857535704 </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>OtherMissense</td><td>0.12048515</td><td>0.09136339</td><td>0.14960690</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>OtherMissense</td><td>0.12091587</td><td>0.09068329</td><td>0.15114845</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>MM </td><td>Synonymous   </td><td>0.12402883</td><td>0.09335873</td><td>0.15469892</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "\t<tr><td>SSC</td><td>Synonymous   </td><td>0.12280518</td><td>0.09330564</td><td>0.15230472</td><td>NA                </td><td>NA                </td><td>NA                </td><td>All genes</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 36 × 9\n",
       "\\begin{tabular}{lllllllll}\n",
       " cohort & category & mean & lower & upper & mean\\_diff & mean\\_diff\\_lower & mean\\_diff\\_upper & Gene\\\\\n",
       " <fct> & <fct> & <dbl> & <dbl> & <dbl> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t MM  & NA            & 1.27395323 & 1.18086405 & 1.36704240 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t SSC & NA            & 1.25450214 & 1.15469614 & 1.35430813 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t MM  & NA            & 0.33133415 & 0.28467739 & 0.37799091 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t SSC & NA            & 0.28906450 & 0.24442039 & 0.33370860 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t MM  & NA            & 0.27463526 & 0.23174229 & 0.31752822 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t SSC & NA            & 0.26072484 & 0.21812915 & 0.30332054 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t MM  & LGD           & 0.11516962 & 0.08656898 & 0.14377027 & 0.0603796209933105 & 0.0249566763471755 & 0.0958025656394456 & All genes\\\\\n",
       "\t SSC & LGD           & 0.05479000 & 0.03384624 & 0.07573377 & 0.0603796209933105 & 0.0249566763471755 & 0.0958025656394456 & All genes\\\\\n",
       "\t MM  & NA            & 0.15415011 & 0.12278819 & 0.18551203 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t SSC & NA            & 0.13980897 & 0.10865755 & 0.17096040 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t MM  & DMis          & 0.06024257 & 0.03979874 & 0.08068640 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t SSC & DMis          & 0.03967552 & 0.02291462 & 0.05643641 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t MM  & NA            & 0.15237827 & 0.12116656 & 0.18358999 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t SSC & NA            & 0.13225173 & 0.10181715 & 0.16268631 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t MM  & NA            & 0.18427140 & 0.14981844 & 0.21872436 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t SSC & NA            & 0.17003794 & 0.13545440 & 0.20462148 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t MM  & NA            & 0.13288803 & 0.10382803 & 0.16194803 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t SSC & NA            & 0.11146932 & 0.08315469 & 0.13978394 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t MM  & NA            & 0.17186852 & 0.13870134 & 0.20503569 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t SSC & NA            & 0.15303415 & 0.11984811 & 0.18622018 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t MM  & NA            & 0.26931974 & 0.22696776 & 0.31167171 & 0.0747207598965531 & 0.017680361675465  & 0.131761158117641  & All genes\\\\\n",
       "\t SSC & NA            & 0.19459898 & 0.15632169 & 0.23287626 & 0.0747207598965531 & 0.017680361675465  & 0.131761158117641  & All genes\\\\\n",
       "\t MM  & DamagingS     & 0.17541220 & 0.14125679 & 0.20956761 & 0.0809466741234898 & 0.0374997132181427 & 0.124393635028837  & All genes\\\\\n",
       "\t SSC & DamagingS     & 0.09446552 & 0.06755928 & 0.12137176 & 0.0809466741234898 & 0.0374997132181427 & 0.124393635028837  & All genes\\\\\n",
       "\t MM  & NA            & 0.26754790 & 0.22528289 & 0.30981290 & 0.080506161324414  & 0.0238792822303671 & 0.137133040418461  & All genes\\\\\n",
       "\t SSC & NA            & 0.18704173 & 0.14928703 & 0.22479644 & 0.080506161324414  & 0.0238792822303671 & 0.137133040418461  & All genes\\\\\n",
       "\t MM  & NA            & 0.29944102 & 0.25514149 & 0.34374056 & 0.0765023894394815 & 0.0163323040955573 & 0.136672474783406  & All genes\\\\\n",
       "\t SSC & NA            & 0.22293863 & 0.18214851 & 0.26372876 & 0.0765023894394815 & 0.0163323040955573 & 0.136672474783406  & All genes\\\\\n",
       "\t MM  & NA            & 0.24805765 & 0.20828111 & 0.28783419 & 0.0817983322017028 & 0.0285486462435022 & 0.135048018159903  & All genes\\\\\n",
       "\t SSC & NA            & 0.16625932 & 0.13079238 & 0.20172626 & 0.0817983322017028 & 0.0285486462435022 & 0.135048018159903  & All genes\\\\\n",
       "\t MM  & NA            & 0.28703814 & 0.24356651 & 0.33050977 & 0.0811033008964945 & 0.0222567442572848 & 0.139949857535704  & All genes\\\\\n",
       "\t SSC & NA            & 0.20593484 & 0.16620177 & 0.24566791 & 0.0811033008964945 & 0.0222567442572848 & 0.139949857535704  & All genes\\\\\n",
       "\t MM  & OtherMissense & 0.12048515 & 0.09136339 & 0.14960690 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t SSC & OtherMissense & 0.12091587 & 0.09068329 & 0.15114845 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t MM  & Synonymous    & 0.12402883 & 0.09335873 & 0.15469892 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\t SSC & Synonymous    & 0.12280518 & 0.09330564 & 0.15230472 & NA                 & NA                 & NA                 & All genes\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 36 × 9\n",
       "\n",
       "| cohort &lt;fct&gt; | category &lt;fct&gt; | mean &lt;dbl&gt; | lower &lt;dbl&gt; | upper &lt;dbl&gt; | mean_diff &lt;chr&gt; | mean_diff_lower &lt;chr&gt; | mean_diff_upper &lt;chr&gt; | Gene &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| MM  | NA            | 1.27395323 | 1.18086405 | 1.36704240 | NA                 | NA                 | NA                 | All genes |\n",
       "| SSC | NA            | 1.25450214 | 1.15469614 | 1.35430813 | NA                 | NA                 | NA                 | All genes |\n",
       "| MM  | NA            | 0.33133415 | 0.28467739 | 0.37799091 | NA                 | NA                 | NA                 | All genes |\n",
       "| SSC | NA            | 0.28906450 | 0.24442039 | 0.33370860 | NA                 | NA                 | NA                 | All genes |\n",
       "| MM  | NA            | 0.27463526 | 0.23174229 | 0.31752822 | NA                 | NA                 | NA                 | All genes |\n",
       "| SSC | NA            | 0.26072484 | 0.21812915 | 0.30332054 | NA                 | NA                 | NA                 | All genes |\n",
       "| MM  | LGD           | 0.11516962 | 0.08656898 | 0.14377027 | 0.0603796209933105 | 0.0249566763471755 | 0.0958025656394456 | All genes |\n",
       "| SSC | LGD           | 0.05479000 | 0.03384624 | 0.07573377 | 0.0603796209933105 | 0.0249566763471755 | 0.0958025656394456 | All genes |\n",
       "| MM  | NA            | 0.15415011 | 0.12278819 | 0.18551203 | NA                 | NA                 | NA                 | All genes |\n",
       "| SSC | NA            | 0.13980897 | 0.10865755 | 0.17096040 | NA                 | NA                 | NA                 | All genes |\n",
       "| MM  | DMis          | 0.06024257 | 0.03979874 | 0.08068640 | NA                 | NA                 | NA                 | All genes |\n",
       "| SSC | DMis          | 0.03967552 | 0.02291462 | 0.05643641 | NA                 | NA                 | NA                 | All genes |\n",
       "| MM  | NA            | 0.15237827 | 0.12116656 | 0.18358999 | NA                 | NA                 | NA                 | All genes |\n",
       "| SSC | NA            | 0.13225173 | 0.10181715 | 0.16268631 | NA                 | NA                 | NA                 | All genes |\n",
       "| MM  | NA            | 0.18427140 | 0.14981844 | 0.21872436 | NA                 | NA                 | NA                 | All genes |\n",
       "| SSC | NA            | 0.17003794 | 0.13545440 | 0.20462148 | NA                 | NA                 | NA                 | All genes |\n",
       "| MM  | NA            | 0.13288803 | 0.10382803 | 0.16194803 | NA                 | NA                 | NA                 | All genes |\n",
       "| SSC | NA            | 0.11146932 | 0.08315469 | 0.13978394 | NA                 | NA                 | NA                 | All genes |\n",
       "| MM  | NA            | 0.17186852 | 0.13870134 | 0.20503569 | NA                 | NA                 | NA                 | All genes |\n",
       "| SSC | NA            | 0.15303415 | 0.11984811 | 0.18622018 | NA                 | NA                 | NA                 | All genes |\n",
       "| MM  | NA            | 0.26931974 | 0.22696776 | 0.31167171 | 0.0747207598965531 | 0.017680361675465  | 0.131761158117641  | All genes |\n",
       "| SSC | NA            | 0.19459898 | 0.15632169 | 0.23287626 | 0.0747207598965531 | 0.017680361675465  | 0.131761158117641  | All genes |\n",
       "| MM  | DamagingS     | 0.17541220 | 0.14125679 | 0.20956761 | 0.0809466741234898 | 0.0374997132181427 | 0.124393635028837  | All genes |\n",
       "| SSC | DamagingS     | 0.09446552 | 0.06755928 | 0.12137176 | 0.0809466741234898 | 0.0374997132181427 | 0.124393635028837  | All genes |\n",
       "| MM  | NA            | 0.26754790 | 0.22528289 | 0.30981290 | 0.080506161324414  | 0.0238792822303671 | 0.137133040418461  | All genes |\n",
       "| SSC | NA            | 0.18704173 | 0.14928703 | 0.22479644 | 0.080506161324414  | 0.0238792822303671 | 0.137133040418461  | All genes |\n",
       "| MM  | NA            | 0.29944102 | 0.25514149 | 0.34374056 | 0.0765023894394815 | 0.0163323040955573 | 0.136672474783406  | All genes |\n",
       "| SSC | NA            | 0.22293863 | 0.18214851 | 0.26372876 | 0.0765023894394815 | 0.0163323040955573 | 0.136672474783406  | All genes |\n",
       "| MM  | NA            | 0.24805765 | 0.20828111 | 0.28783419 | 0.0817983322017028 | 0.0285486462435022 | 0.135048018159903  | All genes |\n",
       "| SSC | NA            | 0.16625932 | 0.13079238 | 0.20172626 | 0.0817983322017028 | 0.0285486462435022 | 0.135048018159903  | All genes |\n",
       "| MM  | NA            | 0.28703814 | 0.24356651 | 0.33050977 | 0.0811033008964945 | 0.0222567442572848 | 0.139949857535704  | All genes |\n",
       "| SSC | NA            | 0.20593484 | 0.16620177 | 0.24566791 | 0.0811033008964945 | 0.0222567442572848 | 0.139949857535704  | All genes |\n",
       "| MM  | OtherMissense | 0.12048515 | 0.09136339 | 0.14960690 | NA                 | NA                 | NA                 | All genes |\n",
       "| SSC | OtherMissense | 0.12091587 | 0.09068329 | 0.15114845 | NA                 | NA                 | NA                 | All genes |\n",
       "| MM  | Synonymous    | 0.12402883 | 0.09335873 | 0.15469892 | NA                 | NA                 | NA                 | All genes |\n",
       "| SSC | Synonymous    | 0.12280518 | 0.09330564 | 0.15230472 | NA                 | NA                 | NA                 | All genes |\n",
       "\n"
      ],
      "text/plain": [
       "   cohort category      mean       lower      upper      mean_diff         \n",
       "1  MM     NA            1.27395323 1.18086405 1.36704240 NA                \n",
       "2  SSC    NA            1.25450214 1.15469614 1.35430813 NA                \n",
       "3  MM     NA            0.33133415 0.28467739 0.37799091 NA                \n",
       "4  SSC    NA            0.28906450 0.24442039 0.33370860 NA                \n",
       "5  MM     NA            0.27463526 0.23174229 0.31752822 NA                \n",
       "6  SSC    NA            0.26072484 0.21812915 0.30332054 NA                \n",
       "7  MM     LGD           0.11516962 0.08656898 0.14377027 0.0603796209933105\n",
       "8  SSC    LGD           0.05479000 0.03384624 0.07573377 0.0603796209933105\n",
       "9  MM     NA            0.15415011 0.12278819 0.18551203 NA                \n",
       "10 SSC    NA            0.13980897 0.10865755 0.17096040 NA                \n",
       "11 MM     DMis          0.06024257 0.03979874 0.08068640 NA                \n",
       "12 SSC    DMis          0.03967552 0.02291462 0.05643641 NA                \n",
       "13 MM     NA            0.15237827 0.12116656 0.18358999 NA                \n",
       "14 SSC    NA            0.13225173 0.10181715 0.16268631 NA                \n",
       "15 MM     NA            0.18427140 0.14981844 0.21872436 NA                \n",
       "16 SSC    NA            0.17003794 0.13545440 0.20462148 NA                \n",
       "17 MM     NA            0.13288803 0.10382803 0.16194803 NA                \n",
       "18 SSC    NA            0.11146932 0.08315469 0.13978394 NA                \n",
       "19 MM     NA            0.17186852 0.13870134 0.20503569 NA                \n",
       "20 SSC    NA            0.15303415 0.11984811 0.18622018 NA                \n",
       "21 MM     NA            0.26931974 0.22696776 0.31167171 0.0747207598965531\n",
       "22 SSC    NA            0.19459898 0.15632169 0.23287626 0.0747207598965531\n",
       "23 MM     DamagingS     0.17541220 0.14125679 0.20956761 0.0809466741234898\n",
       "24 SSC    DamagingS     0.09446552 0.06755928 0.12137176 0.0809466741234898\n",
       "25 MM     NA            0.26754790 0.22528289 0.30981290 0.080506161324414 \n",
       "26 SSC    NA            0.18704173 0.14928703 0.22479644 0.080506161324414 \n",
       "27 MM     NA            0.29944102 0.25514149 0.34374056 0.0765023894394815\n",
       "28 SSC    NA            0.22293863 0.18214851 0.26372876 0.0765023894394815\n",
       "29 MM     NA            0.24805765 0.20828111 0.28783419 0.0817983322017028\n",
       "30 SSC    NA            0.16625932 0.13079238 0.20172626 0.0817983322017028\n",
       "31 MM     NA            0.28703814 0.24356651 0.33050977 0.0811033008964945\n",
       "32 SSC    NA            0.20593484 0.16620177 0.24566791 0.0811033008964945\n",
       "33 MM     OtherMissense 0.12048515 0.09136339 0.14960690 NA                \n",
       "34 SSC    OtherMissense 0.12091587 0.09068329 0.15114845 NA                \n",
       "35 MM     Synonymous    0.12402883 0.09335873 0.15469892 NA                \n",
       "36 SSC    Synonymous    0.12280518 0.09330564 0.15230472 NA                \n",
       "   mean_diff_lower    mean_diff_upper    Gene     \n",
       "1  NA                 NA                 All genes\n",
       "2  NA                 NA                 All genes\n",
       "3  NA                 NA                 All genes\n",
       "4  NA                 NA                 All genes\n",
       "5  NA                 NA                 All genes\n",
       "6  NA                 NA                 All genes\n",
       "7  0.0249566763471755 0.0958025656394456 All genes\n",
       "8  0.0249566763471755 0.0958025656394456 All genes\n",
       "9  NA                 NA                 All genes\n",
       "10 NA                 NA                 All genes\n",
       "11 NA                 NA                 All genes\n",
       "12 NA                 NA                 All genes\n",
       "13 NA                 NA                 All genes\n",
       "14 NA                 NA                 All genes\n",
       "15 NA                 NA                 All genes\n",
       "16 NA                 NA                 All genes\n",
       "17 NA                 NA                 All genes\n",
       "18 NA                 NA                 All genes\n",
       "19 NA                 NA                 All genes\n",
       "20 NA                 NA                 All genes\n",
       "21 0.017680361675465  0.131761158117641  All genes\n",
       "22 0.017680361675465  0.131761158117641  All genes\n",
       "23 0.0374997132181427 0.124393635028837  All genes\n",
       "24 0.0374997132181427 0.124393635028837  All genes\n",
       "25 0.0238792822303671 0.137133040418461  All genes\n",
       "26 0.0238792822303671 0.137133040418461  All genes\n",
       "27 0.0163323040955573 0.136672474783406  All genes\n",
       "28 0.0163323040955573 0.136672474783406  All genes\n",
       "29 0.0285486462435022 0.135048018159903  All genes\n",
       "30 0.0285486462435022 0.135048018159903  All genes\n",
       "31 0.0222567442572848 0.139949857535704  All genes\n",
       "32 0.0222567442572848 0.139949857535704  All genes\n",
       "33 NA                 NA                 All genes\n",
       "34 NA                 NA                 All genes\n",
       "35 NA                 NA                 All genes\n",
       "36 NA                 NA                 All genes"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "\n",
    "df_tmp\n",
    "colnames(df_rr) <- c(\"category\",\"RR\", \"pvalue\", 'MM', \"SSC\", \"Gene\",\"Lower\",\"Upper\", \"MM\", \"SSC\")\n",
    "df_rr\n",
    "df_all_stat"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 198,
   "id": "8a3f49f7",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ERROR while rich displaying an object: \u001b[1m\u001b[33mError\u001b[39m in `geom_errorbar()`:\u001b[22m\n",
      "\u001b[1m\u001b[22m\u001b[33m!\u001b[39m Problem while computing aesthetics.\n",
      "\u001b[36mi\u001b[39m Error occurred in the 1st layer.\n",
      "\u001b[1mCaused by error in `FUN()`:\u001b[22m\n",
      "\u001b[33m!\u001b[39m object 'ci' not found\n",
      "\n",
      "Traceback:\n",
      "1. tryCatch(withCallingHandlers({\n",
      " .     if (!mime %in% names(repr::mime2repr)) \n",
      " .         stop(\"No repr_* for mimetype \", mime, \" in repr::mime2repr\")\n",
      " .     rpr <- repr::mime2repr[[mime]](obj)\n",
      " .     if (is.null(rpr)) \n",
      " .         return(NULL)\n",
      " .     prepare_content(is.raw(rpr), rpr)\n",
      " . }, error = error_handler), error = outer_handler)\n",
      "2. tryCatchList(expr, classes, parentenv, handlers)\n",
      "3. tryCatchOne(expr, names, parentenv, handlers[[1L]])\n",
      "4. doTryCatch(return(expr), name, parentenv, handler)\n",
      "5. withCallingHandlers({\n",
      " .     if (!mime %in% names(repr::mime2repr)) \n",
      " .         stop(\"No repr_* for mimetype \", mime, \" in repr::mime2repr\")\n",
      " .     rpr <- repr::mime2repr[[mime]](obj)\n",
      " .     if (is.null(rpr)) \n",
      " .         return(NULL)\n",
      " .     prepare_content(is.raw(rpr), rpr)\n",
      " . }, error = error_handler)\n",
      "6. repr::mime2repr[[mime]](obj)\n",
      "7. repr_text.default(obj)\n",
      "8. paste(capture.output(print(obj)), collapse = \"\\n\")\n",
      "9. capture.output(print(obj))\n",
      "10. withVisible(...elt(i))\n",
      "11. print(obj)\n",
      "12. print.ggplot(obj)\n",
      "13. ggplot_build(x)\n",
      "14. ggplot_build.ggplot(x)\n",
      "15. by_layer(function(l, d) l$compute_aesthetics(d, plot), layers, \n",
      "  .     data, \"computing aesthetics\")\n",
      "16. try_fetch(for (i in seq_along(data)) {\n",
      "  .     out[[i]] <- f(l = layers[[i]], d = data[[i]])\n",
      "  . }, error = function(cnd) {\n",
      "  .     cli::cli_abort(c(\"Problem while {step}.\", i = \"Error occurred in the {ordinal(i)} layer.\"), \n",
      "  .         call = layers[[i]]$constructor, parent = cnd)\n",
      "  . })\n",
      "17. tryCatch(withCallingHandlers(expr, error = function(cnd) {\n",
      "  .     {\n",
      "  .         .__handler_frame__. <- TRUE\n",
      "  .         .__setup_frame__. <- frame\n",
      "  .     }\n",
      "  .     out <- handlers[[1L]](cnd)\n",
      "  .     if (!inherits(out, \"rlang_zap\")) \n",
      "  .         throw(out)\n",
      "  . }), stackOverflowError = handlers[[1L]])\n",
      "18. tryCatchList(expr, classes, parentenv, handlers)\n",
      "19. tryCatchOne(expr, names, parentenv, handlers[[1L]])\n",
      "20. doTryCatch(return(expr), name, parentenv, handler)\n",
      "21. withCallingHandlers(expr, error = function(cnd) {\n",
      "  .     {\n",
      "  .         .__handler_frame__. <- TRUE\n",
      "  .         .__setup_frame__. <- frame\n",
      "  .     }\n",
      "  .     out <- handlers[[1L]](cnd)\n",
      "  .     if (!inherits(out, \"rlang_zap\")) \n",
      "  .         throw(out)\n",
      "  . })\n",
      "22. f(l = layers[[i]], d = data[[i]])\n",
      "23. l$compute_aesthetics(d, plot)\n",
      "24. compute_aesthetics(..., self = self)\n",
      "25. lapply(aesthetics, eval_tidy, data = data, env = env)\n",
      "26. FUN(X[[i]], ...)\n",
      "27. .handleSimpleError(function (cnd) \n",
      "  . {\n",
      "  .     {\n",
      "  .         .__handler_frame__. <- TRUE\n",
      "  .         .__setup_frame__. <- frame\n",
      "  .     }\n",
      "  .     out <- handlers[[1L]](cnd)\n",
      "  .     if (!inherits(out, \"rlang_zap\")) \n",
      "  .         throw(out)\n",
      "  . }, \"object 'ci' not found\", base::quote(FUN(X[[i]], ...)))\n",
      "28. h(simpleError(msg, call))\n",
      "29. handlers[[1L]](cnd)\n",
      "30. cli::cli_abort(c(\"Problem while {step}.\", i = \"Error occurred in the {ordinal(i)} layer.\"), \n",
      "  .     call = layers[[i]]$constructor, parent = cnd)\n",
      "31. rlang::abort(message, ..., call = call, use_cli_format = TRUE, \n",
      "  .     .frame = .frame)\n",
      "32. signal_abort(cnd, .file)\n"
     ]
    }
   ],
   "source": [
    "figure_a = c('LGD','DMis','DamagingS','OtherMissense', 'Synonymous')\n",
    "label_a = c('LGD','DMis','LGD + DMis','Other missense', 'Synonymous')\n",
    "\n",
    "df_all_stat$category <- factor(df_all_stat$category, levels = figure_a)\n",
    "df_all_stat$cohort <- factor(df_all_stat$cohort, levels = c(\"SSC\", \"MM\"))\n",
    "\n",
    "p_fa <- ggplot(df_all_stat, aes(category, as.numeric(mean))) +\n",
    "  geom_errorbar(aes(x = category, ymin=mean-ci, ymax = mean+ci,group = cohort,color = category),\n",
    "                position = position_dodge(0.5), width = 0.2,size = .3) + \n",
    "  geom_point(data = df_all_stat,\n",
    "             aes(category, as.numeric(mean), group = cohort, color = category, shape = cohort),\n",
    "             size =2.5,\n",
    "             position =position_dodge(0.5)) + \n",
    "  scale_color_manual(values = c(\"#642E44\", \"#A65D7C\",  \"#E5A3B1\", \"#D94929\", \"#A62F14\", \"#C2C0A6\", \"#889C9B\", \"#B2BEBF\")) +\n",
    "  scale_x_discrete(breaks=figure_a,labels=label_a) + \n",
    "  scale_y_continuous(sec.axis = sec_axis(~ . / (100000000 / 2) * total_size, name = 'Theoretical Variant Rate (per child)'), limits= c(0,0.5)) +\n",
    "  labs(fill='Cohort', x='', y='Variant rate (x 10e-8)') +\n",
    "  theme_classic() +\n",
    "  theme(legend.position='bottom', \n",
    "        aspect.ratio = 0.6,\n",
    "        axis.title.y.right = element_text( angle = 90,size = 12),\n",
    "        axis.text.x = element_text(angle = -45, hjust = 0.01,size = 10, color = \"black\", face = \"bold\"),\n",
    "        axis.text.y = element_text(size = 12))+ \n",
    "  #geom_text(data =df_rr, aes(x = category, y = 0.13,\n",
    "     #label = paste(paste(\"RR=\",round(as.numeric(RR),2), sep=\"\"), \n",
    "     #paste(\"p=\",ifelse(pvalue>0.01, round(pvalue,2), formatC(pvalue, format = \"e\", digits = 1)),sep=\"\" ), sep='\\n')), size = 3) + \n",
    "  #geom_text(data =df_rr, \n",
    "            #aes(x = category, y =0.1, label = ifelse(pvalue > 0.05, \"NS\", ifelse(pvalue>0.01, \"*\", ifelse(pvalue > 0.001, \"**\", ifelse(pvalue<= 0.001, \"***\",\"\"))))), size =3) + \n",
    "  ggtitle(\"Without FA\")\n",
    "\n",
    "p_fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 199,
   "id": "e063a1c8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(Intercept)         LGD \n",
      "   1.055991    1.755413 \n",
      "(Intercept)        DMis \n",
      "   1.055325    8.528178 \n",
      "(Intercept)   DamagingS \n",
      "   1.045398    2.391978 \n",
      "  (Intercept) OtherMissense \n",
      "     1.048433      1.560770 \n",
      "(Intercept)  Synonymous \n",
      "   1.064195    1.064966 \n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 39 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Type</th><th scope=col>OR</th><th scope=col>lower</th><th scope=col>upper</th><th scope=col>p</th><th scope=col>tol</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>All          </td><td>1.0149041131636  </td><td>0.915180233716343</td><td>1.12577821990192</td><td>0.779252715938502   </td><td>all genes       </td></tr>\n",
       "\t<tr><td>Nonsynonymous</td><td>1.15367936765932 </td><td>0.927595114899581</td><td>1.43754877838492</td><td>0.200299283879461   </td><td>all genes       </td></tr>\n",
       "\t<tr><td>Missense     </td><td>1.05516810386382 </td><td>0.835910625711855</td><td>1.3334607718297 </td><td>0.651712533618056   </td><td>all genes       </td></tr>\n",
       "\t<tr><td>LGD          </td><td>2.02642265159431 </td><td>1.33320143793986 </td><td>3.16951204591218</td><td>0.0013235577008087  </td><td>all genes       </td></tr>\n",
       "\t<tr><td>DMisL        </td><td>1.10921567949408 </td><td>0.806533181698217</td><td>1.52958442415603</td><td>0.524662389922903   </td><td>all genes       </td></tr>\n",
       "\t<tr><td>DMis         </td><td>1.5194766245253  </td><td>0.888836306573464</td><td>2.66301388081266</td><td>0.132458158206779   </td><td>all genes       </td></tr>\n",
       "\t<tr><td>DMisL2       </td><td>1.16136312213606 </td><td>0.840653844766046</td><td>1.61018939480137</td><td>0.36588472654298    </td><td>all genes       </td></tr>\n",
       "\t<tr><td>DMisM        </td><td>1.08803553532893 </td><td>0.815271648864723</td><td>1.45506965894187</td><td>0.567311918766137   </td><td>all genes       </td></tr>\n",
       "\t<tr><td>DMisMC       </td><td>1.20203967312936 </td><td>0.849189828418962</td><td>1.70981395748909</td><td>0.301464634680291   </td><td>all genes       </td></tr>\n",
       "\t<tr><td>DamagingL    </td><td>1.385311651393   </td><td>1.07964031094929 </td><td>1.78759991198715</td><td>0.0111479222378603  </td><td>all genes       </td></tr>\n",
       "\t<tr><td>DamagingS    </td><td>1.8561006323892  </td><td>1.32694031446135 </td><td>2.63431284814604</td><td>0.000393674585000889</td><td>all genes       </td></tr>\n",
       "\t<tr><td>DamagingL2   </td><td>1.42903160499845 </td><td>1.11086842968071 </td><td>1.84987387704189</td><td>0.00598188932526878 </td><td>all genes       </td></tr>\n",
       "\t<tr><td>DamagingM    </td><td>1.34907158968818 </td><td>1.06562990551066 </td><td>1.71580763630546</td><td>0.0135910285778616  </td><td>all genes       </td></tr>\n",
       "\t<tr><td>DamagingMC   </td><td>1.50803964276796 </td><td>1.15295035037352 </td><td>1.98616505598407</td><td>0.00301938436321749 </td><td>all genes       </td></tr>\n",
       "\t<tr><td>OtherMissense</td><td>0.996551348481482</td><td>0.711914231578562</td><td>1.39663558321953</td><td>0.983915304442229   </td><td>all genes       </td></tr>\n",
       "\t<tr><td>Synonymous   </td><td>1.00955058143754 </td><td>0.725149247299183</td><td>1.4075882152305 </td><td>0.955056006329412   </td><td>all genes       </td></tr>\n",
       "\t<tr><td>All          </td><td>1.19134776469255 </td><td>0.966588209489911</td><td>1.4720833330236 </td><td>0.102278337425742   </td><td>intolerant genes</td></tr>\n",
       "\t<tr><td>Nonsynonymous</td><td>1.98161838161787 </td><td>1.25727692732424 </td><td>3.19352860203635</td><td>0.0038812682237667  </td><td>intolerant genes</td></tr>\n",
       "\t<tr><td>Missense     </td><td>1.88654984749688 </td><td>1.14147233268981 </td><td>3.20111181899537</td><td>0.0152995590188007  </td><td>intolerant genes</td></tr>\n",
       "\t<tr><td>LGD          </td><td>1.75541346867423 </td><td>0.843171769839646</td><td>3.98827300236457</td><td>0.148936385912949   </td><td>intolerant genes</td></tr>\n",
       "\t<tr><td>DMisL        </td><td>2.34911531421198 </td><td>1.18795857671027 </td><td>4.97429693834952</td><td>0.0181565893422255  </td><td>intolerant genes</td></tr>\n",
       "\t<tr><td>DMis         </td><td>8.52817809897646 </td><td>1.59695481666901 </td><td>157.47292072555 </td><td>0.0422457248859166  </td><td>intolerant genes</td></tr>\n",
       "\t<tr><td>DMisL2       </td><td>2.34911531421198 </td><td>1.18795857671027 </td><td>4.97429693834952</td><td>0.0181565893422255  </td><td>intolerant genes</td></tr>\n",
       "\t<tr><td>DMisM        </td><td>2.13469303240336 </td><td>1.12407985417548 </td><td>4.27733366560767</td><td>0.024806512876364   </td><td>intolerant genes</td></tr>\n",
       "\t<tr><td>DMisMC       </td><td>3.11930294905437 </td><td>1.46628509116807 </td><td>7.41372738006986</td><td>0.00526290269613478 </td><td>intolerant genes</td></tr>\n",
       "\t<tr><td>DMisMC2      </td><td>2.13469303240336 </td><td>1.12407985417548 </td><td>4.27733366560767</td><td>0.024806512876364   </td><td>intolerant genes</td></tr>\n",
       "\t<tr><td>DamagingL    </td><td>2.09235621452412 </td><td>1.25600848213888 </td><td>3.62241927562666</td><td>0.00598632067048762 </td><td>intolerant genes</td></tr>\n",
       "\t<tr><td>DamagingS    </td><td>2.39197763672603 </td><td>1.2181270501809  </td><td>5.1466752092237 </td><td>0.0163773851280988  </td><td>intolerant genes</td></tr>\n",
       "\t<tr><td>DamagingL2   </td><td>2.09235621452412 </td><td>1.25600848213888 </td><td>3.62241927562666</td><td>0.00598632067048762 </td><td>intolerant genes</td></tr>\n",
       "\t<tr><td>DamagingM    </td><td>2.08780375381845 </td><td>1.2665426951599  </td><td>3.56893705618837</td><td>0.00509415317797204 </td><td>intolerant genes</td></tr>\n",
       "\t<tr><td>DamagingMC   </td><td>2.39506669264023 </td><td>1.39820013550767 </td><td>4.31137800609215</td><td>0.00220855087667449 </td><td>intolerant genes</td></tr>\n",
       "\t<tr><td>DamagingMC2  </td><td>2.08780375381845 </td><td>1.2665426951599  </td><td>3.56893705618837</td><td>0.00509415317797204 </td><td>intolerant genes</td></tr>\n",
       "\t<tr><td>OtherMissense</td><td>1.56077047084866 </td><td>0.916466386928426</td><td>2.71717377837684</td><td>0.106332325310337   </td><td>intolerant genes</td></tr>\n",
       "\t<tr><td>Synonymous   </td><td>1.06496560853317 </td><td>0.601958007477081</td><td>1.89818191942114</td><td>0.828600840865385   </td><td>intolerant genes</td></tr>\n",
       "\t<tr><td>LGD          </td><td>1.75541346867423 </td><td>0.843171769839646</td><td>3.98827300236457</td><td>0.148936385912949   </td><td>Intol genes     </td></tr>\n",
       "\t<tr><td>DMis         </td><td>8.52817809897646 </td><td>1.59695481666901 </td><td>157.47292072555 </td><td>0.0422457248859166  </td><td>Intol genes     </td></tr>\n",
       "\t<tr><td>DamagingS    </td><td>2.39197763672603 </td><td>1.2181270501809  </td><td>5.1466752092237 </td><td>0.0163773851280988  </td><td>Intol genes     </td></tr>\n",
       "\t<tr><td>OtherMissense</td><td>1.56077047084866 </td><td>0.916466386928426</td><td>2.71717377837684</td><td>0.106332325310337   </td><td>Intol genes     </td></tr>\n",
       "\t<tr><td>Synonymous   </td><td>1.06496560853317 </td><td>0.601958007477081</td><td>1.89818191942114</td><td>0.828600840865385   </td><td>Intol genes     </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 39 × 6\n",
       "\\begin{tabular}{llllll}\n",
       " Type & OR & lower & upper & p & tol\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t All           & 1.0149041131636   & 0.915180233716343 & 1.12577821990192 & 0.779252715938502    & all genes       \\\\\n",
       "\t Nonsynonymous & 1.15367936765932  & 0.927595114899581 & 1.43754877838492 & 0.200299283879461    & all genes       \\\\\n",
       "\t Missense      & 1.05516810386382  & 0.835910625711855 & 1.3334607718297  & 0.651712533618056    & all genes       \\\\\n",
       "\t LGD           & 2.02642265159431  & 1.33320143793986  & 3.16951204591218 & 0.0013235577008087   & all genes       \\\\\n",
       "\t DMisL         & 1.10921567949408  & 0.806533181698217 & 1.52958442415603 & 0.524662389922903    & all genes       \\\\\n",
       "\t DMis          & 1.5194766245253   & 0.888836306573464 & 2.66301388081266 & 0.132458158206779    & all genes       \\\\\n",
       "\t DMisL2        & 1.16136312213606  & 0.840653844766046 & 1.61018939480137 & 0.36588472654298     & all genes       \\\\\n",
       "\t DMisM         & 1.08803553532893  & 0.815271648864723 & 1.45506965894187 & 0.567311918766137    & all genes       \\\\\n",
       "\t DMisMC        & 1.20203967312936  & 0.849189828418962 & 1.70981395748909 & 0.301464634680291    & all genes       \\\\\n",
       "\t DamagingL     & 1.385311651393    & 1.07964031094929  & 1.78759991198715 & 0.0111479222378603   & all genes       \\\\\n",
       "\t DamagingS     & 1.8561006323892   & 1.32694031446135  & 2.63431284814604 & 0.000393674585000889 & all genes       \\\\\n",
       "\t DamagingL2    & 1.42903160499845  & 1.11086842968071  & 1.84987387704189 & 0.00598188932526878  & all genes       \\\\\n",
       "\t DamagingM     & 1.34907158968818  & 1.06562990551066  & 1.71580763630546 & 0.0135910285778616   & all genes       \\\\\n",
       "\t DamagingMC    & 1.50803964276796  & 1.15295035037352  & 1.98616505598407 & 0.00301938436321749  & all genes       \\\\\n",
       "\t OtherMissense & 0.996551348481482 & 0.711914231578562 & 1.39663558321953 & 0.983915304442229    & all genes       \\\\\n",
       "\t Synonymous    & 1.00955058143754  & 0.725149247299183 & 1.4075882152305  & 0.955056006329412    & all genes       \\\\\n",
       "\t All           & 1.19134776469255  & 0.966588209489911 & 1.4720833330236  & 0.102278337425742    & intolerant genes\\\\\n",
       "\t Nonsynonymous & 1.98161838161787  & 1.25727692732424  & 3.19352860203635 & 0.0038812682237667   & intolerant genes\\\\\n",
       "\t Missense      & 1.88654984749688  & 1.14147233268981  & 3.20111181899537 & 0.0152995590188007   & intolerant genes\\\\\n",
       "\t LGD           & 1.75541346867423  & 0.843171769839646 & 3.98827300236457 & 0.148936385912949    & intolerant genes\\\\\n",
       "\t DMisL         & 2.34911531421198  & 1.18795857671027  & 4.97429693834952 & 0.0181565893422255   & intolerant genes\\\\\n",
       "\t DMis          & 8.52817809897646  & 1.59695481666901  & 157.47292072555  & 0.0422457248859166   & intolerant genes\\\\\n",
       "\t DMisL2        & 2.34911531421198  & 1.18795857671027  & 4.97429693834952 & 0.0181565893422255   & intolerant genes\\\\\n",
       "\t DMisM         & 2.13469303240336  & 1.12407985417548  & 4.27733366560767 & 0.024806512876364    & intolerant genes\\\\\n",
       "\t DMisMC        & 3.11930294905437  & 1.46628509116807  & 7.41372738006986 & 0.00526290269613478  & intolerant genes\\\\\n",
       "\t DMisMC2       & 2.13469303240336  & 1.12407985417548  & 4.27733366560767 & 0.024806512876364    & intolerant genes\\\\\n",
       "\t DamagingL     & 2.09235621452412  & 1.25600848213888  & 3.62241927562666 & 0.00598632067048762  & intolerant genes\\\\\n",
       "\t DamagingS     & 2.39197763672603  & 1.2181270501809   & 5.1466752092237  & 0.0163773851280988   & intolerant genes\\\\\n",
       "\t DamagingL2    & 2.09235621452412  & 1.25600848213888  & 3.62241927562666 & 0.00598632067048762  & intolerant genes\\\\\n",
       "\t DamagingM     & 2.08780375381845  & 1.2665426951599   & 3.56893705618837 & 0.00509415317797204  & intolerant genes\\\\\n",
       "\t DamagingMC    & 2.39506669264023  & 1.39820013550767  & 4.31137800609215 & 0.00220855087667449  & intolerant genes\\\\\n",
       "\t DamagingMC2   & 2.08780375381845  & 1.2665426951599   & 3.56893705618837 & 0.00509415317797204  & intolerant genes\\\\\n",
       "\t OtherMissense & 1.56077047084866  & 0.916466386928426 & 2.71717377837684 & 0.106332325310337    & intolerant genes\\\\\n",
       "\t Synonymous    & 1.06496560853317  & 0.601958007477081 & 1.89818191942114 & 0.828600840865385    & intolerant genes\\\\\n",
       "\t LGD           & 1.75541346867423  & 0.843171769839646 & 3.98827300236457 & 0.148936385912949    & Intol genes     \\\\\n",
       "\t DMis          & 8.52817809897646  & 1.59695481666901  & 157.47292072555  & 0.0422457248859166   & Intol genes     \\\\\n",
       "\t DamagingS     & 2.39197763672603  & 1.2181270501809   & 5.1466752092237  & 0.0163773851280988   & Intol genes     \\\\\n",
       "\t OtherMissense & 1.56077047084866  & 0.916466386928426 & 2.71717377837684 & 0.106332325310337    & Intol genes     \\\\\n",
       "\t Synonymous    & 1.06496560853317  & 0.601958007477081 & 1.89818191942114 & 0.828600840865385    & Intol genes     \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 39 × 6\n",
       "\n",
       "| Type &lt;chr&gt; | OR &lt;chr&gt; | lower &lt;chr&gt; | upper &lt;chr&gt; | p &lt;chr&gt; | tol &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "| All           | 1.0149041131636   | 0.915180233716343 | 1.12577821990192 | 0.779252715938502    | all genes        |\n",
       "| Nonsynonymous | 1.15367936765932  | 0.927595114899581 | 1.43754877838492 | 0.200299283879461    | all genes        |\n",
       "| Missense      | 1.05516810386382  | 0.835910625711855 | 1.3334607718297  | 0.651712533618056    | all genes        |\n",
       "| LGD           | 2.02642265159431  | 1.33320143793986  | 3.16951204591218 | 0.0013235577008087   | all genes        |\n",
       "| DMisL         | 1.10921567949408  | 0.806533181698217 | 1.52958442415603 | 0.524662389922903    | all genes        |\n",
       "| DMis          | 1.5194766245253   | 0.888836306573464 | 2.66301388081266 | 0.132458158206779    | all genes        |\n",
       "| DMisL2        | 1.16136312213606  | 0.840653844766046 | 1.61018939480137 | 0.36588472654298     | all genes        |\n",
       "| DMisM         | 1.08803553532893  | 0.815271648864723 | 1.45506965894187 | 0.567311918766137    | all genes        |\n",
       "| DMisMC        | 1.20203967312936  | 0.849189828418962 | 1.70981395748909 | 0.301464634680291    | all genes        |\n",
       "| DamagingL     | 1.385311651393    | 1.07964031094929  | 1.78759991198715 | 0.0111479222378603   | all genes        |\n",
       "| DamagingS     | 1.8561006323892   | 1.32694031446135  | 2.63431284814604 | 0.000393674585000889 | all genes        |\n",
       "| DamagingL2    | 1.42903160499845  | 1.11086842968071  | 1.84987387704189 | 0.00598188932526878  | all genes        |\n",
       "| DamagingM     | 1.34907158968818  | 1.06562990551066  | 1.71580763630546 | 0.0135910285778616   | all genes        |\n",
       "| DamagingMC    | 1.50803964276796  | 1.15295035037352  | 1.98616505598407 | 0.00301938436321749  | all genes        |\n",
       "| OtherMissense | 0.996551348481482 | 0.711914231578562 | 1.39663558321953 | 0.983915304442229    | all genes        |\n",
       "| Synonymous    | 1.00955058143754  | 0.725149247299183 | 1.4075882152305  | 0.955056006329412    | all genes        |\n",
       "| All           | 1.19134776469255  | 0.966588209489911 | 1.4720833330236  | 0.102278337425742    | intolerant genes |\n",
       "| Nonsynonymous | 1.98161838161787  | 1.25727692732424  | 3.19352860203635 | 0.0038812682237667   | intolerant genes |\n",
       "| Missense      | 1.88654984749688  | 1.14147233268981  | 3.20111181899537 | 0.0152995590188007   | intolerant genes |\n",
       "| LGD           | 1.75541346867423  | 0.843171769839646 | 3.98827300236457 | 0.148936385912949    | intolerant genes |\n",
       "| DMisL         | 2.34911531421198  | 1.18795857671027  | 4.97429693834952 | 0.0181565893422255   | intolerant genes |\n",
       "| DMis          | 8.52817809897646  | 1.59695481666901  | 157.47292072555  | 0.0422457248859166   | intolerant genes |\n",
       "| DMisL2        | 2.34911531421198  | 1.18795857671027  | 4.97429693834952 | 0.0181565893422255   | intolerant genes |\n",
       "| DMisM         | 2.13469303240336  | 1.12407985417548  | 4.27733366560767 | 0.024806512876364    | intolerant genes |\n",
       "| DMisMC        | 3.11930294905437  | 1.46628509116807  | 7.41372738006986 | 0.00526290269613478  | intolerant genes |\n",
       "| DMisMC2       | 2.13469303240336  | 1.12407985417548  | 4.27733366560767 | 0.024806512876364    | intolerant genes |\n",
       "| DamagingL     | 2.09235621452412  | 1.25600848213888  | 3.62241927562666 | 0.00598632067048762  | intolerant genes |\n",
       "| DamagingS     | 2.39197763672603  | 1.2181270501809   | 5.1466752092237  | 0.0163773851280988   | intolerant genes |\n",
       "| DamagingL2    | 2.09235621452412  | 1.25600848213888  | 3.62241927562666 | 0.00598632067048762  | intolerant genes |\n",
       "| DamagingM     | 2.08780375381845  | 1.2665426951599   | 3.56893705618837 | 0.00509415317797204  | intolerant genes |\n",
       "| DamagingMC    | 2.39506669264023  | 1.39820013550767  | 4.31137800609215 | 0.00220855087667449  | intolerant genes |\n",
       "| DamagingMC2   | 2.08780375381845  | 1.2665426951599   | 3.56893705618837 | 0.00509415317797204  | intolerant genes |\n",
       "| OtherMissense | 1.56077047084866  | 0.916466386928426 | 2.71717377837684 | 0.106332325310337    | intolerant genes |\n",
       "| Synonymous    | 1.06496560853317  | 0.601958007477081 | 1.89818191942114 | 0.828600840865385    | intolerant genes |\n",
       "| LGD           | 1.75541346867423  | 0.843171769839646 | 3.98827300236457 | 0.148936385912949    | Intol genes      |\n",
       "| DMis          | 8.52817809897646  | 1.59695481666901  | 157.47292072555  | 0.0422457248859166   | Intol genes      |\n",
       "| DamagingS     | 2.39197763672603  | 1.2181270501809   | 5.1466752092237  | 0.0163773851280988   | Intol genes      |\n",
       "| OtherMissense | 1.56077047084866  | 0.916466386928426 | 2.71717377837684 | 0.106332325310337    | Intol genes      |\n",
       "| Synonymous    | 1.06496560853317  | 0.601958007477081 | 1.89818191942114 | 0.828600840865385    | Intol genes      |\n",
       "\n"
      ],
      "text/plain": [
       "   Type          OR                lower             upper           \n",
       "1  All           1.0149041131636   0.915180233716343 1.12577821990192\n",
       "2  Nonsynonymous 1.15367936765932  0.927595114899581 1.43754877838492\n",
       "3  Missense      1.05516810386382  0.835910625711855 1.3334607718297 \n",
       "4  LGD           2.02642265159431  1.33320143793986  3.16951204591218\n",
       "5  DMisL         1.10921567949408  0.806533181698217 1.52958442415603\n",
       "6  DMis          1.5194766245253   0.888836306573464 2.66301388081266\n",
       "7  DMisL2        1.16136312213606  0.840653844766046 1.61018939480137\n",
       "8  DMisM         1.08803553532893  0.815271648864723 1.45506965894187\n",
       "9  DMisMC        1.20203967312936  0.849189828418962 1.70981395748909\n",
       "10 DamagingL     1.385311651393    1.07964031094929  1.78759991198715\n",
       "11 DamagingS     1.8561006323892   1.32694031446135  2.63431284814604\n",
       "12 DamagingL2    1.42903160499845  1.11086842968071  1.84987387704189\n",
       "13 DamagingM     1.34907158968818  1.06562990551066  1.71580763630546\n",
       "14 DamagingMC    1.50803964276796  1.15295035037352  1.98616505598407\n",
       "15 OtherMissense 0.996551348481482 0.711914231578562 1.39663558321953\n",
       "16 Synonymous    1.00955058143754  0.725149247299183 1.4075882152305 \n",
       "17 All           1.19134776469255  0.966588209489911 1.4720833330236 \n",
       "18 Nonsynonymous 1.98161838161787  1.25727692732424  3.19352860203635\n",
       "19 Missense      1.88654984749688  1.14147233268981  3.20111181899537\n",
       "20 LGD           1.75541346867423  0.843171769839646 3.98827300236457\n",
       "21 DMisL         2.34911531421198  1.18795857671027  4.97429693834952\n",
       "22 DMis          8.52817809897646  1.59695481666901  157.47292072555 \n",
       "23 DMisL2        2.34911531421198  1.18795857671027  4.97429693834952\n",
       "24 DMisM         2.13469303240336  1.12407985417548  4.27733366560767\n",
       "25 DMisMC        3.11930294905437  1.46628509116807  7.41372738006986\n",
       "26 DMisMC2       2.13469303240336  1.12407985417548  4.27733366560767\n",
       "27 DamagingL     2.09235621452412  1.25600848213888  3.62241927562666\n",
       "28 DamagingS     2.39197763672603  1.2181270501809   5.1466752092237 \n",
       "29 DamagingL2    2.09235621452412  1.25600848213888  3.62241927562666\n",
       "30 DamagingM     2.08780375381845  1.2665426951599   3.56893705618837\n",
       "31 DamagingMC    2.39506669264023  1.39820013550767  4.31137800609215\n",
       "32 DamagingMC2   2.08780375381845  1.2665426951599   3.56893705618837\n",
       "33 OtherMissense 1.56077047084866  0.916466386928426 2.71717377837684\n",
       "34 Synonymous    1.06496560853317  0.601958007477081 1.89818191942114\n",
       "35 LGD           1.75541346867423  0.843171769839646 3.98827300236457\n",
       "36 DMis          8.52817809897646  1.59695481666901  157.47292072555 \n",
       "37 DamagingS     2.39197763672603  1.2181270501809   5.1466752092237 \n",
       "38 OtherMissense 1.56077047084866  0.916466386928426 2.71717377837684\n",
       "39 Synonymous    1.06496560853317  0.601958007477081 1.89818191942114\n",
       "   p                    tol             \n",
       "1  0.779252715938502    all genes       \n",
       "2  0.200299283879461    all genes       \n",
       "3  0.651712533618056    all genes       \n",
       "4  0.0013235577008087   all genes       \n",
       "5  0.524662389922903    all genes       \n",
       "6  0.132458158206779    all genes       \n",
       "7  0.36588472654298     all genes       \n",
       "8  0.567311918766137    all genes       \n",
       "9  0.301464634680291    all genes       \n",
       "10 0.0111479222378603   all genes       \n",
       "11 0.000393674585000889 all genes       \n",
       "12 0.00598188932526878  all genes       \n",
       "13 0.0135910285778616   all genes       \n",
       "14 0.00301938436321749  all genes       \n",
       "15 0.983915304442229    all genes       \n",
       "16 0.955056006329412    all genes       \n",
       "17 0.102278337425742    intolerant genes\n",
       "18 0.0038812682237667   intolerant genes\n",
       "19 0.0152995590188007   intolerant genes\n",
       "20 0.148936385912949    intolerant genes\n",
       "21 0.0181565893422255   intolerant genes\n",
       "22 0.0422457248859166   intolerant genes\n",
       "23 0.0181565893422255   intolerant genes\n",
       "24 0.024806512876364    intolerant genes\n",
       "25 0.00526290269613478  intolerant genes\n",
       "26 0.024806512876364    intolerant genes\n",
       "27 0.00598632067048762  intolerant genes\n",
       "28 0.0163773851280988   intolerant genes\n",
       "29 0.00598632067048762  intolerant genes\n",
       "30 0.00509415317797204  intolerant genes\n",
       "31 0.00220855087667449  intolerant genes\n",
       "32 0.00509415317797204  intolerant genes\n",
       "33 0.106332325310337    intolerant genes\n",
       "34 0.828600840865385    intolerant genes\n",
       "35 0.148936385912949    Intol genes     \n",
       "36 0.0422457248859166   Intol genes     \n",
       "37 0.0163773851280988   Intol genes     \n",
       "38 0.106332325310337    Intol genes     \n",
       "39 0.828600840865385    Intol genes     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "tol_type = \"Intol genes\"\n",
    "\n",
    "\n",
    "for (eff in figure_a) {\n",
    "    state <- paste(\"MM ~ \", eff)\n",
    "    x <- glm(as.formula(state),\n",
    "             family = binomial,\n",
    "             data = df)\n",
    "    suppressMessages(a<- confint(x))\n",
    "\n",
    "    coef <- exp(coef(x))\n",
    "    print (coef)[2]\n",
    "    result =  c(eff, coef[2], exp(a)[2,1], exp(a)[2,2], summary(x)$coefficient[2,4], tol_type) \n",
    "    #print (result)\n",
    "    df_odds = rbind(df_odds, result)\n",
    "}\n",
    "df_odds"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 200,
   "id": "b0c2627b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 18 × 10</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>category</th><th scope=col>RR</th><th scope=col>pvalue</th><th scope=col>MM</th><th scope=col>SSC</th><th scope=col>Gene</th><th scope=col>Lower</th><th scope=col>Upper</th><th scope=col>MM</th><th scope=col>SSC</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>All          </td><td>1.015</td><td>0.36750219</td><td>MM</td><td>SSC</td><td>All</td><td>0.946314210007404</td><td>Inf</td><td>719</td><td>664</td></tr>\n",
       "\t<tr><td>Nonsynonymous</td><td>1.146</td><td>0.06068071</td><td>MM</td><td>SSC</td><td>All</td><td>0.991864762164627</td><td>Inf</td><td>187</td><td>153</td></tr>\n",
       "\t<tr><td>Missense     </td><td>1.051</td><td>0.3108332 </td><td>MM</td><td>SSC</td><td>All</td><td>0.899748854658688</td><td>Inf</td><td>155</td><td>138</td></tr>\n",
       "\t<tr><td>LGD          </td><td>2.095</td><td>9.07e-06  </td><td>MM</td><td>SSC</td><td>All</td><td>1.55232208946951 </td><td>Inf</td><td>65 </td><td>29 </td></tr>\n",
       "\t<tr><td>DMisL        </td><td>1.102</td><td>0.23590122</td><td>MM</td><td>SSC</td><td>All</td><td>0.89180362641801 </td><td>Inf</td><td>87 </td><td>74 </td></tr>\n",
       "\t<tr><td>DMis         </td><td>1.517</td><td>0.03400915</td><td>MM</td><td>SSC</td><td>All</td><td>1.03904139279005 </td><td>Inf</td><td>34 </td><td>21 </td></tr>\n",
       "\t<tr><td>DMisL2       </td><td>1.143</td><td>0.15889109</td><td>MM</td><td>SSC</td><td>All</td><td>0.922002711183897</td><td>Inf</td><td>86 </td><td>70 </td></tr>\n",
       "\t<tr><td>DMisM        </td><td>1.086</td><td>0.25142529</td><td>MM</td><td>SSC</td><td>All</td><td>0.895992573688703</td><td>Inf</td><td>104</td><td>90 </td></tr>\n",
       "\t<tr><td>DMisMC       </td><td>1.192</td><td>0.1111023 </td><td>MM</td><td>SSC</td><td>All</td><td>0.944142160308373</td><td>Inf</td><td>75 </td><td>59 </td></tr>\n",
       "\t<tr><td>DMisMC2      </td><td>1.124</td><td>0.17644896</td><td>MM</td><td>SSC</td><td>All</td><td>0.919095701303745</td><td>Inf</td><td>97 </td><td>81 </td></tr>\n",
       "\t<tr><td>DamagingL    </td><td>1.387</td><td>0.00059199</td><td>MM</td><td>SSC</td><td>All</td><td>1.17120452064468 </td><td>Inf</td><td>152</td><td>103</td></tr>\n",
       "\t<tr><td>DamagingS    </td><td>1.864</td><td>1.84e-06  </td><td>MM</td><td>SSC</td><td>All</td><td>1.47922093198684 </td><td>Inf</td><td>99 </td><td>50 </td></tr>\n",
       "\t<tr><td>DamagingL2   </td><td>1.427</td><td>0.00024288</td><td>MM</td><td>SSC</td><td>All</td><td>1.20242128398786 </td><td>Inf</td><td>151</td><td>99 </td></tr>\n",
       "\t<tr><td>DamagingM    </td><td>1.345</td><td>0.00089926</td><td>MM</td><td>SSC</td><td>All</td><td>1.14761833255911 </td><td>Inf</td><td>169</td><td>118</td></tr>\n",
       "\t<tr><td>DamagingMC   </td><td>1.489</td><td>0.00010054</td><td>MM</td><td>SSC</td><td>All</td><td>1.24291430217307 </td><td>Inf</td><td>140</td><td>88 </td></tr>\n",
       "\t<tr><td>DamagingMC2  </td><td>1.393</td><td>0.00034555</td><td>MM</td><td>SSC</td><td>All</td><td>1.1826688880035  </td><td>Inf</td><td>162</td><td>109</td></tr>\n",
       "\t<tr><td>OtherMissense</td><td>0.992</td><td>0.5510505 </td><td>MM</td><td>SSC</td><td>All</td><td>0.784862516565366</td><td>Inf</td><td>68 </td><td>64 </td></tr>\n",
       "\t<tr><td>Synonymous   </td><td>1.018</td><td>0.4740654 </td><td>MM</td><td>SSC</td><td>All</td><td>0.807887116097002</td><td>Inf</td><td>70 </td><td>65 </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 18 × 10\n",
       "\\begin{tabular}{llllllllll}\n",
       " category & RR & pvalue & MM & SSC & Gene & Lower & Upper & MM & SSC\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t All           & 1.015 & 0.36750219 & MM & SSC & All & 0.946314210007404 & Inf & 719 & 664\\\\\n",
       "\t Nonsynonymous & 1.146 & 0.06068071 & MM & SSC & All & 0.991864762164627 & Inf & 187 & 153\\\\\n",
       "\t Missense      & 1.051 & 0.3108332  & MM & SSC & All & 0.899748854658688 & Inf & 155 & 138\\\\\n",
       "\t LGD           & 2.095 & 9.07e-06   & MM & SSC & All & 1.55232208946951  & Inf & 65  & 29 \\\\\n",
       "\t DMisL         & 1.102 & 0.23590122 & MM & SSC & All & 0.89180362641801  & Inf & 87  & 74 \\\\\n",
       "\t DMis          & 1.517 & 0.03400915 & MM & SSC & All & 1.03904139279005  & Inf & 34  & 21 \\\\\n",
       "\t DMisL2        & 1.143 & 0.15889109 & MM & SSC & All & 0.922002711183897 & Inf & 86  & 70 \\\\\n",
       "\t DMisM         & 1.086 & 0.25142529 & MM & SSC & All & 0.895992573688703 & Inf & 104 & 90 \\\\\n",
       "\t DMisMC        & 1.192 & 0.1111023  & MM & SSC & All & 0.944142160308373 & Inf & 75  & 59 \\\\\n",
       "\t DMisMC2       & 1.124 & 0.17644896 & MM & SSC & All & 0.919095701303745 & Inf & 97  & 81 \\\\\n",
       "\t DamagingL     & 1.387 & 0.00059199 & MM & SSC & All & 1.17120452064468  & Inf & 152 & 103\\\\\n",
       "\t DamagingS     & 1.864 & 1.84e-06   & MM & SSC & All & 1.47922093198684  & Inf & 99  & 50 \\\\\n",
       "\t DamagingL2    & 1.427 & 0.00024288 & MM & SSC & All & 1.20242128398786  & Inf & 151 & 99 \\\\\n",
       "\t DamagingM     & 1.345 & 0.00089926 & MM & SSC & All & 1.14761833255911  & Inf & 169 & 118\\\\\n",
       "\t DamagingMC    & 1.489 & 0.00010054 & MM & SSC & All & 1.24291430217307  & Inf & 140 & 88 \\\\\n",
       "\t DamagingMC2   & 1.393 & 0.00034555 & MM & SSC & All & 1.1826688880035   & Inf & 162 & 109\\\\\n",
       "\t OtherMissense & 0.992 & 0.5510505  & MM & SSC & All & 0.784862516565366 & Inf & 68  & 64 \\\\\n",
       "\t Synonymous    & 1.018 & 0.4740654  & MM & SSC & All & 0.807887116097002 & Inf & 70  & 65 \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 18 × 10\n",
       "\n",
       "| category &lt;chr&gt; | RR &lt;chr&gt; | pvalue &lt;chr&gt; | MM &lt;chr&gt; | SSC &lt;chr&gt; | Gene &lt;chr&gt; | Lower &lt;chr&gt; | Upper &lt;chr&gt; | MM &lt;chr&gt; | SSC &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|\n",
       "| All           | 1.015 | 0.36750219 | MM | SSC | All | 0.946314210007404 | Inf | 719 | 664 |\n",
       "| Nonsynonymous | 1.146 | 0.06068071 | MM | SSC | All | 0.991864762164627 | Inf | 187 | 153 |\n",
       "| Missense      | 1.051 | 0.3108332  | MM | SSC | All | 0.899748854658688 | Inf | 155 | 138 |\n",
       "| LGD           | 2.095 | 9.07e-06   | MM | SSC | All | 1.55232208946951  | Inf | 65  | 29  |\n",
       "| DMisL         | 1.102 | 0.23590122 | MM | SSC | All | 0.89180362641801  | Inf | 87  | 74  |\n",
       "| DMis          | 1.517 | 0.03400915 | MM | SSC | All | 1.03904139279005  | Inf | 34  | 21  |\n",
       "| DMisL2        | 1.143 | 0.15889109 | MM | SSC | All | 0.922002711183897 | Inf | 86  | 70  |\n",
       "| DMisM         | 1.086 | 0.25142529 | MM | SSC | All | 0.895992573688703 | Inf | 104 | 90  |\n",
       "| DMisMC        | 1.192 | 0.1111023  | MM | SSC | All | 0.944142160308373 | Inf | 75  | 59  |\n",
       "| DMisMC2       | 1.124 | 0.17644896 | MM | SSC | All | 0.919095701303745 | Inf | 97  | 81  |\n",
       "| DamagingL     | 1.387 | 0.00059199 | MM | SSC | All | 1.17120452064468  | Inf | 152 | 103 |\n",
       "| DamagingS     | 1.864 | 1.84e-06   | MM | SSC | All | 1.47922093198684  | Inf | 99  | 50  |\n",
       "| DamagingL2    | 1.427 | 0.00024288 | MM | SSC | All | 1.20242128398786  | Inf | 151 | 99  |\n",
       "| DamagingM     | 1.345 | 0.00089926 | MM | SSC | All | 1.14761833255911  | Inf | 169 | 118 |\n",
       "| DamagingMC    | 1.489 | 0.00010054 | MM | SSC | All | 1.24291430217307  | Inf | 140 | 88  |\n",
       "| DamagingMC2   | 1.393 | 0.00034555 | MM | SSC | All | 1.1826688880035   | Inf | 162 | 109 |\n",
       "| OtherMissense | 0.992 | 0.5510505  | MM | SSC | All | 0.784862516565366 | Inf | 68  | 64  |\n",
       "| Synonymous    | 1.018 | 0.4740654  | MM | SSC | All | 0.807887116097002 | Inf | 70  | 65  |\n",
       "\n"
      ],
      "text/plain": [
       "   category      RR    pvalue     MM SSC Gene Lower             Upper MM  SSC\n",
       "1  All           1.015 0.36750219 MM SSC All  0.946314210007404 Inf   719 664\n",
       "2  Nonsynonymous 1.146 0.06068071 MM SSC All  0.991864762164627 Inf   187 153\n",
       "3  Missense      1.051 0.3108332  MM SSC All  0.899748854658688 Inf   155 138\n",
       "4  LGD           2.095 9.07e-06   MM SSC All  1.55232208946951  Inf   65  29 \n",
       "5  DMisL         1.102 0.23590122 MM SSC All  0.89180362641801  Inf   87  74 \n",
       "6  DMis          1.517 0.03400915 MM SSC All  1.03904139279005  Inf   34  21 \n",
       "7  DMisL2        1.143 0.15889109 MM SSC All  0.922002711183897 Inf   86  70 \n",
       "8  DMisM         1.086 0.25142529 MM SSC All  0.895992573688703 Inf   104 90 \n",
       "9  DMisMC        1.192 0.1111023  MM SSC All  0.944142160308373 Inf   75  59 \n",
       "10 DMisMC2       1.124 0.17644896 MM SSC All  0.919095701303745 Inf   97  81 \n",
       "11 DamagingL     1.387 0.00059199 MM SSC All  1.17120452064468  Inf   152 103\n",
       "12 DamagingS     1.864 1.84e-06   MM SSC All  1.47922093198684  Inf   99  50 \n",
       "13 DamagingL2    1.427 0.00024288 MM SSC All  1.20242128398786  Inf   151 99 \n",
       "14 DamagingM     1.345 0.00089926 MM SSC All  1.14761833255911  Inf   169 118\n",
       "15 DamagingMC    1.489 0.00010054 MM SSC All  1.24291430217307  Inf   140 88 \n",
       "16 DamagingMC2   1.393 0.00034555 MM SSC All  1.1826688880035   Inf   162 109\n",
       "17 OtherMissense 0.992 0.5510505  MM SSC All  0.784862516565366 Inf   68  64 \n",
       "18 Synonymous    1.018 0.4740654  MM SSC All  0.807887116097002 Inf   70  65 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "\n",
    "\n",
    "df_rr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 201,
   "id": "359e3853",
   "metadata": {},
   "outputs": [],
   "source": [
    "figure_a = c('LGD','DMisS','DMisN','DamagingS', 'DamagingN','OtherMissense', 'Synonymous')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6158a572",
   "metadata": {},
   "source": [
    "## Main Fig1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3bd008cf",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "df_all_stat_backup <- df_all_stat\n",
    "\n",
    "\n",
    "figure_a = c('LGD','LGD_FS', 'LGD_STOP','DMisS','DamagingS','OtherMissense', 'Synonymous')\n",
    "\n",
    "df_all_stat$category <- factor(df_all_stat$category, levels = figure_a)\n",
    "\n",
    "p_con <- ggplot(subset(df_all_stat, class %in% figure_a & Gene ==\"Constrained genes\"), aes(category, as.numeric(mean))) +\n",
    "geom_errorbar(aes(x = category, ymin=mean-ci, ymax = mean+ci,group = cohort,color = category),\n",
    "              position = position_dodge(0.5), width = 0.2,size = .3) + \n",
    "geom_point(data = subset(df_all_stat, class %in% figure_a & Gene ==\"Constrained genes\"),\n",
    "           aes(category, as.numeric(mean), group = cohort, color = category, shape = cohort),\n",
    "        size =1.4,\n",
    "          position =position_dodge(0.5)) + \n",
    "#scale_color_manual(values = c(\"#590E0E\", \"#F28157\", \"#A65221\", \"#F28157\", \"#F2B950\")) +\n",
    "scale_color_manual(values = c(\"#261A1A\", \"#642E44\", \"#A65D7C\",  \"#F2DFDF\", \"#C2C0A6\")) +\n",
    "scale_y_continuous(sec.axis = sec_axis(~ . / (100000000 / 2) * total_size, name = 'Theoretical Variant Rate (per child)')) +\n",
    "    labs(fill='Cohort', x='', y='Variant rate (x 10e-8)') +\n",
    "theme_classic() +\n",
    "    theme(legend.position='bottom', \n",
    "          aspect.ratio = 0.6,\n",
    "          axis.title.y.right = element_text( angle = 90,size = 12),\n",
    "         axis.text.x = element_text(angle = -45, hjust = 0.01,size = 12),\n",
    "          axis.text.y = element_text(size = 12))+ \n",
    "geom_text(data =subset(df_rr,  category %in% figure_a & Gene == 'Constrained'), aes(x = category, y = 0.12,\n",
    "                                                          label = paste(paste(\"RR=\",round(as.numeric(RR),2), sep=\"\"), \n",
    "                                                                        paste(\"p=\",pvalue,sep=\"\" ), sep='\\n')), size = 5) +\n",
    "    ggtitle(\"Constrained genes\")\n",
    "\n",
    "\n",
    "\n",
    "p_all <- ggplot() +\n",
    "geom_errorbar(data = subset(df_all_stat, class %in% figure_a & Gene ==\"All genes\"),aes(x = category, ymin=mean-ci, ymax = mean+ci,group = cohort,color = category),\n",
    "              position = position_dodge(0.5), width = 0.2,size = .3) + \n",
    "geom_point(data = subset(df_all_stat, class %in% figure_a & Gene ==\"All genes\"),\n",
    "           aes(category, as.numeric(mean), group = cohort, color = category, shape = cohort),\n",
    "        size =1.4,\n",
    "          position =position_dodge(0.5)) + \n",
    "#scale_color_manual(values = c(\"#590E0E\", \"#F28157\", \"#A65221\", \"#F28157\", \"#F2B950\")) +\n",
    "scale_color_manual(values = c(\"#261A1A\", \"#642E44\", \"#A65D7C\",  \"#F2DFDF\", \"#C2C0A6\")) +\n",
    "scale_y_continuous(sec.axis = sec_axis(~ . / (100000000 / 2) * total_size, name = 'Theoretical Variant Rate (per child)')) +\n",
    "    labs(fill='Cohort', x='', y='Variant rate (x 10e-8)') +\n",
    "theme_classic() +\n",
    "    theme(legend.position='bottom', \n",
    "          aspect.ratio = 0.6,\n",
    "          axis.title.y.right = element_text( angle = 90,size = 20),\n",
    "         axis.text.x = element_text(angle = -45, hjust = 0.01,size = 12),\n",
    "          axis.text.y = element_text(size = 12))+ \n",
    "geom_text(data =subset(df_rr,  category %in% figure_a & Gene == 'All'), aes(x = category, y = 0.4,\n",
    "                                                  label = paste(paste(\"RR=\",round(as.numeric(RR),2), sep=\"\"), \n",
    "                                                                paste(\"p=\",pvalue,sep=\"\" ), sep='\\n')), size = 5) + \n",
    "    ggtitle(\"All genes\")\n",
    "\n",
    "\n",
    "p_combined <- ggarrange(p_con, p_all)\n",
    "ggsave(\"/projects/ps-gleesonlab8/User/hiyoothere/NTD/5.Analysis/DNM/Poisson/NewRate.pdf\", plot= p_combined)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8fbb0849",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_rr\n",
    "p_combined"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "35172e35",
   "metadata": {},
   "source": [
    "## Supplementary FIg"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "583cb201",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_all_stat <- df_all_stat_backup\n",
    "\n",
    "figure_a = c('LGD','DMisS','DamagingS', 'DMisN','DamagingN')\n",
    "\n",
    "df_all_stat$category <- factor(df_all_stat$category, levels = figure_a)\n",
    "\n",
    "p_con <- ggplot(subset(df_all_stat, class %in% figure_a & Gene ==\"All genes\"), aes(category, as.numeric(mean))) +\n",
    "geom_errorbar(aes(x = category, ymin=mean-ci, ymax = mean+ci,group = cohort,color = category),\n",
    "              position = position_dodge(0.5), width = 0.2,size = .3) + \n",
    "geom_point(data = subset(df_all_stat, class %in% figure_a & Gene ==\"All genes\"),\n",
    "           aes(category, as.numeric(mean), group = cohort, color = category, shape = cohort),\n",
    "        size =1.4,\n",
    "          position =position_dodge(0.5)) + \n",
    "#scale_color_manual(values = c(\"#590E0E\", \"#F28157\", \"#A65221\", \"#F28157\", \"#F2B950\")) +\n",
    "scale_color_manual(values = c(\"#261A1A\", \"#642E44\",   \"#A65D7C\", \"#426866\", \"#7D8C82\")) +\n",
    "scale_y_continuous(sec.axis = sec_axis(~ . / (100000000 / 2) * total_size, name = 'Theoretical Variant Rate (per child)')) +\n",
    "    labs(fill='Cohort', x='', y='Variant rate (x 10e-8)') +\n",
    "theme_classic() +\n",
    "    theme(legend.position='bottom', \n",
    "          aspect.ratio = 0.6,\n",
    "          axis.title.y.right = element_text( angle = 90,size = 12),\n",
    "         axis.text.x = element_text(angle = -45, hjust = 0.01,size = 12),\n",
    "          axis.text.y = element_text(size = 12))+ \n",
    "geom_text(data =subset(df_rr,  category %in% figure_a & Gene == 'All'), aes(x = category, y = 0.3,\n",
    "                                                          label = paste(paste(\"RR=\",round(as.numeric(RR),2), sep=\"\"), \n",
    "                                                                        paste(\"p=\",pvalue,sep=\"\" ), sep='\\n')), size = 5) +\n",
    "    ggtitle(\"DamaingL\")\n",
    "\n",
    "#Other damaging by predictors\n",
    "df_all_stat <- df_all_stat_backup\n",
    "figure_a = c('DMisS','DMisMetaLR', \"DMisREVEL\",'DamagingS', 'DamagingMetaLR','DamagingREVEL')\n",
    "df_all_stat$category <- factor(df_all_stat$category, levels = figure_a)\n",
    "p_all <- ggplot() +\n",
    "geom_errorbar(data = subset(df_all_stat, class %in% figure_a & Gene ==\"All genes\"),aes(x = category, ymin=mean-ci, ymax = mean+ci,group = cohort,color = category),\n",
    "              position = position_dodge(0.5), width = 0.2,size = .3) + \n",
    "geom_point(data = subset(df_all_stat, class %in% figure_a & Gene ==\"All genes\"),\n",
    "           aes(category, as.numeric(mean), group = cohort, color = category, shape = cohort),\n",
    "        size =1.4,\n",
    "          position =position_dodge(0.5)) + \n",
    "#scale_color_manual(values = c(\"#590E0E\", \"#F28157\", \"#A65221\", \"#F28157\", \"#F2B950\")) +\n",
    "scale_color_manual(values = c(\"#642E44\", \"#F28444\", \"#204473\",  \"#A65D7C\", \"#F29E38\", \"#607EA6\")) +\n",
    "scale_y_continuous(sec.axis = sec_axis(~ . / (100000000 / 2) * total_size, name = 'Theoretical Variant Rate (per child)')) +\n",
    "    labs(fill='Cohort', x='', y='Variant rate (x 10e-8)') +\n",
    "theme_classic() +\n",
    "    theme(legend.position='bottom', \n",
    "          aspect.ratio = 0.7,\n",
    "          axis.title.y.right = element_text( angle = 90,size = 20),\n",
    "         axis.text.x = element_text(angle = -45, hjust = 0.01,size = 12),\n",
    "          axis.text.y = element_text(size = 12))+ \n",
    "geom_text(data =subset(df_rr,  category %in% figure_a & Gene == 'All'), aes(x = category, y = 0.3,\n",
    "                                                  label = paste(paste(\"RR=\",round(as.numeric(RR),2), sep=\"\"), \n",
    "                                                                paste(\"p=\",pvalue,sep=\"\" ), sep='\\n')), size = 5) + \n",
    "    ggtitle(\"Other predictions\")\n",
    "\n",
    "\n",
    "p_combined <- ggarrange(p_con, p_all)\n",
    "ggsave(\"/projects/ps-gleesonlab8/User/hiyoothere/NTD/15.Final/221020_DNM_v9/figure/NewRate_supple.pdf\", plot= p_combined)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1e7ce520",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_rr\n",
    "#df_all_stat\n",
    "p_combined"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6b1018f0",
   "metadata": {},
   "outputs": [],
   "source": [
    "figure_a = c('LGD','DMisS' ,'DamagingS', 'Missense', 'All')\n",
    "df_odds_sub <- df_odds\n",
    "df_odds_sub$Type <- factor(df_odds_sub$Type, levels = figure_a)\n",
    "colnaames(df_odds_sub) <- c(\"Type\",\"OR\", \"lower\", \"upper\", \"p\", \"tol\")\n",
    "df_odds_sub\n",
    "ggplot(subset(df_odds_sub, Type %in% figure_a), aes(tol, as.numeric(OR))) + \n",
    "geom_hline(yintercept = 1, linetype = \"dashed\", color = \"darkgray\", size=1 ) +\n",
    "geom_pointrange(aes(ymin = as.numeric(lower),ymax=as.numeric(upper),group = Type, \n",
    "                    color = Type), position = position_dodge(0.8), size = 1.5)+\n",
    "ylim(c(0.67,4)) +\n",
    "theme_bw() +\n",
    "scale_color_manual(values = c(\"#590E0E\", \"#F28157\", \"#A65221\", \"#F28157\", \"#F2B950\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fe2e5fd1",
   "metadata": {},
   "outputs": [],
   "source": [
    "figure_a = c('DMisS','DMisL', 'LGD','DMisS', 'DMisL' ,'DamagingS','DamagingL', 'Missense', 'All')\n",
    "\n",
    "colnames(df_odds_sub) <- c(\"Type\",\"OR\", \"lower\", \"upper\", \"p\", \"tol\")\n",
    "df_odds_sub\n",
    "ggplot(subset(df_odds_sub, Type %in% figure_a), aes(tol, log2(as.numeric(OR)))) + \n",
    "geom_pointrange(aes(ymin = log2(as.numeric(lower)),ymax=log2(as.numeric(upper)),group = Type, color = Type), position = position_dodge(0.3))+\n",
    "ylim(c(0,10)) # + scale_color_manual(values = c(\"#00AFBB\", \"#E7B800\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a98c8264",
   "metadata": {},
   "outputs": [],
   "source": [
    "head(df)\n",
    "dim(df)\n",
    "df$MM <- ifelse(df$cohort == \"MM\", 1, 0)\n",
    "head(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b5b8a910",
   "metadata": {},
   "outputs": [],
   "source": [
    "tol_type = \"all genes\"\n",
    "df_odds = data.frame()\n",
    "\n",
    "for (eff in figure_a) {\n",
    "    state <- paste(\"MM ~ \", eff)\n",
    "    x <- glm(as.formula(state),\n",
    "             family = binomial,\n",
    "             data = df)\n",
    "    suppressMessages(a<- confint(x))\n",
    "\n",
    "    coef <- exp(coef(x))\n",
    "    print (coef)[2]\n",
    "    result =  c(eff, coef[2], exp(a)[2,1], exp(a)[2,2], summary(x)$coefficient[2,4], tol_type) \n",
    "    #print (result)\n",
    "    df_odds = rbind(df_odds, result)\n",
    "}\n",
    "\n",
    "#df_odds\n",
    "colnames(df_odds) <- c(\"Type\",\"OR\", \"2.5%\", \"97.5%\", \"p\", \"tol\")\n",
    "df_odds"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2667cafa",
   "metadata": {},
   "outputs": [],
   "source": [
    "summary(x)\n",
    "\n",
    "\n",
    "t = exp(coef(reduced.model))\n",
    "t\n",
    "t[2][1]\n",
    "x = reduced.model\n",
    "suppressMessages(a<- confint(x))\n",
    "exp(a)\n",
    "exp(a)[2,2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4c083185",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4e4020a4",
   "metadata": {},
   "outputs": [],
   "source": [
    "MM <- c(57,66,69,66)\n",
    "SSC <- c(36,56,60,56)\n",
    "df_type <- data.frame()\n",
    "df_type <- rbind(df_type, MM)\n",
    "df_type <- rbind(df_type, SSC)\n",
    "colnames(df_type) <- c(\"LGD\", \"DMisS\", \"OtherMis\", \"Syn\")\n",
    "rownames(df_type) <- c(\"MM\", \"SSC\")\n",
    "df_type\n",
    "df_type.melt <- melt(as.matrix(df_type))\n",
    "df_type.melt\n",
    "\n",
    "ggplot(data=df_type.melt, aes(x=Var2,y=as.integer(df_type.melt$value), fill= Var1)) + \n",
    "stat_density()\n",
    "#stat_smooth(geom=\"area\", method = \"loess\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a91599ba",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "f26f6c0b",
   "metadata": {},
   "source": [
    "# Get Damaging variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8c7fd2f8",
   "metadata": {},
   "outputs": [],
   "source": [
    "data.raw = data.fil\n",
    "data_missense <- subset(data.raw, 'missense_variant' %in% data.raw$Consequence & data.raw$batch != 'SSC' & MetaLR_score != 'NA')\n",
    "#data_missense\n",
    "head(data.raw)\n",
    "p1 <- ggplot(data_missense, aes(y=as.numeric(metasvm_score))) + \n",
    "geom_point(aes(x=as.numeric(data_missense$MetaLR_score),color = as.factor(data_missense$metasvm),shape = as.factor(data_missense$MetaLR_pred)), size = 3) + \n",
    "scale_color_manual(values = c(\"#8C0428\", \"#B2BEBF\")) +  \n",
    "scale_shape_manual(values=c (17, 19)) + \n",
    "theme(legend.position='bottom', \n",
    "          aspect.ratio = 1.2,\n",
    "          axis.title.y.right = element_text( angle = 90,size = 12),\n",
    "         axis.text.x = element_text(angle = -45, hjust = 0.01,size = 12),\n",
    "          axis.text.y = element_text(size = 12)) + \n",
    "xlab(\"MetaLR score\") + \n",
    "ylab(\"MetaSVM score\")\n",
    "\n",
    "\n",
    "p2 <- ggplot(data_missense, aes(y=as.numeric(metasvm_score))) + \n",
    "geom_point(aes(x=as.numeric(data_missense$REVEL_score),color = data_missense$metasvm ), size = 3) + \n",
    "geom_vline(aes(xintercept = 0.5), linetype = \"dashed\") + \n",
    "scale_color_manual(values = c(\"#8C0428\", \"#B2BEBF\")) +  \n",
    " theme(legend.position='bottom', \n",
    "          aspect.ratio = 1.2,\n",
    "          axis.title.y.right = element_text( angle = 90,size = 12),\n",
    "         axis.text.x = element_text(angle = -45, hjust = 0.01,size = 12),\n",
    "          axis.text.y = element_text(size = 12))+\n",
    "xlab(\"REVEL score\") + \n",
    "ylab(\"MetaSVM score\")\n",
    "\n",
    "p3<- ggplot(data_missense, aes(y=as.numeric(metasvm_score))) + \n",
    "geom_point(aes(x=as.numeric(data_missense$cadd),color = data_missense$metasvm ), size = 3) + \n",
    "geom_vline(aes(xintercept = 20), linetype = \"dashed\") + \n",
    "scale_color_manual(values = c(\"#8C0428\", \"#B2BEBF\")) +  \n",
    " theme(legend.position='bottom', \n",
    "          aspect.ratio = 1.2,\n",
    "          axis.title.y.right = element_text( angle = 90,size = 12),\n",
    "         axis.text.x = element_text(angle = -45, hjust = 0.01,size = 12),\n",
    "          axis.text.y = element_text(size = 12))+\n",
    "xlab(\"CADD score\") + \n",
    "ylab(\"MetaSVM score\")\n",
    "\n",
    "\n",
    "p_predict <- ggarrange(p1, p2, p3, ncol = 3)\n",
    "p_predict\n",
    "\n",
    "ggsave(\"/projects/ps-gleesonlab8/User/hiyoothere/NTD/15.Final/221020_DNM_v9/figure/Predict_Damaging.pdf\", plot= p_predict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d832d895",
   "metadata": {},
   "outputs": [],
   "source": [
    "#percent= '70'\n",
    "#data.poi = read.delim(paste(\"/projects/ps-gleesonlab8/User/hiyoothere/NTD/15.Final/221020_DNM_v9/NTD_WES_HC_all.VQSR.CGP.DNV.DeNovo.vep.filtered.DP12.RESCUE.PF2.consensus\",percent, \".den_input\", sep=\"\"), header=T)\n",
    "#head(data_poi)\n",
    "head(data.raw)\n",
    "data_missense_ntd <- subset(data.raw, data.raw$Consequence == 'missense_variant' & data.raw$batch != 'SSC' )\n",
    "data_dmis_ntd <- subset(data_missense_ntd, data_missense_ntd$metasvm == 'D' )\n",
    "data_lgd_ntd <- subset(data.raw, data.raw$impact == 'HIGH' & data.raw$batch != 'SSC' &  data.raw$Consequence != 'missense_variant' )\n",
    "data_dmis_ntd_constrained <- subset(data_dmis_ntd, data_dmis_ntd$pLI >= 0.9)\n",
    "data_lgd_ntd_constrained <- subset(data_lgd_ntd, data_lgd_ntd$pLI >= 0.9)\n",
    "dim(data_dmis_ntd_constrained)\n",
    "data_dmis_ntd_constrained$gene\n",
    "\n",
    "dim(data_lgd_ntd_constrained)\n",
    "data_lgd_ntd_constrained$gene\n",
    "\n",
    "#two LGD must be filtered out based on poisson filter 6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "daec2649",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "\n",
    "\n",
    "\n",
    "data.raw.mm <- subset(data.raw, data.raw$batch != 'SSC' & as.numeric(data.raw$pLI >=0.9))\n",
    "#data.raw.mm\n",
    "data.raw.ssc <- subset(data.raw, data.raw$batch == 'SSC'& as.numeric(data.raw$pLI >=0.9))\n",
    "\n",
    "#ggplot(data.raw.mm, aes(y=oe_lof, x=metasvm_score)) + \n",
    "#geom_point(aes(color = data.raw.mm$impact), alpha = 0.4, size = 3)\n",
    "\n",
    "#MetaLR_pred\tMetaLR_rankscore\tMetaLR_score\tREVEL_rankscore\tREVEL_score"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "32ebd4fa",
   "metadata": {},
   "outputs": [],
   "source": [
    "data.raw.mm.LGD <- subset(data.raw.mm, data.raw.mm$impact == \"HIGH\")\n",
    "data.raw.mm.DMisS <- subset(data.raw.mm, data.raw.mm$metasvm == \"D\")\n",
    "\n",
    "data.raw.ssc.LGD <- subset(data.raw.ssc, data.raw.ssc$impact == \"HIGH\")\n",
    "data.raw.ssc.DMisS <- subset(data.raw.ssc, data.raw.ssc$metasvm == \"D\")\n",
    "\n",
    "summary(data.raw.mm.LGD$oe_lof)\n",
    "summary(data.raw.ssc.LGD$oe_lof)\n",
    "\n",
    "summary(data.raw.mm.DMisS$oe_mis_pphen)\n",
    "summary(data.raw.ssc.DMisS$oe_mis_pphen)\n",
    "\n",
    "wilcox.test(data.raw.mm.LGD$oe_lof,data.raw.ssc.LGD$oe_lof )\n",
    "\n",
    "wilcox.test(data.raw.mm.DMisS$oe_mis_pphen,data.raw.ssc.DMisS$oe_mis_pphen )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "de19f140",
   "metadata": {},
   "outputs": [],
   "source": [
    "data.raw.mm <- subset(data.raw, data.raw$batch != 'SSC' & as.numeric(data.raw$pLI >=0.9))\n",
    "ggplot() + \n",
    "\n",
    "geom_density(data=subset(data.raw.mm, data.raw.mm$impact == 'HIGH'),aes(x=oe_lof), color=\"red\") + \n",
    "geom_density(data=subset(data.raw.mm, data.raw.mm$metasvm == 'D' & data.raw.mm$impact != 'HIGH'), aes(x=oe_lof), color=\"green\")  +\n",
    "#geom_density(data=subset(data.raw.mm, data.raw.mm$metasvm == 'D'), aes(x=oe_mis_pphen), color=\"blue\")  +\n",
    "#geom_vline(data=subset(data.raw.mm, data.raw.mm$impact == 'HIGH'),aes(xintercept=median(oe_lof)), color=\"red\") + \n",
    "#geom_vline(data=subset(data.raw.mm, data.raw.mm$impact == 'D'),aes(xintercept=median(oe_lof)), color=\"green\") +\n",
    "#geom_vline(data=subset(data.raw.mm, data.raw.mm$impact == 'D'),aes(xintercept=median(oe_mis_pphen)), color=\"blue\") +\n",
    "xlim(c(0,1.5))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f95ab019",
   "metadata": {},
   "outputs": [],
   "source": [
    "data.raw.mm <- subset(data.raw, data.raw$batch != 'SSC' & as.numeric(data.raw$pLI >=0.9))\n",
    "ggplot() + \n",
    "#DMisS gene의 lof vs Mis oe\n",
    "geom_density(data=subset(data.raw.mm, data.raw.mm$metasvm == 'D' & data.raw.mm$impact != 'HIGH'),aes(x=oe_lof), color=\"red\") + \n",
    "geom_density(data=subset(data.raw.mm, data.raw.mm$metasvm == 'D' & data.raw.mm$impact != 'HIGH'), aes(x=oe_mis_pphen), color=\"green\")  +\n",
    "#geom_density(data=subset(data.raw.mm, data.raw.mm$metasvm == 'D'), aes(x=oe_mis_pphen), color=\"blue\")  +\n",
    "#geom_vline(data=subset(data.raw.mm, data.raw.mm$impact == 'HIGH'),aes(xintercept=median(oe_lof)), color=\"red\") + \n",
    "#geom_vline(data=subset(data.raw.mm, data.raw.mm$impact == 'D'),aes(xintercept=median(oe_lof)), color=\"green\") +\n",
    "#geom_vline(data=subset(data.raw.mm, data.raw.mm$impact == 'D'),aes(xintercept=median(oe_mis_pphen)), color=\"blue\") +\n",
    "xlim(c(0,1.5))\n",
    "\n",
    "\n",
    "#DMisS gene의 lof vs Mis oe\n",
    "ggplot() + \n",
    "geom_density(data=subset(data.raw.mm, data.raw.mm$impact != 'HIGH'),aes(x=oe_lof), color=\"red\") + \n",
    "geom_density(data=subset(data.raw.mm, data.raw.mm$impact != 'HIGH'), aes(x=oe_mis_pphen), color=\"green\")  +\n",
    "#geom_density(data=subset(data.raw.mm, data.raw.mm$metasvm == 'D'), aes(x=oe_mis_pphen), color=\"blue\")  +\n",
    "#geom_vline(data=subset(data.raw.mm, data.raw.mm$impact == 'HIGH'),aes(xintercept=median(oe_lof)), color=\"red\") + \n",
    "#geom_vline(data=subset(data.raw.mm, data.raw.mm$impact == 'D'),aes(xintercept=median(oe_lof)), color=\"green\") +\n",
    "#geom_vline(data=subset(data.raw.mm, data.raw.mm$impact == 'D'),aes(xintercept=median(oe_mis_pphen)), color=\"blue\") +\n",
    "xlim(c(0,1.5))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7f84eab4",
   "metadata": {},
   "outputs": [],
   "source": [
    "oe = read.delim('/projects/ps-gleesonlab5/user/sangmoon/0_DB/gnomAD_constraint/gnomad.v2.1.1.lof_metrics.by_gene.txt')\n",
    "dim(oe)\n",
    "oe = oe[!duplicated(oe$gene), ]\n",
    "head(oe)\n",
    "oe$gene = factor(oe$gene, levels=oe$gene)\n",
    "oe = oe[, c(1,10,21,22,23,24,29,30)]\n",
    "oe_unique <- unique(oe)\n",
    "head(oe)\n",
    "#oe<- subset(oe, oe$pLI >= 0.9)\n",
    "dim(oe)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "414ba91b",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "#data.raw.mm\n",
    "data.raw.mm <- subset(data.raw, data.raw$batch != 'SSC' & as.numeric(data.raw$pLI >=0.9))\n",
    "oe<- subset(oe, oe$pLI >= 0.9)\n",
    "ggplot() + \n",
    "\n",
    "geom_density(data=subset(data.raw.mm, data.raw.mm$impact == 'HIGH'),aes(x=oe_lof), color=\"red\") + \n",
    "geom_density(data=subset(data.raw.mm, data.raw.mm$metasvm == 'D'), aes(x=oe_lof), color=\"green\")  +\n",
    "#geom_density(data=subset(data.raw.mm, data.raw.mm$metasvm == 'D'), aes(x=oe_mis_pphen), fill=\"blue\", alpha = 0.4)  +\n",
    "geom_vline(data=subset(data.raw.mm, data.raw.mm$impact == 'HIGH'),aes(xintercept=median(oe_lof)), color=\"red\") + \n",
    "geom_vline(data=subset(data.raw.mm, data.raw.mm$impact == 'D'),aes(xintercept=median(oe_lof)), color=\"green\") +\n",
    "#geom_vline(data=subset(data.raw.mm, data.raw.mm$impact == 'D'),aes(xintercept=median(oe_mis_pphen)), color=\"blue\") + \n",
    "xlim(c(0,1.5)) + \n",
    "geom_density(data=oe, aes(x=oe_lof), color = \"black\")\n",
    "\n",
    "ggplot() + \n",
    "geom_density(data=subset(data.raw.mm, data.raw.mm$impact == 'HIGH'),aes(x=oe_mis_pphen), color=\"red\") + \n",
    "geom_density(data=subset(data.raw.mm, data.raw.mm$metasvm == 'D'), aes(x=oe_mis_pphen), color=\"green\")  +\n",
    "#geom_density(data=subset(data.raw.mm, data.raw.mm$metasvm == 'D'), aes(x=oe_mis_pphen), fill=\"blue\", alpha = 0.4)  +\n",
    "geom_vline(data=subset(data.raw.mm, data.raw.mm$impact == 'HIGH'),aes(xintercept=median(oe_lof)), color=\"red\") + \n",
    "geom_vline(data=subset(data.raw.mm, data.raw.mm$impact == 'D'),aes(xintercept=median(oe_lof)), color=\"green\") +\n",
    "#geom_vline(data=subset(data.raw.mm, data.raw.mm$impact == 'D'),aes(xintercept=median(oe_mis_pphen)), color=\"blue\") + \n",
    "xlim(c(0,1.5)) + \n",
    "geom_density(data=oe, aes(x=oe_mis_pphen), color = \"black\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dfa86ef2",
   "metadata": {},
   "outputs": [],
   "source": [
    "data.raw.ssc <- subset(data.raw, data.raw$batch == 'SSC'& as.numeric(data.raw$pLI >= 0))\n",
    "ggplot() + \n",
    "geom_density(data=subset(data.raw.ssc, data.raw.ssc$impact == 'HIGH'),aes(x=oe_lof, y= ..density..), color=\"red\") + \n",
    "geom_density(data=subset(data.raw.ssc, data.raw.ssc$metasvm == 'D'),aes(x=oe_lof, y= ..density..), color=\"green\")  +\n",
    "#geom_density(data=subset(data.raw.ssc, data.raw.ssc$impact == 'HIGH'), aes(x=oe_mis_pphen, y= -..density..), color=\"orange\") + \n",
    "geom_density(data=subset(data.raw.ssc, data.raw.ssc$metasvm == 'D'), aes(x=oe_mis_pphen, y= ..density..), color=\"blue\")  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d51672ad",
   "metadata": {},
   "outputs": [],
   "source": [
    "ggplot(data=subset(data.raw.mm, data.raw.mm$impact == 'HIGH')) + \n",
    "\n",
    "geom_density(data=subset(data.raw.mm, data.raw.mm$impact == 'HIGH'),aes(x=oe_lof), color=\"red\") + \n",
    "geom_density(data=subset(data.raw.mm, data.raw.mm$metasvm == 'D'), aes(x=oe_lof), color=\"green\")  +\n",
    "geom_vline(aes(xintercept=median(oe_lof)), color=\"red\") \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "691a1d09",
   "metadata": {},
   "outputs": [],
   "source": [
    "head(data.raw)\n",
    "NTD_LGD <- subset(data.raw, data.raw$batch != \"SSC\" &  data.raw$impact == \"HIGH\" )\n",
    "NTD_DMis <- subset(data.raw, data.raw$batch != \"SSC\" &  data.raw$metasvm== 'D' )\n",
    "dim(NTD_LGD)\n",
    "\n",
    "NTD_Dam <- subset(data.raw, data.raw$batch != \"SSC\" & (( data.raw$impact == \"HIGH\" |  data.raw$metasvm== 'D') )) \n",
    "dim(data.raw)[1]\n",
    "dim(NTD_Dam)[1]\n",
    "#table(NTD_Dam$term)\n",
    "\n",
    "\n",
    "NTD_Dam2 <- subset(data.raw, data.raw$batch != \"SSC\" & (( data.raw$impact == \"HIGH\" |  grepl(\"damaging\",data.raw$polyphen)) )) \n",
    "dim(NTD_Dam2)[1]\n",
    "DamagingL = subset(data.raw, data.raw$batch != \"SSC\" & (data.raw$impact =='HIGH' |  grepl(\"damaging\",data.raw$polyphen)))\n",
    "\"DamaingL\"\n",
    "head(DamagingL)\n",
    "dim(DamagingL)\n",
    "\n",
    "NTD_Dam3 <- subset(data.raw, data.raw$batch != \"SSC\" & (( data.raw$impact == \"HIGH\" | \n",
    "                                                         data.raw$metasvm== 'D' |\n",
    "                                                         (data.raw$term == 'missense_variant' & \n",
    "                        \n",
    "                                                                                      grepl(\"damaging\",data.raw$polyphen) &\n",
    "                                                                                      grepl(\"deleterious\",data.raw$sift) &\n",
    "                                                                                      as.numeric(data.raw$cadd) > 20\n",
    "                                                                                                as.numeric(data.raw$metasvm_score) >=0.7))) ))\n",
    "\"Dam3: \" \n",
    "dim(NTD_Dam3)[1]\n",
    "\n",
    "write.table(NTD_Dam,\"/projects/ps-gleesonlab8/User/hiyoothere/NTD/15.Final/221020_DNM_v9/NTD_DamS.den_input\", sep='\\t', quote = F, row.names = F )\n",
    "write.table(NTD_Dam2,\"/projects/ps-gleesonlab8/User/hiyoothere/NTD/15.Final/221020_DNM_v9/NTD_DamL.den_input\", sep='\\t', quote = F, row.names = F )\n",
    "write.table(NTD_Dam3,\"/projects/ps-gleesonlab8/User/hiyoothere/NTD/15.Final/221020_DNM_v9/NTD_DamN.den_input\", sep='\\t', quote = F, row.names = F )\n",
    "\n",
    "SSC_Dam <- subset(data.raw, data.raw$batch == \"SSC\" & (( data.raw$impact == \"HIGH\" | (data.raw$term == 'missense_variant' & data.raw$metasvm== 'D') )) )\n",
    "dim(data.raw)[1]\n",
    "dim(SSC_Dam)[1]\n",
    "#table(SSC_Dam$term)\n",
    "\n",
    "SSC_Dam2 <- subset(data.raw, data.raw$batch == \"SSC\" & (( data.raw$impact == \"HIGH\" | (data.raw$term == 'missense_variant' & grepl(\"damaging\",data.raw$polyphen)) )) )\n",
    "dim(SSC_Dam2)[1]\n",
    "\n",
    "write.table(SSC_Dam,\"/projects/ps-gleesonlab8/User/hiyoothere/NTD/15.Final/221020_DNM_v9/SSC_DamS.den_input\", sep='\\t', quote = F, row.names = F )\n",
    "write.table(SSC_Dam2,\"a\", sep='\\t', quote = F, row.names = F )\n",
    "\n",
    "    table(NTD_Dam$term) \n",
    "      table(NTD_Dam3$term)     \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b68450d8",
   "metadata": {},
   "outputs": [],
   "source": [
    "NTD_DMis_Con <- subset(NTD_DMis, NTD_DMis$pLI >= 0.9)\n",
    "\n",
    "dim(NTD_DMis)\n",
    "dim(NTD_DMis_Con)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4716345e",
   "metadata": {},
   "outputs": [],
   "source": [
    "install.packages(\"enrichplot\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "809d77a0",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "abcec009",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "data.fil <- data.con\n",
    "\n",
    "length(table(data.fil$Consequence))\n",
    "\n",
    "table(data.fil$Consequence)\n",
    "#Replace pattern according to the vep consequence table\n",
    "pattern1 = \"frameshift_variant&NMD_transcript_variant\"\n",
    "replace1 = \"frameshift_variant\"\n",
    "\n",
    "pattern2 = \"frameshift_variant&splice_region_variant\"\n",
    "replace2 = \"frameshift_variant\"\n",
    "\n",
    "pattern3= \"frameshift_variant&start_lost\"\n",
    "replace3 = \"frameshift_variant\"\n",
    "\n",
    "pattern4 = \"frameshift_variant&stop_lost\"\n",
    "replace4 = \"frameshift_variant\"\n",
    "\n",
    "pattern5 = \"inframe_deletion&NMD_transcript_variant\"\n",
    "replace5= \"inframe_deletion\"\n",
    "\n",
    "pattern6 = \"inframe_deletion&splice_region_variant\"\n",
    "replace6= \"inframe_deletion\"\n",
    "\n",
    "pattern7 = \"inframe_insertion&NMD_transcript_variant\"\n",
    "replace7= \"inframe_insertion\"\n",
    "\n",
    "pattern8 = \"inframe_insertion&NMD_transcript_variant\"\n",
    "replace8= \"inframe_insertion\"\n",
    "\n",
    "pattern9 = \"inframe_insertion&splice_region_variant\"\n",
    "replace9= \"inframe_insertion\"\n",
    "\n",
    "pattern10 = \"missense_variant&NMD_transcript_variant\"\n",
    "replace10= \"missense_variant\"\n",
    "\n",
    "pattern11 = \"missense_variant&&splice_region_variant\"\n",
    "replace11 = \"missense_variant\"\n",
    "\n",
    "pattern12 = \"missense_variant&splice_region_variant&NMD_transcript_variant\"\n",
    "replace12 = \"missense_variant\"\n",
    "\n",
    "pattern13 = \"protein_altering_variant&splice_region_variant\"\n",
    "replace13 = \"protein_altering_variant\"\n",
    "\n",
    "pattern14 = \"splice_acceptor_variant&5_prime_UTR_variant\"\n",
    "replace14 = \"splice_acceptor_variant\"\n",
    "\n",
    "pattern15 = \"splice_acceptor_variant&coding_sequence_variant\"\n",
    "replace15 = \"splice_acceptor_variant\"\n",
    "\n",
    "pattern16 = \"splice_acceptor_variant&NMD_transcript_variant\"\n",
    "replace16 = \"splice_acceptor_variant\"\n",
    "\n",
    "pattern17 = \"splice_acceptor_variant&non_coding_transcript_variant\"\n",
    "replace17 = \"splice_acceptor_variant\"\n",
    "\n",
    "pattern18 = \"splice_acceptor_variant&NMD_transcript_variant\"\n",
    "replace18 = \"splice_acceptor_variant\"\n",
    "\n",
    "pattern19 = \"splice_acceptor_variant&splice_polypyrimidine_tract_variant&3_prime_UTR_variant&intron_variant&NMD_transcript_variant\"\n",
    "replace19 = \"splice_acceptor_variant\"\n",
    "\n",
    "pattern20 = \"splice_acceptor_variant&splice_polypyrimidine_tract_variant&5_prime_UTR_variant&intron_variant\"\n",
    "replace20 = \"splice_acceptor_variant\"\n",
    "\n",
    "pattern21 = \"splice_acceptor_variant&splice_polypyrimidine_tract_variant&5_prime_UTR_variant&intron_variant&NMD_transcript_variant\"\n",
    "replace21 = \"splice_acceptor_variant\"\n",
    "\n",
    "pattern22 = \"splice_acceptor_variant&splice_polypyrimidine_tract_variant&coding_sequence_variant&intron_variant\"\n",
    "replace22 = \"splice_acceptor_variant\"\n",
    "\n",
    "pattern23 = \"splice_acceptor_variant&splice_polypyrimidine_tract_variant&intron_variant\"\n",
    "replace23 = \"splice_acceptor_variant\"\n",
    "\n",
    "pattern24 = \"splice_acceptor_variant&splice_polypyrimidine_tract_variant&non_coding_transcript_exon_variant&intron_variant\"\n",
    "replace24 = \"splice_acceptor_variant\"\n",
    "\n",
    "pattern25 = \"splice_donor_variant&coding_sequence_variant\"\n",
    "replace25 = \"splice_donor_variant\"\n",
    "\n",
    "pattern26 = \"splice_donor_variant&NMD_transcript_variant\"\n",
    "replace26 = \"splice_donor_variant\"\n",
    "\n",
    "pattern27 = \"splice_donor_variant&non_coding_transcript_variant\"\n",
    "replace27 = \"splice_donor_variant\"\n",
    "\n",
    "pattern28 = \"splice_donor_variant&splice_acceptor_variant&splice_donor_5th_base_variant&splice_polypyrimidine_tract_variant&coding_sequence_variant&intron_variant\"\n",
    "replace28 = \"splice_donor_variant\"\n",
    "\n",
    "pattern29 = \"splice_donor_variant&splice_donor_5th_base_variant&coding_sequence_variant&intron_variant\"\n",
    "replace29 = \"splice_donor_variant\"\n",
    "\n",
    "pattern30 = \"splice_donor_variant&splice_donor_5th_base_variant&intron_variant\"\n",
    "replace30 = \"splice_donor_variant\"\n",
    "\n",
    "pattern31 = \"splice_donor_variant&splice_donor_5th_base_variant&non_coding_transcript_exon_variant&intron_variant\"\n",
    "replace31 = \"splice_donor_variant\"\n",
    "\n",
    "pattern32 = \"splice_donor_variant&splice_donor_5th_base_variant&splice_polypyrimidine_tract_variant&coding_sequence_variant&intron_variant\"\n",
    "replace32 = \"splice_donor_variant\"\n",
    "\n",
    "\n",
    "pattern33 = \"splice_donor_variant&splice_donor_region_variant&coding_sequence_variant&intron_variant\"\n",
    "replace33 = \"splice_donor_variant\"\n",
    "\n",
    "pattern34 = \"splice_donor_variant&splice_donor_region_variant&intron_variant\"\n",
    "replace34 = \"splice_donor_variant\"\n",
    "\n",
    "pattern35 = \"splice_donor_variant&splice_donor_region_variant&coding_sequence_variant&intron_variant\"\n",
    "replace35 = \"splice_donor_variant\"\n",
    "\n",
    "pattern36 = \"splice_donor_variant&splice_donor_region_variant&coding_sequence_variant&intron_variant\"\n",
    "replace36 = \"splice_donor_variant\"\n",
    "\n",
    "pattern37 = \"start_lost&5_prime_UTR_variant\"\n",
    "replace37 = \"start_lost\"\n",
    "\n",
    "pattern38 = \"start_lost&inframe_deletion\"\n",
    "replace38 = \"start_lost\"\n",
    "\n",
    "pattern39 = \"start_lost&NMD_transcript_variant\"\n",
    "replace39 = \"start_lost\"\n",
    "###stop_gained&frameshift_varaint --> \n",
    "pattern40 = \"stop_gained&frameshift_variant\"\n",
    "replace40 = \"stop_gained\"\n",
    "\n",
    "pattern41 = \"stop_gained&inframe_insertion\"\n",
    "replace41 = \"stop_gained\"\n",
    "\n",
    "pattern42 = \"stop_gained&inframe_insertion&splice_region_variant\"\n",
    "replace42 = \"stop_gained\"\n",
    "\n",
    "pattern43 = \"stop_gained&NMD_transcript_variant\"\n",
    "replace43 = \"stop_gained\"\n",
    "\n",
    "pattern44 = \"stop_gained&protein_altering_variant\"\n",
    "replace44 = \"stop_gained\"\n",
    "\n",
    "pattern45 = \"stop_gained&protein_altering_variant&NMD_transcript_variant\"\n",
    "replace45 = \"stop_gained\"\n",
    "\n",
    "pattern46 = \"stop_gained&protein_altering_variant&splice_region_variant\"\n",
    "replace46 = \"stop_gained\"\n",
    "\n",
    "pattern47 = \"stop_gained&splice_region_variant\"\n",
    "replace47 = \"stop_gained\"\n",
    "\n",
    "pattern48 = \"stop_lost&3_prime_UTR_variant\"\n",
    "replace48 = \"stop_lost\"\n",
    "\n",
    "pattern49 = \"synonymous_variant&NMD_transcript_variant\"\n",
    "replace49 = \"synonymous_variant\"\n",
    "\n",
    "pattern50 = \"splice_region_variant&synonymous_variant\"\n",
    "replace50 = \"synonymous_variant\"\n",
    "\n",
    "pattern51 = \"missense_variant&splice_region_variant\"\n",
    "replace51 = \"missense_variant\"\n",
    "\n",
    "pattern52 = \"stop_lost&NMD_transcript_variant\"\n",
    "replace52 = \"stop_lost\"\n",
    "\n",
    "pattern53 = \"stop_gained&inframe_insertion\"\n",
    "replace53 = \"stop_gained\"\n",
    "\n",
    "pattern53 = \"stop_gained&inframe_insertion\"\n",
    "replace53 = \"stop_gained\"\n",
    "\n",
    "pattern54 = \"stop_gained&NMD_transcript_variant\"\n",
    "replace54 = \"stop_gained\"\n",
    "\n",
    "pattern55 = \"stop_gained&splice_region_variant\"\n",
    "replace55 = \"stop_gained\"\n",
    "\n",
    "\n",
    "\n",
    "#i = 1\n",
    "#data.fil[] <- lapply(data.fil[], gsub, pattern = get(paste(\"pattern\",i,sep = \"\")), replacement = replace49, fixed = TRUE)\n",
    "\n",
    "\n",
    "for (i in 1:55){\n",
    "    data.fil[] <- lapply(data.fil[], gsub, pattern = get(paste(\"pattern\",i, sep='')), replacement = get(paste(\"replace\",i, sep='')), fixed = TRUE)\n",
    "}\n",
    "#data.fil[] <- lapply(data.fil[], gsub, pattern = pattern1, replacement = replace1, fixed = TRUE)\n",
    "#data.fil[] <- lapply(data.fil[], gsub, pattern = pattern49, replacement = replace49, fixed = TRUE)\n",
    "\n",
    "\n",
    "head(data.fil)\n",
    "length(table(data.fil$Consequence))\n",
    "table(data.fil$Consequence)\n",
    "\n",
    "coding_term = c('frameshift_variant', 'inframe_deletion', 'inframe_insertion', \n",
    "                'synonymous_variant', 'missense_variant', \n",
    "                'splice_acceptor_variant', 'splice_donor_variant', \n",
    "                'start_lost', \n",
    "                'stop_gained', 'stop_lost', 'stop_retained_variant', \"protein_altering_variant\")\n",
    "dim(data.fil)\n",
    "#data.fil = data.fil[data.fil$Consequence %in% coding_term, ]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0ddb219d",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
